# Aligning sequence data

## What is an alignment ?

We want to compare individuals, species, whatever. To do this we need to compare what is comparable. Alignment to the rescue.

-   We want to find similarities in 2 sequences and put these similarities together, so we can compare sequences.

-   When we align 2 sequences together we talk about **pairwise** alignment as opposed to **multiple** alignment where we align more than 2 sequences all together. We will first focus on pairwise alignment.

2 ways of going about for pairwise alignment:

-   **Global alignment**: we look at the entirety of the 2 sequences and take all that information into account when aligning

-   **Local alignment**: we look at the sequences portion by portion, trying to find the best place where they match up.

This is very similar to another well known problem in computer science: the string-edit problem (**REFS?**)

### Why align ?

-   We do global alignment when we expect 2 sequences to be related, for example if we sequence the same genome twice we can globally align the two resulting genomes, if we find positions where they differ then that might be a sequencing or assembly error.

-   Similarly if we sequence two related organisms, like a human and a chimp we can see which regions of the genome are identical, which are different and learn what makes a human a human or a chimp a chimp.

-   For local alignment there we can align a read to a reference sequence in order to correct sequencing errors or find out where the read comes from.

### How to align ?

The seminal method for pairwise alignment was the Needleman-WÃ¼nsch algorithm [@needlemanGeneralMethodApplicable1970] based on a dynamic programming method. A decade later, the Smith-Waterman algorithm [@smithIdentificationCommonMolecular1981] was developed with similar ideas to perform local alignment. Both are still used today for pairwise alignment.

Dynamic programming is a method to solve complex problems by breaking it
into smaller sub-problems and solving each one optimally and separately [@bradleyAppliedMathematicalProgramming1977], it is widely used when we wish to have a precise alignment between 2 sequences.

#### Global alignment

-   short presentation of NW algo:

    -   Fill out DP matrix

    -   Traceback to find optimal alignment

-   Example...

-   This algorithm although very precise, has a time complexity of $O(nm)$ where $n$ and $m$ are the lengths of the sequences to align [@sungAlgorithmsBioinformaticsPractical2011]. Some methods have been proposed to speed up [@masekFasterAlgorithmComputing1980], however the complexity is still $O(nm/\log(n))$. Lower bounds have been studied and there is not much optimization we can do if we want to keep an optimal exact alignment [@vinhInformationTheoreticMeasures2010; @ullmanBoundsComplexityLongest1976]. If we want to do better we have to rely on heuristics.

-   Another issue is space complexity since we need to store the matrix, we get $O(nm)$ as well for space, so if we wish to align 2 human genomes we would need to store $\approx 10^{19}$ matrix cells, which would amount to 10 Exabytes of storage if we use 8bit integers (meaning it would take an entire data center to store that).

    -   However in practice we can do much better than that, and construct an optimal alignment in linear space complexity $O(n+m)$ [@hirschbergLinearSpaceAlgorithm1975] meaning we would only need a couple gigabytes to store the matrix for 2 human genomes.

-   Use in software: ...

#### Local alignment 

-   Similar ideas to NW

-   Basic example of NW

-   In terms of complexity is the same as NW in quadratic in time and space [@sungAlgorithmsBioinformaticsPractical2011] but can be taken down to linear space with the same approach as NW.

-   Used in software: ...

### Substitution models

mismatch depends on models, BLOSUM, PAML, ...

#### Dealing with gaps

-   biologically longer deletions are more likely than plenty of shorter ones

-   Affine gap penalties

## How do we speed up alignment ?

### Heuristic algorithms 

-   Better local alignment

-   Better global alignment

### Indexes + heuristics

Pre-computing indexes -\> when we know we will have lots of queries:

-   BLAST

-   FASTA

## MSA (short)

When we need to compare a lot of individuals together we can do MSA.

NP-hard problem so we need heuristics or tricks

Even if we align all sequences pairwise we need to then combine all gaps and stuff -\> complicated.

#### Progressive

guide tree, clustering of sequences then refine alignment. Good heuristic but with larger datasets, becomes harder.

#### HMMs / profile alignments / pairwise

Example of COVID where homology is high so we can get away with using HMMS / pairwise to root sequence. *(point to appendix with covid align ?)*

#### Other optimization methods (short)

-   Simulated annealing

-   Genetic algorithm

## Mapping

-   short intro
-   context of short read on large reference sequence
-   Build an index:
    -   FM-index (bowtie2, bwa-mem?)

    -   minimizers (minimap2)
-   Problems:
    -   Repetitive regions (centromeres, telomeres)

    -   low homology / sequencing errors...
-   specific heuristics like minimizers / FM-index help *(Needed here since I introduce it earlier ? )*

\printbibliography[segment=\therefsegment,heading=subbibintoc,title={References for chapter \thechapter}]
