# Aligning sequence data

## What is an alignment ?

We want to compare individuals, species, whatever. To do this we need to compare what is comparable. Alignment to the rescue.

-   We want to find similarities in 2 sequences and put these similarities together, so we can compare sequences.

-   When we align 2 sequences together we talk about **pairwise** alignment as opposed to **multiple** alignment where we align more than 2 sequences all together. We will first focus on pairwise alignment.

2 ways of going about for pairwise alignment [@sungAlgorithmsBioinformaticsPractical2011]:

-   **Global alignment**: we look at the entirety of the 2 sequences and take all that information into account when aligning

-   **Local alignment**: we look at the sequences portion by portion, trying to find the best place where they match up.

### Why align ?

-   hamming distance is an easy method [@hammingCodingInformationTheory1980]

    -   Although it is easy to compute -\> sequences must have same length

    -   When dealing with DNA/proteins we have to be able to deal with insertions/deletions and hamming cannot do that...

-   This is very similar to other well known problems in computer science: the string-edit problem [@gusfieldAlgorithmsStringsTrees1997] and the Levenshtein distance [@levenshteinBinaryCodesCapable1966]

-   Alignments are used in many cases so that:

    -   we can examine similarities/differences between them (i.e. comparative genomics ) [@hardisonComparativeGenomics2003]

    -   we can infer (usually with multiple alignment):

        -   evolutionary relationships (phylogenetics), and most methods to reconstruct phylogenetic trees take as input a multiple alignment: [@felsensteinEvolutionaryTreesDNA1981; @kumarMEGAMolecularEvolutionary1994; @kozlovRAxMLNGFastScalable2019; @guindonNewAlgorithmsMethods2010; @priceFastTreeApproximatelyMaximumLikelihood2010]

        -   protein:

            -   structure [@jumperHighlyAccurateProtein2021; @karplusPredictingProteinStructure1999 ]

            -   function [@watsonPredictingProteinFunction2005; @leePredictingProteinFunction2007]

    -   we can correct sequencing errors [@hacklProovreadLargescaleHighaccuracy2014; @korenHybridErrorCorrection2012; @salmelaCorrectingErrorsShort2011].

    -   Structural variant detection [@medvedevComputationalMethodsDiscovering2009; @mahmoudStructuralVariantCalling2019]

### How to align two sequences ?

The seminal method for pairwise alignment was the Needleman-WÃ¼nsch algorithm [@needlemanGeneralMethodApplicable1970] based on a dynamic programming method. A decade later, the Smith-Waterman algorithm [@smithIdentificationCommonMolecular1981] was developed with similar ideas to perform local alignment. Both are still used today for pairwise alignment.

Dynamic programming is a method to solve complex problems by breaking it into smaller sub-problems and solving each one optimally and separately [@bradleyAppliedMathematicalProgramming1977; @bellmanTheoryDynamicProgramming1954], it is widely used when we wish to have a precise alignment between 2 sequences.

#### Global alignment

-   We do global alignment when we expect two sequences to be related and of similar genes/proteins, etc...
-   short presentation of NW algo (c.f. Figure \@ref(fig:nwAlign)):
    -   The score of an alignment can be defined recursively -\> dynamic programming

    -   Fill out DP matrix:

    -   Traceback to find optimal alignment: start in the bottom right and follow the arrows up to the upper left. Diagonal: Match/mismatch, vertical arrow: gap in "columns" sequence, horizontal arrow: gap in "rows" sequence.

```{r, nwCaption}
nwCaption <- "**Example global alignment with the Needleman-Wunsch algorithm**\ The match score is 1, the mismatch and indel scores are both set to -1. This results in 3 equally optimal global alignments."
```

```{r, nwAlign, label="nwAlign", fig.cap=nwCaption, eval=knitr::is_html_output(), out.width="60%"}
knitr::include_graphics("./figures/Align-Intro/Nw-dp.png")
```

```{=tex}
\begin{figure}[h]
\centering
\includegraphics[width=0.6\linewidth]{./figures/Align-Intro/Nw-dp.pdf}
\extcaption{Example global alignment with the Needleman-Wunsch algorithm}{The match score is 1, the mismatch and indel scores are both set to -1. This results in 3 equally optimal global alignments.}
\label{fig:nwAlign}
\end{figure}
```
-   This algorithm although very precise, has a time complexity of $O(nm)$ where $n$ and $m$ are the lengths of the sequences to align [@sungAlgorithmsBioinformaticsPractical2011]. Some methods have been proposed to speed up [@masekFasterAlgorithmComputing1980], however the complexity is still $O(nm/\log(n))$. Lower bounds have been studied and there is not much optimization we can do if we want to keep an optimal exact alignment [@vinhInformationTheoreticMeasures2010; @ullmanBoundsComplexityLongest1976]. If we want to do better we have to rely on heuristics.
-   Another issue is space complexity since we need to store the matrix, we get $O(nm)$ as well for space, so if we wish to align 2 human genomes we would need to store $\approx 10^{19}$ matrix cells, which would amount to 10 Exabytes of storage if we use 8bit integers (meaning it would take an entire data center to store that).
    -   However in practice we can do much better than that, and construct an optimal alignment in linear space complexity $O(n+m)$ [@hirschbergLinearSpaceAlgorithm1975] meaning we would only need a couple gigabytes to store the matrix for 2 human genomes.

    -   Above idea resulting in Myers-miller algorithm [@myersOptimalAlignmentsLinear1988], implemented in the EMBOSS stretcher tool [@riceEMBOSSEuropeanMolecular2000]

#### Local alignment

-   We want to align a subsequence to a subsequence.

-   Similar ideas to NW, except we start the traceback at the maximally valued cell and stop when we hit a score of 0. i.e.

-   In the example in Figure \@ref(fig:nwAlign) there are two possible starting cells with a score of 2. This will yield 2 optimal local alignments: AC to AC and GA to GA.

-   In terms of complexity is the same as NW in quadratic in time and space [@sungAlgorithmsBioinformaticsPractical2011] but can be taken down to linear space with the same approach as NW.

-   Optimizations were made and resulted in the Huang and Miller algorithm [@huangTimeefficientLinearspaceLocal1991] which is implemented in the EMBOSS Lalign tool [@riceEMBOSSEuropeanMolecular2000], and the Waterman Eggert algorithm [@watermanNewAlgorithmBest1987].

Both methods are implemented in many different software tools and are used when performing pairwise alignments of short sequences [@stajichBioperlToolkitPerl2002; @gentlemanBioconductorOpenSoftware2004; @riceEMBOSSEuropeanMolecular2000] with version implemented for specific CPU instruction sets [@dailyParasailSIMDLibrary2016] or GPUs [@frohmbergGPASImprovedVersion2012] to speed up alignment.

### Substitution models / scoring

When scoring an alignment we can use different scoring models/substitution matrices [@altschulSubstitutionMatrices2013]:

-   A lot of work has been done on protein scoring matrices

    -   Log-Odds models, based on the fact that mutations are not equiprobable, and some mutations will be much more common (I $\leftrightarrow$ L in proteins)

    -   PAM (Point Accepted Mutations) [@dayhoffModelEvolutionaryChange1978], gotten from studying closely related protein sequences, estimate the probability of one amino acid changing to another one over time. With more data refinements have been made for PAM-like matrices [@mullerModelingAminoAcid2000].

    -   BLOSUM [@henikoffAminoAcidSubstitution1992], similar idea to PAM except they were constructed on several "blocks". A block is a segment of a protein that is very well conserved within a family and computed the probabilities on these blocks.

    -   Some matrices estimated with ML instad of log-odds methods [@whelanGeneralEmpiricalModel2001; @leImprovedGeneralAmino2008]

    -   Model specific matrices:

        -   Transmembrane matrices [@mullerNonsymmetricScoreMatrices2001; @ngPHATTransmembranespecificSubstitution2000]

        -   Disordered regions in proteins: [@trivediAminoAcidSubstitution2019]

        -   context-specific matrices [@goonesekereContextspecificAminoAcid2008]

        -   Specific organisms like *P. falciparum* with pfSSM [@pailaGenomeBiasInfluences2008] or HIV [@nickleHIVSpecificProbabilisticModels2007]

        -   specific to global alignment [@sardiuScoreStatisticsGlobal2005]

-   Some on DNA alignment

    -   You can derive a matrix with methods similar to PAM [@chiaromonteScoringPairwiseGenomic2001]
    -   You can do codon substitution matrices [@schneiderEmpiricalCodonSubstitution2005], or combine codon matrices with AA matrices [@doron-faigenboimCombinedEmpiricalMechanistic2007]

### Dealing with gaps

-   biologically longer deletions are more likely than plenty of shorter ones

    -   Here a short example of 2 alignments with different gap strategies

-   Semi global alignment (i.e. gaps on the ends of the sequence are free...)

-   Affine gap penalties, proposed by Gotoh [@gotohImprovedAlgorithmMatching1982]

-   Non affine gap penalties [@bennerEmpiricalStructuralModels1993; @guSizeDistributionInsertions1995; @watermanEfficientSequenceAlignment1984]

## How do we speed up pairwise alignment ?

Although NW and SW give us optimal alignments [@pearson27DynamicProgramming1992] when dealing with large sequences they become impractical due to time and space complexity. We need heuristics, review [@chaoDevelopmentsAlgorithmsSequence2022], .

### Change the method

-   Bounded DP [@spougeSpeedingDynamicProgramming1989a; @fickettFastOptimalAlignment1984], we can make an assumption about the relationship between the sequences, there will probably not be many gaps, therefore the scores we will used will be concentrated around the diagonal of the DP matrix c.f. \@ref(fig:optimMethods)

-   New Algorithms like the wavefront algorithm [@marco-solaFastGapaffinePairwise2020]

-   HMMs:

    -   PairHMMs can be used to align 2 sequences, in some cases it is mathematically equivalent to NW [@durbinBiologicalSequenceAnalysis1998].

    -   Software: HHsearch [@sodingProteinHomologyDetection2005], HMMer [@finnHMMERWebServer2011], MCALIGN2 [@wangMCALIGN2FasterAccurate2006] is used to efficiently search for alignments in large databases

-   You can also align with Fast Fourier Transforms (FFT) as in MAFFT [@katohMAFFTNovelMethod2002] by quickly computing correlations with FFT to find homologous regions and then using these to align.

### Seed and extend with data structures

-   General idea:

    -   Seed + extend for local alignment

    -   Find anchors for global alignment

    -   In both cases: divide and conquer approach: as you can restrict the DP matrix by blocks defined by the anchors/seeds c.f. \@ref(fig:optimMethods)

    -   Many ways of finding seeds [@sunChoosingBestHeuristic2006] and to index them [@liSurveySequenceAlignment2010]

    -   2 use cases:

        -   global alignment: align 2 large sequences that would overload computing resources.

        -   local alignment:

            -   to a database: ie. find hits in many sequences.

            -   to a reference: ie find best hit in one sequence.

```{r}
optimCaption <- "**2 heuristic methods to speed up alignment:**\
divide and conquer and bounded dynamic programming. Adapted from [@chaoDevelopmentsAlgorithmsSequence2022] **(Original figure here as a placeholder, I will adapt it)**"
```

```{r, optimMethods, label="optimMethods", fig.cap=optimCaption, out.width="80%", eval=knitr::is_html_output()}
knitr::include_graphics("./figures/Align-Intro/optim-to-adapt.png")
```

```{=tex}
\begin{figure}[h]
\centering
\includegraphics[width=\linewidth]{./figures/Align-Intro/optim-to-adapt.png}
\extcaption{2 heuristic methods to speed up alignment:}{divide and conquer and bounded dynamic programming. Adapted from \autocite{chaoDevelopmentsAlgorithmsSequence2022} \textbf{(Original figure here as a placeholder, I will adapt it)}
}
\label{fig:optimMethods}
\end{figure}
```
-   Building indices/databases (useful when you want to try aligning a query sequence to a bunch of possible targets, i.e. in order to search for homology, this does approximate local alignment):

#### Hash tables

-   BLAST [@altschulBasicLocalAlignment1990]

    -   We break up our sequences into overlapping "words" i.e. all possible short sub-sequences.

    -   Construct a database of words with positions in the reference sequence(s)

    -   Get list of words for your query sequence and generate possible "hits", i.e. words that align with your query words with a score higher than a threshold.

    -   Scan the database for these generated words, if you find one it is a hit and these 2 words are a candidate position for a local alignment.

    -   extend the local alignment with SW in both directions from the hit to get a local alignment, stop extending if the alignment score gets too low.

    -   Plenty of variants: BLASTP for proteins, BLASTN or MEGABLAST for DNA, BLASTX for comparing DNA query to a protein database, PSI-BLAST that iteratively refines the alignment by building multiple alignments from the statistically significant alignments, learns a specific scoring matrix from this and starts over, other people have refined the heuristic for even faster seed prediction as in UBLAST [@edgarSearchClusteringOrders2010].

    -   It is important to note that it is a heuristic method, it has a decreased sensitivity to the SW algorithm which is optimal [@shpaerSensitivitySelectivityProtein1996], it is however much faster and can allow users to search for similarities among millions of sequences at once

    -   Particularly useful, one of the most cited papers in the world: quickly search for homologs of unknown sequences, ... available as a web service hosted by NCBI (<https://blast.ncbi.nlm.nih.gov/Blast.cgi>).

-   FASTA [@pearsonImprovedToolsBiological1988], derived from FASTP [@lipmanRapidSensitiveProtein1985] which could not deal with gaps, preceded BLAST.

    -   Similarly to BLAST, break query up into overlapping words

    -   Scan the library for exact-matching words to build up regions with high similarity (save 10 best regions)

    -   Score regions with a substitution matrix

    -   High scoring regions are combined to build an approximate alignment

    -   Highest scoring ungapped alignment is realigned with banded SW.

-   Both methods are very Fast, taking only a couple seconds to find approximate local alignments for 100 sequences [@saripellaBenchmarkingNextGeneration2016] in a database of over 80 million sequences [@finnPfamProteinFamilies2016], with HMMs being quite slow. It is much faster to use these than to try to do the same task with SW or NW [@essoussiComparisonFourPairwise2007].

-   Both BLAST and FASTA compute a statistical p-value for a pairwise alignment and an expected number of random hits when scanning the database (E-value) [@karlinMethodsAssessingStatistical1990a; @mottAlignmentStatisticalSignificance2005]

-   Minimap2 [@liMinimap2PairwiseAlignment2018] minimizer (Figure \@ref(fig:minimizers)) hash table

-   Diamond [@buchfinkFastSensitiveProtein2015; @buchfinkSensitiveProteinAlignments2021] which indexes both the reference and query at the same time.

```{r, minimizercaption}
progCaption <- "**Example of kmers sharing a minimizer**"
```

```{r, minimizers, label="minimizers", eval=knitr::is_html_output(), fig.cap=progCaption, out.width="70%"}
knitr::include_graphics("./figures/Align-Intro/minimizers.png")
```

```{=tex}
\begin{figure}[h]
\centering
\includegraphics[width=0.7\textwidth]{./figures/Align-Intro/minimizers.pdf}
\extcaption{Example of kmers sharing a minimizer}{}
\label{fig:minimizers}
\end{figure}
```
#### Suffix trees / suffix arrays

-   Used in many pattern matching problems [@gusfieldAlgorithmsStringsTrees1997].

-   Software: AVID uses suffix trees to find anchors [@brayAVIDGlobalAlignment2003], MUMmer finds largest identical subsequences between 2 sequences to anchor alignments using suffix trees [@delcherFastAlgorithmsLargescale2002]

#### FM Index

-   FM index [@ferraginaOpportunisticDataStructures2000]:

-   In some cases when using suffix trees take too much memory we can use an FM index which is based on the burrows wheeler transform [@burrowsBlockSortingLosslessData1994].

-   used for in exact and approximate string matching [@sungAlgorithmsBioinformaticsPractical2011]

-   Software: BWT-SW uses an FM-index to speed up local alignment [@lamCompressedIndexingLocal2008], Bowtie 2 uses an FM index to find seeds [@langmeadFastGappedreadAlignment2012], BWA and BWA-SW use a similar idea [@liFastAccurateShort2009; @liFastAccurateLongread2010], BWA-MEM [@liAligningSequenceReads2013] and CUSHAW [@liuLongReadAlignment2012] also uses FM indices to find exact matches to seed a local alignment .

## Multiple sequence alignment

When we need to compare a lot of individuals together we can do MSA, essential task in many bioinformatics analyses [@russellMultipleSequenceAlignment2014].

NP-hard [@wangComplexityMultipleSequence1994; @justComputationalComplexityMultiple2001] problem if you do it with DP so we need heuristics or tricks

Even if we align all sequences pairwise we need to then combine all gaps and stuff -\> complicated.

### Progressive alignment

guide tree, clustering of sequences then refine alignment. Good heuristic but with larger datasets, becomes harder. [@fengProgressiveSequenceAlignment1987] Main MSA method.

1.  Compute pairwise distance matrix for sequence set:
    -   either by doing N(N-1)/2 pairwise alignments

    -   Or alignment free methods to speed things up [@jonesRapidGenerationMutation1992; @blaisdellMeasureSimilaritySets1986]
2.  Build guide tree from distances (neighbor joining, UPGMA, ...)
3.  Align sequences one by one according to the tree, from the leafs (i.e sequences) to the root (full MSA).

```{r, progcaption}
progCaption <- "**Overview of the progressive alignment process.**\
A) sequences to align B) guide tree constructed from distances C) Alignment steps along the guide tree and resulting MSA. Adapted from [@sungAlgorithmsBioinformaticsPractical2011]"
```

```{r, progAlign, label="progAlign", eval=knitr::is_html_output(), fig.cap=progCaption, out.width="70%"}
knitr::include_graphics("./figures/Align-Intro/progressive-aln.png")
```

```{=tex}
\begin{figure}[h]
\centering
\includegraphics[width=0.7\textwidth]{./figures/Align-Intro/progressive-aln.pdf}
\extcaption{Overview of the progressive alignment process.}{A) sequences to align B) guide tree constructed from distances C) Alignment steps along the guide tree and resulting MSA. Adapted from \autocite{sungAlgorithmsBioinformaticsPractical2011}}
\label{fig:progAlign}
\end{figure}
```
Problems -\> keeps gaps and if bad alignment at first steps error propagates ("once a gap, always a gap" [@fengProgressiveSequenceAlignment1987]).

In order to curtail this problem we have iterative refinement [@russellMultipleSequenceAlignment2014]:

1.  create MSA (e.g. progressive)
2.  divide MSA into 2 groups + remove columns with only gaps
3.  realign with profile alignment
4.  Redo 2+3 until no improvement is made, according to some scoring function (Weighted sum-of-pair [@altschulWeightsDataRelated1989], or others like log-odds, correlation [@edgarComparisonScoringFunctions2004] or consistency [@notredameCOFFEEObjectiveFunction1998])

Some of the most used MSA software uses these methods of progressive/iterative refinement:

-   CLUSTAL-W [@thompsonCLUSTALImprovingSensitivity1994] and CLUSTAL-X [@thompsonCLUSTALWindowsInterface1997]

-   T-Coffee [@notredameTcoffeeNovelMethod2000]

-   MUSCLE [@edgarMUSCLEMultipleSequence2004a; @edgarMUSCLEMultipleSequence2004]

-   MAFFT [@katohMAFFTNovelMethod2002]

-   ProbCons [@doProbConsProbabilisticConsistencybased2005]

Different strenghts/weaknesses so many reviews and benchmarks to make a choice [@notredameRecentEvolutionsMultiple2007; @notredameRecentProgressMultiple2002; @edgarMultipleSequenceAlignment2006; @paisAssessingEfficiencyMultiple2014; @thompsonBAliBASEBenchmarkAlignment1999]

### HMMs in profile-profile or seq-profile alignments

HMM method similar performance to clustal-w [@eddyMultipleAlignmentUsing], HHMer [@finnHMMERWebServer2011], MSAProbs [@liuMSAProbsMultipleSequence2010] and COVID-align [@lemoineCOVIDAlignAccurateOnline2020]

Example of COVID where homology is high so we can get away with aligning pairwise to ancestral-sequence NextClade/NextAlign [@aksamentovNextcladeCladeAssignment2021]

### Other methods (short)

-   Simulated annealing to speed up DP [@kimMultipleSequenceAlignment1994; @ishikawaMultipleSequenceAlignment1993; @huoSimulatedAnnealingAlgorithm2007]

-   Genetic algorithm, review [@chowdhuryReviewMultipleSequence2017], can speed things up [@zhangGeneticAlgorithmMultiple1997]

    -   SAGA [@notredameSAGASequenceAlignment1996], VGDA [@nazninVerticalDecompositionGenetic2011], GAPAM [@nazninProgressiveAlignmentMethod2012]

-   recently: regressive method [@garrigaLargeMultipleSequence2019] root to leaf allows to align a very large number of sequences: 1.4 million!

## The specificities of read-mapping {#the-specificities-of-read-mapping}

Huge review of mapping in [@alserTechnologyDictatesAlgorithms2021]

### What is read-mapping ?

Task: locally/semi-globally align independent sequencing reads to a reference genome.

Plenty of aligners both for short reads (reviews [@schbathMappingReadsGenomic2012; @hatemBenchmarkingShortSequence2013; @canzarShortReadMapping2017]) and long-reads ([@liMinimap2PairwiseAlignment2018]. Active field and there are benchmarks [@brindaRNFGeneralFramework2016] and reviews [@alserTechnologyDictatesAlgorithms2021; @ruffaloComparativeAnalysisAlgorithms2011].

-   Mapping quality:
    -   Intro and definition, quite a loosely defined term.

        -   Different definitions/implementations ?

    -   Mapping quality from tandem simulation [@langmeadTandemSimulationFramework2017]
-   Technical aspects to speed up, since we usually have a couple refs and and many many query reads, the index strategy described earlier is particularly well suited:
    -   hash tables: [@liMinimap2PairwiseAlignment2018]

    -   FM index / BWT: BWT-SW [@lamCompressedIndexingLocal2008], Bowtie 2 [@langmeadFastGappedreadAlignment2012], BWA, BWA-SW [@liFastAccurateShort2009; @liFastAccurateLongread2010], BWA-MEM [@liAligningSequenceReads2013] and CUSHAW [@liuLongReadAlignment2012].
-   Divide and conquer as well: [@linKartDivideandconquerAlgorithm2017]
-   These heuristics also become more and more needed as sequencing technologies yield longer and longer reads.
-   Also hardware acceleration for short [@olsonHardwareAccelerationShort2012] and long [@chenAcceleratingNextGeneration2014; @suzukiIntroducingDifferenceRecurrence2018; @zeniLOGANHighPerformanceGPUBased2020] reads.

### Challenges of read-mapping

-   low homology / sequencing errors... make mapping and other tasks hard [@gusfieldAlgorithmsStringsTrees1997]
    -   Particularly with long reads since sequencing error rate is higher: specific long read mappers that take this into account:

        -   PB: BLASR [@chaissonMappingSingleMolecule2012], lordFAST [@haghshenasLordFASTSensitiveFast2019]

        -   ONT: GraphMap [@sovicFastSensitiveMapping2016]

        -   Both: NGMLR [@sedlazeckAccurateDetectionComplex2018], MashMap [@jainFastApproximateAlgorithm2018], DuploMap [@prodanovSensitiveAlignmentUsing2020]
-   Repetitive regions (centromeres, telomeres) make it hard to map [@alserTechnologyDictatesAlgorithms2021]. Specific tools have been developed for this:
    -   winnowmap [@jainWeightedMinimizerSampling2020], winnowmap2 [@jainLongreadMappingRepetitive2022]

    -   tandemtools [@mikheenkoTandemToolsMappingLong2020] for tandem repeats.
-   homopolymers can also be a problem, so HPC is implemented in mappers *(c.f.* \@ref(hpc-trick)*)*

\printbibliography[segment=\therefsegment,heading=subbibintoc,title={References for chapter \thechapter}]
