# Aligning sequence data

## What is an alignment ?

We want to compare individuals, species, whatever. To do this we need to compare what is comparable. Alignment to the rescue.

Local vs. global alignment

### How to align ?

Usually done with dynamic programming by assigning a score to mismatches, indels, ... and then choosing the alignment with minimal score.

short example of local alignment

gap open, gap extend, etc...

### Substitution models

mismatch depends on models, BLOSUM, PAML, ...

### Why align ?

-   pairwise alignment to compare two similar sequences

-   mapping to get an idea of where a short sequence comes from on a larger reference. (sequencing reads)

## How do we speed up local alignment ?

heuristic methods, based on kmers

-   BLAST

-   FM-index (bowtie2, bwa-mem?)

-   minimizers (minimap2)

## MSA (short)

When we need to compare a lot of individuals together we can do MSA.

NP-hard problem so we need heuristics or tricks

Even if we align all sequences pairwise we need to then combine all gaps and stuff -\> complicated.

#### Progressive

guide tree, clustering of sequences then refine alignment. Good heuristic but with larger datasets, becomes harder.

#### HMMs / profile alignments / pairwise

Example of COVID where homology is high so we can get away with using HMMS / pairwise to root sequence. *(point to appendix with covid align ?)*

#### Other optimization methods (short)

-   Simulated annealing

-   Genetic algorithm

## Mapping

-   short intro
-   context of short read on large reference sequence
-   Problems:
    -   Repetitive regions (centromeres, telomeres)

    -   low homology / sequencing errors...
-   specific heuristics like minimizers / FM-index help *(Needed here since I introduce it earlier ? )*

\printbibliography[segment=\therefsegment,heading=subbibintoc,title={References for chapter \thechapter}]
