# Viruses, HIV and drug resistance

## What are viruses ?

Viruses occupy a strange place in the tree of life, with many debating if they are actually alive or not. André Lwoff gave what is probably the most fitting definition: *"viruses are viruses"* [@lwoffConceptVirus1957]. Despite this ambiguity, viruses share some common characteristics which allow us to define them as intracellular parasites [@minorViruses2014]:

1.  Viruses have some type of genetic information, contained in either DNA or RNA
2.  This genetic information is protected by some form of envelope
3.  They use the cellular machinery of host cells to make copies of themselves.

While we all know that viruses can be pathogenic and dangerous (the recent example of SARS-CoV2 springs to mind), that is not necessarily the case and some viruses like GBV-C [@stapletonGBVirusesReview2011] and certain strains of H5N1 *Influenza* [@yamamotoCharacterizationNonpathogenicH5N12011] are non pathogenic and essentially harmless.

Viruses have been discovered for all three domains of life: Eukaryota, Bacteria and Archea. In Eukaryota many viruses have been discovered for animals (both vertebrate [@shiEvolutionaryHistoryVertebrate2018] and invertebrate [@adamsAtlasInvertebrateViruses2017]), plants [@lefeuvreEvolutionEcologyPlant2019], protozoa [@wangVirusesParasiticProtozoa1991], chromista [@ferminVirusesProkaryotesProtozoa2018] and even fungi [@sutelaVirusesFungiOomycetes2019]. Bacterial viruses known as phages have been known to exists since the beginning of the 20th century [@twortINVESTIGATIONNATUREULTRAMICROSCOPIC1915; @delbrockBacterialVirusesBacteriophages1946]. These bacteriophages are being considered as a therapeutic alternative to antibiotics [@clarkBacterialVirusesHuman2004; @vankan-davelaarUsingVirusesNanomedicines2014] which could help with multi-drug-resistant bacterial pathogens. Archea are also known to have their own viral infections [@prangishviliVirusesArchaea2016; @prangishviliVirusesArchaeaUnifying2006].

Strangely even viruses of viruses seem to exist such as the plant satellite virus [@franckiPlantVirusSatellites; @xuPlantVirusSatellites2011] or hepatitis delta virus [@laiMolecularBiologyHepatitis1995; @hughesHepatitisDeltaVirus2011]. These "viroids" do not infect viral hosts *per se* but they cannot replicate on their own. Replication must happen during co-infection with a larger virus.

There is a huge diversity of viruses affecting all types of life, and new viruses are being discovered all the time [@edgarPetabasescaleSequenceAlignment2022]. This diversity hints at a rich and long evolutionary history. When and where viruses originated is still under study [@nasirInvestigatingConceptOrigin2020; @forterreOriginViruses2009] and we might never know how they emerged, it is however believed that they may have played an important role in the emergence of eukaryotic cells [@forterreOriginVirusesTheir2006]. This co-evolution between virus and host cell shows a strong link between the two organisms and some parts of the human genome are likely of ancient viral origin [@boekeRetrotransposonsEndogenousRetroviruses1997; @kojimaViruslikeInsertionsSequence2021]. It has been estimated that 1% to 8% of the human genome are endogenous retroviral sequences [@lowerVirusesAllUs1996; @griffithsEndogenousRetrovirusesHuman2001].

The rich diversity of viruses is reflected in the variety of genetic information support, replication strategy, physical and genomic size, as well as shape. The differences in genetic information support and replication strategy form the basis of the Baltimore virus classification system [@baltimoreExpressionAnimalVirus1971] , still used today [@kooninBaltimoreClassificationViruses2021] to classify virus lineages.

As stated above all viruses have some genetic information, this information is stored either as DNA or as RNA, which is the molecule of choice for 70% of human pathogenic viruses [@domingoRNAVirusGenomes2018] *(HIV and SARS-CoV 2 are RNA viruses)*.

For DNA viruses, the molecule can be double-stranded as for *Herpesvirus* [@mcgeochTopicsHerpesvirusGenomics2006; @boehmerHerpesVirusReplication2003], single-stranded like in the case of *Papillomavirus* [@brentjensHumanPapillomavirusReview2002] or even circular in the case of the Hepatitis B virus [@kayHepatitisVirusGenetic2007]. This molecular diversity is also present in RNA viruses where the RNA molecule can be double-stranded like for *Rotavirus* [@parasharRotavirus1998], or single-stranded. Furthermore, for single-stranded RNA viruses the strand can either be positive *(i.e. can be directly translated into a protein)* like Hepatitis C virus [@simmondsVariabilityHepatitisVirus1995] or *Poliovirus* [@wimmerGeneticsPoliovirus1993; @racanielloOneHundredYears2006]; conversely there are negative-strand RNA viruses, for which the complementary strand of RNA must be synthesized before translation into a protein, such as the Influenza or Measles viruses [@paleseNegativestrandRNAViruses1996].

This diversity in genetic information support implies a necessary diversity in replication strategy. The main replication strategies are as follows [@domingoVirusEvolution2014]:

-   The RNA molecule is directly copied as RNA. This is the strategy followed by single-stranded RNA coronaviruses [@vkovskiCoronavirusBiologyReplication2021], Dengue viruses [@backDengueVirusesOverview2013] or Hepatitis C virus [@dustinHepatitisVirusLife2016].

-   The DNA molecule is directly replicated as DNA. this can happen for both single-stranded DNA viruses like *Papillomavirus* [@kadajaPapillomavirusDNAReplication2009]and double-stranded DNA viruses like Herpes simplex virus [@wellerHerpesSimplexViruses2012].

-   The DNA molecule is replicated by going through an RNA intermediary like Hepatitis B virus [@beckHepatitisVirusReplication2007].

-   The RNA molecule is replicated by going through a DNA intermediary. This strategy is used by retroviruses that integrate this viral DNA intermediary into the host DNA, like HIV-1 (see Section \@ref(life-cycle-of-hiv)).

Finally, the genetic diversity of viruses is reflected in their physical characteristics: viruses come in all shapes and sizes. Physical size range from 17nm for plant satellite viruses [@pyleChapter58Biology2017] to the giant 400nm *Mimivirus* [@raoult2megabaseGenomeSequence2004]. Genomic size is also quite variable, there is a stark contrast between the 860 bp *Circovirus SFBeef* and the 2.5 Mbp *Pandoravirus salinus* genomes [@campillo-balderasViralGenomeSize2015]. Viruses come in a variety of shapes [@cannVirusStructure2015]: icosahedral for HIV, helical for the tobacco mosaic virus or a distinctive head-tail shape for bacteriophages.

Although there are a large number of viruses, and many of them are of great importance for human health, we will now focus on one virus of particular importance: Human Immunodeficiency Virus otherwise known as HIV.

## Getting to know HIV

### Quick Presentation of HIV

HIV is a single-stranded RNA retrovirus that is responsible for the Acquired Immune Deficiency Syndrome (AIDS) pandemic that has been around for the last couple decades. This virus is transmitted through sexual contact or through blood. Sexual activity is the largest transmission factor followed by intravenous drug use [@hladikSettingStageHost2008; @shawHIVTransmission2012].

The HIV infects cells of the host immune system, specifically CD4 T-cell lymphocytes and destroys them due to it's replication process [@weissHowDoesHIV1993]. An HIV infection typically start with an asymptomatic phase that can last years, followed by a growth in viral replication leading to a decrease in CD4 cells which progresses into AIDS [@melhuishNaturalHistoryHIV2018]. During AIDS, when the CD4 cell count is low enough opportunistic diseases such as pneumonia or tuberculosis [@murrayPulmonaryComplicationsAcquired1984] can easily infect the host, leading to death when the immune system is weak enough.

The HIV/AIDS is one of the deadliest pandemics in history, estimated to have lead to the death of 36 million people [@sampathPandemicsThroughoutHistory2021]. In 2010 [@worldhealthorganizationGlobalReportUNAIDS2010] approximately 33 million people were infected with HIV, 2.6 million of which were due to new infections, and 1.8 million died of AIDS. Most of the new infections happened in economically developing regions of the world, 70% of them coming from sub-Saharan Africa [@worldhealthorganizationGlobalReportUNAIDS2010]. As of 2020, these numbers have decreased with "only" 1.5 million new infections and 680,000 AIDS deaths, which is encouraging from a public health perspective.

The HIV-1 virus was discovered simultaneously in 1983 by Françoise Barré-Sinoussi, Luc Montagnier [@barre-sinoussiIsolationTlymphotropicRetrovirus1983] and Robert Gallo [@galloIsolationHumanTcell1983]. There exists a second HIV-2 virus discovered shortly after HIV-1 [@clavelIsolationNewHuman1986], it is however less transmissible than HIV-1 which is largely responsible for the global HIV/AIDS pandemic [@gilbert2003]. In Africa in 2006, HIV-1 infections were rising where HIV-2 were declining [@vanderloeffSixteenYearsHIV2006].

While both viruses are of zoonotic origin, from transmissions of Simian Immunodeficiency Virus (SIV) from primates to humans. HIV-1 most likely originates from an SIV present in chimpanzees [@gaoOriginHIV1Chimpanzee1999; @hamelTwentyYearsProspective2007;@sharpOriginsHIVAIDS2011], and HIV-2 from an SIV present in Sooty mangabeys [@hirschAfricanPrimateLentivirus1989; @gaoHumanInfectionGenetically1992; @chenGeneticCharacterizationNew1996].

Several independent such transmissions have resulted in 4 lineages of HIV-1 labeled groups M, N, O and P [@hemelaarOriginDiversityHIV12012] (similarly HIV-2 is split into groups A to H also resulting from independent zoonotic transmissions). While Groups N and P have been identified in only a handful of individuals in Cameroon , and group O is estimated to a few thousand cases in western Africa, the majority of the pandemic is due to viruses from group M.

The most most recent common ancestor, *i.e. the putative virus that founded group M*, is estimated to have originated in what is now the Democratic Republic of Congo [@worobeyDirectEvidenceExtensive2008; @vidalUnprecedentedDegreeHuman2000; @fariaEarlySpreadEpidemic2014] at some point between 1910 and 1931 [@korberTimingAncestorHIV12000; @worobeyDirectEvidenceExtensive2008; @rambautCausesConsequencesHIV2004].

Group M is further subdivided into 9 subtypes each with distinct genetic characteristics, labeled A to K [@mccutchanGlobalEpidemiologyHIV2006; @sharpOriginsHIVAIDS2011]. Like in many viruses [@perez-losadaRecombinationVirusesMechanisms2015], when 2 genetically different strains of HIV co-infect a single host there is a risk genetic recombination leading to a new strain [@robertsonRecombinationAIDSViruses1995]. During recombination, a new genome is formed from parts of the original genomes. This can lead to new strains that can spread and form lineages of their own. HIV strains resulting from recombination are called Cirulating Recombinant Forms (CRFs). There are currently 118 identifed HIV-1 CRFs in the Los Alamos National Laboratory HIV sequence database [@HIVCirculatingRecombinant] (1 for HIV-2). There also exist many unique recombinant forms (URFs). Recombination can be particulary bothersome, complicating evolutionary analyses [@posadaRecombinationEvolutionaryGenomics2002], or facilitating drug resistance and hindering vaccine development [@taylorChallengeHIV1Subtype2008].

While subtype C represented almost half of global infections from 2004 to 2007, subtype B is the majority subtype in richer countries of North America and Western Europe [@hemelaarGlobalTrendsMolecular2011] where sequencing efforts are more common. This accounts for an over-representation of subtype B sequences in public databases such as the Los Alamos sequence database where 54% of sequences are of the B subtype and only 15% are C [@DistributionAllHIV1].

### The life cycle of HIV

-   Cycle [@freedHIV1Replication2001; @fergusonHIV1ReplicationCycle2002] shown in Figure \@ref(fig:hivCycle):
    1.  Virion attaches to CD4 host cell through membrane proteins

    2.  Membrane fusion and entry of viral material into cell

    3.  Reverse transcription of viral RNA into viral DNA (RT)

    4.  integration of viral DNA into host DNA (IN)

    5.  transcription of viral DNA into viral RNA

    6.  translation of viral RNA into polyprotein

    7.  cleaving of polyprotein to create mature viral proteins (PR)

    8.  assembly of viral RNA + mature viral proteins into new virion

    9.  Exit of new virion

```{r, hivCycleCaption}
hivCycleCaption <- "**Simplified HIV-1 life cycle**\
"
```

```{r, hivCycle, label="hivCycle", fig.cap=hivCycleCaption, eval=knitr::is_html_output(), out.width="60%", cache=FALSE}
knitr::include_graphics("./figures/HIV-Intro/HIV-cycle.png")
```

```{=tex}
\begin{figure}[h]
\centering
\includegraphics[width=0.6\textwidth]{./figures/HIV-Intro/HIV-cycle.pdf}
\extcaption{Simplified HIV-1 life cycle}{}
\label{fig:hivCycle}
\end{figure}
```
### Genetics of HIV

-   9 genes coding for 15 proteins [@frankelHIV1FifteenProteins1998], structure visible in Figure \@ref(fig:hivStructure):

    -   gag polyprotein: -\> structural proteins

        -   p17/MA: mature protein matrix, line the internal virion membrane

        -   p24/CA: capsid, core of the viral particle, protects the RNA by encapsulating it

        -   p7/NC: nucleocapsid, protects the genetic information in the viral particle by binding with viral RNA

        -   p6: promotes budding / virion release [@gottlingerEffectMutationsAffecting1991; @huangP6GagRequiredParticle1995], largely unstructured

    -   pol polyprotein: -\> essential functional proteins

        -   PR: cleaves polyproteins to get individual proteins.

        -   RT/p51/p66: turns viral RNA into DNA

        -   IN/p32: intergrates viral DNA in host DNA

    -   env polyprotein: -\> envelope

        -   SU (surface)/gp 120: responsible for binding to CD4 cells [@bourHumanImmunodeficiencyVirus1995]

        -   TM (transmembrane)/gp 41: together with gp120 causes membrane fusion, anchors SU to the viral particle [@hernandezViruscellCellcellFusion1996]

    -   regulatory elements:

        -   tat (trans-activator of transcription) (!!) activates transcription -\> essential promoter and leads to longer transcripts [@jonesControlRnaInitiation1994]

        -   rev (Regulator of viron) (!!) helps RNA get out of nucleus for translation [@hopeViralRNAExport1997]

        -   nef (Negative regulatory factor) (accs) prevents host cell from producing defence proteins (CD4) and boosting infectivity [@mangasarianMultifacetedRoleHIV1997]

        -   vif (viral infectivity factor) (accs) probably boosts infection as mutants have less infectivity [@cohenRoleAuxiliaryProteins1996]

        -   vpu (viral prot U) (accs) helps during release [@lambVpuVprHuman1997; @cohenRoleAuxiliaryProteins1996]

        -   vpr (viral prot R) (accs), prevents mitosis and helps entry in nucleus of host [@emermanHIV1VprCell1996]

    -   A 10th gene ?

        -   First suggested in 1988 [@millerHumanImmunodeficiencyVirus1988]

        -   overlaps env and is read in the reverse sense than the rest of the genome

        -   Under selective pressure and correlated with the spread of the pandemic (specific to HIV-M) [@cassanConcomitantEmergenceAntisense2016]

        -   under active research [@savoretPilotStudyHumoral2020]

```{r, hivStructCaption}
structCaption <- "**Structure and main components of HIV-1 virion.**\ 
Structural proteins are represented in Blue, functional enzymes in pink, RNA proteins in orange and accessory proteins in green.  
Figure adapted from PDB101 [@zardecki2022] ([PDB101.rcsb.org](https://PDB101.rcsb.org), *CC By 4.0 License*, detailed list of structures used available in Appendix \\@ref(HIV-intro-appendix))."
```

```{r, hivStructure, label="hivStructure", eval=knitr::is_html_output(), fig.cap=structCaption, out.width="90%", cache=FALSE}
knitr::include_graphics("./figures/HIV-Intro/HIV-structure.png")
```

```{=tex}
\begin{figure}[h]
  \centering
  \includegraphics[width=0.9\textwidth]{./figures/HIV-Intro/HIV-structure.png}
  \extcaption{Structure and main components of HIV-1 virion.}{Structural proteins are represented in Blue, functional enzymes in pink, RNA proteins in orange and accessory proteins in green. \\
Figure adapted from PDB101 \autocite{zardecki2022} (\href{https://PDB101.rcsb.org}{PDB101.rcsb.org}, \textit{CC By 4.0 License}, detailed list of structures used available in Appendix \autoref(sec:HIV-intro-appendix)).}
\end{figure}
```
### Main protein mechanisms in replication

ALl 3 are cleaved and matured from the pol polyprotein

-   RT: [@sarafianosStructureFunctionHIV12009], c.f. Figure \@ref(fig:rtStruct)

    -   2 subunits: p66 and p51 (i.e. same chain but 1 is cleaved). p66=functional, p51=structural mainly (except for NNIBP).

    -   Fold to form a canal -\> fingers/palm/thumb with the polymerase active site in the palm

    -   RNA strand and DNA go through the canal and are synthesized in the palm

    -   RNase domain at the other end of p66 to cleave the RNA strand off of the DNA once it's RTed.

    -   Once the RNA has been RTed the RT does another pass to synthesize the complementary DNA brand, this time the RNase is not active.

```{r, rtCaption}
rtCaption <- "**3D structure of HIV-1 Reverse-transcriptase**\ 
the different domains of the p66 subunit are labeled and shown in different shades of blue and green. The structural p51 subunit is shown in orange. The RNA template is shown in dark gray and the newly synthesized DNA stran in light gray. The polymerase active site is shown in red although mostly hidden by the RNA template. The 3D visualization was produced with Illustrate [@goodsellIllustrateSoftwareBiomolecular2019] using the [2hmi](https://www.rcsb.org/structure/2HMI) PDB structure."
```

```{r, rtStruct, lavel="rtStruct", fig.cap=rtCaption, out.width="80%", eval=knitr::is_html_output(), cache=FALSE}
knitr::include_graphics("./figures/HIV-Intro/rt.png")
```

```{=tex}
\begin{figure}[h]
\centering
\includegraphics[width=0.8\textwidth]{./figures/HIV-Intro/rt.png}
\extcaption{3D structure of HIV-1 Reverse-transcriptase}{the different domains of the p66 subunit are labeled and shown in different shades of blue and green. The structural p51 subunit is shown in orange. The RNA template is shown in dark gray and the newly synthesized DNA stran in light gray. The polymerase active site is shown in red although mostly hidden by the RNA template. The 3D visualization was produced with Illustrate \autocite{goodsellIllustrateSoftwareBiomolecular2019} using the \href{https://www.rcsb.org/structure/2HMI}{2hmi} PDB structure.}
\label{fig:rtStruct}
\end{figure}
```
-   IN: It pre-processes the ends of the viral DNA, cleaves the host DNA and binds the viral DNA to the host DNA (i.e. strand transfer). [@chiuStructureFunctionHIV12004; @espositoHIVIntegraseStructure1999; @delelisIntegraseIntegrationBiochemical2008]. Multimer -\> chains link up. Each chain has:

    -   Core catalytic domain: Binds with viral DNA and performs the integration

    -   C-terminal domain: Binds with any DNA, adding stability

    -   N-terminal domain: Binds with Zn^2+^ and helps multimerisation (i.e integrase stability)

-   PR: cleaves gag and pol polyproteins to produce individual structural and regulatory proteins. Symetric dimer structure [@pearlStructuralModelRetroviral1987; @gulnikHIVProteaseEnzyme2000], 2 identical chains.

    -   Ionization of an Aspartate residue in each active site produces acid-base reaction that cleaves precursor [@silvaInhibitionCatalyticMechanism1996]. (active site)

    -   Flaps are flexible parts of PR that can open and close [@hornakHIV1ProteaseFlaps2006], probably controls access of substrates to active site [@yuStructuralInsightsHIV12017]

## Drug resistance in HIV

When on ART, virus evolves under selective pressure and develops resistance -\> treatment failure.

### What is ART ?

ART is very important in reducing the viral load and therefore reducing transmission [@eisingerHIVViralLoad2019], however it does not "cure" HIV, it controls it drastically reducing mortality and morbidity [@palellaDecliningMorbidityMortality1998].

Mostly target the RT and PR proteins

History [@zolopaEvolutionHIVTreatment2010; @forsytheTwentyYearsAntiretroviral2019]:

-   azidothymidine (AZT), an RTI, was shown to be effective against HIV in 1987 [@fischlEfficacyAzidothymidineAZT1987], only 3 years later resistant strains of the virus were circulating [@richmanSusceptibilityNucleosideAnalogues1990], because it was a monotherapy.

-   Between 1088 and 1995 four more RTIs were approved by the FDA and a combination therapy was shown to be effective [@gulickTreatmentIndinavirZidovudine1997]

-   In 2008 there were 30 antiretroviral drugs have approved by the FDA [@palmisanoBriefHistoryAntiretroviral2011], of which 12 were RTI, 10 PIs and 5 combination drugs, often available as a single pill.

-   In 2019 there are 11 licensed single pill regimens in Europe [@OnepilladayHIVTreatments].

Other drugs:

-   entry inhibitors: have been shown to be effective 2 have been approved in the USA [@tiltonEntryInhibitorsTreatment2010]

-   integrase inhibitors (INI): recent focus on development, 5 approved drugs since 2007 [@scarsiHIV1IntegraseInhibitors2020]

-   Pre-exposure prophylaxis (PrEP), putting uninfected but at risk populations on ART before any known exposure. Has been shown effectively lower risk of infection in several studies [@grantPreexposureChemoprophylaxisHIV2010; @baetenAntiretroviralProphylaxisHIV2012; @buchbinderPreexposureProphylaxisPromise2011]. When adherence is maintained studies have shown a infection risk between 44% and 100% [@riddellHIVPreexposureProphylaxis2018].\
    In 2022 there are 1 PrEP regimen approved for use in Europe [@emaTruvada2018].

-   Action of ART drugs:

    -   RTI:

        -   NRTI: nucleoside analogue RTI work by binding with the active site of RT blocking DNA polymerization -\> chain terminating inhibitors

        -   NNRTI: non-nucleoside analogue RTI, bind to other a hydrophobic pocket behind the active site the RT, changing its flexibility and rendering it incative [@esnoufUniqueFeaturesStructure1997]

    -   PI: bind to active site of protease to hinder action [@robertsRationalDesignPeptideBased1990]

    -   Entry inhibitors are being found [@esteHIVEntryInhibitors2007; @kilbyNovelTherapiesBased2003]. enfuvirtide binds to gp41 and prevents fusion [@fletcherEnfuvirtideNewDrug2003]

    -   INI: prevent strand transfer where viral DNA is first linked to host DNA [@pommierIntegraseInhibitorsTreat2005]

However Drug resistance has emerged for most treatment types even PrEP [@knoxMultidrugResistantHIV1Infection2017; @hurtPreexposureProphylaxisAntiretroviral2011] but seem to be rare and due to pre-treatment but unknown HIV infection [@gibasDrugResistanceHIV2019]

### How can HIV evade ART ?

review [@ammaranondMechanismHIVAntiretroviral2012; @clavelHIVDrugResistance2004]

-   NRTI [@menendez-ariasMechanismsResistanceNucleoside2008; @sluis-cremerMolecularMechanismsHIV12000]:
    -   Prevent incorporation of NRTI, like M184V [@sarafianosLamivudine3TCResistance1999]

    -   Remove incorporated NRTI: TAMs [@meyerMechanismAZTResistance1999; @boyerSelectiveExcisionAZTMP2001]
-   NNRTI: mutations limiting access to the binding pocket like Y181C or changing the shape of it like K103N [@deeksNonnucleosideReverseTranscriptase2001]. dangerous because a single mutation produces viruses that are both fit and highly resistant [@ammaranondMechanismHIVAntiretroviral2012] + cross-resistance
-   PI: make active site bigger reducing affinity to inhibitors but also to natural substrate [@wensingFifteenYearsHIV2010], also change flap conformation to deny access to active site to PI [@kurtyilmazImprovingViralProtease2016].
-   INI: reduce affinity to inhibitors [@blancoHIV1IntegraseInhibitor2011]

### Consequences on global health

-   Transmission of drug resistant viruses, particularly between treatment-naive individuals which is the main transmission mode in the UK [@mouradPhylotypebasedAnalysisHighlights2015] and Switzerland [@drescherTreatmentNaiveIndividualsAre2014], creating long lasting drug resistance reservoirs in naive population.

-   Dangerous because there can be therapeutic failure of first line regimen. [@boermaHighLevelsPretreatment2016]

-   Fitness is lower for some DRMs [@kuhnertQuantifyingFitnessCost2018; @mespledeViralFitnessCost2013] which means there is evolutionary pressure for these mutations to disappear in the absence of treatment

-   the reversion to WT can last a long time with a wide range of median loss times from 1 to 13 years [@castroPersistenceHIV1Transmitted2013], depending on the type of mutation (NRTI \> NNRTI).

### Finding DRMS

(This sections build upon a review we did [@blasselDrugResistanceMutations2021])

-   Finding and categorizing DRMs is an important task.
-   Sequence databases useful in discovery:
    -   UK-CHIC [@ukchicsteeringcommitteeCreationLargeUKbased2004] and the uk HIV drug resistance database (<https://www.hivrdb.org.uk/>)

    -   Swiss cohort study (<https://www.shcs.ch/>)

    -   PANGEA [@abeler-dornerPANGEAHIVPhylogeneticsNetworks2019]

    -   Stanford HIV Drug resistance database (<https://hivdb.stanford.edu/>) with sequences [@rheeHumanImmunodeficiencyVirus2003]
-   Databases of mutations are kept [@shaferRationaleUsesPublic2006] with regularly update lists of important DRMs [@wensing2019UpdateDrug2019; @clarkMutationsRetroviralGenes2007] and tools to help practitioners detect and interpret resistance [@liuWebResourcesHIV2006d].
-   stat tests
    -   multiple testing
    -   phylogenetic correlation
-   assays
-   novel approaches
    -   deep learning
    -   ...

\printbibliography[segment=\therefsegment,heading=subbibintoc,title={References for chapter \thechapter}]
