---
title: "From sequences to knowledge, improving and learning from sequence alignments"
author: "Luc Blassel"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: scrbook
fontsize: 11pt
classoption:
  - twoside
  - listof=totoc
csl: "./styles/nature.csl"
biblio-style: authoryear
bibliography: 
  - ./references/HPC-MSRs.bib
  - ./references/HIV-DRMs.bib
  - ./references/Sequence-Data.bib
  - ./references/Aligning-Sequences.bib
  - ./references/HIV-context.bib
  - ./references/Learning-from-alignments.bib
biblatexoptions: [backref=true,refsegment=chapter,defernumbers=true,sorting=none,citestyle=numeric,bibstyle=numeric,maxnames=3,maxbibnames=3,minnames=2,uniquelist=false,backend=biber]
link-citations: yes
github-repo: lucblassel/phd-manuscript
biblio-title: Global References
lof: yes
lot: yes
titlepage: tex/titlepage.tex
---

```{r, setup, include=FALSE}
library(tidyr)
library(readr)
library(dplyr)
library(ggplot2)
library(kableExtra)
library(flextable)
library(grid)
library(tiff)

options(tinytex.verbose=TRUE)
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE, out.width="90%", fig.align="center", cache=TRUE)

# embed files in html output
xfun::pkg_load2(c("htmltools", "mime"))
```

\pagestyle{empty}

```{r, include=knitr::is_latex_output()}
knitr::asis_output('\n\n\\begin{comment}')
```

#  {.unnumbered}

## Abstract {.unnumbered}

```{r, results='asis'}
cat(readLines('abstract/abstract_en.txt'), sep='\n')
```

## Résumé {.unnumbered}

```{r, results='asis'}
cat(readLines('abstract/abstract_fr.txt'), sep='\n')
```

# Acknowledgments {.unnumbered}

Thanks everybody

# List of Acronyms and abbreviations {.unnumbered}

**DNA**: Desoxy Ribonucleic Acid

**RNA**: Ribonucleic Acid

**PCR**: Polymerase Chain Reaction

**NGS**: Next Generation Sequencing

**NTP**: Nucleotide Triphosphate

**dNTP**: deoxyNucleotide Triphosphate

**SMRT**: Single Molecule Real-Time Sequencing

**PacBio**: Pacific Biosciences

**ONT**: Oxford Nanopore Technologies

**ZMW**: Zero-Mode Waveguide

**ORF**: Open Reading Frame

**HiFi**: High-Fidelity

**CCS**: Circular Consensus Sequencing

**HMM**: Hidden Markov Model

**T2T**: Telomere to Telomere Consortium

**HPC**: Homopolymer Compression

**NW**: Needleman Wunsch algorithm

**SW**: Smith Waterman algorithm

**PAM**: Point Accepted Mutation

**BLOSUM**: Block Substitution

**FFT**: Fast Fourier Transform

**WFA**: WaveFront algorithm

**BLAST**: Basic Local Alignment Search Tool

**MEM**: Maximal Exact Match

**MUM**: Maximal Unique Match

**BWT**: Burrows Wheeler Transform

**ETR**: Extra-long Tandem Repeat

**MSA**: Multiple Sequence Alignment

**mapq**: mapping quality

**MSR**: Mapping-friendly sequence reduction

**SSR**: Streaming sequence reduction

**RC**: reverse complement

**HIV**: Human Immunodeficiency Virus

**DRM**: Drug Resistance Mutation

**AIDS**: Acquired Immunodeficience Syndrome

**SIV**: Human Immunodeficiency Virus

**CRF**: Circulating Recombinant Form

**URF**: Unique Recombinant Form

**RT**: Reverse Transcriptase

**NRTI**: Nucleoside Reverse Transcriptase Inhibitor

**NNRTI**: Non-Nucleoside Reverse Transcriptase Inhibitor

**NNIBP**: Non-Nucleoside Reverse Transcriptase Inhibitor Binding Pocket

**HAART**: Highly Active Antiretroviral therapy

**RTI**: Reverse Transcriptase Inhibitor

**IN**: Integrase

**FI**: Fusion Inhibitor

**EI**: Entry Inhipitor

**PE**: Pharmacokinetic Enhancer

**INSTI**: Integrase Strand Transfer Inhibitor

**PR**: Protease

**MA**: Matrix

**CA**: Capsid

**NC**: Nucleocapsid

**SU**: Surface

**TM**: Transmembrane

**ART**: Antiretroviral therapy

**PI**: Protease Inhibitor

**FDC**: Fixed-Dose Combination

**TAM**: Thymidine analog mutation

**FWER**: Family-wise Error Rate

**FDR**: Flase Discovery Rate

**RAM**: Resistance Associated Mutation

**SVM**: Support Vector Machine

**KNN**: k-Nearest Neighbors

**RF**: Random Forest

**ANN**: Artificial Neural Network

**PRAM**: Potentially Resistance Associated Mutation

**NB**: Naive Bayes

**LR**: Logisitic Regression

**FC**: Fischer Classifier

**AMI**: Adjusted Mutual Information

**MI**: Mutual Information

**ML**: Machine Learning

**RR**: Risk Ratio

**AA**: Amino Acid

```{r, include=knitr::is_latex_output()}
knitr::asis_output('\n\n\\end{comment}')
```

```{=tex}
\chapter{List of Acronyms and Abbreviations}
\begin{tabular}{ll}
\textbf{DNA}:& Desoxy Ribonucleic Acid \\
\textbf{RNA}:& Ribonucleic Acid \\
\textbf{PCR}:& Polymerase Chain Reaction \\
\textbf{NGS}:& Next Generation Sequencing \\
\textbf{NTP}:& Nucleotide Triphosphate \\
\textbf{dNTP}:& deoxyNucleotide Triphosphate \\
\textbf{SMRT}:& Single Molecule Real-Time Sequencing \\
\textbf{PacBio}:& Pacific Biosciences \\
\textbf{ONT}:& Oxford Nanopore Technologies \\
\textbf{ZMW}:& Zero-Mode Waveguide \\
\textbf{ORF}:& Open Reading Frame \\
\textbf{HiFi}:& High-Fidelity \\
\textbf{CCS}:& Circular Consensus Sequencing\\
\textbf{HMM}:& Hidden Markov Model\\
\textbf{T2T}:& Telomere to Telomere Consortium \\
\textbf{HPC}:& Homopolymer Compression\\
\textbf{NW}:& Needleman Wunsch algorithm \\
\textbf{SW}:& Smith Waterman algorithm \\
\textbf{PAM}:& Point Accepted Mutation \\
\textbf{BLOSUM}:& Block Substitution \\
\textbf{FFT}:& Fast Fourier Transform \\
\textbf{WFA}:& WaveFront algorithm \\
\textbf{BLAST}:& Basic Local Alignment Search Tool \\
\textbf{MEM}:& Maximal Exact Match \\
\textbf{MUM}:& Maximal Unique Match \\
\textbf{BWT}:& Burrows Wheeler Transform \\
\textbf{ETR}:& Extra-long Tandem Repeat \\
\textbf{MSA}:& Multiple Sequence Alignment \\
\textbf{mapq}:& mapping quality \\
\textbf{MSR}:& Mapping-friendly sequence reduction \\
\textbf{SSR}:& Streaming sequence reduction \\
\textbf{RC}:& reverse complement \\
\textbf{HIV}:& Human Immunodeficiency Virus  \\
\textbf{DRM}:& Drug Resistance Mutation  \\
\textbf{AIDS}:& Acquired Immunodeficience Syndrome  \\
\textbf{SIV}:& Human Immunodeficiency Virus  \\
\textbf{CRF}:& Circulating Recombinant Form  \\
\textbf{URF}:& Unique Recombinant Form  \\
\textbf{RT}:& Reverse Transcriptase  \\
\textbf{NRTI}:& Nucleoside Reverse Transcriptase Inhibitor  \\
\textbf{NNRTI}:& Non-Nucleoside Reverse Transcriptase Inhibitor  \\
\textbf{NNIBP}:& Non-Nucleoside Reverse Transcriptase Inhibitor Binding Pocket  \\
\textbf{HAART}:& Highly Active Antiretroviral therapy  \\
\textbf{RTI}:& Reverse Transcriptase Inhibitor  \\
\textbf{IN}:& Integrase  \\
\textbf{FI}:& Fusion Inhibitor  \\
\textbf{EI}:& Entry Inhipitor  \\
\textbf{PE}:& Pharmacokinetic Enhancer  \\
\textbf{INSTI}:& Integrase Strand Transfer Inhibitor  \\
\textbf{PR}:& Protease  \\
\textbf{MA}:& Matrix  \\
\textbf{CA}:& Capsid  \\
\textbf{NC}:& Nucleocapsid  \\
\textbf{SU}:& Surface  \\
\textbf{TM}:& Transmembrane  \\
\textbf{ART}:& Antiretroviral therapy  \\
\textbf{PI}:& Protease Inhibitor  \\
\textbf{FDC}:& Fixed-Dose Combination  \\
\textbf{TAM}:& Thymidine analog mutation  \\
\textbf{FWER}:& Family-wise Error Rate  \\
\textbf{FDR}:& Flase Discovery Rate  \\
\textbf{RAM}:& Resistance Associated Mutation \\
\textbf{SVM}:& Support Vector Machine \\
\textbf{KNN}:& k-Nearest Neighbors \\
\textbf{RF}:& Random Forest \\
\textbf{ANN}:& Artificial Neural Network \\
\textbf{PRAM}:& Potentially Resistance Associated Mutation \\
\textbf{NB}:& Naive Bayes \\
\textbf{LR}:& Logisitic Regression \\
\textbf{FC}:& Fischer Classifier \\
\textbf{AMI}:& Adjusted Mutual Information \\
\textbf{MI}:& Mutual Information \\
\textbf{ML}:& Machine Learning \\
\textbf{RR}:& Risk Ratio \\
\textbf{AA}:& Amino Acid \\
\end{tabular}

\mainmatter

\pagestyle{fancy}
\fancyhf{}
\fancyhead[LE]{CHAPTER \thechapter} 
\fancyhead[RO]{\rightmark}
\fancyfoot[CO,CE]{\thepage}
```
# General Introduction {.unnumbered}

-   Explain shortly that 2 quite different subjects linked by alignment and sequence data.

## Organization of this manuscript {.unnumbered}

-   Organisation of the manuscript

## Research output {.unnumbered}

During this thesis, my work on finding drug resistance mutations with machine learning resulted in two publications: a first author article describing our method published in *PLOS Computational Biology* as well as a co-first author review article published in *Current Opinion in Virology*.

The second half of my PhD work, on improving read-mapping resulted in a first-author paper, presented at the RECOMB-SEQ 2022 conference and to be published in the *iScience* proceedings of that conference.

In 2020, during the early stages of the COVID-19 pandemic and the lockdowns, I participated in some work resulting in the COVID-Align web-service and a middle-authorship in the corresponding *Bioinformatics* publication. This work also led to middle-authorship on the origins of SARS-CoV-2 in the *Comptes Rendus. Biologies* journal of the French Science Academy.

### Journal publications {.unnumbered}

-   **Blassel, Luc**, Paul Medvedev and Rayan Chikhi. 2022. **"Mapping-friendly sequence reductions: going beyond homopolymer compression"**. accepted as part of the [RECOMB-SEQ 2022](https://recomb2022.net/recomb-seq/) proceedings in *iScience*,\
    To be published *(Adapted as Chapter* \@ref(HPC-paper)*)*

-   **Blassel, Luc**[^index-1], Anna Zhukova^1^, Christian J Villabona-Arenas, Katherine E Atkins, Stéphane Hué, and Olivier Gascuel. 2021. **"Drug Resistance Mutations in HIV: New Bioinformatics Approaches and Challenges."** *Current Opinion in Virology* 51 (December): 56--64.\
    [10.1016/j.coviro.2021.09.009](https://doi.org/10.1016/j.coviro.2021.09.009) *(Used as the basis for Section* \@ref(finding-drms)*)*

-   **Blassel, Luc**, Anna Tostevin, Christian Julian Villabona-Arenas, MartinePeeters, Stéphane Hué, and Olivier Gascuel. 2021. **"Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV."** *PLOS Computational Biology* 17 (8): e1008873.\
    [10.1371/journal.pcbi.1008873](https://doi.org/10.1371/journal.pcbi.1008873). *(Adapted as Chapter* \@ref(HIV-paper)*)*

-   Zhukova, Anna, **Luc Blassel**, Frédéric Lemoine, Marie Morel, JakubVoznica, and Olivier Gascuel. 2021. **"Origin, Evolution and Global Spread of SARS-CoV-2."** *Comptes Rendus. Biologies* 344 (1): 57--75.\
    [10.5802/crbiol.29](https://doi.org/10.5802/crbiol.29).

-   Lemoine, Frédéric, **Luc Blassel**, Jakub Voznica, and Olivier Gascuel.2020. **"COVID-Align: accurate online alignment of hCoV-19 genomes using a profile HMM"** *Bioinformatics*, 37 (12): 1761-1762.\
    [10.1093/bioinformatics/btaa871](https://doi.org/10.1093/bioinformatics/btaa871).

[^index-1]: Co-first authors: Luc Blassel and Anna Zhukova

### Presentations and posters {.unnumbered}

-   **"Mapping-friendly sequence reductions: going beyond homopolymer compression"** proceedings talk, [RECOMB-SEQ 2022](https://recomb2022.net/recomb-seq/). San Diego, USA *(May 21^st^ 2022)*

-   **"Can we improve analyses be transforming DNA?"** Joint RECOMB-SEQ RECOMB-CCB scientific [communication session](https://recomb2022.net/recomb-ccb-seq-scientific-communication/)[^index-2]. San Diego, USA (*May 21^st^* *2022).*

-   "**Machine learning approaches to reveal resistance mutations in HIV"** Poster at [MCEB 2019](https://www.lirmm.fr/mceb2019/). Porquerolles, France *(May 29^th^ 2019)*

[^index-2]: 2nd place prize awarded
