---
title: "From sequences to knowledge, improving and learning from sequence alignments"
author: "Luc Blassel"
date: "2022-12-02"
site: bookdown::bookdown_site
documentclass: scrbook
fontsize: 11pt
classoption:
  - twoside
  - BCOR=10mm
  - listof=totoc
  - cleardoubleempty
csl: "./styles/nature.csl"
biblio-style: authoryear
bibliography: ./references/references.bib
biblatexoptions: [backref=true, refsegment=chapter, defernumbers=true, sorting=none, uniquelist=false, style=numeric-comp, maxcitenames=2, maxnames = 5, giveninits=true, uniquename=init, eprint=false, texencoding=utf8, bibencoding=utf8, autocite=superscript, backend=biber]
link-citations: yes
github-repo: lucblassel/phd-manuscript
biblio-title: Global References
lof: yes
lot: yes
titlepage: tex/titlepage.tex
---

```{r, setup, include=FALSE}
library(tidyr)
library(readr)
library(dplyr)
library(ggplot2)
library(kableExtra)
library(flextable)
library(grid)
library(tiff)

options(tinytex.verbose=TRUE, tinytex.clean=FALSE)
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE, out.width="90%", fig.align="center")

# embed files in html output
xfun::pkg_load2(c("htmltools", "mime"))

# simple tex abstract to md
tex2md <- function(file) {

  return (
    stringr::str_replace_all(
      readLines(file), "\\\\textit\\{([^\\}]+)\\}", "*\\1*"
    ) %>%
      stringr::str_replace_all("\\\\textbf\\{([^\\}]+)\\}", "**\\1**")
  )

}
```

\pagestyle{empty}

```{r, include=knitr::is_latex_output()}
knitr::asis_output('\n\n\\begin{comment}')
```

#  {.unnumbered}

## Abstract {.unnumbered}

```{r, results='asis'}
cat(tex2md("tex_content/abstract_en.tex"), sep='\n')
```

## Résumé {.unnumbered}

```{r, results='asis'}
cat(tex2md("tex_content/abstract_fr.tex"), sep='\n')
```

# List of Acronyms and Abbreviations {.unnumbered}

```{r}
acros <- readr::read_csv("data/acronyms.csv") %>%
  mutate(md=paste0("**", acronym, "**: ", definition, "  ")) %>%
  mutate(tex=paste0("\\textbf{", acronym, "}:& ", definition, "\\\\"))
```

```{r, htmlAcros, include=knitr::is_html_output()}
knitr::asis_output(
  stringr::str_c(acros$md, collapse="\n")
)
```

```{r, texAcros, include=knitr::is_latex_output()}
knitr::asis_output('\n\n\\end{comment}\n')

knitr::asis_output('\\chapter{List of Acronyms and Abbreviations}\n')
knitr::asis_output('\\begin{tabular}{ll}\n')
knitr::asis_output(
  stringr::str_c(acros$tex, collapse="\n")
)
knitr::asis_output('\n\\end{tabular}\n')
```

\mainmatter

# General Introduction {.unnumbered}

This manuscript is the result of my years at the Institut Pasteur, where I built upon work initiated during an internship in 2018. During my time at the Institut Pasteur I have worked on two very distinct subjects:

1.  The study of drug resistance mutations in HIV sequences with machine learning.
2.  The study of sequence transformation functions to improve long-read mapping.

These two subjects, though distinct, do share some common characteristics: mainly that they are based on sequence data and specifically alignments. Although the research on drug resistance in HIV was conducted before that on long-read mapping, I have forgone the chronological ordering of my work in this manuscript for the sake of thematic coherence. Through the organization of this manuscript, I have tried to link all the facets of my PhD work, and it is my hope that readers will be able to follow the flow without too much jumping around.

This manuscript is articulated around seven chapters, listed as follows:

1.  An introduction to biological sequence data, how it is obtained and specific characteristics and problems inherent to long reads.
2.  An introduction to sequence alignment, and how and why read-mapping is performed.
3.  A presentation of my work on sequence transformation functions to improve long-read mapping, which was written as a standalone research article.
4.  An introduction to machine learning on biological sequence data, with a focus on techniques used later in the manuscript.
5.  An introduction to viruses and HIV in particular, with a focus on proteins important to drug resistance.
6.  A presentation of my work on drug resistance in HIV, which was written and published as a standalone research article.
7.  A short introduction to deep learning in sequence alignment and perspectives to the work presented in chapter 3.

## Research output {.unnumbered}

During this thesis, my work on finding drug resistance mutations with machine learning resulted in two publications: a first author article describing our method published in *PLOS Computational Biology* as well as a co-first author review article published in *Current Opinion in Virology*.

The second half of my PhD work, on improving read-mapping resulted in a first-author paper, presented at the RECOMB-SEQ 2022 conference and to be published in the *iScience* proceedings of that conference.

In 2020, during the early stages of the COVID-19 pandemic and the lockdowns, I participated in some work resulting in the COVID-Align web-service and a middle-authorship in the corresponding *Bioinformatics* publication. This work also led to middle-authorship in an article concerning the origins of SARS-CoV-2 in the *Comptes Rendus. Biologies* journal of the French Science Academy.

### Journal publications {.unnumbered}

This list contains the formal references of the publications mentioned above, along with my contribution represented using the [CRediT taxonomy](https://credit.niso.org/).

-   **Blassel, Luc**, Paul Medvedev and Rayan Chikhi. 2022. **"Mapping-friendly sequence reductions: going beyond homopolymer compression"**. In press as part of the [RECOMB-SEQ 2022](https://recomb2022.net/recomb-seq/) proceedings in *iScience*, *(Adapted as Chapter* \@ref(HPC-paper)*)\
    *Contributions: Formal Analysis, Investigation, Methodology, Software, Validation, Visualization, Writing -- original draft, Writing -- review & editing.

-   **Blassel, Luc**[^index-1], Anna Zhukova^1^, Christian J Villabona-Arenas, Katherine E Atkins, Stéphane Hué, and Olivier Gascuel. 2021. **"Drug Resistance Mutations in HIV: New Bioinformatics Approaches and Challenges."** *Current Opinion in Virology* 51 (December): 56--64. [10.1016/j.coviro.2021.09.009](https://doi.org/10.1016/j.coviro.2021.09.009) *(Used as the basis for Section* \@ref(finding-drms)*)\
    *Contributions: Visualization, Writing -- original draft, Writing -- review & editing.

-   **Blassel, Luc**, Anna Tostevin, Christian Julian Villabona-Arenas, MartinePeeters, Stéphane Hué, and Olivier Gascuel. 2021. **"Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV."** *PLOS Computational Biology* 17 (8): e1008873. [10.1371/journal.pcbi.1008873](https://doi.org/10.1371/journal.pcbi.1008873). *(Adapted as Chapter* \@ref(HIV-paper)*)\
    *Contributions: Formal Analysis, Investigation, Methodology, Software, Validation, Visualization, Writing -- original draft, Writing -- review & editing.

-   Zhukova, Anna, **Luc Blassel**, Frédéric Lemoine, Marie Morel, JakubVoznica, and Olivier Gascuel. 2021. **"Origin, Evolution and Global Spread of SARS-CoV-2."** *Comptes Rendus. Biologies* 344 (1): 57--75. [10.5802/crbiol.29](https://doi.org/10.5802/crbiol.29).\
    Contributions: Writing -- review & editing.

-   Lemoine, Frédéric, **Luc Blassel**, Jakub Voznica, and Olivier Gascuel.2020. **"COVID-Align: accurate online alignment of hCoV-19 genomes using a profile HMM"** *Bioinformatics*, 37 (12): 1761-1762. [10.1093/bioinformatics/btaa871](https://doi.org/10.1093/bioinformatics/btaa871).\
    Contributions: Software, Writing -- review & editing

[^index-1]: Co-first authors: Luc Blassel and Anna Zhukova

### Presentations and posters {.unnumbered}

-   **"Mapping-friendly sequence reductions: going beyond homopolymer compression"** proceedings talk, [RECOMB-SEQ 2022](https://recomb2022.net/recomb-seq/). San Diego, USA *(May 21^st^ 2022)*

-   **"Can we improve analyses be transforming DNA?"** Joint RECOMB-SEQ RECOMB-CCB scientific [communication session](https://recomb2022.net/recomb-ccb-seq-scientific-communication/)[^index-2]. San Diego, USA (*May 21^st^* *2022).*

-   "**Machine learning approaches to reveal resistance mutations in HIV"** Poster at [MCEB 2019](https://www.lirmm.fr/mceb2019/). Porquerolles, France *(May 29^th^ 2019)*

[^index-2]: 2nd place prize awarded

```{=tex}
\pagestyle{fancy}
\fancyhf{}
\fancyhead[LE]{CHAPTER \thechapter}
\fancyhead[RO]{\rightmark}
\fancyfoot[CO,CE]{\thepage}

% Number footnotes in main text with letters (not to confuse with Nature style citations)
\renewcommand*\thefootnote{\alph{footnote}}
```
# Acknowledgements {.unnumbered}

First, I would like to thank all the members of my jury for accepting to evaluate the work that I have done over the past few years. In particular, I would like to thank Macha Nikolski and Brona Brejova for reading my manuscript, reviewing it, and providing valuable feedback on my work. Second, I would like to thank the people that supervised me and helped me along the way. Thank you to Paul for your valuable insights. Thank you to Rayan for supervising me in these last years of this project, and to Olivier for doing so in the first. I would also like to thank Didier for helping me navigate the turmoil.

For most of my PhD, except for a few months in early 2020..., I spent my time on the 6th floor of the Omics building of the Institut Pasteur, in the *Evolutionary Bioinformatics* lab. Thank you to all my office mates for making my time on the 6th floor so pleasant. Thank you Anna, Fred and Jakub for all the discussions and coffee breaks. A special thanks to my good-***colleague*** Marie, who has  ingrained in me the response of looking up and to the right whenever I hear an office chair move. Thank you also to the 6th floor adoptees: Jerome, Andrew, and Arthur, the earliest of them. I would also like to thank my labmates from the *Sequence Bioinformatics* lab: Camilla, Francesco, Yoann, Riccardo, Luca and Téo.

Of course, as studious as I might wish to be, I did spend 4 years at the office. I would therefore like to thank all my friends that have made them enjoyable. Grimault, Nathalie, Théo and Élise for the German escapades and the *(first of many I hope)* canyoning outings. Thank you as well to Eva M., Max, and Flore for being there throughout the years, and thank you to Eva C., Celia and all the other Spanish-sectionites for making me tolerate reggaeton. It has been a while since the Balzac days, but I am very glad we all still hang out! Dylan and Joshua, I started this adventure with you as roommates and enjoyed it very much, so thank you as well! Finally, thank you to Marie, Anna and Jean for many, many *(many!)* celebrations! Without all of you, my time as a PhD student would definitely have been less fun.

Next up on the thank you list is, of course, my family. I would like to thank my mom, dad and brothers, Noé and Tom, for supporting me along my journey. Thank you as well to my grandparents: Mamette, André and Grandma for believing in me and helping me along the way. To all the others from Tokyo to Santa Cruz, it warms my heart to know I have a worldwide support network, thank you!

Finally, Fatima. I am so glad I met you during this adventure. Thank you for supporting me, talking things through with me and helping me along the way. I am very grateful that you are part of my life, and I feel better looking to the future knowing you will be by my side.

As parting words, I would like to dedicate this work to Pierre Blassel and John F. Murray. They made me, in part, what I am today and were essential in instilling in me the curiosity and scientific interest that are so necessary to pursue this type of endeavor. Although they could not see me finish, I would like to thank them from the bottom of my heart and I can only hope that they would be proud of me.

\
\
