<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Global References | From sequences to knowledge, improving and learning from sequence alignments.</title>
  <meta name="description" content="Global References | From sequences to knowledge, improving and learning from sequence alignments." />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Global References | From sequences to knowledge, improving and learning from sequence alignments." />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Global References | From sequences to knowledge, improving and learning from sequence alignments." />
  
  
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-06-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="global-conclusion.html"/>
<link rel="next" href="HPC-appendix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />
<link href="libs/tabwid-1.0.0/scrool.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">My PhD thesis.</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="résumé.html"><a href="résumé.html"><i class="fa fa-check"></i>Résumé</a></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="glossary.html"><a href="glossary.html"><i class="fa fa-check"></i>Glossary</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html"><i class="fa fa-check"></i>General Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#organization-of-this-manuscript"><i class="fa fa-check"></i>Organization of this manuscript</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#publications-produced"><i class="fa fa-check"></i>Publications produced</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>1</b> What is Sequence data ?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>1.1</b> Biological sequences, a primer</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>1.1.1</b> What is DNA ?</a></li>
<li class="chapter" data-level="1.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#from-information-to-action"><i class="fa fa-check"></i><b>1.1.2</b> From Information to action</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining sequence data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough"><i class="fa fa-check"></i><b>1.2.1</b> Sanger sequencing, a breakthrough</a></li>
<li class="chapter" data-level="1.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> Next-generation sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>1.3</b> Sequencing errors, how to account for them ?</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#error-correction-methods"><i class="fa fa-check"></i><b>1.3.1</b> Error correction methods</a></li>
<li class="chapter" data-level="1.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#more-accurate-sequencing-methods"><i class="fa fa-check"></i><b>1.3.2</b> More accurate sequencing methods</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#the-special-case-of-homopolymers"><i class="fa fa-check"></i><b>1.4</b> The special case of homopolymers</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-the-genome"><i class="fa fa-check"></i><b>1.4.1</b> Homopolymers and the genome</a></li>
<li class="chapter" data-level="1.4.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-long-reads"><i class="fa fa-check"></i><b>1.4.2</b> Homopolymers and long reads</a></li>
<li class="chapter" data-level="1.4.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#accounting-for-homopolymers"><i class="fa fa-check"></i><b>1.4.3</b> Accounting for homopolymers</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>2</b> Aligning sequence data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>2.1</b> What is an alignment ?</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#why-align"><i class="fa fa-check"></i><b>2.1.1</b> Why align ?</a></li>
<li class="chapter" data-level="2.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-align-two-sequences"><i class="fa fa-check"></i><b>2.1.2</b> How to align two sequences ?</a></li>
<li class="chapter" data-level="2.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#substitution-models-scoring"><i class="fa fa-check"></i><b>2.1.3</b> Substitution models / scoring</a></li>
<li class="chapter" data-level="2.1.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#dealing-with-gaps"><i class="fa fa-check"></i><b>2.1.4</b> Dealing with gaps</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-do-we-speed-up-pairwise-alignment"><i class="fa fa-check"></i><b>2.2</b> How do we speed up pairwise alignment ?</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#change-the-method"><i class="fa fa-check"></i><b>2.2.1</b> Change the method</a></li>
<li class="chapter" data-level="2.2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#seed-and-extend-with-data-structures"><i class="fa fa-check"></i><b>2.2.2</b> Seed and extend with data structures</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.3</b> Multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#progressive-alignment"><i class="fa fa-check"></i><b>2.3.1</b> Progressive alignment</a></li>
<li class="chapter" data-level="2.3.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#hmms-in-profile-profile-or-seq-profile-alignments"><i class="fa fa-check"></i><b>2.3.2</b> HMMs in profile-profile or seq-profile alignments</a></li>
<li class="chapter" data-level="2.3.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#other-optimization-methods-short"><i class="fa fa-check"></i><b>2.3.3</b> Other optimization methods (short)</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#the-specifities-of-read-mapping"><i class="fa fa-check"></i><b>2.4</b> The specifities of read-mapping</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="HPC-paper.html"><a href="HPC-paper.html"><i class="fa fa-check"></i><b>3</b> Contribution 1: Improving read alignment by exploring a sequence transformation space</a>
<ul>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="3.1" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-methods"><i class="fa fa-check"></i><b>3.2</b> Methods</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:msr-def"><i class="fa fa-check"></i><b>3.2.1</b> Streaming sequence reductions</a></li>
<li class="chapter" data-level="3.2.2" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:enum"><i class="fa fa-check"></i><b>3.2.2</b> Restricting the space of Streaming sequence reductions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="HPC-paper.html"><a href="HPC-paper.html#datasets-and-pipelines"><i class="fa fa-check"></i><b>3.3</b> Datasets and Pipelines</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="HPC-paper.html"><a href="HPC-paper.html#datasets"><i class="fa fa-check"></i><b>3.3.1</b> Datasets</a></li>
<li class="chapter" data-level="3.3.2" data-path="HPC-paper.html"><a href="HPC-paper.html#simulation-pipeline"><i class="fa fa-check"></i><b>3.3.2</b> Simulation pipeline</a></li>
<li class="chapter" data-level="3.3.3" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:evalpipeline"><i class="fa fa-check"></i><b>3.3.3</b> Evaluation pipeline</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-results"><i class="fa fa-check"></i><b>3.4</b> Results</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="HPC-paper.html"><a href="HPC-paper.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>3.4.1</b> Selection of mapping-friendly sequence reductions</a></li>
<li class="chapter" data-level="3.4.2" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>3.4.2</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes</a></li>
<li class="chapter" data-level="3.4.3" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>3.4.3</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome</a></li>
<li class="chapter" data-level="3.4.4" data-path="HPC-paper.html"><a href="HPC-paper.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>3.4.4</b> Raw mapping improves upon HPC on centromeric regions</a></li>
<li class="chapter" data-level="3.4.5" data-path="HPC-paper.html"><a href="HPC-paper.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>3.4.5</b> Positions of incorrectly mapped reads across the entire human genome</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-discussion"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="HPC-paper.html"><a href="HPC-paper.html#limitations-of-this-study"><i class="fa fa-check"></i><b>3.6</b> Limitations of this study</a></li>
<li class="chapter" data-level="3.7" data-path="HPC-paper.html"><a href="HPC-paper.html#code-availability"><i class="fa fa-check"></i><b>3.7</b> Code availability</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#supplementary-information"><i class="fa fa-check"></i>Supplementary information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html"><i class="fa fa-check"></i><b>4</b> Learning from alignments</a>
<ul>
<li class="chapter" data-level="4.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#alignments-are-a-rich-source-of-information"><i class="fa fa-check"></i><b>4.1</b> Alignments are a rich source of information</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#pairwise-alns"><i class="fa fa-check"></i><b>4.1.1</b> Pairwise alns</a></li>
<li class="chapter" data-level="4.1.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#msa"><i class="fa fa-check"></i><b>4.1.2</b> MSA</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#preprocessing-the-alignment-for-machine-learning"><i class="fa fa-check"></i><b>4.2</b> Preprocessing the alignment for machine learning</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#embedding-the-alignment"><i class="fa fa-check"></i><b>4.2.1</b> Embedding the alignment</a></li>
<li class="chapter" data-level="4.2.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#choosing-a-learning-target"><i class="fa fa-check"></i><b>4.2.2</b> Choosing a learning target</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#how-to-learn-from-alns"><i class="fa fa-check"></i><b>4.3</b> How to learn from ALNs</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>4.3.1</b> Tests and statistical learning</a></li>
<li class="chapter" data-level="4.3.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#taking-interactions-into-account"><i class="fa fa-check"></i><b>4.3.2</b> Taking interactions into account</a></li>
<li class="chapter" data-level="4.3.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#deep-learning"><i class="fa fa-check"></i><b>4.3.3</b> Deep Learning</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html"><i class="fa fa-check"></i><b>5</b> HIV and DRMs</a>
<ul>
<li class="chapter" data-level="5.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-are-viruses"><i class="fa fa-check"></i><b>5.1</b> What are viruses ?</a></li>
<li class="chapter" data-level="5.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-is-hiv"><i class="fa fa-check"></i><b>5.2</b> What is HIV ?</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#quick-presentation-of-hiv"><i class="fa fa-check"></i><b>5.2.1</b> Quick Presentation of HIV</a></li>
<li class="chapter" data-level="5.2.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#replication-cycle-of-hiv"><i class="fa fa-check"></i><b>5.2.2</b> Replication cycle of HIV</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#drug-resistance-in-hiv"><i class="fa fa-check"></i><b>5.3</b> Drug resistance in HIV</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#how-does-art-work"><i class="fa fa-check"></i><b>5.3.1</b> How does ART work</a></li>
<li class="chapter" data-level="5.3.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#different-types-of-resistance"><i class="fa fa-check"></i><b>5.3.2</b> different types of resistance</a></li>
<li class="chapter" data-level="5.3.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#consequences-on-global-health"><i class="fa fa-check"></i><b>5.3.3</b> Consequences on global health</a></li>
<li class="chapter" data-level="5.3.4" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#finding-drms"><i class="fa fa-check"></i><b>5.3.4</b> Finding DRMS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>6</b> Contribution 2: Inferring mutation roles from sequence alignments using machine learning</a>
<ul>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#abstract-paper"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#author-summary"><i class="fa fa-check"></i>Author summary</a></li>
<li class="chapter" data-level="6.1" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>6.2</b> Materials and methods</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>6.2.2</b> Classifier training</a></li>
<li class="chapter" data-level="6.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>6.2.3</b> Measuring classifier performance</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-results"><i class="fa fa-check"></i><b>6.3</b> Results</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>6.3.1</b> Classifier performance &amp; interpretation</a></li>
<li class="chapter" data-level="6.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>6.3.2</b> Additional classification results</a></li>
<li class="chapter" data-level="6.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>6.3.3</b> Identifying new mutations from classifiers</a></li>
<li class="chapter" data-level="6.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>6.3.4</b> Detailed analysis of potentially resistance-associated mutations</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>6.4</b> Discussion and perspectives</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-acknowledgments"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#supporting-information"><i class="fa fa-check"></i>Supporting Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>7</b> Learning alignments, an interesting perspective</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-pairwise-alignment"><i class="fa fa-check"></i><b>7.1</b> Learning pairwise alignment</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#dedal"><i class="fa fa-check"></i><b>7.1.1</b> DEDAL</a></li>
<li class="chapter" data-level="7.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-an-alignment"><i class="fa fa-check"></i><b>7.1.2</b> predicting an alignment</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#what-else-could-we-learn"><i class="fa fa-check"></i><b>7.2</b> What else could we learn ?</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-to-predict-seeds-or-starting-positions"><i class="fa fa-check"></i><b>7.2.1</b> Learn to predict seeds or starting positions</a></li>
<li class="chapter" data-level="7.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-pre-processing-functions"><i class="fa fa-check"></i><b>7.2.2</b> Learn pre-processing functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html"><i class="fa fa-check"></i>Global conclusion</a>
<ul>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hpc-part"><i class="fa fa-check"></i>HPC part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hiv-part"><i class="fa fa-check"></i>HIV part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#final-words"><i class="fa fa-check"></i>Final words</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-references.html"><a href="global-references.html"><i class="fa fa-check"></i>Global References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”</a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation</a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#msr-performance-comparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison</a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#origin-of-incorrectly-mapped-reads-of-high-mapping-quality-on-whole-human-genome."><i class="fa fa-check"></i><b>A.3</b> Origin of incorrectly mapped reads of high mapping quality on whole human genome.</a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome"><i class="fa fa-check"></i><b>A.4</b> Analyzing read origin on whole human genome</a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.5</b> Performance of MSRs on the Drosophila genome</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”</a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>B.1</b> S1 Appendix (Technical appendix).</a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>B.1.1</b> Data</a></li>
<li class="chapter" data-level="B.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>B.1.2</b> Classifiers</a></li>
<li class="chapter" data-level="B.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>B.1.3</b> Scoring</a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>B.2</b> S1 Fig.</a></li>
<li class="chapter" data-level="B.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>B.3</b> S2 Fig.</a></li>
<li class="chapter" data-level="B.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>B.4</b> S3 Fig.</a></li>
<li class="chapter" data-level="B.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>B.5</b> S1 Table.</a></li>
<li class="chapter" data-level="B.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>B.6</b> S2 Appendix. (Fisher exact tests)</a></li>
<li class="chapter" data-level="B.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>B.7</b> S1 Data.</a></li>
<li class="chapter" data-level="B.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>B.8</b> S2 Data.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequences to knowledge, improving and learning from sequence alignments.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="global-references" class="section level1 unnumbered hasAnchor">
<h1>Global References<a href="global-references.html#global-references" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Watson, J. D. &amp; Crick, F. H. C. <a href="https://doi.org/10.1101/SQB.1953.018.01.020">The Structure of Dna</a>. <em>Cold Spring Harbor Symposia on Quantitative Biology</em> <strong>18</strong>, 123–131 (1953).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Sanger, F. <em>et al.</em> <a href="https://doi.org/10.1038/265687a0">Nucleotide sequence of bacteriophage <span class="nocase">φX174 DNA</span></a>. <em>Nature</em> <strong>265</strong>, 687–695 (1977).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Archer, C. T. <em>et al.</em> <a href="https://doi.org/10.1186/1471-2164-12-9">The genome sequence of e. Coli w (ATCC 9637): Comparative genome analysis and an improved genome-scale reconstruction of e. coli</a>. <em>BMC Genomics</em> <strong>12</strong>, 9 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Nurk, S. <em>et al.</em> <a href="https://doi.org/10.1126/science.abj6987">The complete sequence of a human genome</a>. <em>Science</em> <strong>376</strong>, 44–53 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Pellicer, J., Fay, M. F. &amp; Leitch, I. J. <a href="https://doi.org/10.1111/j.1095-8339.2010.01072.x">The largest eukaryotic genome of them all?</a> <em>Botanical Journal of the Linnean Society</em> <strong>164</strong>, 10–15 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Macgregor, H. C. C-Value Paradox. in (eds. Brenner, S. &amp; Miller, J. H.) 249–250 (Academic Press, 2001). doi:<a href="https://doi.org/10.1006/rwgn.2001.0301">10.1006/rwgn.2001.0301</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Alberts, B. <em>et al.</em> <em><a href="https://www.ncbi.nlm.nih.gov/books/NBK26916/">Molecular Biology of the Cell. 4th edition</a></em>. (Garland Science, 2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Crick, F. H. C., Barnett, L., Brenner, S. &amp; Watts-Tobin, R. J. <a href="https://doi.org/10.1038/1921227a0">General Nature of the Genetic Code for Proteins</a>. <em>Nature</em> <strong>192</strong>, 1227–1232 (1961).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">International Human Genome Sequencing Consortium. <a href="https://doi.org/10.1038/nature03001">Finishing the euchromatic sequence of the human genome</a>. <em>Nature</em> <strong>431</strong>, 931–945 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Elkon, R. &amp; Agami, R. <a href="https://doi.org/10.1038/nbt.3863">Characterization of noncoding regulatory DNA in the human genome</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 732–746 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Omenn, G. S. <a href="https://doi.org/10.1016/j.mcpro.2021.100062">Reflections on the HUPO human proteome project, the flagship project of the human proteome organization, at 10 years</a>. <em>Molecular &amp; Cellular Proteomics : MCP</em> <strong>20</strong>, 100062 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Shabalina, S. A. &amp; Spiridonov, N. A. <a href="https://doi.org/10.1186/gb-2004-5-4-105">The mammalian transcriptome and the function of non-coding DNA sequences</a>. <em>Genome Biology</em> <strong>5</strong>, 105 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">ENCODE Project Consortium. <a href="https://doi.org/10.1038/nature11247">An integrated encyclopedia of DNA elements in the human genome</a>. <em>Nature</em> <strong>489</strong>, 57–74 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Chatterjee, N. &amp; Walker, G. C. <a href="https://doi.org/10.1002/em.22087">Mechanisms of DNA damage, repair, and mutagenesis: DNA Damage and Repair</a>. <em>Environmental and Molecular Mutagenesis</em> <strong>58</strong>, 235–263 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Fijalkowska, I. J., Schaaper, R. M. &amp; Jonczyk, P. <a href="https://doi.org/10.1111/j.1574-6976.2012.00338.x">DNA replication fidelity in escherichia coli: A multi-DNA polymerase affair</a>. <em>FEMS microbiology reviews</em> <strong>36</strong>, 1105–1121 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Pray, L. DNA replication and causes of mutation. <em>Nature education</em> <strong>1</strong>, 214 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Gout, J.-F., Thomas, W. K., Smith, Z., Okamoto, K. &amp; Lynch, M. <a href="https://doi.org/10.1073/pnas.1309843110">Large-scale detection of in vivo transcription errors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>110</strong>, 18584–18589 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Gout, J.-F. <em>et al.</em> <a href="https://doi.org/10.1126/sciadv.1701484">The landscape of transcription errors in eukaryotic cells</a>. <em>Science Advances</em> <strong>3</strong>, e1701484 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Desouky, O., Ding, N. &amp; Zhou, G. <a href="https://doi.org/10.1016/j.jrras.2015.03.003">Targeted and non-targeted effects of ionizing radiation</a>. <em>Journal of Radiation Research and Applied Sciences</em> <strong>8</strong>, 247–254 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Kiefer, J. Effects of Ultraviolet Radiation on DNA. in (eds. Obe, G. &amp; Vijayalaxmi) 39–53 (Springer, 2007). doi:<a href="https://doi.org/10.1007/978-3-540-71414-9_3">10.1007/978-3-540-71414-9_3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Bennett, J. W. &amp; Klich, M. <a href="https://doi.org/10.1128/CMR.16.3.497-516.2003">Mycotoxins</a>. <em>Clinical Microbiology Reviews</em> <strong>16</strong>, 497–516 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Kantidze, O. L., Velichko, A. K., Luzhin, A. V. &amp; Razin, S. V. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4947990/">Heat stress-induced DNA damage</a>. <em>Acta Naturae</em> <strong>8</strong>, 75–78 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Gregory, C. D. &amp; Milner, A. E. <a href="https://doi.org/10.1002/ijc.2910570321">Regulation of cell survival in Burkitt lymphoma: implications from studies of apoptosis following cold-shock treatment</a>. <em>International Journal of Cancer</em> <strong>57</strong>, 419–426 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Gafter-Gvili, A. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0068341">Oxidative stress-induced DNA damage and repair in human peripheral blood mononuclear cells: Protective role of hemoglobin</a>. <em>PLoS ONE</em> <strong>8</strong>, e68341 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Kujovich, J. L. <a href="https://doi.org/10.1097/GIM.0b013e3181faa0f2">Factor V Leiden thrombophilia</a>. <em>Genetics in Medicine</em> <strong>13</strong>, 1–16 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Cutting, G. R. <a href="https://doi.org/10.1038/nrg3849">Cystic fibrosis genetics: from molecular understanding to clinical application</a>. <em>Nature Reviews Genetics</em> <strong>16</strong>, 45–56 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Fuchsberger, C. <em>et al.</em> <a href="https://doi.org/10.1038/nature18642">The genetic architecture of type 2 diabetes</a>. <em>Nature</em> <strong>536</strong>, 41–47 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Morris, A. P. <em>et al.</em> <a href="https://doi.org/10.1038/ng.2383">Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes</a>. <em>Nature genetics</em> <strong>44</strong>, 981–990 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Rhee, S.-Y. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkg100">Human immunodeficiency virus reverse transcriptase and protease sequence database</a>. <em>Nucleic Acids Research</em> <strong>31</strong>, 298–303 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Woodford, N. &amp; Ellington, M. J. <a href="https://doi.org/10.1111/j.1469-0691.2006.01492.x">The emergence of antibiotic resistance by mutation</a>. <em>Clinical Microbiology and Infection</em> <strong>13</strong>, 5–18 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Sanger, F., Nicklen, S. &amp; Coulson, A. R. <a href="https://doi.org/10.1073/pnas.74.12.5463"><span>DNA</span> sequencing with chain-terminating inhibitors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>74</strong>, 5463–5467 (1977).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Smith, L. M., Fung, S., Hunkapiller, M. W., Hunkapiller, T. J. &amp; Hood, L. E. <a href="https://doi.org/10.1093/nar/13.7.2399">The synthesis of oligonucleotides containing an aliphatic amino group at the 5′ terminus: Synthesis of fluorescent <span>DNA</span> primers for use in <span>DNA</span> sequence analysis</a>. <em>Nucleic Acids Research</em> <strong>13</strong>, 2399–2412 (1985).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Smith, L. M. <em>et al.</em> <a href="https://doi.org/10.1038/321674a0">Fluorescence detection in automated <span>DNA</span> sequence analysis</a>. <em>Nature</em> <strong>321</strong>, 674–679 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Shendure, J. &amp; Ji, H. <a href="https://doi.org/10.1038/nbt1486">Next-generation DNA sequencing</a>. <em>Nature Biotechnology</em> <strong>26</strong>, 1135–1145 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><a href="https://www.genome.gov/about-genomics/fact-sheets/Sequencing-Human-Genome-cost">The Cost of Sequencing a Human Genome</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Collins, F. S., Morgan, M. &amp; Patrinos, A. <a href="https://doi.org/10.1126/science.1084564">The human genome project: Lessons from large-scale biology</a>. <em>Science</em> <strong>300</strong>, 286–290 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Liu, L. <em>et al.</em> <a href="https://doi.org/10.1155/2012/251364">Comparison of <span>Next-Generation Sequencing Systems</span></a>. <em>Journal of Biomedicine and Biotechnology</em> <strong>2012</strong>, e251364 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Metzker, M. L. <a href="https://doi.org/10.1038/nrg2626">Sequencing technologies — the next generation</a>. <em>Nat Rev Genet</em> <strong>11</strong>, 31–46 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Mardis, E. R. <a href="https://doi.org/10.1038/nature09796">A decade<span>’</span>s perspective on DNA sequencing technology</a>. <em>Nature</em> <strong>470</strong>, 198–203 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline">Eid, J. <em>et al.</em> <a href="https://doi.org/10.1126/science.1162986">Real-time DNA sequencing from single polymerase molecules</a>. <em>Science</em> <strong>323</strong>, 133–138 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Rhoads, A. &amp; Au, K. F. <a href="https://doi.org/10.1016/j.gpb.2015.08.002">PacBio Sequencing and Its Applications</a>. <em>Genomics, Proteomics &amp; Bioinformatics</em> <strong>13</strong>, 278–289 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Chaisson, M. J. P. <em>et al.</em> <a href="https://doi.org/10.1038/nature13907">Resolving the complexity of the human genome using single-molecule sequencing</a>. <em>Nature</em> <strong>517</strong>, 608–611 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Logsdon, G. A., Vollger, M. R. &amp; Eichler, E. E. <a href="https://doi.org/10.1038/s41576-020-0236-x">Long-read human genome sequencing and its applications</a>. <em>Nat Rev Genet</em> <strong>21</strong>, 597–614 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Murigneux, V. <em>et al.</em> <a href="https://doi.org/10.1093/gigascience/giaa146">Comparison of long-read methods for sequencing and assembly of a plant genome</a>. <em>GigaScience</em> <strong>9</strong>, giaa146 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Clarke, J. <em>et al.</em> <a href="https://doi.org/10.1038/nnano.2009.12">Continuous base identification for single-molecule nanopore DNA sequencing</a>. <em>Nature Nanotechnology</em> <strong>4</strong>, 265–270 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline">Ip, C. L. C. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.7201.1">MinION analysis and reference consortium: Phase 1 data release and analysis</a>. <em>F1000Research</em> <strong>4</strong>, 1075 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Jain, M. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.4060">Nanopore sequencing and assembly of a human genome with ultra-long reads</a>. <em>Nature Biotechnology</em> <strong>36</strong>, 338–345 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><a href="http://lab.loman.net/2017/03/09/ultrareads-for-nanopore/">Thar she blows! Ultra long read method for nanopore sequencing · loman labs</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">Payne, A., Holmes, N., Rakyan, V. &amp; Loose, M. <a href="https://doi.org/10.1093/bioinformatics/bty841">BulkVis: A graphical viewer for oxford nanopore bulk FAST5 files</a>. <em>Bioinformatics</em> <strong>35</strong>, 2193–2198 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">Jain, M., Olsen, H. E., Paten, B. &amp; Akeson, M. <a href="https://doi.org/10.1186/s13059-016-1103-0">The Oxford Nanopore MinION: delivery of nanopore sequencing to the genomics community</a>. <em>Genome Biology</em> <strong>17</strong>, 239 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Hunt, D. F., Yates, J. R., Shabanowitz, J., Winston, S. &amp; Hauer, C. R. <a href="https://doi.org/10.1073/pnas.83.17.6233">Protein sequencing by tandem mass spectrometry.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>83</strong>, 6233–6237 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Smith, B. J. <em>Protein Sequencing Protocols</em>. (Springer Science &amp; Business Media, 2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Restrepo-Pérez, L., Joo, C. &amp; Dekker, C. <a href="https://doi.org/10.1038/s41565-018-0236-6">Paving the way to single-molecule protein sequencing</a>. <em>Nature Nanotechnology</em> <strong>13</strong>, 786–796 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline">Ma, X. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-019-1659-6">Analysis of error profiles in deep next-generation sequencing data</a>. <em>Genome Biology</em> <strong>20</strong>, 50 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline">Lima, L. <em>et al.</em> <a href="https://doi.org/10.1093/bib/bbz058">Comparative assessment of long-read error correction software applied to nanopore RNA-sequencing data</a>. <em>Briefings in Bioinformatics</em> <strong>21</strong>, 1164–1181 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline">Amarasinghe, S. L. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-020-1935-5">Opportunities and challenges in long-read sequencing data analysis</a>. <em>Genome Biology</em> <strong>21</strong>, 30 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline">Ruan, J. &amp; Li, H. <a href="https://doi.org/10.1038/s41592-019-0669-3">Fast and accurate long-read assembly with wtdbg2</a>. <em>Nature Methods</em> <strong>17</strong>, 155–158 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline">Koren, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.215087.116">Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation</a>. <em>Genome Research</em> <strong>27</strong>, 722–736 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline">Tischler, G. &amp; Myers, E. W. <em>Non Hybrid Long Read Consensus Using Local De Bruijn Graph Assembly</em>. 106252 <a href="https://www.biorxiv.org/content/10.1101/106252v1">https://www.biorxiv.org/content/10.1101/106252v1</a> (2017) doi:<a href="https://doi.org/10.1101/106252">10.1101/106252</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline">Hackl, T., Hedrich, R., Schultz, J. &amp; Förster, F. <a href="https://doi.org/10.1093/bioinformatics/btu392">Proovread : Large-scale high-accuracy PacBio correction through iterative short read consensus</a>. <em>Bioinformatics</em> <strong>30</strong>, 3004–3011 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline">Miclotte, G. <em>et al.</em> <a href="https://doi.org/10.1186/s13015-016-0075-7">Jabba: Hybrid error correction for long sequencing reads</a>. <em>Algorithms for Molecular Biology</em> <strong>11</strong>, 10 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline">Koren, S. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.2280">Hybrid error correction and de novo assembly of single-molecule sequencing reads</a>. <em>Nature Biotechnology</em> <strong>30</strong>, 693–700 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline">Salmela, L. &amp; Rivals, E. <a href="https://doi.org/10.1093/bioinformatics/btu538">LoRDEC: Accurate and efficient long read error correction</a>. <em>Bioinformatics</em> <strong>30</strong>, 3506–3514 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline">Hepler, N. L. <em>et al.</em> An improved circular consensus algorithm with an application to detect HIV-1 drug-resistance associated mutations (DRAMs). in (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline">Simpson, J. T. <em>et al.</em> <a href="https://doi.org/10.1038/nmeth.4184">Detecting DNA cytosine methylation using nanopore sequencing</a>. <em>Nature Methods</em> <strong>14</strong>, 407–410 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline">Walker, B. J. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</a>. <em>PLOS ONE</em> <strong>9</strong>, e112963 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Vaser, R., Sović, I., Nagarajan, N. &amp; Šikić, M. <a href="https://doi.org/10.1101/gr.214270.116">Fast and accurate de novo genome assembly from long uncorrected reads</a>. <em>Genome Research</em> <strong>27</strong>, 737–746 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Fu, S., Wang, A. &amp; Au, K. F. <a href="https://doi.org/10.1186/s13059-018-1605-z">A comparative evaluation of hybrid error correction methods for error-prone long reads</a>. <em>Genome Biology</em> <strong>20</strong>, 26 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline">Zhang, H., Jain, C. &amp; Aluru, S. <a href="https://doi.org/10.1186/s12864-020-07227-0">A comprehensive evaluation of long read error correction methods</a>. <em>BMC Genomics</em> <strong>21</strong>, 889 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline">Wenger, A. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-019-0217-9">Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome</a>. <em>Nature Biotechnology</em> <strong>37</strong>, 1155–1162 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline">Karst, S. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-020-01041-y">High-accuracy long-read amplicon sequences using unique molecular identifiers with Nanopore or PacBio sequencing</a>. <em>Nature Methods</em> <strong>18</strong>, 165–169 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline">Perešíni, P., Boža, V., Brejová, B. &amp; Vinař, T. <a href="https://doi.org/10.1093/bioinformatics/btab528">Nanopore base calling on the edge</a>. <em>Bioinformatics</em> <strong>37</strong>, 4661–4667 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline">Boža, V., Brejová, B. &amp; Vinař, T. <a href="https://doi.org/10.1371/journal.pone.0178751">DeepNano: Deep recurrent neural networks for base calling in MinION nanopore reads</a>. <em>PLOS ONE</em> <strong>12</strong>, e0178751 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline">Wick, R. R., Judd, L. M. &amp; Holt, K. E. <a href="https://doi.org/10.1186/s13059-019-1727-y">Performance of neural network basecalling tools for Oxford Nanopore sequencing</a>. <em>Genome Biology</em> <strong>20</strong>, 129 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline">Chen, Z. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.3982">Highly accurate fluorogenic DNA sequencing with information theory<span></span>based error correction</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 1170–1178 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline"><a href="https://www.illumina.com/content/illumina-marketing/amr/en_US/science/genomics-research/articles/infinity-high-performance-long-read-assay.html">High Performance Long Read Assay Enables Contiguous Data up to 10Kb on Existing Illumina Platforms</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline">Ivády, G. <em>et al.</em> <a href="https://doi.org/10.1186/s12864-018-4544-x">Analytical parameters and validation of homopolymer detection in a pyrosequencing-based next generation sequencing system</a>. <em>BMC Genomics</em> <strong>19</strong>, 158 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline">Booeshaghi, A. S. &amp; Pachter, L. Pseudoalignment facilitates assignment of error-prone Ultima Genomics reads. doi:<a href="https://doi.org/10.1101/2022.06.04.494845">10.1101/2022.06.04.494845</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline">Delahaye, C. &amp; Nicolas, J. <a href="https://doi.org/10.1371/journal.pone.0257521">Sequencing DNA with nanopores: Troubles and biases</a>. <em>PLOS ONE</em> <strong>16</strong>, e0257521 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline">Goodwin, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.191395.115">Oxford Nanopore sequencing, hybrid error correction, and de novo assembly of a eukaryotic genome</a>. <em>Genome Research</em> <strong>25</strong>, 1750–1756 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline">Dohm, J. C., Peters, P., Stralis-Pavese, N. &amp; Himmelbauer, H. <a href="https://doi.org/10.1093/nargab/lqaa037">Benchmarking of long-read correction methods</a>. <em>NAR Genomics and Bioinformatics</em> <strong>2</strong>, (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline">Weirather, J. L. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.10571.2">Comprehensive comparison of pacific biosciences and oxford nanopore technologies and their applications to transcriptome analysis</a>. <em>F1000Research</em> <strong>6</strong>, 100 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline">Foox, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-021-01049-5">Performance assessment of DNA sequencing platforms in the ABRF Next-Generation Sequencing Study</a>. <em>Nature Biotechnology</em> <strong>39</strong>, 1129–1140 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline">Huang, Y.-T., Liu, P.-Y. &amp; Shih, P.-W. <a href="https://doi.org/10.1186/s13059-021-02282-6">Homopolish: A method for the removal of systematic errors in nanopore sequencing by homologous polishing</a>. <em>Genome Biology</em> <strong>22</strong>, 95 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline">Rang, F. J., Kloosterman, W. P. &amp; Ridder, J. de. <a href="https://doi.org/10.1186/s13059-018-1462-9">From squiggle to basepair: Computational approaches for improving nanopore sequencing read accuracy</a>. <em>Genome Biology</em> <strong>19</strong>, 90 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline">Sarkozy, P., Jobbágy, Á. &amp; Antal, P. Calling Homopolymer Stretches from Raw Nanopore Reads by Analyzing k-mer Dwell Times. in (eds. Eskola, H., Väisänen, O., Viik, J. &amp; Hyttinen, J.) 241–244 (Springer, 2018). doi:<a href="https://doi.org/10.1007/978-981-10-5122-7_61">10.1007/978-981-10-5122-7_61</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline"><a href="http://nanoporetech.com/about-us/news/r103-newest-nanopore-high-accuracy-nanopore-sequencing-now-available-store">R10.3: the newest nanopore for high accuracy nanopore sequencing <span></span> now available in store</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline">Zhou, L. <em>et al.</em> <a href="https://doi.org/10.1116/1.5116295">Detection of DNA homopolymer with graphene nanopore</a>. <em>Journal of Vacuum Science &amp; Technology B</em> <strong>37</strong>, 061809 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline">Goto, Y., Yanagi, I., Matsui, K., Yokoi, T. &amp; Takeda, K. <a href="https://doi.org/10.1039/C8NR04238A">Identification of four single-stranded DNA homopolymers with a solid-state nanopore in alkaline CsCl solution</a>. <em>Nanoscale</em> <strong>10</strong>, 20844–20850 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline">Hawkins, J. A., Jones, S. K., Finkelstein, I. J. &amp; Press, W. H. <a href="https://doi.org/10.1073/pnas.1802640115">Indel-correcting DNA barcodes for high-throughput sequencing</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>115</strong>, E6217–E6226 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline">Srivathsan, A. <em>et al.</em> <a href="https://doi.org/10.1111/1755-0998.12890">A MinION<span></span>-based pipeline for fast and cost-effective DNA barcoding</a>. <em>Molecular Ecology Resources</em> <strong>18</strong>, 1035–1049 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">92. </div><div class="csl-right-inline">Wang, Y., Noor-A-Rahim, Md., Gunawan, E., Guan, Y. L. &amp; Poh, C. L. <a href="https://doi.org/10.1109/LCOMM.2019.2912572">Construction of bio-constrained code for DNA data storage</a>. <em>IEEE Communications Letters</em> <strong>23</strong>, 963–966 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">93. </div><div class="csl-right-inline">Au, K. F., Underwood, J. G., Lee, L. &amp; Wong, W. H. <a href="https://doi.org/10.1371/journal.pone.0046679">Improving <span>PacBio Long Read Accuracy</span> by <span>Short Read Alignment</span></a>. <em>PLOS ONE</em> <strong>7</strong>, e46679 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">94. </div><div class="csl-right-inline">Nurk, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.263566.120"><span>HiCanu</span>: Accurate assembly of segmental duplications, satellites, and allelic variants from high-fidelity long reads</a>. <em>Genome Res.</em> <strong>30</strong>, 1291–1305 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">95. </div><div class="csl-right-inline">Li, H. <a href="https://doi.org/10.1093/bioinformatics/bty191">Minimap2: Pairwise alignment for nucleotide sequences</a>. <em>Bioinformatics</em> <strong>34</strong>, 3094–3100 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">96. </div><div class="csl-right-inline">Shafin, K. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-020-0503-6">Nanopore sequencing and the Shasta toolkit enable efficient de novo assembly of eleven human genomes</a>. <em>Nature Biotechnology</em> <strong>38</strong>, 1044–1053 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">97. </div><div class="csl-right-inline">Sahlin, K. &amp; Medvedev, P. <a href="https://doi.org/10.1089/cmb.2019.0299">De <span>Novo Clustering</span> of <span>Long-Read Transcriptome Data Using</span> a <span>Greedy</span>, <span>Quality Value-Based Algorithm</span></a>. <em>Journal of Computational Biology</em> <strong>27</strong>, 472–484 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">98. </div><div class="csl-right-inline">Ekim, B., Berger, B. &amp; Chikhi, R. <a href="https://doi.org/10.1016/j.cels.2021.08.009">Minimizer-space de <span>Bruijn</span> graphs: Whole-genome assembly of long reads in minutes on a personal computer</a>. <em>Cell Systems</em> <strong>12</strong>, 958–968.e6 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">99. </div><div class="csl-right-inline">Miller, J. R. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btn548">Aggressive assembly of pyrosequencing reads with mates</a>. <em>Bioinformatics</em> <strong>24</strong>, 2818–2824 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">100. </div><div class="csl-right-inline">Sung, W.-K. <em>Algorithms in bioinformatics: A practical introduction</em>. (Chapman; Hall/CRC, 2011). doi:<a href="https://doi.org/10.1201/9781420070347">10.1201/9781420070347</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">101. </div><div class="csl-right-inline">Hamming, R. W. <em>Coding and Information Theory</em>. (Prentice-Hall, 1980).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">102. </div><div class="csl-right-inline">Gusfield, D. <em>Algorithms on strings, trees, and sequences: Computer science and computational biology</em>. (Cambridge University Press, 1997). doi:<a href="https://doi.org/10.1017/CBO9780511574931">10.1017/CBO9780511574931</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">103. </div><div class="csl-right-inline">Levenshtein, V. I. <a href="https://ui.adsabs.harvard.edu/abs/1966SPhD...10..707L">Binary codes capable of correcting deletions, insertions and reversals</a>. <em>Soviet Physics Doklady</em> <strong>10</strong>, 707 (1966).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">104. </div><div class="csl-right-inline">Hardison, R. C. <a href="https://doi.org/10.1371/journal.pbio.0000058">Comparative Genomics</a>. <em>PLOS Biology</em> <strong>1</strong>, e58 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">105. </div><div class="csl-right-inline">Felsenstein, J. <a href="https://doi.org/10.1007/BF01734359">Evolutionary trees from DNA sequences: A maximum likelihood approach</a>. <em>Journal of Molecular Evolution</em> <strong>17</strong>, 368–376 (1981).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">106. </div><div class="csl-right-inline">Kumar, S., Tamura, K. &amp; Nei, M. <a href="https://doi.org/10.1093/bioinformatics/10.2.189">MEGA: Molecular evolutionary genetics analysis software for microcomputers</a>. <em>Bioinformatics</em> <strong>10</strong>, 189–191 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">107. </div><div class="csl-right-inline">Kozlov, A. M., Darriba, D., Flouri, T., Morel, B. &amp; Stamatakis, A. <a href="https://doi.org/10.1093/bioinformatics/btz305">RAxML-NG: A fast, scalable and user-friendly tool for maximum likelihood phylogenetic inference</a>. <em>Bioinformatics</em> <strong>35</strong>, 4453–4455 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">108. </div><div class="csl-right-inline">Guindon, S. <em>et al.</em> <a href="https://doi.org/10.1093/sysbio/syq010">New algorithms and methods to estimate maximum-likelihood phylogenies: Assessing the performance of PhyML 3.0</a>. <em>Systematic Biology</em> <strong>59</strong>, 307–321 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">109. </div><div class="csl-right-inline">Price, M. N., Dehal, P. S. &amp; Arkin, A. P. <a href="https://doi.org/10.1371/journal.pone.0009490">FastTree 2 <span></span> Approximately Maximum-Likelihood Trees for Large Alignments</a>. <em>PLOS ONE</em> <strong>5</strong>, e9490 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">110. </div><div class="csl-right-inline">Jumper, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-021-03819-2">Highly accurate protein structure prediction with AlphaFold</a>. <em>Nature</em> <strong>596</strong>, 583–589 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">111. </div><div class="csl-right-inline">Karplus, K. <em>et al.</em> <a href="https://doi.org/10.1002/(SICI)1097-0134(1999)37:3+&lt;121::AID-PROT16&gt;3.0.CO;2-Q">Predicting protein structure using only sequence information</a>. <em>Proteins: Structure, Function, and Bioinformatics</em> <strong>37</strong>, 121–125 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">112. </div><div class="csl-right-inline">Watson, J. D., Laskowski, R. A. &amp; Thornton, J. M. <a href="https://doi.org/10.1016/j.sbi.2005.04.003">Predicting protein function from sequence and structural data</a>. <em>Current Opinion in Structural Biology</em> <strong>15</strong>, 275–284 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">113. </div><div class="csl-right-inline">Lee, D., Redfern, O. &amp; Orengo, C. <a href="https://doi.org/10.1038/nrm2281">Predicting protein function from sequence and structure</a>. <em>Nature Reviews Molecular Cell Biology</em> <strong>8</strong>, 995–1005 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">114. </div><div class="csl-right-inline">Salmela, L. &amp; Schröder, J. <a href="https://doi.org/10.1093/bioinformatics/btr170">Correcting errors in short reads by multiple alignments</a>. <em>Bioinformatics</em> <strong>27</strong>, 1455–1461 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">115. </div><div class="csl-right-inline">Medvedev, P., Stanciu, M. &amp; Brudno, M. <a href="https://doi.org/10.1038/nmeth.1374">Computational methods for discovering structural variation with next-generation sequencing</a>. <em>Nature Methods</em> <strong>6</strong>, S13–S20 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">116. </div><div class="csl-right-inline">Mahmoud, M. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-019-1828-7">Structural variant calling: the long and the short of it</a>. <em>Genome Biology</em> <strong>20</strong>, 246 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">117. </div><div class="csl-right-inline">Needleman, S. B. &amp; Wunsch, C. D. <a href="https://doi.org/10.1016/0022-2836(70)90057-4">A general method applicable to the search for similarities in the amino acid sequence of two proteins</a>. <em>Journal of Molecular Biology</em> <strong>48</strong>, 443–453 (1970).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">118. </div><div class="csl-right-inline">Smith, T. F. &amp; Waterman, M. S. <a href="https://doi.org/10.1016/0022-2836(81)90087-5">Identification of common molecular subsequences</a>. <em>Journal of Molecular Biology</em> <strong>147</strong>, 195–197 (1981).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">119. </div><div class="csl-right-inline">Bradley, S. P., Hax, A. C. &amp; Magnanti, T. L. <em>Applied Mathematical Programming</em>. (Addison-Wesley Publishing Company, 1977).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">120. </div><div class="csl-right-inline">Bellman, R. <a href="https://doi.org/10.1090/S0002-9904-1954-09848-8">The theory of dynamic programming</a>. <em>Bulletin of the American Mathematical Society</em> <strong>60</strong>, 503–515 (1954).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">121. </div><div class="csl-right-inline">Masek, W. J. &amp; Paterson, M. S. <a href="https://doi.org/10.1016/0022-0000(80)90002-1">A faster algorithm computing string edit distances</a>. <em>Journal of Computer and System Sciences</em> <strong>20</strong>, 18–31 (1980).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">122. </div><div class="csl-right-inline">Vinh, N. X., Epps, J. &amp; Bailey, J. Information <span>Theoretic Measures</span> for <span>Clusterings Comparison</span>: <span>Variants</span>, <span>Properties</span>, <span>Normalization</span> and <span>Correction</span> for <span>Chance</span>. <em>Journal of Machine Learning Research</em> <strong>11</strong>, 18 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">123. </div><div class="csl-right-inline">Ullman, J. D., Aho, A. V. &amp; Hirschberg, D. S. <a href="https://doi.org/10.1145/321921.321922">Bounds on the complexity of the longest common subsequence problem</a>. <em>Journal of the ACM</em> <strong>23</strong>, 112 (1976).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">124. </div><div class="csl-right-inline">Hirschberg, D. S. <a href="https://doi.org/10.1145/360825.360861">A linear space algorithm for computing maximal common subsequences</a>. <em>Communications of the ACM</em> <strong>18</strong>, 341343 (1975).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">125. </div><div class="csl-right-inline">Myers, E. W. &amp; Miller, W. <a href="https://doi.org/10.1093/bioinformatics/4.1.11">Optimal alignments in linear space</a>. <em>Bioinformatics</em> <strong>4</strong>, 11–17 (1988).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">126. </div><div class="csl-right-inline">Rice, P., Longden, I. &amp; Bleasby, A. EMBOSS: The european molecular biology open software suite. <em>Trends in genetics</em> <strong>16</strong>, 276277 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">127. </div><div class="csl-right-inline">Huang, X. &amp; Miller, W. <a href="https://doi.org/10.1016/0196-8858(91)90017-D">A time-efficient, linear-space local similarity algorithm</a>. <em>Advances in Applied Mathematics</em> <strong>12</strong>, 337–357 (1991).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">128. </div><div class="csl-right-inline">Waterman, M. S. &amp; Eggert, M. <a href="https://doi.org/10.1016/0022-2836(87)90478-5">A new algorithm for best subsequence alignments with application to tRNA-rRNA comparisons</a>. <em>Journal of Molecular Biology</em> <strong>197</strong>, 723–728 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">129. </div><div class="csl-right-inline">Stajich, J. E. <em>et al.</em> <a href="https://doi.org/10.1101/gr.361602">The Bioperl Toolkit: Perl Modules for the Life Sciences</a>. <em>Genome Research</em> <strong>12</strong>, 1611–1618 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">130. </div><div class="csl-right-inline">Gentleman, R. C. <em>et al.</em> <a href="https://doi.org/10.1186/gb-2004-5-10-r80">Bioconductor: open software development for computational biology and bioinformatics</a>. <em>Genome Biology</em> <strong>5</strong>, R80 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">131. </div><div class="csl-right-inline">Daily, J. <a href="https://doi.org/10.1186/s12859-016-0930-z">Parasail: SIMD c library for global, semi-global, and local pairwise sequence alignments</a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 81 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">132. </div><div class="csl-right-inline">Frohmberg, W., Kierzynka, M., Blazewicz, J. &amp; Wojciechowski, P. <a href="https://doi.org/10.2478/v10175-012-0062-1">G-PAS 2.0 <span></span> an improved version of protein alignment tool with an efficient backtracking routine on multiple GPUs</a>. <em>Bulletin of the Polish Academy of Sciences: Technical Sciences</em> <strong>60</strong>, 491–494 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">133. </div><div class="csl-right-inline">Altschul, S. F. Substitution Matrices. in (John Wiley &amp; Sons, Ltd, 2013). doi:<a href="https://doi.org/10.1002/9780470015902.a0005265.pub3">10.1002/9780470015902.a0005265.pub3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">134. </div><div class="csl-right-inline">Dayhoff, M. O., Schwartz, R. M. &amp; Orcutt, B. C. A model of evolutionary change in proteins. <em>A Model of Evolutionary Change in Proteins</em> 345–352 (1978).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">135. </div><div class="csl-right-inline">Müller, T. &amp; Vingron, M. <a href="https://doi.org/10.1089/10665270050514918">Modeling amino acid replacement</a>. <em>Journal of Computational Biology: A Journal of Computational Molecular Cell Biology</em> <strong>7</strong>, 761–776 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">136. </div><div class="csl-right-inline">Henikoff, S. &amp; Henikoff, J. G. <a href="https://doi.org/10.1073/pnas.89.22.10915">Amino acid substitution matrices from protein blocks</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>89</strong>, 10915–10919 (1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">137. </div><div class="csl-right-inline">Whelan, S. &amp; Goldman, N. <a href="https://doi.org/10.1093/oxfordjournals.molbev.a003851">A general empirical model of protein evolution derived from multiple protein families using a maximum-likelihood approach</a>. <em>Molecular Biology and Evolution</em> <strong>18</strong>, 691–699 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">138. </div><div class="csl-right-inline">Le, S. Q. &amp; Gascuel, O. <a href="https://doi.org/10.1093/molbev/msn067">An improved general amino acid replacement matrix</a>. <em>Molecular Biology and Evolution</em> <strong>25</strong>, 1307–1320 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">139. </div><div class="csl-right-inline">Müller, T., Rahmann, S. &amp; Rehmsmeier, M. <a href="https://doi.org/10.1093/bioinformatics/17.suppl_1.S182">Non-symmetric score matrices and the detection of homologous transmembrane proteins</a>. <em>Bioinformatics</em> <strong>17</strong>, S182–S189 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">140. </div><div class="csl-right-inline">Ng, P. C., Henikoff, J. G. &amp; Henikoff, S. <a href="https://doi.org/10.1093/bioinformatics/16.9.760">PHAT: a transmembrane-specific substitution matrix. Predicted hydrophobic and transmembrane</a>. <em>Bioinformatics (Oxford, England)</em> <strong>16</strong>, 760–766 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">141. </div><div class="csl-right-inline">Trivedi, R. &amp; Nagarajaram, H. A. <a href="https://doi.org/10.1038/s41598-019-52532-8">Amino acid substitution scoring matrices specific to intrinsically disordered regions in proteins</a>. <em>Scientific Reports</em> <strong>9</strong>, 16380 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">142. </div><div class="csl-right-inline">Goonesekere, N. C. W. &amp; Lee, B. <a href="https://doi.org/10.1002/prot.21775">Context-specific amino acid substitution matrices and their use in the detection of protein homologs</a>. <em>Proteins: Structure, Function, and Bioinformatics</em> <strong>71</strong>, 910–919 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">143. </div><div class="csl-right-inline">Paila, U., Kondam, R. &amp; Ranjan, A. <a href="https://doi.org/10.1093/nar/gkn635">Genome bias influences amino acid choices: analysis of amino acid substitution and re-compilation of substitution matrices exclusive to an AT-biased genome</a>. <em>Nucleic Acids Research</em> <strong>36</strong>, 6664–6675 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">144. </div><div class="csl-right-inline">Nickle, D. C. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0000503">HIV-specific probabilistic models of protein evolution</a>. <em>PLoS ONE</em> <strong>2</strong>, e503 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">145. </div><div class="csl-right-inline">Sardiu, M. E., Alves, G. &amp; Yu, Y.-K. <a href="https://doi.org/10.1103/PhysRevE.72.061917">Score statistics of global sequence alignment from the energy distribution of a modified directed polymer and directed percolation problem</a>. <em>Physical Review. E, Statistical, Nonlinear, and Soft Matter Physics</em> <strong>72</strong>, 061917 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">146. </div><div class="csl-right-inline">Chiaromonte, F., Yap, V. B. &amp; Miller, W. Scoring pairwise genomic sequence alignments. in 115–126 (WORLD SCIENTIFIC, 2001). doi:<a href="https://doi.org/10.1142/9789812799623_0012">10.1142/9789812799623_0012</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">147. </div><div class="csl-right-inline">Schneider, A., Cannarozzi, G. M. &amp; Gonnet, G. H. <a href="https://doi.org/10.1186/1471-2105-6-134">Empirical codon substitution matrix</a>. <em>BMC bioinformatics</em> <strong>6</strong>, 134 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">148. </div><div class="csl-right-inline">Doron-Faigenboim, A. &amp; Pupko, T. <a href="https://doi.org/10.1093/molbev/msl175">A combined empirical and mechanistic codon model</a>. <em>Molecular Biology and Evolution</em> <strong>24</strong>, 388–397 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">149. </div><div class="csl-right-inline">Gotoh, O. <a href="https://doi.org/10.1016/0022-2836(82)90398-9">An improved algorithm for matching biological sequences</a>. <em>Journal of Molecular Biology</em> <strong>162</strong>, 705–708 (1982).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">150. </div><div class="csl-right-inline">Benner, S. A., Cohen, M. A. &amp; Gonnet, G. H. <a href="https://doi.org/10.1006/jmbi.1993.1105">Empirical and Structural Models for Insertions and Deletions in the Divergent Evolution of Proteins</a>. <em>Journal of Molecular Biology</em> <strong>229</strong>, 1065–1082 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">151. </div><div class="csl-right-inline">Gu, X. &amp; Li, W.-H. <a href="https://doi.org/10.1007/BF00164032">The size distribution of insertions and deletions in human and rodent pseudogenes suggests the logarithmic gap penalty for sequence alignment</a>. <em>Journal of Molecular Evolution</em> <strong>40</strong>, 464–473 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">152. </div><div class="csl-right-inline">Waterman, M. S. <a href="https://doi.org/10.1016/S0022-5193(84)80037-5">Efficient sequence alignment algorithms</a>. <em>Journal of Theoretical Biology</em> <strong>108</strong>, 333–337 (1984).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">153. </div><div class="csl-right-inline">Pearson, W. R. &amp; Miller, W. <a href="https://doi.org/10.1016/0076-6879(92)10029-D">[27] Dynamic programming algorithms for biological sequence comparison</a>. in vol. 210 575–601 (Academic Press, 1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">154. </div><div class="csl-right-inline">Chao, J., Tang, F. &amp; Xu, L. <a href="https://doi.org/10.3390/biom12040546">Developments in Algorithms for Sequence Alignment: A Review</a>. <em>Biomolecules</em> <strong>12</strong>, 546 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">155. </div><div class="csl-right-inline">Spouge, J. L. <a href="https://doi.org/10.1137/0149094">Speeding up dynamic programming algorithms for finding optimal lattice paths</a>. <em>SIAM Journal on Applied Mathematics</em> <strong>49</strong>, 1552–1566 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">156. </div><div class="csl-right-inline">Fickett, J. W. <a href="https://doi.org/10.1093/nar/12.1Part1.175">Fast optimal alignment</a>. <em>Nucleic Acids Research</em> <strong>12</strong>, 175–179 (1984).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">157. </div><div class="csl-right-inline">Durbin, R., Eddy, S. R., Krogh, A. &amp; Mitchison, G. <em>Biological sequence analysis: Probabilistic models of proteins and nucleic acids</em>. (Cambridge University Press, 1998). doi:<a href="https://doi.org/10.1017/CBO9780511790492">10.1017/CBO9780511790492</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">158. </div><div class="csl-right-inline">Söding, J. <a href="https://doi.org/10.1093/bioinformatics/bti125">Protein homology detection by HMM-HMM comparison</a>. <em>Bioinformatics (Oxford, England)</em> <strong>21</strong>, 951–960 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">159. </div><div class="csl-right-inline">Finn, R. D., Clements, J. &amp; Eddy, S. R. <a href="https://doi.org/10.1093/nar/gkr367">HMMER web server: Interactive sequence similarity searching</a>. <em>Nucleic Acids Research</em> <strong>39</strong>, W29–W37 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">160. </div><div class="csl-right-inline">Wang, J., Keightley, P. D. &amp; Johnson, T. <a href="https://doi.org/10.1186/1471-2105-7-292">MCALIGN2: Faster, accurate global pairwise alignment of non-coding DNA sequences based on explicit models of indel evolution</a>. <em>BMC Bioinformatics</em> <strong>7</strong>, 292 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">161. </div><div class="csl-right-inline">Katoh, K., Misawa, K., Kuma, K. &amp; Miyata, T. <a href="https://doi.org/10.1093/nar/gkf436">MAFFT: A novel method for rapid multiple sequence alignment based on fast fourier transform</a>. <em>Nucleic Acids Research</em> <strong>30</strong>, 3059–3066 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">162. </div><div class="csl-right-inline">Sun, Y. &amp; Buhler, J. <a href="https://doi.org/10.1186/1471-2105-7-133">Choosing the best heuristic for seeded alignment of DNA sequences</a>. <em>BMC Bioinformatics</em> <strong>7</strong>, 133 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">163. </div><div class="csl-right-inline">Li, H. &amp; Homer, N. <a href="https://doi.org/10.1093/bib/bbq015">A survey of sequence alignment algorithms for next-generation sequencing</a>. <em>Briefings in Bioinformatics</em> <strong>11</strong>, 473–483 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">164. </div><div class="csl-right-inline">Altschul, S. F., Gish, W., Miller, W., Myers, E. W. &amp; Lipman, D. J. <a href="https://doi.org/10.1016/S0022-2836(05)80360-2">Basic local alignment search tool</a>. <em>Journal of Molecular Biology</em> <strong>215</strong>, 403–410 (1990).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">165. </div><div class="csl-right-inline">Edgar, R. C. <a href="https://doi.org/10.1093/bioinformatics/btq461">Search and clustering orders of magnitude faster than BLAST</a>. <em>Bioinformatics</em> <strong>26</strong>, 2460–2461 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">166. </div><div class="csl-right-inline">Shpaer, E. G. <em>et al.</em> <a href="https://doi.org/10.1006/geno.1996.0614">Sensitivity and Selectivity in Protein Similarity Searches: A Comparison of Smith<span></span>Waterman in Hardware to BLAST and FASTA</a>. <em>Genomics</em> <strong>38</strong>, 179–191 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">167. </div><div class="csl-right-inline">Pearson, W. R. &amp; Lipman, D. J. <a href="https://doi.org/10.1073/pnas.85.8.2444">Improved tools for biological sequence comparison</a>. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>85</strong>, 2444–2448 (1988).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">168. </div><div class="csl-right-inline">Lipman, D. J. &amp; Pearson, W. R. <a href="https://doi.org/10.1126/science.2983426">Rapid and sensitive protein similarity searches</a>. <em>Science (New York, N.Y.)</em> <strong>227</strong>, 1435–1441 (1985).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">169. </div><div class="csl-right-inline">Saripella, G. V., Sonnhammer, E. L. L. &amp; Forslund, K. <a href="https://doi.org/10.1093/bioinformatics/btw305">Benchmarking the next generation of homology inference tools</a>. <em>Bioinformatics</em> <strong>32</strong>, 2636 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">170. </div><div class="csl-right-inline">Finn, R. D. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkv1344">The Pfam protein families database: towards a more sustainable future</a>. <em>Nucleic Acids Research</em> <strong>44</strong>, D279 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">171. </div><div class="csl-right-inline">Essoussi, N. &amp; Fayech, S. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2255065/">A comparison of four pair-wise sequence alignment methods</a>. <em>Bioinformation</em> <strong>2</strong>, 166–168 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">172. </div><div class="csl-right-inline">Karlin, S. &amp; Altschul, S. F. <a href="https://doi.org/10.1073/pnas.87.6.2264">Methods for assessing the statistical significance of molecular sequence features by using general scoring schemes.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>87</strong>, 2264–2268 (1990).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">173. </div><div class="csl-right-inline">Mott, R. Alignment: Statistical Significance. in (John Wiley &amp; Sons, Ltd, 2005). doi:<a href="https://doi.org/10.1038/npg.els.0005264">10.1038/npg.els.0005264</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">174. </div><div class="csl-right-inline">Buchfink, B., Xie, C. &amp; Huson, D. H. <a href="https://doi.org/10.1038/nmeth.3176">Fast and sensitive protein alignment using DIAMOND</a>. <em>Nature Methods</em> <strong>12</strong>, 59–60 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">175. </div><div class="csl-right-inline">Buchfink, B., Reuter, K. &amp; Drost, H.-G. <a href="https://doi.org/10.1038/s41592-021-01101-x">Sensitive protein alignments at tree-of-life scale using DIAMOND</a>. <em>Nature Methods</em> <strong>18</strong>, 366–368 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">176. </div><div class="csl-right-inline">Bray, N., Dubchak, I. &amp; Pachter, L. <a href="https://doi.org/10.1101/gr.789803">AVID: A global alignment program</a>. <em>Genome Research</em> <strong>13</strong>, 97–102 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">177. </div><div class="csl-right-inline">Delcher, A. L., Phillippy, A., Carlton, J. &amp; Salzberg, S. L. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC117189/">Fast algorithms for large-scale genome alignment and comparison</a>. <em>Nucleic Acids Research</em> <strong>30</strong>, 2478–2483 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">178. </div><div class="csl-right-inline">Ferragina, P. &amp; Manzini, G. Proceedings 41st annual symposium on foundations of computer science. in 390–398 (2000). doi:<a href="https://doi.org/10.1109/SFCS.2000.892127">10.1109/SFCS.2000.892127</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">179. </div><div class="csl-right-inline">Burrows, M. &amp; Wheeler, D. <em>A block-sorting lossless data compression algorithm</em>. (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">180. </div><div class="csl-right-inline">Lam, T. W., Sung, W. K., Tam, S. L., Wong, C. K. &amp; Yiu, S. M. <a href="https://doi.org/10.1093/bioinformatics/btn032">Compressed indexing and local alignment of DNA</a>. <em>Bioinformatics</em> <strong>24</strong>, 791–797 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">181. </div><div class="csl-right-inline">Langmead, B. &amp; Salzberg, S. L. <a href="https://doi.org/10.1038/nmeth.1923">Fast gapped-read alignment with <span>Bowtie</span> 2</a>. <em>Nat Methods</em> <strong>9</strong>, 357–359 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">182. </div><div class="csl-right-inline">Li, H. &amp; Durbin, R. <a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with burrows<span></span>wheeler transform</a>. <em>Bioinformatics</em> <strong>25</strong>, 1754–1760 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">183. </div><div class="csl-right-inline">Li, H. &amp; Durbin, R. <a href="https://doi.org/10.1093/bioinformatics/btp698">Fast and accurate long-read alignment with burrows<span></span>wheeler transform</a>. <em>Bioinformatics</em> <strong>26</strong>, 589–595 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">184. </div><div class="csl-right-inline">Li, H. Aligning sequence reads, clone sequences and assembly contigs with <span>BWA</span>-<span>MEM</span>. <a href="https://arxiv.org/abs/1303.3997">https://arxiv.org/abs/1303.3997</a> (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">185. </div><div class="csl-right-inline">Liu, Y. &amp; Schmidt, B. <a href="https://doi.org/10.1093/bioinformatics/bts414">Long read alignment based on maximal exact match seeds</a>. <em>Bioinformatics</em> <strong>28</strong>, i318–i324 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">186. </div><div class="csl-right-inline"><em><a href="https://doi.org/10.1007/978-1-62703-646-7">Multiple Sequence Alignment Methods</a></em>. vol. 1079 (Humana Press, 2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">187. </div><div class="csl-right-inline">Wang, L. &amp; Jiang, T. <a href="https://doi.org/10.1089/cmb.1994.1.337">On the complexity of multiple sequence alignment</a>. <em>Journal of Computational Biology</em> <strong>1</strong>, 337–348 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">188. </div><div class="csl-right-inline">Just, W. <a href="https://doi.org/10.1089/106652701753307511">Computational complexity of multiple sequence alignment with SP-score</a>. <em>Journal of Computational Biology</em> <strong>8</strong>, 615–623 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">189. </div><div class="csl-right-inline">Feng, D.-F. &amp; Doolittle, R. F. <a href="https://doi.org/10.1007/BF02603120">Progressive sequence alignment as a prerequisitetto correct phylogenetic trees</a>. <em>Journal of Molecular Evolution</em> <strong>25</strong>, 351–360 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">190. </div><div class="csl-right-inline">Jones, D. T., Taylor, W. R. &amp; Thornton, J. M. <a href="https://doi.org/10.1093/bioinformatics/8.3.275">The rapid generation of mutation data matrices from protein sequences</a>. <em>Bioinformatics</em> <strong>8</strong>, 275–282 (1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">191. </div><div class="csl-right-inline">Blaisdell, B. E. <a href="https://doi.org/10.1073/pnas.83.14.5155">A measure of the similarity of sets of sequences not requiring sequence alignment.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>83</strong>, 5155–5159 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">192. </div><div class="csl-right-inline">Altschul, S. F., Carroll, R. J. &amp; Lipman, D. J. <a href="https://doi.org/10.1016/0022-2836(89)90234-9">Weights for data related by a tree</a>. <em>Journal of Molecular Biology</em> <strong>207</strong>, 647–653 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">193. </div><div class="csl-right-inline">Edgar, R. C. &amp; Sjölander, K. <a href="https://doi.org/10.1093/bioinformatics/bth090">A comparison of scoring functions for protein sequence profile alignment</a>. <em>Bioinformatics</em> <strong>20</strong>, 1301–1308 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">194. </div><div class="csl-right-inline">Notredame, C., Holm, L. &amp; Higgins, D. G. <a href="https://doi.org/10.1093/bioinformatics/14.5.407">COFFEE: An objective function for multiple sequence alignments.</a> <em>Bioinformatics</em> <strong>14</strong>, 407–422 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">195. </div><div class="csl-right-inline">Thompson, J. D., Higgins, D. G. &amp; Gibson, T. J. <a href="https://doi.org/10.1093/nar/22.22.4673">CLUSTAL W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice</a>. <em>Nucleic Acids Research</em> <strong>22</strong>, 4673–4680 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">196. </div><div class="csl-right-inline">Thompson, J. D., Gibson, T. J., Plewniak, F., Jeanmougin, F. &amp; Higgins, D. G. <a href="https://doi.org/10.1093/nar/25.24.4876">The CLUSTAL<span>_</span>x windows interface: Flexible strategies for multiple sequence alignment aided by quality analysis tools</a>. <em>Nucleic Acids Research</em> <strong>25</strong>, 4876–4882 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">197. </div><div class="csl-right-inline">Notredame, C., Higgins, D. G. &amp; Heringa, J. <a href="https://doi.org/10.1006/jmbi.2000.4042">T-coffee: a novel method for fast and accurate multiple sequence alignment11Edited by J. Thornton</a>. <em>Journal of Molecular Biology</em> <strong>302</strong>, 205–217 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">198. </div><div class="csl-right-inline">Edgar, R. C. <a href="https://doi.org/10.1186/1471-2105-5-113">MUSCLE: A multiple sequence alignment method with reduced time and space complexity</a>. <em>BMC Bioinformatics</em> <strong>5</strong>, 113 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">199. </div><div class="csl-right-inline">Edgar, R. C. <a href="https://doi.org/10.1093/nar/gkh340">MUSCLE: Multiple sequence alignment with high accuracy and high throughput</a>. <em>Nucleic Acids Research</em> <strong>32</strong>, 1792–1797 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">200. </div><div class="csl-right-inline">Do, C. B., Mahabhashyam, M. S. P., Brudno, M. &amp; Batzoglou, S. <a href="https://doi.org/10.1101/gr.2821705">ProbCons: Probabilistic consistency-based multiple sequence alignment</a>. <em>Genome Research</em> <strong>15</strong>, 330–340 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">201. </div><div class="csl-right-inline">Notredame, C. <a href="https://doi.org/10.1371/journal.pcbi.0030123">Recent Evolutions of Multiple Sequence Alignment Algorithms</a>. <em>PLOS Computational Biology</em> <strong>3</strong>, e123 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">202. </div><div class="csl-right-inline">Notredame, C. <a href="https://doi.org/10.1517/14622416.3.1.131">Recent progress in multiple sequence alignment: A survey</a>. <em>Pharmacogenomics</em> <strong>3</strong>, 131–144 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">203. </div><div class="csl-right-inline">Edgar, R. C. &amp; Batzoglou, S. <a href="https://doi.org/10.1016/j.sbi.2006.04.004">Multiple sequence alignment</a>. <em>Current Opinion in Structural Biology</em> <strong>16</strong>, 368–373 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">204. </div><div class="csl-right-inline">Pais, F. S.-M., Ruy, P. de C., Oliveira, G. &amp; Coimbra, R. S. <a href="https://doi.org/10.1186/1748-7188-9-4">Assessing the efficiency of multiple sequence alignment programs</a>. <em>Algorithms for Molecular Biology</em> <strong>9</strong>, 4 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">205. </div><div class="csl-right-inline">Thompson, J. D., Plewniak, F. &amp; Poch, O. <a href="https://doi.org/10.1093/bioinformatics/15.1.87">BAliBASE: a benchmark alignment database for the evaluation of multiple alignment programs.</a> <em>Bioinformatics</em> <strong>15</strong>, 87–88 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">206. </div><div class="csl-right-inline">Eddy, S. R. Multiple Alignment Using Hidden Markov Models. 7.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">207. </div><div class="csl-right-inline">Lemoine, F., Blassel, L., Voznica, J. &amp; Gascuel, O. COVID-align: Accurate online alignment of hCoV-19 genomes using a profile HMM. <em>Bioinformatics</em> (2020) doi:<a href="https://doi.org/10.1093/bioinformatics/btaa871">10.1093/bioinformatics/btaa871</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">208. </div><div class="csl-right-inline">Aksamentov, I., Roemer, C., Hodcroft, E. &amp; Neher, R. <a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes</a>. <em>Journal of Open Source Software</em> <strong>6</strong>, 3773 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">209. </div><div class="csl-right-inline">Kim, J., Pramanik, S. &amp; Chung, M. J. <a href="https://doi.org/10.1093/bioinformatics/10.4.419">Multiple sequence alignment using simulated annealing</a>. <em>Bioinformatics</em> <strong>10</strong>, 419–426 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">210. </div><div class="csl-right-inline">Ishikawa, M. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/9.3.267">Multiple sequence alignment by parallel simulated annealing</a>. <em>Bioinformatics</em> <strong>9</strong>, 267–273 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">211. </div><div class="csl-right-inline">Chowdhury, B. &amp; Garai, G. <a href="https://doi.org/10.1016/j.ygeno.2017.06.007">A review on multiple sequence alignment from the perspective of genetic algorithm</a>. <em>Genomics</em> <strong>109</strong>, 419–431 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">212. </div><div class="csl-right-inline">Notredame, C. &amp; Higgins, D. G. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC145823/">SAGA: Sequence alignment by genetic algorithm.</a> <em>Nucleic Acids Research</em> <strong>24</strong>, 1515–1524 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">213. </div><div class="csl-right-inline">Garriga, E. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-019-0333-6">Large multiple sequence alignments with a root-to-leaf regressive method</a>. <em>Nature Biotechnology</em> <strong>37</strong>, 1466–1470 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">214. </div><div class="csl-right-inline">Alser, M. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-021-02443-7">Technology dictates algorithms: Recent developments in read alignment</a>. <em>Genome Biology</em> <strong>22</strong>, 249 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">215. </div><div class="csl-right-inline">Schbath, S. <em>et al.</em> <a href="https://doi.org/10.1089/cmb.2012.0022">Mapping reads on a genomic sequence: An algorithmic overview and a practical comparative analysis</a>. <em>Journal of Computational Biology</em> <strong>19</strong>, 796–813 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">216. </div><div class="csl-right-inline">Hatem, A., Bozdağ, D., Toland, A. E. &amp; Çatalyürek, Ü. V. <a href="https://doi.org/10.1186/1471-2105-14-184">Benchmarking short sequence mapping tools</a>. <em>BMC Bioinformatics</em> <strong>14</strong>, 184 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">217. </div><div class="csl-right-inline">Canzar, S. &amp; Salzberg, S. L. <a href="https://doi.org/10.1109/JPROC.2015.2455551">Short read mapping: An algorithmic tour</a>. <em>Proceedings of the IEEE</em> <strong>105</strong>, 436–458 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">218. </div><div class="csl-right-inline">Olson, C. B. <em>et al.</em> 2012 IEEE 20th international symposium on field-programmable custom computing machines. in 161–168 (2012). doi:<a href="https://doi.org/10.1109/FCCM.2012.36">10.1109/FCCM.2012.36</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">219. </div><div class="csl-right-inline">Jain, C. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btaa435">Weighted minimizer sampling improves long read mapping</a>. <em>Bioinformatics</em> <strong>36</strong>, i111–i118 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">220. </div><div class="csl-right-inline">Jain, C., Rhie, A., Hansen, N. F., Koren, S. &amp; Phillippy, A. M. <a href="https://doi.org/10.1038/s41592-022-01457-8">Long-read mapping to repetitive reference sequences using Winnowmap2</a>. <em>Nature Methods</em> <strong>19</strong>, 705–710 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">221. </div><div class="csl-right-inline">Mikheenko, A., Bzikadze, A. V., Gurevich, A., Miga, K. H. &amp; Pevzner, P. A. <a href="https://doi.org/10.1093/bioinformatics/btaa440"><span>TandemTools</span>: Mapping long reads and assessing/improving assembly quality in extra-long tandem repeats</a>. <em>Bioinformatics</em> <strong>36</strong>, i75–i83 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">222. </div><div class="csl-right-inline">Langmead, B. <a href="https://doi.org/10.1186/s13059-017-1290-3">A tandem simulation framework for predicting mapping quality</a>. <em>Genome Biology</em> <strong>18</strong>, 152 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">223. </div><div class="csl-right-inline">Břinda, K., Boeva, V. &amp; Kucherov, G. <a href="https://doi.org/10.1093/bioinformatics/btv524">RNF: A general framework to evaluate NGS read mappers</a>. <em>Bioinformatics</em> <strong>32</strong>, 136–139 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">224. </div><div class="csl-right-inline">Bragg, L., Stone, G., Imelfort, M., Hugenholtz, P. &amp; Tyson, G. W. <a href="https://doi.org/10.1038/nmeth.1990">Fast, accurate error-correction of amplicon pyrosequences using <span>Acacia</span></a>. <em>Nat Methods</em> <strong>9</strong>, 425–426 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">225. </div><div class="csl-right-inline">Sahlin, K. &amp; Medvedev, P. <a href="https://doi.org/10.1038/s41467-020-20340-8">Error correction enables use of <span>Oxford Nanopore</span> technology for reference-free transcriptome analysis</a>. <em>Nat Commun</em> <strong>12</strong>, 2 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">226. </div><div class="csl-right-inline">Liu, H. <em>et al.</em> <a href="https://doi.org/10.46471/gigabyte.15"><span>SMARTdenovo</span>: A de novo assembler using long noisy reads</a>. <em>Gigabyte</em> <strong>2021</strong>, 1–9 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">227. </div><div class="csl-right-inline">Graham, R. L., Knuth, D. E. &amp; Patashnik, O. <em>Concrete mathematics: A foundation for computer science</em>. (<span>Addison-Wesley</span>, 1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">228. </div><div class="csl-right-inline">Adams, M. D. <em>et al.</em> <a href="https://doi.org/10.1126/science.287.5461.2185">The genome sequence of <span>Drosophila</span> melanogaster</a>. <em>Science</em> <strong>287</strong>, 2185–2195 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">229. </div><div class="csl-right-inline">Yang, C., Chu, J., Warren, R. L. &amp; Birol, I. <a href="https://doi.org/10.1093/gigascience/gix010"><span>NanoSim</span>: Nanopore sequence read simulator based on statistical characterization</a>. <em>GigaScience</em> <strong>6</strong>, (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">230. </div><div class="csl-right-inline">Rhie, A., Walenz, B. P., Koren, S. &amp; Phillippy, A. M. <a href="https://doi.org/10.1186/s13059-020-02134-9">Merqury: Reference-free quality, completeness, and phasing assessment for genome assemblies</a>. <em>Genome Biology</em> <strong>21</strong>, 245 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">231. </div><div class="csl-right-inline">Prodanov, T. &amp; Bansal, V. <a href="https://doi.org/10.1093/nar/gkaa829">Sensitive alignment using paralogous sequence variants improves long-read mapping and variant calling in segmental duplications</a>. <em>Nucleic Acids Research</em> <strong>48</strong>, e114 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">232. </div><div class="csl-right-inline">Li, H. New strategies to improve Minimap2 alignment accuracy. <a href="https://arxiv.org/abs/2108.03515">https://arxiv.org/abs/2108.03515</a> (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">233. </div><div class="csl-right-inline">Li, H. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-018-0054-7">A synthetic-diploid benchmark for accurate variant-calling evaluation</a>. <em>Nat Methods</em> <strong>15</strong>, 595–597 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">234. </div><div class="csl-right-inline">Lepri, A. C. <em>et al.</em> <a href="https://doi.org/10.1086/315301">Resistance <span>Profiles</span> in <span>Patients</span> with <span>Viral Rebound</span> on <span>Potent Antiretroviral Therapy</span></a>. <em>The Journal of Infectious Diseases</em> <strong>181</strong>, 1143–1147 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">235. </div><div class="csl-right-inline">Verhofstede, C. <em>et al.</em> <a href="https://doi.org/10.1002/jmv.20950">Detection of drug resistance mutations as a predictor of subsequent virological failure in patients with <span>HIV</span>-1 viral rebounds of less than 1,000 <span>RNA</span> copies/ml</a>. <em>Journal of Medical Virology</em> <strong>79</strong>, 1254–1260 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">236. </div><div class="csl-right-inline">Hué, S. <em>et al.</em> <a href="https://doi.org/10.1128/JVI.01556-08">Demonstration of <span>Sustained Drug</span>-<span>Resistant Human Immunodeficiency Virus Type</span> 1 <span>Lineages Circulating</span> among <span>Treatment</span>-<span>Naïve Individuals</span></a>. <em>Journal of Virology</em> <strong>83</strong>, 2645–2654 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">237. </div><div class="csl-right-inline">Mourad, R. <em>et al.</em> <a href="https://doi.org/10.1097/QAD.0000000000000768">A phylotype-based analysis highlights the role of drug-naive <span>HIV</span>-positive individuals in the transmission of antiretroviral resistance in the <span>UK</span></a>. <em>Aids</em> <strong>29</strong>, 1917–1925 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">238. </div><div class="csl-right-inline">Zhukova, A., Cutino-Moguel, T., Gascuel, O. &amp; Pillay, D. <a href="https://doi.org/10.1093/infdis/jix411">The <span>Role</span> of <span>Phylogenetics</span> as a <span>Tool</span> to <span>Predict</span> the <span>Spread</span> of <span>Resistance</span></a>. <em>The Journal of Infectious Diseases</em> <strong>216</strong>, S820–S823 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">239. </div><div class="csl-right-inline">Bennett, D. E. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0004724">Drug <span>Resistance Mutations</span> for <span>Surveillance</span> of <span>Transmitted HIV</span>-1 <span>Drug</span>-<span>Resistance</span>: 2009 <span>Update</span></a>. <em>PLOS ONE</em> <strong>4</strong>, e4724 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">240. </div><div class="csl-right-inline">Hammond, J., Calef, C., Larder, B., Schinazi, R. &amp; Mellors, J. W. Mutations in <span>Retroviral Genes Associated</span> with <span>Drug Resistance</span>. <em>Human retroviruses and AIDS</em> 11136–11179 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">241. </div><div class="csl-right-inline">Wensing, A. M. <em>et al.</em> <a href="https://www.ncbi.nlm.nih.gov/pubmed/28208121">2017 <span>Update</span> of the <span>Drug Resistance Mutations</span> in <span>HIV</span>-1., 2017 <span>Update</span> of the <span>Drug Resistance Mutations</span> in <span>HIV</span>-1</a>. <em>Topics in antiviral medicine, Topics in Antiviral Medicine</em> <strong>24, 24</strong>, 132, 132–133 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">242. </div><div class="csl-right-inline">Dudoit, S. &amp; Laan, M. J. van der. <em>Multiple <span>Testing Procedures</span> with <span>Applications</span> to <span>Genomics</span></em>. (<span>Springer Science &amp; Business Media</span>, 2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">243. </div><div class="csl-right-inline">Villabona-Arenas, C. J. <em>et al.</em> <a href="https://doi.org/10.1097/QAD.0000000000001233">In-depth analysis of <span>HIV</span>-1 drug resistance mutations in <span>HIV</span>-infected individuals failing first-line regimens in <span>West</span> and <span>Central Africa</span></a>. <em>AIDS</em> <strong>30</strong>, 2577 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">244. </div><div class="csl-right-inline">Maddison, W. P. &amp; FitzJohn, R. G. <a href="https://doi.org/10.1093/sysbio/syu070">The <span>Unsolved Challenge</span> to <span>Phylogenetic Correlation Tests</span> for <span>Categorical Characters</span></a>. <em>Systematic Biology</em> <strong>64</strong>, 127–136 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">245. </div><div class="csl-right-inline">Sham, P. C. &amp; Purcell, S. M. <a href="https://doi.org/10.1038/nrg3706">Statistical power and significance testing in large-scale genetic studies</a>. <em>Nature Reviews Genetics</em> <strong>15</strong>, 335–346 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">246. </div><div class="csl-right-inline">Lengauer, T. &amp; Sing, T. <a href="https://doi.org/10.1038/nrmicro1477">Bioinformatics-assisted anti-<span>HIV</span> therapy</a>. <em>Nature Reviews Microbiology</em> <strong>4</strong>, 790–797 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">247. </div><div class="csl-right-inline">Zhang, J., Rhee, S.-Y., Taylor, J. &amp; Shafer, R. W. <a href="https://doi.org/10.1097/01.qai.0000147526.64863.53">Comparison of the <span>Precision</span> and <span>Sensitivity</span> of the <span>Antivirogram</span> and <span>PhenoSense HIV Drug Susceptibility Assays</span></a>. <em>JAIDS Journal of Acquired Immune Deficiency Syndromes</em> <strong>38</strong>, 439–444 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">248. </div><div class="csl-right-inline">Beerenwinkel, N. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkg575">Geno2pheno: Estimating phenotypic drug resistance from <span>HIV</span>-1 genotypes</a>. <em>Nucleic Acids Research</em> <strong>31</strong>, 3850–3855 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">249. </div><div class="csl-right-inline">Shen, C., Yu, X., Harrison, R. W. &amp; Weber, I. T. <a href="https://doi.org/10.1186/s12859-016-1114-6">Automated prediction of <span>HIV</span> drug resistance from genotype data</a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 278 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">250. </div><div class="csl-right-inline">Yu, X., Weber, I. T. &amp; Harrison, R. W. <a href="https://doi.org/10.1186/1471-2164-15-S5-S1">Prediction of <span>HIV</span> drug resistance from genotype with encoded three-dimensional protein structure</a>. <em>BMC Genomics</em> <strong>15</strong>, S1 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">251. </div><div class="csl-right-inline">Sheik Amamuddy, O., Bishop, N. T. &amp; Tastan Bishop, Ö. <a href="https://doi.org/10.1186/s12859-017-1782-x">Improving fold resistance prediction of <span>HIV</span>-1 against protease and reverse transcriptase inhibitors using artificial neural networks</a>. <em>BMC Bioinformatics</em> <strong>18</strong>, 369 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">252. </div><div class="csl-right-inline">Beerenwinkel, N. <em>et al.</em> <a href="https://doi.org/10.1109/5254.972080">Geno2pheno: Interpreting genotypic <span>HIV</span> drug resistance tests</a>. <em>IEEE Intelligent Systems</em> <strong>16</strong>, 35–41 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">253. </div><div class="csl-right-inline">Araya, S. T. &amp; Hazelhurst, S. <a href="https://doi.org/10.1080/00359190909519238">Support vector machine prediction of <span>HIV</span>-1 drug resistance using the viral nucleotide patterns</a>. <em>Transactions of the Royal Society of South Africa</em> <strong>64</strong>, 62–72 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">254. </div><div class="csl-right-inline">Riemenschneider, M., Senge, R., Neumann, U., Hüllermeier, E. &amp; Heider, D. <a href="https://doi.org/10.1186/s13040-016-0089-1">Exploiting <span>HIV</span>-1 protease and reverse transcriptase cross-resistance information for improved drug resistance prediction by means of multi-label classification</a>. <em>BioData Mining</em> <strong>9</strong>, 10 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">255. </div><div class="csl-right-inline">Heider, D., Senge, R., Cheng, W. &amp; Hüllermeier, E. <a href="https://doi.org/10.1093/bioinformatics/btt331">Multilabel classification for exploiting cross-resistance information in <span>HIV</span>-1 drug resistance prediction</a>. <em>Bioinformatics</em> <strong>29</strong>, 1946–1952 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">256. </div><div class="csl-right-inline">Drăghici, S. &amp; Potter, R. B. <a href="https://doi.org/10.1093/bioinformatics/19.1.98">Predicting <span>HIV</span> drug resistance with neural networks</a>. <em>Bioinformatics</em> <strong>19</strong>, 98–107 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">257. </div><div class="csl-right-inline">Steiner, M. C., Gibson, K. M. &amp; Crandall, K. A. <a href="https://doi.org/10.3390/v12050560">Drug <span>Resistance Prediction Using Deep Learning Techniques</span> on <span>HIV</span>-1 <span>Sequence Data</span></a>. <em>Viruses</em> <strong>12</strong>, 560 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">258. </div><div class="csl-right-inline">Mooney, A. C. <em>et al.</em> <a href="https://doi.org/10.1007/s10461-018-2155-9">Beyond <span>Social Desirability Bias</span>: <span>Investigating Inconsistencies</span> in <span>Self</span>-<span>Reported HIV Testing</span> and <span>Treatment Behaviors Among HIV</span>-<span>Positive Adults</span> in <span>North West Province</span>, <span>South Africa</span></a>. <em>AIDS and Behavior</em> <strong>22</strong>, 2368–2379 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">259. </div><div class="csl-right-inline">Tibshirani, R. <a href="https://doi.org/10.1111/j.2517-6161.1996.tb02080.x">Regression <span>Shrinkage</span> and <span>Selection Via</span> the <span>Lasso</span></a>. <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> <strong>58</strong>, 267–288 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">260. </div><div class="csl-right-inline">Brier, G. W. Verification of <span>Forecasts</span> <span>Expressed</span> in <span>Terms</span> of <span>Probability</span>. <em>Monthly Weather Review</em> <strong>78</strong>, 1–3 (1950).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">261. </div><div class="csl-right-inline">Gascuel, O. <em>et al.</em> <a href="https://doi.org/10.1142/S0218001498000336">Twelve <span>Numerical</span>, <span>Symbolic</span> and <span>Hybrid Supervised Classification Methods</span></a>. <em>International Journal of Pattern Recognition and Artificial Intelligence</em> <strong>12</strong>, 517–571 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">262. </div><div class="csl-right-inline">Goeman, J. J. &amp; Solari, A. <a href="https://doi.org/10.1002/sim.6082">Multiple hypothesis testing in genomics</a>. <em>Statistics in Medicine</em> <strong>33</strong>, 1946–1978 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">263. </div><div class="csl-right-inline">Rennie, J. D., Shih, L., Teevan, J. &amp; Karger, D. R. Tackling the <span>Poor Assumptions</span> of <span>Naive Bayes Text Classifiers</span>. in <em>Proceedings of the 20th international conference on machine learning (<span>ICML</span>-03)</em> 616–623 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">264. </div><div class="csl-right-inline">Breiman, L. <a href="https://doi.org/10.1023/A:1010933404324">Random <span>Forests</span></a>. <em>Machine Learning</em> <strong>45</strong>, 5–32 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">265. </div><div class="csl-right-inline">Alvarez Melis, D. &amp; Jaakkola, T. Towards <span>Robust Interpretability</span> with <span>Self</span>-<span>Explaining Neural Networks</span>. in <em>Advances in <span>Neural Information Processing Systems</span> 31</em> (eds. Bengio, S. et al.) 7775–7784 (<span>Curran Associates, Inc.</span>, 2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">266. </div><div class="csl-right-inline">Hastie, T., Tibshirani, R. &amp; Friedman, J. <em>The <span>Elements</span> of <span>Statistical Learning</span>: <span>Data Mining</span>, <span>Inference</span>, and <span>Prediction</span>, <span>Second Edition</span></em>. (<span>Springer Science &amp; Business Media</span>, 2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">267. </div><div class="csl-right-inline">Zhang, Q., Wu, Y. N. &amp; Zhu, S.-C. Interpretable <span>Convolutional Neural Networks</span>. in <em>Proceedings of the <span>IEEE Conference</span> on <span>Computer Vision</span> and <span>Pattern Recognition</span></em> 8827–8836 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">268. </div><div class="csl-right-inline">Brodersen, K. H., Ong, C. S., Stephan, K. E. &amp; Buhmann, J. M. The <span>Balanced Accuracy</span> and <span>Its Posterior Distribution</span>. in <em>2010 20th <span>International Conference</span> on <span>Pattern Recognition</span></em> 3121–3124 (2010). doi:<a href="https://doi.org/10.1109/ICPR.2010.764">10.1109/ICPR.2010.764</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">269. </div><div class="csl-right-inline">Schrödinger, LLC. The <span>PyMOL</span> molecular graphics system, version 1.8. (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">270. </div><div class="csl-right-inline">Sarafianos, S. G. <em>et al.</em> <a href="https://doi.org/10.1016/j.jmb.2008.10.071">Structure and function of <span>HIV</span>-1 reverse transcriptase: Molecular mechanisms of polymerization and inhibition</a>. <em>Journal of molecular biology</em> <strong>385</strong>, 693–713 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">271. </div><div class="csl-right-inline">Rhee, S.-Y., Liu, T. F., Holmes, S. P. &amp; Shafer, R. W. <a href="https://doi.org/10.1371/journal.pcbi.0030087"><span>HIV</span>-1 <span>Subtype B Protease</span> and <span>Reverse Transcriptase Amino Acid Covariation</span></a>. <em>PLOS Computational Biology</em> <strong>3</strong>, e87 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">272. </div><div class="csl-right-inline">De Luca, A. <em>et al.</em> <a href="https://doi.org/10.1086/522231">Improved <span>Interpretation</span> of <span>Genotypic Changes</span> in the <span>HIV</span>-1 <span>Reverse Transcriptase Coding Region That Determine</span> the <span>Virological Response</span> to <span>Didanosine</span></a>. <em>The Journal of Infectious Diseases</em> <strong>196</strong>, 1645–1653 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">273. </div><div class="csl-right-inline">Marcelin, A.-G. <em>et al.</em> Impact of <span>HIV</span>-1 reverse transcriptase polymorphism at codons 211 and 228 on virological response to didanosine. <em>Antiviral Therapy</em> 8 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">274. </div><div class="csl-right-inline">Brown, A. J. L. <em>et al.</em> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC110900">Reduced <span>Susceptibility</span> of <span>Human Immunodeficiency Virus Type</span> 1 (<span>HIV</span>-1) from <span>Patients</span> with <span>Primary HIV Infection</span> to <span>Nonnucleoside Reverse Transcriptase Inhibitors</span> <span>Is Associated</span> with <span>Variation</span> at <span>Novel Amino Acid Sites</span></a>. <em>Journal of Virology</em> <strong>74</strong>, 10269–10273 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">275. </div><div class="csl-right-inline">Clark, S. A., Shulman, N. S., Bosch, R. J. &amp; Mellors, J. W. <a href="https://doi.org/10.1097/01.aids.0000222069.14878.44">Reverse transcriptase mutations <span>118I</span>, <span>208Y</span>, and <span>215Y</span> cause <span>HIV</span>-1 hypersusceptibility to non-nucleoside reverse transcriptase inhibitors</a>. <em>AIDS</em> <strong>20</strong>, 981–984 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">276. </div><div class="csl-right-inline">Nebbia, G., Sabin, C. A., Dunn, D. T. &amp; Geretti, A. M. <a href="https://doi.org/10.1093/jac/dkm067">Emergence of the <span>H208Y</span> mutation in the reverse transcriptase (<span>RT</span>) of <span>HIV</span>-1 in association with nucleoside <span>RT</span> inhibitor therapy</a>. <em>Journal of Antimicrobial Chemotherapy</em> <strong>59</strong>, 1013–1016 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">277. </div><div class="csl-right-inline">Saracino, A. <em>et al.</em> <a href="https://doi.org/10.1002/jmv.20500">Impact of unreported <span>HIV</span>-1 reverse transcriptase mutations on phenotypic resistance to nucleoside and non-nucleoside inhibitors</a>. <em>Journal of Medical Virology</em> <strong>78</strong>, 9–17 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">278. </div><div class="csl-right-inline">Wu, T. T., Chen, Y. F., Hastie, T., Sobel, E. &amp; Lange, K. <a href="https://doi.org/10.1093/bioinformatics/btp041">Genome-wide association analysis by lasso penalized logistic regression</a>. <em>Bioinformatics</em> <strong>25</strong>, 714–721 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">279. </div><div class="csl-right-inline">Bzikadze, A. V. &amp; Pevzner, P. A. <a href="https://doi.org/10.1038/s41587-020-0582-4">Automated assembly of centromeres from ultra-long error-prone reads</a>. <em>Nat Biotechnol</em> <strong>38</strong>, 1309–1316 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">280. </div><div class="csl-right-inline">Castresana, J. <a href="https://doi.org/10.1093/oxfordjournals.molbev.a026334">Selection of <span>Conserved Blocks</span> from <span>Multiple Alignments</span> for <span>Their Use</span> in <span>Phylogenetic Analysis</span></a>. <em>Molecular Biology and Evolution</em> <strong>17</strong>, 540–552 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">281. </div><div class="csl-right-inline">McGinnis, W. <em>et al.</em> Scikit-<span>Learn</span>-<span>Contrib</span>/<span>Categorical</span>-<span>Encoding</span>: <span>Release For Zenodo</span>. (2018) doi:<a href="https://doi.org/10.5281/ZENODO.1157110">10.5281/ZENODO.1157110</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">282. </div><div class="csl-right-inline">Pedregosa, F. <em>et al.</em> Scikit-learn: <span>Machine Learning</span> in <span>Python</span>. <em>Journal of Machine Learning Research</em> <strong>12</strong>, 2825–2830 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">283. </div><div class="csl-right-inline">Virtanen, P. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-019-0686-2"><span>SciPy</span> 1.0: Fundamental algorithms for scientific computing in <span>Python</span></a>. <em>Nature Methods</em> <strong>17</strong>, 261–272 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">284. </div><div class="csl-right-inline">Seabold, Skipper &amp; Perktold, Josef. <span>S</span>tatsmodels: <span>E</span>conometric and <span>S</span>tatistical <span>M</span>odeling with <span>P</span>ython. in <em><span>P</span>roceedings of the 9th <span>P</span>ython in <span>S</span>cience <span>C</span>onference</em> (eds. Walt, Stéfan van der &amp; Millman, Jarrod) 92–96 (2010). doi:<a href="https://doi.org/ 10.25080/Majora-92bf1922-011 ">10.25080/Majora-92bf1922-011 </a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">285. </div><div class="csl-right-inline">Vinh, N. X. &amp; Epps, J. A <span>Novel Approach</span> for <span>Automatic Number</span> of <span>Clusters Detection</span> in <span>Microarray Data Based</span> on <span>Consensus Clustering</span>. in <em>2009 <span>Ninth IEEE International Conference</span> on <span>Bioinformatics</span> and <span>BioEngineering</span></em> 84–91 (2009). doi:<a href="https://doi.org/10.1109/BIBE.2009.19">10.1109/BIBE.2009.19</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">286. </div><div class="csl-right-inline">Harremoes, P. Mutual information of contingency tables and related inequalities. in <em>2014 <span>IEEE International Symposium</span> on <span>Information Theory</span></em> 2474–2478 (<span>IEEE</span>, 2014). doi:<a href="https://doi.org/10.1109/ISIT.2014.6875279">10.1109/ISIT.2014.6875279</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">287. </div><div class="csl-right-inline">Kawashima, S. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkm998">AAindex: amino acid index database, progress report 2008</a>. <em>Nucleic Acids Research</em> <strong>36</strong>, D202–D205 (2008).</div>
</div>
</div>
</div>



</div>
            </section>

          </div>
        </div>
      </div>
<a href="global-conclusion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="HPC-appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/91-HPC-Appendix.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
