<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Global References | From sequences to knowledge, improving and learning from sequence alignments</title>
  <meta name="description" content="Global References | From sequences to knowledge, improving and learning from sequence alignments" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Global References | From sequences to knowledge, improving and learning from sequence alignments" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Global References | From sequences to knowledge, improving and learning from sequence alignments" />
  
  
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-09-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="global-conclusion.html"/>
<link rel="next" href="HPC-appendix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />
<link href="libs/tabwid-1.0.0/scrool.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">From sequences to knowledge,</br> improving and learning from sequence alignments.</a></li>

<li class="divider"></li>
<li><a href="index.html#section"></a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#résumé"><i class="fa fa-check"></i>Résumé</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="list-of-acronyms-and-abbreviations.html"><a href="list-of-acronyms-and-abbreviations.html"><i class="fa fa-check"></i>List of Acronyms and Abbreviations</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html"><i class="fa fa-check"></i>General Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#research-output"><i class="fa fa-check"></i>Research output</a>
<ul>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#journal-publications"><i class="fa fa-check"></i>Journal publications</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#presentations-and-posters"><i class="fa fa-check"></i>Presentations and posters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>1</b> What is Sequence data ?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>1.1</b> Biological sequences, a primer</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>1.1.1</b> What is DNA ?</a></li>
<li class="chapter" data-level="1.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#from-information-to-action"><i class="fa fa-check"></i><b>1.1.2</b> From Information to action</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining sequence data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough"><i class="fa fa-check"></i><b>1.2.1</b> Sanger sequencing, a breakthrough</a></li>
<li class="chapter" data-level="1.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> Next-generation sequencing</a></li>
<li class="chapter" data-level="1.2.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#long-read-sequencing"><i class="fa fa-check"></i><b>1.2.3</b> Long read sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>1.3</b> Sequencing errors, how to account for them ?</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#error-correction-methods"><i class="fa fa-check"></i><b>1.3.1</b> Error correction methods</a></li>
<li class="chapter" data-level="1.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#more-accurate-sequencing-methods"><i class="fa fa-check"></i><b>1.3.2</b> More accurate sequencing methods</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#the-special-case-of-homopolymers"><i class="fa fa-check"></i><b>1.4</b> The special case of homopolymers</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-the-human-genome"><i class="fa fa-check"></i><b>1.4.1</b> Homopolymers and the human genome</a></li>
<li class="chapter" data-level="1.4.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-long-reads"><i class="fa fa-check"></i><b>1.4.2</b> Homopolymers and long reads</a></li>
<li class="chapter" data-level="1.4.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#accounting-for-homopolymers"><i class="fa fa-check"></i><b>1.4.3</b> Accounting for homopolymers</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>2</b> Aligning sequence data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>2.1</b> What is an alignment ?</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#why-align"><i class="fa fa-check"></i><b>2.1.1</b> Why align ?</a></li>
<li class="chapter" data-level="2.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-align-two-sequences"><i class="fa fa-check"></i><b>2.1.2</b> How to align two sequences ?</a></li>
<li class="chapter" data-level="2.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#scoring-and-substitution-models"><i class="fa fa-check"></i><b>2.1.3</b> Scoring and substitution models</a></li>
<li class="chapter" data-level="2.1.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#dealing-with-gaps"><i class="fa fa-check"></i><b>2.1.4</b> Dealing with gaps</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-do-we-speed-up-pairwise-alignment"><i class="fa fa-check"></i><b>2.2</b> How do we speed up pairwise alignment ?</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#changing-the-method"><i class="fa fa-check"></i><b>2.2.1</b> Changing the method</a></li>
<li class="chapter" data-level="2.2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#seed-and-extend-with-data-structures"><i class="fa fa-check"></i><b>2.2.2</b> Seed and extend with data structures</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#the-specificities-of-read-mapping"><i class="fa fa-check"></i><b>2.3</b> The specificities of read-mapping</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-read-mapping"><i class="fa fa-check"></i><b>2.3.1</b> What is read-mapping ?</a></li>
<li class="chapter" data-level="2.3.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#challenges-of-read-mapping"><i class="fa fa-check"></i><b>2.3.2</b> Challenges of read-mapping</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.4</b> Multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#progressive-alignment"><i class="fa fa-check"></i><b>2.4.1</b> Progressive alignment</a></li>
<li class="chapter" data-level="2.4.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#other-methods"><i class="fa fa-check"></i><b>2.4.2</b> Other methods</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#conclusion-1"><i class="fa fa-check"></i><b>2.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="HPC-paper.html"><a href="HPC-paper.html"><i class="fa fa-check"></i><b>3</b> Contribution 1: Improving read alignment by exploring a sequence transformation space</a>
<ul>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#highlights"><i class="fa fa-check"></i>Highlights</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#graphical-abstract"><i class="fa fa-check"></i>Graphical Abstract</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="3.1" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="HPC-paper.html"><a href="HPC-paper.html#methods"><i class="fa fa-check"></i><b>3.2</b> Methods</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:msr-def"><i class="fa fa-check"></i><b>3.2.1</b> Streaming sequence reductions</a></li>
<li class="chapter" data-level="3.2.2" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:enum"><i class="fa fa-check"></i><b>3.2.2</b> Restricting the space of streaming sequence reductions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="HPC-paper.html"><a href="HPC-paper.html#datasets-and-pipelines"><i class="fa fa-check"></i><b>3.3</b> Datasets and Pipelines</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="HPC-paper.html"><a href="HPC-paper.html#datasets"><i class="fa fa-check"></i><b>3.3.1</b> Datasets</a></li>
<li class="chapter" data-level="3.3.2" data-path="HPC-paper.html"><a href="HPC-paper.html#simulation-pipeline"><i class="fa fa-check"></i><b>3.3.2</b> Simulation pipeline</a></li>
<li class="chapter" data-level="3.3.3" data-path="HPC-paper.html"><a href="HPC-paper.html#evaluation-pipeline"><i class="fa fa-check"></i><b>3.3.3</b> Evaluation pipeline</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-results"><i class="fa fa-check"></i><b>3.4</b> Results</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="HPC-paper.html"><a href="HPC-paper.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>3.4.1</b> Selection of mapping-friendly sequence reductions</a></li>
<li class="chapter" data-level="3.4.2" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>3.4.2</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes</a></li>
<li class="chapter" data-level="3.4.3" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>3.4.3</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome</a></li>
<li class="chapter" data-level="3.4.4" data-path="HPC-paper.html"><a href="HPC-paper.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>3.4.4</b> Raw mapping improves upon HPC on centromeric regions</a></li>
<li class="chapter" data-level="3.4.5" data-path="HPC-paper.html"><a href="HPC-paper.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>3.4.5</b> Positions of incorrectly mapped reads across the entire human genome</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="HPC-paper.html"><a href="HPC-paper.html#discussion"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="HPC-paper.html"><a href="HPC-paper.html#limitations-of-this-study"><i class="fa fa-check"></i><b>3.6</b> Limitations of this study</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#author-contributions"><i class="fa fa-check"></i>Author contributions</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#declaration-of-interests"><i class="fa fa-check"></i>Declaration of interests</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#star-methods"><i class="fa fa-check"></i>STAR Methods</a>
<ul>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#lead-contact"><i class="fa fa-check"></i>Lead contact</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#materials-availability"><i class="fa fa-check"></i>Materials availability</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#data-and-code-availability"><i class="fa fa-check"></i>Data and code availability</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#method-details"><i class="fa fa-check"></i>Method details</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#supplementary-information"><i class="fa fa-check"></i>Supplementary information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html"><i class="fa fa-check"></i><b>4</b> Learning from sequences and alignments</a>
<ul>
<li class="chapter" data-level="4.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#why-learn-from-alignments"><i class="fa fa-check"></i><b>4.1</b> Why learn from alignments ?</a></li>
<li class="chapter" data-level="4.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#what-to-learn"><i class="fa fa-check"></i><b>4.2</b> What to learn ?</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#supervised-learning"><i class="fa fa-check"></i><b>4.2.1</b> Supervised learning</a></li>
<li class="chapter" data-level="4.2.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#unsupervised-learning"><i class="fa fa-check"></i><b>4.2.2</b> Unsupervised learning</a></li>
<li class="chapter" data-level="4.2.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#others-paradigms"><i class="fa fa-check"></i><b>4.2.3</b> Others paradigms</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#how-to-learn"><i class="fa fa-check"></i><b>4.3</b> How to learn ?</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#general-setting"><i class="fa fa-check"></i><b>4.3.1</b> General setting</a></li>
<li class="chapter" data-level="4.3.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>4.3.2</b> Tests and statistical learning</a></li>
<li class="chapter" data-level="4.3.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#more-complex-methods"><i class="fa fa-check"></i><b>4.3.3</b> More complex methods</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#preprocessing-the-alignment-for-machine-learning"><i class="fa fa-check"></i><b>4.4</b> Preprocessing the alignment for machine learning</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#general-purpose-encodings"><i class="fa fa-check"></i><b>4.4.1</b> General purpose encodings</a></li>
<li class="chapter" data-level="4.4.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#biological-sequence-specific-encodings"><i class="fa fa-check"></i><b>4.4.2</b> Biological sequence-specific encodings</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#conclusion-2"><i class="fa fa-check"></i><b>4.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html"><i class="fa fa-check"></i><b>5</b> Viruses, HIV and drug resistance</a>
<ul>
<li class="chapter" data-level="5.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#what-are-viruses"><i class="fa fa-check"></i><b>5.1</b> What are viruses ?</a></li>
<li class="chapter" data-level="5.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#getting-to-know-hiv"><i class="fa fa-check"></i><b>5.2</b> Getting to know HIV</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#quick-presentation-of-hiv"><i class="fa fa-check"></i><b>5.2.1</b> Quick Presentation of HIV</a></li>
<li class="chapter" data-level="5.2.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#the-replication-cycle-of-hiv"><i class="fa fa-check"></i><b>5.2.2</b> The replication cycle of HIV</a></li>
<li class="chapter" data-level="5.2.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#genetics-of-hiv"><i class="fa fa-check"></i><b>5.2.3</b> Genetics of HIV</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#drug-resistance-in-hiv"><i class="fa fa-check"></i><b>5.3</b> Drug resistance in HIV</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#a-quick-history-of-art"><i class="fa fa-check"></i><b>5.3.1</b> A quick history of ART</a></li>
<li class="chapter" data-level="5.3.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#drug-mechanisms"><i class="fa fa-check"></i><b>5.3.2</b> Main mechanisms of viral proteins, antiretroviral drugs and associated resistance.</a></li>
<li class="chapter" data-level="5.3.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#consequences-of-resistance-on-global-health"><i class="fa fa-check"></i><b>5.3.3</b> Consequences of resistance on global health</a></li>
<li class="chapter" data-level="5.3.4" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#finding-drms"><i class="fa fa-check"></i><b>5.3.4</b> Finding DRMs </a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#conclusion-3"><i class="fa fa-check"></i><b>5.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>6</b> Contribution 2: Inferring mutation roles from sequence alignments using machine learning</a>
<ul>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#abstract-paper"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#author-summary"><i class="fa fa-check"></i>Author summary</a></li>
<li class="chapter" data-level="6.1" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>6.2</b> Materials and methods</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>6.2.2</b> Classifier training</a></li>
<li class="chapter" data-level="6.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>6.2.3</b> Measuring classifier performance</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-results"><i class="fa fa-check"></i><b>6.3</b> Results</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>6.3.1</b> Classifier performance &amp; interpretation</a></li>
<li class="chapter" data-level="6.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>6.3.2</b> Additional classification results</a></li>
<li class="chapter" data-level="6.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>6.3.3</b> Identifying new mutations from classifiers</a></li>
<li class="chapter" data-level="6.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>6.3.4</b> Detailed analysis of potentially resistance-associated mutations</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>6.4</b> Discussion and perspectives</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-acknowledgments"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#supporting-information"><i class="fa fa-check"></i>Supporting Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>7</b> Learning alignments, an interesting perspective</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#deep-learning-and-sequences"><i class="fa fa-check"></i><b>7.1</b> Deep learning and sequences</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#intro-to-deep-learning"><i class="fa fa-check"></i><b>7.1.1</b> Intro to deep learning</a></li>
<li class="chapter" data-level="7.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learned-sequence-embeddings"><i class="fa fa-check"></i><b>7.1.2</b> Learned sequence embeddings</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-pairwise-alignment"><i class="fa fa-check"></i><b>7.2</b> Learning pairwise alignment</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-a-substitution-matrix"><i class="fa fa-check"></i><b>7.2.1</b> Predicting a substitution matrix</a></li>
<li class="chapter" data-level="7.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-an-alignment"><i class="fa fa-check"></i><b>7.2.2</b> predicting an alignment</a></li>
<li class="chapter" data-level="7.2.3" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-seeds"><i class="fa fa-check"></i><b>7.2.3</b> Learning seeds</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html"><i class="fa fa-check"></i>Global conclusion</a>
<ul>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hpc-part"><i class="fa fa-check"></i>HPC part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hiv-part"><i class="fa fa-check"></i>HIV part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#final-words"><i class="fa fa-check"></i>Final words</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-references.html"><a href="global-references.html"><i class="fa fa-check"></i>Global References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”</a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation</a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#msr-performance-comparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison</a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome"><i class="fa fa-check"></i><b>A.3</b> Analyzing read origin on whole human genome</a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.4</b> Performance of MSRs on the Drosophila genome</a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#key-resource-table"><i class="fa fa-check"></i><b>A.5</b> Key Resource Table</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “HIV and DRMs”</a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html#detailed-list-of-hiv-1-protein-structures-used-for-figure-generation."><i class="fa fa-check"></i><b>B.1</b> Detailed list of HIV-1 protein structures used for figure generation.</a></li>
<li class="chapter" data-level="B.2" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html#list-of-all-antiretroviral-drugs"><i class="fa fa-check"></i><b>B.2</b> List of all antiretroviral drugs</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>C</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”</a>
<ul>
<li class="chapter" data-level="C.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>C.1</b> S1 Appendix (Technical appendix).</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>C.1.1</b> Data</a></li>
<li class="chapter" data-level="C.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>C.1.2</b> Classifiers</a></li>
<li class="chapter" data-level="C.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>C.1.3</b> Scoring</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>C.2</b> S1 Fig.</a></li>
<li class="chapter" data-level="C.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>C.3</b> S2 Fig.</a></li>
<li class="chapter" data-level="C.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>C.4</b> S3 Fig.</a></li>
<li class="chapter" data-level="C.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>C.5</b> S1 Table.</a></li>
<li class="chapter" data-level="C.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>C.6</b> S2 Appendix. (Fisher exact tests)</a></li>
<li class="chapter" data-level="C.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>C.7</b> S1 Data.</a></li>
<li class="chapter" data-level="C.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>C.8</b> S2 Data.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequences to knowledge, improving and learning from sequence alignments</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="global-references" class="section level1 unnumbered hasAnchor">
<h1>Global References<a href="global-references.html#global-references" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Watson, J. D. &amp; Crick, F. H. C. <a href="https://doi.org/10.1101/SQB.1953.018.01.020">The Structure of Dna</a>. <em>Cold Spring Harbor Symposia on Quantitative Biology</em> <strong>18</strong>, 123–131 (1953).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Sanger, F. <em>et al.</em> <a href="https://doi.org/10.1038/265687a0">Nucleotide sequence of bacteriophage <span class="nocase">φX174 DNA</span></a>. <em>Nature</em> <strong>265</strong>, 687–695 (1977).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Archer, C. T. <em>et al.</em> <a href="https://doi.org/10.1186/1471-2164-12-9">The genome sequence of e. Coli w (ATCC 9637): Comparative genome analysis and an improved genome-scale reconstruction of e. coli</a>. <em>BMC Genomics</em> <strong>12</strong>, 9 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Nurk, S. <em>et al.</em> <a href="https://doi.org/10.1126/science.abj6987">The complete sequence of a human genome</a>. <em>Science</em> <strong>376</strong>, 44–53 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Pellicer, J., Fay, M. F. &amp; Leitch, I. J. <a href="https://doi.org/10.1111/j.1095-8339.2010.01072.x">The largest eukaryotic genome of them all?</a> <em>Botanical Journal of the Linnean Society</em> <strong>164</strong>, 10–15 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Macgregor, H. C. C-Value Paradox. in (eds. Brenner, S. &amp; Miller, J. H.) 249–250 (Academic Press, 2001). doi:<a href="https://doi.org/10.1006/rwgn.2001.0301">10.1006/rwgn.2001.0301</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Alberts, B. <em>et al.</em> <em>Molecular Biology of the Cell. 4th edition</em>. (Garland Science, 2002). doi:<a href="https://doi.org/10.1201/9781315735368">10.1201/9781315735368</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Crick, F. H. C., Barnett, L., Brenner, S. &amp; Watts-Tobin, R. J. <a href="https://doi.org/10.1038/1921227a0">General Nature of the Genetic Code for Proteins</a>. <em>Nature</em> <strong>192</strong>, 1227–1232 (1961).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">International Human Genome Sequencing Consortium. <a href="https://doi.org/10.1038/nature03001">Finishing the euchromatic sequence of the human genome</a>. <em>Nature</em> <strong>431</strong>, 931–945 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Elkon, R. &amp; Agami, R. <a href="https://doi.org/10.1038/nbt.3863">Characterization of noncoding regulatory DNA in the human genome</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 732–746 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Omenn, G. S. <a href="https://doi.org/10.1016/j.mcpro.2021.100062">Reflections on the HUPO human proteome project, the flagship project of the human proteome organization, at 10 years</a>. <em>Molecular &amp; Cellular Proteomics : MCP</em> <strong>20</strong>, 100062 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Shabalina, S. A. &amp; Spiridonov, N. A. <a href="https://doi.org/10.1186/gb-2004-5-4-105">The mammalian transcriptome and the function of non-coding DNA sequences</a>. <em>Genome Biology</em> <strong>5</strong>, 105 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">ENCODE Project Consortium. <a href="https://doi.org/10.1038/nature11247">An integrated encyclopedia of DNA elements in the human genome</a>. <em>Nature</em> <strong>489</strong>, 57–74 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Chatterjee, N. &amp; Walker, G. C. <a href="https://doi.org/10.1002/em.22087">Mechanisms of DNA damage, repair, and mutagenesis: DNA Damage and Repair</a>. <em>Environmental and Molecular Mutagenesis</em> <strong>58</strong>, 235–263 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Fijalkowska, I. J., Schaaper, R. M. &amp; Jonczyk, P. <a href="https://doi.org/10.1111/j.1574-6976.2012.00338.x">DNA replication fidelity in escherichia coli: A multi-DNA polymerase affair</a>. <em>FEMS microbiology reviews</em> <strong>36</strong>, 1105–1121 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Pray, L. <a href="https://www.nature.com/scitable/topicpage/dna-replication-and-causes-of-mutation-409/">DNA replication and causes of mutation</a>. <em>Nature education</em> <strong>1</strong>, 214 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Gout, J.-F., Thomas, W. K., Smith, Z., Okamoto, K. &amp; Lynch, M. <a href="https://doi.org/10.1073/pnas.1309843110">Large-scale detection of in vivo transcription errors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>110</strong>, 18584–18589 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Gout, J.-F. <em>et al.</em> <a href="https://doi.org/10.1126/sciadv.1701484">The landscape of transcription errors in eukaryotic cells</a>. <em>Science Advances</em> <strong>3</strong>, e1701484 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Shcherbakov, D. <em>et al.</em> <a href="https://doi.org/10.1038/s42003-019-0626-9">Ribosomal mistranslation leads to silencing of the unfolded protein response and increased mitochondrial biogenesis</a>. <em>Communications Biology</em> <strong>2</strong>, 1–16 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Desouky, O., Ding, N. &amp; Zhou, G. <a href="https://doi.org/10.1016/j.jrras.2015.03.003">Targeted and non-targeted effects of ionizing radiation</a>. <em>Journal of Radiation Research and Applied Sciences</em> <strong>8</strong>, 247–254 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Kiefer, J. Effects of Ultraviolet Radiation on DNA. in (eds. Obe, G. &amp; Vijayalaxmi) 39–53 (Springer, 2007). doi:<a href="https://doi.org/10.1007/978-3-540-71414-9_3">10.1007/978-3-540-71414-9_3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Bennett, J. W. &amp; Klich, M. <a href="https://doi.org/10.1128/CMR.16.3.497-516.2003">Mycotoxins</a>. <em>Clinical Microbiology Reviews</em> <strong>16</strong>, 497–516 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Kantidze, O. L., Velichko, A. K., Luzhin, A. V. &amp; Razin, S. V. Heat stress-induced DNA damage. <em>Acta Naturae</em> <strong>8</strong>, 75–78 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Gregory, C. D. &amp; Milner, A. E. <a href="https://doi.org/10.1002/ijc.2910570321">Regulation of cell survival in Burkitt lymphoma: implications from studies of apoptosis following cold-shock treatment</a>. <em>International Journal of Cancer</em> <strong>57</strong>, 419–426 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Gafter-Gvili, A. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0068341">Oxidative stress-induced DNA damage and repair in human peripheral blood mononuclear cells: Protective role of hemoglobin</a>. <em>PLoS ONE</em> <strong>8</strong>, e68341 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Roth, J. R. <a href="https://doi.org/10.1146/annurev.ge.08.120174.001535">Frameshift mutations</a>. <em>Annual Review of Genetics</em> <strong>8</strong>, 319–346 (1974).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Kujovich, J. L. <a href="https://doi.org/10.1097/GIM.0b013e3181faa0f2">Factor V Leiden thrombophilia</a>. <em>Genetics in Medicine</em> <strong>13</strong>, 1–16 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Cutting, G. R. <a href="https://doi.org/10.1038/nrg3849">Cystic fibrosis genetics: from molecular understanding to clinical application</a>. <em>Nature Reviews Genetics</em> <strong>16</strong>, 45–56 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Fuchsberger, C. <em>et al.</em> <a href="https://doi.org/10.1038/nature18642">The genetic architecture of type 2 diabetes</a>. <em>Nature</em> <strong>536</strong>, 41–47 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Morris, A. P. <em>et al.</em> <a href="https://doi.org/10.1038/ng.2383">Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes</a>. <em>Nature genetics</em> <strong>44</strong>, 981–990 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Woodford, N. &amp; Ellington, M. J. <a href="https://doi.org/10.1111/j.1469-0691.2006.01492.x">The emergence of antibiotic resistance by mutation</a>. <em>Clinical Microbiology and Infection</em> <strong>13</strong>, 5–18 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Rhee, S.-Y. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkg100">Human immunodeficiency virus reverse transcriptase and protease sequence database</a>. <em>Nucleic Acids Research</em> <strong>31</strong>, 298–303 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Sanger, F., Nicklen, S. &amp; Coulson, A. R. <a href="https://doi.org/10.1073/pnas.74.12.5463"><span>DNA</span> sequencing with chain-terminating inhibitors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>74</strong>, 5463–5467 (1977).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Smith, L. M., Fung, S., Hunkapiller, M. W., Hunkapiller, T. J. &amp; Hood, L. E. <a href="https://doi.org/10.1093/nar/13.7.2399">The synthesis of oligonucleotides containing an aliphatic amino group at the 5′ terminus: Synthesis of fluorescent <span>DNA</span> primers for use in <span>DNA</span> sequence analysis</a>. <em>Nucleic Acids Research</em> <strong>13</strong>, 2399–2412 (1985).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline">Smith, L. M. <em>et al.</em> <a href="https://doi.org/10.1038/321674a0">Fluorescence detection in automated <span>DNA</span> sequence analysis</a>. <em>Nature</em> <strong>321</strong>, 674–679 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Ansorge, W., Sproat, B., Stegemann, J., Schwager, C. &amp; Zenke, M. <a href="https://doi.org/10.1093/nar/15.11.4593">Automated DNA sequencing: Ultrasensitive detection of fluorescent bands during electrophoresis</a>. <em>Nucleic Acids Research</em> <strong>15</strong>, 4593–4602 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Shendure, J. &amp; Ji, H. <a href="https://doi.org/10.1038/nbt1486">Next-generation DNA sequencing</a>. <em>Nature Biotechnology</em> <strong>26</strong>, 1135–1145 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Collins, F. S., Morgan, M. &amp; Patrinos, A. <a href="https://doi.org/10.1126/science.1084564">The human genome project: Lessons from large-scale biology</a>. <em>Science</em> <strong>300</strong>, 286–290 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Liu, L. <em>et al.</em> <a href="https://doi.org/10.1155/2012/251364">Comparison of <span>Next-Generation Sequencing Systems</span></a>. <em>Journal of Biomedicine and Biotechnology</em> <strong>2012</strong>, e251364 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><a href="https://www.genome.gov/about-genomics/fact-sheets/Sequencing-Human-Genome-cost">The Cost of Sequencing a Human Genome</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Metzker, M. L. <a href="https://doi.org/10.1038/nrg2626">Sequencing technologies — the next generation</a>. <em>Nat Rev Genet</em> <strong>11</strong>, 31–46 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Canard, B. &amp; Sarfati, R. S. <a href="https://doi.org/10.1016/0378-1119(94)90226-7">DNA polymerase fluorescent substrates with reversible 3<span>’</span>-tags</a>. <em>Gene</em> <strong>148</strong>, 1–6 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Nyren, P., Pettersson, B. &amp; Uhlen, M. <a href="https://doi.org/10.1006/abio.1993.1024">Solid Phase DNA Minisequencing by an Enzymatic Luminometric Inorganic Pyrophosphate Detection Assay</a>. <em>Analytical Biochemistry</em> <strong>208</strong>, 171–175 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Mardis, E. R. <a href="https://doi.org/10.1038/nature09796">A decade<span>’</span>s perspective on DNA sequencing technology</a>. <em>Nature</em> <strong>470</strong>, 198–203 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Stoler, N. &amp; Nekrutenko, A. <a href="https://doi.org/10.1093/nargab/lqab019">Sequencing error profiles of illumina sequencing instruments</a>. <em>NAR Genomics and Bioinformatics</em> <strong>3</strong>, lqab019 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><a href="https://www.illumina.com/science/technology/next-generation-sequencing/sequencing-technology.html">Sequencing Technology | Sequencing by synthesis</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Pollard, M. O., Gurdasani, D., Mentzer, A. J., Porter, T. &amp; Sandhu, M. S. <a href="https://doi.org/10.1093/hmg/ddy177">Long reads: Their purpose and place</a>. <em>Human Molecular Genetics</em> <strong>27</strong>, R234–R241 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline">Eid, J. <em>et al.</em> <a href="https://doi.org/10.1126/science.1162986">Real-time DNA sequencing from single polymerase molecules</a>. <em>Science</em> <strong>323</strong>, 133–138 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">Levene, M. J. <em>et al.</em> <a href="https://doi.org/10.1126/science.1079700">Zero-mode waveguides for single-molecule analysis at high concentrations</a>. <em>Science</em> <strong>299</strong>, 682–686 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">Clarke, J. <em>et al.</em> <a href="https://doi.org/10.1038/nnano.2009.12">Continuous base identification for single-molecule nanopore DNA sequencing</a>. <em>Nature Nanotechnology</em> <strong>4</strong>, 265–270 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Deamer, D., Akeson, M. &amp; Branton, D. <a href="https://doi.org/10.1038/nbt.3423">Three decades of nanopore sequencing</a>. <em>Nature Biotechnology</em> <strong>34</strong>, 518–524 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Wick, R. R., Judd, L. M. &amp; Holt, K. E. <a href="https://doi.org/10.1186/s13059-019-1727-y">Performance of neural network basecalling tools for Oxford Nanopore sequencing</a>. <em>Genome Biology</em> <strong>20</strong>, 129 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Rhoads, A. &amp; Au, K. F. <a href="https://doi.org/10.1016/j.gpb.2015.08.002">PacBio Sequencing and Its Applications</a>. <em>Genomics, Proteomics &amp; Bioinformatics</em> <strong>13</strong>, 278–289 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline">Ip, C. L. C. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.7201.1">MinION analysis and reference consortium: Phase 1 data release and analysis</a>. <em>F1000Research</em> <strong>4</strong>, 1075 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline">Logsdon, G. A., Vollger, M. R. &amp; Eichler, E. E. <a href="https://doi.org/10.1038/s41576-020-0236-x">Long-read human genome sequencing and its applications</a>. <em>Nat Rev Genet</em> <strong>21</strong>, 597–614 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline">Jain, M. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.4060">Nanopore sequencing and assembly of a human genome with ultra-long reads</a>. <em>Nature Biotechnology</em> <strong>36</strong>, 338–345 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><a href="http://lab.loman.net/2017/03/09/ultrareads-for-nanopore/">Thar she blows! Ultra long read method for nanopore sequencing · loman labs</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline">Payne, A., Holmes, N., Rakyan, V. &amp; Loose, M. <a href="https://doi.org/10.1093/bioinformatics/bty841">BulkVis: A graphical viewer for oxford nanopore bulk FAST5 files</a>. <em>Bioinformatics</em> <strong>35</strong>, 2193–2198 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline">Murigneux, V. <em>et al.</em> <a href="https://doi.org/10.1093/gigascience/giaa146">Comparison of long-read methods for sequencing and assembly of a plant genome</a>. <em>GigaScience</em> <strong>9</strong>, giaa146 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline">Chaisson, M. J. P. <em>et al.</em> <a href="https://doi.org/10.1038/nature13907">Resolving the complexity of the human genome using single-molecule sequencing</a>. <em>Nature</em> <strong>517</strong>, 608–611 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline">Jain, M., Olsen, H. E., Paten, B. &amp; Akeson, M. <a href="https://doi.org/10.1186/s13059-016-1103-0">The Oxford Nanopore MinION: delivery of nanopore sequencing to the genomics community</a>. <em>Genome Biology</em> <strong>17</strong>, 239 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline">Hong, M. <em>et al.</em> <a href="https://doi.org/10.1186/s13045-020-01005-x">RNA sequencing: New technologies and applications in cancer research</a>. <em>Journal of Hematology &amp; Oncology</em> <strong>13</strong>, 166 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline">Ozsolak, F. &amp; Milos, P. M. <a href="https://doi.org/10.1038/nrg2934">RNA sequencing: advances, challenges and opportunities</a>. <em>Nature Reviews Genetics</em> <strong>12</strong>, 87–98 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline">Hunt, D. F., Yates, J. R., Shabanowitz, J., Winston, S. &amp; Hauer, C. R. <a href="https://doi.org/10.1073/pnas.83.17.6233">Protein sequencing by tandem mass spectrometry.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>83</strong>, 6233–6237 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline">Smith, B. J. <em>Protein Sequencing Protocols</em>. (Springer Science &amp; Business Media, 2002). doi:<a href="https://doi.org/10.1385/1592593429">10.1385/1592593429</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline">Restrepo-Pérez, L., Joo, C. &amp; Dekker, C. <a href="https://doi.org/10.1038/s41565-018-0236-6">Paving the way to single-molecule protein sequencing</a>. <em>Nature Nanotechnology</em> <strong>13</strong>, 786–796 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Weirather, J. L. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.10571.2">Comprehensive comparison of pacific biosciences and oxford nanopore technologies and their applications to transcriptome analysis</a>. <em>F1000Research</em> <strong>6</strong>, 100 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Wang, Y., Zhao, Y., Bollas, A., Wang, Y. &amp; Au, K. F. <a href="https://doi.org/10.1038/s41587-021-01108-x">Nanopore sequencing technology, bioinformatics and applications</a>. <em>Nature Biotechnology</em> <strong>39</strong>, 1348–1365 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline">Ma, X. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-019-1659-6">Analysis of error profiles in deep next-generation sequencing data</a>. <em>Genome Biology</em> <strong>20</strong>, 50 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline">Lima, L. <em>et al.</em> <a href="https://doi.org/10.1093/bib/bbz058">Comparative assessment of long-read error correction software applied to nanopore RNA-sequencing data</a>. <em>Briefings in Bioinformatics</em> <strong>21</strong>, 1164–1181 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline">Fu, S., Wang, A. &amp; Au, K. F. <a href="https://doi.org/10.1186/s13059-018-1605-z">A comparative evaluation of hybrid error correction methods for error-prone long reads</a>. <em>Genome Biology</em> <strong>20</strong>, 26 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline">Zhang, H., Jain, C. &amp; Aluru, S. <a href="https://doi.org/10.1186/s12864-020-07227-0">A comprehensive evaluation of long read error correction methods</a>. <em>BMC Genomics</em> <strong>21</strong>, 889 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline">Amarasinghe, S. L. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-020-1935-5">Opportunities and challenges in long-read sequencing data analysis</a>. <em>Genome Biology</em> <strong>21</strong>, 30 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline">Ruan, J. &amp; Li, H. <a href="https://doi.org/10.1038/s41592-019-0669-3">Fast and accurate long-read assembly with wtdbg2</a>. <em>Nature Methods</em> <strong>17</strong>, 155–158 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline">Koren, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.215087.116">Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation</a>. <em>Genome Research</em> <strong>27</strong>, 722–736 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline">Tischler, G. &amp; Myers, E. W. <em>Non Hybrid Long Read Consensus Using Local De Bruijn Graph Assembly</em>. 106252 (2017) doi:<a href="https://doi.org/10.1101/106252">10.1101/106252</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline">Warren, R. L. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btz400">ntEdit: Scalable genome sequence polishing</a>. <em>Bioinformatics</em> <strong>35</strong>, 4430–4432 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline">Hepler, N. L. <em>et al.</em> <a href="https://www.pacb.com/wp-content/uploads/improved-circular-consensus-algorithm-with-application-to-detection-hiv-1-drams.pdf">An improved circular consensus algorithm with an application to detect HIV-1 drug-resistance associated mutations (DRAMs)</a>. in (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline">Simpson, J. T. <em>et al.</em> <a href="https://doi.org/10.1038/nmeth.4184">Detecting DNA cytosine methylation using nanopore sequencing</a>. <em>Nature Methods</em> <strong>14</strong>, 407–410 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline">Hackl, T., Hedrich, R., Schultz, J. &amp; Förster, F. <a href="https://doi.org/10.1093/bioinformatics/btu392">Proovread : Large-scale high-accuracy PacBio correction through iterative short read consensus</a>. <em>Bioinformatics</em> <strong>30</strong>, 3004–3011 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline">Miclotte, G. <em>et al.</em> <a href="https://doi.org/10.1186/s13015-016-0075-7">Jabba: Hybrid error correction for long sequencing reads</a>. <em>Algorithms for Molecular Biology</em> <strong>11</strong>, 10 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline">Koren, S. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.2280">Hybrid error correction and de novo assembly of single-molecule sequencing reads</a>. <em>Nature Biotechnology</em> <strong>30</strong>, 693–700 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline">Salmela, L. &amp; Rivals, E. <a href="https://doi.org/10.1093/bioinformatics/btu538">LoRDEC: Accurate and efficient long read error correction</a>. <em>Bioinformatics</em> <strong>30</strong>, 3506–3514 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline">Walker, B. J. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</a>. <em>PLOS ONE</em> <strong>9</strong>, e112963 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline">Vaser, R., Sović, I., Nagarajan, N. &amp; Šikić, M. <a href="https://doi.org/10.1101/gr.214270.116">Fast and accurate de novo genome assembly from long uncorrected reads</a>. <em>Genome Research</em> <strong>27</strong>, 737–746 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline">Wenger, A. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-019-0217-9">Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome</a>. <em>Nature Biotechnology</em> <strong>37</strong>, 1155–1162 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline">Timp, W., Comer, J. &amp; Aksimentiev, A. <a href="https://doi.org/10.1016/j.bpj.2012.04.009">DNA Base-Calling from a Nanopore Using a Viterbi Algorithm</a>. <em>Biophysical Journal</em> <strong>102</strong>, L37–L39 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline">Perešíni, P., Boža, V., Brejová, B. &amp; Vinař, T. <a href="https://doi.org/10.1093/bioinformatics/btab528">Nanopore base calling on the edge</a>. <em>Bioinformatics</em> <strong>37</strong>, 4661–4667 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline">Boža, V., Brejová, B. &amp; Vinař, T. <a href="https://doi.org/10.1371/journal.pone.0178751">DeepNano: Deep recurrent neural networks for base calling in MinION nanopore reads</a>. <em>PLOS ONE</em> <strong>12</strong>, e0178751 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline">Tyler, A. D. <em>et al.</em> <a href="https://doi.org/10.1038/s41598-018-29334-5">Evaluation of Oxford Nanopore<span>’</span>s MinION Sequencing Device for Microbial Whole Genome Sequencing Applications</a>. <em>Scientific Reports</em> <strong>8</strong>, 10931 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline">Lin, B., Hui, J. &amp; Mao, H. <a href="https://doi.org/10.3390/bios11070214">Nanopore Technology and Its Applications in Gene Sequencing</a>. <em>Biosensors</em> <strong>11</strong>, 214 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">92. </div><div class="csl-right-inline"><a href="http://nanoporetech.com/about-us/news/oxford-nanopore-tech-update-new-duplex-method-q30-nanopore-single-molecule-reads-0">Oxford Nanopore Tech Update: new Duplex method for Q30 nanopore single molecule reads, PromethION 2, and more</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">93. </div><div class="csl-right-inline">Sanderson, N. <em>et al.</em> Comparison of R9.4.1/Kit10 and R10/Kit12 Oxford Nanopore flowcells and chemistries in bacterial genome reconstruction. doi:<a href="https://doi.org/10.1101/2022.04.29.490057">10.1101/2022.04.29.490057</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">94. </div><div class="csl-right-inline">Karst, S. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-020-01041-y">High-accuracy long-read amplicon sequences using unique molecular identifiers with Nanopore or PacBio sequencing</a>. <em>Nature Methods</em> <strong>18</strong>, 165–169 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">95. </div><div class="csl-right-inline">Chen, Z. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.3982">Highly accurate fluorogenic DNA sequencing with information theory<span></span>based error correction</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 1170–1178 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">96. </div><div class="csl-right-inline"><a href="https://www.illumina.com/content/illumina-marketing/amr/en_US/science/genomics-research/articles/infinity-high-performance-long-read-assay.html">High Performance Long Read Assay Enables Contiguous Data up to 10Kb on Existing Illumina Platforms</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">97. </div><div class="csl-right-inline">Booeshaghi, A. S. &amp; Pachter, L. Pseudoalignment facilitates assignment of error-prone Ultima Genomics reads. doi:<a href="https://doi.org/10.1101/2022.06.04.494845">10.1101/2022.06.04.494845</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">98. </div><div class="csl-right-inline">Delahaye, C. &amp; Nicolas, J. <a href="https://doi.org/10.1371/journal.pone.0257521">Sequencing DNA with nanopores: Troubles and biases</a>. <em>PLOS ONE</em> <strong>16</strong>, e0257521 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">99. </div><div class="csl-right-inline">Goodwin, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.191395.115">Oxford Nanopore sequencing, hybrid error correction, and de novo assembly of a eukaryotic genome</a>. <em>Genome Research</em> <strong>25</strong>, 1750–1756 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">100. </div><div class="csl-right-inline">Dohm, J. C., Peters, P., Stralis-Pavese, N. &amp; Himmelbauer, H. <a href="https://doi.org/10.1093/nargab/lqaa037">Benchmarking of long-read correction methods</a>. <em>NAR Genomics and Bioinformatics</em> <strong>2</strong>, (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">101. </div><div class="csl-right-inline">Foox, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-021-01049-5">Performance assessment of DNA sequencing platforms in the ABRF Next-Generation Sequencing Study</a>. <em>Nature Biotechnology</em> <strong>39</strong>, 1129–1140 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">102. </div><div class="csl-right-inline">Huang, Y.-T., Liu, P.-Y. &amp; Shih, P.-W. <a href="https://doi.org/10.1186/s13059-021-02282-6">Homopolish: A method for the removal of systematic errors in nanopore sequencing by homologous polishing</a>. <em>Genome Biology</em> <strong>22</strong>, 95 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">103. </div><div class="csl-right-inline">Rang, F. J., Kloosterman, W. P. &amp; Ridder, J. de. <a href="https://doi.org/10.1186/s13059-018-1462-9">From squiggle to basepair: Computational approaches for improving nanopore sequencing read accuracy</a>. <em>Genome Biology</em> <strong>19</strong>, 90 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">104. </div><div class="csl-right-inline">Sarkozy, P., Jobbágy, Á. &amp; Antal, P. Calling Homopolymer Stretches from Raw Nanopore Reads by Analyzing k-mer Dwell Times. in (eds. Eskola, H., Väisänen, O., Viik, J. &amp; Hyttinen, J.) 241–244 (Springer, 2018). doi:<a href="https://doi.org/10.1007/978-981-10-5122-7_61">10.1007/978-981-10-5122-7_61</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">105. </div><div class="csl-right-inline">Hawkins, J. A., Jones, S. K., Finkelstein, I. J. &amp; Press, W. H. <a href="https://doi.org/10.1073/pnas.1802640115">Indel-correcting DNA barcodes for high-throughput sequencing</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>115</strong>, E6217–E6226 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">106. </div><div class="csl-right-inline">Srivathsan, A. <em>et al.</em> <a href="https://doi.org/10.1111/1755-0998.12890">A MinION<span></span>-based pipeline for fast and cost-effective DNA barcoding</a>. <em>Molecular Ecology Resources</em> <strong>18</strong>, 1035–1049 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">107. </div><div class="csl-right-inline">Wang, Y., Noor-A-Rahim, Md., Gunawan, E., Guan, Y. L. &amp; Poh, C. L. <a href="https://doi.org/10.1109/LCOMM.2019.2912572">Construction of bio-constrained code for DNA data storage</a>. <em>IEEE Communications Letters</em> <strong>23</strong>, 963–966 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">108. </div><div class="csl-right-inline"><a href="http://nanoporetech.com/about-us/news/r103-newest-nanopore-high-accuracy-nanopore-sequencing-now-available-store">R10.3: the newest nanopore for high accuracy nanopore sequencing <span></span> now available in store</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">109. </div><div class="csl-right-inline">Zhou, L. <em>et al.</em> <a href="https://doi.org/10.1116/1.5116295">Detection of DNA homopolymer with graphene nanopore</a>. <em>Journal of Vacuum Science &amp; Technology B</em> <strong>37</strong>, 061809 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">110. </div><div class="csl-right-inline">Goto, Y., Yanagi, I., Matsui, K., Yokoi, T. &amp; Takeda, K. <a href="https://doi.org/10.1039/C8NR04238A">Identification of four single-stranded DNA homopolymers with a solid-state nanopore in alkaline CsCl solution</a>. <em>Nanoscale</em> <strong>10</strong>, 20844–20850 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">111. </div><div class="csl-right-inline">Nurk, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.263566.120"><span>HiCanu</span>: Accurate assembly of segmental duplications, satellites, and allelic variants from high-fidelity long reads</a>. <em>Genome Res.</em> <strong>30</strong>, 1291–1305 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">112. </div><div class="csl-right-inline">Ekim, B., Berger, B. &amp; Chikhi, R. <a href="https://doi.org/10.1016/j.cels.2021.08.009">Minimizer-space de <span>Bruijn</span> graphs: Whole-genome assembly of long reads in minutes on a personal computer</a>. <em>Cell Systems</em> <strong>12</strong>, 958–968.e6 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">113. </div><div class="csl-right-inline">Shafin, K. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-020-0503-6">Nanopore sequencing and the Shasta toolkit enable efficient de novo assembly of eleven human genomes</a>. <em>Nature Biotechnology</em> <strong>38</strong>, 1044–1053 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">114. </div><div class="csl-right-inline">Miller, J. R. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btn548">Aggressive assembly of pyrosequencing reads with mates</a>. <em>Bioinformatics</em> <strong>24</strong>, 2818–2824 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">115. </div><div class="csl-right-inline">Sahlin, K. &amp; Medvedev, P. <a href="https://doi.org/10.1089/cmb.2019.0299">De <span>Novo Clustering</span> of <span>Long-Read Transcriptome Data Using</span> a <span>Greedy</span>, <span>Quality Value-Based Algorithm</span></a>. <em>Journal of Computational Biology</em> <strong>27</strong>, 472–484 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">116. </div><div class="csl-right-inline">Au, K. F., Underwood, J. G., Lee, L. &amp; Wong, W. H. <a href="https://doi.org/10.1371/journal.pone.0046679">Improving <span>PacBio Long Read Accuracy</span> by <span>Short Read Alignment</span></a>. <em>PLOS ONE</em> <strong>7</strong>, e46679 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">117. </div><div class="csl-right-inline">Hu, R., Sun, G. &amp; Sun, X. <a href="https://doi.org/10.1186/s12859-016-1316-y">LSCplus: a fast solution for improving long read accuracy by short read alignment</a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 451 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">118. </div><div class="csl-right-inline">Li, H. <a href="https://doi.org/10.1093/bioinformatics/bty191">Minimap2: Pairwise alignment for nucleotide sequences</a>. <em>Bioinformatics</em> <strong>34</strong>, 3094–3100 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">119. </div><div class="csl-right-inline">Jain, C. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btaa435">Weighted minimizer sampling improves long read mapping</a>. <em>Bioinformatics</em> <strong>36</strong>, i111–i118 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">120. </div><div class="csl-right-inline">Van Neste, C., Van Nieuwerburgh, F., Van Hoofstat, D. &amp; Deforce, D. <a href="https://doi.org/10.1016/j.fsigen.2012.03.004">Forensic STR analysis using massive parallel sequencing</a>. <em>Forensic Science International: Genetics</em> <strong>6</strong>, 810–818 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">121. </div><div class="csl-right-inline"><a href="https://www.pacb.com/technology/sequencing-by-binding/">Short-read sequencing by binding</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">122. </div><div class="csl-right-inline">Cetin, A. E. <em>et al.</em> <a href="https://doi.org/10.1021/acssensors.7b00957">Plasmonic sensor could enable label-free DNA sequencing</a>. <em>ACS Sensors</em> <strong>3</strong>, 561–568 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">123. </div><div class="csl-right-inline">Almogy, G. <em>et al.</em> Cost-efficient whole genome-sequencing using novel mostly natural sequencing-by-synthesis chemistry and open fluidics platform. doi:<a href="https://doi.org/10.1101/2022.05.29.493900">10.1101/2022.05.29.493900</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">124. </div><div class="csl-right-inline">Sunagawa, S. <em>et al.</em> <a href="https://doi.org/10.1038/s41579-020-0364-5">Tara Oceans: towards global ocean ecosystems biology</a>. <em>Nature Reviews Microbiology</em> <strong>18</strong>, 428–445 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">125. </div><div class="csl-right-inline">Lewin, H. A. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.1720115115">Earth BioGenome project: Sequencing life for the future of life</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>115</strong>, 4325–4333 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">126. </div><div class="csl-right-inline">Lightbody, G. <em>et al.</em> <a href="https://doi.org/10.1093/bib/bby051">Review of applications of high-throughput sequencing in personalized medicine: Barriers and facilitators of future progress in research and clinical application</a>. <em>Briefings in Bioinformatics</em> <strong>20</strong>, 1795–1811 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">127. </div><div class="csl-right-inline">Hamming, R. W. <em>Coding and Information Theory</em>. (Prentice-Hall, 1980).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">128. </div><div class="csl-right-inline">Gusfield, D. <em>Algorithms on strings, trees, and sequences: Computer science and computational biology</em>. (Cambridge University Press, 1997). doi:<a href="https://doi.org/10.1017/CBO9780511574931">10.1017/CBO9780511574931</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">129. </div><div class="csl-right-inline">Levenshtein, V. I. Binary codes capable of correcting deletions, insertions and reversals. <em>Soviet Physics Doklady</em> <strong>10</strong>, 707 (1966).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">130. </div><div class="csl-right-inline">Hardison, R. C. <a href="https://doi.org/10.1371/journal.pbio.0000058">Comparative Genomics</a>. <em>PLOS Biology</em> <strong>1</strong>, e58 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">131. </div><div class="csl-right-inline">Felsenstein, J. <a href="https://doi.org/10.1007/BF01734359">Evolutionary trees from DNA sequences: A maximum likelihood approach</a>. <em>Journal of Molecular Evolution</em> <strong>17</strong>, 368–376 (1981).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">132. </div><div class="csl-right-inline">Kumar, S., Tamura, K. &amp; Nei, M. <a href="https://doi.org/10.1093/bioinformatics/10.2.189">MEGA: Molecular evolutionary genetics analysis software for microcomputers</a>. <em>Bioinformatics</em> <strong>10</strong>, 189–191 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">133. </div><div class="csl-right-inline">Kozlov, A. M., Darriba, D., Flouri, T., Morel, B. &amp; Stamatakis, A. <a href="https://doi.org/10.1093/bioinformatics/btz305">RAxML-NG: A fast, scalable and user-friendly tool for maximum likelihood phylogenetic inference</a>. <em>Bioinformatics</em> <strong>35</strong>, 4453–4455 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">134. </div><div class="csl-right-inline">Guindon, S. <em>et al.</em> <a href="https://doi.org/10.1093/sysbio/syq010">New algorithms and methods to estimate maximum-likelihood phylogenies: Assessing the performance of PhyML 3.0</a>. <em>Systematic Biology</em> <strong>59</strong>, 307–321 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">135. </div><div class="csl-right-inline">Price, M. N., Dehal, P. S. &amp; Arkin, A. P. <a href="https://doi.org/10.1371/journal.pone.0009490">FastTree 2 <span></span> Approximately Maximum-Likelihood Trees for Large Alignments</a>. <em>PLOS ONE</em> <strong>5</strong>, e9490 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">136. </div><div class="csl-right-inline">Jumper, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-021-03819-2">Highly accurate protein structure prediction with AlphaFold</a>. <em>Nature</em> <strong>596</strong>, 583–589 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">137. </div><div class="csl-right-inline">Karplus, K. <em>et al.</em> <a href="https://doi.org/10.1002/(SICI)1097-0134(1999)37:3+&lt;121::AID-PROT16&gt;3.0.CO;2-Q">Predicting protein structure using only sequence information</a>. <em>Proteins: Structure, Function, and Bioinformatics</em> <strong>37</strong>, 121–125 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">138. </div><div class="csl-right-inline">Watson, J. D., Laskowski, R. A. &amp; Thornton, J. M. <a href="https://doi.org/10.1016/j.sbi.2005.04.003">Predicting protein function from sequence and structural data</a>. <em>Current Opinion in Structural Biology</em> <strong>15</strong>, 275–284 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">139. </div><div class="csl-right-inline">Lee, D., Redfern, O. &amp; Orengo, C. <a href="https://doi.org/10.1038/nrm2281">Predicting protein function from sequence and structure</a>. <em>Nature Reviews Molecular Cell Biology</em> <strong>8</strong>, 995–1005 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">140. </div><div class="csl-right-inline">Salmela, L. &amp; Schröder, J. <a href="https://doi.org/10.1093/bioinformatics/btr170">Correcting errors in short reads by multiple alignments</a>. <em>Bioinformatics</em> <strong>27</strong>, 1455–1461 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">141. </div><div class="csl-right-inline">Medvedev, P., Stanciu, M. &amp; Brudno, M. <a href="https://doi.org/10.1038/nmeth.1374">Computational methods for discovering structural variation with next-generation sequencing</a>. <em>Nature Methods</em> <strong>6</strong>, S13–S20 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">142. </div><div class="csl-right-inline">Mahmoud, M. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-019-1828-7">Structural variant calling: the long and the short of it</a>. <em>Genome Biology</em> <strong>20</strong>, 246 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">143. </div><div class="csl-right-inline">Sung, W.-K. <em>Algorithms in bioinformatics: A practical introduction</em>. (Chapman; Hall/CRC, 2011). doi:<a href="https://doi.org/10.1201/9781420070347">10.1201/9781420070347</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">144. </div><div class="csl-right-inline">Needleman, S. B. &amp; Wunsch, C. D. <a href="https://doi.org/10.1016/0022-2836(70)90057-4">A general method applicable to the search for similarities in the amino acid sequence of two proteins</a>. <em>Journal of Molecular Biology</em> <strong>48</strong>, 443–453 (1970).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">145. </div><div class="csl-right-inline">Smith, T. F. &amp; Waterman, M. S. <a href="https://doi.org/10.1016/0022-2836(81)90087-5">Identification of common molecular subsequences</a>. <em>Journal of Molecular Biology</em> <strong>147</strong>, 195–197 (1981).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">146. </div><div class="csl-right-inline">Bradley, S. P., Hax, A. C. &amp; Magnanti, T. L. <em>Applied Mathematical Programming</em>. (Addison-Wesley Publishing Company, 1977).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">147. </div><div class="csl-right-inline">Bellman, R. <a href="https://doi.org/10.1090/S0002-9904-1954-09848-8">The theory of dynamic programming</a>. <em>Bulletin of the American Mathematical Society</em> <strong>60</strong>, 503–515 (1954).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">148. </div><div class="csl-right-inline">Masek, W. J. &amp; Paterson, M. S. <a href="https://doi.org/10.1016/0022-0000(80)90002-1">A faster algorithm computing string edit distances</a>. <em>Journal of Computer and System Sciences</em> <strong>20</strong>, 18–31 (1980).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">149. </div><div class="csl-right-inline">Vinh, N. X., Epps, J. &amp; Bailey, J. Information <span>Theoretic Measures</span> for <span>Clusterings Comparison</span>: <span>Variants</span>, <span>Properties</span>, <span>Normalization</span> and <span>Correction</span> for <span>Chance</span>. <em>Journal of Machine Learning Research</em> <strong>11</strong>, 18 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">150. </div><div class="csl-right-inline">Ullman, J. D., Aho, A. V. &amp; Hirschberg, D. S. <a href="https://doi.org/10.1145/321921.321922">Bounds on the complexity of the longest common subsequence problem</a>. <em>Journal of the ACM</em> <strong>23</strong>, 112 (1976).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">151. </div><div class="csl-right-inline">Hirschberg, D. S. <a href="https://doi.org/10.1145/360825.360861">A linear space algorithm for computing maximal common subsequences</a>. <em>Communications of the ACM</em> <strong>18</strong>, 341343 (1975).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">152. </div><div class="csl-right-inline">Myers, E. W. &amp; Miller, W. <a href="https://doi.org/10.1093/bioinformatics/4.1.11">Optimal alignments in linear space</a>. <em>Bioinformatics</em> <strong>4</strong>, 11–17 (1988).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">153. </div><div class="csl-right-inline">Rice, P., Longden, I. &amp; Bleasby, A. EMBOSS: The european molecular biology open software suite. <em>Trends in genetics</em> <strong>16</strong>, 276277 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">154. </div><div class="csl-right-inline">Huang, X. &amp; Miller, W. <a href="https://doi.org/10.1016/0196-8858(91)90017-D">A time-efficient, linear-space local similarity algorithm</a>. <em>Advances in Applied Mathematics</em> <strong>12</strong>, 337–357 (1991).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">155. </div><div class="csl-right-inline">Waterman, M. S. &amp; Eggert, M. <a href="https://doi.org/10.1016/0022-2836(87)90478-5">A new algorithm for best subsequence alignments with application to tRNA-rRNA comparisons</a>. <em>Journal of Molecular Biology</em> <strong>197</strong>, 723–728 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">156. </div><div class="csl-right-inline">Stajich, J. E. <em>et al.</em> <a href="https://doi.org/10.1101/gr.361602">The Bioperl Toolkit: Perl Modules for the Life Sciences</a>. <em>Genome Research</em> <strong>12</strong>, 1611–1618 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">157. </div><div class="csl-right-inline">Gentleman, R. C. <em>et al.</em> <a href="https://doi.org/10.1186/gb-2004-5-10-r80">Bioconductor: open software development for computational biology and bioinformatics</a>. <em>Genome Biology</em> <strong>5</strong>, R80 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">158. </div><div class="csl-right-inline">Daily, J. <a href="https://doi.org/10.1186/s12859-016-0930-z">Parasail: SIMD c library for global, semi-global, and local pairwise sequence alignments</a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 81 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">159. </div><div class="csl-right-inline">Frohmberg, W., Kierzynka, M., Blazewicz, J. &amp; Wojciechowski, P. <a href="https://doi.org/10.2478/v10175-012-0062-1">G-PAS 2.0 <span></span> an improved version of protein alignment tool with an efficient backtracking routine on multiple GPUs</a>. <em>Bulletin of the Polish Academy of Sciences: Technical Sciences</em> <strong>60</strong>, 491–494 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">160. </div><div class="csl-right-inline">Altschul, S. F. Substitution Matrices. in (John Wiley &amp; Sons, Ltd, 2013). doi:<a href="https://doi.org/10.1002/9780470015902.a0005265.pub3">10.1002/9780470015902.a0005265.pub3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">161. </div><div class="csl-right-inline">Dayhoff, M. O., Schwartz, R. M. &amp; Orcutt, B. C. A model of evolutionary change in proteins. <em>A Model of Evolutionary Change in Proteins</em> 345–352 (1978).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">162. </div><div class="csl-right-inline">Müller, T. &amp; Vingron, M. <a href="https://doi.org/10.1089/10665270050514918">Modeling amino acid replacement</a>. <em>Journal of Computational Biology: A Journal of Computational Molecular Cell Biology</em> <strong>7</strong>, 761–776 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">163. </div><div class="csl-right-inline">Henikoff, S. &amp; Henikoff, J. G. <a href="https://doi.org/10.1073/pnas.89.22.10915">Amino acid substitution matrices from protein blocks</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>89</strong>, 10915–10919 (1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">164. </div><div class="csl-right-inline">Whelan, S. &amp; Goldman, N. <a href="https://doi.org/10.1093/oxfordjournals.molbev.a003851">A general empirical model of protein evolution derived from multiple protein families using a maximum-likelihood approach</a>. <em>Molecular Biology and Evolution</em> <strong>18</strong>, 691–699 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">165. </div><div class="csl-right-inline">Le, S. Q. &amp; Gascuel, O. <a href="https://doi.org/10.1093/molbev/msn067">An improved general amino acid replacement matrix</a>. <em>Molecular Biology and Evolution</em> <strong>25</strong>, 1307–1320 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">166. </div><div class="csl-right-inline">Müller, T., Rahmann, S. &amp; Rehmsmeier, M. <a href="https://doi.org/10.1093/bioinformatics/17.suppl_1.S182">Non-symmetric score matrices and the detection of homologous transmembrane proteins</a>. <em>Bioinformatics</em> <strong>17</strong>, S182–S189 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">167. </div><div class="csl-right-inline">Ng, P. C., Henikoff, J. G. &amp; Henikoff, S. <a href="https://doi.org/10.1093/bioinformatics/16.9.760">PHAT: a transmembrane-specific substitution matrix. Predicted hydrophobic and transmembrane</a>. <em>Bioinformatics (Oxford, England)</em> <strong>16</strong>, 760–766 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">168. </div><div class="csl-right-inline">Trivedi, R. &amp; Nagarajaram, H. A. <a href="https://doi.org/10.1038/s41598-019-52532-8">Amino acid substitution scoring matrices specific to intrinsically disordered regions in proteins</a>. <em>Scientific Reports</em> <strong>9</strong>, 16380 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">169. </div><div class="csl-right-inline">Goonesekere, N. C. W. &amp; Lee, B. <a href="https://doi.org/10.1002/prot.21775">Context-specific amino acid substitution matrices and their use in the detection of protein homologs</a>. <em>Proteins: Structure, Function, and Bioinformatics</em> <strong>71</strong>, 910–919 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">170. </div><div class="csl-right-inline">Paila, U., Kondam, R. &amp; Ranjan, A. <a href="https://doi.org/10.1093/nar/gkn635">Genome bias influences amino acid choices: analysis of amino acid substitution and re-compilation of substitution matrices exclusive to an AT-biased genome</a>. <em>Nucleic Acids Research</em> <strong>36</strong>, 6664–6675 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">171. </div><div class="csl-right-inline">Nickle, D. C. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0000503">HIV-specific probabilistic models of protein evolution</a>. <em>PLoS ONE</em> <strong>2</strong>, e503 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">172. </div><div class="csl-right-inline">Sardiu, M. E., Alves, G. &amp; Yu, Y.-K. <a href="https://doi.org/10.1103/PhysRevE.72.061917">Score statistics of global sequence alignment from the energy distribution of a modified directed polymer and directed percolation problem</a>. <em>Physical Review. E, Statistical, Nonlinear, and Soft Matter Physics</em> <strong>72</strong>, 061917 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">173. </div><div class="csl-right-inline">Chiaromonte, F., Yap, V. B. &amp; Miller, W. Scoring pairwise genomic sequence alignments. in 115–126 (WORLD SCIENTIFIC, 2001). doi:<a href="https://doi.org/10.1142/9789812799623_0012">10.1142/9789812799623_0012</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">174. </div><div class="csl-right-inline">Schneider, A., Cannarozzi, G. M. &amp; Gonnet, G. H. <a href="https://doi.org/10.1186/1471-2105-6-134">Empirical codon substitution matrix</a>. <em>BMC bioinformatics</em> <strong>6</strong>, 134 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">175. </div><div class="csl-right-inline">Doron-Faigenboim, A. &amp; Pupko, T. <a href="https://doi.org/10.1093/molbev/msl175">A combined empirical and mechanistic codon model</a>. <em>Molecular Biology and Evolution</em> <strong>24</strong>, 388–397 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">176. </div><div class="csl-right-inline">Cartwright, R. A. <a href="https://doi.org/10.1093/molbev/msn275">Problems and solutions for estimating indel rates and length distributions</a>. <em>Molecular Biology and Evolution</em> <strong>26</strong>, 473–480 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">177. </div><div class="csl-right-inline">Fitch, W. M. &amp; Smith, T. F. <a href="https://doi.org/10.1073/pnas.80.5.1382">Optimal sequence alignments</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>80</strong>, 1382–1386 (1983).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">178. </div><div class="csl-right-inline">Waterman, M. S., Smith, T. F. &amp; Beyer, W. A. <a href="https://doi.org/10.1016/0001-8708(76)90202-4">Some biological sequence metrics</a>. <em>Advances in Mathematics</em> <strong>20</strong>, 367–387 (1976).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">179. </div><div class="csl-right-inline">Gotoh, O. <a href="https://doi.org/10.1016/0022-2836(82)90398-9">An improved algorithm for matching biological sequences</a>. <em>Journal of Molecular Biology</em> <strong>162</strong>, 705–708 (1982).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">180. </div><div class="csl-right-inline">Altschul, S. F. &amp; Erickson, B. W. <a href="https://doi.org/10.1016/S0092-8240(86)90010-8">Optimal sequence alignment using affine gap costs</a>. <em>Bulletin of Mathematical Biology</em> <strong>48</strong>, 603–616 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">181. </div><div class="csl-right-inline">Waterman, M. S. <a href="https://doi.org/10.1016/S0022-5193(84)80037-5">Efficient sequence alignment algorithms</a>. <em>Journal of Theoretical Biology</em> <strong>108</strong>, 333–337 (1984).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">182. </div><div class="csl-right-inline">Miller, W. &amp; Myers, E. W. <a href="https://doi.org/10.1007/BF02459948">Sequence comparison with concave weighting functions</a>. <em>Bulletin of Mathematical Biology</em> <strong>50</strong>, 97–120 (1988).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">183. </div><div class="csl-right-inline">Cartwright, R. A. <a href="https://doi.org/10.1186/1471-2105-7-527">Logarithmic gap costs decrease alignment accuracy</a>. <em>BMC Bioinformatics</em> <strong>7</strong>, 527 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">184. </div><div class="csl-right-inline">Goonesekere, N. C. W. &amp; Lee, B. <a href="https://doi.org/10.1093/nar/gkh610">Frequency of gaps observed in a structurally aligned protein pair database suggests a simple gap penalty function</a>. <em>Nucleic Acids Research</em> <strong>32</strong>, 2838–2843 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">185. </div><div class="csl-right-inline">Benner, S. A., Cohen, M. A. &amp; Gonnet, G. H. <a href="https://doi.org/10.1006/jmbi.1993.1105">Empirical and Structural Models for Insertions and Deletions in the Divergent Evolution of Proteins</a>. <em>Journal of Molecular Biology</em> <strong>229</strong>, 1065–1082 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">186. </div><div class="csl-right-inline">Wrabl, J. O. &amp; Grishin, N. V. <a href="https://doi.org/10.1002/prot.10508">Gaps in structurally similar proteins: Towards improvement of multiple sequence alignment</a>. <em>Proteins: Structure, Function, and Bioinformatics</em> <strong>54</strong>, 71–87 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">187. </div><div class="csl-right-inline">Zhang, W., Liu, S. &amp; Zhou, Y. <a href="https://doi.org/10.1371/journal.pone.0002325">SP5: Improving Protein Fold Recognition by Using Torsion Angle Profiles and Profile-Based Gap Penalty Model</a>. <em>PLOS ONE</em> <strong>3</strong>, e2325 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">188. </div><div class="csl-right-inline">Jeanmougin, F., Thompson, J. D., Gouy, M., Higgins, D. G. &amp; Gibson, T. J. <a href="https://doi.org/10.1016/S0968-0004(98)01285-7">Multiple sequence alignment with clustal x</a>. <em>Trends in Biochemical Sciences</em> <strong>23</strong>, 403–405 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">189. </div><div class="csl-right-inline">Wang, C., Yan, R.-X., Wang, X.-F., Si, J.-N. &amp; Zhang, Z. <a href="https://doi.org/10.1016/j.compbiolchem.2011.07.006">Comparison of linear gap penalties and profile-based variable gap penalties in profile<span></span>profile alignments</a>. <em>Computational Biology and Chemistry</em> <strong>35</strong>, 308–318 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">190. </div><div class="csl-right-inline">Marco-Sola, S., Moure, J. C., Moreto, M. &amp; Espinosa, A. Fast gap-affine pairwise alignment using the wavefront algorithm. <em>Bioinformatics</em> (2020) doi:<a href="https://doi.org/10.1093/bioinformatics/btaa777">10.1093/bioinformatics/btaa777</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">191. </div><div class="csl-right-inline">Pearson, W. R. &amp; Miller, W. <a href="https://doi.org/10.1016/0076-6879(92)10029-D">[27] Dynamic programming algorithms for biological sequence comparison</a>. in vol. 210 575–601 (Academic Press, 1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">192. </div><div class="csl-right-inline">Spouge, J. L. <a href="https://doi.org/10.1137/0149094">Speeding up dynamic programming algorithms for finding optimal lattice paths</a>. <em>SIAM Journal on Applied Mathematics</em> <strong>49</strong>, 1552–1566 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">193. </div><div class="csl-right-inline">Fickett, J. W. <a href="https://doi.org/10.1093/nar/12.1Part1.175">Fast optimal alignment</a>. <em>Nucleic Acids Research</em> <strong>12</strong>, 175–179 (1984).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">194. </div><div class="csl-right-inline">Chao, J., Tang, F. &amp; Xu, L. <a href="https://doi.org/10.3390/biom12040546">Developments in Algorithms for Sequence Alignment: A Review</a>. <em>Biomolecules</em> <strong>12</strong>, 546 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">195. </div><div class="csl-right-inline">Katoh, K., Misawa, K., Kuma, K. &amp; Miyata, T. <a href="https://doi.org/10.1093/nar/gkf436">MAFFT: A novel method for rapid multiple sequence alignment based on fast fourier transform</a>. <em>Nucleic Acids Research</em> <strong>30</strong>, 3059–3066 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">196. </div><div class="csl-right-inline">Sun, Y. &amp; Buhler, J. <a href="https://doi.org/10.1186/1471-2105-7-133">Choosing the best heuristic for seeded alignment of DNA sequences</a>. <em>BMC Bioinformatics</em> <strong>7</strong>, 133 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">197. </div><div class="csl-right-inline">Li, H. &amp; Homer, N. <a href="https://doi.org/10.1093/bib/bbq015">A survey of sequence alignment algorithms for next-generation sequencing</a>. <em>Briefings in Bioinformatics</em> <strong>11</strong>, 473–483 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">198. </div><div class="csl-right-inline">Altschul, S. F., Gish, W., Miller, W., Myers, E. W. &amp; Lipman, D. J. <a href="https://doi.org/10.1016/S0022-2836(05)80360-2">Basic local alignment search tool</a>. <em>Journal of Molecular Biology</em> <strong>215</strong>, 403–410 (1990).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">199. </div><div class="csl-right-inline">Altschul, S. F. <em>et al.</em> <a href="https://doi.org/10.1093/nar/25.17.3389">Gapped BLAST and PSI-BLAST: A new generation of protein database search programs</a>. <em>Nucleic Acids Research</em> <strong>25</strong>, 3389–3402 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">200. </div><div class="csl-right-inline">Schwartz, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.809403">Human<span></span>Mouse Alignments with BLASTZ</a>. <em>Genome Research</em> <strong>13</strong>, 103–107 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">201. </div><div class="csl-right-inline">Ma, B., Tromp, J. &amp; Li, M. <a href="https://doi.org/10.1093/bioinformatics/18.3.440">PatternHunter: Faster and more sensitive homology search</a>. <em>Bioinformatics</em> <strong>18</strong>, 440–445 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">202. </div><div class="csl-right-inline">Edgar, R. C. <a href="https://doi.org/10.1093/bioinformatics/btq461">Search and clustering orders of magnitude faster than BLAST</a>. <em>Bioinformatics</em> <strong>26</strong>, 2460–2461 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">203. </div><div class="csl-right-inline">Buchfink, B., Xie, C. &amp; Huson, D. H. <a href="https://doi.org/10.1038/nmeth.3176">Fast and sensitive protein alignment using DIAMOND</a>. <em>Nature Methods</em> <strong>12</strong>, 59–60 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">204. </div><div class="csl-right-inline">Buchfink, B., Reuter, K. &amp; Drost, H.-G. <a href="https://doi.org/10.1038/s41592-021-01101-x">Sensitive protein alignments at tree-of-life scale using DIAMOND</a>. <em>Nature Methods</em> <strong>18</strong>, 366–368 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">205. </div><div class="csl-right-inline">Pearson, W. R. &amp; Lipman, D. J. <a href="https://doi.org/10.1073/pnas.85.8.2444">Improved tools for biological sequence comparison</a>. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>85</strong>, 2444–2448 (1988).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">206. </div><div class="csl-right-inline">Lipman, D. J. &amp; Pearson, W. R. <a href="https://doi.org/10.1126/science.2983426">Rapid and sensitive protein similarity searches</a>. <em>Science (New York, N.Y.)</em> <strong>227</strong>, 1435–1441 (1985).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">207. </div><div class="csl-right-inline">Saripella, G. V., Sonnhammer, E. L. L. &amp; Forslund, K. <a href="https://doi.org/10.1093/bioinformatics/btw305">Benchmarking the next generation of homology inference tools</a>. <em>Bioinformatics</em> <strong>32</strong>, 2636 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">208. </div><div class="csl-right-inline">Finn, R. D. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkv1344">The Pfam protein families database: towards a more sustainable future</a>. <em>Nucleic Acids Research</em> <strong>44</strong>, D279 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">209. </div><div class="csl-right-inline">Essoussi, N. &amp; Fayech, S. A comparison of four pair-wise sequence alignment methods. <em>Bioinformation</em> <strong>2</strong>, 166–168 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">210. </div><div class="csl-right-inline">Shpaer, E. G. <em>et al.</em> <a href="https://doi.org/10.1006/geno.1996.0614">Sensitivity and Selectivity in Protein Similarity Searches: A Comparison of Smith<span></span>Waterman in Hardware to BLAST and FASTA</a>. <em>Genomics</em> <strong>38</strong>, 179–191 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">211. </div><div class="csl-right-inline">Schleimer, S., Wilkerson, D. S. &amp; Aiken, A. Winnowing: Local algorithms for document fingerprinting. in 7685 (Association for Computing Machinery, 2003). doi:<a href="https://doi.org/10.1145/872757.872770">10.1145/872757.872770</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">212. </div><div class="csl-right-inline">Roberts, M., Hayes, W., Hunt, B. R., Mount, S. M. &amp; Yorke, J. A. <a href="https://doi.org/10.1093/bioinformatics/bth408">Reducing storage requirements for biological sequence comparison</a>. <em>Bioinformatics</em> <strong>20</strong>, 3363–3369 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">213. </div><div class="csl-right-inline">Li, H. <a href="https://doi.org/10.1093/bioinformatics/btw152">Minimap and miniasm: Fast mapping and de novo assembly for noisy long sequences</a>. <em>Bioinformatics</em> <strong>32</strong>, 2103–2110 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">214. </div><div class="csl-right-inline">Jain, C., Koren, S., Dilthey, A., Phillippy, A. M. &amp; Aluru, S. <a href="https://doi.org/10.1093/bioinformatics/bty597">A fast adaptive algorithm for computing whole-genome homology maps</a>. <em>Bioinformatics</em> <strong>34</strong>, i748–i756 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">215. </div><div class="csl-right-inline">Orenstein, Y., Pellow, D., Marçais, G., Shamir, R. &amp; Kingsford, C. Compact Universal k-mer Hitting Sets. in (eds. Frith, M. &amp; Storm Pedersen, C. N.) 257–268 (Springer International Publishing, 2016). doi:<a href="https://doi.org/10.1007/978-3-319-43681-4_21">10.1007/978-3-319-43681-4_21</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">216. </div><div class="csl-right-inline">Marçais, G. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btx235">Improving the performance of minimizers and winnowing schemes</a>. <em>Bioinformatics</em> <strong>33</strong>, i110–i117 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">217. </div><div class="csl-right-inline">Chikhi, R., Limasset, A., Jackman, S., Simpson, J. T. &amp; Medvedev, P. On the Representation of de Bruijn Graphs. in 35–55 (Springer International Publishing, 2014). doi:<a href="https://doi.org/10.1007/978-3-319-05269-4_4">10.1007/978-3-319-05269-4_4</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">218. </div><div class="csl-right-inline">Edgar, R. <a href="https://doi.org/10.7717/peerj.10805">Syncmers are more sensitive than minimizers for selecting conserved k<span>-</span>mers in biological sequences</a>. <em>PeerJ</em> <strong>9</strong>, e10805 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">219. </div><div class="csl-right-inline">Sahlin, K. <a href="https://doi.org/10.1101/gr.275648.121">Effective sequence similarity detection with strobemers</a>. <em>Genome Research</em> <strong>31</strong>, 2080–2094 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">220. </div><div class="csl-right-inline">Sahlin, K. Flexible seed size enables ultra-fast and accurate read alignment. doi:<a href="https://doi.org/10.1101/2021.06.18.449070">10.1101/2021.06.18.449070</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">221. </div><div class="csl-right-inline">Weiner, P. 14th annual symposium on switching and automata theory (swat 1973). in 1–11 (1973). doi:<a href="https://doi.org/10.1109/SWAT.1973.13">10.1109/SWAT.1973.13</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">222. </div><div class="csl-right-inline">Manber, U. &amp; Myers, G. <a href="https://doi.org/10.1137/0222058">Suffix arrays: A new method for on-line string searches</a>. <em>SIAM Journal on Computing</em> <strong>22</strong>, 935–948 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">223. </div><div class="csl-right-inline">Abouelhoda, M. I., Kurtz, S. &amp; Ohlebusch, E. The Enhanced Suffix Array and Its Applications to Genome Analysis. in (eds. Guigó, R. &amp; Gusfield, D.) 449–463 (Springer, 2002). doi:<a href="https://doi.org/10.1007/3-540-45784-4_35">10.1007/3-540-45784-4_35</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">224. </div><div class="csl-right-inline">Ferragina, P. &amp; Manzini, G. Proceedings 41st annual symposium on foundations of computer science. in 390–398 (2000). doi:<a href="https://doi.org/10.1109/SFCS.2000.892127">10.1109/SFCS.2000.892127</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">225. </div><div class="csl-right-inline">Bray, N., Dubchak, I. &amp; Pachter, L. <a href="https://doi.org/10.1101/gr.789803">AVID: A global alignment program</a>. <em>Genome Research</em> <strong>13</strong>, 97–102 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">226. </div><div class="csl-right-inline">Delcher, A. L., Phillippy, A., Carlton, J. &amp; Salzberg, S. L. Fast algorithms for large-scale genome alignment and comparison. <em>Nucleic Acids Research</em> <strong>30</strong>, 2478–2483 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">227. </div><div class="csl-right-inline">Abouelhoda, M. I., Kurtz, S. &amp; Ohlebusch, E. <a href="https://doi.org/10.1016/S1570-8667(03)00065-0">Replacing suffix trees with enhanced suffix arrays</a>. <em>Journal of Discrete Algorithms</em> <strong>2</strong>, 53–86 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">228. </div><div class="csl-right-inline">Marçais, G. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pcbi.1005944">MUMmer4: A fast and versatile genome alignment system</a>. <em>PLOS Computational Biology</em> <strong>14</strong>, e1005944 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">229. </div><div class="csl-right-inline">Burrows, M. &amp; Wheeler, D. <em>A block-sorting lossless data compression algorithm</em>. (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">230. </div><div class="csl-right-inline">Vyverman, M., De Baets, B., Fack, V. &amp; Dawyndt, P. <a href="https://doi.org/10.1093/nar/gks408">Prospects and limitations of full-text index structures in genome analysis</a>. <em>Nucleic Acids Research</em> <strong>40</strong>, 6993–7015 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">231. </div><div class="csl-right-inline">Cheng, H., Wu, M. &amp; Xu, Y. <a href="https://doi.org/10.1093/bioinformatics/btx596">FMtree: A fast locating algorithm of FM-indexes for genomic data</a>. <em>Bioinformatics</em> <strong>34</strong>, 416–424 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">232. </div><div class="csl-right-inline">Lam, T. W., Sung, W. K., Tam, S. L., Wong, C. K. &amp; Yiu, S. M. <a href="https://doi.org/10.1093/bioinformatics/btn032">Compressed indexing and local alignment of DNA</a>. <em>Bioinformatics</em> <strong>24</strong>, 791–797 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">233. </div><div class="csl-right-inline">Li, H. &amp; Durbin, R. <a href="https://doi.org/10.1093/bioinformatics/btp324">Fast and accurate short read alignment with burrows<span></span>wheeler transform</a>. <em>Bioinformatics</em> <strong>25</strong>, 1754–1760 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">234. </div><div class="csl-right-inline">Li, H. &amp; Durbin, R. <a href="https://doi.org/10.1093/bioinformatics/btp698">Fast and accurate long-read alignment with burrows<span></span>wheeler transform</a>. <em>Bioinformatics</em> <strong>26</strong>, 589–595 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">235. </div><div class="csl-right-inline">Li, H. Aligning sequence reads, clone sequences and assembly contigs with <span>BWA</span>-<span>MEM</span>. <a href="https://arxiv.org/abs/1303.3997">https://arxiv.org/abs/1303.3997</a> (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">236. </div><div class="csl-right-inline">Liu, Y. &amp; Schmidt, B. <a href="https://doi.org/10.1093/bioinformatics/bts414">Long read alignment based on maximal exact match seeds</a>. <em>Bioinformatics</em> <strong>28</strong>, i318–i324 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">237. </div><div class="csl-right-inline">Langmead, B. &amp; Salzberg, S. L. <a href="https://doi.org/10.1038/nmeth.1923">Fast gapped-read alignment with <span>Bowtie</span> 2</a>. <em>Nat Methods</em> <strong>9</strong>, 357–359 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">238. </div><div class="csl-right-inline">Song, B. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.2113075119">AnchorWave: Sensitive alignment of genomes with high sequence diversity, extensive structural polymorphism, and whole-genome duplication</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>119</strong>, e2113075119 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">239. </div><div class="csl-right-inline">Durbin, R., Eddy, S. R., Krogh, A. &amp; Mitchison, G. <em>Biological sequence analysis: Probabilistic models of proteins and nucleic acids</em>. (Cambridge University Press, 1998). doi:<a href="https://doi.org/10.1017/CBO9780511790492">10.1017/CBO9780511790492</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">240. </div><div class="csl-right-inline">Söding, J. <a href="https://doi.org/10.1093/bioinformatics/bti125">Protein homology detection by HMM-HMM comparison</a>. <em>Bioinformatics (Oxford, England)</em> <strong>21</strong>, 951–960 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">241. </div><div class="csl-right-inline">Finn, R. D., Clements, J. &amp; Eddy, S. R. <a href="https://doi.org/10.1093/nar/gkr367">HMMER web server: Interactive sequence similarity searching</a>. <em>Nucleic Acids Research</em> <strong>39</strong>, W29–W37 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">242. </div><div class="csl-right-inline">Wang, J., Keightley, P. D. &amp; Johnson, T. <a href="https://doi.org/10.1186/1471-2105-7-292">MCALIGN2: Faster, accurate global pairwise alignment of non-coding DNA sequences based on explicit models of indel evolution</a>. <em>BMC Bioinformatics</em> <strong>7</strong>, 292 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">243. </div><div class="csl-right-inline">Ruffalo, M., LaFramboise, T. &amp; Koyutürk, M. <a href="https://doi.org/10.1093/bioinformatics/btr477">Comparative analysis of algorithms for next-generation sequencing read alignment</a>. <em>Bioinformatics (Oxford, England)</em> <strong>27</strong>, 2790–2796 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">244. </div><div class="csl-right-inline">Schbath, S. <em>et al.</em> <a href="https://doi.org/10.1089/cmb.2012.0022">Mapping reads on a genomic sequence: An algorithmic overview and a practical comparative analysis</a>. <em>Journal of Computational Biology</em> <strong>19</strong>, 796–813 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">245. </div><div class="csl-right-inline">Hatem, A., Bozdağ, D., Toland, A. E. &amp; Çatalyürek, Ü. V. <a href="https://doi.org/10.1186/1471-2105-14-184">Benchmarking short sequence mapping tools</a>. <em>BMC Bioinformatics</em> <strong>14</strong>, 184 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">246. </div><div class="csl-right-inline">Canzar, S. &amp; Salzberg, S. L. <a href="https://doi.org/10.1109/JPROC.2015.2455551">Short read mapping: An algorithmic tour</a>. <em>Proceedings of the IEEE</em> <strong>105</strong>, 436–458 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">247. </div><div class="csl-right-inline">Alser, M. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-021-02443-7">Technology dictates algorithms: Recent developments in read alignment</a>. <em>Genome Biology</em> <strong>22</strong>, 249 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">248. </div><div class="csl-right-inline">Břinda, K., Boeva, V. &amp; Kucherov, G. <a href="https://doi.org/10.1093/bioinformatics/btv524">RNF: A general framework to evaluate NGS read mappers</a>. <em>Bioinformatics</em> <strong>32</strong>, 136–139 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">249. </div><div class="csl-right-inline">Lin, H.-N. &amp; Hsu, W.-L. <a href="https://doi.org/10.1093/bioinformatics/btx189">Kart: a divide-and-conquer algorithm for NGS read alignment</a>. <em>Bioinformatics (Oxford, England)</em> <strong>33</strong>, 2281–2287 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">250. </div><div class="csl-right-inline">Olson, C. B. <em>et al.</em> 2012 IEEE 20th international symposium on field-programmable custom computing machines. in 161–168 (2012). doi:<a href="https://doi.org/10.1109/FCCM.2012.36">10.1109/FCCM.2012.36</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">251. </div><div class="csl-right-inline">Chen, P., Wang, C., Li, X. &amp; Zhou, X. <a href="https://doi.org/10.1109/TCBB.2014.2326876">Accelerating the next generation long read mapping with the FPGA-based system</a>. <em>IEEE/ACM Transactions on Computational Biology and Bioinformatics</em> <strong>11</strong>, 840–852 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">252. </div><div class="csl-right-inline">Suzuki, H. &amp; Kasahara, M. <a href="https://doi.org/10.1186/s12859-018-2014-8">Introducing difference recurrence relations for faster semi-global alignment of long sequences</a>. <em>BMC Bioinformatics</em> <strong>19</strong>, 45 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">253. </div><div class="csl-right-inline">Zeni, A. <em>et al.</em> 2020 IEEE international parallel and distributed processing symposium (IPDPS). in 462–471 (2020). doi:<a href="https://doi.org/10.1109/IPDPS47924.2020.00055">10.1109/IPDPS47924.2020.00055</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">254. </div><div class="csl-right-inline">Chaisson, M. J. &amp; Tesler, G. <a href="https://doi.org/10.1186/1471-2105-13-238">Mapping single molecule sequencing reads using basic local alignment with successive refinement (BLASR): Application and theory</a>. <em>BMC Bioinformatics</em> <strong>13</strong>, 238 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">255. </div><div class="csl-right-inline">Haghshenas, E., Sahinalp, S. C. &amp; Hach, F. <a href="https://doi.org/10.1093/bioinformatics/bty544">lordFAST: sensitive and Fast Alignment Search Tool for LOng noisy Read sequencing Data</a>. <em>Bioinformatics (Oxford, England)</em> <strong>35</strong>, 20–27 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">256. </div><div class="csl-right-inline">Sović, I. <em>et al.</em> <a href="https://doi.org/10.1038/ncomms11307">Fast and sensitive mapping of nanopore sequencing reads with GraphMap</a>. <em>Nature Communications</em> <strong>7</strong>, 11307 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">257. </div><div class="csl-right-inline">Sedlazeck, F. J. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-018-0001-7">Accurate detection of complex structural variations using single-molecule sequencing</a>. <em>Nature Methods</em> <strong>15</strong>, 461–468 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">258. </div><div class="csl-right-inline">Jain, C., Dilthey, A., Koren, S., Aluru, S. &amp; Phillippy, A. M. <a href="https://doi.org/10.1089/cmb.2018.0036">A fast approximate algorithm for mapping long reads to large reference databases</a>. <em>Journal of Computational Biology</em> <strong>25</strong>, 766–779 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">259. </div><div class="csl-right-inline">Prodanov, T. &amp; Bansal, V. <a href="https://doi.org/10.1093/nar/gkaa829">Sensitive alignment using paralogous sequence variants improves long-read mapping and variant calling in segmental duplications</a>. <em>Nucleic Acids Research</em> <strong>48</strong>, e114 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">260. </div><div class="csl-right-inline">Jain, C., Rhie, A., Hansen, N. F., Koren, S. &amp; Phillippy, A. M. <a href="https://doi.org/10.1038/s41592-022-01457-8">Long-read mapping to repetitive reference sequences using Winnowmap2</a>. <em>Nature Methods</em> <strong>19</strong>, 705–710 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">261. </div><div class="csl-right-inline">Mikheenko, A., Bzikadze, A. V., Gurevich, A., Miga, K. H. &amp; Pevzner, P. A. <a href="https://doi.org/10.1093/bioinformatics/btaa440"><span>TandemTools</span>: Mapping long reads and assessing/improving assembly quality in extra-long tandem repeats</a>. <em>Bioinformatics</em> <strong>36</strong>, i75–i83 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">262. </div><div class="csl-right-inline">Li, H., Ruan, J. &amp; Durbin, R. <a href="https://doi.org/10.1101/gr.078212.108">Mapping short DNA sequencing reads and calling variants using mapping quality scores</a>. <em>Genome Research</em> <strong>18</strong>, 1851–1858 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">263. </div><div class="csl-right-inline">Li, H. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btp352">The sequence alignment/map format and SAMtools</a>. <em>Bioinformatics</em> <strong>25</strong>, 2078–2079 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">264. </div><div class="csl-right-inline"><a href="http://www.acgt.me/blog/2014/12/16/understanding-mapq-scores-in-sam-files-does-37-42">Understanding MAPQ scores in SAM files: Does 37 = 42?</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">265. </div><div class="csl-right-inline">Lee, H. &amp; Schatz, M. C. <a href="https://doi.org/10.1093/bioinformatics/bts330">Genomic dark matter: The reliability of short read mapping illustrated by the genome mappability score</a>. <em>Bioinformatics</em> <strong>28</strong>, 2097–2105 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">266. </div><div class="csl-right-inline">Langmead, B. <a href="https://doi.org/10.1186/s13059-017-1290-3">A tandem simulation framework for predicting mapping quality</a>. <em>Genome Biology</em> <strong>18</strong>, 152 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">267. </div><div class="csl-right-inline">Ruffalo, M., Koyutürk, M., Ray, S. &amp; LaFramboise, T. <a href="https://doi.org/10.1093/bioinformatics/bts408">Accurate estimation of short read mapping quality for next-generation genome sequencing</a>. <em>Bioinformatics</em> <strong>28</strong>, i349–i355 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">268. </div><div class="csl-right-inline"><em><a href="https://doi.org/10.1007/978-1-62703-646-7">Multiple Sequence Alignment Methods</a></em>. vol. 1079 (Humana Press, 2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">269. </div><div class="csl-right-inline">Wang, L. &amp; Jiang, T. <a href="https://doi.org/10.1089/cmb.1994.1.337">On the complexity of multiple sequence alignment</a>. <em>Journal of Computational Biology</em> <strong>1</strong>, 337–348 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">270. </div><div class="csl-right-inline">Just, W. <a href="https://doi.org/10.1089/106652701753307511">Computational complexity of multiple sequence alignment with SP-score</a>. <em>Journal of Computational Biology</em> <strong>8</strong>, 615–623 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">271. </div><div class="csl-right-inline">Tang, F. <em>et al.</em> <a href="https://doi.org/10.1093/molbev/msac166">HAlign 3: Fast multiple alignment of ultra-large numbers of similar DNA/RNA sequences</a>. <em>Molecular Biology and Evolution</em> <strong>39</strong>, msac166 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">272. </div><div class="csl-right-inline">Feng, D.-F. &amp; Doolittle, R. F. <a href="https://doi.org/10.1007/BF02603120">Progressive sequence alignment as a prerequisitetto correct phylogenetic trees</a>. <em>Journal of Molecular Evolution</em> <strong>25</strong>, 351–360 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">273. </div><div class="csl-right-inline">Jones, D. T., Taylor, W. R. &amp; Thornton, J. M. <a href="https://doi.org/10.1093/bioinformatics/8.3.275">The rapid generation of mutation data matrices from protein sequences</a>. <em>Bioinformatics</em> <strong>8</strong>, 275–282 (1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">274. </div><div class="csl-right-inline">Blaisdell, B. E. <a href="https://doi.org/10.1073/pnas.83.14.5155">A measure of the similarity of sets of sequences not requiring sequence alignment.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>83</strong>, 5155–5159 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">275. </div><div class="csl-right-inline">Gronau, I. &amp; Moran, S. <a href="https://doi.org/10.1016/j.ipl.2007.07.002">Optimal implementations of UPGMA and other common clustering algorithms</a>. <em>Information Processing Letters</em> <strong>104</strong>, 205–210 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">276. </div><div class="csl-right-inline">Saitou, N. &amp; Nei, M. <a href="https://doi.org/10.1093/oxfordjournals.molbev.a040454">The neighbor-joining method: A new method for reconstructing phylogenetic trees.</a> <em>Molecular Biology and Evolution</em> <strong>4</strong>, 406–425 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">277. </div><div class="csl-right-inline">Katoh, K. &amp; Toh, H. <a href="https://doi.org/10.1093/bioinformatics/btl592">PartTree: An algorithm to build an approximate tree from a large number of unaligned sequences</a>. <em>Bioinformatics</em> <strong>23</strong>, 372–374 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">278. </div><div class="csl-right-inline">Sievers, F. <em>et al.</em> <a href="https://doi.org/10.1038/msb.2011.75">Fast, scalable generation of high<span>-</span>quality protein multiple sequence alignments using Clustal Omega</a>. <em>Molecular Systems Biology</em> <strong>7</strong>, 539 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">279. </div><div class="csl-right-inline">Blackshields, G., Sievers, F., Shi, W., Wilm, A. &amp; Higgins, D. G. <a href="https://doi.org/10.1186/1748-7188-5-21">Sequence embedding for fast construction of guide trees for multiple sequence alignment</a>. <em>Algorithms for Molecular Biology</em> <strong>5</strong>, 21 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">280. </div><div class="csl-right-inline">Altschul, S. F. <a href="https://doi.org/10.1016/S0022-5193(89)80196-1">Gap costs for multiple sequence alignment</a>. <em>Journal of Theoretical Biology</em> <strong>138</strong>, 297–309 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">281. </div><div class="csl-right-inline">Altschul, S. F., Carroll, R. J. &amp; Lipman, D. J. <a href="https://doi.org/10.1016/0022-2836(89)90234-9">Weights for data related by a tree</a>. <em>Journal of Molecular Biology</em> <strong>207</strong>, 647–653 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">282. </div><div class="csl-right-inline">Edgar, R. C. &amp; Sjölander, K. <a href="https://doi.org/10.1093/bioinformatics/bth090">A comparison of scoring functions for protein sequence profile alignment</a>. <em>Bioinformatics</em> <strong>20</strong>, 1301–1308 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">283. </div><div class="csl-right-inline">Notredame, C., Holm, L. &amp; Higgins, D. G. <a href="https://doi.org/10.1093/bioinformatics/14.5.407">COFFEE: An objective function for multiple sequence alignments.</a> <em>Bioinformatics</em> <strong>14</strong>, 407–422 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">284. </div><div class="csl-right-inline">Notredame, C., Higgins, D. G. &amp; Heringa, J. <a href="https://doi.org/10.1006/jmbi.2000.4042">T-coffee: a novel method for fast and accurate multiple sequence alignment11Edited by J. Thornton</a>. <em>Journal of Molecular Biology</em> <strong>302</strong>, 205–217 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">285. </div><div class="csl-right-inline">Edgar, R. C. <a href="https://doi.org/10.1186/1471-2105-5-113">MUSCLE: A multiple sequence alignment method with reduced time and space complexity</a>. <em>BMC Bioinformatics</em> <strong>5</strong>, 113 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">286. </div><div class="csl-right-inline">Edgar, R. C. <a href="https://doi.org/10.1093/nar/gkh340">MUSCLE: Multiple sequence alignment with high accuracy and high throughput</a>. <em>Nucleic Acids Research</em> <strong>32</strong>, 1792–1797 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">287. </div><div class="csl-right-inline">Do, C. B., Mahabhashyam, M. S. P., Brudno, M. &amp; Batzoglou, S. <a href="https://doi.org/10.1101/gr.2821705">ProbCons: Probabilistic consistency-based multiple sequence alignment</a>. <em>Genome Research</em> <strong>15</strong>, 330–340 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">288. </div><div class="csl-right-inline">Thompson, J. D., Higgins, D. G. &amp; Gibson, T. J. <a href="https://doi.org/10.1093/nar/22.22.4673">CLUSTAL W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap penalties and weight matrix choice</a>. <em>Nucleic Acids Research</em> <strong>22</strong>, 4673–4680 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">289. </div><div class="csl-right-inline">Thompson, J. D., Gibson, T. J., Plewniak, F., Jeanmougin, F. &amp; Higgins, D. G. <a href="https://doi.org/10.1093/nar/25.24.4876">The CLUSTAL<span>_</span>x windows interface: Flexible strategies for multiple sequence alignment aided by quality analysis tools</a>. <em>Nucleic Acids Research</em> <strong>25</strong>, 4876–4882 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">290. </div><div class="csl-right-inline">Liu, Y., Schmidt, B. &amp; Maskell, D. L. <a href="https://doi.org/10.1093/bioinformatics/btq338">MSAProbs: Multiple sequence alignment based on pair hidden markov models and partition function posterior probabilities</a>. <em>Bioinformatics</em> <strong>26</strong>, 1958–1964 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">291. </div><div class="csl-right-inline">Lemoine, F., Blassel, L., Voznica, J. &amp; Gascuel, O. COVID-align: Accurate online alignment of hCoV-19 genomes using a profile HMM. <em>Bioinformatics</em> (2020) doi:<a href="https://doi.org/10.1093/bioinformatics/btaa871">10.1093/bioinformatics/btaa871</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">292. </div><div class="csl-right-inline">Eddy, S. R. Multiple Alignment Using Hidden Markov Models. 7.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">293. </div><div class="csl-right-inline">Kim, J., Pramanik, S. &amp; Chung, M. J. <a href="https://doi.org/10.1093/bioinformatics/10.4.419">Multiple sequence alignment using simulated annealing</a>. <em>Bioinformatics</em> <strong>10</strong>, 419–426 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">294. </div><div class="csl-right-inline">Ishikawa, M. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/9.3.267">Multiple sequence alignment by parallel simulated annealing</a>. <em>Bioinformatics</em> <strong>9</strong>, 267–273 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">295. </div><div class="csl-right-inline">Huo, H. &amp; Stojkovic, V. <a href="https://doi.org/10.1109/ICNC.2007.139">Third international conference on natural computation (ICNC 2007)</a>. in vol. 2 270–274 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">296. </div><div class="csl-right-inline">Chowdhury, B. &amp; Garai, G. <a href="https://doi.org/10.1016/j.ygeno.2017.06.007">A review on multiple sequence alignment from the perspective of genetic algorithm</a>. <em>Genomics</em> <strong>109</strong>, 419–431 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">297. </div><div class="csl-right-inline">Zhang, C. &amp; Wong, A. K. C. <a href="https://doi.org/10.1093/bioinformatics/13.6.565">A genetic algorithm for multiple molecular sequence alignment</a>. <em>Bioinformatics</em> <strong>13</strong>, 565–581 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">298. </div><div class="csl-right-inline">Naznin, F., Sarker, R. &amp; Essam, D. <a href="https://doi.org/10.1186/1471-2105-12-353">Vertical decomposition with genetic algorithm for multiple sequence alignment</a>. <em>BMC Bioinformatics</em> <strong>12</strong>, 353 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">299. </div><div class="csl-right-inline">Naznin, F., Sarker, R. &amp; Essam, D. <a href="https://doi.org/10.1109/TEVC.2011.2162849">Progressive alignment method using genetic algorithm for multiple sequence alignment</a>. <em>IEEE Transactions on Evolutionary Computation</em> <strong>16</strong>, 615–631 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">300. </div><div class="csl-right-inline">Notredame, C. &amp; Higgins, D. G. SAGA: Sequence alignment by genetic algorithm. <em>Nucleic Acids Research</em> <strong>24</strong>, 1515–1524 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">301. </div><div class="csl-right-inline">Aksamentov, I., Roemer, C., Hodcroft, E. &amp; Neher, R. <a href="https://doi.org/10.21105/joss.03773">Nextclade: clade assignment, mutation calling and quality control for viral genomes</a>. <em>Journal of Open Source Software</em> <strong>6</strong>, 3773 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">302. </div><div class="csl-right-inline">Garriga, E. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-019-0333-6">Large multiple sequence alignments with a root-to-leaf regressive method</a>. <em>Nature Biotechnology</em> <strong>37</strong>, 1466–1470 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">303. </div><div class="csl-right-inline">Notredame, C. <a href="https://doi.org/10.1371/journal.pcbi.0030123">Recent Evolutions of Multiple Sequence Alignment Algorithms</a>. <em>PLOS Computational Biology</em> <strong>3</strong>, e123 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">304. </div><div class="csl-right-inline">Notredame, C. <a href="https://doi.org/10.1517/14622416.3.1.131">Recent progress in multiple sequence alignment: A survey</a>. <em>Pharmacogenomics</em> <strong>3</strong>, 131–144 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">305. </div><div class="csl-right-inline">Edgar, R. C. &amp; Batzoglou, S. <a href="https://doi.org/10.1016/j.sbi.2006.04.004">Multiple sequence alignment</a>. <em>Current Opinion in Structural Biology</em> <strong>16</strong>, 368–373 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">306. </div><div class="csl-right-inline">Pais, F. S.-M., Ruy, P. de C., Oliveira, G. &amp; Coimbra, R. S. <a href="https://doi.org/10.1186/1748-7188-9-4">Assessing the efficiency of multiple sequence alignment programs</a>. <em>Algorithms for Molecular Biology</em> <strong>9</strong>, 4 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">307. </div><div class="csl-right-inline">Thompson, J. D., Plewniak, F. &amp; Poch, O. <a href="https://doi.org/10.1093/bioinformatics/15.1.87">BAliBASE: a benchmark alignment database for the evaluation of multiple alignment programs.</a> <em>Bioinformatics</em> <strong>15</strong>, 87–88 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">308. </div><div class="csl-right-inline">Bragg, L., Stone, G., Imelfort, M., Hugenholtz, P. &amp; Tyson, G. W. <a href="https://doi.org/10.1038/nmeth.1990">Fast, accurate error-correction of amplicon pyrosequences using <span>Acacia</span></a>. <em>Nat Methods</em> <strong>9</strong>, 425–426 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">309. </div><div class="csl-right-inline">Sahlin, K. &amp; Medvedev, P. <a href="https://doi.org/10.1038/s41467-020-20340-8">Error correction enables use of <span>Oxford Nanopore</span> technology for reference-free transcriptome analysis</a>. <em>Nat Commun</em> <strong>12</strong>, 2 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">310. </div><div class="csl-right-inline">Liu, H. <em>et al.</em> <a href="https://doi.org/10.46471/gigabyte.15"><span>SMARTdenovo</span>: A de novo assembler using long noisy reads</a>. <em>Gigabyte</em> <strong>2021</strong>, 1–9 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">311. </div><div class="csl-right-inline">Graham, R. L., Knuth, D. E. &amp; Patashnik, O. <em>Concrete mathematics: A foundation for computer science</em>. (<span>Addison-Wesley</span>, 1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">312. </div><div class="csl-right-inline">Adams, M. D. <em>et al.</em> <a href="https://doi.org/10.1126/science.287.5461.2185">The genome sequence of <span>Drosophila</span> melanogaster</a>. <em>Science</em> <strong>287</strong>, 2185–2195 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">313. </div><div class="csl-right-inline">Yang, C., Chu, J., Warren, R. L. &amp; Birol, I. <a href="https://doi.org/10.1093/gigascience/gix010"><span>NanoSim</span>: Nanopore sequence read simulator based on statistical characterization</a>. <em>GigaScience</em> <strong>6</strong>, (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">314. </div><div class="csl-right-inline">Rhie, A., Walenz, B. P., Koren, S. &amp; Phillippy, A. M. <a href="https://doi.org/10.1186/s13059-020-02134-9">Merqury: Reference-free quality, completeness, and phasing assessment for genome assemblies</a>. <em>Genome Biology</em> <strong>21</strong>, 245 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">315. </div><div class="csl-right-inline">Li, H. New strategies to improve Minimap2 alignment accuracy. <a href="https://arxiv.org/abs/2108.03515">https://arxiv.org/abs/2108.03515</a> (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">316. </div><div class="csl-right-inline">Li, H. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-018-0054-7">A synthetic-diploid benchmark for accurate variant-calling evaluation</a>. <em>Nat Methods</em> <strong>15</strong>, 595–597 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">317. </div><div class="csl-right-inline">Martin, J. A. &amp; Wang, Z. <a href="https://doi.org/10.1038/nrg3068">Next-generation transcriptome assembly</a>. <em>Nature Reviews Genetics</em> <strong>12</strong>, 671–682 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">318. </div><div class="csl-right-inline">Kyriakidou, M., Tai, H. H., Anglin, N. L., Ellis, D. &amp; Strömvik, M. V. Current strategies of polyploid plant genome sequence assembly. <em>Frontiers in Plant Science</em> <strong>9</strong>, (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">319. </div><div class="csl-right-inline">Paszkiewicz, K. &amp; Studholme, D. J. <a href="https://doi.org/10.1093/bib/bbq020">De novo assembly of short sequence reads</a>. <em>Briefings in Bioinformatics</em> <strong>11</strong>, 457–472 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">320. </div><div class="csl-right-inline">Sohn, J. &amp; Nam, J.-W. <a href="https://doi.org/10.1093/bib/bbw096">The present and future of de novo whole-genome assembly</a>. <em>Briefings in Bioinformatics</em> <strong>19</strong>, 23–40 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">321. </div><div class="csl-right-inline">Sleator, R. D. &amp; Walsh, P. <a href="https://doi.org/10.1007/s00203-010-0549-9">An overview of in silico protein function prediction</a>. <em>Archives of Microbiology</em> <strong>192</strong>, 151–155 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">322. </div><div class="csl-right-inline">Koboldt, D. C. <a href="https://doi.org/10.1186/s13073-020-00791-w">Best practices for variant calling in clinical sequencing</a>. <em>Genome Medicine</em> <strong>12</strong>, 91 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">323. </div><div class="csl-right-inline">Alkan, C., Coe, B. P. &amp; Eichler, E. E. <a href="https://doi.org/10.1038/nrg2958">Genome structural variation discovery and genotyping</a>. <em>Nature Reviews Genetics</em> <strong>12</strong>, 363–376 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">324. </div><div class="csl-right-inline">Ho, S. S., Urban, A. E. &amp; Mills, R. E. <a href="https://doi.org/10.1038/s41576-019-0180-9">Structural variation in the sequencing era</a>. <em>Nature Reviews Genetics</em> <strong>21</strong>, 171–189 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">325. </div><div class="csl-right-inline">Morrison, D. A. <a href="https://doi.org/10.1016/0020-7519(96)00044-6">Phylogenetic tree-building</a>. <em>International Journal for Parasitology</em> <strong>26</strong>, 589–617 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">326. </div><div class="csl-right-inline">Kapli, P., Yang, Z. &amp; Telford, M. J. <a href="https://doi.org/10.1038/s41576-020-0233-0">Phylogenetic tree building in the genomic age</a>. <em>Nature Reviews Genetics</em> <strong>21</strong>, 428–444 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">327. </div><div class="csl-right-inline">Kuhlman, B. &amp; Bradley, P. <a href="https://doi.org/10.1038/s41580-019-0163-x">Advances in protein structure prediction and design</a>. <em>Nature Reviews Molecular Cell Biology</em> <strong>20</strong>, 681–697 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">328. </div><div class="csl-right-inline">Ammad-ud-din, M., Khan, S. A., Wennerberg, K. &amp; Aittokallio, T. <a href="https://doi.org/10.1093/bioinformatics/btx266">Systematic identification of feature combinations for predicting drug response with bayesian multi-view multi-task linear regression</a>. <em>Bioinformatics</em> <strong>33</strong>, i359–i368 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">329. </div><div class="csl-right-inline">Steiner, M. C., Gibson, K. M. &amp; Crandall, K. A. <a href="https://doi.org/10.3390/v12050560">Drug <span>Resistance Prediction Using Deep Learning Techniques</span> on <span>HIV</span>-1 <span>Sequence Data</span></a>. <em>Viruses</em> <strong>12</strong>, 560 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">330. </div><div class="csl-right-inline">Noé, F., De Fabritiis, G. &amp; Clementi, C. <a href="https://doi.org/10.1016/j.sbi.2019.12.005">Machine learning for protein folding and dynamics</a>. <em>Current Opinion in Structural Biology</em> <strong>60</strong>, 77–84 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">331. </div><div class="csl-right-inline">Pearce, R. &amp; Zhang, Y. <a href="https://doi.org/10.1016/j.jbc.2021.100870">Toward the solution of the protein structure prediction problem</a>. <em>Journal of Biological Chemistry</em> <strong>297</strong>, (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">332. </div><div class="csl-right-inline">Tunyasuvunakool, K. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-021-03828-1">Highly accurate protein structure prediction for the human proteome</a>. <em>Nature</em> <strong>596</strong>, 590–596 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">333. </div><div class="csl-right-inline">Cheng, J., Tegge, A. N. &amp; Baldi, P. <a href="https://doi.org/10.1109/RBME.2008.2008239">Machine learning methods for protein structure prediction</a>. <em>IEEE Reviews in Biomedical Engineering</em> <strong>1</strong>, 41–49 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">334. </div><div class="csl-right-inline">AlQuraishi, M. <a href="https://doi.org/10.1016/j.cbpa.2021.04.005">Machine learning in protein structure prediction</a>. <em>Current Opinion in Chemical Biology</em> <strong>65</strong>, 1–8 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">335. </div><div class="csl-right-inline">Wittmann, B. J., Johnston, K. E., Wu, Z. &amp; Arnold, F. H. <a href="https://doi.org/10.1016/j.sbi.2021.01.008">Advances in machine learning for directed evolution</a>. <em>Current Opinion in Structural Biology</em> <strong>69</strong>, 11–18 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">336. </div><div class="csl-right-inline">Yang, K. K., Wu, Z. &amp; Arnold, F. H. <a href="https://doi.org/10.1038/s41592-019-0496-6">Machine-learning-guided directed evolution for protein engineering</a>. <em>Nature Methods</em> <strong>16</strong>, 687–694 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">337. </div><div class="csl-right-inline">Li, G., Dong, Y. &amp; Reetz, M. T. <a href="https://doi.org/10.1002/adsc.201900149">Can Machine Learning Revolutionize Directed Evolution of Selective Enzymes?</a> <em>Advanced Synthesis &amp; Catalysis</em> <strong>361</strong>, 2377–2386 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">338. </div><div class="csl-right-inline">Xie, R., Wen, J., Quitadamo, A., Cheng, J. &amp; Shi, X. <a href="https://doi.org/10.1186/s12864-017-4226-0">A deep auto-encoder model for gene expression prediction</a>. <em>BMC Genomics</em> <strong>18</strong>, 845 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">339. </div><div class="csl-right-inline">Ortuño, F. M. <em>et al.</em> <a href="https://doi.org/10.1016/j.neucom.2015.01.080">Comparing different machine learning and mathematical regression models to evaluate multiple sequence alignments</a>. <em>Neurocomputing</em> <strong>164</strong>, 123–136 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">340. </div><div class="csl-right-inline">Wang, S., Sun, S., Li, Z., Zhang, R. &amp; Xu, J. <a href="https://doi.org/10.1371/journal.pcbi.1005324">Accurate De Novo Prediction of Protein Contact Map by Ultra-Deep Learning Model</a>. <em>PLOS Computational Biology</em> <strong>13</strong>, e1005324 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">341. </div><div class="csl-right-inline">Haga, H. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0242028">A machine learning-based treatment prediction model using whole genome variants of hepatitis C virus</a>. <em>PLOS ONE</em> <strong>15</strong>, e0242028 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">342. </div><div class="csl-right-inline">Zazzi, M. <em>et al.</em> <a href="https://doi.org/10.1159/000332008">Predicting response to antiretroviral treatment by machine learning: The EuResist project</a>. <em>Intervirology</em> <strong>55</strong>, 123–127 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">343. </div><div class="csl-right-inline">Ren, Y. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btab681">Prediction of antimicrobial resistance based on whole-genome sequencing and machine learning</a>. <em>Bioinformatics</em> <strong>38</strong>, 325–334 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">344. </div><div class="csl-right-inline">Kim, J. I. <em>et al.</em> <a href="https://doi.org/10.1128/cmr.00179-21">Machine learning for antimicrobial resistance prediction: Current practice, limitations, and clinical perspective</a>. <em>Clinical Microbiology Reviews</em> <strong>0</strong>, e00179–21 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">345. </div><div class="csl-right-inline">Wang, Y. <em>et al.</em> <a href="https://doi.org/10.1038/srep19598">Predicting DNA Methylation State of CpG Dinucleotide Using Genome Topological Features and Deep Networks</a>. <em>Scientific Reports</em> <strong>6</strong>, 19598 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">346. </div><div class="csl-right-inline">Rätsch, G., Sonnenburg, S. &amp; Schäfer, C. <a href="https://doi.org/10.1186/1471-2105-7-S1-S9">Learning Interpretable SVMs for Biological Sequence Classification</a>. <em>BMC Bioinformatics</em> <strong>7</strong>, S9 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">347. </div><div class="csl-right-inline">Jones, D. T. <a href="https://doi.org/10.1006/jmbi.1999.3091">Protein secondary structure prediction based on position-specific scoring matrices</a>. <em>Journal of Molecular Biology</em> <strong>292</strong>, 195–202 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">348. </div><div class="csl-right-inline">Wei, L., Ding, Y., Su, R., Tang, J. &amp; Zou, Q. <a href="https://doi.org/10.1016/j.jpdc.2017.08.009">Prediction of human protein subcellular localization using deep learning</a>. <em>Journal of Parallel and Distributed Computing</em> <strong>117</strong>, 212–217 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">349. </div><div class="csl-right-inline">Wang, H., Yan, L., Huang, H. &amp; Ding, C. <a href="https://doi.org/10.1109/TCBB.2016.2591529">From protein sequence to protein function via multi-label linear discriminant analysis</a>. <em>IEEE/ACM Transactions on Computational Biology and Bioinformatics</em> <strong>14</strong>, 503–513 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">350. </div><div class="csl-right-inline">Kelley, D. R., Snoek, J. &amp; Rinn, J. L. <a href="https://doi.org/10.1101/gr.200535.115">Basset: Learning the regulatory code of the accessible genome with deep convolutional neural networks</a>. <em>Genome Research</em> <strong>26</strong>, 990–999 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">351. </div><div class="csl-right-inline">Hastie, T., Tibshirani, R. &amp; Friedman, J. <em>The <span>Elements</span> of <span>Statistical Learning</span>: <span>Data Mining</span>, <span>Inference</span>, and <span>Prediction</span>, <span>Second Edition</span></em>. (<span>Springer Science &amp; Business Media</span>, 2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">352. </div><div class="csl-right-inline">Kriventseva, E. V., Biswas, M. &amp; Apweiler, R. <a href="https://doi.org/10.1016/S0959-440X(00)00211-6">Clustering and analysis of protein families</a>. <em>Current Opinion in Structural Biology</em> <strong>11</strong>, 334–339 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">353. </div><div class="csl-right-inline">Fu, L., Niu, B., Zhu, Z., Wu, S. &amp; Li, W. <a href="https://doi.org/10.1093/bioinformatics/bts565">CD-HIT: Accelerated for clustering the next-generation sequencing data</a>. <em>Bioinformatics</em> <strong>28</strong>, 3150–3152 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">354. </div><div class="csl-right-inline">Balaban, M., Moshiri, N., Mai, U., Jia, X. &amp; Mirarab, S. <a href="https://doi.org/10.1371/journal.pone.0221068">TreeCluster: Clustering biological sequences using phylogenetic trees</a>. <em>PLOS ONE</em> <strong>14</strong>, e0221068 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">355. </div><div class="csl-right-inline">Zorita, E., Cuscó, P. &amp; Filion, G. J. <a href="https://doi.org/10.1093/bioinformatics/btv053">Starcode: Sequence clustering based on all-pairs search</a>. <em>Bioinformatics</em> <strong>31</strong>, 1913–1919 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">356. </div><div class="csl-right-inline">Ondov, B. D. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-016-0997-x">Mash: Fast genome and metagenome distance estimation using MinHash</a>. <em>Genome Biology</em> <strong>17</strong>, 132 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">357. </div><div class="csl-right-inline">Baker, D. N. &amp; Langmead, B. <a href="https://doi.org/10.1186/s13059-019-1875-0">Dashing: Fast and accurate genomic distances with HyperLogLog</a>. <em>Genome Biology</em> <strong>20</strong>, 265 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">358. </div><div class="csl-right-inline">Corso, G. <em>et al.</em> Neural distance embeddings for biological sequences. in vol. 34 1853918551 (Curran Associates, Inc., 2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">359. </div><div class="csl-right-inline">Hopf, T. A. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.3769">Mutation effects predicted from sequence co-variation</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 128–135 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">360. </div><div class="csl-right-inline">Castro, B. M., Lemes, R. B., Cesar, J., Hünemeier, T. &amp; Leonardi, F. <a href="https://doi.org/10.1016/j.jmva.2018.05.006">A model selection approach for multiple sequence segmentation and dimensionality reduction</a>. <em>Journal of Multivariate Analysis</em> <strong>167</strong>, 319–330 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">361. </div><div class="csl-right-inline">Haschka, T., Ponger, L., Escudé, C. &amp; Mozziconacci, J. <a href="https://doi.org/10.1093/bioinformatics/btab430">MNHN-tree-tools: A toolbox for tree inference using multi-scale clustering of a set of sequences</a>. <em>Bioinformatics</em> <strong>37</strong>, 3947–3949 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">362. </div><div class="csl-right-inline">Konishi, T. <em>et al.</em> <a href="https://doi.org/10.1038/s41598-019-55253-0">Principal Component Analysis applied directly to Sequence Matrix</a>. <em>Scientific Reports</em> <strong>9</strong>, 19297 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">363. </div><div class="csl-right-inline">Ben-Hur, A. &amp; Guyon, I. Detecting Stable Clusters Using Principal Component Analysis. in 159–182 (Humana Press, 2003). doi:<a href="https://doi.org/10.1385/1-59259-364-X:159">10.1385/1-59259-364-X:159</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">364. </div><div class="csl-right-inline">Ding, C. &amp; He, X. K-means clustering via principal component analysis. in 29 (Association for Computing Machinery, 2004). doi:<a href="https://doi.org/10.1145/1015330.1015408">10.1145/1015330.1015408</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">365. </div><div class="csl-right-inline">Casari, G., Sander, C. &amp; Valencia, A. Sequencespace: A tool for family analysis. <strong>2</strong>, 171178 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">366. </div><div class="csl-right-inline">Clamp, M., Cuff, J., Searle, S. M. &amp; Barton, G. J. <a href="https://doi.org/10.1093/bioinformatics/btg430">The jalview java alignment editor</a>. <em>Bioinformatics</em> <strong>20</strong>, 426–427 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">367. </div><div class="csl-right-inline">Xia, Z., Wu, L.-Y., Zhou, X. &amp; Wong, S. T. <a href="https://doi.org/10.1186/1752-0509-4-S2-S6">Semi-supervised drug-protein interaction prediction from heterogeneous biological spaces</a>. <em>BMC Systems Biology</em> <strong>4</strong>, S6 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">368. </div><div class="csl-right-inline">Tamposis, I. A., Tsirigos, K. D., Theodoropoulou, M. C., Kontou, P. I. &amp; Bagos, P. G. <a href="https://doi.org/10.1093/bioinformatics/bty910">Semi-supervised learning of hidden markov models for biological sequence analysis</a>. <em>Bioinformatics</em> <strong>35</strong>, 2208–2215 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">369. </div><div class="csl-right-inline">Elnaggar, A. <em>et al.</em> ProtTrans: Towards cracking the language of life’s code through self-supervised deep learning and high performance computing. doi:<a href="https://doi.org/10.48550/arXiv.2007.06225">10.48550/arXiv.2007.06225</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">370. </div><div class="csl-right-inline">Lu, A. X. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pcbi.1010238">Discovering molecular features of intrinsically disordered regions by using evolution for contrastive learning</a>. <em>PLOS Computational Biology</em> <strong>18</strong>, e1010238 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">371. </div><div class="csl-right-inline">Townshend, R., Bedi, R., Suriana, P. &amp; Dror, R. End-to-end learning on 3D protein structure for interface prediction. in vol. 32 (Curran Associates, Inc., 2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">372. </div><div class="csl-right-inline">Lee, B., Baek, J., Park, S. &amp; Yoon, S. deepTarget: End-to-end learning framework for microRNA target prediction using deep recurrent neural networks. in 434442 (Association for Computing Machinery, 2016). doi:<a href="https://doi.org/10.1145/2975167.2975212">10.1145/2975167.2975212</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">373. </div><div class="csl-right-inline">Goodfellow, I., Bengio, Y. &amp; Courville, A. <em>Deep learning</em>. (MIT Press, 2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">374. </div><div class="csl-right-inline">Wang, Q., Ma, Y., Zhao, K. &amp; Tian, Y. <a href="https://doi.org/10.1007/s40745-020-00253-5">A Comprehensive Survey of Loss Functions in Machine Learning</a>. <em>Annals of Data Science</em> <strong>9</strong>, 187–212 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">375. </div><div class="csl-right-inline">Jiao, Y. &amp; Du, P. <a href="https://doi.org/10.1007/s40484-016-0081-2">Performance measures in evaluating machine learning based bioinformatics predictors for classifications</a>. <em>Quantitative Biology</em> <strong>4</strong>, 320–330 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">376. </div><div class="csl-right-inline">Brodersen, K. H., Ong, C. S., Stephan, K. E. &amp; Buhmann, J. M. The <span>Balanced Accuracy</span> and <span>Its Posterior Distribution</span>. in <em>2010 20th <span>International Conference</span> on <span>Pattern Recognition</span></em> 3121–3124 (2010). doi:<a href="https://doi.org/10.1109/ICPR.2010.764">10.1109/ICPR.2010.764</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">377. </div><div class="csl-right-inline">Kaufman, S., Rosset, S. &amp; Perlich, C. Leakage in data mining: Formulation, detection, and avoidance. in 556563 (Association for Computing Machinery, 2011). doi:<a href="https://doi.org/10.1145/2020408.2020496">10.1145/2020408.2020496</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">378. </div><div class="csl-right-inline">Whalen, S., Schreiber, J., Noble, W. S. &amp; Pollard, K. S. <a href="https://doi.org/10.1038/s41576-021-00434-9">Navigating the pitfalls of applying machine learning in genomics</a>. <em>Nature Reviews Genetics</em> <strong>23</strong>, 169–181 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">379. </div><div class="csl-right-inline">Fisher, R. A. <a href="https://doi.org/10.2307/2340521">On the interpretation of χ2 from contingency tables, and the calculation of p</a>. <em>Journal of the Royal Statistical Society</em> <strong>85</strong>, 87–94 (1922).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">380. </div><div class="csl-right-inline">Pearson, K. <a href="https://doi.org/10.1080/14786440009463897">X. On the criterion that a given system of deviations from the probable in the case of a correlated system of variables is such that it can be reasonably supposed to have arisen from random sampling</a>. <em>The London, Edinburgh, and Dublin Philosophical Magazine and Journal of Science</em> <strong>50</strong>, 157–175 (1900).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">381. </div><div class="csl-right-inline">Hoerl, A. E. &amp; Kennard, R. W. <a href="https://doi.org/10.1080/00401706.1970.10488634">Ridge regression: Biased estimation for nonorthogonal problems</a>. <em>Technometrics</em> <strong>12</strong>, 55–67 (1970).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">382. </div><div class="csl-right-inline">Tibshirani, R. <a href="https://doi.org/10.1111/j.2517-6161.1996.tb02080.x">Regression <span>Shrinkage</span> and <span>Selection Via</span> the <span>Lasso</span></a>. <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> <strong>58</strong>, 267–288 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">383. </div><div class="csl-right-inline">Zhang, H. The Optimality of Naive Bayes. 6.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">384. </div><div class="csl-right-inline">Rish, I. An empirical study of the naive Bayes classi<span>fi</span>er. 6.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">385. </div><div class="csl-right-inline">Vapnik, V. <em>Estimation of dependences based on empirical data: Springer series in statistics (springer series in statistics)</em>. (Springer-Verlag, 1982).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">386. </div><div class="csl-right-inline">Boser, B. E., Guyon, I. M. &amp; Vapnik, V. N. A training algorithm for optimal margin classifiers. in 144152 (Association for Computing Machinery, 1992). doi:<a href="https://doi.org/10.1145/130385.130401">10.1145/130385.130401</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">387. </div><div class="csl-right-inline">Cortes, C. &amp; Vapnik, V. <a href="https://doi.org/10.1007/BF00994018">Support-vector networks</a>. <em>Machine Learning</em> <strong>20</strong>, 273–297 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">388. </div><div class="csl-right-inline">Drucker, H., Burges, C. J. C., Kaufman, L., Smola, A. &amp; Vapnik, V. Support vector regression machines. in vol. 9 (MIT Press, 1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">389. </div><div class="csl-right-inline">Breiman, L. <a href="https://doi.org/10.1023/A:1010933404324">Random <span>Forests</span></a>. <em>Machine Learning</em> <strong>45</strong>, 5–32 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">390. </div><div class="csl-right-inline">Breiman, L., Friedman, J. H., Olshen, R. A. &amp; Stone, C. J. <em>Classification and regression trees</em>. (1983).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">391. </div><div class="csl-right-inline">Kingsford, C. &amp; Salzberg, S. L. <a href="https://doi.org/10.1038/nbt0908-1011">What are decision trees?</a> <em>Nature Biotechnology</em> <strong>26</strong>, 1011–1013 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">392. </div><div class="csl-right-inline">Caruana, R. &amp; Niculescu-Mizil, A. An empirical comparison of supervised learning algorithms. in 161168 (Association for Computing Machinery, 2006). doi:<a href="https://doi.org/10.1145/1143844.1143865">10.1145/1143844.1143865</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">393. </div><div class="csl-right-inline">Yang, P., Hwa Yang, Y., B. Zhou, B. &amp; Y. Zomaya, A. A review of ensemble methods in bioinformatics. <em>Current Bioinformatics</em> <strong>5</strong>, 296–308 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">394. </div><div class="csl-right-inline">Potdar, K., S., T. &amp; D., C. <a href="https://doi.org/10.5120/ijca2017915495">A Comparative Study of Categorical Variable Encoding Techniques for Neural Network Classifiers</a>. <em>International Journal of Computer Applications</em> <strong>175</strong>, 7–9 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">395. </div><div class="csl-right-inline">Hassani Saadi, H., Sameni, R. &amp; Zollanvari, A. <a href="https://doi.org/10.1186/s12859-017-1524-0">Interpretive time-frequency analysis of genomic sequences</a>. <em>BMC Bioinformatics</em> <strong>18</strong>, 154 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">396. </div><div class="csl-right-inline">Brouwer, R. K. <a href="https://doi.org/10.1016/S0893-6080(02)00090-4">A feed-forward network for input that is both categorical and quantitative</a>. <em>Neural Networks</em> <strong>15</strong>, 881–890 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">397. </div><div class="csl-right-inline">Kunanbayev, K., Temirbek, I. &amp; Zollanvari, A. 2021 international joint conference on neural networks (IJCNN). in 1–6 (2021). doi:<a href="https://doi.org/10.1109/IJCNN52387.2021.9534094">10.1109/IJCNN52387.2021.9534094</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">398. </div><div class="csl-right-inline">Dufresne, Y. <em>et al.</em> The k-mer file format: A standardized and compact disk representation of sets of k-mers. <em>Bioinformatics</em> btac528 (2022) doi:<a href="https://doi.org/10.1093/bioinformatics/btac528">10.1093/bioinformatics/btac528</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">399. </div><div class="csl-right-inline">Wright, E. S. <a href="https://doi.org/10.32614/RJ-2016-025">Using DECIPHER v2.0 to analyze big biological sequence data in r</a>. <em>The R Journal</em> <strong>8</strong>, 352–359 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">400. </div><div class="csl-right-inline">Zamani, M. &amp; Kremer, S. C. 2011 IEEE international conference on bioinformatics and biomedicine workshops (BIBMW). in 327–333 (2011). doi:<a href="https://doi.org/10.1109/BIBMW.2011.6112394">10.1109/BIBMW.2011.6112394</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">401. </div><div class="csl-right-inline">Singh, D., Singh, P. &amp; Sisodia, D. S. <a href="https://doi.org/10.1016/j.eswa.2018.05.003">Evolutionary based optimal ensemble classifiers for HIV-1 protease cleavage sites prediction</a>. <em>Expert Systems with Applications</em> <strong>109</strong>, 86–99 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">402. </div><div class="csl-right-inline">Qian, N. &amp; Sejnowski, T. J. <a href="https://doi.org/10.1016/0022-2836(88)90564-5">Predicting the secondary structure of globular proteins using neural network models</a>. <em>Journal of Molecular Biology</em> <strong>202</strong>, 865–884 (1988).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">403. </div><div class="csl-right-inline">Budach, S. &amp; Marsico, A. <a href="https://doi.org/10.1093/bioinformatics/bty222">Pysster: Classification of biological sequences by learning sequence and structure motifs with convolutional neural networks</a>. <em>Bioinformatics</em> <strong>34</strong>, 3035–3037 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">404. </div><div class="csl-right-inline">Choong, A. C. H. &amp; Lee, N. K. 2017 international conference on computer and drone applications (IConDA). in 60–65 (2017). doi:<a href="https://doi.org/10.1109/ICONDA.2017.8270400">10.1109/ICONDA.2017.8270400</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">405. </div><div class="csl-right-inline">McGinnis, W. <em>et al.</em> Scikit-<span>Learn</span>-<span>Contrib</span>/<span>Categorical</span>-<span>Encoding</span>: <span>Release For Zenodo</span>. (2018) doi:<a href="https://doi.org/10.5281/ZENODO.1157110">10.5281/ZENODO.1157110</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">406. </div><div class="csl-right-inline">Kawashima, S. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkm998">AAindex: amino acid index database, progress report 2008</a>. <em>Nucleic Acids Research</em> <strong>36</strong>, D202–D205 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">407. </div><div class="csl-right-inline">Li, Z.-C., Zhou, X.-B., Dai, Z. &amp; Zou, X.-Y. <a href="https://doi.org/10.1007/s00726-008-0170-2">Prediction of protein structural classes by Chou<span>’</span>s pseudo amino acid composition: approached using continuous wavelet transform and principal component analysis</a>. <em>Amino Acids</em> <strong>37</strong>, 415 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">408. </div><div class="csl-right-inline">Nanni, L. &amp; Lumini, A. <a href="https://doi.org/10.1016/j.eswa.2010.09.005">A new encoding technique for peptide classification</a>. <em>Expert Systems with Applications</em> <strong>38</strong>, 3185–3191 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">409. </div><div class="csl-right-inline">Chen, Z. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/bty140">iFeature: A python package and web server for features extraction and selection from protein and peptide sequences</a>. <em>Bioinformatics</em> <strong>34</strong>, 2499–2502 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">410. </div><div class="csl-right-inline">Taylor, W. R. <a href="https://doi.org/10.1016/S0022-5193(86)80075-3">The classification of amino acid conservation</a>. <em>Journal of Theoretical Biology</em> <strong>119</strong>, 205–218 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">411. </div><div class="csl-right-inline">Zvelebil, M. J., Barton, G. J., Taylor, W. R. &amp; Sternberg, M. J. E. <a href="https://doi.org/10.1016/0022-2836(87)90501-8">Prediction of protein secondary structure and active sites using the alignment of homologous sequences</a>. <em>Journal of Molecular Biology</em> <strong>195</strong>, 957–961 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">412. </div><div class="csl-right-inline">Kremer, S. &amp; Lac, H. <a href="https://patents.google.com/patent/US20090024375A1/en">Method, system and computer program product for levinthal process induction from known structure using machine learning</a>. (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">413. </div><div class="csl-right-inline">Maetschke, S., Towsey, M. &amp; Bodén, M. <a href="https://doi.org/10.1142/9781860947322_0014">Blomap: An encoding of amino acids which improves signal peptide cleavage site prediction</a>. in vols Volume 1 141–150 (PUBLISHED BY IMPERIAL COLLEGE PRESS AND DISTRIBUTED BY WORLD SCIENTIFIC PUBLISHING CO., 2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">414. </div><div class="csl-right-inline">Gök, M. &amp; Özcerit, A. T. <a href="https://doi.org/10.1007/s00521-012-0967-5">A new feature encoding scheme for HIV-1 protease cleavage site prediction</a>. <em>Neural Computing and Applications</em> <strong>22</strong>, 1757–1761 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">415. </div><div class="csl-right-inline">Saha, S. &amp; Bhattacharya, T. A Novel Approach to Find the Saturation Point of n-Gram Encoding Method for Protein Sequence Classification Involving Data Mining. in 101–108 (Springer, 2019). doi:<a href="https://doi.org/10.1007/978-981-13-2354-6_12">10.1007/978-981-13-2354-6_12</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">416. </div><div class="csl-right-inline">Jeffrey, H. J. <a href="https://doi.org/10.1093/nar/18.8.2163">Chaos game representation of gene structure</a>. <em>Nucleic Acids Research</em> <strong>18</strong>, 2163–2170 (1990).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">417. </div><div class="csl-right-inline">Löchel, H. F. &amp; Heider, D. <a href="https://doi.org/10.1016/j.csbj.2021.11.008">Chaos game representation and its applications in bioinformatics</a>. <em>Computational and Structural Biotechnology Journal</em> <strong>19</strong>, 6263–6271 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">418. </div><div class="csl-right-inline">Cartes, J. A., Anand, S., Ciccolella, S., Bonizzoni, P. &amp; Vedova, G. D. Accurate and Fast Clade Assignment via Deep Learning and Frequency Chaos Game Representation. doi:<a href="https://doi.org/10.1101/2022.06.13.495912">10.1101/2022.06.13.495912</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">419. </div><div class="csl-right-inline">Ni, H., Mu, H. &amp; Qi, D. <a href="https://doi.org/10.1016/j.jmgm.2021.107942">Applying frequency chaos game representation with perceptual image hashing to gene sequence phylogenetic analyses</a>. <em>Journal of Molecular Graphics and Modelling</em> <strong>107</strong>, 107942 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">420. </div><div class="csl-right-inline">Lwoff, A. <a href="https://doi.org/10.1099/00221287-17-2-239">The concept of virus</a>. <em>Journal of General Microbiology</em> <strong>17</strong>, 239–253 (1957).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">421. </div><div class="csl-right-inline">Minor, P. D. Viruses. in (John Wiley &amp; Sons, Ltd, 2014). doi:<a href="https://doi.org/10.1002/9780470015902.a0000441.pub3">10.1002/9780470015902.a0000441.pub3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">422. </div><div class="csl-right-inline">Stapleton, J. T., Foung, S., Muerhoff, A. S., Bukh, J. &amp; Simmonds, P. <a href="https://doi.org/10.1099/vir.0.027490-0">The GB viruses: A review and proposed classification of GBV-a, GBV-c (HGV), and GBV-d in genus pegivirus within the family flaviviridae</a>. <em>The Journal of General Virology</em> <strong>92</strong>, 233–246 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">423. </div><div class="csl-right-inline">Yamamoto, N. <em>et al.</em> <a href="https://doi.org/10.1186/1743-422X-8-65">Characterization of a non-pathogenic H5N1 influenza virus isolated from a migratory duck flying from siberia in hokkaido, japan, in october 2009</a>. <em>Virology Journal</em> <strong>8</strong>, 65 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">424. </div><div class="csl-right-inline">Shi, M. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-018-0012-7">The evolutionary history of vertebrate RNA viruses</a>. <em>Nature</em> <strong>556</strong>, 197–202 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">425. </div><div class="csl-right-inline"><em>Atlas of invertebrate viruses</em>. (CRC Press, 2017). doi:<a href="https://doi.org/10.1201/9781315149929">10.1201/9781315149929</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">426. </div><div class="csl-right-inline">Lefeuvre, P. <em>et al.</em> <a href="https://doi.org/10.1038/s41579-019-0232-3">Evolution and ecology of plant viruses</a>. <em>Nature Reviews Microbiology</em> <strong>17</strong>, 632–644 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">427. </div><div class="csl-right-inline">Wang, A. L. &amp; Wang, C. C. <a href="https://doi.org/10.1016/0169-4758(91)90198-W">Viruses of parasitic protozoa</a>. <em>Parasitology Today</em> <strong>7</strong>, 76–80 (1991).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">428. </div><div class="csl-right-inline">Fermin, G., Mazumdar-Leighton, S. &amp; Tennant, P. Viruses of prokaryotes, protozoa, fungi, and chromista. in 217 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">429. </div><div class="csl-right-inline">Sutela, S., Poimala, A. &amp; Vainio, E. J. <a href="https://doi.org/10.1093/femsec/fiz119">Viruses of fungi and oomycetes in the soil environment</a>. <em>FEMS Microbiology Ecology</em> <strong>95</strong>, fiz119 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">430. </div><div class="csl-right-inline">Twort, F. W. <a href="https://doi.org/10.1016/S0140-6736(01)20383-3">AN INVESTIGATION ON THE NATURE OF ULTRA-MICROSCOPIC VIRUSES.</a> <em>The Lancet</em> <strong>186</strong>, 1241–1243 (1915).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">431. </div><div class="csl-right-inline">Delbrock, M. <a href="https://doi.org/10.1111/j.1469-185X.1946.tb00451.x">Bacterial Viruses or Bacteriophages</a>. <em>Biological Reviews</em> <strong>21</strong>, 30–40 (1946).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">432. </div><div class="csl-right-inline">Clark, J. R. &amp; March, J. B. <a href="https://doi.org/10.1586/14760584.3.4.463">Bacterial viruses as human vaccines?</a> <em>Expert Review of Vaccines</em> <strong>3</strong>, 463–476 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">433. </div><div class="csl-right-inline">Kan-Davelaar, H. E. van, Hest, J. C. M. van, Cornelissen, J. J. L. M. &amp; Koay, M. S. T. <a href="https://doi.org/10.1111/bph.12662">Using viruses as nanomedicines</a>. <em>British Journal of Pharmacology</em> <strong>171</strong>, 4001–4009 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">434. </div><div class="csl-right-inline">Prangishvili, D., Basta, T., Garrett, R. A. &amp; Krupovic, M. Viruses of the Archaea. in 1–9 (John Wiley &amp; Sons, Ltd, 2016). doi:<a href="https://doi.org/10.1002/9780470015902.a0000774.pub3">10.1002/9780470015902.a0000774.pub3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">435. </div><div class="csl-right-inline">Prangishvili, D., Forterre, P. &amp; Garrett, R. A. <a href="https://doi.org/10.1038/nrmicro1527">Viruses of the Archaea: a unifying view</a>. <em>Nature Reviews Microbiology</em> <strong>4</strong>, 837–848 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">436. </div><div class="csl-right-inline">Francki, R. I. B. Plant virus satellites. <em>ANNUAL REVIEW OF MICROBIOLOGY</em>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">437. </div><div class="csl-right-inline">Xu, P. &amp; Roossinck, M. J. Plant Virus Satellites. in (John Wiley &amp; Sons, Ltd, 2011). doi:<a href="https://doi.org/10.1002/9780470015902.a0000771.pub2">10.1002/9780470015902.a0000771.pub2</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">438. </div><div class="csl-right-inline">Lai, M. M. <a href="https://doi.org/10.1146/annurev.bi.64.070195.001355">The molecular biology of hepatitis delta virus</a>. <em>Annual review of biochemistry</em> <strong>64</strong>, 259–286 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">439. </div><div class="csl-right-inline">Hughes, S. A., Wedemeyer, H. &amp; Harrison, P. M. <a href="https://doi.org/10.1016/S0140-6736(10)61931-9">Hepatitis delta virus</a>. <em>The Lancet</em> <strong>378</strong>, 73–85 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">440. </div><div class="csl-right-inline">Desnues, C., Boyer, M. &amp; Raoult, D. <a href="https://doi.org/10.1016/B978-0-12-394621-8.00013-3">Chapter 3 - Sputnik, a Virophage Infecting the Viral Domain of Life</a>. in (eds. Łobocka, M. &amp; Szybalski, W. T.) vol. 82 63–89 (Academic Press, 2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">441. </div><div class="csl-right-inline">Gaia, M. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0094923">Zamilon, a Novel Virophage with Mimiviridae Host Specificity</a>. <em>PLOS ONE</em> <strong>9</strong>, e94923 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">442. </div><div class="csl-right-inline">Edgar, R. C. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-021-04332-2">Petabase-scale sequence alignment catalyses viral discovery</a>. <em>Nature</em> <strong>602</strong>, 142–147 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">443. </div><div class="csl-right-inline">Nasir, A., Romero-Severson, E. &amp; Claverie, J.-M. <a href="https://doi.org/10.1016/j.tim.2020.08.003">Investigating the Concept and Origin of Viruses</a>. <em>Trends in Microbiology</em> <strong>28</strong>, 959–967 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">444. </div><div class="csl-right-inline">Forterre, P. &amp; Prangishvili, D. <a href="https://doi.org/10.1016/j.resmic.2009.07.008">The origin of viruses</a>. <em>Research in Microbiology</em> <strong>160</strong>, 466–472 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">445. </div><div class="csl-right-inline">Forterre, P. <a href="https://doi.org/10.1016/j.virusres.2006.01.010">The origin of viruses and their possible roles in major evolutionary transitions</a>. <em>Virus Research</em> <strong>117</strong>, 5–16 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">446. </div><div class="csl-right-inline">Boeke, J. &amp; Stoye, J. Retrotransposons, Endogenous Retroviruses, and the Evolution of Retroelement. (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">447. </div><div class="csl-right-inline">Kojima, S. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.2010758118">Virus-like insertions with sequence signatures similar to those of endogenous nonretroviral RNA viruses in the human genome</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>118</strong>, e2010758118 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">448. </div><div class="csl-right-inline">Löwer, R., Löwer, J. &amp; R Kurth. <a href="https://doi.org/10.1073/pnas.93.11.5177">The viruses in all of us: Characteristics and biological significance of human endogenous retrovirus sequences.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>93</strong>, 5177–5184 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">449. </div><div class="csl-right-inline">Griffiths, D. J. <a href="https://doi.org/10.1186/gb-2001-2-6-reviews1017">Endogenous retroviruses in the human genome sequence</a>. <em>Genome Biology</em> <strong>2</strong>, reviews1017.1 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">450. </div><div class="csl-right-inline">Baltimore, D. <a href="https://doi.org/10.1128/br.35.3.235-241.1971">Expression of animal virus genomes</a>. <em>Bacteriological Reviews</em> <strong>35</strong>, 235–241 (1971).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">451. </div><div class="csl-right-inline">Koonin, E. V., Krupovic, M. &amp; Agol, V. I. <a href="https://doi.org/10.1128/MMBR.00053-21">The baltimore classification of viruses 50 years later: How does it stand in the light of virus evolution?</a> <em>Microbiology and Molecular Biology Reviews</em> <strong>85</strong>, e00053–21 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">452. </div><div class="csl-right-inline">Domingo, E. &amp; Perales, C. RNA Virus Genomes. in 1–12 (John Wiley &amp; Sons, Ltd, 2018). doi:<a href="https://doi.org/10.1002/9780470015902.a0001488.pub3">10.1002/9780470015902.a0001488.pub3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">453. </div><div class="csl-right-inline">McGeoch, D. J., Rixon, F. J. &amp; Davison, A. J. <a href="https://doi.org/10.1016/j.virusres.2006.01.002">Topics in herpesvirus genomics and evolution</a>. <em>Virus Research</em> <strong>117</strong>, 90–104 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">454. </div><div class="csl-right-inline">Boehmer, P. &amp; Nimonkar, A. <a href="https://doi.org/10.1080/1521654031000070645">Herpes Virus Replication</a>. <em>IUBMB Life</em> <strong>55</strong>, 13–22 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">455. </div><div class="csl-right-inline">Brentjens, M. H., Yeung-Yue, K. A., Lee, P. C. &amp; Tyring, S. K. <a href="https://doi.org/10.1016/S0733-8635(01)00028-6">Human papillomavirus: a review</a>. <em>Dermatologic Clinics</em> <strong>20</strong>, 315–331 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">456. </div><div class="csl-right-inline">Kay, A. &amp; Zoulim, F. <a href="https://doi.org/10.1016/j.virusres.2007.02.021">Hepatitis B virus genetic variability and evolution</a>. <em>Virus Research</em> <strong>127</strong>, 164–176 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">457. </div><div class="csl-right-inline">Parashar, U. D., Bresee, J. S., Gentsch, J. R. &amp; Glass, R. I. Rotavirus. <em>Emerging Infectious Diseases</em> <strong>4</strong>, 561–570 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">458. </div><div class="csl-right-inline">Simmonds, P. <a href="https://doi.org/10.1016/0270-9139(95)90121-3">Variability of hepatitis C virus</a>. <em>Hepatology</em> <strong>21</strong>, 570–583 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">459. </div><div class="csl-right-inline">Wimmer, E., Hellen, C. U. T. &amp; Cao, X. Genetics of poliovirus. <em>Annual Review of Genetics</em> <strong>27</strong>, 353–437 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">460. </div><div class="csl-right-inline">Racaniello, V. R. <a href="https://doi.org/10.1016/j.virol.2005.09.015">One hundred years of poliovirus pathogenesis</a>. <em>Virology</em> <strong>344</strong>, 9–16 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">461. </div><div class="csl-right-inline">Palese, P., Zheng, H., Engelhardt, O. G., Pleschka, S. &amp; García-Sastre, A. <a href="https://doi.org/10.1073/pnas.93.21.11354">Negative-strand RNA viruses: Genetic engineering and applications.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>93</strong>, 11354–11358 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">462. </div><div class="csl-right-inline">Domingo, E. &amp; Perales, C. Virus Evolution. in (John Wiley &amp; Sons, Ltd, 2014). doi:<a href="https://doi.org/10.1002/9780470015902.a0000436.pub3">10.1002/9780470015902.a0000436.pub3</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">463. </div><div class="csl-right-inline">V’kovski, P., Kratzel, A., Steiner, S., Stalder, H. &amp; Thiel, V. <a href="https://doi.org/10.1038/s41579-020-00468-6">Coronavirus biology and replication: implications for SARS-CoV-2</a>. <em>Nature Reviews Microbiology</em> <strong>19</strong>, 155–170 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">464. </div><div class="csl-right-inline">Bäck, A. T. &amp; Lundkvist, Å. <a href="https://doi.org/10.3402/iee.v3i0.19839">Dengue viruses <span></span> an overview</a>. <em>Infection Ecology &amp; Epidemiology</em> <strong>3</strong>, 10.3402/iee.v3i0.19839 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">465. </div><div class="csl-right-inline">Dustin, L. B., Bartolini, B., Capobianchi, M. R. &amp; Pistello, M. <a href="https://doi.org/10.1016/j.cmi.2016.08.025">Hepatitis c virus: Life cycle in cells, infection and host response, and analysis of molecular markers influencing the outcome of infection and response to therapy</a>. <em>Clinical microbiology and infection : the official publication of the European Society of Clinical Microbiology and Infectious Diseases</em> <strong>22</strong>, 826–832 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">466. </div><div class="csl-right-inline">Kadaja, M., Silla, T., Ustav, E. &amp; Ustav, M. <a href="https://doi.org/10.1016/j.virol.2008.11.032">Papillomavirus DNA replication <span></span> From initiation to genomic instability</a>. <em>Virology</em> <strong>384</strong>, 360–368 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">467. </div><div class="csl-right-inline">Weller, S. K. &amp; Coen, D. M. <a href="https://doi.org/10.1101/cshperspect.a013011">Herpes simplex viruses: Mechanisms of DNA replication</a>. <em>Cold Spring Harbor Perspectives in Biology</em> <strong>4</strong>, a013011 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">468. </div><div class="csl-right-inline">Beck, J. &amp; Nassal, M. <a href="https://doi.org/10.3748/wjg.v13.i1.48">Hepatitis b virus replication</a>. <em>World Journal of Gastroenterology : WJG</em> <strong>13</strong>, 48–64 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">469. </div><div class="csl-right-inline">Pyle, J. D. &amp; Scholthof, K.-B. G. Chapter 58 - Biology and Pathogenesis of Satellite Viruses. in (eds. Hadidi, A., Flores, R., Randles, J. W. &amp; Palukaitis, P.) 627–636 (Academic Press, 2017). doi:<a href="https://doi.org/10.1016/B978-0-12-801498-1.00058-9">10.1016/B978-0-12-801498-1.00058-9</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">470. </div><div class="csl-right-inline">Raoult, D. <em>et al.</em> <a href="https://doi.org/10.1126/science.1101485">The 1.2-megabase genome sequence of Mimivirus</a>. <em>Science (New York, N.Y.)</em> <strong>306</strong>, 1344–1350 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">471. </div><div class="csl-right-inline">Campillo-Balderas, J. A., Lazcano, A. &amp; Becerra, A. Viral genome size distribution does not correlate with the antiquity of the host lineages. <em>Frontiers in Ecology and Evolution</em> <strong>3</strong>, (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">472. </div><div class="csl-right-inline">Cann, A. J. Virus Structure. in 1–9 (John Wiley &amp; Sons, Ltd, 2015). doi:<a href="https://doi.org/10.1002/9780470015902.a0000439.pub2">10.1002/9780470015902.a0000439.pub2</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">473. </div><div class="csl-right-inline">Hladik, F. &amp; McElrath, M. J. <a href="https://doi.org/10.1038/nri2302">Setting the stage: host invasion by HIV</a>. <em>Nature Reviews Immunology</em> <strong>8</strong>, 447–457 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">474. </div><div class="csl-right-inline">Shaw, G. M. &amp; Hunter, E. <a href="https://doi.org/10.1101/cshperspect.a006965">HIV Transmission</a>. <em>Cold Spring Harbor Perspectives in Medicine</em> <strong>2</strong>, a006965 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">475. </div><div class="csl-right-inline">Weiss, R. A. <a href="https://doi.org/10.1126/science.8493571">How does HIV cause AIDS?</a> <em>Science</em> <strong>260</strong>, 1273–1279 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">476. </div><div class="csl-right-inline">Melhuish, A. &amp; Lewthwaite, P. <a href="https://doi.org/10.1016/j.mpmed.2018.03.010">Natural history of HIV and AIDS</a>. <em>Medicine</em> <strong>46</strong>, 356–361 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">477. </div><div class="csl-right-inline">Murray, J. F. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM198406213102529">Pulmonary complications of the acquired immunodeficiency syndrome</a>. <em>New England Journal of Medicine</em> <strong>310</strong>, 1682–1688 (1984).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">478. </div><div class="csl-right-inline">Sampath, S. <em>et al.</em> <a href="https://doi.org/10.7759/cureus.18136">Pandemics Throughout the History</a>. <em>Cureus</em> <strong>13</strong>, (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">479. </div><div class="csl-right-inline">World Health Organization. <em>Global report: UNAIDS report on the global AIDS epidemic 2010.</em> (World Health Organization, 2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">480. </div><div class="csl-right-inline">Barré-Sinoussi, F. <em>et al.</em> <a href="https://doi.org/10.1126/science.6189183">Isolation of a T-lymphotropic retrovirus from a patient at risk for acquired immune deficiency syndrome (AIDS)</a>. <em>Science (New York, N.Y.)</em> <strong>220</strong>, 868–871 (1983).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">481. </div><div class="csl-right-inline">Gallo, R. C. <em>et al.</em> <a href="https://doi.org/10.1126/science.6601823">Isolation of human T-cell leukemia virus in acquired immune deficiency syndrome (AIDS)</a>. <em>Science (New York, N.Y.)</em> <strong>220</strong>, 865–867 (1983).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">482. </div><div class="csl-right-inline">Clavel, F. <em>et al.</em> <a href="https://doi.org/10.1126/science.2425430">Isolation of a new human retrovirus from west african patients with AIDS</a>. <em>Science</em> <strong>233</strong>, 343–346 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">483. </div><div class="csl-right-inline">Gilbert, P. B. <em>et al.</em> <a href="https://doi.org/10.1002/sim.1342">Comparison of HIV-1 and HIV-2 infectivity from a prospective cohort study in Senegal</a>. <em>Statistics in Medicine</em> <strong>22</strong>, 573–593 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">484. </div><div class="csl-right-inline">Loeff, M. F. S. van der <em>et al.</em> <a href="https://doi.org/10.1093/ije/dyl037">Sixteen years of HIV surveillance in a West African research clinic reveals divergent epidemic trends of HIV-1 and HIV-2</a>. <em>International Journal of Epidemiology</em> <strong>35</strong>, 1322–1328 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">485. </div><div class="csl-right-inline">Gao, F. <em>et al.</em> <a href="https://doi.org/10.1038/17130">Origin of HIV-1 in the chimpanzee Pan troglodytes troglodytes</a>. <em>Nature</em> <strong>397</strong>, 436–441 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">486. </div><div class="csl-right-inline">Hamel, D. J. <em>et al.</em> <a href="https://doi.org/10.1089/aid.2007.0037">Twenty years of prospective molecular epidemiology in Senegal: changes in HIV diversity</a>. <em>AIDS research and human retroviruses</em> <strong>23</strong>, 1189–1196 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">487. </div><div class="csl-right-inline">Sharp, P. M. &amp; Hahn, B. H. <a href="https://doi.org/10.1101/cshperspect.a006841">Origins of HIV and the AIDS pandemic</a>. <em>Cold Spring Harbor Perspectives in Medicine:</em> <strong>1</strong>, a006841 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">488. </div><div class="csl-right-inline">Hirsch, V. M., Olmsted, R. A., Murphey-Corb, M., Purcell, R. H. &amp; Johnson, P. R. <a href="https://doi.org/10.1038/339389a0">An African primate lentivirus (SIVsmclosely related to HIV-2</a>. <em>Nature</em> <strong>339</strong>, 389–392 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">489. </div><div class="csl-right-inline">Gao, F. <em>et al.</em> <a href="https://doi.org/10.1038/358495a0">Human infection by genetically diverse SIVSM-related HIV-2 in West Africa</a>. <em>Nature</em> <strong>358</strong>, 495–499 (1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">490. </div><div class="csl-right-inline">Chen, Z. <em>et al.</em> <a href="https://doi.org/10.1128/JVI.70.6.3617-3627.1996">Genetic characterization of new West African simian immunodeficiency virus SIVsm: geographic clustering of household-derived SIV strains with human immunodeficiency virus type 2 subtypes and genetically diverse viruses from a single feral sooty mangabey troop</a>. <em>Journal of Virology</em> <strong>70</strong>, 3617–3627 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">491. </div><div class="csl-right-inline">Hemelaar, J. <a href="https://doi.org/10.1016/j.molmed.2011.12.001">The origin and diversity of the HIV-1 pandemic</a>. <em>Trends in Molecular Medicine</em> <strong>18</strong>, 182–192 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">492. </div><div class="csl-right-inline">Worobey, M. <em>et al.</em> <a href="https://doi.org/10.1038/nature07390">Direct evidence of extensive diversity of HIV-1 in Kinshasa by 1960</a>. <em>Nature</em> <strong>455</strong>, 661–664 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">493. </div><div class="csl-right-inline">Vidal, N. <em>et al.</em> <a href="https://doi.org/10.1128/jvi.74.22.10498-10507.2000">Unprecedented degree of human immunodeficiency virus type 1 (HIV-1) group M genetic diversity in the Democratic Republic of Congo suggests that the HIV-1 pandemic originated in Central Africa</a>. <em>Journal of Virology</em> <strong>74</strong>, 10498–10507 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">494. </div><div class="csl-right-inline">Faria, N. R. <em>et al.</em> <a href="https://doi.org/10.1126/science.1256739">The early spread and epidemic ignition of HIV-1 in human populations</a>. <em>Science (New York, N.Y.)</em> <strong>346</strong>, 56–61 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">495. </div><div class="csl-right-inline">Korber, B. <em>et al.</em> <a href="https://doi.org/10.1126/science.288.5472.1789">Timing the ancestor of the HIV-1 pandemic strains</a>. <em>Science (New York, N.Y.)</em> <strong>288</strong>, 1789–1796 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">496. </div><div class="csl-right-inline">Rambaut, A., Posada, D., Crandall, K. A. &amp; Holmes, E. C. <a href="https://doi.org/10.1038/nrg1246">The causes and consequences of HIV evolution</a>. <em>Nature Reviews Genetics</em> <strong>5</strong>, 52–61 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">497. </div><div class="csl-right-inline">McCutchan, F. E. <a href="https://doi.org/10.1002/jmv.20599">Global epidemiology of HIV</a>. <em>Journal of Medical Virology</em> <strong>78</strong>, S7–S12 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">498. </div><div class="csl-right-inline">Pérez-Losada, M., Arenas, M., Galán, J. C., Palero, F. &amp; González-Candelas, F. <a href="https://doi.org/10.1016/j.meegid.2014.12.022">Recombination in viruses: Mechanisms, methods of study, and evolutionary consequences</a>. <em>Infection, Genetics and Evolution</em> <strong>30</strong>, 296–307 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">499. </div><div class="csl-right-inline">Robertson, D. L., Hahn, B. H. &amp; Sharp, P. M. <a href="https://doi.org/10.1007/BF00163230">Recombination in AIDS viruses</a>. <em>Journal of Molecular Evolution</em> <strong>40</strong>, 249–259 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">500. </div><div class="csl-right-inline"><a href="https://www.hiv.lanl.gov/content/sequence/HIV/CRFs/CRFs.html">HIV circulating recombinant forms (CRFs)</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">501. </div><div class="csl-right-inline">Lau, K. A. &amp; Wong, J. J. L. <a href="https://doi.org/10.4081/idr.2013.s1.e4">Current trends of HIV recombination worldwide</a>. <em>Infectious Disease Reports</em> <strong>5</strong>, e4 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">502. </div><div class="csl-right-inline">Posada, D., Crandall, K. A. &amp; Holmes, E. C. <a href="https://doi.org/10.1146/annurev.genet.36.040202.111115">Recombination in evolutionary genomics</a>. <em>Annual Review of Genetics</em> <strong>36</strong>, 75–97 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">503. </div><div class="csl-right-inline">Taylor, B. S., Sobieszczyk, M. E., McCutchan, F. E. &amp; Hammer, S. M. <a href="https://doi.org/10.1056/NEJMra0706737">The challenge of HIV-1 subtype diversity</a>. <em>New England Journal of Medicine</em> <strong>358</strong>, 1590–1602 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">504. </div><div class="csl-right-inline">Hemelaar, J., Gouws, E., Ghys, P. D. &amp; Osmanov, S. <a href="https://doi.org/10.1097/QAD.0b013e328342ff93">Global trends in molecular epidemiology of HIV-1 during 2000<span></span>2007</a>. <em>AIDS (London, England)</em> <strong>25</strong>, 679–689 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">505. </div><div class="csl-right-inline"><a href="https://www.hiv.lanl.gov/components/sequence/HIV/geo/geo.comp">Distribution of all HIV-1 sequences: WORLD</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">506. </div><div class="csl-right-inline">Freed, E. O. <a href="https://doi.org/10.1023/A:1021070512287">HIV-1 Replication</a>. <em>Somatic Cell and Molecular Genetics</em> <strong>26</strong>, 13–33 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">507. </div><div class="csl-right-inline">Ferguson, M. R., Rojo, D. R., Lindern, J. J. von &amp; O’Brien, W. A. <a href="https://doi.org/10.1016/s0272-2712(02)00015-x">HIV-1 replication cycle</a>. <em>Clinics in Laboratory Medicine</em> <strong>22</strong>, 611–635 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">508. </div><div class="csl-right-inline">Gougeon, M. L., Laurent-Crawford, A. G., Hovanessian, A. G. &amp; Montagnier, L. <a href="https://doi.org/10.1006/smim.1993.1022">Direct and indirect mechanisms mediating apoptosis during HIV infection: contribution to in vivo CD4 T cell depletion</a>. <em>Seminars in Immunology</em> <strong>5</strong>, 187–194 (1993).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">509. </div><div class="csl-right-inline">Vidya Vijayan, K. K., Karthigeyan, K. P., Tripathi, S. P. &amp; Hanna, L. E. <a href="https://doi.org/10.3389/fimmu.2017.00580">Pathophysiology of CD4+ t-cell depletion in HIV-1 and HIV-2 infections</a>. <em>Frontiers in Immunology</em> <strong>8</strong>, 580 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">510. </div><div class="csl-right-inline">Frankel, A. D. &amp; Young, J. A. <a href="https://doi.org/10.1146/annurev.biochem.67.1.1">HIV-1: fifteen proteins and an RNA</a>. <em>Annual Review of Biochemistry</em> <strong>67</strong>, 1–25 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">511. </div><div class="csl-right-inline">Fossen, T. <em>et al.</em> <a href="https://doi.org/10.1074/jbc.M507375200">Solution structure of the human immunodeficiency virus type 1 p6 protein *</a>. <em>Journal of Biological Chemistry</em> <strong>280</strong>, 42515–42527 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">512. </div><div class="csl-right-inline">Göttlinger, H. G., Dorfman, T., Sodroski, J. G. &amp; Haseltine, W. A. <a href="https://doi.org/10.1073/pnas.88.8.3195">Effect of mutations affecting the p6 gag protein on human immunodeficiency virus particle release.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>88</strong>, 3195–3199 (1991).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">513. </div><div class="csl-right-inline">Huang, M., Orenstein, J. M., Martin, M. A. &amp; Freed, E. O. <a href="https://doi.org/10.1128/jvi.69.11.6810-6818.1995">p6Gag is required for particle production from full-length human immunodeficiency virus type 1 molecular clones expressing protease</a>. <em>Journal of Virology</em> <strong>69</strong>, 6810–6818 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">514. </div><div class="csl-right-inline">Bour, S., Geleziunas, R. &amp; Wainberg, M. A. <a href="https://doi.org/10.1128/mr.59.1.63-93.1995">The human immunodeficiency virus type 1 (HIV-1) CD4 receptor and its central role in promotion of HIV-1 infection</a>. <em>Microbiological Reviews</em> <strong>59</strong>, 63–93 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">515. </div><div class="csl-right-inline">Hernandez, L. D., Hoffman, L. R., Wolfsberg, T. G. &amp; White, J. M. <a href="https://doi.org/10.1146/annurev.cellbio.12.1.627">Virus-cell and cell-cell fusion</a>. <em>Annual Review of Cell and Developmental Biology</em> <strong>12</strong>, 627–661 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">516. </div><div class="csl-right-inline">Jones, K. &amp; Peterlin, B. <a href="https://doi.org/10.1146/annurev.bi.63.070194.003441">Control of rna initiation and elongation at the hiv-1 promoter</a>. <em>Annual Review of Biochemistry</em> <strong>63</strong>, 717–743 (1994).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">517. </div><div class="csl-right-inline">Hope, T. J. <a href="https://doi.org/10.1016/S1074-5521(97)90124-1">Viral RNA export</a>. <em>Chemistry &amp; Biology</em> <strong>4</strong>, 335–344 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">518. </div><div class="csl-right-inline">Mangasarian, A. &amp; Trono, D. <a href="https://doi.org/10.1016/S0923-2516(97)81909-7">The multifaceted role of HIV Nef</a>. <em>Research in Virology</em> <strong>148</strong>, 30–33 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">519. </div><div class="csl-right-inline">Cohen, é. A., Subbramanian, R. A. &amp; Göttlinger, H. G. Role of Auxiliary Proteins in Retroviral Morphogenesis. in (ed. Kräusslich, H.-G.) 219–235 (Springer, 1996). doi:<a href="https://doi.org/10.1007/978-3-642-80145-7_7">10.1007/978-3-642-80145-7_7</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">520. </div><div class="csl-right-inline">Lamb, R. A. &amp; Pinto, L. H. <a href="https://doi.org/10.1006/viro.1997.8451">Do Vpu and Vpr of Human Immunodeficiency Virus Type 1 and NB of Influenza B Virus Have Ion Channel Activities in the Viral Life Cycles?</a> <em>Virology</em> <strong>229</strong>, 1–11 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">521. </div><div class="csl-right-inline">Khan, N. &amp; Geiger, J. D. <a href="https://doi.org/10.3390/v13081466">Role of viral protein u (vpu) in HIV-1 infection and pathogenesis</a>. <em>Viruses</em> <strong>13</strong>, 1466 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">522. </div><div class="csl-right-inline">Emerman, M. <a href="https://doi.org/10.1016/S0960-9822(02)00676-0">HIV-1, Vpr and the cell cycle</a>. <em>Current Biology</em> <strong>6</strong>, 1096–1103 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">523. </div><div class="csl-right-inline">Miller, R. H. <a href="https://doi.org/10.1126/science.3347840">Human immunodeficiency virus may encode a novel protein on the genomic DNA plus strand</a>. <em>Science (New York, N.Y.)</em> <strong>239</strong>, 1420–1422 (1988).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">524. </div><div class="csl-right-inline">Briquet, S. &amp; Vaquero, C. <a href="https://doi.org/10.1006/viro.2001.1224">Immunolocalization Studies of an Antisense Protein in HIV-1-Infected Cells and Viral Particles</a>. <em>Virology</em> <strong>292</strong>, 177–184 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">525. </div><div class="csl-right-inline">Cassan, E., Arigon-Chifolleau, A.-M., Mesnard, J.-M., Gross, A. &amp; Gascuel, O. <a href="https://doi.org/10.1073/pnas.1605739113">Concomitant emergence of the antisense protein gene of HIV-1 and of the pandemic</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>113</strong>, 11537–11542 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">526. </div><div class="csl-right-inline">Savoret, J. <em>et al.</em> A pilot study of the humoral response against the AntiSense protein (ASP) in HIV-1-infected patients. <em>Frontiers in Microbiology</em> <strong>11</strong>, (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">527. </div><div class="csl-right-inline">Zardecki, C. <em>et al.</em> <a href="https://doi.org/10.1002/pro.4200">PDB-101: Educational resources supporting molecular explorations through biology and medicine</a>. <em>Protein Science</em> <strong>31</strong>, 129–140 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">528. </div><div class="csl-right-inline">Eisinger, R. W., Dieffenbach, C. W. &amp; Fauci, A. S. <a href="https://doi.org/10.1001/jama.2018.21167">HIV viral load and transmissibility of HIV infection: Undetectable equals untransmittable</a>. <em>JAMA</em> <strong>321</strong>, 451–452 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">529. </div><div class="csl-right-inline">Palella, F. J. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM199803263381301">Declining morbidity and mortality among patients with advanced human immunodeficiency virus infection</a>. <em>New England Journal of Medicine</em> <strong>338</strong>, 853–860 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">530. </div><div class="csl-right-inline">Forsythe, S. S. <em>et al.</em> <a href="https://doi.org/10.1377/hlthaff.2018.05391">Twenty years of antiretroviral therapy for people living with HIV: Global costs, health achievements, economic benefits</a>. <em>Health Affairs</em> <strong>38</strong>, 1163–1172 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">531. </div><div class="csl-right-inline">Fischl, M. A. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM198707233170401">The efficacy of azidothymidine (AZT) in the treatment of patients with AIDS and AIDS-related complex</a>. <em>New England Journal of Medicine</em> <strong>317</strong>, 185–191 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">532. </div><div class="csl-right-inline">Richman, D. D. <a href="https://doi.org/10.1016/0002-9343(90)90414-9">Susceptibility to nucleoside analogues of zidovudine-resistant isolates of human immunodeficiency virus</a>. <em>The American Journal of Medicine</em> <strong>88</strong>, S8–S10 (1990).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">533. </div><div class="csl-right-inline">Yeo, J. Y., Goh, G.-R., Su, C. T.-T. &amp; Gan, S. K.-E. <a href="https://doi.org/10.3390/v12030297">The Determination of HIV-1 RT Mutation Rate, Its Possible Allosteric Effects, and Its Implications on Drug Resistance</a>. <em>Viruses</em> <strong>12</strong>, 297 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">534. </div><div class="csl-right-inline">Cuevas, J. M., Geller, R., Garijo, R., López-Aldeguer, J. &amp; Sanjuán, R. <a href="https://doi.org/10.1371/journal.pbio.1002251">Extremely High Mutation Rate of HIV-1 In Vivo</a>. <em>PLOS Biology</em> <strong>13</strong>, e1002251 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">535. </div><div class="csl-right-inline">Carvajal-Rodríguez, A., Crandall, K. A. &amp; Posada, D. <a href="https://doi.org/10.1016/j.meegid.2007.02.001">Recombination favors the evolution of drug resistance in HIV-1 during antiretroviral therapy</a>. <em>Infection, genetics and evolution : journal of molecular epidemiology and evolutionary genetics in infectious diseases</em> <strong>7</strong>, 476–483 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">536. </div><div class="csl-right-inline">Gulick, R. M. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM199709113371102">Treatment with indinavir, zidovudine, and lamivudine in adults with human immunodeficiency virus infection and prior antiretroviral therapy</a>. <em>The New England Journal of Medicine</em> <strong>337</strong>, 734–739 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">537. </div><div class="csl-right-inline">Wensing, A. M. J., Maarseveen, N. M. van &amp; Nijhuis, M. <a href="https://doi.org/10.1016/j.antiviral.2009.10.003">Fifteen years of HIV Protease Inhibitors: raising the barrier to resistance</a>. <em>Antiviral Research</em> <strong>85</strong>, 59–74 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">538. </div><div class="csl-right-inline">Pedersen, O. S. &amp; Pedersen, E. B. <a href="https://doi.org/10.1177/095632029901000601">Non-Nucleoside Reverse Transcriptase Inhibitors: The NNRTI Boom</a>. <em>Antiviral Chemistry and Chemotherapy</em> <strong>10</strong>, 285–314 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">539. </div><div class="csl-right-inline">Scarsi, K. K., Havens, J. P., Podany, A. T., Avedissian, S. N. &amp; Fletcher, C. V. <a href="https://doi.org/10.1007/s40265-020-01379-9">HIV-1 Integrase Inhibitors: A Comparative Review of Efficacy and Safety</a>. <em>Drugs</em> <strong>80</strong>, 1649–1676 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">540. </div><div class="csl-right-inline">Fletcher, C. V. <a href="https://doi.org/10.1016/S0140-6736(03)13323-5">Enfuvirtide, a new drug for HIV infection</a>. <em>The Lancet</em> <strong>361</strong>, 1577–1578 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">541. </div><div class="csl-right-inline">Esté, J. A. &amp; Telenti, A. <a href="https://doi.org/10.1016/S0140-6736(07)61052-6">HIV entry inhibitors</a>. <em>The Lancet</em> <strong>370</strong>, 81–88 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">542. </div><div class="csl-right-inline">Kilby, J. M. &amp; Eron, J. J. <a href="https://doi.org/10.1056/NEJMra022812">Novel therapies based on mechanisms of HIV-1 cell entry</a>. <em>New England Journal of Medicine</em> <strong>348</strong>, 2228–2238 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">543. </div><div class="csl-right-inline">Yeni, P. <a href="https://doi.org/10.1016/j.jhep.2005.11.021">Update on HAART in HIV</a>. <em>Journal of Hepatology</em> <strong>44</strong>, S100–S103 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">544. </div><div class="csl-right-inline">Palmisano, L. &amp; Vella, S. <a href="https://doi.org/10.4415/ANN_11_01_10">A brief history of antiretroviral therapy of HIV infection: success and challenges</a>. <em>Annali dell’Istituto Superiore Di Sanita</em> <strong>47</strong>, 44–48 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">545. </div><div class="csl-right-inline">Pennings, P. S. <a href="https://doi.org/10.4081/idr.2013.s1.e5">HIV drug resistance: problems and perspectives</a>. <em>Infectious Disease Reports</em> <strong>5</strong>, e5 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">546. </div><div class="csl-right-inline">Mehta, S., Moore, R. D. &amp; Graham, N. M. H. Potential factors affecting adherence with HIV therapy. <em>AIDS</em> <strong>11</strong>, 16651670 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">547. </div><div class="csl-right-inline">Miller, N. H. <a href="https://doi.org/10.1016/S0002-9343(97)00467-1">Compliance with treatment regimens in chronic asymptomatic diseases</a>. <em>The American Journal of Medicine</em> <strong>102</strong>, 43–49 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">548. </div><div class="csl-right-inline">Chesney, M. A., Morin, M. &amp; Sherr, L. <a href="https://doi.org/10.1016/S0277-9536(99)00468-2">Adherence to HIV combination therapy</a>. <em>Social Science &amp; Medicine</em> <strong>50</strong>, 1599–1605 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">549. </div><div class="csl-right-inline">Aldir, I., Horta, A. &amp; Serrado, M. <a href="https://doi.org/10.1185/03007995.2013.844685">Single-tablet regimens in HIV: Does it really make a difference?</a> <em>Current Medical Research and Opinion</em> <strong>30</strong>, 89–97 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">550. </div><div class="csl-right-inline">Grant, R. M. <em>et al.</em> <a href="https://doi.org/10.1056/NEJMoa1011205">Preexposure chemoprophylaxis for HIV prevention in men who have sex with men</a>. <em>New England Journal of Medicine</em> <strong>363</strong>, 2587–2599 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">551. </div><div class="csl-right-inline">Baeten, J. M. <em>et al.</em> <a href="https://doi.org/10.1056/NEJMoa1108524">Antiretroviral prophylaxis for HIV prevention in heterosexual men and women</a>. <em>The New England Journal of Medicine</em> <strong>367</strong>, 399–410 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">552. </div><div class="csl-right-inline">Buchbinder, S. P. &amp; Liu, A. <a href="https://doi.org/10.1007/s10461-011-9894-1">Pre-exposure prophylaxis and the promise of combination prevention approaches</a>. <em>AIDS and behavior</em> <strong>15 Suppl 1</strong>, S72–79 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">553. </div><div class="csl-right-inline">Riddell, J. I., Amico, K. R. &amp; Mayer, K. H. <a href="https://doi.org/10.1001/jama.2018.1917">HIV preexposure prophylaxis: A review</a>. <em>JAMA</em> <strong>319</strong>, 1261–1268 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">554. </div><div class="csl-right-inline">EMA. <a href="https://www.ema.europa.eu/en/medicines/human/EPAR/truvada">Truvada</a>. (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">555. </div><div class="csl-right-inline"><a href="https://www.cdc.gov/hiv/basics/prep/about-prep.html">About PrEP | PrEP | HIV basics | HIV/AIDS | CDC</a>. (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">556. </div><div class="csl-right-inline">Zolopa, A. R. <a href="https://doi.org/10.1016/j.antiviral.2009.10.018">The evolution of HIV treatment guidelines: Current state-of-the-art of ART</a>. <em>Antiviral Research</em> <strong>85</strong>, 241–244 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">557. </div><div class="csl-right-inline">World Health Organization. <em>Consolidated guidelines on HIV prevention, testing, treatment, service delivery and monitoring: recommendations for a public health approach</em>. (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">558. </div><div class="csl-right-inline">Ammaranond, P. &amp; Sanguansittianan, S. <a href="https://doi.org/10.1111/j.1472-8206.2011.01009.x">Mechanism of HIV antiretroviral drugs progress toward drug resistance</a>. <em>Fundamental &amp; Clinical Pharmacology</em> <strong>26</strong>, 146–161 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">559. </div><div class="csl-right-inline">Clavel, F. &amp; Hance, A. J. <a href="https://doi.org/10.1056/NEJMra025195">HIV drug resistance</a>. <em>New England Journal of Medicine</em> <strong>350</strong>, 1023–1035 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">560. </div><div class="csl-right-inline">Sarafianos, S. G. <em>et al.</em> <a href="https://doi.org/10.1016/j.jmb.2008.10.071">Structure and function of <span>HIV</span>-1 reverse transcriptase: Molecular mechanisms of polymerization and inhibition</a>. <em>Journal of molecular biology</em> <strong>385</strong>, 693–713 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">561. </div><div class="csl-right-inline">Goodsell, D. S., Autin, L. &amp; Olson, A. J. <a href="https://doi.org/10.1016/j.str.2019.08.011">Illustrate: Software for Biomolecular Illustration</a>. <em>Structure (London, England: 1993)</em> <strong>27</strong>, 1716–1720.e1 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">562. </div><div class="csl-right-inline">Esnouf, R. M. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.94.8.3984">Unique features in the structure of the complex between HIV-1 reverse transcriptase and the bis(heteroaryl)piperazine (BHAP) U-90152 explain resistance mutations for this nonnucleoside inhibitor</a>. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>94</strong>, 3984–3989 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">563. </div><div class="csl-right-inline">Hang, J. Q. <em>et al.</em> <a href="https://doi.org/10.1016/j.bbrc.2004.03.061">Activity of the isolated HIV RNase H domain and specific inhibition by N-hydroxyimides</a>. <em>Biochemical and Biophysical Research Communications</em> <strong>317</strong>, 321–329 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">564. </div><div class="csl-right-inline">Klumpp, K. &amp; Mirzadegan, T. <a href="https://doi.org/10.2174/138161206776873653">Recent progress in the design of small molecule inhibitors of HIV RNase h</a>. <em>Current Pharmaceutical Design</em> <strong>12</strong>, 1909–1922 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">565. </div><div class="csl-right-inline">Menéndez-Arias, L. <a href="https://doi.org/10.1016/j.virusres.2007.12.015">Mechanisms of resistance to nucleoside analogue inhibitors of HIV-1 reverse transcriptase</a>. <em>Virus Research</em> <strong>134</strong>, 124–146 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">566. </div><div class="csl-right-inline">Sluis-Cremer, N., Arion, D. &amp; Parniak*, M. A. <a href="https://doi.org/10.1007/PL00000626">Molecular mechanisms of HIV-1 resistance to nucleoside reverse transcriptase inhibitors (NRTIs)</a>. <em>Cellular and Molecular Life Sciences CMLS</em> <strong>57</strong>, 1408–1422 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">567. </div><div class="csl-right-inline">Sarafianos, S. G. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.96.18.10027">Lamivudine (3TC) resistance in HIV-1 reverse transcriptase involves steric hindrance with beta-branched amino acids</a>. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>96</strong>, 10027–10032 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">568. </div><div class="csl-right-inline">Meyer, P. R., Matsuura, S. E., Mian, A. M., So, A. G. &amp; Scott, W. A. <a href="https://doi.org/10.1016/s1097-2765(00)80185-9">A mechanism of AZT resistance: an increase in nucleotide-dependent primer unblocking by mutant HIV-1 reverse transcriptase</a>. <em>Molecular Cell</em> <strong>4</strong>, 35–43 (1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">569. </div><div class="csl-right-inline">Boyer, P. L., Sarafianos, S. G., Arnold, E. &amp; Hughes, S. H. <a href="https://doi.org/10.1128/JVI.75.10.4832-4842.2001">Selective excision of AZTMP by drug-resistant human immunodeficiency virus reverse transcriptase</a>. <em>Journal of Virology</em> <strong>75</strong>, 4832–4842 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">570. </div><div class="csl-right-inline">Deeks, S. G. Nonnucleoside reverse transcriptase inhibitor resistance. <em>JAIDS Journal of Acquired Immune Deficiency Syndromes</em> <strong>26</strong>, S25 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">571. </div><div class="csl-right-inline">Ren, J. &amp; Stammers, D. K. <a href="https://doi.org/10.1016/j.virusres.2007.12.018">Structural basis for drug resistance mechanisms for non-nucleoside inhibitors of HIV reverse transcriptase</a>. <em>Virus Research</em> <strong>134</strong>, 157–170 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">572. </div><div class="csl-right-inline">Lloyd, S. B., Kent, S. J. &amp; Winnall, W. R. <a href="https://doi.org/10.1089/aid.2013.0153">The high cost of fidelity</a>. <em>AIDS Research and Human Retroviruses</em> <strong>30</strong>, 8–16 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">573. </div><div class="csl-right-inline">Pearl, L. H. &amp; Taylor, W. R. <a href="https://doi.org/10.1038/329351a0">A structural model for the retroviral proteases</a>. <em>Nature</em> <strong>329</strong>, 351–354 (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">574. </div><div class="csl-right-inline">Gulnik, S., Erickson, J. W. &amp; Xie, D. <a href="https://doi.org/10.1016/S0083-6729(00)58026-1">HIV protease: Enzyme function and drug resistance</a>. in vol. 58 213–256 (Academic Press, 2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">575. </div><div class="csl-right-inline">Silva, A. M., Cachau, R. E., Sham, H. L. &amp; Erickson, J. W. <a href="https://doi.org/10.1006/jmbi.1996.0026">Inhibition and catalytic mechanism of HIV-1 aspartic protease</a>. <em>Journal of Molecular Biology</em> <strong>255</strong>, 321–340 (1996).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">576. </div><div class="csl-right-inline">Hornak, V., Okur, A., Rizzo, R. C. &amp; Simmerling, C. <a href="https://doi.org/10.1073/pnas.0508452103">HIV-1 protease flaps spontaneously open and reclose in molecular dynamics simulations</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>103</strong>, 915–920 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">577. </div><div class="csl-right-inline">Freedberg, D. I. <em>et al.</em> Rapid structural fluctuations of the free HIV protease flaps in solution: Relationship to crystal structures and comparison with predictions of dynamics calculations. <em>Protein Science : A Publication of the Protein Society</em> <strong>11</strong>, 221–232 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">578. </div><div class="csl-right-inline">Yu, Y. <em>et al.</em> <a href="https://doi.org/10.1039/C7RA09691G">Structural insights into HIV-1 protease flap opening processes and key intermediates</a>. <em>RSC Advances</em> <strong>7</strong>, 45121–45128 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">579. </div><div class="csl-right-inline">Roberts, N. A. <em>et al.</em> <a href="https://doi.org/10.1126/science.2183354">Rational design of peptide-based HIV proteinase inhibitors</a>. <em>Science</em> <strong>248</strong>, 358–361 (1990).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">580. </div><div class="csl-right-inline">Lv, Z., Chu, Y. &amp; Wang, Y. <a href="https://doi.org/10.2147/HIV.S79956">HIV protease inhibitors: A review of molecular selectivity and toxicity</a>. <em>HIV/AIDS (Auckland, N.Z.)</em> <strong>7</strong>, 95–104 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">581. </div><div class="csl-right-inline">Prabu-Jeyabalan, M., Nalivaika, E. &amp; Schiffer, C. A. <a href="https://doi.org/10.1016/S0969-2126(02)00720-7">Substrate shape determines specificity of recognition for HIV-1 protease: Analysis of crystal structures of six substrate complexes</a>. <em>Structure</em> <strong>10</strong>, 369–381 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">582. </div><div class="csl-right-inline">Prabu-Jeyabalan, M. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.50.4.1518-1521.2006">Substrate envelope and drug resistance: Crystal structure of RO1 in complex with wild-type human immunodeficiency virus type 1 protease</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>50</strong>, 1518–1521 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">583. </div><div class="csl-right-inline">Kurt Yilmaz, N., Swanstrom, R. &amp; Schiffer, C. A. <a href="https://doi.org/10.1016/j.tim.2016.03.010">Improving Viral Protease Inhibitors to Counter Drug Resistance</a>. <em>Trends in Microbiology</em> <strong>24</strong>, 547–557 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">584. </div><div class="csl-right-inline">Chiu, T. K. &amp; Davies, D. R. <a href="https://doi.org/10.2174/1568026043388547">Structure and function of HIV-1 integrase</a>. <em>Current Topics in Medicinal Chemistry</em> <strong>4</strong>, 965–977 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">585. </div><div class="csl-right-inline">Esposito, D. &amp; Craigie, R. HIV Integrase Structure and Function. in (eds. Rlaramorosch, K., Murphy, F. A. &amp; Shawn, A. J.) vol. 52 319–333 (Academic Press, 1999).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">586. </div><div class="csl-right-inline">Delelis, O., Carayon, K., Saïb, A., Deprez, E. &amp; Mouscadet, J.-F. <a href="https://doi.org/10.1186/1742-4690-5-114">Integrase and integration: Biochemical activities of HIV-1 integrase</a>. <em>Retrovirology</em> <strong>5</strong>, 114 (2008).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">587. </div><div class="csl-right-inline">Maertens, G. N., Engelman, A. N. &amp; Cherepanov, P. <a href="https://doi.org/10.1038/s41579-021-00586-9">Structure and function of retroviral integrase</a>. <em>Nature Reviews Microbiology</em> <strong>20</strong>, 20–34 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">588. </div><div class="csl-right-inline">Pommier, Y., Johnson, A. A. &amp; Marchand, C. <a href="https://doi.org/10.1038/nrd1660">Integrase inhibitors to treat HIV/Aids</a>. <em>Nature Reviews Drug Discovery</em> <strong>4</strong>, 236–248 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">589. </div><div class="csl-right-inline">Blanco, J.-L., Varghese, V., Rhee, S.-Y., Gatell, J. M. &amp; Shafer, R. W. <a href="https://doi.org/10.1093/infdis/jir025">HIV-1 integrase inhibitor resistance and its clinical implications</a>. <em>The Journal of Infectious Diseases</em> <strong>203</strong>, 1204–1214 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">590. </div><div class="csl-right-inline">Geretti, A. M., Armenia, D. &amp; Ceccherini-Silberstein, F. <a href="https://doi.org/10.1097/QCO.0b013e32835a1de7">Emerging patterns and implications of HIV-1 integrase inhibitor resistance</a>. <em>Current Opinion in Infectious Diseases</em> <strong>25</strong>, 677686 (2012).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">591. </div><div class="csl-right-inline">Knox, D. C., Anderson, P. L., Harrigan, P. R. &amp; Tan, D. H. S. <a href="https://doi.org/10.1056/NEJMc1611639">Multidrug-resistant HIV-1 infection despite preexposure prophylaxis</a>. <em>New England Journal of Medicine</em> <strong>376</strong>, 501–502 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">592. </div><div class="csl-right-inline">Hurt, C. B., Eron, J. J. &amp; Cohen, M. S. <a href="https://doi.org/10.1093/cid/cir684">Pre-exposure prophylaxis and antiretroviral resistance: HIV prevention at a cost?</a> <em>Clinical Infectious Diseases: An Official Publication of the Infectious Diseases Society of America</em> <strong>53</strong>, 1265–1270 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">593. </div><div class="csl-right-inline">Gibas, K. M., Berg, P. van den, Powell, V. E. &amp; Krakower, D. S. <a href="https://doi.org/10.1007/s40265-019-01108-x">Drug Resistance During HIV Pre-Exposure Prophylaxis</a>. <em>Drugs</em> <strong>79</strong>, 609–619 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">594. </div><div class="csl-right-inline">Mourad, R. <em>et al.</em> <a href="https://doi.org/10.1097/QAD.0000000000000768">A phylotype-based analysis highlights the role of drug-naive <span>HIV</span>-positive individuals in the transmission of antiretroviral resistance in the <span>UK</span></a>. <em>Aids</em> <strong>29</strong>, 1917–1925 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">595. </div><div class="csl-right-inline">Hué, S. <em>et al.</em> <a href="https://doi.org/10.1128/JVI.01556-08">Demonstration of <span>Sustained Drug</span>-<span>Resistant Human Immunodeficiency Virus Type</span> 1 <span>Lineages Circulating</span> among <span>Treatment</span>-<span>Naïve Individuals</span></a>. <em>Journal of Virology</em> <strong>83</strong>, 2645–2654 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">596. </div><div class="csl-right-inline">Drescher, S. M. <em>et al.</em> <a href="https://doi.org/10.1093/cid/cit694">Treatment-naive individuals are the major source of transmitted HIV-1 drug resistance in men who have sex with men in the swiss HIV cohort study</a>. <em>Clinical Infectious Diseases</em> <strong>58</strong>, 285–294 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">597. </div><div class="csl-right-inline">Boerma, R. S. <em>et al.</em> <a href="https://doi.org/10.7448/IAS.19.1.21140">High levels of pre-treatment HIV drug resistance and treatment failure in Nigerian children</a>. <em>Journal of the International AIDS Society</em> <strong>19</strong>, 21140 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">598. </div><div class="csl-right-inline">Clutter, D. S., Jordan, M. R., Bertagnolio, S. &amp; Shafer, R. W. <a href="https://doi.org/10.1016/j.meegid.2016.08.031">HIV-1 drug resistance and resistance testing</a>. <em>Infection, Genetics and Evolution</em> <strong>46</strong>, 292–307 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">599. </div><div class="csl-right-inline">Kühnert, D. <em>et al.</em> <a href="https://doi.org/10.1371/journal.ppat.1006895">Quantifying the fitness cost of HIV-1 drug resistance mutations through phylodynamics</a>. <em>PLOS Pathogens</em> <strong>14</strong>, e1006895 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">600. </div><div class="csl-right-inline">Mesplède, T. <em>et al.</em> <a href="https://doi.org/10.1186/1742-4690-10-22">Viral fitness cost prevents HIV-1 from evading dolutegravir drug pressure</a>. <em>Retrovirology</em> <strong>10</strong>, 22 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">601. </div><div class="csl-right-inline">Castro, H. <em>et al.</em> <a href="https://doi.org/10.1093/infdis/jit345">Persistence of HIV-1 Transmitted Drug Resistance Mutations</a>. <em>The Journal of Infectious Diseases</em> <strong>208</strong>, 1459–1463 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">602. </div><div class="csl-right-inline">Blassel, L. <em>et al.</em> <a href="https://doi.org/10.1016/j.coviro.2021.09.009">Drug resistance mutations in HIV: new bioinformatics approaches and challenges</a>. <em>Current Opinion in Virology</em> <strong>51</strong>, 56–64 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">603. </div><div class="csl-right-inline">UK CHIC Steering Committee. <a href="https://doi.org/10.1111/j.1468-1293.2004.00197.x">The creation of a large UK-based multicentre cohort of HIV-infected individuals: The UK Collaborative HIV Cohort (UK CHIC) Study</a>. <em>HIV Medicine</em> <strong>5</strong>, 115–124 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">604. </div><div class="csl-right-inline">Abeler-Dörner, L. <em>et al.</em> <a href="https://doi.org/10.1097/COH.0000000000000542">PANGEA-HIV 2: Phylogenetics and networks for generalised epidemics in africa</a>. <em>Current Opinion in HIV and AIDS</em> <strong>14</strong>, 173180 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">605. </div><div class="csl-right-inline">Shafer, R. W. <a href="https://doi.org/10.1086/505356">Rationale and Uses of a Public HIV Drug<span>-</span>Resistance Database</a>. <em>The Journal of Infectious Diseases</em> <strong>194</strong>, S51–S58 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">606. </div><div class="csl-right-inline">Kuiken, C., Korber, B. &amp; Shafer, R. W. HIV sequence databases. <em>AIDS reviews</em> <strong>5</strong>, 52–61 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">607. </div><div class="csl-right-inline">Wensing, A. M. <em>et al.</em> 2019 update of the drug resistance mutations in HIV-1. <em>Topics in Antiviral Medicine</em> <strong>27</strong>, 111–121 (2019).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">608. </div><div class="csl-right-inline">Clark, S. A., Calef, C. &amp; Mellors, J. W. Mutations in retroviral genes associated with drug resistance. <em>HIV sequence compendium</em> 58158 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">609. </div><div class="csl-right-inline">Liu, T. F. &amp; Shafer, R. W. <a href="https://doi.org/10.1086/503914">Web <span>Resources</span> for <span>HIV Type</span> 1 <span>Genotypic</span>-<span>Resistance Test Interpretation</span></a>. <em>Clinical Infectious Diseases</em> <strong>42</strong>, 1608–1618 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">610. </div><div class="csl-right-inline">Johnson, V. A. <em>et al.</em> Update of the drug resistance mutations in HIV-1: March 2013. <em>Topics in Antiviral Medicine</em> <strong>21</strong>, 6–7 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">611. </div><div class="csl-right-inline">Villabona-Arenas, C. J. <em>et al.</em> <a href="https://doi.org/10.1097/QAD.0000000000001233">In-depth analysis of <span>HIV</span>-1 drug resistance mutations in <span>HIV</span>-infected individuals failing first-line regimens in <span>West</span> and <span>Central Africa</span></a>. <em>AIDS</em> <strong>30</strong>, 2577 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">612. </div><div class="csl-right-inline">Shulman, N. S., Bosch, R. J., Mellors, J. W., Albrecht, M. A. &amp; Katzenstein, D. A. Genetic correlates of efavirenz hypersusceptibility. <em>AIDS</em> <strong>18</strong>, 17811785 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">613. </div><div class="csl-right-inline">Miller, M. D. <em>et al.</em> <a href="https://doi.org/10.1086/381784">Genotypic and phenotypic predictors of the magnitude of response to tenofovir disoproxil fumarate treatment in antiretroviral-experienced patients</a>. <em>The Journal of Infectious Diseases</em> <strong>189</strong>, 837–846 (2004).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">614. </div><div class="csl-right-inline">Brown, B. W. &amp; Russell, K. <a href="https://doi.org/10.1002/(SICI)1097-0258(19971130)16:22&lt;2511::AID-SIM693&gt;3.0.CO;2-4">Methods correcting for multiple testing: operating characteristics</a>. <em>Statistics in Medicine</em> <strong>16</strong>, 2511–2528 (1997).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">615. </div><div class="csl-right-inline">Austin, P. C., Mamdani, M. M., Juurlink, D. N. &amp; Hux, J. E. <a href="https://doi.org/10.1016/j.jclinepi.2006.01.012">Testing multiple statistical hypotheses resulted in spurious associations: a study of astrological signs and health</a>. <em>Journal of Clinical Epidemiology</em> <strong>59</strong>, 964–969 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">616. </div><div class="csl-right-inline">Hochberg, Y. &amp; Tamhane, A. C. <em>Multiple comparison procedures</em>. (1987).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">617. </div><div class="csl-right-inline">Benjamini, Y. &amp; Hochberg, Y. Controlling the false discovery rate: A practical and powerful approach to multiple testing. <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> <strong>57</strong>, 289–300 (1995).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">618. </div><div class="csl-right-inline">Gonzales, M. J. <em>et al.</em> <a href="https://doi.org/10.1097/01.aids.0000050860.71999.23">Extended spectrum of HIV-1 reverse transcriptase mutations in patients receiving multiple nucleoside analog inhibitors</a>. <em>AIDS (London, England)</em> <strong>17</strong>, 791–799 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">619. </div><div class="csl-right-inline">Seoighe, C. <em>et al.</em> <a href="https://doi.org/10.1093/molbev/msm021">A model of directional selection applied to the evolution of drug resistance in HIV-1</a>. <em>Molecular Biology and Evolution</em> <strong>24</strong>, 1025–1031 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">620. </div><div class="csl-right-inline">Sham, P. C. &amp; Purcell, S. M. <a href="https://doi.org/10.1038/nrg3706">Statistical power and significance testing in large-scale genetic studies</a>. <em>Nature Reviews Genetics</em> <strong>15</strong>, 335–346 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">621. </div><div class="csl-right-inline">Alizon, S. <em>et al.</em> <a href="https://doi.org/10.1371/journal.ppat.1001123">Phylogenetic Approach Reveals That Virus Genotype Largely Determines HIV Set-Point Viral Load</a>. <em>PLOS Pathogens</em> <strong>6</strong>, e1001123 (2010).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">622. </div><div class="csl-right-inline">Flynn, W. F. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pcbi.1004249">Deep Sequencing of Protease Inhibitor Resistant HIV Patient Isolates Reveals Patterns of Correlated Mutations in Gag and Protease</a>. <em>PLOS Computational Biology</em> <strong>11</strong>, e1004249 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">623. </div><div class="csl-right-inline">Petropoulos, C. J. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.44.4.920-928.2000">A novel phenotypic drug susceptibility assay for human immunodeficiency virus type 1</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>44</strong>, 920–928 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">624. </div><div class="csl-right-inline">Hertogs, K. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.42.2.269">A rapid method for simultaneous detection of phenotypic resistance to inhibitors of protease and reverse transcriptase in recombinant human immunodeficiency virus type 1 isolates from patients treated with antiretroviral drugs</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>42</strong>, 269–276 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">625. </div><div class="csl-right-inline">Heilek-Snyder, G. &amp; Bean, P. Role of HIV phenotypic assays in the management of HIV infection. <em>American Clinical Laboratory</em> <strong>21</strong>, 40–43 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">626. </div><div class="csl-right-inline">Moyle, G. J. <em>et al.</em> <a href="https://doi.org/10.1086/428096">Epidemiology and predictive factors for chemokine receptor use in HIV-1 infection</a>. <em>The Journal of Infectious Diseases</em> <strong>191</strong>, 866–872 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">627. </div><div class="csl-right-inline">Gartland, M. <em>et al.</em> <a href="https://doi.org/10.1093/jac/dkaa474">Susceptibility of global HIV-1 clinical isolates to fostemsavir using the PhenoSense® entry assay</a>. <em>Journal of Antimicrobial Chemotherapy</em> <strong>76</strong>, 648–652 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">628. </div><div class="csl-right-inline">Masquelier, B. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.45.6.1836-1842.2001">Genotypic and phenotypic resistance patterns of human immunodeficiency virus type 1 variants with insertions or deletions in the reverse transcriptase (RT): Multicenter study of patients treated with RT inhibitors</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>45</strong>, 1836–1842 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">629. </div><div class="csl-right-inline">Larder, B. A. &amp; Kemp, S. D. <a href="https://doi.org/10.1126/science.2479983">Multiple mutations in HIV-1 reverse transcriptase confer high-level resistance to zidovudine (AZT)</a>. <em>Science (New York, N.Y.)</em> <strong>246</strong>, 1155–1158 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">630. </div><div class="csl-right-inline">Vreese, K. de <em>et al.</em> <a href="https://doi.org/10.1016/0042-6822(92)90550-9">Resistance of human immunodeficiency virus type 1 reverse transcriptase to TIBO derivatives induced by site-directed mutagenesis</a>. <em>Virology</em> <strong>188</strong>, 900–904 (1992).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">631. </div><div class="csl-right-inline">Tambuyzer, L., Nijs, S., Daems, B., Picchio, G. &amp; Vingerhoets, J. <a href="https://doi.org/10.1097/QAI.0b013e3182237f74">Effect of mutations at position E138 in HIV-1 reverse transcriptase on phenotypic susceptibility and virologic response to etravirine</a>. <em>JAIDS Journal of Acquired Immune Deficiency Syndromes</em> <strong>58</strong>, 1822 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">632. </div><div class="csl-right-inline">Katzenstein, D. A. <em>et al.</em> Phenotypic susceptibility and virological outcome in nucleoside-experienced patients receiving three or four antiretroviral drugs. <em>AIDS</em> <strong>17</strong>, 821830 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">633. </div><div class="csl-right-inline">Blassel, L. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pcbi.1008873">Using machine learning and big data to explore the drug resistance landscape in HIV</a>. <em>PLOS Computational Biology</em> <strong>17</strong>, e1008873 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">634. </div><div class="csl-right-inline">Sheik Amamuddy, O., Bishop, N. T. &amp; Tastan Bishop, Ö. <a href="https://doi.org/10.1186/s12859-017-1782-x">Improving fold resistance prediction of <span>HIV</span>-1 against protease and reverse transcriptase inhibitors using artificial neural networks</a>. <em>BMC Bioinformatics</em> <strong>18</strong>, 369 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">635. </div><div class="csl-right-inline">Beerenwinkel, N. <em>et al.</em> <a href="https://doi.org/10.1109/5254.972080">Geno2pheno: Interpreting genotypic <span>HIV</span> drug resistance tests</a>. <em>IEEE Intelligent Systems</em> <strong>16</strong>, 35–41 (2001).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">636. </div><div class="csl-right-inline">Riemenschneider, M., Hummel, T. &amp; Heider, D. <a href="https://doi.org/10.1186/s12859-016-1179-2"><span>SHIVA</span> - a web application for drug resistance and tropism testing in <span>HIV</span></a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 314 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">637. </div><div class="csl-right-inline">Beerenwinkel, N. <em>et al.</em> Diversity and complexity of <span>HIV</span>-1 drug resistance: <span>A</span> bioinformatics approach to predicting phenotype from genotype. <em>Proceedings of the National Academy of Sciences</em> <strong>99</strong>, 8271–8276 (2002).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">638. </div><div class="csl-right-inline">Heider, D., Senge, R., Cheng, W. &amp; Hüllermeier, E. <a href="https://doi.org/10.1093/bioinformatics/btt331">Multilabel classification for exploiting cross-resistance information in <span>HIV</span>-1 drug resistance prediction</a>. <em>Bioinformatics</em> <strong>29</strong>, 1946–1952 (2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin"><a href="https://doi.org/10.1086/315301">639.Lepri, A. C. <em>et al.</em> <em>The Journal of Infectious Diseases</em> <strong>181</strong>, 1143–1147 (2000)</a></div>.
</div>
<div class="csl-entry">
<div class="csl-left-margin">640. </div><div class="csl-right-inline">Verhofstede, C. <em>et al.</em> <a href="https://doi.org/10.1002/jmv.20950">Detection of drug resistance mutations as a predictor of subsequent virological failure in patients with <span>HIV</span>-1 viral rebounds of less than 1,000 <span>RNA</span> copies/ml</a>. <em>Journal of Medical Virology</em> <strong>79</strong>, 1254–1260 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">641. </div><div class="csl-right-inline">Zhukova, A., Cutino-Moguel, T., Gascuel, O. &amp; Pillay, D. <a href="https://doi.org/10.1093/infdis/jix411">The <span>Role</span> of <span>Phylogenetics</span> as a <span>Tool</span> to <span>Predict</span> the <span>Spread</span> of <span>Resistance</span></a>. <em>The Journal of Infectious Diseases</em> <strong>216</strong>, S820–S823 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">642. </div><div class="csl-right-inline">Bennett, D. E. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0004724">Drug <span>Resistance Mutations</span> for <span>Surveillance</span> of <span>Transmitted HIV</span>-1 <span>Drug</span>-<span>Resistance</span>: 2009 <span>Update</span></a>. <em>PLOS ONE</em> <strong>4</strong>, e4724 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">643. </div><div class="csl-right-inline">Hammond, J., Calef, C., Larder, B., Schinazi, R. &amp; Mellors, J. W. Mutations in <span>Retroviral Genes Associated</span> with <span>Drug Resistance</span>. <em>Human retroviruses and AIDS</em> 11136–11179 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">644. </div><div class="csl-right-inline">Wensing, A. M. <em>et al.</em> <a href="https://www.ncbi.nlm.nih.gov/pubmed/28208121">2017 <span>Update</span> of the <span>Drug Resistance Mutations</span> in <span>HIV</span>-1., 2017 <span>Update</span> of the <span>Drug Resistance Mutations</span> in <span>HIV</span>-1</a>. <em>Topics in antiviral medicine, Topics in Antiviral Medicine</em> <strong>24, 24</strong>, 132, 132–133 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">645. </div><div class="csl-right-inline">Dudoit, S. &amp; Laan, M. J. van der. <em>Multiple <span>Testing Procedures</span> with <span>Applications</span> to <span>Genomics</span></em>. (<span>Springer Science &amp; Business Media</span>, 2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">646. </div><div class="csl-right-inline">Maddison, W. P. &amp; FitzJohn, R. G. <a href="https://doi.org/10.1093/sysbio/syu070">The <span>Unsolved Challenge</span> to <span>Phylogenetic Correlation Tests</span> for <span>Categorical Characters</span></a>. <em>Systematic Biology</em> <strong>64</strong>, 127–136 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">647. </div><div class="csl-right-inline">Lengauer, T. &amp; Sing, T. <a href="https://doi.org/10.1038/nrmicro1477">Bioinformatics-assisted anti-<span>HIV</span> therapy</a>. <em>Nature Reviews Microbiology</em> <strong>4</strong>, 790–797 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">648. </div><div class="csl-right-inline">Zhang, J., Rhee, S.-Y., Taylor, J. &amp; Shafer, R. W. <a href="https://doi.org/10.1097/01.qai.0000147526.64863.53">Comparison of the <span>Precision</span> and <span>Sensitivity</span> of the <span>Antivirogram</span> and <span>PhenoSense HIV Drug Susceptibility Assays</span></a>. <em>JAIDS Journal of Acquired Immune Deficiency Syndromes</em> <strong>38</strong>, 439–444 (2005).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">649. </div><div class="csl-right-inline">Beerenwinkel, N. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkg575">Geno2pheno: Estimating phenotypic drug resistance from <span>HIV</span>-1 genotypes</a>. <em>Nucleic Acids Research</em> <strong>31</strong>, 3850–3855 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">650. </div><div class="csl-right-inline">Shen, C., Yu, X., Harrison, R. W. &amp; Weber, I. T. <a href="https://doi.org/10.1186/s12859-016-1114-6">Automated prediction of <span>HIV</span> drug resistance from genotype data</a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 278 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">651. </div><div class="csl-right-inline">Yu, X., Weber, I. T. &amp; Harrison, R. W. <a href="https://doi.org/10.1186/1471-2164-15-S5-S1">Prediction of <span>HIV</span> drug resistance from genotype with encoded three-dimensional protein structure</a>. <em>BMC Genomics</em> <strong>15</strong>, S1 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">652. </div><div class="csl-right-inline">Araya, S. T. &amp; Hazelhurst, S. <a href="https://doi.org/10.1080/00359190909519238">Support vector machine prediction of <span>HIV</span>-1 drug resistance using the viral nucleotide patterns</a>. <em>Transactions of the Royal Society of South Africa</em> <strong>64</strong>, 62–72 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">653. </div><div class="csl-right-inline">Riemenschneider, M., Senge, R., Neumann, U., Hüllermeier, E. &amp; Heider, D. <a href="https://doi.org/10.1186/s13040-016-0089-1">Exploiting <span>HIV</span>-1 protease and reverse transcriptase cross-resistance information for improved drug resistance prediction by means of multi-label classification</a>. <em>BioData Mining</em> <strong>9</strong>, 10 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">654. </div><div class="csl-right-inline">Drăghici, S. &amp; Potter, R. B. <a href="https://doi.org/10.1093/bioinformatics/19.1.98">Predicting <span>HIV</span> drug resistance with neural networks</a>. <em>Bioinformatics</em> <strong>19</strong>, 98–107 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">655. </div><div class="csl-right-inline">Mooney, A. C. <em>et al.</em> <a href="https://doi.org/10.1007/s10461-018-2155-9">Beyond <span>Social Desirability Bias</span>: <span>Investigating Inconsistencies</span> in <span>Self</span>-<span>Reported HIV Testing</span> and <span>Treatment Behaviors Among HIV</span>-<span>Positive Adults</span> in <span>North West Province</span>, <span>South Africa</span></a>. <em>AIDS and Behavior</em> <strong>22</strong>, 2368–2379 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">656. </div><div class="csl-right-inline">Brier, G. W. Verification of <span>Forecasts</span> <span>Expressed</span> in <span>Terms</span> of <span>Probability</span>. <em>Monthly Weather Review</em> <strong>78</strong>, 1–3 (1950).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">657. </div><div class="csl-right-inline">Gascuel, O. <em>et al.</em> <a href="https://doi.org/10.1142/S0218001498000336">Twelve <span>Numerical</span>, <span>Symbolic</span> and <span>Hybrid Supervised Classification Methods</span></a>. <em>International Journal of Pattern Recognition and Artificial Intelligence</em> <strong>12</strong>, 517–571 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">658. </div><div class="csl-right-inline">Goeman, J. J. &amp; Solari, A. <a href="https://doi.org/10.1002/sim.6082">Multiple hypothesis testing in genomics</a>. <em>Statistics in Medicine</em> <strong>33</strong>, 1946–1978 (2014).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">659. </div><div class="csl-right-inline">Rennie, J. D., Shih, L., Teevan, J. &amp; Karger, D. R. Tackling the <span>Poor Assumptions</span> of <span>Naive Bayes Text Classifiers</span>. in <em>Proceedings of the 20th international conference on machine learning (<span>ICML</span>-03)</em> 616–623 (2003).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">660. </div><div class="csl-right-inline">Alvarez Melis, D. &amp; Jaakkola, T. Towards <span>Robust Interpretability</span> with <span>Self</span>-<span>Explaining Neural Networks</span>. in <em>Advances in <span>Neural Information Processing Systems</span> 31</em> (eds. Bengio, S. et al.) 7775–7784 (<span>Curran Associates, Inc.</span>, 2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">661. </div><div class="csl-right-inline">Zhang, Q., Wu, Y. N. &amp; Zhu, S.-C. Interpretable <span>Convolutional Neural Networks</span>. in <em>Proceedings of the <span>IEEE Conference</span> on <span>Computer Vision</span> and <span>Pattern Recognition</span></em> 8827–8836 (2018).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">662. </div><div class="csl-right-inline">Schrödinger, LLC. The <span>PyMOL</span> molecular graphics system, version 1.8. (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">663. </div><div class="csl-right-inline">Rhee, S.-Y., Liu, T. F., Holmes, S. P. &amp; Shafer, R. W. <a href="https://doi.org/10.1371/journal.pcbi.0030087"><span>HIV</span>-1 <span>Subtype B Protease</span> and <span>Reverse Transcriptase Amino Acid Covariation</span></a>. <em>PLOS Computational Biology</em> <strong>3</strong>, e87 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">664. </div><div class="csl-right-inline">De Luca, A. <em>et al.</em> <a href="https://doi.org/10.1086/522231">Improved <span>Interpretation</span> of <span>Genotypic Changes</span> in the <span>HIV</span>-1 <span>Reverse Transcriptase Coding Region That Determine</span> the <span>Virological Response</span> to <span>Didanosine</span></a>. <em>The Journal of Infectious Diseases</em> <strong>196</strong>, 1645–1653 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">665. </div><div class="csl-right-inline">Marcelin, A.-G. <em>et al.</em> Impact of <span>HIV</span>-1 reverse transcriptase polymorphism at codons 211 and 228 on virological response to didanosine. <em>Antiviral Therapy</em> 8 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">666. </div><div class="csl-right-inline">Brown, A. J. L. <em>et al.</em> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC110900">Reduced <span>Susceptibility</span> of <span>Human Immunodeficiency Virus Type</span> 1 (<span>HIV</span>-1) from <span>Patients</span> with <span>Primary HIV Infection</span> to <span>Nonnucleoside Reverse Transcriptase Inhibitors</span> <span>Is Associated</span> with <span>Variation</span> at <span>Novel Amino Acid Sites</span></a>. <em>Journal of Virology</em> <strong>74</strong>, 10269–10273 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">667. </div><div class="csl-right-inline">Clark, S. A., Shulman, N. S., Bosch, R. J. &amp; Mellors, J. W. <a href="https://doi.org/10.1097/01.aids.0000222069.14878.44">Reverse transcriptase mutations <span>118I</span>, <span>208Y</span>, and <span>215Y</span> cause <span>HIV</span>-1 hypersusceptibility to non-nucleoside reverse transcriptase inhibitors</a>. <em>AIDS</em> <strong>20</strong>, 981–984 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">668. </div><div class="csl-right-inline">Nebbia, G., Sabin, C. A., Dunn, D. T. &amp; Geretti, A. M. <a href="https://doi.org/10.1093/jac/dkm067">Emergence of the <span>H208Y</span> mutation in the reverse transcriptase (<span>RT</span>) of <span>HIV</span>-1 in association with nucleoside <span>RT</span> inhibitor therapy</a>. <em>Journal of Antimicrobial Chemotherapy</em> <strong>59</strong>, 1013–1016 (2007).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">669. </div><div class="csl-right-inline">Saracino, A. <em>et al.</em> <a href="https://doi.org/10.1002/jmv.20500">Impact of unreported <span>HIV</span>-1 reverse transcriptase mutations on phenotypic resistance to nucleoside and non-nucleoside inhibitors</a>. <em>Journal of Medical Virology</em> <strong>78</strong>, 9–17 (2006).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">670. </div><div class="csl-right-inline">Wu, T. T., Chen, Y. F., Hastie, T., Sobel, E. &amp; Lange, K. <a href="https://doi.org/10.1093/bioinformatics/btp041">Genome-wide association analysis by lasso penalized logistic regression</a>. <em>Bioinformatics</em> <strong>25</strong>, 714–721 (2009).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">671. </div><div class="csl-right-inline">Rosenblatt, F. <a href="https://doi.org/10.1037/h0042519">The perceptron: A probabilistic model for information storage and organization in the brain</a>. <em>Psychological Review</em> <strong>65</strong>, 386–408 (1958).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">672. </div><div class="csl-right-inline">Rumelhart, D. E., Hinton, G. E. &amp; Williams, R. J. <a href="https://doi.org/10.1038/323533a0">Learning representations by back-propagating errors</a>. <em>Nature</em> <strong>323</strong>, 533–536 (1986).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">673. </div><div class="csl-right-inline">Li, Y. &amp; Yuan, Y. <a href="https://proceedings.neurips.cc/paper/2017/hash/a96b65a721e561e1e3de768ac819ffbb-Abstract.html">Convergence analysis of two-layer neural networks with ReLU activation</a>. in vol. 30 (Curran Associates, Inc., 2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">674. </div><div class="csl-right-inline">Murtagh, F. <a href="https://doi.org/10.1016/0925-2312(91)90023-5">Multilayer perceptrons for classification and regression</a>. <em>Neurocomputing</em> <strong>2</strong>, 183–197 (1991).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">675. </div><div class="csl-right-inline">Cybenko, G. <a href="https://doi.org/10.1007/BF02551274">Approximation by superpositions of a sigmoidal function</a>. <em>Mathematics of Control, Signals and Systems</em> <strong>2</strong>, 303–314 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">676. </div><div class="csl-right-inline">Hornik, K., Stinchcombe, M. &amp; White, H. <a href="https://doi.org/10.1016/0893-6080(89)90020-8">Multilayer feedforward networks are universal approximators</a>. <em>Neural Networks</em> <strong>2</strong>, 359–366 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">677. </div><div class="csl-right-inline">Hornik, K. <a href="https://doi.org/10.1016/0893-6080(91)90009-T">Approximation capabilities of multilayer feedforward networks</a>. <em>Neural Networks</em> <strong>4</strong>, 251–257 (1991).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">678. </div><div class="csl-right-inline">LeCun, Y. <em>et al.</em> <a href="https://doi.org/10.1162/neco.1989.1.4.541">Backpropagation applied to handwritten zip code recognition</a>. <em>Neural Computation</em> <strong>1</strong>, 541–551 (1989).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">679. </div><div class="csl-right-inline">Lecun, Y., Bottou, L., Bengio, Y. &amp; Haffner, P. <a href="https://doi.org/10.1109/5.726791">Gradient-based learning applied to document recognition</a>. <em>Proceedings of the IEEE</em> <strong>86</strong>, 2278–2324 (1998).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">680. </div><div class="csl-right-inline">Voznica, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41467-022-31511-0">Deep learning from phylogenies to uncover the epidemiological dynamics of outbreaks</a>. <em>Nature Communications</em> <strong>13</strong>, 3896 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">681. </div><div class="csl-right-inline">Krizhevsky, A., Sutskever, I. &amp; Hinton, G. E. <a href="https://doi.org/10.1145/3065386">ImageNet classification with deep convolutional neural networks</a>. <em>Communications of the ACM</em> <strong>60</strong>, 8490 (2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">682. </div><div class="csl-right-inline">He, K., Zhang, X., Ren, S. &amp; Sun, J. <a href="https://openaccess.thecvf.com/content_cvpr_2016/html/He_Deep_Residual_Learning_CVPR_2016_paper.html">Proceedings of the IEEE conference on computer vision and pattern recognition</a>. in 770–778 (2016).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">683. </div><div class="csl-right-inline">Bahdanau, D., Cho, K. &amp; Bengio, Y. Neural machine translation by jointly learning to align and translate. doi:<a href="https://doi.org/10.48550/arXiv.1409.0473">10.48550/arXiv.1409.0473</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">684. </div><div class="csl-right-inline">Vaswani, A. <em>et al.</em> Attention is all you need. in vol. 30 (Curran Associates, Inc., 2017).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">685. </div><div class="csl-right-inline">Moreta, L. S. <em>et al.</em> International Conference on Learning Representations. in (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">686. </div><div class="csl-right-inline">Wu, Z., Johnston, K. E., Arnold, F. H. &amp; Yang, K. K. <a href="https://doi.org/10.1016/j.cbpa.2021.04.004">Protein sequence design with deep generative models</a>. <em>Current Opinion in Chemical Biology</em> <strong>65</strong>, 18–27 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">687. </div><div class="csl-right-inline">Stanton, S. <em>et al.</em> Accelerating bayesian optimization for biological sequence design with denoising autoencoders. doi:<a href="https://doi.org/10.48550/arXiv.2203.12742">10.48550/arXiv.2203.12742</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">688. </div><div class="csl-right-inline"><a href="https://www.merriam-webster.com/help/faq-how-many-english-words">How many words are there in english? | merriam-webster</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">689. </div><div class="csl-right-inline">Mikolov, T., Chen, K., Corrado, G. &amp; Dean, J. Efficient estimation of word representations in vector space. doi:<a href="https://doi.org/10.48550/arXiv.1301.3781">10.48550/arXiv.1301.3781</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">690. </div><div class="csl-right-inline">Mikolov, T., Sutskever, I., Chen, K., Corrado, G. S. &amp; Dean, J. Distributed representations of words and phrases and their compositionality. in vol. 26 (Curran Associates, Inc., 2013).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">691. </div><div class="csl-right-inline">Goldberg, Y. &amp; Levy, O. word2vec explained: Deriving mikolov et al.’s negative-sampling word-embedding method. doi:<a href="https://doi.org/10.48550/arXiv.1402.3722">10.48550/arXiv.1402.3722</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">692. </div><div class="csl-right-inline">Ng, P. dna2vec: Consistent vector representations of variable-length k-mers. doi:<a href="https://doi.org/10.48550/arXiv.1701.06279">10.48550/arXiv.1701.06279</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">693. </div><div class="csl-right-inline">Liang, Y. <em>et al.</em> <a href="https://doi.org/10.1186/s12859-022-04789-6">Hyb4mC: a hybrid DNA2vec-based model for DNA N4-methylcytosine sites prediction</a>. <em>BMC Bioinformatics</em> <strong>23</strong>, 258 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">694. </div><div class="csl-right-inline">Kimothi, D., Soni, A., Biyani, P. &amp; Hogan, J. M. Distributed representations for biological sequence analysis. doi:<a href="https://doi.org/10.48550/arXiv.1608.05949">10.48550/arXiv.1608.05949</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">695. </div><div class="csl-right-inline">Asgari, E. &amp; Mofrad, M. R. K. <a href="https://doi.org/10.1371/journal.pone.0141287">Continuous distributed representation of biological sequences for deep proteomics and genomics</a>. <em>PLoS ONE</em> <strong>10</strong>, e0141287 (2015).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">696. </div><div class="csl-right-inline">Kimothi, D., Shukla, A., Biyani, P., Anand, S. &amp; Hogan, J. M. 2017 IEEE 18th international workshop on signal processing advances in wireless communications (SPAWC). in 1–5 (2017). doi:<a href="https://doi.org/10.1109/SPAWC.2017.8227769">10.1109/SPAWC.2017.8227769</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">697. </div><div class="csl-right-inline">Devlin, J., Chang, M.-W., Lee, K. &amp; Toutanova, K. BERT: Pre-training of deep bidirectional transformers for language understanding. doi:<a href="https://doi.org/10.48550/arXiv.1810.04805">10.48550/arXiv.1810.04805</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">698. </div><div class="csl-right-inline">Brown, T. <em>et al.</em> Language models are few-shot learners. in vol. 33 18771901 (Curran Associates, Inc., 2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">699. </div><div class="csl-right-inline">Song, B. <em>et al.</em> <a href="https://doi.org/10.1093/bfgp/elab025">Pretraining model for biological sequence data.</a> <em>Briefings in Functional Genomics</em> <strong>20</strong>, 181–195 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">700. </div><div class="csl-right-inline">Madani, A. <em>et al.</em> ProGen: Language modeling for protein generation. <em>bioRxiv</em> (2020) doi:<a href="https://doi.org/10.1101/2020.03.07.982272">10.1101/2020.03.07.982272</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">701. </div><div class="csl-right-inline">Erik Nijkamp, Jeffrey A. Ruffolo, Eli N. Weinstein, Nikhil Naik &amp; Ali Madani. ProGen2: Exploring the boundaries of protein language models. <em>ArXiv</em> (2022) doi:<a href="https://doi.org/10.48550/arxiv.2206.13517">10.48550/arxiv.2206.13517</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">702. </div><div class="csl-right-inline">Ji, Y., Zhou, Z., Liu, H. &amp; Davuluri, R. V. <a href="https://doi.org/10.1093/bioinformatics/btab083">DNABERT: Pre-trained bidirectional encoder representations from transformers model for DNA-language in genome</a>. <em>Bioinformatics</em> <strong>37</strong>, 2112–2120 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">703. </div><div class="csl-right-inline">Bepler, T. &amp; Berger, B. <a href="https://doi.org/10.1016/j.cels.2021.05.017">Learning the protein language: Evolution, structure, and function.</a> <em>Cell systems</em> <strong>12</strong>, (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">704. </div><div class="csl-right-inline">Rao, R., Meier, J., Sercu, T., Ovchinnikov, S. &amp; Rives, A. Transformer protein language models are unsupervised structure learners. doi:<a href="https://doi.org/10.1101/2020.12.15.422761">10.1101/2020.12.15.422761</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">705. </div><div class="csl-right-inline">Rives, A. <em>et al.</em> Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences. <em>bioRxiv</em> 622803 (2019) doi:<a href="https://doi.org/10.1101/622803">10.1101/622803</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">706. </div><div class="csl-right-inline">Bhattacharya, N. <em>et al.</em> Single Layers of Attention Suffice to Predict Protein Contacts. doi:<a href="https://doi.org/10.1101/2020.12.21.423882">10.1101/2020.12.21.423882</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">707. </div><div class="csl-right-inline">Hu, M. <em>et al.</em> Exploring evolution-based &amp; -free protein language models as protein function predictors. doi:<a href="https://doi.org/10.48550/arXiv.2206.06583">10.48550/arXiv.2206.06583</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">708. </div><div class="csl-right-inline">Meier, J. <em>et al.</em> <a href="https://doi.org/10.1101/2021.07.09.450648">Language models enable zero-shot prediction of the effects of mutations on protein function</a>. <em>bioRxiv</em> <strong>34</strong>, (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">709. </div><div class="csl-right-inline">Hie, B., Kevin K Yang &amp; Kim, S. K. Evolutionary velocity with protein language models predicts evolutionary dynamics of diverse proteins. <em>Cell systems</em> (2022) doi:<a href="https://doi.org/10.1016/j.cels.2022.01.003">10.1016/j.cels.2022.01.003</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">710. </div><div class="csl-right-inline">Benegas, G., Batra, S. S. &amp; Song, Y. S. DNA language models are powerful zero-shot predictors of non-coding variant effects. doi:<a href="https://doi.org/10.1101/2022.08.22.504706">10.1101/2022.08.22.504706</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">711. </div><div class="csl-right-inline">Cai, T. <em>et al.</em> Genome-wide Prediction of Small Molecule Binding to Remote Orphan Proteins Using Distilled Sequence Alignment Embedding. doi:<a href="https://doi.org/10.1101/2020.08.04.236729">10.1101/2020.08.04.236729</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">712. </div><div class="csl-right-inline">Rao, R. <em>et al.</em> MSA transformer. <em>bioRxiv</em> (2021) doi:<a href="https://doi.org/10.1101/2021.02.12.430858">10.1101/2021.02.12.430858</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">713. </div><div class="csl-right-inline">Sturmfels, P., Vig, J., Madani, A. &amp; Rajani, N. F. Profile prediction: An alignment-based pre-training task for protein sequence models. doi:<a href="https://doi.org/10.48550/arXiv.2012.00195">10.48550/arXiv.2012.00195</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">714. </div><div class="csl-right-inline">Sercu, T. <em>et al.</em> Neural Potts Model. doi:<a href="https://doi.org/10.1101/2021.04.08.439084">10.1101/2021.04.08.439084</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">715. </div><div class="csl-right-inline">Baid, G. <em>et al.</em> DeepConsensus improves the accuracy of sequences with a gap-aware sequence transformer. <em>Nature Biotechnology</em> 1–7 (2022) doi:<a href="https://doi.org/10.1038/s41587-022-01435-7">10.1038/s41587-022-01435-7</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">716. </div><div class="csl-right-inline">Vig, J. <em>et al.</em> BERTology meets biology: Interpreting attention in protein language models. doi:<a href="https://doi.org/10.48550/arXiv.2006.15222">10.48550/arXiv.2006.15222</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">717. </div><div class="csl-right-inline">Gao, M. &amp; Skolnick, J. <a href="https://doi.org/10.1093/bioinformatics/btaa810">A novel sequence alignment algorithm based on deep learning of the protein folding code</a>. <em>Bioinformatics</em> <strong>37</strong>, 490–496 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">718. </div><div class="csl-right-inline">Morton, J. T. <em>et al.</em> Protein Structural Alignments From Sequence. doi:<a href="https://doi.org/10.1101/2020.11.03.365932">10.1101/2020.11.03.365932</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">719. </div><div class="csl-right-inline">Llinares-López, F., Berthet, Q., Blondel, M., Teboul, O. &amp; Vert, J.-P. Deep embedding and alignment of protein sequences. doi:<a href="https://doi.org/10.1101/2021.11.15.468653">10.1101/2021.11.15.468653</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">720. </div><div class="csl-right-inline">Petti, S. <em>et al.</em> End-to-end learning of multiple sequence alignments with differentiable Smith-Waterman. doi:<a href="https://doi.org/10.1101/2021.10.23.465204">10.1101/2021.10.23.465204</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">721. </div><div class="csl-right-inline">Guo, Y., Wu, J., Ma, H., Wang, S. &amp; Huang, J. <a href="https://doi.org/10.1089/cmb.2020.0416">Comprehensive study on enhancing low-quality position-specific scoring matrix with deep learning for accurate protein structure property prediction: Using bagging multiple sequence alignment learning</a>. <em>Journal of Computational Biology</em> <strong>28</strong>, 346–361 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">722. </div><div class="csl-right-inline">Dotan, E. <em>et al.</em> Harnessing machine translation methods for sequence alignment. doi:<a href="https://doi.org/10.1101/2022.07.22.501063">10.1101/2022.07.22.501063</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">723. </div><div class="csl-right-inline">Choromanski, K. <em>et al.</em> Masked language modeling for proteins via linearly scalable long-context transformers. doi:<a href="https://doi.org/10.48550/arXiv.2006.03555">10.48550/arXiv.2006.03555</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">724. </div><div class="csl-right-inline">Bhattacharya, N. <em>et al.</em> Interpreting potts and transformer protein models through the lens of simplified attention. in 34–45 (WORLD SCIENTIFIC, 2021). doi:<a href="https://doi.org/10.1142/9789811250477_0004">10.1142/9789811250477_0004</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">725. </div><div class="csl-right-inline">Sukhbaatar, S., Grave, E., Bojanowski, P. &amp; Joulin, A. Adaptive attention span in transformers. doi:<a href="https://doi.org/10.48550/arXiv.1905.07799">10.48550/arXiv.1905.07799</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">726. </div><div class="csl-right-inline">Wu, Z., Liu, Z., Lin, J., Lin, Y. &amp; Han, S. Lite transformer with long-short range attention. doi:<a href="https://doi.org/10.48550/arXiv.2004.11886">10.48550/arXiv.2004.11886</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">727. </div><div class="csl-right-inline">Child, R., Gray, S., Radford, A. &amp; Sutskever, I. Generating long sequences with sparse transformers. doi:<a href="https://doi.org/10.48550/arXiv.1904.10509">10.48550/arXiv.1904.10509</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">728. </div><div class="csl-right-inline">Correia, G. M., Niculae, V. &amp; Martins, A. F. T. EMNLP-IJCNLP 2019. in 21742184 (Association for Computational Linguistics, 2019). doi:<a href="https://doi.org/10.18653/v1/D19-1223">10.18653/v1/D19-1223</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">729. </div><div class="csl-right-inline">Kitaev, N., Kaiser, Ł. &amp; Levskaya, A. Reformer: The efficient transformer. doi:<a href="https://doi.org/10.48550/arXiv.2001.04451">10.48550/arXiv.2001.04451</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">730. </div><div class="csl-right-inline">Kraska, T., Beutel, A., Chi, E. H., Dean, J. &amp; Polyzotis, N. The case for learned index structures. in 489504 (Association for Computing Machinery, 2018). doi:<a href="https://doi.org/10.1145/3183713.3196909">10.1145/3183713.3196909</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">731. </div><div class="csl-right-inline">Jung, Y. &amp; Han, D. <a href="https://doi.org/10.1093/bioinformatics/btac137">BWA-MEME: BWA-MEM emulated with a machine learning approach</a>. <em>Bioinformatics</em> <strong>38</strong>, 2404–2413 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">732. </div><div class="csl-right-inline">Kirsche, M., Das, A. &amp; Schatz, M. C. <a href="https://doi.org/10.1093/bioinformatics/btaa911">Sapling: Accelerating suffix array queries with learned data models</a>. <em>Bioinformatics</em> <strong>37</strong>, 744–749 (2021).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">733. </div><div class="csl-right-inline">Ho, D. <em>et al.</em> LISA: Learned Indexes for Sequence Analysis. doi:<a href="https://doi.org/10.1101/2020.12.22.423964">10.1101/2020.12.22.423964</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">734. </div><div class="csl-right-inline">Hoang, M., Zheng, H. &amp; Kingsford, C. Differentiable learning of sequence-specific minimizer schemes with DeepMinimizer. <em>Journal of Computational Biology</em> (2022) doi:<a href="https://doi.org/10.1089/cmb.2022.0275">10.1089/cmb.2022.0275</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">735. </div><div class="csl-right-inline">Min, S., Lee, B. &amp; Yoon, S. <a href="https://doi.org/10.1093/bioinformatics/btab733">TargetNet: Functional microRNA target prediction with deep neural networks</a>. <em>Bioinformatics</em> <strong>38</strong>, 671–677 (2022).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">736. </div><div class="csl-right-inline">Bzikadze, A. V. &amp; Pevzner, P. A. <a href="https://doi.org/10.1038/s41587-020-0582-4">Automated assembly of centromeres from ultra-long error-prone reads</a>. <em>Nat Biotechnol</em> <strong>38</strong>, 1309–1316 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">737. </div><div class="csl-right-inline">Castresana, J. <a href="https://doi.org/10.1093/oxfordjournals.molbev.a026334">Selection of <span>Conserved Blocks</span> from <span>Multiple Alignments</span> for <span>Their Use</span> in <span>Phylogenetic Analysis</span></a>. <em>Molecular Biology and Evolution</em> <strong>17</strong>, 540–552 (2000).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">738. </div><div class="csl-right-inline">Pedregosa, F. <em>et al.</em> Scikit-learn: <span>Machine Learning</span> in <span>Python</span>. <em>Journal of Machine Learning Research</em> <strong>12</strong>, 2825–2830 (2011).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">739. </div><div class="csl-right-inline">Virtanen, P. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-019-0686-2"><span>SciPy</span> 1.0: Fundamental algorithms for scientific computing in <span>Python</span></a>. <em>Nature Methods</em> <strong>17</strong>, 261–272 (2020).</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">740. </div><div class="csl-right-inline">Seabold, Skipper &amp; Perktold, Josef. <span>S</span>tatsmodels: <span>E</span>conometric and <span>S</span>tatistical <span>M</span>odeling with <span>P</span>ython. in <em><span>P</span>roceedings of the 9th <span>P</span>ython in <span>S</span>cience <span>C</span>onference</em> (eds. Walt, Stéfan van der &amp; Millman, Jarrod) 92–96 (2010). doi:<a href="https://doi.org/ 10.25080/Majora-92bf1922-011 ">10.25080/Majora-92bf1922-011 </a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">741. </div><div class="csl-right-inline">Vinh, N. X. &amp; Epps, J. A <span>Novel Approach</span> for <span>Automatic Number</span> of <span>Clusters Detection</span> in <span>Microarray Data Based</span> on <span>Consensus Clustering</span>. in <em>2009 <span>Ninth IEEE International Conference</span> on <span>Bioinformatics</span> and <span>BioEngineering</span></em> 84–91 (2009). doi:<a href="https://doi.org/10.1109/BIBE.2009.19">10.1109/BIBE.2009.19</a>.</div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">742. </div><div class="csl-right-inline">Harremoes, P. Mutual information of contingency tables and related inequalities. in <em>2014 <span>IEEE International Symposium</span> on <span>Information Theory</span></em> 2474–2478 (<span>IEEE</span>, 2014). doi:<a href="https://doi.org/10.1109/ISIT.2014.6875279">10.1109/ISIT.2014.6875279</a>.</div>
</div>
</div>
</div>



</div>
            </section>

          </div>
        </div>
      </div>
<a href="global-conclusion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="HPC-appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/93-HPC-paper-Appendix.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
