<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression” | From sequences to knowledge, improving and learning from sequence alignments.</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.24.11 and GitBook 2.6.7" />

  <meta property="og:title" content="A Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression” | From sequences to knowledge, improving and learning from sequence alignments." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression” | From sequences to knowledge, improving and learning from sequence alignments." />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-06-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="global-references.html"/>
<link rel="next" href="HIV-appendix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">My PhD thesis.</a></li>

<li class="divider"></li>
<li><a href="index.html#abstract">Abstract<span></span></a></li>
<li><a href="résumé.html#résumé">Résumé<span></span></a></li>
<li><a href="acknowledgments.html#acknowledgments">Acknowledgments<span></span></a></li>
<li><a href="glossary.html#glossary">Glossary<span></span></a></li>
<li><a href="general-introduction.html#general-introduction">General Introduction<span></span></a></li>
<li class="chapter" data-level="1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>1</b> What is Sequence data ?<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>1.1</b> Biological sequences, a primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>1.1.1</b> What is DNA ?<span></span></a></li>
<li class="chapter" data-level="1.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#from-information-to-action"><i class="fa fa-check"></i><b>1.1.2</b> From Information to action<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining sequence data<span></span></a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough-wip"><i class="fa fa-check"></i><b>1.2.1</b> Sanger sequencing, a breakthrough (WIP)<span></span></a></li>
<li class="chapter" data-level="1.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> Next-generation sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>1.3</b> Sequencing errors, how to account for them ?<span></span></a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#long-read-errors"><i class="fa fa-check"></i><b>1.3.1</b> Long read errors<span></span></a></li>
<li class="chapter" data-level="1.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#hpc"><i class="fa fa-check"></i><b>1.3.2</b> HPC<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-computational-view-here"><i class="fa fa-check"></i><b>1.4</b> Biological sequences, a computational view (here ?)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>2</b> Aligning sequence data<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>2.1</b> What is an alignment ?<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-align"><i class="fa fa-check"></i><b>2.1.1</b> How to align ?<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#substitution-models"><i class="fa fa-check"></i><b>2.1.2</b> Substitution models<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#why-align"><i class="fa fa-check"></i><b>2.1.3</b> Why align ?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-do-we-speed-up-local-alignment"><i class="fa fa-check"></i><b>2.2</b> How do we speed up local alignment ?<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#msa-short"><i class="fa fa-check"></i><b>2.3</b> MSA (short)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><i class="fa fa-check"></i><b>3</b> Mapping-friendly sequence reductions: going beyond homopolymer compression<span></span></a>
<ul>
<li><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-abstract">Abstract<span></span></a></li>
<li class="chapter" data-level="3.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-methods"><i class="fa fa-check"></i><b>3.2</b> Methods<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:msr-def"><i class="fa fa-check"></i><b>3.2.1</b> Streaming sequence reductions<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:enum"><i class="fa fa-check"></i><b>3.2.2</b> Restricting the space of Streaming sequence reductions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#datasets-and-pipelines"><i class="fa fa-check"></i><b>3.3</b> Datasets and Pipelines<span></span></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#datasets"><i class="fa fa-check"></i><b>3.3.1</b> Datasets<span></span></a></li>
<li class="chapter" data-level="3.3.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#simulation-pipeline"><i class="fa fa-check"></i><b>3.3.2</b> Simulation pipeline<span></span></a></li>
<li class="chapter" data-level="3.3.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:evalpipeline"><i class="fa fa-check"></i><b>3.3.3</b> Evaluation pipeline<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-results"><i class="fa fa-check"></i><b>3.4</b> Results<span></span></a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>3.4.1</b> Selection of mapping-friendly sequence reductions<span></span></a></li>
<li class="chapter" data-level="3.4.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>3.4.2</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes<span></span></a></li>
<li class="chapter" data-level="3.4.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>3.4.3</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome<span></span></a></li>
<li class="chapter" data-level="3.4.4" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>3.4.4</b> Raw mapping improves upon HPC on centromeric regions<span></span></a></li>
<li class="chapter" data-level="3.4.5" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>3.4.5</b> Positions of incorrectly mapped reads across the entire human genome<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-discussion"><i class="fa fa-check"></i><b>3.5</b> Discussion<span></span></a></li>
<li class="chapter" data-level="3.6" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#limitations-of-this-study"><i class="fa fa-check"></i><b>3.6</b> Limitations of this study<span></span></a></li>
<li class="chapter" data-level="3.7" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#code-availability"><i class="fa fa-check"></i><b>3.7</b> Code availability<span></span></a></li>
<li><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#supplementary-information">Supplementary information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html"><i class="fa fa-check"></i><b>4</b> Learning from alignments<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#alignments-are-a-rich-source-of-information"><i class="fa fa-check"></i><b>4.1</b> Alignments are a rich source of information<span></span></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#pairwise-alns"><i class="fa fa-check"></i><b>4.1.1</b> Pairwise alns<span></span></a></li>
<li class="chapter" data-level="4.1.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#msa"><i class="fa fa-check"></i><b>4.1.2</b> MSA<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#preprocessing-the-alignment-for-machine-learning"><i class="fa fa-check"></i><b>4.2</b> Preprocessing the alignment for machine learning<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#embedding-the-alignment"><i class="fa fa-check"></i><b>4.2.1</b> Embedding the alignment<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#choosing-a-learning-target"><i class="fa fa-check"></i><b>4.2.2</b> Choosing a learning target<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#how-to-learn-from-alns"><i class="fa fa-check"></i><b>4.3</b> How to learn from ALNs<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>4.3.1</b> Tests and statistical learning<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#taking-interactions-into-account"><i class="fa fa-check"></i><b>4.3.2</b> Taking interactions into account<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#deep-learning"><i class="fa fa-check"></i><b>4.3.3</b> Deep Learning<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html"><i class="fa fa-check"></i><b>5</b> HIV and DRMs<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-are-viruses"><i class="fa fa-check"></i><b>5.1</b> What are viruses ?<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-is-hiv"><i class="fa fa-check"></i><b>5.2</b> What is HIV ?<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#quick-presentation-of-hiv"><i class="fa fa-check"></i><b>5.2.1</b> Quick Presentation of HIV<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#replication-cycle-of-hiv"><i class="fa fa-check"></i><b>5.2.2</b> Replication cycle of HIV<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#drug-resistance-in-hiv"><i class="fa fa-check"></i><b>5.3</b> Drug resistance in HIV<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#how-does-art-work"><i class="fa fa-check"></i><b>5.3.1</b> How does ART work<span></span></a></li>
<li class="chapter" data-level="5.3.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#different-types-of-resistance"><i class="fa fa-check"></i><b>5.3.2</b> different types of resistance<span></span></a></li>
<li class="chapter" data-level="5.3.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#consequences-on-global-health"><i class="fa fa-check"></i><b>5.3.3</b> Consequences on global health<span></span></a></li>
<li class="chapter" data-level="5.3.4" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#finding-drms"><i class="fa fa-check"></i><b>5.3.4</b> Finding DRMS<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>6</b> Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV<span></span></a>
<ul>
<li><a href="HIV-paper.html#abstract-paper">Abstract<span></span></a></li>
<li><a href="HIV-paper.html#author-summary">Author summary<span></span></a></li>
<li class="chapter" data-level="6.1" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>6.2</b> Materials and methods<span></span></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>6.2.1</b> Data<span></span></a></li>
<li class="chapter" data-level="6.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>6.2.2</b> Classifier training<span></span></a></li>
<li class="chapter" data-level="6.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>6.2.3</b> Measuring classifier performance<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-results"><i class="fa fa-check"></i><b>6.3</b> Results<span></span></a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>6.3.1</b> Classifier performance &amp; interpretation<span></span></a></li>
<li class="chapter" data-level="6.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>6.3.2</b> Additional classification results<span></span></a></li>
<li class="chapter" data-level="6.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>6.3.3</b> Identifying new mutations from classifiers<span></span></a></li>
<li class="chapter" data-level="6.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>6.3.4</b> Detailed analysis of potentially resistance-associated mutations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>6.4</b> Discussion and perspectives<span></span></a></li>
<li><a href="HIV-paper.html#hiv-acknowledgments">Acknowledgments<span></span></a></li>
<li><a href="HIV-paper.html#supporting-information">Supporting Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>7</b> Learning alignments, an interesting perspective<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-pairwise-alignment"><i class="fa fa-check"></i><b>7.1</b> Learning pairwise alignment<span></span></a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#dedal"><i class="fa fa-check"></i><b>7.1.1</b> DEDAL<span></span></a></li>
<li class="chapter" data-level="7.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-an-alignment"><i class="fa fa-check"></i><b>7.1.2</b> predicting an alignment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#what-else-could-we-learn"><i class="fa fa-check"></i><b>7.2</b> What else could we learn ?<span></span></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-to-predict-seeds-or-starting-positions"><i class="fa fa-check"></i><b>7.2.1</b> Learn to predict seeds or starting positions<span></span></a></li>
<li class="chapter" data-level="7.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-pre-processing-functions"><i class="fa fa-check"></i><b>7.2.2</b> Learn pre-processing functions<span></span></a></li>
</ul></li>
</ul></li>
<li><a href="global-references.html#global-references">Global References<span></span></a></li>
<li class="appendix"><span><b>Appendix<span></span></b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”<span></span></a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation<span></span></a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#msr-performance-comparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison<span></span></a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#origin-of-incorrectly-mapped-reads-of-high-mapping-quality-on-whole-human-genome."><i class="fa fa-check"></i><b>A.3</b> Origin of incorrectly mapped reads of high mapping quality on whole human genome.<span></span></a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome"><i class="fa fa-check"></i><b>A.4</b> Analyzing read origin on whole human genome<span></span></a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.5</b> Performance of MSRs on the Drosophila genome<span></span></a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”<span></span></a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>B.1</b> S1 Appendix (Technical appendix).<span></span></a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>B.1.1</b> Data<span></span></a></li>
<li class="chapter" data-level="B.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>B.1.2</b> Classifiers<span></span></a></li>
<li class="chapter" data-level="B.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>B.1.3</b> Scoring<span></span></a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>B.2</b> S1 Fig.<span></span></a></li>
<li class="chapter" data-level="B.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>B.3</b> S2 Fig.<span></span></a></li>
<li class="chapter" data-level="B.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>B.4</b> S3 Fig.<span></span></a></li>
<li class="chapter" data-level="B.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>B.5</b> S1 Table.<span></span></a></li>
<li class="chapter" data-level="B.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>B.6</b> S2 Appendix. (Fisher exact tests)<span></span></a></li>
<li class="chapter" data-level="B.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>B.7</b> S1 Data.<span></span></a></li>
<li class="chapter" data-level="B.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>B.8</b> S2 Data.<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequences to knowledge, improving and learning from sequence alignments.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="HPC-appendix" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">A</span> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”<a href="HPC-appendix.html#HPC-appendix" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="appendix:tandemtools" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">A.1</span> “TandemTools” dataset generation<a href="HPC-appendix.html#appendix:tandemtools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This dataset was obtained by taking a human X chromosome HOR sequence,
concatenating it 500 times with added mutations in order to obtain an
approximately 1 Mbp long sequence. Then 1200 reads were simulated from
the sequence using <code>nanosim</code> <span class="citation">[<a href="#ref-yangNanoSimNanoporeSequence2017" role="doc-biblioref">42</a>]</span> and
assembled using a centromere-tailored pipeline
<span class="citation">[<a href="#ref-bzikadzeAutomatedAssemblyCentromeres2020" role="doc-biblioref">94</a>]</span>. A 10kbp deletion was then
added to this assembly. The resulting sequence is the one we refer to as
the “Centromeric sequence”.</p>
</div>
<div id="msr-performance-comparison" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">A.2</span> MSR performance comparison<a href="HPC-appendix.html#msr-performance-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<template id="90ccb91c-4dba-4b60-b996-bc83b4214842"><style>
.tabwid table{
  border-spacing:0px !important;
  border-collapse:collapse;
  line-height:1;
  margin-left:auto;
  margin-right:auto;
  border-width: 0;
  display: table;
  margin-top: 1.275em;
  margin-bottom: 1.275em;
  border-color: transparent;
}
.tabwid_left table{
  margin-left:0;
}
.tabwid_right table{
  margin-right:0;
}
.tabwid td {
    padding: 0;
}
.tabwid a {
  text-decoration: none;
}
.tabwid thead {
    background-color: transparent;
}
.tabwid tfoot {
    background-color: transparent;
}
.tabwid table tr {
background-color: transparent;
}
</style><div class="tabwid"><style>.cl-baa56c78{table-layout:auto;width:100%;}.cl-ba9a3100{font-family:'Helvetica';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-ba9a3114{font-family:'Helvetica';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-ba9a311e{font-family:'Helvetica';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;top:3.3pt;}.cl-ba9a4c44{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-ba9b6048{background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ba9b605c{background-color:transparent;vertical-align: middle;border-bottom: 2pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ba9b605d{background-color:transparent;vertical-align: middle;border-bottom: 2pt solid rgba(102, 102, 102, 1.00);border-top: 2pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ba9b6066{background-color:transparent;vertical-align: middle;border-bottom: 1pt solid rgba(102, 102, 102, 1.00);border-top: 2pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ba9b6067{background-color:transparent;vertical-align: middle;border-bottom: 2pt solid rgba(102, 102, 102, 1.00);border-top: 1pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table class='cl-baa56c78'><thead><tr style="overflow-wrap:break-word;"><td  rowspan="2"class="cl-ba9b605d"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">label</span></p></td><td  colspan="2"class="cl-ba9b6066"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">mapq = 60</span></p></td><td  colspan="2"class="cl-ba9b6066"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">mapq ≥ 50</span></p></td><td  colspan="2"class="cl-ba9b6066"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">any mapq</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6067"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">fraction</span></p></td><td class="cl-ba9b6067"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">error</span></p></td><td class="cl-ba9b6067"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">fraction</span></p></td><td class="cl-ba9b6067"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">error</span></p></td><td class="cl-ba9b6067"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">fraction</span></p></td><td class="cl-ba9b6067"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">error</span></p></td></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Whole</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">Drosophila</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">melanogaster</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">genome</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">minimap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=25764)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.957</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.27e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.963</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.34e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.998</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.48e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.958</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.27e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.962</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.34e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.997</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.17e-02</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-21%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.952</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.18e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-48%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.960</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.37e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-41%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.998</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.36e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-8%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.946</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-100%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.954</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-100%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.998</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.53e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+3%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.950</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">4.90e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-78%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.957</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">8.11e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-65%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.998</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.39e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-6%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Whole</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">Drosophila</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">melanogaster</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">genome</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">winnowmap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=25764)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.923</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.51e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.930</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.59e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.989</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.50e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.949</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+3%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.92e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+27%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.954</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+3%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.99e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+26%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.995</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.33e-02</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-12%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.918</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.27e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-16%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.925</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.30e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-18%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.987</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.37e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-9%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.905</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.33e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-12%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.912</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.53e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-3%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.983</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.40e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-7%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.905</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.42e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-6%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.912</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.49e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-6%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.983</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.44e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-4%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Synthetic</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">centromeric</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">sequence</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">minimap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=12673)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.870</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.36e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.964</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.56e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.000</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">9.00e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.936</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+8%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.86e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+36%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.984</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.09e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+34%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.000</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">4.50e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-50%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.885</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">3.39e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+149%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.962</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">3.53e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+127%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.000</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.20e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+33%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.850</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.04e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+50%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.968</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.12e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+36%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.000</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">6.63e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-26%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.898</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+3%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.58e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+16%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.968</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.79e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+15%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.000</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">9.78e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+9%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Synthetic</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">centromeric</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">sequence</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">winnowmap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=12673)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.775</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.32e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.822</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.82e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.997</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">8.37e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.850</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+10%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.04e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+54%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.890</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+8%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.95e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+7%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.999</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">4.60e-02</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-45%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.795</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.28e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+73%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.846</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+3%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.52e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+38%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.997</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">6.96e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-17%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.820</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+6%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.83e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+38%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.867</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+6%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.27e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+25%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.997</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">5.97e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-29%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.780</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.62e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+22%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.829</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.09e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+15%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.997</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">8.65e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+3%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Whole</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">human</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">genome</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">minimap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=655594)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.935</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.85e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.942</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.85e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.000</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.46e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.921</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.86e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.927</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.86e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.998</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.29e-02</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-11%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.926</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">6.92e-05</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-96%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.936</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.17e-04</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-94%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.999</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.76e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+20%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.929</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.20e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-88%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.938</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">4.15e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-78%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.999</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.55e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+6%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.930</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.09e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-41%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.938</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.29e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-30%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.000</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.51e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+4%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Whole</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">human</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">genome</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">winnowmap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=655594)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.894</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.43e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.902</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.49e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.988</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.92e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.932</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+4%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.75e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+23%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.937</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+4%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.79e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+20%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.994</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.43e-02</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-26%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.874</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.81e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-80%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.886</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">3.82e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-74%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.984</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.94e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+1%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.795</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-11%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">6.33e-05</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-96%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.820</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-9%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">8.93e-05</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-94%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.971</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.08e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+9%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.826</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-8%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">8.68e-05</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-94%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.845</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-6%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.14e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-92%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.975</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.11e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+10%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Whole</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">Human</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">genome</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(repeated</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">regions)</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">minimap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=68811)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.619</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">3.29e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.656</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">3.10e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.998</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">7.79e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.514</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-17%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.98e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-40%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.539</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-18%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.16e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-30%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.981</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">6.69e-02</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-14%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.601</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-3%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.18e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-34%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.640</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">2.27e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-27%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.998</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">8.15e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+5%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.618</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.41e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-57%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.658</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.55e-04</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-50%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.997</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">8.23e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+6%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.616</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.18e-04</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-64%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.656</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.99e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-36%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.997</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">8.31e-02</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+7%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td  colspan="7"class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">Whole</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">Human</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">genome</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(repeated</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">regions)</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">winnowmap</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">(n=68811)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">HPC</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.525</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.24e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.557</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.49e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.950</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.19e-01</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">raw</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.648</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+23%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.26e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.672</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+21%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.49e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+0%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">0.968</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+2%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">8.09e-02</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-32%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">F</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.482</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-8%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">1.63e-03</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">+31%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.516</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-7%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.83e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+23%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.940</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-1%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.21e-01</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+2%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">E</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.366</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-30%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">6.35e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-49%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.405</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-27%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3114">9.32e-04</span><span class="cl-ba9a3114"> </span><span class="cl-ba9a3114">-37%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.911</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-4%</span></p></td><td class="cl-ba9b6048"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.38e-01</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+17%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ba9b605c"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">MSR</span><span class="cl-ba9a311e">P</span></p></td><td class="cl-ba9b605c"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.415</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-21%</span></p></td><td class="cl-ba9b605c"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">9.45e-04</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-24%</span></p></td><td class="cl-ba9b605c"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.451</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-19%</span></p></td><td class="cl-ba9b605c"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.16e-03</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-22%</span></p></td><td class="cl-ba9b605c"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">0.920</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">-3%</span></p></td><td class="cl-ba9b605c"><p class="cl-ba9a4c44"><span class="cl-ba9a3100">1.39e-01</span><span class="cl-ba9a3100"> </span><span class="cl-ba9a3100">+17%</span></p></td></tr></tbody>
<caption class="caption">
<p><span id="tab:mapperComparison">Table A.1: </span><strong>Comparing performance of MSRs on the whole human genome, whole <em>Drosophila melanogaster</em> genome, repeated regions of the whole human genome and synthetic centromeric sequence.</strong>
Results using <code>minimap2</code> <span class="citation">[<a href="#ref-liMinimap2PairwiseAlignment2018" role="doc-biblioref">30</a>]</span> and winnowmap2 <span class="citation">[<a href="#ref-jainWeightedMinimizerSampling2020" role="doc-biblioref">43</a>]</span>. The number of simulated reads for each reference sequence is given in parentheses and called <span class="math inline">\(n\)</span>. Results are reported for mapq thresholds of 60, 50 and 0. The best performance for each category is highlighted in bold. The percentage difference are computed w.r.t HPC at each given threshold.</p>
</caption>
</table></div></template>
<div class="flextable-shadow-host" id="dd93a6c8-c3c1-4c59-81c1-0da438699ad7"></div>
<script>
var dest = document.getElementById("dd93a6c8-c3c1-4c59-81c1-0da438699ad7");
var template = document.getElementById("90ccb91c-4dba-4b60-b996-bc83b4214842");
var caption = template.content.querySelector("caption");
if(caption) {
  caption.style.cssText = "display:block;text-align:center;";
  var newcapt = document.createElement("p");
  newcapt.appendChild(caption)
  dest.parentNode.insertBefore(newcapt, dest.nextSibling);
}
var fantome = dest.attachShadow({mode: 'open'});
var templateContent = template.content;
fantome.appendChild(templateContent);
</script>

</div>
<div id="origin-of-incorrectly-mapped-reads-of-high-mapping-quality-on-whole-human-genome." class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">A.3</span> Origin of incorrectly mapped reads of high mapping quality on whole human genome.<a href="HPC-appendix.html#origin-of-incorrectly-mapped-reads-of-high-mapping-quality-on-whole-human-genome." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:error-hists"></span>
<img src="figures/HPC-MSRs/ridgelines.new_selection.png" alt="**Histogram of the original simulated positions for the incorrectly
mapped reads using `minimap2` at high mapping qualities across the whole
human genome, for several transformation methods.**
For a given chromosome, each row represents the number of simulated
reads starting at that particular region. The dark gray rectangle
represents the position of the centromere for that chromosome, obtained
from annotations provided by the T2T consortium
(&lt;http://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.1/&gt;). Similarly for
chromosomes 13, 14, 15, 21 and 22, a lighter gray rectangle represents
the position of the &quot;stalk&quot; satellites also containing repetitive
regions. For HPC and raw reads only alignments of mapping quality 60
were considered. To provide a fair comparison, alignments with mapping
qualities $\geq 50$ were considered for MSRs E, F and P." width="90%" />
<p class="caption">
Figure A.1: <strong>Histogram of the original simulated positions for the incorrectly
mapped reads using <code>minimap2</code> at high mapping qualities across the whole
human genome, for several transformation methods.</strong>
For a given chromosome, each row represents the number of simulated
reads starting at that particular region. The dark gray rectangle
represents the position of the centromere for that chromosome, obtained
from annotations provided by the T2T consortium
(<a href="http://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.1/" class="uri">http://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.1/</a>). Similarly for
chromosomes 13, 14, 15, 21 and 22, a lighter gray rectangle represents
the position of the “stalk” satellites also containing repetitive
regions. For HPC and raw reads only alignments of mapping quality 60
were considered. To provide a fair comparison, alignments with mapping
qualities <span class="math inline">\(\geq 50\)</span> were considered for MSRs E, F and P.
</p>
</div>
</div>
<div id="analyzing-read-origin-on-whole-human-genome" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">A.4</span> Analyzing read origin on whole human genome<a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hist-raw"></span>
<img src="figures/HPC-MSRs/raw_distrib.centro.rstart.png" alt="**Origin of correctly (teal) and incorrectly (red) mapped raw reads.**
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
&quot;stalk&quot;, another repetitive region." width="90%" />
<p class="caption">
Figure A.2: <strong>Origin of correctly (teal) and incorrectly (red) mapped raw reads.</strong>
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
“stalk”, another repetitive region.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hist-hpc"></span>
<img src="figures/HPC-MSRs/hpc_distrib.centro.png" alt="**Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with HPC.**
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
&quot;stalk&quot;, another repetitive region." width="90%" />
<p class="caption">
Figure A.3: <strong>Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with HPC.</strong>
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
“stalk”, another repetitive region.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hist-msr-e"></span>
<img src="figures/HPC-MSRs/msr_e_distrib.centro.png" alt="**Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with MSR~E~.**
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
&quot;stalk&quot;, another repetitive region." width="90%" />
<p class="caption">
Figure A.4: <strong>Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with MSR<sub>E</sub>.</strong>
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
“stalk”, another repetitive region.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hist-msr-p"></span>
<img src="figures/HPC-MSRs/msr_p_distrib.centro.png" alt="**Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with MSR~P~.**
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
&quot;stalk&quot;, another repetitive region." width="90%" />
<p class="caption">
Figure A.5: <strong>Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with MSR<sub>P</sub>.</strong>
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
“stalk”, another repetitive region.
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hist-msr-f"></span>
<img src="figures/HPC-MSRs/msr_f2_distrib.centro.png" alt="**Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with MSR~F~.**
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
&quot;stalk&quot;, another repetitive region." width="90%" />
<p class="caption">
Figure A.6: <strong>Origin of correctly (teal) and incorrectly (red) mapped reads,
transformed with MSR<sub>F</sub>.</strong>
Distribution of the origin of correctly and incorrectly mapped simulated
reads (in teal and red respectively) on the different chromosomes of the
whole human genome. The dark grey rectangle for each chromosome
represents the centromere of that chromosome. The lighter gray rectangle
on chromosomes 13, 14, 15, 21 and 22 correspond to satellites denoted as
“stalk”, another repetitive region.
</p>
</div>
</div>
<div id="performance-of-msrs-on-the-drosophila-genome" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">A.5</span> Performance of MSRs on the Drosophila genome<a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:drosophila-results"></span>
<img src="figures/HPC-MSRs/droso_ecoli.png" alt="**Results of the `paftools mapeval` evaluation on reads simulated and
mapped to whole *Drosophila melanogaster* and *Escherichia coli* genomes.**
MSRs E, F and P are shown in different shades of blue to
differentiate them from other MSRs. Reads were simulated with `nanosim`,
and mapped with `minimap2`. The *E. coli* genome was obtained from Genbank ID [U00096.2](https://www.ncbi.nlm.nih.gov/nuccore/U00096.2)" width="90%" />
<p class="caption">
Figure A.7: <strong>Results of the <code>paftools mapeval</code> evaluation on reads simulated and
mapped to whole <em>Drosophila melanogaster</em> and <em>Escherichia coli</em> genomes.</strong>
MSRs E, F and P are shown in different shades of blue to
differentiate them from other MSRs. Reads were simulated with <code>nanosim</code>,
and mapped with <code>minimap2</code>. The <em>E. coli</em> genome was obtained from Genbank ID <a href="https://www.ncbi.nlm.nih.gov/nuccore/U00096.2">U00096.2</a>
</p>
</div>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body">
<div id="ref-liMinimap2PairwiseAlignment2018" class="csl-entry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Li H. Minimap2: Pairwise alignment for nucleotide sequences. Bioinformatics. 2018;34: 3094–3100. doi:<a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a></div>
</div>
<div id="ref-yangNanoSimNanoporeSequence2017" class="csl-entry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Yang C, Chu J, Warren RL, Birol I. <span>NanoSim</span>: Nanopore sequence read simulator based on statistical characterization. GigaScience. 2017;6. doi:<a href="https://doi.org/10.1093/gigascience/gix010">10.1093/gigascience/gix010</a></div>
</div>
<div id="ref-jainWeightedMinimizerSampling2020" class="csl-entry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Jain C, Rhie A, Zhang H, Chu C, Walenz BP, Koren S, et al. Weighted minimizer sampling improves long read mapping. Bioinformatics. 2020;36: i111–i118. doi:<a href="https://doi.org/10.1093/bioinformatics/btaa435">10.1093/bioinformatics/btaa435</a></div>
</div>
<div id="ref-bzikadzeAutomatedAssemblyCentromeres2020" class="csl-entry">
<div class="csl-left-margin">94. </div><div class="csl-right-inline">Bzikadze AV, Pevzner PA. Automated assembly of centromeres from ultra-long error-prone reads. Nat Biotechnol. 2020;38: 1309–1316. doi:<a href="https://doi.org/10.1038/s41587-020-0582-4">10.1038/s41587-020-0582-4</a></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="global-references.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="HIV-appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/91-HPC-Appendix.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
