<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Sequence analysis, combinatorics and machine learning</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.24.11 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Sequence analysis, combinatorics and machine learning" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Sequence analysis, combinatorics and machine learning" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-03-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="conclusion.html"/>
<link rel="next" href="HIV-appendix.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li><a href="index.html#abstract">Abstract<span></span></a></li>
<li><a href="resume.html#resume">Resume<span></span></a></li>
<li><a href="aknowledgments.html#aknowledgments">Aknowledgments<span></span></a></li>
<li><a href="list-of-figures.html#list-of-figures">List of Figures<span></span></a></li>
<li><a href="list-of-tables.html#list-of-tables">List of Tables<span></span></a></li>
<li><a href="glosssary.html#glosssary">Glosssary<span></span></a></li>
<li class="chapter" data-level="1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html"><i class="fa fa-check"></i><b>1</b> HIV and DRMs<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#hiv"><i class="fa fa-check"></i><b>1.1</b> HIV<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#drms"><i class="fa fa-check"></i><b>1.2</b> DRMs<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><i class="fa fa-check"></i><b>2</b> Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV<span></span></a>
<ul>
<li><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#abstract-paper">Abstract<span></span></a></li>
<li><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#author-summary">Author summary<span></span></a></li>
<li class="chapter" data-level="2.1" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#materials-and-methods"><i class="fa fa-check"></i><b>2.2</b> Materials and methods<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#data"><i class="fa fa-check"></i><b>2.2.1</b> Data<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#classifier-training"><i class="fa fa-check"></i><b>2.2.2</b> Classifier training<span></span></a></li>
<li class="chapter" data-level="2.2.3" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>2.2.3</b> Measuring classifier performance<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#results"><i class="fa fa-check"></i><b>2.3</b> Results<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>2.3.1</b> Classifier performance &amp; interpretation<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#additional-classification-results"><i class="fa fa-check"></i><b>2.3.2</b> Additional classification results<span></span></a></li>
<li class="chapter" data-level="2.3.3" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>2.3.3</b> Identifying new mutations from classifiers<span></span></a></li>
<li class="chapter" data-level="2.3.4" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>2.3.4</b> Detailed analysis of potentially resistance-associated mutations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>2.4</b> Discussion and perspectives<span></span></a></li>
<li><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#acknowledgments">Acknowledgments<span></span></a></li>
<li><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#supporting-information">Supporting Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hpc-some-context.html"><a href="hpc-some-context.html"><i class="fa fa-check"></i><b>3</b> HPC, some context<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="hpc-some-context.html"><a href="hpc-some-context.html#sequencing"><i class="fa fa-check"></i><b>3.1</b> Sequencing<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="hpc-some-context.html"><a href="hpc-some-context.html#downstream-analysis"><i class="fa fa-check"></i><b>3.2</b> downstream analysis<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="hpc-some-context.html"><a href="hpc-some-context.html#hpc"><i class="fa fa-check"></i><b>3.3</b> HPC<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="hpc-paper.html"><a href="hpc-paper.html"><i class="fa fa-check"></i><b>4</b> HPC paper!<span></span></a></li>
<li class="chapter" data-level="5" data-path="conclusion.html"><a href="conclusion.html"><i class="fa fa-check"></i><b>5</b> Conclusion<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="conclusion.html"><a href="conclusion.html#hiv-drm-conclusion"><i class="fa fa-check"></i><b>5.1</b> HIV DRM conclusion<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="conclusion.html"><a href="conclusion.html#hpc-conclusion"><i class="fa fa-check"></i><b>5.2</b> HPC conclusion<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="conclusion.html"><a href="conclusion.html#general-conclusion"><i class="fa fa-check"></i><b>5.3</b> general conclusion<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="appendix"><span><b>Appendix<span></span></b></span></li>
<li class="chapter" data-level="A" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”<span></span></a>
<ul>
<li class="chapter" data-level="A.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>A.1</b> S1 Appendix (Technical appendix).<span></span></a>
<ul>
<li class="chapter" data-level="A.1.1" data-path="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html"><a href="using-machine-learning-and-big-data-to-explore-the-drug-resistance-landscape-in-hiv.html#data"><i class="fa fa-check"></i><b>A.1.1</b> Data<span></span></a></li>
<li class="chapter" data-level="A.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>A.1.2</b> Classifiers<span></span></a></li>
<li class="chapter" data-level="A.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>A.1.3</b> Scoring<span></span></a></li>
</ul></li>
<li class="chapter" data-level="A.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>A.2</b> S1 Fig.<span></span></a></li>
<li class="chapter" data-level="A.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>A.3</b> S2 Fig.<span></span></a></li>
<li class="chapter" data-level="A.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>A.4</b> S3 Fig.<span></span></a></li>
<li class="chapter" data-level="A.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>A.5</b> S1 Table.<span></span></a></li>
<li class="chapter" data-level="A.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>A.6</b> S2 Appendix. (Fisher exact tests)<span></span></a></li>
<li class="chapter" data-level="A.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>A.7</b> S1 Data.<span></span></a></li>
<li class="chapter" data-level="A.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>A.8</b> S2 Data.<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Sequence analysis, combinatorics and machine learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered hasAnchor">
<h1>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="refs" class="references csl-bib-body">
<div class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Lepri AC, Sabin CA, Staszewski S, Hertogs K, Müller A, Rabenau H, et al. Resistance <span>Profiles</span> in <span>Patients</span> with <span>Viral Rebound</span> on <span>Potent Antiretroviral Therapy</span>. The Journal of Infectious Diseases. 2000;181: 1143–1147. doi:<a href="https://doi.org/10.1086/315301">10.1086/315301</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Verhofstede C, Wanzeele FV, Gucht BVD, Pelgrom J, Vandekerckhove L, Plum J, et al. Detection of drug resistance mutations as a predictor of subsequent virological failure in patients with <span>HIV</span>-1 viral rebounds of less than 1,000 <span>RNA</span> copies/ml. Journal of Medical Virology. 2007;79: 1254–1260. doi:<a href="https://doi.org/10.1002/jmv.20950">10.1002/jmv.20950</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Hué S, Gifford RJ, Dunn D, Fernhill E, Pillay D, Resistance on B of the UCG on HDR. Demonstration of <span>Sustained Drug</span>-<span>Resistant Human Immunodeficiency Virus Type</span> 1 <span>Lineages Circulating</span> among <span>Treatment</span>-<span>Naïve Individuals</span>. Journal of Virology. 2009;83: 2645–2654. doi:<a href="https://doi.org/10.1128/JVI.01556-08">10.1128/JVI.01556-08</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Mourad R, Chevennet F, Dunn D, Fearnhill E, Delpech V, Asboe D, et al. A phylotype-based analysis highlights the role of drug-naive <span>HIV</span>-positive individuals in the transmission of antiretroviral resistance in the <span>UK</span>. Aids. 2015;29: 1917–1925. doi:<a href="https://doi.org/10.1097/QAD.0000000000000768">10.1097/QAD.0000000000000768</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Zhukova A, Cutino-Moguel T, Gascuel O, Pillay D. The <span>Role</span> of <span>Phylogenetics</span> as a <span>Tool</span> to <span>Predict</span> the <span>Spread</span> of <span>Resistance</span>. The Journal of Infectious Diseases. 2017;216: S820–S823. doi:<a href="https://doi.org/10.1093/infdis/jix411">10.1093/infdis/jix411</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Bennett DE, Camacho RJ, Otelea D, Kuritzkes DR, Fleury H, Kiuchi M, et al. Drug <span>Resistance Mutations</span> for <span>Surveillance</span> of <span>Transmitted HIV</span>-1 <span>Drug</span>-<span>Resistance</span>: 2009 <span>Update</span>. PLOS ONE. 2009;4: e4724. doi:<a href="https://doi.org/10.1371/journal.pone.0004724">10.1371/journal.pone.0004724</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Hammond J, Calef C, Larder B, Schinazi R, Mellors JW. Mutations in <span>Retroviral Genes Associated</span> with <span>Drug Resistance</span>. Human retroviruses and AIDS. 1998; 11136–11179. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Wensing AM, Calvez V, Günthard HF, Johnson VA, Paredes R, Pillay D, et al. <a href="https://www.ncbi.nlm.nih.gov/pubmed/28208121">2017 <span>Update</span> of the <span>Drug Resistance Mutations</span> in <span>HIV</span>-1., 2017 <span>Update</span> of the <span>Drug Resistance Mutations</span> in <span>HIV</span>-1</a>. Topics in antiviral medicine, Topics in Antiviral Medicine. 2016;24, 24: 132, 132–133. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Dudoit S, Laan MJ van der. Multiple <span>Testing Procedures</span> with <span>Applications</span> to <span>Genomics</span>. <span>Springer Science &amp; Business Media</span>; 2007. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Villabona-Arenas CJ, Vidal N, Guichet E, Serrano L, Delaporte E, Gascuel O, et al. In-depth analysis of <span>HIV</span>-1 drug resistance mutations in <span>HIV</span>-infected individuals failing first-line regimens in <span>West</span> and <span>Central Africa</span>. AIDS. 2016;30: 2577. doi:<a href="https://doi.org/10.1097/QAD.0000000000001233">10.1097/QAD.0000000000001233</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Maddison WP, FitzJohn RG. The <span>Unsolved Challenge</span> to <span>Phylogenetic Correlation Tests</span> for <span>Categorical Characters</span>. Systematic Biology. 2015;64: 127–136. doi:<a href="https://doi.org/10.1093/sysbio/syu070">10.1093/sysbio/syu070</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Sham PC, Purcell SM. Statistical power and significance testing in large-scale genetic studies. Nature Reviews Genetics. 2014;15: 335–346. doi:<a href="https://doi.org/10.1038/nrg3706">10.1038/nrg3706</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">Lengauer T, Sing T. Bioinformatics-assisted anti-<span>HIV</span> therapy. Nature Reviews Microbiology. 2006;4: 790–797. doi:<a href="https://doi.org/10.1038/nrmicro1477">10.1038/nrmicro1477</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Zhang J, Rhee S-Y, Taylor J, Shafer RW. Comparison of the <span>Precision</span> and <span>Sensitivity</span> of the <span>Antivirogram</span> and <span>PhenoSense HIV Drug Susceptibility Assays</span>. JAIDS Journal of Acquired Immune Deficiency Syndromes. 2005;38: 439–444. doi:<a href="https://doi.org/10.1097/01.qai.0000147526.64863.53">10.1097/01.qai.0000147526.64863.53</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Beerenwinkel N, Däumer M, Oette M, Korn K, Hoffmann D, Kaiser R, et al. Geno2pheno: Estimating phenotypic drug resistance from <span>HIV</span>-1 genotypes. Nucleic Acids Research. 2003;31: 3850–3855. doi:<a href="https://doi.org/10.1093/nar/gkg575">10.1093/nar/gkg575</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Shen C, Yu X, Harrison RW, Weber IT. Automated prediction of <span>HIV</span> drug resistance from genotype data. BMC Bioinformatics. 2016;17: 278. doi:<a href="https://doi.org/10.1186/s12859-016-1114-6">10.1186/s12859-016-1114-6</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Yu X, Weber IT, Harrison RW. Prediction of <span>HIV</span> drug resistance from genotype with encoded three-dimensional protein structure. BMC Genomics. 2014;15: S1. doi:<a href="https://doi.org/10.1186/1471-2164-15-S5-S1">10.1186/1471-2164-15-S5-S1</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Sheik Amamuddy O, Bishop NT, Tastan Bishop Ö. Improving fold resistance prediction of <span>HIV</span>-1 against protease and reverse transcriptase inhibitors using artificial neural networks. BMC Bioinformatics. 2017;18: 369. doi:<a href="https://doi.org/10.1186/s12859-017-1782-x">10.1186/s12859-017-1782-x</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Beerenwinkel N, Lengauer T, Selbig J, Schmidt B, Walter H, Korn K, et al. Geno2pheno: Interpreting genotypic <span>HIV</span> drug resistance tests. IEEE Intelligent Systems. 2001;16: 35–41. doi:<a href="https://doi.org/10.1109/5254.972080">10.1109/5254.972080</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Araya ST, Hazelhurst S. Support vector machine prediction of <span>HIV</span>-1 drug resistance using the viral nucleotide patterns. Transactions of the Royal Society of South Africa. 2009;64: 62–72. doi:<a href="https://doi.org/10.1080/00359190909519238">10.1080/00359190909519238</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Riemenschneider M, Senge R, Neumann U, Hüllermeier E, Heider D. Exploiting <span>HIV</span>-1 protease and reverse transcriptase cross-resistance information for improved drug resistance prediction by means of multi-label classification. BioData Mining. 2016;9: 10. doi:<a href="https://doi.org/10.1186/s13040-016-0089-1">10.1186/s13040-016-0089-1</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Heider D, Senge R, Cheng W, Hüllermeier E. Multilabel classification for exploiting cross-resistance information in <span>HIV</span>-1 drug resistance prediction. Bioinformatics. 2013;29: 1946–1952. doi:<a href="https://doi.org/10.1093/bioinformatics/btt331">10.1093/bioinformatics/btt331</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Drăghici S, Potter RB. Predicting <span>HIV</span> drug resistance with neural networks. Bioinformatics. 2003;19: 98–107. doi:<a href="https://doi.org/10.1093/bioinformatics/19.1.98">10.1093/bioinformatics/19.1.98</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Steiner MC, Gibson KM, Crandall KA. Drug <span>Resistance Prediction Using Deep Learning Techniques</span> on <span>HIV</span>-1 <span>Sequence Data</span>. Viruses. 2020;12: 560. doi:<a href="https://doi.org/10.3390/v12050560">10.3390/v12050560</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Mooney AC, Campbell CK, Ratlhagana M-J, Grignon JS, Mazibuko S, Agnew E, et al. Beyond <span>Social Desirability Bias</span>: <span>Investigating Inconsistencies</span> in <span>Self</span>-<span>Reported HIV Testing</span> and <span>Treatment Behaviors Among HIV</span>-<span>Positive Adults</span> in <span>North West Province</span>, <span>South Africa</span>. AIDS and Behavior. 2018;22: 2368–2379. doi:<a href="https://doi.org/10.1007/s10461-018-2155-9">10.1007/s10461-018-2155-9</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Tibshirani R. Regression <span>Shrinkage</span> and <span>Selection Via</span> the <span>Lasso</span>. Journal of the Royal Statistical Society: Series B (Methodological). 1996;58: 267–288. doi:<a href="https://doi.org/10.1111/j.2517-6161.1996.tb02080.x">10.1111/j.2517-6161.1996.tb02080.x</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Brier GW. Verification of <span>Forecasts</span> <span>Expressed</span> in <span>Terms</span> of <span>Probability</span>. Monthly Weather Review. 1950;78: 1–3. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Gascuel O, Bouchon-Meunier B, Caraux G, Gallinari P, Guénoche A, Guermeur Y, et al. Twelve <span>Numerical</span>, <span>Symbolic</span> and <span>Hybrid Supervised Classification Methods</span>. International Journal of Pattern Recognition and Artificial Intelligence. 1998;12: 517–571. doi:<a href="https://doi.org/10.1142/S0218001498000336">10.1142/S0218001498000336</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Goeman JJ, Solari A. Multiple hypothesis testing in genomics. Statistics in Medicine. 2014;33: 1946–1978. doi:<a href="https://doi.org/10.1002/sim.6082">10.1002/sim.6082</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Rennie JD, Shih L, Teevan J, Karger DR. Tackling the <span>Poor Assumptions</span> of <span>Naive Bayes Text Classifiers</span>. Proceedings of the 20th international conference on machine learning (<span>ICML</span>-03). 2003. pp. 616–623. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Breiman L. Random <span>Forests</span>. Machine Learning. 2001;45: 5–32. doi:<a href="https://doi.org/10.1023/A:1010933404324">10.1023/A:1010933404324</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Alvarez Melis D, Jaakkola T. Towards <span>Robust Interpretability</span> with <span>Self</span>-<span>Explaining Neural Networks</span>. In: Bengio S, Wallach H, Larochelle H, Grauman K, Cesa-Bianchi N, Garnett R, editors. Advances in <span>Neural Information Processing Systems</span> 31. <span>Curran Associates, Inc.</span>; 2018. pp. 7775–7784. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Hastie T, Tibshirani R, Friedman J. The <span>Elements</span> of <span>Statistical Learning</span>: <span>Data Mining</span>, <span>Inference</span>, and <span>Prediction</span>, <span>Second Edition</span>. <span>Springer Science &amp; Business Media</span>; 2009. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Zhang Q, Wu YN, Zhu S-C. Interpretable <span>Convolutional Neural Networks</span>. Proceedings of the <span>IEEE Conference</span> on <span>Computer Vision</span> and <span>Pattern Recognition</span>. 2018. pp. 8827–8836. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline">Brodersen KH, Ong CS, Stephan KE, Buhmann JM. The <span>Balanced Accuracy</span> and <span>Its Posterior Distribution</span>. 2010 20th <span>International Conference</span> on <span>Pattern Recognition</span>. 2010. pp. 3121–3124. doi:<a href="https://doi.org/10.1109/ICPR.2010.764">10.1109/ICPR.2010.764</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Vinh NX, Epps J, Bailey J. Information <span>Theoretic Measures</span> for <span>Clusterings Comparison</span>: <span>Variants</span>, <span>Properties</span>, <span>Normalization</span> and <span>Correction</span> for <span>Chance</span>. Journal of Machine Learning Research. 2010;11: 18. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Schrödinger, LLC. The <span>PyMOL</span> molecular graphics system, version 1.8. 2015. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Sarafianos SG, Marchand B, Das K, Himmel D, Parniak MA, Hughes SH, et al. Structure and function of <span>HIV</span>-1 reverse transcriptase: Molecular mechanisms of polymerization and inhibition. Journal of molecular biology. 2009;385: 693–713. doi:<a href="https://doi.org/10.1016/j.jmb.2008.10.071">10.1016/j.jmb.2008.10.071</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Rhee S-Y, Liu TF, Holmes SP, Shafer RW. <span>HIV</span>-1 <span>Subtype B Protease</span> and <span>Reverse Transcriptase Amino Acid Covariation</span>. PLOS Computational Biology. 2007;3: e87. doi:<a href="https://doi.org/10.1371/journal.pcbi.0030087">10.1371/journal.pcbi.0030087</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline">De Luca A, Di Giambenedetto S, Trotta MP, Colafigli M, Prosperi M, Ruiz L, et al. Improved <span>Interpretation</span> of <span>Genotypic Changes</span> in the <span>HIV</span>-1 <span>Reverse Transcriptase Coding Region That Determine</span> the <span>Virological Response</span> to <span>Didanosine</span>. The Journal of Infectious Diseases. 2007;196: 1645–1653. doi:<a href="https://doi.org/10.1086/522231">10.1086/522231</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Marcelin A-G, Flandre P, Furco A, Wirden M, Molina J-M, Calvez V. Impact of <span>HIV</span>-1 reverse transcriptase polymorphism at codons 211 and 228 on virological response to didanosine. Antiviral Therapy. 2006; 8. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Brown AJL, Precious HM, Whitcomb JM, Wong JK, Quigg M, Huang W, et al. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC110900">Reduced <span>Susceptibility</span> of <span>Human Immunodeficiency Virus Type</span> 1 (<span>HIV</span>-1) from <span>Patients</span> with <span>Primary HIV Infection</span> to <span>Nonnucleoside Reverse Transcriptase Inhibitors</span> <span>Is Associated</span> with <span>Variation</span> at <span>Novel Amino Acid Sites</span></a>. Journal of Virology. 2000;74: 10269–10273. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Clark SA, Shulman NS, Bosch RJ, Mellors JW. Reverse transcriptase mutations <span>118I</span>, <span>208Y</span>, and <span>215Y</span> cause <span>HIV</span>-1 hypersusceptibility to non-nucleoside reverse transcriptase inhibitors. AIDS. 2006;20: 981–984. doi:<a href="https://doi.org/10.1097/01.aids.0000222069.14878.44">10.1097/01.aids.0000222069.14878.44</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Nebbia G, Sabin CA, Dunn DT, Geretti AM. Emergence of the <span>H208Y</span> mutation in the reverse transcriptase (<span>RT</span>) of <span>HIV</span>-1 in association with nucleoside <span>RT</span> inhibitor therapy. Journal of Antimicrobial Chemotherapy. 2007;59: 1013–1016. doi:<a href="https://doi.org/10.1093/jac/dkm067">10.1093/jac/dkm067</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Saracino A, Monno L, Scudeller L, Cibelli DC, Tartaglia A, Punzi G, et al. Impact of unreported <span>HIV</span>-1 reverse transcriptase mutations on phenotypic resistance to nucleoside and non-nucleoside inhibitors. Journal of Medical Virology. 2006;78: 9–17. doi:<a href="https://doi.org/10.1002/jmv.20500">10.1002/jmv.20500</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline">Wu TT, Chen YF, Hastie T, Sobel E, Lange K. Genome-wide association analysis by lasso penalized logistic regression. Bioinformatics. 2009;25: 714–721. doi:<a href="https://doi.org/10.1093/bioinformatics/btp041">10.1093/bioinformatics/btp041</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Castresana J. Selection of <span>Conserved Blocks</span> from <span>Multiple Alignments</span> for <span>Their Use</span> in <span>Phylogenetic Analysis</span>. Molecular Biology and Evolution. 2000;17: 540–552. doi:<a href="https://doi.org/10.1093/oxfordjournals.molbev.a026334">10.1093/oxfordjournals.molbev.a026334</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline">McGinnis W, Hbghhy, Tao W, Andrethrill, Siu C, Davison C, et al. Scikit-<span>Learn</span>-<span>Contrib</span>/<span>Categorical</span>-<span>Encoding</span>: <span>Release For Zenodo</span>. Zenodo; 2018. doi:<a href="https://doi.org/10.5281/ZENODO.1157110">10.5281/ZENODO.1157110</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">Pedregosa F, Varoquaux G, Gramfort A, Michel V, Thirion B, Grisel O, et al. Scikit-learn: <span>Machine Learning</span> in <span>Python</span>. Journal of Machine Learning Research. 2011;12: 2825–2830. </div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">Virtanen P, Gommers R, Oliphant TE, Haberland M, Reddy T, Cournapeau D, et al. <span>SciPy</span> 1.0: Fundamental algorithms for scientific computing in <span>Python</span>. Nature Methods. 2020;17: 261–272. doi:<a href="https://doi.org/10.1038/s41592-019-0686-2">10.1038/s41592-019-0686-2</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Seabold Skipper, Perktold Josef. <span>S</span>tatsmodels: <span>E</span>conometric and <span>S</span>tatistical <span>M</span>odeling with <span>P</span>ython. In: Walt Stéfan van der, Millman Jarrod, editors. <span>P</span>roceedings of the 9th <span>P</span>ython in <span>S</span>cience <span>C</span>onference. 2010. pp. 92–96. doi:<a href="https://doi.org/ 10.25080/Majora-92bf1922-011 "> 10.25080/Majora-92bf1922-011 </a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Vinh NX, Epps J. A <span>Novel Approach</span> for <span>Automatic Number</span> of <span>Clusters Detection</span> in <span>Microarray Data Based</span> on <span>Consensus Clustering</span>. 2009 <span>Ninth IEEE International Conference</span> on <span>Bioinformatics</span> and <span>BioEngineering</span>. 2009. pp. 84–91. doi:<a href="https://doi.org/10.1109/BIBE.2009.19">10.1109/BIBE.2009.19</a></div>
</div>
<div class="csl-entry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Harremoes P. Mutual information of contingency tables and related inequalities. 2014 <span>IEEE International Symposium</span> on <span>Information Theory</span>. <span>Honolulu, HI, USA</span>: <span>IEEE</span>; 2014. pp. 2474–2478. doi:<a href="https://doi.org/10.1109/ISIT.2014.6875279">10.1109/ISIT.2014.6875279</a></div>
</div>
</div>
</div>



</div>
            </section>

          </div>
        </div>
      </div>
<a href="conclusion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="HIV-appendix.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/08-HIV-Appendix.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
