<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Viruses, HIV and drug resistance | From sequences to knowledge, improving and learning from sequence alignments</title>
  <meta name="description" content="Chapter 5 Viruses, HIV and drug resistance | From sequences to knowledge, improving and learning from sequence alignments" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Viruses, HIV and drug resistance | From sequences to knowledge, improving and learning from sequence alignments" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Viruses, HIV and drug resistance | From sequences to knowledge, improving and learning from sequence alignments" />
  
  
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-09-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="learning-from-sequences-and-alignments.html"/>
<link rel="next" href="HIV-paper.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />
<link href="libs/tabwid-1.0.0/scrool.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">From sequences to knowledge,</br> improving and learning from sequence alignments.</a></li>

<li class="divider"></li>
<li><a href="index.html#section"></a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#résumé"><i class="fa fa-check"></i>Résumé</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="list-of-acronyms-and-abbreviations.html"><a href="list-of-acronyms-and-abbreviations.html"><i class="fa fa-check"></i>List of Acronyms and abbreviations</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html"><i class="fa fa-check"></i>General Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#organization-of-this-manuscript"><i class="fa fa-check"></i>Organization of this manuscript</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#research-output"><i class="fa fa-check"></i>Research output</a>
<ul>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#journal-publications"><i class="fa fa-check"></i>Journal publications</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#presentations-and-posters"><i class="fa fa-check"></i>Presentations and posters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>1</b> What is Sequence data ?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>1.1</b> Biological sequences, a primer</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>1.1.1</b> What is DNA ?</a></li>
<li class="chapter" data-level="1.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#from-information-to-action"><i class="fa fa-check"></i><b>1.1.2</b> From Information to action</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining sequence data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough"><i class="fa fa-check"></i><b>1.2.1</b> Sanger sequencing, a breakthrough</a></li>
<li class="chapter" data-level="1.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> Next-generation sequencing</a></li>
<li class="chapter" data-level="1.2.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#long-read-sequencing"><i class="fa fa-check"></i><b>1.2.3</b> Long read sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>1.3</b> Sequencing errors, how to account for them ?</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#error-correction-methods"><i class="fa fa-check"></i><b>1.3.1</b> Error correction methods</a></li>
<li class="chapter" data-level="1.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#more-accurate-sequencing-methods"><i class="fa fa-check"></i><b>1.3.2</b> More accurate sequencing methods</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#the-special-case-of-homopolymers"><i class="fa fa-check"></i><b>1.4</b> The special case of homopolymers</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-the-human-genome"><i class="fa fa-check"></i><b>1.4.1</b> Homopolymers and the human genome</a></li>
<li class="chapter" data-level="1.4.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-long-reads"><i class="fa fa-check"></i><b>1.4.2</b> Homopolymers and long reads</a></li>
<li class="chapter" data-level="1.4.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#accounting-for-homopolymers"><i class="fa fa-check"></i><b>1.4.3</b> Accounting for homopolymers</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>2</b> Aligning sequence data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>2.1</b> What is an alignment ?</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#why-align"><i class="fa fa-check"></i><b>2.1.1</b> Why align ?</a></li>
<li class="chapter" data-level="2.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-align-two-sequences"><i class="fa fa-check"></i><b>2.1.2</b> How to align two sequences ?</a></li>
<li class="chapter" data-level="2.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#scoring-and-substitution-models"><i class="fa fa-check"></i><b>2.1.3</b> Scoring and substitution models</a></li>
<li class="chapter" data-level="2.1.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#dealing-with-gaps"><i class="fa fa-check"></i><b>2.1.4</b> Dealing with gaps</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-do-we-speed-up-pairwise-alignment"><i class="fa fa-check"></i><b>2.2</b> How do we speed up pairwise alignment ?</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#changing-the-method"><i class="fa fa-check"></i><b>2.2.1</b> Changing the method</a></li>
<li class="chapter" data-level="2.2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#seed-and-extend-with-data-structures"><i class="fa fa-check"></i><b>2.2.2</b> Seed and extend with data structures</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#the-specificities-of-read-mapping"><i class="fa fa-check"></i><b>2.3</b> The specificities of read-mapping</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-read-mapping"><i class="fa fa-check"></i><b>2.3.1</b> What is read-mapping ?</a></li>
<li class="chapter" data-level="2.3.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#challenges-of-read-mapping"><i class="fa fa-check"></i><b>2.3.2</b> Challenges of read-mapping</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.4</b> Multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#progressive-alignment"><i class="fa fa-check"></i><b>2.4.1</b> Progressive alignment</a></li>
<li class="chapter" data-level="2.4.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#other-methods"><i class="fa fa-check"></i><b>2.4.2</b> Other methods</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#conclusion-1"><i class="fa fa-check"></i><b>2.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="HPC-paper.html"><a href="HPC-paper.html"><i class="fa fa-check"></i><b>3</b> Contribution 1: Improving read alignment by exploring a sequence transformation space</a>
<ul>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#highlights"><i class="fa fa-check"></i>Highlights</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#graphical-abstract"><i class="fa fa-check"></i>Graphical Abstract</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="3.1" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="HPC-paper.html"><a href="HPC-paper.html#methods"><i class="fa fa-check"></i><b>3.2</b> Methods</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:msr-def"><i class="fa fa-check"></i><b>3.2.1</b> Streaming sequence reductions</a></li>
<li class="chapter" data-level="3.2.2" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:enum"><i class="fa fa-check"></i><b>3.2.2</b> Restricting the space of streaming sequence reductions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="HPC-paper.html"><a href="HPC-paper.html#datasets-and-pipelines"><i class="fa fa-check"></i><b>3.3</b> Datasets and Pipelines</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="HPC-paper.html"><a href="HPC-paper.html#datasets"><i class="fa fa-check"></i><b>3.3.1</b> Datasets</a></li>
<li class="chapter" data-level="3.3.2" data-path="HPC-paper.html"><a href="HPC-paper.html#simulation-pipeline"><i class="fa fa-check"></i><b>3.3.2</b> Simulation pipeline</a></li>
<li class="chapter" data-level="3.3.3" data-path="HPC-paper.html"><a href="HPC-paper.html#evaluation-pipeline"><i class="fa fa-check"></i><b>3.3.3</b> Evaluation pipeline</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-results"><i class="fa fa-check"></i><b>3.4</b> Results</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="HPC-paper.html"><a href="HPC-paper.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>3.4.1</b> Selection of mapping-friendly sequence reductions</a></li>
<li class="chapter" data-level="3.4.2" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>3.4.2</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes</a></li>
<li class="chapter" data-level="3.4.3" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>3.4.3</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome</a></li>
<li class="chapter" data-level="3.4.4" data-path="HPC-paper.html"><a href="HPC-paper.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>3.4.4</b> Raw mapping improves upon HPC on centromeric regions</a></li>
<li class="chapter" data-level="3.4.5" data-path="HPC-paper.html"><a href="HPC-paper.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>3.4.5</b> Positions of incorrectly mapped reads across the entire human genome</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="HPC-paper.html"><a href="HPC-paper.html#discussion"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="HPC-paper.html"><a href="HPC-paper.html#limitations-of-this-study"><i class="fa fa-check"></i><b>3.6</b> Limitations of this study</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#author-contributions"><i class="fa fa-check"></i>Author contributions</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#declaration-of-interests"><i class="fa fa-check"></i>Declaration of interests</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#star-methods"><i class="fa fa-check"></i>STAR Methods</a>
<ul>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#lead-contact"><i class="fa fa-check"></i>Lead contact</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#materials-availability"><i class="fa fa-check"></i>Materials availability</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#data-and-code-availability"><i class="fa fa-check"></i>Data and code availability</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#method-details"><i class="fa fa-check"></i>Method details</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#supplementary-information"><i class="fa fa-check"></i>Supplementary information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html"><i class="fa fa-check"></i><b>4</b> Learning from sequences and alignments</a>
<ul>
<li class="chapter" data-level="4.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#what-to-learn"><i class="fa fa-check"></i><b>4.1</b> What to learn ?</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#supervised"><i class="fa fa-check"></i><b>4.1.1</b> Supervised</a></li>
<li class="chapter" data-level="4.1.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#unsupervised"><i class="fa fa-check"></i><b>4.1.2</b> Unsupervised</a></li>
<li class="chapter" data-level="4.1.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#others"><i class="fa fa-check"></i><b>4.1.3</b> Others</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#how-to-learn"><i class="fa fa-check"></i><b>4.2</b> How to learn ?</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>4.2.1</b> Tests and statistical learning</a></li>
<li class="chapter" data-level="4.2.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#more-complex-methods-machine-learning"><i class="fa fa-check"></i><b>4.2.2</b> More complex methods, machine learning</a></li>
<li class="chapter" data-level="4.2.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#mediadeep-learning"><i class="fa fa-check"></i><b>4.2.3</b> mediaDeep Learning</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#preprocessing-the-alignment-for-machine-learning"><i class="fa fa-check"></i><b>4.3</b> Preprocessing the alignment for machine learning</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#general-purpose-embeddings"><i class="fa fa-check"></i><b>4.3.1</b> General purpose embeddings</a></li>
<li class="chapter" data-level="4.3.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#biological-sequence-specific-embeddings"><i class="fa fa-check"></i><b>4.3.2</b> Biological sequence-specific embeddings</a></li>
<li class="chapter" data-level="4.3.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#learned-embeddings"><i class="fa fa-check"></i><b>4.3.3</b> Learned embeddings</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#conclusion-2"><i class="fa fa-check"></i><b>4.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html"><i class="fa fa-check"></i><b>5</b> Viruses, HIV and drug resistance</a>
<ul>
<li class="chapter" data-level="5.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#what-are-viruses"><i class="fa fa-check"></i><b>5.1</b> What are viruses ?</a></li>
<li class="chapter" data-level="5.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#getting-to-know-hiv"><i class="fa fa-check"></i><b>5.2</b> Getting to know HIV</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#quick-presentation-of-hiv"><i class="fa fa-check"></i><b>5.2.1</b> Quick Presentation of HIV</a></li>
<li class="chapter" data-level="5.2.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#the-replication-cycle-of-hiv"><i class="fa fa-check"></i><b>5.2.2</b> The replication cycle of HIV</a></li>
<li class="chapter" data-level="5.2.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#genetics-of-hiv"><i class="fa fa-check"></i><b>5.2.3</b> Genetics of HIV</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#drug-resistance-in-hiv"><i class="fa fa-check"></i><b>5.3</b> Drug resistance in HIV</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#a-quick-history-of-art"><i class="fa fa-check"></i><b>5.3.1</b> A quick history of ART</a></li>
<li class="chapter" data-level="5.3.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#drug-mechanisms"><i class="fa fa-check"></i><b>5.3.2</b> Main mechanisms of viral proteins, antiretroviral drugs and associated resistance.</a></li>
<li class="chapter" data-level="5.3.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#consequences-of-resistance-on-global-health"><i class="fa fa-check"></i><b>5.3.3</b> Consequences of resistance on global health</a></li>
<li class="chapter" data-level="5.3.4" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#finding-drms"><i class="fa fa-check"></i><b>5.3.4</b> Finding DRMS </a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#conclusion-3"><i class="fa fa-check"></i><b>5.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>6</b> Contribution 2: Inferring mutation roles from sequence alignments using machine learning</a>
<ul>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#abstract-paper"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#author-summary"><i class="fa fa-check"></i>Author summary</a></li>
<li class="chapter" data-level="6.1" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>6.2</b> Materials and methods</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>6.2.2</b> Classifier training</a></li>
<li class="chapter" data-level="6.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>6.2.3</b> Measuring classifier performance</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-results"><i class="fa fa-check"></i><b>6.3</b> Results</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>6.3.1</b> Classifier performance &amp; interpretation</a></li>
<li class="chapter" data-level="6.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>6.3.2</b> Additional classification results</a></li>
<li class="chapter" data-level="6.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>6.3.3</b> Identifying new mutations from classifiers</a></li>
<li class="chapter" data-level="6.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>6.3.4</b> Detailed analysis of potentially resistance-associated mutations</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>6.4</b> Discussion and perspectives</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-acknowledgments"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#supporting-information"><i class="fa fa-check"></i>Supporting Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>7</b> Learning alignments, an interesting perspective</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-pairwise-alignment"><i class="fa fa-check"></i><b>7.1</b> Learning pairwise alignment</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#dedal"><i class="fa fa-check"></i><b>7.1.1</b> DEDAL</a></li>
<li class="chapter" data-level="7.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-an-alignment"><i class="fa fa-check"></i><b>7.1.2</b> predicting an alignment</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#what-else-could-we-learn"><i class="fa fa-check"></i><b>7.2</b> What else could we learn ?</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-to-predict-seeds-or-starting-positions"><i class="fa fa-check"></i><b>7.2.1</b> Learn to predict seeds or starting positions</a></li>
<li class="chapter" data-level="7.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-pre-processing-functions"><i class="fa fa-check"></i><b>7.2.2</b> Learn pre-processing functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html"><i class="fa fa-check"></i>Global conclusion</a>
<ul>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hpc-part"><i class="fa fa-check"></i>HPC part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hiv-part"><i class="fa fa-check"></i>HIV part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#final-words"><i class="fa fa-check"></i>Final words</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-references.html"><a href="global-references.html"><i class="fa fa-check"></i>Global References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”</a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation</a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#msr-performance-comparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison</a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome"><i class="fa fa-check"></i><b>A.3</b> Analyzing read origin on whole human genome</a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.4</b> Performance of MSRs on the Drosophila genome</a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#key-resource-table"><i class="fa fa-check"></i><b>A.5</b> Key Resource Table</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “HIV and DRMs”</a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html#detailed-list-of-hiv-1-protein-structures-used-for-figure-generation."><i class="fa fa-check"></i><b>B.1</b> Detailed list of HIV-1 protein structures used for figure generation.</a></li>
<li class="chapter" data-level="B.2" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html#list-of-all-antiretroviral-drugs"><i class="fa fa-check"></i><b>B.2</b> List of all antiretroviral drugs</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>C</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”</a>
<ul>
<li class="chapter" data-level="C.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>C.1</b> S1 Appendix (Technical appendix).</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>C.1.1</b> Data</a></li>
<li class="chapter" data-level="C.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>C.1.2</b> Classifiers</a></li>
<li class="chapter" data-level="C.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>C.1.3</b> Scoring</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>C.2</b> S1 Fig.</a></li>
<li class="chapter" data-level="C.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>C.3</b> S2 Fig.</a></li>
<li class="chapter" data-level="C.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>C.4</b> S3 Fig.</a></li>
<li class="chapter" data-level="C.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>C.5</b> S1 Table.</a></li>
<li class="chapter" data-level="C.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>C.6</b> S2 Appendix. (Fisher exact tests)</a></li>
<li class="chapter" data-level="C.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>C.7</b> S1 Data.</a></li>
<li class="chapter" data-level="C.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>C.8</b> S2 Data.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequences to knowledge, improving and learning from sequence alignments</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="viruses-hiv-and-drug-resistance" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Viruses, HIV and drug resistance<a href="viruses-hiv-and-drug-resistance.html#viruses-hiv-and-drug-resistance" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="what-are-viruses" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> What are viruses ?<a href="viruses-hiv-and-drug-resistance.html#what-are-viruses" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Viruses occupy a strange place in the tree of life, with many debating if they are actually alive or not. André Lwoff gave what is probably the most fitting definition: <em>“viruses are viruses”</em><span class="citation"><sup><a href="#ref-lwoffConceptVirus1957" role="doc-biblioref">402</a></sup></span>. Despite this ambiguity, viruses share some common characteristics which allow us to define them as intracellular parasites<span class="citation"><sup><a href="#ref-minorViruses2014" role="doc-biblioref">403</a></sup></span>:</p>
<ol style="list-style-type: decimal">
<li>Viruses have some type of genetic information, contained in either DNA or RNA</li>
<li>This genetic information is protected by some form of envelope</li>
<li>They use the cellular machinery of host cells to make copies of themselves.</li>
</ol>
<p>While we all know that viruses can be pathogenic and dangerous (the recent example of SARS-CoV2 springs to mind), that is not necessarily the case and some viruses like GBV-C<span class="citation"><sup><a href="#ref-stapletonGBVirusesReview2011" role="doc-biblioref">404</a></sup></span> and certain strains of H5N1 <em>Influenza</em><span class="citation"><sup><a href="#ref-yamamotoCharacterizationNonpathogenicH5N12011" role="doc-biblioref">405</a></sup></span> are non pathogenic and essentially harmless.</p>
<p>Viruses have been discovered for all three domains of life: Eukaryota, Bacteria and Archea. In Eukaryota many viruses have been discovered for animals (both vertebrate<span class="citation"><sup><a href="#ref-shiEvolutionaryHistoryVertebrate2018" role="doc-biblioref">406</a></sup></span> and invertebrate<span class="citation"><sup><a href="#ref-adamsAtlasInvertebrateViruses2017" role="doc-biblioref">407</a></sup></span>), plants<span class="citation"><sup><a href="#ref-lefeuvreEvolutionEcologyPlant2019" role="doc-biblioref">408</a></sup></span>, protozoa<span class="citation"><sup><a href="#ref-wangVirusesParasiticProtozoa1991" role="doc-biblioref">409</a></sup></span>, chromista<span class="citation"><sup><a href="#ref-ferminVirusesProkaryotesProtozoa2018" role="doc-biblioref">410</a></sup></span> and even fungi<span class="citation"><sup><a href="#ref-sutelaVirusesFungiOomycetes2019" role="doc-biblioref">411</a></sup></span>. Bacterial viruses known as phages have been known to exists since the beginning of the 20th century<span class="citation"><sup><a href="#ref-twortINVESTIGATIONNATUREULTRAMICROSCOPIC1915" role="doc-biblioref">412</a>,<a href="#ref-delbrockBacterialVirusesBacteriophages1946" role="doc-biblioref">413</a></sup></span>. These bacteriophages are being considered as a therapeutic alternative to antibiotics<span class="citation"><sup><a href="#ref-clarkBacterialVirusesHuman2004" role="doc-biblioref">414</a>,<a href="#ref-vankan-davelaarUsingVirusesNanomedicines2014" role="doc-biblioref">415</a></sup></span> which could help with multi-drug-resistant bacterial pathogens. Archea are also known to have their own viral infections<span class="citation"><sup><a href="#ref-prangishviliVirusesArchaea2016" role="doc-biblioref">416</a>,<a href="#ref-prangishviliVirusesArchaeaUnifying2006" role="doc-biblioref">417</a></sup></span>.</p>
<p>Strangely even viruses of viruses seem to exist such as the plant satellite virus<span class="citation"><sup><a href="#ref-franckiPlantVirusSatellites" role="doc-biblioref">418</a>,<a href="#ref-xuPlantVirusSatellites2011" role="doc-biblioref">419</a></sup></span> or hepatitis delta virus<span class="citation"><sup><a href="#ref-laiMolecularBiologyHepatitis1995" role="doc-biblioref">420</a>,<a href="#ref-hughesHepatitisDeltaVirus2011" role="doc-biblioref">421</a></sup></span>. These “viroids” do not infect viral hosts <em>per se</em> but they cannot replicate on their own. Replication must happen during co-infection with a larger virus. More recently, true viruses of viruses called virophages have been discovered. These virophages like sputnik<span class="citation"><sup><a href="#ref-desnuesChapterSputnikVirophage2012" role="doc-biblioref">422</a></sup></span> or zamilon<span class="citation"><sup><a href="#ref-gaiaZamilonNovelVirophage2014" role="doc-biblioref">423</a></sup></span> specifically infect giant viruses.</p>
<p>There is a huge diversity of viruses affecting all types of life, and new viruses are being discovered all the time<span class="citation"><sup><a href="#ref-edgarPetabasescaleSequenceAlignment2022" role="doc-biblioref">424</a></sup></span>. This diversity hints at a rich and long evolutionary history. When and where viruses originated is still under study<span class="citation"><sup><a href="#ref-nasirInvestigatingConceptOrigin2020" role="doc-biblioref">425</a>,<a href="#ref-forterreOriginViruses2009" role="doc-biblioref">426</a></sup></span> and we might never know how they emerged, it is however believed that they may have played an important role in the emergence of eukaryotic cells<span class="citation"><sup><a href="#ref-forterreOriginVirusesTheir2006" role="doc-biblioref">427</a></sup></span>. This co-evolution between virus and host cell shows a strong link between the two organisms and some parts of the human genome are likely of ancient viral origin<span class="citation"><sup><a href="#ref-boekeRetrotransposonsEndogenousRetroviruses1997" role="doc-biblioref">428</a>,<a href="#ref-kojimaViruslikeInsertionsSequence2021" role="doc-biblioref">429</a></sup></span>. It has been estimated that 1% to 8% of the human genome are endogenous retroviral sequences<span class="citation"><sup><a href="#ref-lowerVirusesAllUs1996" role="doc-biblioref">430</a>,<a href="#ref-griffithsEndogenousRetrovirusesHuman2001" role="doc-biblioref">431</a></sup></span>.</p>
<p>The rich diversity of viruses is reflected in the variety of genetic information support, replication strategy, physical and genomic size, as well as shape. The differences in genetic information support and replication strategy form the basis of the Baltimore virus classification system<span class="citation"><sup><a href="#ref-baltimoreExpressionAnimalVirus1971" role="doc-biblioref">432</a></sup></span> , still used today<span class="citation"><sup><a href="#ref-kooninBaltimoreClassificationViruses2021" role="doc-biblioref">433</a></sup></span> to classify virus lineages.</p>
<p>As stated above all viruses have some genetic information, this information is stored either as DNA or as RNA, which is the molecule of choice for 70% of human pathogenic viruses<span class="citation"><sup><a href="#ref-domingoRNAVirusGenomes2018" role="doc-biblioref">434</a></sup></span> <em>(HIV and SARS-CoV 2 are RNA viruses)</em>.</p>
<p>For DNA viruses, the molecule can be double-stranded as for <em>Herpesvirus</em><span class="citation"><sup><a href="#ref-mcgeochTopicsHerpesvirusGenomics2006" role="doc-biblioref">435</a>,<a href="#ref-boehmerHerpesVirusReplication2003" role="doc-biblioref">436</a></sup></span>, single-stranded like in the case of <em>Papillomavirus</em><span class="citation"><sup><a href="#ref-brentjensHumanPapillomavirusReview2002" role="doc-biblioref">437</a></sup></span> or even circular in the case of the Hepatitis B virus<span class="citation"><sup><a href="#ref-kayHepatitisVirusGenetic2007" role="doc-biblioref">438</a></sup></span>. This molecular diversity is also present in RNA viruses where the RNA molecule can be double-stranded like for <em>Rotavirus</em><span class="citation"><sup><a href="#ref-parasharRotavirus1998" role="doc-biblioref">439</a></sup></span>, or single-stranded. Furthermore, for single-stranded RNA viruses the strand can either be positive <em>(i.e. can be directly translated into a protein)</em> like Hepatitis C virus<span class="citation"><sup><a href="#ref-simmondsVariabilityHepatitisVirus1995" role="doc-biblioref">440</a></sup></span> or <em>Poliovirus</em><span class="citation"><sup><a href="#ref-wimmerGeneticsPoliovirus1993" role="doc-biblioref">441</a>,<a href="#ref-racanielloOneHundredYears2006" role="doc-biblioref">442</a></sup></span>; conversely there are negative-strand RNA viruses, for which the complementary strand of RNA must be synthesized before translation into a protein, such as the Influenza or Measles viruses<span class="citation"><sup><a href="#ref-paleseNegativestrandRNAViruses1996" role="doc-biblioref">443</a></sup></span>.</p>
<p>This diversity in genetic information support implies a necessary diversity in replication strategy. The main replication strategies are as follows<span class="citation"><sup><a href="#ref-domingoVirusEvolution2014" role="doc-biblioref">444</a></sup></span>:</p>
<ul>
<li><p>The RNA molecule is directly copied as RNA. This is the strategy followed by single-stranded RNA coronaviruses<span class="citation"><sup><a href="#ref-vkovskiCoronavirusBiologyReplication2021" role="doc-biblioref">445</a></sup></span>, Dengue viruses<span class="citation"><sup><a href="#ref-backDengueVirusesOverview2013" role="doc-biblioref">446</a></sup></span> or Hepatitis C virus<span class="citation"><sup><a href="#ref-dustinHepatitisVirusLife2016" role="doc-biblioref">447</a></sup></span>.</p></li>
<li><p>The DNA molecule is directly replicated as DNA. this can happen for both single-stranded DNA viruses like <em>Papillomavirus</em><span class="citation"><sup><a href="#ref-kadajaPapillomavirusDNAReplication2009" role="doc-biblioref">448</a></sup></span>and double-stranded DNA viruses like Herpes simplex virus<span class="citation"><sup><a href="#ref-wellerHerpesSimplexViruses2012" role="doc-biblioref">449</a></sup></span>.</p></li>
<li><p>The DNA molecule is replicated by going through an RNA intermediary like Hepatitis B virus<span class="citation"><sup><a href="#ref-beckHepatitisVirusReplication2007" role="doc-biblioref">450</a></sup></span>.</p></li>
<li><p>The RNA molecule is replicated by going through a DNA intermediary. This strategy is used by retroviruses that integrate this viral DNA intermediary into the host DNA, like HIV-1 (see Section <a href="viruses-hiv-and-drug-resistance.html#the-replication-cycle-of-hiv">5.2.2</a>).</p></li>
</ul>
<p>Finally, the genetic diversity of viruses is reflected in their physical characteristics: viruses come in all shapes and sizes. Physical size range from 17nm for plant satellite viruses<span class="citation"><sup><a href="#ref-pyleChapter58Biology2017" role="doc-biblioref">451</a></sup></span> to the giant 400nm <em>Mimivirus</em><span class="citation"><sup><a href="#ref-raoult2megabaseGenomeSequence2004" role="doc-biblioref">452</a></sup></span>. Genomic size is also quite variable, there is a stark contrast between the 860 bp <em>Circovirus SFBeef</em> and the 2.5 Mbp <em>Pandoravirus salinus</em> genomes<span class="citation"><sup><a href="#ref-campillo-balderasViralGenomeSize2015" role="doc-biblioref">453</a></sup></span>. Viruses come in a variety of shapes<span class="citation"><sup><a href="#ref-cannVirusStructure2015" role="doc-biblioref">454</a></sup></span>: icosahedral for HIV, helical for the tobacco mosaic virus or a distinctive head-tail shape for bacteriophages.</p>
<p>Although there are a large number of viruses, and many of them are of great importance for human health, we will now focus on one virus of particular importance: Human Immunodeficiency Virus otherwise known as HIV.</p>
</div>
<div id="getting-to-know-hiv" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Getting to know HIV<a href="viruses-hiv-and-drug-resistance.html#getting-to-know-hiv" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="quick-presentation-of-hiv" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Quick Presentation of HIV<a href="viruses-hiv-and-drug-resistance.html#quick-presentation-of-hiv" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>HIV is a single-stranded RNA retrovirus that is responsible for the Acquired Immune Deficiency Syndrome (AIDS) pandemic that has been around for the last couple decades. This virus is transmitted through sexual contact or through blood. Sexual activity is the largest transmission factor followed by intravenous drug use<span class="citation"><sup><a href="#ref-hladikSettingStageHost2008" role="doc-biblioref">455</a>,<a href="#ref-shawHIVTransmission2012" role="doc-biblioref">456</a></sup></span>.</p>
<p>The HIV infects cells of the host immune system, specifically CD4 T-cell lymphocytes and destroys them due to it’s replication process<span class="citation"><sup><a href="#ref-weissHowDoesHIV1993" role="doc-biblioref">457</a></sup></span>. CD4 T-cells are an essential part of the immune system response helping fight against infection in humans. An HIV infection typically start with an asymptomatic phase that can last years, followed by a growth in viral replication leading to a decrease in CD4 cells which progresses into AIDS<span class="citation"><sup><a href="#ref-melhuishNaturalHistoryHIV2018" role="doc-biblioref">458</a></sup></span>. During AIDS, when the CD4 cell count is low enough opportunistic diseases such as pneumonia or tuberculosis<span class="citation"><sup><a href="#ref-murrayPulmonaryComplicationsAcquired1984" role="doc-biblioref">459</a></sup></span> can easily infect the host, leading to death when the immune system is weak enough.</p>
<p>The HIV/AIDS is one of the deadliest pandemics in history, estimated to have lead to the death of 36 million people<span class="citation"><sup><a href="#ref-sampathPandemicsThroughoutHistory2021" role="doc-biblioref">460</a></sup></span>. In 2010<span class="citation"><sup><a href="#ref-worldhealthorganizationGlobalReportUNAIDS2010" role="doc-biblioref">461</a></sup></span> approximately 33 million people were infected with HIV, 2.6 million of which were due to new infections, and 1.8 million died of AIDS. Most of the new infections happened in economically developing regions of the world, 70% of them coming from sub-Saharan Africa<span class="citation"><sup><a href="#ref-worldhealthorganizationGlobalReportUNAIDS2010" role="doc-biblioref">461</a></sup></span>. As of 2020, these numbers have decreased with “only” 1.5 million new infections and 680,000 AIDS deaths, which is encouraging from a public health perspective.</p>
<p>The HIV-1 virus was discovered simultaneously in 1983 by Françoise Barré-Sinoussi, Luc Montagnier<span class="citation"><sup><a href="#ref-barre-sinoussiIsolationTlymphotropicRetrovirus1983" role="doc-biblioref">462</a></sup></span> and Robert Gallo<span class="citation"><sup><a href="#ref-galloIsolationHumanTcell1983" role="doc-biblioref">463</a></sup></span>. There exists a second HIV-2 virus discovered shortly after HIV-1<span class="citation"><sup><a href="#ref-clavelIsolationNewHuman1986" role="doc-biblioref">464</a></sup></span>, it is however less transmissible than HIV-1 which is largely responsible for the global HIV/AIDS pandemic<span class="citation"><sup><a href="#ref-gilbert2003" role="doc-biblioref">465</a></sup></span>. In Africa in 2006, HIV-1 infections were rising where HIV-2 were declining<span class="citation"><sup><a href="#ref-vanderloeffSixteenYearsHIV2006" role="doc-biblioref">466</a></sup></span>.</p>
<p>While both viruses are of zoonotic origin, from transmissions of Simian Immunodeficiency Virus (SIV) from primates to humans. HIV-1 most likely originates from an SIV present in chimpanzees<span class="citation"><sup><a href="#ref-gaoOriginHIV1Chimpanzee1999" role="doc-biblioref">467</a>–<a href="#ref-sharpOriginsHIVAIDS2011" role="doc-biblioref">469</a></sup></span>, and HIV-2 from an SIV present in Sooty mangabeys<span class="citation"><sup><a href="#ref-hirschAfricanPrimateLentivirus1989" role="doc-biblioref">470</a>–<a href="#ref-chenGeneticCharacterizationNew1996" role="doc-biblioref">472</a></sup></span>.</p>
<p>Several independent such transmissions have resulted in 4 lineages of HIV-1 labeled groups M, N, O and P<span class="citation"><sup><a href="#ref-hemelaarOriginDiversityHIV12012" role="doc-biblioref">473</a></sup></span> (similarly HIV-2 is split into groups A to H also resulting from independent zoonotic transmissions). While Groups N and P have been identified in only a handful of individuals in Cameroon , and group O is estimated to a few thousand cases in western Africa, the majority of the pandemic is due to viruses from group M.</p>
<p>The most most recent common ancestor, <em>i.e. the putative virus that founded group M</em>, is estimated to have originated in what is now the Democratic Republic of Congo<span class="citation"><sup><a href="#ref-worobeyDirectEvidenceExtensive2008" role="doc-biblioref">474</a>–<a href="#ref-fariaEarlySpreadEpidemic2014" role="doc-biblioref">476</a></sup></span> at some point between 1910 and 1931<span class="citation"><sup><a href="#ref-worobeyDirectEvidenceExtensive2008" role="doc-biblioref">474</a>,<a href="#ref-korberTimingAncestorHIV12000" role="doc-biblioref">477</a>,<a href="#ref-rambautCausesConsequencesHIV2004" role="doc-biblioref">478</a></sup></span>.</p>
<p>Group M is further subdivided into 9 subtypes each with distinct genetic characteristics, labeled A to K<span class="citation"><sup><a href="#ref-sharpOriginsHIVAIDS2011" role="doc-biblioref">469</a>,<a href="#ref-mccutchanGlobalEpidemiologyHIV2006" role="doc-biblioref">479</a></sup></span>. Like in many viruses<span class="citation"><sup><a href="#ref-perez-losadaRecombinationVirusesMechanisms2015" role="doc-biblioref">480</a></sup></span>, when 2 genetically different strains of HIV co-infect a single host there is a risk genetic recombination leading to a new strain<span class="citation"><sup><a href="#ref-robertsonRecombinationAIDSViruses1995" role="doc-biblioref">481</a></sup></span>. During recombination, a new genome is formed from parts of the original genomes. This can lead to new strains that can spread and form lineages of their own. HIV strains resulting from recombination are called Cirulating Recombinant Forms (CRFs). There are currently 118 identifed HIV-1 CRFs in the Los Alamos National Laboratory HIV sequence database<span class="citation"><sup><a href="#ref-HIVCirculatingRecombinant" role="doc-biblioref">482</a></sup></span> (1 for HIV-2). There also exist many unique recombinant forms (URFs). Recombination can be particulary bothersome, complicating evolutionary analyses<span class="citation"><sup><a href="#ref-posadaRecombinationEvolutionaryGenomics2002" role="doc-biblioref">483</a></sup></span>, or facilitating drug resistance and hindering vaccine development<span class="citation"><sup><a href="#ref-taylorChallengeHIV1Subtype2008" role="doc-biblioref">484</a></sup></span>.</p>
<p>While subtype C represented almost half of global infections from 2004 to 2007, subtype B is the majority subtype in richer countries of North America and Western Europe<span class="citation"><sup><a href="#ref-hemelaarGlobalTrendsMolecular2011" role="doc-biblioref">485</a></sup></span> where sequencing efforts are more common. This accounts for an over-representation of subtype B sequences in public databases such as the Los Alamos sequence database where 54% of sequences are of the B subtype and only 15% are C<span class="citation"><sup><a href="#ref-DistributionAllHIV1" role="doc-biblioref">486</a></sup></span>.</p>
</div>
<div id="the-replication-cycle-of-hiv" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> The replication cycle of HIV<a href="viruses-hiv-and-drug-resistance.html#the-replication-cycle-of-hiv" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The viruses replication cycle and its immune-cell host specificity are what makes it particularly dangerous. This replication cycle can broadly be categorized into 9 separate steps<span class="citation"><sup><a href="#ref-freedHIV1Replication2001" role="doc-biblioref">487</a>,<a href="#ref-fergusonHIV1ReplicationCycle2002" role="doc-biblioref">488</a></sup></span> shown in Figure <a href="viruses-hiv-and-drug-resistance.html#fig:hivCycle">5.1</a>.</p>
<ol style="list-style-type: decimal">
<li><p>An HIV virion binds itself to the CD4 host cell through membrane proteins.</p></li>
<li><p>The virion envelope and host cell membrane fuse together allowing the viral genetic material and proteins to enter the host cell.</p></li>
<li><p>The viral RNA is reverse-transcripted into viral DNA</p></li>
<li><p>The viral DNA is integrated into the host cell genome</p></li>
<li><p>The integrated viral DNA is transcribed by the host cell machinery into multiple copies of viral RNA</p></li>
<li><p>The viral RNA is translated into immature viral polyproteins</p></li>
<li><p>The viral polyproteins are cleaved to form individual viral proteins.</p></li>
<li><p>The newly synthesized viral RNA and viral proteins gather around the host-cell membrane which starts budding to create a new virion</p></li>
<li><p>Once the budding is complete, the virion is released from the host cell and matures before being able to infect other CD4 cells and replicate again.</p></li>
</ol>
<p>The successive infection of CD4 cells by HIV virions leads to cellular death due to inflammatory response and/or activation of apoptosis<span class="citation"><sup><a href="#ref-gougeonDirectIndirectMechanisms1993" role="doc-biblioref">489</a>,<a href="#ref-vidyavijayanPathophysiologyCD4TCell2017" role="doc-biblioref">490</a></sup></span>. The gradual depletion of CD4 cells in the infected individual’s body lead to the suppression of the immune system, eventually leading to AIDS.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hivCycle"></span>
<img src="figures/HIV-Intro/HIV-cycle.png" alt="**Main steps of HIV-1 replication cycle**
The HIV virion contains viral RNA and three essential proteins: Reverse Transcriptase (RT) represented in red, Integrase (IN) represented in cyan and Protease (PR) represented in yellow.
" width="60%" />
<p class="caption">
Figure 5.1: <strong>Main steps of HIV-1 replication cycle</strong>
The HIV virion contains viral RNA and three essential proteins: Reverse Transcriptase (RT) represented in red, Integrase (IN) represented in cyan and Protease (PR) represented in yellow.
</p>
</div>
</div>
<div id="genetics-of-hiv" class="section level3 hasAnchor" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> Genetics of HIV<a href="viruses-hiv-and-drug-resistance.html#genetics-of-hiv" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The replication cycle described in Section <a href="viruses-hiv-and-drug-resistance.html#the-replication-cycle-of-hiv">5.2.2</a> is made possible by the 15 proteins of HIV. These proteins are coded by 9 separate genes<span class="citation"><sup><a href="#ref-frankelHIV1FifteenProteins1998" role="doc-biblioref">491</a></sup></span>. An overview of the HIV proteins, their structure and localization within the viral particle can be seen in Figure <a href="viruses-hiv-and-drug-resistance.html#fig:hivStructure">5.2</a>.</p>
<p>The HIV genome is made up of three main genes each coding for polyproteins and six genes coding for proteins with regulatory and accessory roles. The three polyproteins correspond to long chains of amino acids which are subsequently cleaved at specific positions to result in separate viral proteins.</p>
<p>The <em>gag</em> (“group-specific-antigen”) gene codes for the Gag polyprotein which, once cleaved, results in four proteins with mainly structural roles:</p>
<ul>
<li><p>The Matrix protein (MA or p17) lines the internal surface if the virion membrane maintaining the shape and structural integrity of the virion.</p></li>
<li><p>The Capsid protein (CA or p24) forms an inner core (the capsid) inside the virion around the viral RNA. It helps protect the viral genetic information.</p></li>
<li><p>The Nucleocapsid protein (NC or p7) binds with the viral RNA inside the capsid, stabilizing the molecule and further protecting the genetic information.</p></li>
<li><p>The p6 protein is a small, largely unstructured protein<span class="citation"><sup><a href="#ref-fossenSolutionStructureHuman2005" role="doc-biblioref">492</a></sup></span> that is suspected of playing a role in virion budding and release from the host cell at the end of the replication cycle<span class="citation"><sup><a href="#ref-gottlingerEffectMutationsAffecting1991" role="doc-biblioref">493</a>,<a href="#ref-huangP6GagRequiredParticle1995" role="doc-biblioref">494</a></sup></span>.</p></li>
</ul>
<p>The <em>pol</em> (“polymerase”) gene coding for the Pol polyprotein. After cleaving this results in the three essential viral enzymes at the heart of the replication cycle:</p>
<ul>
<li><p>The Protease (PR) is responsible for cleaving the Gag, Pol and Env polyproteins to get the individual viral proteins. Without it the individual viral proteins cannot come into being and therefore cannot function, stopping viral replication.</p></li>
<li><p>The Reverse Transcriptase (RT or p51/p66) is responsible for synthesizing viral DNA from the viral RNA template contained in the virion. This is the first step in hijacking the cellular machinery for replication. Without viral DNA, HIV replication is impossible.</p></li>
<li><p>The Integrase (IN) is responsible for integrating the viral DNA produced by RT in to the host cell DNA. Once the viral DNA is inside the host genome it can be transcribed and then translated (as described in Section <a href="what-is-sequence-data.html#biological-sequences-a-primer">1.1</a>) to produce new copies of the viral RNA and proteins. Without this integration step the viral genetic information cannot be expressed and the replication cycle is stopped.</p></li>
</ul>
<p>These three proteins are of particular importance and we will go into more detail about them in Section <a href="viruses-hiv-and-drug-resistance.html#drug-mechanisms">5.3.2</a>.</p>
<p>The <em>env</em> (“envelope”) gene code for the Env, the third and last polyprotein. The two resulting proteins coat the membrane of the virion and are responsible for binding with the CD4 host cells.</p>
<ul>
<li><p>The Surface protein (SU or gp120) binds to receptors on the surface of CD4 cells and allow the virion to attach itself to the host cell<span class="citation"><sup><a href="#ref-bourHumanImmunodeficiencyVirus1995" role="doc-biblioref">495</a></sup></span>. It also enables membrane fusion, the essential first step in the viral replication cycle<span class="citation"><sup><a href="#ref-hernandezViruscellCellcellFusion1996" role="doc-biblioref">496</a></sup></span>.</p></li>
<li><p>The Transmembrane protein (TM or gp41) anchors SU into the virion membrane.</p></li>
</ul>
<p>The 6 remaining genes all code for single proteins. Two of these have essential regulatory roles and the remaining four have accessory roles.</p>
<p>The <em>tat</em> (“trans-activator of transcription”) gene codes for Tat, the first essential regulatory protein. Tat activates and promotes transcription leading to more numerous and longer copies of the viral RNA<span class="citation"><sup><a href="#ref-jonesControlRnaInitiation1994" role="doc-biblioref">497</a></sup></span>. The <em>rev</em> (for “regulator of virion”) gene codes for Rev, the second essential regulatory protein. Rev helps transcribed viral RNA exist the nucleus of the host cell in order to be translated to viral proteins or be packages in new, budding virions<span class="citation"><sup><a href="#ref-hopeViralRNAExport1997" role="doc-biblioref">498</a></sup></span>.</p>
<p>The remaining four accessory genes are as follows: <em>nef</em> (“negative regulatory factor”) code for the Nef protein which prevents the production of the CD4 cellular defense proteins increasing infectivity<span class="citation"><sup><a href="#ref-mangasarianMultifacetedRoleHIV1997" role="doc-biblioref">499</a></sup></span>; <em>vif</em> (“viral infectivity factor”) codes for the Vif protein which also increases viral infectivity<span class="citation"><sup><a href="#ref-cohenRoleAuxiliaryProteins1996" role="doc-biblioref">500</a></sup></span>; <em>vpu</em> (“viral protein U”) codes for Vpu which likely helps during release of new virions<span class="citation"><sup><a href="#ref-cohenRoleAuxiliaryProteins1996" role="doc-biblioref">500</a>,<a href="#ref-lambVpuVprHuman1997" role="doc-biblioref">501</a></sup></span> as well as preventing production of CD4 in the host cell, it is not believed to be present in the mature virion as it binds to host cellulat membranes<span class="citation"><sup><a href="#ref-khanRoleViralProtein2021" role="doc-biblioref">502</a></sup></span>; <em>vpr</em> (“viral protein R”) likely helps viral DNA enter the host cell nucleus and prevents the natural host cell reproduction cycle<span class="citation"><sup><a href="#ref-emermanHIV1VprCell1996" role="doc-biblioref">503</a></sup></span>.</p>
<p>The existence of a 10th HIV-1 gene was suggested in 1988<span class="citation"><sup><a href="#ref-millerHumanImmunodeficiencyVirus1988" role="doc-biblioref">504</a></sup></span>, overlapping the <em>env</em> gene and coding for proteins on the other strand of viral DNA than the other genes. This putative gene was named <em>asp</em> (“antisense protein”) and Asp transcripts were isolated during an HIV-1 infection in 2002<span class="citation"><sup><a href="#ref-briquetImmunolocalizationStudiesAntisense2002" role="doc-biblioref">505</a></sup></span>. The function of this protein is still unknown but it has been shown to have a strong evolutionary correlation with HIV-1 group M responsible for the pandemic<span class="citation"><sup><a href="#ref-cassanConcomitantEmergenceAntisense2016" role="doc-biblioref">506</a></sup></span>. Although this Asp protein is still a source of debate, it is under active research<span class="citation"><sup><a href="#ref-savoretPilotStudyHumoral2020" role="doc-biblioref">507</a></sup></span>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hivStructure"></span>
<img src="figures/HIV-Intro/HIV-structure.png" alt="**Structure and main components of a mature HIV-1 virion.** 
Structural proteins MA, CA, SU and TM are represented in Blue, functional enzymes RT, IN and PR in pink, RNA binding proteins Rev, Tat and NC in orange and accessory proteins Vif, Nef, Vpr and Vpu in green. Viral RNA is shown in yellow. The phospholipd membrane of the virion is shown in a light purple color. The p6 protein is not represented as it is largely unsctructured. Vpu is not believed to be present in the HIV virion.  
Figure adapted from PDB101 [@zardecki2022] ([PDB101.rcsb.org](https://PDB101.rcsb.org), *CC By 4.0 License*, detailed list of structures used available in Appendix \@ref(HIV-intro-appendix))." width="90%" />
<p class="caption">
Figure 5.2: <strong>Structure and main components of a mature HIV-1 virion.</strong>
Structural proteins MA, CA, SU and TM are represented in Blue, functional enzymes RT, IN and PR in pink, RNA binding proteins Rev, Tat and NC in orange and accessory proteins Vif, Nef, Vpr and Vpu in green. Viral RNA is shown in yellow. The phospholipd membrane of the virion is shown in a light purple color. The p6 protein is not represented as it is largely unsctructured. Vpu is not believed to be present in the HIV virion.<br />
Figure adapted from PDB101<span class="citation"><sup><a href="#ref-zardecki2022" role="doc-biblioref">508</a></sup></span> (<a href="https://PDB101.rcsb.org">PDB101.rcsb.org</a>, <em>CC By 4.0 License</em>, detailed list of structures used available in Appendix <a href="HIV-intro-appendix.html#HIV-intro-appendix">B</a>).
</p>
</div>
</div>
</div>
<div id="drug-resistance-in-hiv" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Drug resistance in HIV<a href="viruses-hiv-and-drug-resistance.html#drug-resistance-in-hiv" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Although the HIV/AIDS pandemic has been very deadly around the world, we are not completely defenseless against it. The first antiretroviral therapy (ART) drugs were made available in the late eighties, only a couple years after discovery of the virus. ART reduce the viral load in an HIV positive patient reducing its transmissibility<span class="citation"><sup><a href="#ref-eisingerHIVViralLoad2019" role="doc-biblioref">509</a></sup></span>, while ART is not a cure for an HIV infection it has been shown to drastically reduce mortality and morbidity<span class="citation"><sup><a href="#ref-palellaDecliningMorbidityMortality1998" role="doc-biblioref">510</a></sup></span>. ART is estimated to have saved the lives of 9.5 million individuals between 1995 and 2015<span class="citation"><sup><a href="#ref-forsytheTwentyYearsAntiretroviral2019" role="doc-biblioref">511</a></sup></span>.</p>
<div id="a-quick-history-of-art" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> A quick history of ART<a href="viruses-hiv-and-drug-resistance.html#a-quick-history-of-art" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The first available anti-HIV drug was Zidovudine (ZDV, also known as azidothymidine or AZT) approved by the FDA for usage in the USA in 1987<span class="citation"><sup><a href="#ref-fischlEfficacyAzidothymidineAZT1987" role="doc-biblioref">512</a></sup></span>, a few years only after the discovery of the virus. This drug was a reverse transcriptase inhibitor (RTI) therefore preventing the viral RNA from being transcribed into viral DNA. Unfortunately, 3 years later strains of HIV resistant to ZDV were circulating<span class="citation"><sup><a href="#ref-richmanSusceptibilityNucleosideAnalogues1990" role="doc-biblioref">513</a></sup></span>. This rapid emergence of resistance to treatment is common for HIV<span class="citation"><sup><a href="#ref-yeoDeterminationHIV1RT2020" role="doc-biblioref">514</a></sup></span> due to its very high evolution rate<span class="citation"><sup><a href="#ref-cuevasExtremelyHighMutation2015" role="doc-biblioref">515</a></sup></span> allowing it to explore many possible mutations in response to selective pressures. To counter this resistance new drugs were rapidly developed and, between 1988 and 1995, four more RTIs were approved by the FDA. Using a combination of these drugs was also shown to be effective and led to a slower rise of resistance<span class="citation"><sup><a href="#ref-gulickTreatmentIndinavirZidovudine1997" role="doc-biblioref">516</a></sup></span>.</p>
<p>Then focus was shifted to the development of a new type of drug: Protease Inhibitors (PI). Between 1995 and 1997 4 of them were approved. These, taken in combination with RTI made if harder for the virus to develop resistance<span class="citation"><sup><a href="#ref-wensingFifteenYearsHIV2010" role="doc-biblioref">517</a></sup></span>. A new class of RTIs was also explored, Non-Nucleoside RTIs (NNRTIs) that block the RT action in another manner than the previously approved Nucleoside RTIs (NRTIs). When taken in combination with other drugs they are also highly effective<span class="citation"><sup><a href="#ref-pedersenNonNucleosideReverseTranscriptase1999" role="doc-biblioref">518</a></sup></span>. As the years advanced even more drug targets were explored, with 5 Integrase inhibitors (INSTI) being approved since 2007<span class="citation"><sup><a href="#ref-scarsiHIV1IntegraseInhibitors2020" role="doc-biblioref">519</a></sup></span>, A Fusion Inhibitor (FI) in 2003<span class="citation"><sup><a href="#ref-fletcherEnfuvirtideNewDrug2003" role="doc-biblioref">520</a></sup></span>, and 3 other Entry inhibitors (EI)<span class="citation"><sup><a href="#ref-esteHIVEntryInhibitors2007" role="doc-biblioref">521</a>,<a href="#ref-kilbyNovelTherapiesBased2003" role="doc-biblioref">522</a></sup></span> since 2007 all targeting different steps in the replication cycle of HIV <em>(see Table</em> <a href="HIV-intro-appendix.html#tab:tableDrugs">B.1</a> and Figure <a href="viruses-hiv-and-drug-resistance.html#fig:timeline">5.3</a><em>)</em>.</p>
<p>In response to the rapid emergence of resistance in HIV when treated with a single drug, clinicians started systematically treating HIV with a combination of multiple drugs targeting different proteins, as early as 1996. This is now referred to as Highly active antiretroviral combination therapy (HAART, also known as tritherapy). HAART usually consists of 2 NRTIs coupled with another drug: NNRTI or PI at first and later FI or INSTI<span class="citation"><sup><a href="#ref-yeniUpdateHAARTHIV2006" role="doc-biblioref">523</a></sup></span>. As of 2008 22 anti-HIV single drugs were approved by the FDA<span class="citation"><sup><a href="#ref-palmisanoBriefHistoryAntiretroviral2011" role="doc-biblioref">524</a></sup></span>, and 27 as of today. This large array of available drugs made HAART possible and gave options to clinicians to switch targets when the multi-resistant HIV emerged.</p>
<p>With the advent of HAART, patients had access to more potent treatments, however complexity of the treatment regimen grew. Treatments often involved several pills a day to take at precise intervals. Complex drug regimens have been associated with poorer treatment adherence<span class="citation"><sup><a href="#ref-mehtaPotentialFactorsAffecting1997" role="doc-biblioref">525</a>,<a href="#ref-millerComplianceTreatmentRegimens1997" role="doc-biblioref">526</a></sup></span>, which can lead to poor treatment outcome as well as the emergence of multi-resistant HIV strains<span class="citation"><sup><a href="#ref-chesneyAdherenceHIVCombination2000" role="doc-biblioref">527</a></sup></span> and its spread within the population. To avoid this issue, increasingly more single pill regimens are being developed with a staggering 7 new drugs approved by the FDA in 2018. These single pill regimens greatly reduce the burden of adherence for patients leading to better therapeutic outcomes and reduced healthcare costs<span class="citation"><sup><a href="#ref-aldirSingletabletRegimensHIV2014" role="doc-biblioref">528</a></sup></span>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:timeline"></span>
<img src="figures/HIV-Intro/complete.png" alt="**Timeline of ART drugs FDA approvals.**
Colored by drug type: Nucleoside Reverse transcriptase inhibitors (NRTI), Non-Nucleoside Reverse transcriptase inhibitors (NNRTI), Protease Inhibitors (PI), Integrase inhibitors (INSTI), Entry Inhibitors (EI) and pharmacokinetic enhancers (PE). Fixed Dose Combination (FDC) single pill regimens are also shown.  
* RPV is often also used as a pharmacokinetic enhancer in combination with other drugs.  
✝ These drugs are no longer approved by the FDA or no longer recommended as first line regiment treatment.  
*Information collected from &lt;https://hivinfo.nih.gov/understanding-hiv/fact-sheets/fda-approved-hiv-medicines&gt;, &lt;https://hivinfo.nih.gov/understanding-hiv/infographics/fda-approval-hiv-medicines&gt; and &lt;https://www.accessdata.fda.gov/scripts/cder/daf/index.cfm&gt;.*  
See also Table \@ref(tab:tableDrugs)." width="100%" />
<p class="caption">
Figure 5.3: <strong>Timeline of ART drugs FDA approvals.</strong>
Colored by drug type: Nucleoside Reverse transcriptase inhibitors (NRTI), Non-Nucleoside Reverse transcriptase inhibitors (NNRTI), Protease Inhibitors (PI), Integrase inhibitors (INSTI), Entry Inhibitors (EI) and pharmacokinetic enhancers (PE). Fixed Dose Combination (FDC) single pill regimens are also shown.<br />
* RPV is often also used as a pharmacokinetic enhancer in combination with other drugs.<br />
✝ These drugs are no longer approved by the FDA or no longer recommended as first line regiment treatment.<br />
<em>Information collected from <a href="https://hivinfo.nih.gov/understanding-hiv/fact-sheets/fda-approved-hiv-medicines" class="uri">https://hivinfo.nih.gov/understanding-hiv/fact-sheets/fda-approved-hiv-medicines</a>, <a href="https://hivinfo.nih.gov/understanding-hiv/infographics/fda-approval-hiv-medicines" class="uri">https://hivinfo.nih.gov/understanding-hiv/infographics/fda-approval-hiv-medicines</a> and <a href="https://www.accessdata.fda.gov/scripts/cder/daf/index.cfm" class="uri">https://www.accessdata.fda.gov/scripts/cder/daf/index.cfm</a>.</em><br />
See also Table <a href="HIV-intro-appendix.html#tab:tableDrugs">B.1</a>.
</p>
</div>
<p>Most recently, some studies explored using some of these single pill regimens (such as Truvada, c.f. Table <a href="HIV-intro-appendix.html#tab:tableDrugs">B.1</a>) as prophylactics <em>i.e.</em> Pre-exposure prophylaxis (PrEP). Putting uninfected but at risk populations on ART before any known exposure has been shown to effectively lower the risk of infection<span class="citation"><sup><a href="#ref-grantPreexposureChemoprophylaxisHIV2010" role="doc-biblioref">529</a>–<a href="#ref-buchbinderPreexposureProphylaxisPromise2011" role="doc-biblioref">531</a></sup></span>. When adherence is maintained, this risk reduction has been estimated to be between 44% and 100%<span class="citation"><sup><a href="#ref-riddellHIVPreexposureProphylaxis2018" role="doc-biblioref">532</a></sup></span>. As of 2022, Truvada is the only authorized drug for PrEP in Europe<span class="citation"><sup><a href="#ref-emaTruvada2018" role="doc-biblioref">533</a></sup></span>, although Descovy and Apretude are also authorized for PrEP in the USA<span class="citation"><sup><a href="#ref-PrEPPrEPHIV2022" role="doc-biblioref">534</a></sup></span>.</p>
<p>All of these drugs are widely used and are by now very well studied, therefore detailed guidelines on all the aspects of ART; when to start, which drugs to use, when to change drugs; are issued and updated regularly by practitioners<span class="citation"><sup><a href="#ref-zolopaEvolutionHIVTreatment2010" role="doc-biblioref">535</a></sup></span> and global organisms<span class="citation"><sup><a href="#ref-worldhealthorganizationConsolidatedGuidelinesHIV2021" role="doc-biblioref">536</a></sup></span> alike.</p>
</div>
<div id="drug-mechanisms" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Main mechanisms of viral proteins, antiretroviral drugs and associated resistance.<a href="viruses-hiv-and-drug-resistance.html#drug-mechanisms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Each ART drug targets a specific protein, most of them target one of the three <em>pol</em> proteins: RT, PR and IN. The structure of these proteins is inherently linked to their function, and as such is essential to take into account when developing ART. Similarly the structure of these proteins is very important when studying the resistance mechanisms developed by the virus<span class="citation"><sup><a href="#ref-ammaranondMechanismHIVAntiretroviral2012" role="doc-biblioref">537</a>,<a href="#ref-clavelHIVDrugResistance2004" role="doc-biblioref">538</a></sup></span>. In this section we will go over; for RT, IN and PR; the main structural elements and how they relate to treatment and resistance.</p>
<div id="reverse-transcriptase" class="section level4 hasAnchor" number="5.3.2.1">
<h4><span class="header-section-number">5.3.2.1</span> Reverse Transcriptase<a href="viruses-hiv-and-drug-resistance.html#reverse-transcriptase" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The reverse transcriptase protein is the most targeted protein in number of ART drugs <em>(c.f. Figure</em> <a href="viruses-hiv-and-drug-resistance.html#fig:timeline">5.3</a> <em>and Table</em> <a href="HIV-intro-appendix.html#tab:tableDrugs">B.1</a><em>)</em>. The mature protein is formed of two subunits: p51 and p66. These two subunits are translated from the same section of the <em>pol</em> gene, and such have the same amino acid sequence, but p51 is cleaved and as such is shorter than p66. The p66 subunit contains the active sites of RT whereas p51 plays a mainly structural role.</p>
<p>The p66 sububit can be separated into 5 domains<span class="citation"><sup><a href="#ref-sarafianosStructureFunctionHIV12009" role="doc-biblioref">539</a></sup></span>. The “fingers”, “palm”, and “thumb” domains are linked together and folded to form a canal through which the RNA template and newly synthesized viral DNA can pass through. The polymerase active site, responsible for incorporating nucleotides to the viral DNA molecule, is situated in the “palm” domain at the bottom of the canal. The “RNase” domain of RT contains a secondary active site responsible for cleaving the viral RNA template from the viral DNA so that the RT can fill out the complementary strand of viral DNA before integration into the host genome. The final “connection” domain is simply a link between the “RNase” and the “thumb” domains. A three dimensional view of RT with these domains highlighted can be seen in Figure <a href="viruses-hiv-and-drug-resistance.html#fig:rtStruct">5.4</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:rtStruct"></span>
<img src="figures/HIV-Intro/rt.png" alt="**3D structure of HIV-1 Reverse-transcriptase.** The different domains of the p66 subunit are labeled and shown in different shades of blue and green. The structural p51 subunit is shown in orange. The RNA template is shown in dark gray and the newly synthesized DNA stran in light gray. The polymerase active site is shown in red although mostly hidden by the RNA template. The 3D visualization was produced with Illustrate [@goodsellIllustrateSoftwareBiomolecular2019] using the [2hmi](https://www.rcsb.org/structure/2HMI) PDB structure." width="80%" />
<p class="caption">
Figure 5.4: <strong>3D structure of HIV-1 Reverse-transcriptase.</strong> The different domains of the p66 subunit are labeled and shown in different shades of blue and green. The structural p51 subunit is shown in orange. The RNA template is shown in dark gray and the newly synthesized DNA stran in light gray. The polymerase active site is shown in red although mostly hidden by the RNA template. The 3D visualization was produced with Illustrate<span class="citation"><sup><a href="#ref-goodsellIllustrateSoftwareBiomolecular2019" role="doc-biblioref">540</a></sup></span> using the <a href="https://www.rcsb.org/structure/2HMI">2hmi</a> PDB structure.
</p>
</div>
<p>Reverse Transcriptase inhibitors can be separated into two classes: Nucleoside RTIs (NRTIs) and Non-Nucleoside RTIs (NNRTIs), inhibiting the action of RT in two disctinct manners:</p>
<ul>
<li><p>NRTIs are analogues of free nucleotides in the host cell. They competitively inhibit RT and can be used to elongate the viral DNA chain. Once an NRTI is incorporated, further elongation of the DNA molecule is impossible and the viral DNA cannot be synthesized anymore. This is similar to the chain terminating nucleotides introduced in Section <a href="what-is-sequence-data.html#obtaining-sequence-data">1.2</a>.</p></li>
<li><p>NNRTIs bind to a specific region of the p51 subunit: the Non Nucleoside Inhibitor Binding Pocket (NNIBP) <em>(A view of RT with the NNIBP visible is shown in Figure</em> <a href="HIV-paper.html#fig:figStructure">6.4</a><em>)</em>. This pocket, although it is on the p51 subunit is spatially situated very close to the polymerase active site. NNRTIs bind to the NNIBP to change the conformation of the active site, lowering its flexibility<span class="citation"><sup><a href="#ref-esnoufUniqueFeaturesStructure1997" role="doc-biblioref">541</a></sup></span>, and thusly non-competitively inhibiting the action of RT.</p></li>
</ul>
<p>Research has been conducted into inhibition of the RNase active site of RT<span class="citation"><sup><a href="#ref-hangActivityIsolatedHIV2004" role="doc-biblioref">542</a>,<a href="#ref-klumppRecentProgressDesign2006" role="doc-biblioref">543</a></sup></span> which could also inhibit the action of RT. There is however, to this day, no approved treatment that inhibits the RNase action of RT.</p>
<p>Drug resistance mutations (DRMs) that arise in HIV from the selective pressures resulting from RTI exposure can similarly be grouped into two categories: NRTI and NNRTI resistance mutations.</p>
<p>NRTI resistance mutations can further be subcategorized into two groups<span class="citation"><sup><a href="#ref-menendez-ariasMechanismsResistanceNucleoside2008" role="doc-biblioref">544</a>,<a href="#ref-sluis-cremerMolecularMechanismsHIV12000" role="doc-biblioref">545</a></sup></span>. The first type of NRTI resistance mutations are mutations that prevent the incorporation of NRTIs into the viral DNA molecule. M184V and M184I, indicating the replacement, at site number 184, of a Methionine by a Valine or an Isoleucine respectively, are very common NRTI resistance mutations. These V and I amino acids have a different structure than the original M, interfering with the incorporation of lamiduvine (3TC) but not dNTP<span class="citation"><sup><a href="#ref-sarafianosLamivudine3TCResistance1999" role="doc-biblioref">546</a></sup></span>. The second type of mutation, allows RT to remove an incorporated NRTI from the viral DNA to resume synthesis. Thymidine Analog Mutations (TAMs), M41L, D67N, K70R, L210W, T215Y/F and K219Q/E confer resistance to azidothymidine (AZT) through this mechanism<span class="citation"><sup><a href="#ref-meyerMechanismAZTResistance1999" role="doc-biblioref">547</a>,<a href="#ref-boyerSelectiveExcisionAZTMP2001" role="doc-biblioref">548</a></sup></span>.</p>
<p>Similarly, NNRTI resistance mutations work via several different mechanisms<span class="citation"><sup><a href="#ref-deeksNonnucleosideReverseTranscriptase2001" role="doc-biblioref">549</a>,<a href="#ref-renStructuralBasisDrug2008" role="doc-biblioref">550</a></sup></span>. Some NNRTI resistance mutations, like Y181C, lower the affinity of the NNIBP to NNRTIs preventing binding of drugs to RT. Others, like K103N change the conformation of the p51 subunit, making the NNIBP disappear. NNRTI resistance mutations are particularly dangerous because they often confer cross-resistance to multiple NNRTIs without affecting the polymerase action very much<span class="citation"><sup><a href="#ref-ammaranondMechanismHIVAntiretroviral2012" role="doc-biblioref">537</a></sup></span>, giving rise to viruses that are both fit and highly resistant. This is contrast to NRTI resistance mutations that generally incur a fitness cost for the virus, lowering its efficacy<span class="citation"><sup><a href="#ref-lloydHighCostFidelity2014" role="doc-biblioref">551</a></sup></span>.</p>
</div>
<div id="protease" class="section level4 hasAnchor" number="5.3.2.2">
<h4><span class="header-section-number">5.3.2.2</span> Protease<a href="viruses-hiv-and-drug-resistance.html#protease" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The Protease protein, also a major drug target for ART, cleaves the <em>gag</em> and <em>pol</em> polyproteins in order to produce functional viral proteins, essential to replication. It has a symmetric dimeric structure, that is two say that it is composed of two identical chains of amino acids<span class="citation"><sup><a href="#ref-pearlStructuralModelRetroviral1987" role="doc-biblioref">552</a>,<a href="#ref-gulnikHIVProteaseEnzyme2000" role="doc-biblioref">553</a></sup></span>. A structural view of PR is shown in Figure <a href="viruses-hiv-and-drug-resistance.html#fig:prStruct">5.5</a>.</p>
<p>These two chains are folded in order to create a “tunnel” through which the polyproteins enter. In the middle of this “tunnel”, at the bottom, is the active site. The active site is composed of two Aspartate residues, one on each chain, with water they can participate in a chemical reaction that cleaves the polyprotein at a specific position<span class="citation"><sup><a href="#ref-silvaInhibitionCatalyticMechanism1996" role="doc-biblioref">554</a></sup></span>.</p>
<p>The roof of the “tunnel” is formed by the flaps, a flexible region from each of the two chains that can open or close the “tunnel”<span class="citation"><sup><a href="#ref-hornakHIV1ProteaseFlaps2006" role="doc-biblioref">555</a></sup></span>. These flaps most likely control the access of polyproteins to the active site<span class="citation"><sup><a href="#ref-freedbergRapidStructuralFluctuations2002" role="doc-biblioref">556</a>,<a href="#ref-yuStructuralInsightsHIV12017" role="doc-biblioref">557</a></sup></span>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:prStruct"></span>
<img src="figures/HIV-Intro/pr.png" alt="**3D structure of HIV-1 Protease.** The two identical chains are colored in orange and blue shades respectively. The flexible, flaps form the the &quot;roof&quot; of a tunnel, at the bottom of which is the active site: 2 Asp residues, one on each chain. The 3D visualization was produced with Illustrate [@goodsellIllustrateSoftwareBiomolecular2019] using the [2p3b](https://www.rcsb.org/structure/2P3B) PDB structure." width="70%" />
<p class="caption">
Figure 5.5: <strong>3D structure of HIV-1 Protease.</strong> The two identical chains are colored in orange and blue shades respectively. The flexible, flaps form the the “roof” of a tunnel, at the bottom of which is the active site: 2 Asp residues, one on each chain. The 3D visualization was produced with Illustrate<span class="citation"><sup><a href="#ref-goodsellIllustrateSoftwareBiomolecular2019" role="doc-biblioref">540</a></sup></span> using the <a href="https://www.rcsb.org/structure/2P3B">2p3b</a> PDB structure.
</p>
</div>
<p>All the approve Protease Inhibitors (PIs) share a similar mode of action, each PI binds to the active site of the PR, denying access to the “tunnel” for polyproteins and stopping the catalytic action of PR<span class="citation"><sup><a href="#ref-robertsRationalDesignPeptideBased1990" role="doc-biblioref">558</a>,<a href="#ref-lvHIVProteaseInhibitors2015" role="doc-biblioref">559</a></sup></span>. Tipranavir, one of the more recent PIs, also binds with the flaps<span class="citation"><sup><a href="#ref-lvHIVProteaseInhibitors2015" role="doc-biblioref">559</a></sup></span>.</p>
<p>According to Prabu-Jeyabalan and colleagues, the Protease structure does not recognize the specific sequence of the polyprotein cleavage site but rather its shape<span class="citation"><sup><a href="#ref-prabu-jeyabalanSubstrateShapeDetermines2002" role="doc-biblioref">560</a></sup></span>. They proposed an inhibitor based on this shape for all polyproteins combined, which establishes more bonds with PR, making it suposedly more efficient<span class="citation"><sup><a href="#ref-prabu-jeyabalanSubstrateEnvelopeDrug2006" role="doc-biblioref">561</a></sup></span> than current approved PIs.</p>
<p>As is the case with RTIs, when under selective pressure due to PIs, the virus tends to develop PI associated DRMs. Most PI resistance mutations result in an enlarged “tunnel”, this tends to lower the affinity of the PIs to the active site, but also the affinity of polyproteins, lowering the fitness of the virus significantly<span class="citation"><sup><a href="#ref-wensingFifteenYearsHIV2010" role="doc-biblioref">517</a></sup></span>. In addition, some mutations on the <em>gag</em> polyprotein seem to lower the efficacy of PIs, although the underlying mechanism is not well known<span class="citation"><sup><a href="#ref-wensingFifteenYearsHIV2010" role="doc-biblioref">517</a></sup></span>.</p>
<p>Some mutations in the flaps of PR have also been shown to confer PI resistance. It seems likely that these mutations change conformation of the flaps, opening them leading to the release of inhibitors from the active site<span class="citation"><sup><a href="#ref-kurtyilmazImprovingViralProtease2016" role="doc-biblioref">562</a></sup></span>.</p>
</div>
<div id="integrase" class="section level4 hasAnchor" number="5.3.2.3">
<h4><span class="header-section-number">5.3.2.3</span> Integrase<a href="viruses-hiv-and-drug-resistance.html#integrase" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The integrase protein is the third major anti-retroviral drug target, it is responsible for integrating the viral DNA into the host genome. It is a tetramer composed of four identical amino acid chains.<span class="citation"><sup><a href="#ref-chiuStructureFunctionHIV12004" role="doc-biblioref">563</a>,<a href="#ref-espositoHIVIntegraseStructure1999" role="doc-biblioref">564</a></sup></span>. Each of these chains contain three domains linked together by flexible linker sequences: the N-terminal domain, the catalytic core and the C-terminal domain. In each tetramer, two chains provide the active site for the integration reaction while the other two have a mostly structural role. It is probable that the N-terminal domain which is very conserved is necessary for stable tetramerization of IN monomers<span class="citation"><sup><a href="#ref-delelisIntegraseIntegrationBiochemical2008" role="doc-biblioref">565</a></sup></span>, this tetrameric structure is shown in Figure <a href="viruses-hiv-and-drug-resistance.html#fig:inStruct">5.6</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:inStruct"></span>
<img src="figures/HIV-Intro/in.png" alt="**3D structure of an Integrase.** This Integrase tetramer is binded with viral (red) and host (orange) DNA, linked to the two light blue functional subunits via the C-terminal domain. The active site formed by the the catalytic cores of the two functional subunits *(not visible in this representation)*, is where the strand transfer reaction will take place. The two dark blue IN subunits have a structural role. 
This figure was adapted from the PDB 101 molecule of the month Integrase entry by David S. Goodsell and the RCSB PDB ([pdb101.rcsb.org/motm/135](https://pdb101.rcsb.org/motm/135)) with a CC By 4.0 license. " width="95%" />
<p class="caption">
Figure 5.6: <strong>3D structure of an Integrase.</strong> This Integrase tetramer is binded with viral (red) and host (orange) DNA, linked to the two light blue functional subunits via the C-terminal domain. The active site formed by the the catalytic cores of the two functional subunits <em>(not visible in this representation)</em>, is where the strand transfer reaction will take place. The two dark blue IN subunits have a structural role.
This figure was adapted from the PDB 101 molecule of the month Integrase entry by David S. Goodsell and the RCSB PDB (<a href="https://pdb101.rcsb.org/motm/135">pdb101.rcsb.org/motm/135</a>) with a CC By 4.0 license.
</p>
</div>
<p>Several steps are needed in order to integrate the viral DNA with the host genome<span class="citation"><sup><a href="#ref-maertensStructureFunctionRetroviral2022" role="doc-biblioref">566</a></sup></span>. First, IN binds to the both ends of the viral DNA, using the C-terminal domains, forming a closed loop. Secondly, both ends of the viral DNA molecule are then prepared for integration by the catalytic core. Third, the host DNA is captured with C-terminal domains. Then, the strand-transfer is done within the catalytic core: the host DNA is cut in two places and a single strand from each end of the viral DNA are attached to these two breakpoints. Finally, the IN tetramer detaches from the linked molecules and the final steps necessary to create a single hybrid DNA molecule are done by the host cellular machinery. A graphical representation of this process can be found in Figure 1 of the 2022 article by Maertens et al.<span class="citation"><sup><a href="#ref-maertensStructureFunctionRetroviral2022" role="doc-biblioref">566</a></sup></span>.</p>
<p>Integrase Strand Transfer Inhibitors (INSTIs), as their name indicates, block the strand transfer reaction. They achieve this by strongly binding to the active site of the IN tetramer after it has formed a complex with the viral DNA<span class="citation"><sup><a href="#ref-maertensStructureFunctionRetroviral2022" role="doc-biblioref">566</a>,<a href="#ref-pommierIntegraseInhibitorsTreat2005" role="doc-biblioref">567</a></sup></span>. In doing so, INSTIs prevent the IN / viral DNA complex from binding to the host DNA, effectively preventing strand transfer.</p>
<p>In the presence of INSTIs during therapy, once more the HIV virus develops resistance mutations over time. These mutations all lower affinity of IN to INSTIs, preventing bonding<span class="citation"><sup><a href="#ref-maertensStructureFunctionRetroviral2022" role="doc-biblioref">566</a>,<a href="#ref-blancoHIV1IntegraseInhibitor2011" role="doc-biblioref">568</a></sup></span>. Since most INSTIs behave similarly, this means that cross-resistance to INSTIs is quite common for INSTI DRMs<span class="citation"><sup><a href="#ref-blancoHIV1IntegraseInhibitor2011" role="doc-biblioref">568</a>,<a href="#ref-gerettiEmergingPatternsImplications2012" role="doc-biblioref">569</a></sup></span>. Once more, these mutations tend to lower the overall viral fitness necessitating secondary compensatory mutations to restore fitness<span class="citation"><sup><a href="#ref-blancoHIV1IntegraseInhibitor2011" role="doc-biblioref">568</a>,<a href="#ref-gerettiEmergingPatternsImplications2012" role="doc-biblioref">569</a></sup></span>.</p>
</div>
<div id="other-drug-targets" class="section level4 hasAnchor" number="5.3.2.4">
<h4><span class="header-section-number">5.3.2.4</span> Other drug targets<a href="viruses-hiv-and-drug-resistance.html#other-drug-targets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For now, resistance has not been observed for novel drugs like entry inhibitors. This might be because the genetic barrier to resistance is higher and because not enough time has passed since their introduction for resistance to emerge.</p>
<p>For all the other drug targets however, as stated earlier in this section, resistance is documented and problematic. Resistance has even been detected for PrEP which is prophylactic<span class="citation"><sup><a href="#ref-knoxMultidrugResistantHIV1Infection2017" role="doc-biblioref">570</a>,<a href="#ref-hurtPreexposureProphylaxisAntiretroviral2011" role="doc-biblioref">571</a></sup></span>. This seems to be rare however and mostly due to an unknown pre-treatment HIV infection<span class="citation"><sup><a href="#ref-gibasDrugResistanceHIV2019" role="doc-biblioref">572</a></sup></span>.</p>
</div>
</div>
<div id="consequences-of-resistance-on-global-health" class="section level3 hasAnchor" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> Consequences of resistance on global health<a href="viruses-hiv-and-drug-resistance.html#consequences-of-resistance-on-global-health" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>HIV resistance to ART drugs is problematic from a global health perspective. Indeed circulation of resistant strains of HIV within populations can lead to treatment-naive individuals that will not respond well to treatment.</p>
<p>More concerning is the fact that transmission of resistant strains of HIV between treatment-naive individuals is the main mode of resistance transmission in the UK<span class="citation"><sup><a href="#ref-mouradPhylotypebasedAnalysisHighlights2015" role="doc-biblioref">573</a>,<a href="#ref-hueDemonstrationSustainedDrugResistant2009" role="doc-biblioref">574</a></sup></span> and Switzerland<span class="citation"><sup><a href="#ref-drescherTreatmentNaiveIndividualsAre2014" role="doc-biblioref">575</a></sup></span>. This treatment-naive to treatment-naive transmission is particularly insidious since it can go undetected and creates long lasting drug resistant strain reservoirs in the treatment-naive population. This of course is dangerous since some infected individuals might experience poor therapeutic outcomes and even failure when administered first line regimens<span class="citation"><sup><a href="#ref-boermaHighLevelsPretreatment2016" role="doc-biblioref">576</a></sup></span>. To avoid this, genotypic resistance testing has become standard practice when choosing the therapeutic strategy in high-income countries, but more effort must be done to make resistance testing less expensive and more cost-efficient in lower and middle income countries<span class="citation"><sup><a href="#ref-clutterHIV1DrugResistance2016" role="doc-biblioref">577</a></sup></span>.</p>
<p>Although the transmitted drug resistance described above is problematic, a large portion of DRMs incur a fitness cost for the resistant strain<span class="citation"><sup><a href="#ref-kuhnertQuantifyingFitnessCost2018" role="doc-biblioref">578</a>,<a href="#ref-mespledeViralFitnessCost2013" role="doc-biblioref">579</a></sup></span>. This means that although they are selected when exposed to the evolutionary pressure of ART, when the treatment is interrupted there is another pressure leading these costly mutations to disappear. This reversion is commonly observed after interruption of treatment, however the median reversion times vary widely from 1 to 13 years<span class="citation"><sup><a href="#ref-castroPersistenceHIV1Transmitted2013" role="doc-biblioref">580</a></sup></span> depending on the severity of the fitness loss and type of mutation. This means that although reversion can possibly lead to loss of resistance this can potentially take a long time and possibly longer than the treatment interruption.</p>
<p>In practice it is therefore very important to keep an eye on all drug resistance mutations, their population dynamics and spread as well as their presence or absence in a particular strain before starting treatment.</p>
</div>
<div id="finding-drms" class="section level3 hasAnchor" number="5.3.4">
<h3><span class="header-section-number">5.3.4</span> Finding DRMS <a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a><a href="viruses-hiv-and-drug-resistance.html#finding-drms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finding and categorizing mutations as DRMs is an important task in light of the public health implications mentioned in Section <a href="viruses-hiv-and-drug-resistance.html#consequences-of-resistance-on-global-health">5.3.3</a>, as such this is an active part of the HIV research field.</p>
<p>The most important thing needed in order to study DRMs is of course viral sequences. To facilitate the search for DRMs several sequence databases exist. Sequences are often linked to metadata related to the treatment status of the patient from which the sequence was obtained. This metadata can be quite variable: from a coarse level binary indicator of treatment to a finely detailed list of all treatments recieved and associated phenotypic measurements like viral load.</p>
<p>Databases like the UK-CHIC<span class="citation"><sup><a href="#ref-ukchicsteeringcommitteeCreationLargeUKbased2004" role="doc-biblioref">582</a></sup></span>, uk HIV drug resistance database (<a href="https://www.hivrdb.org.uk/" class="uri">https://www.hivrdb.org.uk/</a>) and Swiss cohort study (<a href="https://www.shcs.ch/" class="uri">https://www.shcs.ch/</a>) host sequences on a national level, although access can be granted to international researchers. Other databases like the PANGEA database<span class="citation"><sup><a href="#ref-abeler-dornerPANGEAHIVPhylogeneticsNetworks2019" role="doc-biblioref">583</a></sup></span> host sequences from multiple countries in sub-Saharan Africa. The Stanford HIV drug resistance database (<a href="https://hivdb.stanford.edu/" class="uri">https://hivdb.stanford.edu/</a>) hosts HIV sequences with some phenotypic data<span class="citation"><sup><a href="#ref-rheeHumanImmunodeficiencyVirus2003" role="doc-biblioref">32</a>,<a href="#ref-shaferRationaleUsesPublic2006" role="doc-biblioref">584</a></sup></span>. Finally some database only host sequences, such as the Los Alamos HIV sequence database (<a href="http://www.hiv.lanl.gov/" class="uri">http://www.hiv.lanl.gov/</a>), however without any specific treatment or resistance related metadata<span class="citation"><sup><a href="#ref-kuikenHIVSequenceDatabases2003" role="doc-biblioref">585</a></sup></span> these have less direct applicability to the DRM search task.</p>
<p>Some databases, like the Standford HIV resistance database, also store specific knowledge about known resistance mutations, keeping and regularly updating lists of clinically important DRMs as well as their impact on ART<span class="citation"><sup><a href="#ref-wensing2019UpdateDrug2019" role="doc-biblioref">586</a>,<a href="#ref-clarkMutationsRetroviralGenes2007" role="doc-biblioref">587</a></sup></span>. Additionaly, Stanford also offers tools for clinicians to do genotypic resistance testing with interpretable results<span class="citation"><sup><a href="#ref-liuWebResourcesHIV2006d" role="doc-biblioref">588</a></sup></span>.</p>
<p>The first step of mutations discovery is usually some kind of statistical association analysis<span class="citation"><sup><a href="#ref-wensing2019UpdateDrug2019" role="doc-biblioref">586</a>,<a href="#ref-johnsonUpdateDrugResistance2016" role="doc-biblioref">589</a></sup></span> where the association between treatment status (coarse of fine grained) and specific mutations is statistically tested. This is usually done with Fisher association tests<span class="citation"><sup><a href="#ref-villabona-arenasIndepthAnalysisHIV12016" role="doc-biblioref">590</a>,<a href="#ref-shulmanGeneticCorrelatesEfavirenz2004" role="doc-biblioref">591</a></sup></span> or correlation testing with the Spearman correlation for example<span class="citation"><sup><a href="#ref-millerGenotypicPhenotypicPredictors2004" role="doc-biblioref">592</a></sup></span>. This results in a list of mutations that are significantly associated with a given treatment and corresponding p-values.</p>
<p>Since, on a given sequence dataset, several mutations are usually tested at once this can lead to inflated false positive<span class="citation"><sup><a href="#ref-brownMethodsCorrectingMultiple1997" role="doc-biblioref">593</a></sup></span> and spurious<span class="citation"><sup><a href="#ref-austinTestingMultipleStatistical2006" role="doc-biblioref">594</a></sup></span> associations. Fortunately this is a well studied problem and many methods exist to control this effect by controlling the Familywise Error Rate (FWER) <em>e.g.</em> with the Bonferroni procedure,<span class="citation"><sup><a href="#ref-hochbergMultipleComparisonProcedures1987" role="doc-biblioref">595</a></sup></span> or the False Discovery Rate (FDR) <em>e.g.</em> with the Benjamini-Hochberg procedure<span class="citation"><sup><a href="#ref-benjaminiControllingFalseDiscovery1995" role="doc-biblioref">596</a></sup></span>. These methods are often applied when testing for resistance association<span class="citation"><sup><a href="#ref-villabona-arenasIndepthAnalysisHIV12016" role="doc-biblioref">590</a>,<a href="#ref-gonzalesExtendedSpectrumHIV12003" role="doc-biblioref">597</a>,<a href="#ref-seoigheModelDirectionalSelection2007" role="doc-biblioref">598</a></sup></span>. However these correction methods are a double-edged sword, some of them can be very conservative and lead to falsely rejecting true associations<span class="citation"><sup><a href="#ref-shamStatisticalPowerSignificance2014" role="doc-biblioref">599</a></sup></span>. In some studies on resistance phylogenetic correlation between the sequences is also accounted for in statistical analyses<span class="citation"><sup><a href="#ref-alizonPhylogeneticApproachReveals2010" role="doc-biblioref">600</a>,<a href="#ref-flynnDeepSequencingProtease2015" role="doc-biblioref">601</a></sup></span>.</p>
<p>Statistical testing on treatment status, while informative, can only associate a mutation with a treatment. In order to actually validate whether a mutation causes resistance or not biological analyses are needed<span class="citation"><sup><a href="#ref-wensing2019UpdateDrug2019" role="doc-biblioref">586</a>,<a href="#ref-johnsonUpdateDrugResistance2016" role="doc-biblioref">589</a></sup></span>. The easiest of these are <em>in vitro</em> analyses where live viruses are subjected to a phenotypical assay. These assays measure the susceptibility of HIV viruses to a wide array of drugs, which can then be statistically associated with genetic traits like specific mutations. These assays like phenosense<span class="citation"><sup><a href="#ref-petropoulosNovelPhenotypicDrug2000" role="doc-biblioref">602</a></sup></span> or antivirogram<span class="citation"><sup><a href="#ref-hertogsRapidMethodSimultaneous1998" role="doc-biblioref">603</a></sup></span> are widely used<span class="citation"><sup><a href="#ref-heilek-snyderRoleHIVPhenotypic2002" role="doc-biblioref">604</a>–<a href="#ref-gartlandSusceptibilityGlobalHIV12021" role="doc-biblioref">606</a></sup></span>. The viruses can be obtained from clinical isolates<span class="citation"><sup><a href="#ref-masquelierGenotypicPhenotypicResistance2001" role="doc-biblioref">607</a></sup></span> or, viruses with specific mutations can be manufactured with site directed mutagenesis<span class="citation"><sup><a href="#ref-larderMultipleMutationsHIV11989" role="doc-biblioref">608</a>,<a href="#ref-devreeseResistanceHumanImmunodeficiency1992" role="doc-biblioref">609</a></sup></span>. <em>In vivo</em> can be conducted by sequencing viruses from patients failing ART, following over time and studying the association between their treatment response and HIV genetics<span class="citation"><sup><a href="#ref-tambuyzerEffectMutationsPosition2011" role="doc-biblioref">610</a>,<a href="#ref-katzensteinPhenotypicSusceptibilityVirological2003" role="doc-biblioref">611</a></sup></span>.</p>
<p>More recently, as sequence database grow bigger and bigger <em>(The UK-CHIC database contains more than 80,000 HIV sequences with treatment status)</em>, methods based on statistical and machine learning are being used to study resistance. Most approaches rely on training models to predict some type of resistance: either classifying sequences as resistant or not<span class="citation"><sup><a href="#ref-steinerDrugResistancePrediction2020a" role="doc-biblioref">327</a>,<a href="#ref-blasselUsingMachineLearning2021" role="doc-biblioref">612</a></sup></span> of predicting a phenotypic response like fold resistance compared to wild type<span class="citation"><sup><a href="#ref-sheikamamuddyImprovingFoldResistance2017b" role="doc-biblioref">613</a></sup></span>. Initial approaches were mainly designed for clinical testing, rather than new DRM search, and distributed via web services<span class="citation"><sup><a href="#ref-beerenwinkelGeno2phenoInterpretingGenotypic2001" role="doc-biblioref">614</a>,<a href="#ref-riemenschneiderSHIVAWebApplication2016b" role="doc-biblioref">615</a></sup></span>.</p>
<p>Initially these approaches were based on models like decision trees<span class="citation"><sup><a href="#ref-beerenwinkelDiversityComplexityHIV12002c" role="doc-biblioref">616</a></sup></span>, SVMs<span class="citation"><sup><a href="#ref-beerenwinkelGeno2phenoInterpretingGenotypic2001" role="doc-biblioref">614</a></sup></span> or logistic regression<span class="citation"><sup><a href="#ref-heiderMultilabelClassificationExploiting2013b" role="doc-biblioref">617</a></sup></span>. Over time the use of more complex models such as neural networks has increased, with increased prediction accuracy<span class="citation"><sup><a href="#ref-sheikamamuddyImprovingFoldResistance2017b" role="doc-biblioref">613</a></sup></span>.</p>
<p>By analyzing the important features used by trained models to predict resistance, it is possible to find features, corresponding to mutations, that are useful for predicting, and therefore likely associated with, drug resistance (see Chapter <a href="HIV-paper.html#HIV-paper">6</a>). With the improvement in methods to interpret and extract features from complex models such as deep neural networks, this approach has been used with deep learning models<span class="citation"><sup><a href="#ref-steinerDrugResistancePrediction2020a" role="doc-biblioref">327</a></sup></span>. This novel way of finding resistance associated mutations has the potential to uncover complex mutational effects that simple association testing cannot.</p>
</div>
</div>
<div id="conclusion-3" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Conclusion<a href="viruses-hiv-and-drug-resistance.html#conclusion-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Viruses are surprisingly complex in light of their apparent simplicity. They are ubiquitous and present an extreme diversity. Whether they are pathogenic or not, the role of viruses in a myriad of processes and niches make them interesting and important to study. The sequences of these viruses, although small can be very useful for evolutionary as well as clinical analyses.</p>
<p>Although the study of viruses as a whole can be very useful, HIV is particularly important to study. The impact of the HIV pandemic on global health has been severe, both in Lower and Higher income countries. It is therefore paramount to fully understand the underlying mechanisms and evolutionary adaptations of this virus. It’s high mutation rate allow it to quickly explore evolutionary alternatives when exposed to drugs, making anti HIV therapy a complex endeavor.</p>
<p>Fortunately, with large scale sequencing efforts it is possible to study and track these evolutionary adaptations to treatments. This allows us to adapt therapeutic strategies as well as developing new compounds and approaches. In this context, studying and finding the virus’ mutational processes is paramount. This is especially important when studying resistance to RTIs as they form the backbone of first line regimen combination therapies, and are the most common type of anti-HIV drug. This process is made easier by the large scale sequence repositories now available, and the usage of machine and statistical learning to leverage that data.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-rheeHumanImmunodeficiencyVirus2003" class="csl-entry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Rhee, S.-Y. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkg100">Human immunodeficiency virus reverse transcriptase and protease sequence database</a>. <em>Nucleic Acids Research</em> <strong>31</strong>, 298–303 (2003).</div>
</div>
<div id="ref-steinerDrugResistancePrediction2020a" class="csl-entry">
<div class="csl-left-margin">327. </div><div class="csl-right-inline">Steiner, M. C., Gibson, K. M. &amp; Crandall, K. A. <a href="https://doi.org/10.3390/v12050560">Drug <span>Resistance Prediction Using Deep Learning Techniques</span> on <span>HIV</span>-1 <span>Sequence Data</span></a>. <em>Viruses</em> <strong>12</strong>, 560 (2020).</div>
</div>
<div id="ref-lwoffConceptVirus1957" class="csl-entry">
<div class="csl-left-margin">402. </div><div class="csl-right-inline">Lwoff, A. <a href="https://doi.org/10.1099/00221287-17-2-239">The concept of virus</a>. <em>Journal of General Microbiology</em> <strong>17</strong>, 239–253 (1957).</div>
</div>
<div id="ref-minorViruses2014" class="csl-entry">
<div class="csl-left-margin">403. </div><div class="csl-right-inline">Minor, P. D. Viruses. in (John Wiley &amp; Sons, Ltd, 2014). doi:<a href="https://doi.org/10.1002/9780470015902.a0000441.pub3">10.1002/9780470015902.a0000441.pub3</a>.</div>
</div>
<div id="ref-stapletonGBVirusesReview2011" class="csl-entry">
<div class="csl-left-margin">404. </div><div class="csl-right-inline">Stapleton, J. T., Foung, S., Muerhoff, A. S., Bukh, J. &amp; Simmonds, P. <a href="https://doi.org/10.1099/vir.0.027490-0">The GB viruses: A review and proposed classification of GBV-a, GBV-c (HGV), and GBV-d in genus pegivirus within the family flaviviridae</a>. <em>The Journal of General Virology</em> <strong>92</strong>, 233–246 (2011).</div>
</div>
<div id="ref-yamamotoCharacterizationNonpathogenicH5N12011" class="csl-entry">
<div class="csl-left-margin">405. </div><div class="csl-right-inline">Yamamoto, N. <em>et al.</em> <a href="https://doi.org/10.1186/1743-422X-8-65">Characterization of a non-pathogenic H5N1 influenza virus isolated from a migratory duck flying from siberia in hokkaido, japan, in october 2009</a>. <em>Virology Journal</em> <strong>8</strong>, 65 (2011).</div>
</div>
<div id="ref-shiEvolutionaryHistoryVertebrate2018" class="csl-entry">
<div class="csl-left-margin">406. </div><div class="csl-right-inline">Shi, M. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-018-0012-7">The evolutionary history of vertebrate RNA viruses</a>. <em>Nature</em> <strong>556</strong>, 197–202 (2018).</div>
</div>
<div id="ref-adamsAtlasInvertebrateViruses2017" class="csl-entry">
<div class="csl-left-margin">407. </div><div class="csl-right-inline"><em>Atlas of invertebrate viruses</em>. (CRC Press, 2017). doi:<a href="https://doi.org/10.1201/9781315149929">10.1201/9781315149929</a>.</div>
</div>
<div id="ref-lefeuvreEvolutionEcologyPlant2019" class="csl-entry">
<div class="csl-left-margin">408. </div><div class="csl-right-inline">Lefeuvre, P. <em>et al.</em> <a href="https://doi.org/10.1038/s41579-019-0232-3">Evolution and ecology of plant viruses</a>. <em>Nature Reviews Microbiology</em> <strong>17</strong>, 632–644 (2019).</div>
</div>
<div id="ref-wangVirusesParasiticProtozoa1991" class="csl-entry">
<div class="csl-left-margin">409. </div><div class="csl-right-inline">Wang, A. L. &amp; Wang, C. C. <a href="https://doi.org/10.1016/0169-4758(91)90198-W">Viruses of parasitic protozoa</a>. <em>Parasitology Today</em> <strong>7</strong>, 76–80 (1991).</div>
</div>
<div id="ref-ferminVirusesProkaryotesProtozoa2018" class="csl-entry">
<div class="csl-left-margin">410. </div><div class="csl-right-inline">Fermin, G., Mazumdar-Leighton, S. &amp; Tennant, P. Viruses of prokaryotes, protozoa, fungi, and chromista. in 217 (2018).</div>
</div>
<div id="ref-sutelaVirusesFungiOomycetes2019" class="csl-entry">
<div class="csl-left-margin">411. </div><div class="csl-right-inline">Sutela, S., Poimala, A. &amp; Vainio, E. J. <a href="https://doi.org/10.1093/femsec/fiz119">Viruses of fungi and oomycetes in the soil environment</a>. <em>FEMS Microbiology Ecology</em> <strong>95</strong>, fiz119 (2019).</div>
</div>
<div id="ref-twortINVESTIGATIONNATUREULTRAMICROSCOPIC1915" class="csl-entry">
<div class="csl-left-margin">412. </div><div class="csl-right-inline">Twort, F. W. <a href="https://doi.org/10.1016/S0140-6736(01)20383-3">AN INVESTIGATION ON THE NATURE OF ULTRA-MICROSCOPIC VIRUSES.</a> <em>The Lancet</em> <strong>186</strong>, 1241–1243 (1915).</div>
</div>
<div id="ref-delbrockBacterialVirusesBacteriophages1946" class="csl-entry">
<div class="csl-left-margin">413. </div><div class="csl-right-inline">Delbrock, M. <a href="https://doi.org/10.1111/j.1469-185X.1946.tb00451.x">Bacterial Viruses or Bacteriophages</a>. <em>Biological Reviews</em> <strong>21</strong>, 30–40 (1946).</div>
</div>
<div id="ref-clarkBacterialVirusesHuman2004" class="csl-entry">
<div class="csl-left-margin">414. </div><div class="csl-right-inline">Clark, J. R. &amp; March, J. B. <a href="https://doi.org/10.1586/14760584.3.4.463">Bacterial viruses as human vaccines?</a> <em>Expert Review of Vaccines</em> <strong>3</strong>, 463–476 (2004).</div>
</div>
<div id="ref-vankan-davelaarUsingVirusesNanomedicines2014" class="csl-entry">
<div class="csl-left-margin">415. </div><div class="csl-right-inline">Kan-Davelaar, H. E. van, Hest, J. C. M. van, Cornelissen, J. J. L. M. &amp; Koay, M. S. T. <a href="https://doi.org/10.1111/bph.12662">Using viruses as nanomedicines</a>. <em>British Journal of Pharmacology</em> <strong>171</strong>, 4001–4009 (2014).</div>
</div>
<div id="ref-prangishviliVirusesArchaea2016" class="csl-entry">
<div class="csl-left-margin">416. </div><div class="csl-right-inline">Prangishvili, D., Basta, T., Garrett, R. A. &amp; Krupovic, M. Viruses of the Archaea. in 1–9 (John Wiley &amp; Sons, Ltd, 2016). doi:<a href="https://doi.org/10.1002/9780470015902.a0000774.pub3">10.1002/9780470015902.a0000774.pub3</a>.</div>
</div>
<div id="ref-prangishviliVirusesArchaeaUnifying2006" class="csl-entry">
<div class="csl-left-margin">417. </div><div class="csl-right-inline">Prangishvili, D., Forterre, P. &amp; Garrett, R. A. <a href="https://doi.org/10.1038/nrmicro1527">Viruses of the Archaea: a unifying view</a>. <em>Nature Reviews Microbiology</em> <strong>4</strong>, 837–848 (2006).</div>
</div>
<div id="ref-franckiPlantVirusSatellites" class="csl-entry">
<div class="csl-left-margin">418. </div><div class="csl-right-inline">Francki, R. I. B. Plant virus satellites. <em>ANNUAL REVIEW OF MICROBIOLOGY</em>.</div>
</div>
<div id="ref-xuPlantVirusSatellites2011" class="csl-entry">
<div class="csl-left-margin">419. </div><div class="csl-right-inline">Xu, P. &amp; Roossinck, M. J. Plant Virus Satellites. in (John Wiley &amp; Sons, Ltd, 2011). doi:<a href="https://doi.org/10.1002/9780470015902.a0000771.pub2">10.1002/9780470015902.a0000771.pub2</a>.</div>
</div>
<div id="ref-laiMolecularBiologyHepatitis1995" class="csl-entry">
<div class="csl-left-margin">420. </div><div class="csl-right-inline">Lai, M. M. <a href="https://doi.org/10.1146/annurev.bi.64.070195.001355">The molecular biology of hepatitis delta virus</a>. <em>Annual review of biochemistry</em> <strong>64</strong>, 259–286 (1995).</div>
</div>
<div id="ref-hughesHepatitisDeltaVirus2011" class="csl-entry">
<div class="csl-left-margin">421. </div><div class="csl-right-inline">Hughes, S. A., Wedemeyer, H. &amp; Harrison, P. M. <a href="https://doi.org/10.1016/S0140-6736(10)61931-9">Hepatitis delta virus</a>. <em>The Lancet</em> <strong>378</strong>, 73–85 (2011).</div>
</div>
<div id="ref-desnuesChapterSputnikVirophage2012" class="csl-entry">
<div class="csl-left-margin">422. </div><div class="csl-right-inline">Desnues, C., Boyer, M. &amp; Raoult, D. <a href="https://doi.org/10.1016/B978-0-12-394621-8.00013-3">Chapter 3 - Sputnik, a Virophage Infecting the Viral Domain of Life</a>. in (eds. Łobocka, M. &amp; Szybalski, W. T.) vol. 82 63–89 (Academic Press, 2012).</div>
</div>
<div id="ref-gaiaZamilonNovelVirophage2014" class="csl-entry">
<div class="csl-left-margin">423. </div><div class="csl-right-inline">Gaia, M. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0094923">Zamilon, a Novel Virophage with Mimiviridae Host Specificity</a>. <em>PLOS ONE</em> <strong>9</strong>, e94923 (2014).</div>
</div>
<div id="ref-edgarPetabasescaleSequenceAlignment2022" class="csl-entry">
<div class="csl-left-margin">424. </div><div class="csl-right-inline">Edgar, R. C. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-021-04332-2">Petabase-scale sequence alignment catalyses viral discovery</a>. <em>Nature</em> <strong>602</strong>, 142–147 (2022).</div>
</div>
<div id="ref-nasirInvestigatingConceptOrigin2020" class="csl-entry">
<div class="csl-left-margin">425. </div><div class="csl-right-inline">Nasir, A., Romero-Severson, E. &amp; Claverie, J.-M. <a href="https://doi.org/10.1016/j.tim.2020.08.003">Investigating the Concept and Origin of Viruses</a>. <em>Trends in Microbiology</em> <strong>28</strong>, 959–967 (2020).</div>
</div>
<div id="ref-forterreOriginViruses2009" class="csl-entry">
<div class="csl-left-margin">426. </div><div class="csl-right-inline">Forterre, P. &amp; Prangishvili, D. <a href="https://doi.org/10.1016/j.resmic.2009.07.008">The origin of viruses</a>. <em>Research in Microbiology</em> <strong>160</strong>, 466–472 (2009).</div>
</div>
<div id="ref-forterreOriginVirusesTheir2006" class="csl-entry">
<div class="csl-left-margin">427. </div><div class="csl-right-inline">Forterre, P. <a href="https://doi.org/10.1016/j.virusres.2006.01.010">The origin of viruses and their possible roles in major evolutionary transitions</a>. <em>Virus Research</em> <strong>117</strong>, 5–16 (2006).</div>
</div>
<div id="ref-boekeRetrotransposonsEndogenousRetroviruses1997" class="csl-entry">
<div class="csl-left-margin">428. </div><div class="csl-right-inline">Boeke, J. &amp; Stoye, J. Retrotransposons, Endogenous Retroviruses, and the Evolution of Retroelement. (1997).</div>
</div>
<div id="ref-kojimaViruslikeInsertionsSequence2021" class="csl-entry">
<div class="csl-left-margin">429. </div><div class="csl-right-inline">Kojima, S. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.2010758118">Virus-like insertions with sequence signatures similar to those of endogenous nonretroviral RNA viruses in the human genome</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>118</strong>, e2010758118 (2021).</div>
</div>
<div id="ref-lowerVirusesAllUs1996" class="csl-entry">
<div class="csl-left-margin">430. </div><div class="csl-right-inline">Löwer, R., Löwer, J. &amp; R Kurth. <a href="https://doi.org/10.1073/pnas.93.11.5177">The viruses in all of us: Characteristics and biological significance of human endogenous retrovirus sequences.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>93</strong>, 5177–5184 (1996).</div>
</div>
<div id="ref-griffithsEndogenousRetrovirusesHuman2001" class="csl-entry">
<div class="csl-left-margin">431. </div><div class="csl-right-inline">Griffiths, D. J. <a href="https://doi.org/10.1186/gb-2001-2-6-reviews1017">Endogenous retroviruses in the human genome sequence</a>. <em>Genome Biology</em> <strong>2</strong>, reviews1017.1 (2001).</div>
</div>
<div id="ref-baltimoreExpressionAnimalVirus1971" class="csl-entry">
<div class="csl-left-margin">432. </div><div class="csl-right-inline">Baltimore, D. <a href="https://doi.org/10.1128/br.35.3.235-241.1971">Expression of animal virus genomes</a>. <em>Bacteriological Reviews</em> <strong>35</strong>, 235–241 (1971).</div>
</div>
<div id="ref-kooninBaltimoreClassificationViruses2021" class="csl-entry">
<div class="csl-left-margin">433. </div><div class="csl-right-inline">Koonin, E. V., Krupovic, M. &amp; Agol, V. I. <a href="https://doi.org/10.1128/MMBR.00053-21">The baltimore classification of viruses 50 years later: How does it stand in the light of virus evolution?</a> <em>Microbiology and Molecular Biology Reviews</em> <strong>85</strong>, e00053–21 (2021).</div>
</div>
<div id="ref-domingoRNAVirusGenomes2018" class="csl-entry">
<div class="csl-left-margin">434. </div><div class="csl-right-inline">Domingo, E. &amp; Perales, C. RNA Virus Genomes. in 1–12 (John Wiley &amp; Sons, Ltd, 2018). doi:<a href="https://doi.org/10.1002/9780470015902.a0001488.pub3">10.1002/9780470015902.a0001488.pub3</a>.</div>
</div>
<div id="ref-mcgeochTopicsHerpesvirusGenomics2006" class="csl-entry">
<div class="csl-left-margin">435. </div><div class="csl-right-inline">McGeoch, D. J., Rixon, F. J. &amp; Davison, A. J. <a href="https://doi.org/10.1016/j.virusres.2006.01.002">Topics in herpesvirus genomics and evolution</a>. <em>Virus Research</em> <strong>117</strong>, 90–104 (2006).</div>
</div>
<div id="ref-boehmerHerpesVirusReplication2003" class="csl-entry">
<div class="csl-left-margin">436. </div><div class="csl-right-inline">Boehmer, P. &amp; Nimonkar, A. <a href="https://doi.org/10.1080/1521654031000070645">Herpes Virus Replication</a>. <em>IUBMB Life</em> <strong>55</strong>, 13–22 (2003).</div>
</div>
<div id="ref-brentjensHumanPapillomavirusReview2002" class="csl-entry">
<div class="csl-left-margin">437. </div><div class="csl-right-inline">Brentjens, M. H., Yeung-Yue, K. A., Lee, P. C. &amp; Tyring, S. K. <a href="https://doi.org/10.1016/S0733-8635(01)00028-6">Human papillomavirus: a review</a>. <em>Dermatologic Clinics</em> <strong>20</strong>, 315–331 (2002).</div>
</div>
<div id="ref-kayHepatitisVirusGenetic2007" class="csl-entry">
<div class="csl-left-margin">438. </div><div class="csl-right-inline">Kay, A. &amp; Zoulim, F. <a href="https://doi.org/10.1016/j.virusres.2007.02.021">Hepatitis B virus genetic variability and evolution</a>. <em>Virus Research</em> <strong>127</strong>, 164–176 (2007).</div>
</div>
<div id="ref-parasharRotavirus1998" class="csl-entry">
<div class="csl-left-margin">439. </div><div class="csl-right-inline">Parashar, U. D., Bresee, J. S., Gentsch, J. R. &amp; Glass, R. I. Rotavirus. <em>Emerging Infectious Diseases</em> <strong>4</strong>, 561–570 (1998).</div>
</div>
<div id="ref-simmondsVariabilityHepatitisVirus1995" class="csl-entry">
<div class="csl-left-margin">440. </div><div class="csl-right-inline">Simmonds, P. <a href="https://doi.org/10.1016/0270-9139(95)90121-3">Variability of hepatitis C virus</a>. <em>Hepatology</em> <strong>21</strong>, 570–583 (1995).</div>
</div>
<div id="ref-wimmerGeneticsPoliovirus1993" class="csl-entry">
<div class="csl-left-margin">441. </div><div class="csl-right-inline">Wimmer, E., Hellen, C. U. T. &amp; Cao, X. Genetics of poliovirus. <em>Annual Review of Genetics</em> <strong>27</strong>, 353–437 (1993).</div>
</div>
<div id="ref-racanielloOneHundredYears2006" class="csl-entry">
<div class="csl-left-margin">442. </div><div class="csl-right-inline">Racaniello, V. R. <a href="https://doi.org/10.1016/j.virol.2005.09.015">One hundred years of poliovirus pathogenesis</a>. <em>Virology</em> <strong>344</strong>, 9–16 (2006).</div>
</div>
<div id="ref-paleseNegativestrandRNAViruses1996" class="csl-entry">
<div class="csl-left-margin">443. </div><div class="csl-right-inline">Palese, P., Zheng, H., Engelhardt, O. G., Pleschka, S. &amp; García-Sastre, A. <a href="https://doi.org/10.1073/pnas.93.21.11354">Negative-strand RNA viruses: Genetic engineering and applications.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>93</strong>, 11354–11358 (1996).</div>
</div>
<div id="ref-domingoVirusEvolution2014" class="csl-entry">
<div class="csl-left-margin">444. </div><div class="csl-right-inline">Domingo, E. &amp; Perales, C. Virus Evolution. in (John Wiley &amp; Sons, Ltd, 2014). doi:<a href="https://doi.org/10.1002/9780470015902.a0000436.pub3">10.1002/9780470015902.a0000436.pub3</a>.</div>
</div>
<div id="ref-vkovskiCoronavirusBiologyReplication2021" class="csl-entry">
<div class="csl-left-margin">445. </div><div class="csl-right-inline">V’kovski, P., Kratzel, A., Steiner, S., Stalder, H. &amp; Thiel, V. <a href="https://doi.org/10.1038/s41579-020-00468-6">Coronavirus biology and replication: implications for SARS-CoV-2</a>. <em>Nature Reviews Microbiology</em> <strong>19</strong>, 155–170 (2021).</div>
</div>
<div id="ref-backDengueVirusesOverview2013" class="csl-entry">
<div class="csl-left-margin">446. </div><div class="csl-right-inline">Bäck, A. T. &amp; Lundkvist, Å. <a href="https://doi.org/10.3402/iee.v3i0.19839">Dengue viruses <span></span> an overview</a>. <em>Infection Ecology &amp; Epidemiology</em> <strong>3</strong>, 10.3402/iee.v3i0.19839 (2013).</div>
</div>
<div id="ref-dustinHepatitisVirusLife2016" class="csl-entry">
<div class="csl-left-margin">447. </div><div class="csl-right-inline">Dustin, L. B., Bartolini, B., Capobianchi, M. R. &amp; Pistello, M. <a href="https://doi.org/10.1016/j.cmi.2016.08.025">Hepatitis c virus: Life cycle in cells, infection and host response, and analysis of molecular markers influencing the outcome of infection and response to therapy</a>. <em>Clinical microbiology and infection : the official publication of the European Society of Clinical Microbiology and Infectious Diseases</em> <strong>22</strong>, 826–832 (2016).</div>
</div>
<div id="ref-kadajaPapillomavirusDNAReplication2009" class="csl-entry">
<div class="csl-left-margin">448. </div><div class="csl-right-inline">Kadaja, M., Silla, T., Ustav, E. &amp; Ustav, M. <a href="https://doi.org/10.1016/j.virol.2008.11.032">Papillomavirus DNA replication <span></span> From initiation to genomic instability</a>. <em>Virology</em> <strong>384</strong>, 360–368 (2009).</div>
</div>
<div id="ref-wellerHerpesSimplexViruses2012" class="csl-entry">
<div class="csl-left-margin">449. </div><div class="csl-right-inline">Weller, S. K. &amp; Coen, D. M. <a href="https://doi.org/10.1101/cshperspect.a013011">Herpes simplex viruses: Mechanisms of DNA replication</a>. <em>Cold Spring Harbor Perspectives in Biology</em> <strong>4</strong>, a013011 (2012).</div>
</div>
<div id="ref-beckHepatitisVirusReplication2007" class="csl-entry">
<div class="csl-left-margin">450. </div><div class="csl-right-inline">Beck, J. &amp; Nassal, M. <a href="https://doi.org/10.3748/wjg.v13.i1.48">Hepatitis b virus replication</a>. <em>World Journal of Gastroenterology : WJG</em> <strong>13</strong>, 48–64 (2007).</div>
</div>
<div id="ref-pyleChapter58Biology2017" class="csl-entry">
<div class="csl-left-margin">451. </div><div class="csl-right-inline">Pyle, J. D. &amp; Scholthof, K.-B. G. Chapter 58 - Biology and Pathogenesis of Satellite Viruses. in (eds. Hadidi, A., Flores, R., Randles, J. W. &amp; Palukaitis, P.) 627–636 (Academic Press, 2017). doi:<a href="https://doi.org/10.1016/B978-0-12-801498-1.00058-9">10.1016/B978-0-12-801498-1.00058-9</a>.</div>
</div>
<div id="ref-raoult2megabaseGenomeSequence2004" class="csl-entry">
<div class="csl-left-margin">452. </div><div class="csl-right-inline">Raoult, D. <em>et al.</em> <a href="https://doi.org/10.1126/science.1101485">The 1.2-megabase genome sequence of Mimivirus</a>. <em>Science (New York, N.Y.)</em> <strong>306</strong>, 1344–1350 (2004).</div>
</div>
<div id="ref-campillo-balderasViralGenomeSize2015" class="csl-entry">
<div class="csl-left-margin">453. </div><div class="csl-right-inline">Campillo-Balderas, J. A., Lazcano, A. &amp; Becerra, A. Viral genome size distribution does not correlate with the antiquity of the host lineages. <em>Frontiers in Ecology and Evolution</em> <strong>3</strong>, (2015).</div>
</div>
<div id="ref-cannVirusStructure2015" class="csl-entry">
<div class="csl-left-margin">454. </div><div class="csl-right-inline">Cann, A. J. Virus Structure. in 1–9 (John Wiley &amp; Sons, Ltd, 2015). doi:<a href="https://doi.org/10.1002/9780470015902.a0000439.pub2">10.1002/9780470015902.a0000439.pub2</a>.</div>
</div>
<div id="ref-hladikSettingStageHost2008" class="csl-entry">
<div class="csl-left-margin">455. </div><div class="csl-right-inline">Hladik, F. &amp; McElrath, M. J. <a href="https://doi.org/10.1038/nri2302">Setting the stage: host invasion by HIV</a>. <em>Nature Reviews Immunology</em> <strong>8</strong>, 447–457 (2008).</div>
</div>
<div id="ref-shawHIVTransmission2012" class="csl-entry">
<div class="csl-left-margin">456. </div><div class="csl-right-inline">Shaw, G. M. &amp; Hunter, E. <a href="https://doi.org/10.1101/cshperspect.a006965">HIV Transmission</a>. <em>Cold Spring Harbor Perspectives in Medicine</em> <strong>2</strong>, a006965 (2012).</div>
</div>
<div id="ref-weissHowDoesHIV1993" class="csl-entry">
<div class="csl-left-margin">457. </div><div class="csl-right-inline">Weiss, R. A. <a href="https://doi.org/10.1126/science.8493571">How does HIV cause AIDS?</a> <em>Science</em> <strong>260</strong>, 1273–1279 (1993).</div>
</div>
<div id="ref-melhuishNaturalHistoryHIV2018" class="csl-entry">
<div class="csl-left-margin">458. </div><div class="csl-right-inline">Melhuish, A. &amp; Lewthwaite, P. <a href="https://doi.org/10.1016/j.mpmed.2018.03.010">Natural history of HIV and AIDS</a>. <em>Medicine</em> <strong>46</strong>, 356–361 (2018).</div>
</div>
<div id="ref-murrayPulmonaryComplicationsAcquired1984" class="csl-entry">
<div class="csl-left-margin">459. </div><div class="csl-right-inline">Murray, J. F. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM198406213102529">Pulmonary complications of the acquired immunodeficiency syndrome</a>. <em>New England Journal of Medicine</em> <strong>310</strong>, 1682–1688 (1984).</div>
</div>
<div id="ref-sampathPandemicsThroughoutHistory2021" class="csl-entry">
<div class="csl-left-margin">460. </div><div class="csl-right-inline">Sampath, S. <em>et al.</em> <a href="https://doi.org/10.7759/cureus.18136">Pandemics Throughout the History</a>. <em>Cureus</em> <strong>13</strong>, (2021).</div>
</div>
<div id="ref-worldhealthorganizationGlobalReportUNAIDS2010" class="csl-entry">
<div class="csl-left-margin">461. </div><div class="csl-right-inline">World Health Organization. <em>Global report: UNAIDS report on the global AIDS epidemic 2010.</em> (World Health Organization, 2010).</div>
</div>
<div id="ref-barre-sinoussiIsolationTlymphotropicRetrovirus1983" class="csl-entry">
<div class="csl-left-margin">462. </div><div class="csl-right-inline">Barré-Sinoussi, F. <em>et al.</em> <a href="https://doi.org/10.1126/science.6189183">Isolation of a T-lymphotropic retrovirus from a patient at risk for acquired immune deficiency syndrome (AIDS)</a>. <em>Science (New York, N.Y.)</em> <strong>220</strong>, 868–871 (1983).</div>
</div>
<div id="ref-galloIsolationHumanTcell1983" class="csl-entry">
<div class="csl-left-margin">463. </div><div class="csl-right-inline">Gallo, R. C. <em>et al.</em> <a href="https://doi.org/10.1126/science.6601823">Isolation of human T-cell leukemia virus in acquired immune deficiency syndrome (AIDS)</a>. <em>Science (New York, N.Y.)</em> <strong>220</strong>, 865–867 (1983).</div>
</div>
<div id="ref-clavelIsolationNewHuman1986" class="csl-entry">
<div class="csl-left-margin">464. </div><div class="csl-right-inline">Clavel, F. <em>et al.</em> <a href="https://doi.org/10.1126/science.2425430">Isolation of a new human retrovirus from west african patients with AIDS</a>. <em>Science</em> <strong>233</strong>, 343–346 (1986).</div>
</div>
<div id="ref-gilbert2003" class="csl-entry">
<div class="csl-left-margin">465. </div><div class="csl-right-inline">Gilbert, P. B. <em>et al.</em> <a href="https://doi.org/10.1002/sim.1342">Comparison of HIV-1 and HIV-2 infectivity from a prospective cohort study in Senegal</a>. <em>Statistics in Medicine</em> <strong>22</strong>, 573–593 (2003).</div>
</div>
<div id="ref-vanderloeffSixteenYearsHIV2006" class="csl-entry">
<div class="csl-left-margin">466. </div><div class="csl-right-inline">Loeff, M. F. S. van der <em>et al.</em> <a href="https://doi.org/10.1093/ije/dyl037">Sixteen years of HIV surveillance in a West African research clinic reveals divergent epidemic trends of HIV-1 and HIV-2</a>. <em>International Journal of Epidemiology</em> <strong>35</strong>, 1322–1328 (2006).</div>
</div>
<div id="ref-gaoOriginHIV1Chimpanzee1999" class="csl-entry">
<div class="csl-left-margin">467. </div><div class="csl-right-inline">Gao, F. <em>et al.</em> <a href="https://doi.org/10.1038/17130">Origin of HIV-1 in the chimpanzee Pan troglodytes troglodytes</a>. <em>Nature</em> <strong>397</strong>, 436–441 (1999).</div>
</div>
<div id="ref-sharpOriginsHIVAIDS2011" class="csl-entry">
<div class="csl-left-margin">469. </div><div class="csl-right-inline">Sharp, P. M. &amp; Hahn, B. H. <a href="https://doi.org/10.1101/cshperspect.a006841">Origins of HIV and the AIDS pandemic</a>. <em>Cold Spring Harbor Perspectives in Medicine:</em> <strong>1</strong>, a006841 (2011).</div>
</div>
<div id="ref-hirschAfricanPrimateLentivirus1989" class="csl-entry">
<div class="csl-left-margin">470. </div><div class="csl-right-inline">Hirsch, V. M., Olmsted, R. A., Murphey-Corb, M., Purcell, R. H. &amp; Johnson, P. R. <a href="https://doi.org/10.1038/339389a0">An African primate lentivirus (SIVsmclosely related to HIV-2</a>. <em>Nature</em> <strong>339</strong>, 389–392 (1989).</div>
</div>
<div id="ref-chenGeneticCharacterizationNew1996" class="csl-entry">
<div class="csl-left-margin">472. </div><div class="csl-right-inline">Chen, Z. <em>et al.</em> <a href="https://doi.org/10.1128/JVI.70.6.3617-3627.1996">Genetic characterization of new West African simian immunodeficiency virus SIVsm: geographic clustering of household-derived SIV strains with human immunodeficiency virus type 2 subtypes and genetically diverse viruses from a single feral sooty mangabey troop</a>. <em>Journal of Virology</em> <strong>70</strong>, 3617–3627 (1996).</div>
</div>
<div id="ref-hemelaarOriginDiversityHIV12012" class="csl-entry">
<div class="csl-left-margin">473. </div><div class="csl-right-inline">Hemelaar, J. <a href="https://doi.org/10.1016/j.molmed.2011.12.001">The origin and diversity of the HIV-1 pandemic</a>. <em>Trends in Molecular Medicine</em> <strong>18</strong>, 182–192 (2012).</div>
</div>
<div id="ref-worobeyDirectEvidenceExtensive2008" class="csl-entry">
<div class="csl-left-margin">474. </div><div class="csl-right-inline">Worobey, M. <em>et al.</em> <a href="https://doi.org/10.1038/nature07390">Direct evidence of extensive diversity of HIV-1 in Kinshasa by 1960</a>. <em>Nature</em> <strong>455</strong>, 661–664 (2008).</div>
</div>
<div id="ref-fariaEarlySpreadEpidemic2014" class="csl-entry">
<div class="csl-left-margin">476. </div><div class="csl-right-inline">Faria, N. R. <em>et al.</em> <a href="https://doi.org/10.1126/science.1256739">The early spread and epidemic ignition of HIV-1 in human populations</a>. <em>Science (New York, N.Y.)</em> <strong>346</strong>, 56–61 (2014).</div>
</div>
<div id="ref-korberTimingAncestorHIV12000" class="csl-entry">
<div class="csl-left-margin">477. </div><div class="csl-right-inline">Korber, B. <em>et al.</em> <a href="https://doi.org/10.1126/science.288.5472.1789">Timing the ancestor of the HIV-1 pandemic strains</a>. <em>Science (New York, N.Y.)</em> <strong>288</strong>, 1789–1796 (2000).</div>
</div>
<div id="ref-rambautCausesConsequencesHIV2004" class="csl-entry">
<div class="csl-left-margin">478. </div><div class="csl-right-inline">Rambaut, A., Posada, D., Crandall, K. A. &amp; Holmes, E. C. <a href="https://doi.org/10.1038/nrg1246">The causes and consequences of HIV evolution</a>. <em>Nature Reviews Genetics</em> <strong>5</strong>, 52–61 (2004).</div>
</div>
<div id="ref-mccutchanGlobalEpidemiologyHIV2006" class="csl-entry">
<div class="csl-left-margin">479. </div><div class="csl-right-inline">McCutchan, F. E. <a href="https://doi.org/10.1002/jmv.20599">Global epidemiology of HIV</a>. <em>Journal of Medical Virology</em> <strong>78</strong>, S7–S12 (2006).</div>
</div>
<div id="ref-perez-losadaRecombinationVirusesMechanisms2015" class="csl-entry">
<div class="csl-left-margin">480. </div><div class="csl-right-inline">Pérez-Losada, M., Arenas, M., Galán, J. C., Palero, F. &amp; González-Candelas, F. <a href="https://doi.org/10.1016/j.meegid.2014.12.022">Recombination in viruses: Mechanisms, methods of study, and evolutionary consequences</a>. <em>Infection, Genetics and Evolution</em> <strong>30</strong>, 296–307 (2015).</div>
</div>
<div id="ref-robertsonRecombinationAIDSViruses1995" class="csl-entry">
<div class="csl-left-margin">481. </div><div class="csl-right-inline">Robertson, D. L., Hahn, B. H. &amp; Sharp, P. M. <a href="https://doi.org/10.1007/BF00163230">Recombination in AIDS viruses</a>. <em>Journal of Molecular Evolution</em> <strong>40</strong>, 249–259 (1995).</div>
</div>
<div id="ref-HIVCirculatingRecombinant" class="csl-entry">
<div class="csl-left-margin">482. </div><div class="csl-right-inline">HIV circulating recombinant forms (CRFs).</div>
</div>
<div id="ref-posadaRecombinationEvolutionaryGenomics2002" class="csl-entry">
<div class="csl-left-margin">483. </div><div class="csl-right-inline">Posada, D., Crandall, K. A. &amp; Holmes, E. C. <a href="https://doi.org/10.1146/annurev.genet.36.040202.111115">Recombination in evolutionary genomics</a>. <em>Annual Review of Genetics</em> <strong>36</strong>, 75–97 (2002).</div>
</div>
<div id="ref-taylorChallengeHIV1Subtype2008" class="csl-entry">
<div class="csl-left-margin">484. </div><div class="csl-right-inline">Taylor, B. S., Sobieszczyk, M. E., McCutchan, F. E. &amp; Hammer, S. M. <a href="https://doi.org/10.1056/NEJMra0706737">The challenge of HIV-1 subtype diversity</a>. <em>New England Journal of Medicine</em> <strong>358</strong>, 1590–1602 (2008).</div>
</div>
<div id="ref-hemelaarGlobalTrendsMolecular2011" class="csl-entry">
<div class="csl-left-margin">485. </div><div class="csl-right-inline">Hemelaar, J., Gouws, E., Ghys, P. D. &amp; Osmanov, S. <a href="https://doi.org/10.1097/QAD.0b013e328342ff93">Global trends in molecular epidemiology of HIV-1 during 2000<span></span>2007</a>. <em>AIDS (London, England)</em> <strong>25</strong>, 679–689 (2011).</div>
</div>
<div id="ref-DistributionAllHIV1" class="csl-entry">
<div class="csl-left-margin">486. </div><div class="csl-right-inline">Distribution of all HIV-1 sequences: WORLD.</div>
</div>
<div id="ref-freedHIV1Replication2001" class="csl-entry">
<div class="csl-left-margin">487. </div><div class="csl-right-inline">Freed, E. O. <a href="https://doi.org/10.1023/A:1021070512287">HIV-1 Replication</a>. <em>Somatic Cell and Molecular Genetics</em> <strong>26</strong>, 13–33 (2001).</div>
</div>
<div id="ref-fergusonHIV1ReplicationCycle2002" class="csl-entry">
<div class="csl-left-margin">488. </div><div class="csl-right-inline">Ferguson, M. R., Rojo, D. R., Lindern, J. J. von &amp; O’Brien, W. A. <a href="https://doi.org/10.1016/s0272-2712(02)00015-x">HIV-1 replication cycle</a>. <em>Clinics in Laboratory Medicine</em> <strong>22</strong>, 611–635 (2002).</div>
</div>
<div id="ref-gougeonDirectIndirectMechanisms1993" class="csl-entry">
<div class="csl-left-margin">489. </div><div class="csl-right-inline">Gougeon, M. L., Laurent-Crawford, A. G., Hovanessian, A. G. &amp; Montagnier, L. <a href="https://doi.org/10.1006/smim.1993.1022">Direct and indirect mechanisms mediating apoptosis during HIV infection: contribution to in vivo CD4 T cell depletion</a>. <em>Seminars in Immunology</em> <strong>5</strong>, 187–194 (1993).</div>
</div>
<div id="ref-vidyavijayanPathophysiologyCD4TCell2017" class="csl-entry">
<div class="csl-left-margin">490. </div><div class="csl-right-inline">Vidya Vijayan, K. K., Karthigeyan, K. P., Tripathi, S. P. &amp; Hanna, L. E. <a href="https://doi.org/10.3389/fimmu.2017.00580">Pathophysiology of CD4+ t-cell depletion in HIV-1 and HIV-2 infections</a>. <em>Frontiers in Immunology</em> <strong>8</strong>, 580 (2017).</div>
</div>
<div id="ref-frankelHIV1FifteenProteins1998" class="csl-entry">
<div class="csl-left-margin">491. </div><div class="csl-right-inline">Frankel, A. D. &amp; Young, J. A. <a href="https://doi.org/10.1146/annurev.biochem.67.1.1">HIV-1: fifteen proteins and an RNA</a>. <em>Annual Review of Biochemistry</em> <strong>67</strong>, 1–25 (1998).</div>
</div>
<div id="ref-fossenSolutionStructureHuman2005" class="csl-entry">
<div class="csl-left-margin">492. </div><div class="csl-right-inline">Fossen, T. <em>et al.</em> <a href="https://doi.org/10.1074/jbc.M507375200">Solution structure of the human immunodeficiency virus type 1 p6 protein *</a>. <em>Journal of Biological Chemistry</em> <strong>280</strong>, 42515–42527 (2005).</div>
</div>
<div id="ref-gottlingerEffectMutationsAffecting1991" class="csl-entry">
<div class="csl-left-margin">493. </div><div class="csl-right-inline">Göttlinger, H. G., Dorfman, T., Sodroski, J. G. &amp; Haseltine, W. A. <a href="https://doi.org/10.1073/pnas.88.8.3195">Effect of mutations affecting the p6 gag protein on human immunodeficiency virus particle release.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>88</strong>, 3195–3199 (1991).</div>
</div>
<div id="ref-huangP6GagRequiredParticle1995" class="csl-entry">
<div class="csl-left-margin">494. </div><div class="csl-right-inline">Huang, M., Orenstein, J. M., Martin, M. A. &amp; Freed, E. O. <a href="https://doi.org/10.1128/jvi.69.11.6810-6818.1995">p6Gag is required for particle production from full-length human immunodeficiency virus type 1 molecular clones expressing protease</a>. <em>Journal of Virology</em> <strong>69</strong>, 6810–6818 (1995).</div>
</div>
<div id="ref-bourHumanImmunodeficiencyVirus1995" class="csl-entry">
<div class="csl-left-margin">495. </div><div class="csl-right-inline">Bour, S., Geleziunas, R. &amp; Wainberg, M. A. <a href="https://doi.org/10.1128/mr.59.1.63-93.1995">The human immunodeficiency virus type 1 (HIV-1) CD4 receptor and its central role in promotion of HIV-1 infection</a>. <em>Microbiological Reviews</em> <strong>59</strong>, 63–93 (1995).</div>
</div>
<div id="ref-hernandezViruscellCellcellFusion1996" class="csl-entry">
<div class="csl-left-margin">496. </div><div class="csl-right-inline">Hernandez, L. D., Hoffman, L. R., Wolfsberg, T. G. &amp; White, J. M. <a href="https://doi.org/10.1146/annurev.cellbio.12.1.627">Virus-cell and cell-cell fusion</a>. <em>Annual Review of Cell and Developmental Biology</em> <strong>12</strong>, 627–661 (1996).</div>
</div>
<div id="ref-jonesControlRnaInitiation1994" class="csl-entry">
<div class="csl-left-margin">497. </div><div class="csl-right-inline">Jones, K. &amp; Peterlin, B. <a href="https://doi.org/10.1146/annurev.bi.63.070194.003441">Control of rna initiation and elongation at the hiv-1 promoter</a>. <em>Annual Review of Biochemistry</em> <strong>63</strong>, 717–743 (1994).</div>
</div>
<div id="ref-hopeViralRNAExport1997" class="csl-entry">
<div class="csl-left-margin">498. </div><div class="csl-right-inline">Hope, T. J. <a href="https://doi.org/10.1016/S1074-5521(97)90124-1">Viral RNA export</a>. <em>Chemistry &amp; Biology</em> <strong>4</strong>, 335–344 (1997).</div>
</div>
<div id="ref-mangasarianMultifacetedRoleHIV1997" class="csl-entry">
<div class="csl-left-margin">499. </div><div class="csl-right-inline">Mangasarian, A. &amp; Trono, D. <a href="https://doi.org/10.1016/S0923-2516(97)81909-7">The multifaceted role of HIV Nef</a>. <em>Research in Virology</em> <strong>148</strong>, 30–33 (1997).</div>
</div>
<div id="ref-cohenRoleAuxiliaryProteins1996" class="csl-entry">
<div class="csl-left-margin">500. </div><div class="csl-right-inline">Cohen, é. A., Subbramanian, R. A. &amp; Göttlinger, H. G. Role of Auxiliary Proteins in Retroviral Morphogenesis. in (ed. Kräusslich, H.-G.) 219–235 (Springer, 1996). doi:<a href="https://doi.org/10.1007/978-3-642-80145-7_7">10.1007/978-3-642-80145-7_7</a>.</div>
</div>
<div id="ref-lambVpuVprHuman1997" class="csl-entry">
<div class="csl-left-margin">501. </div><div class="csl-right-inline">Lamb, R. A. &amp; Pinto, L. H. <a href="https://doi.org/10.1006/viro.1997.8451">Do Vpu and Vpr of Human Immunodeficiency Virus Type 1 and NB of Influenza B Virus Have Ion Channel Activities in the Viral Life Cycles?</a> <em>Virology</em> <strong>229</strong>, 1–11 (1997).</div>
</div>
<div id="ref-khanRoleViralProtein2021" class="csl-entry">
<div class="csl-left-margin">502. </div><div class="csl-right-inline">Khan, N. &amp; Geiger, J. D. <a href="https://doi.org/10.3390/v13081466">Role of viral protein u (vpu) in HIV-1 infection and pathogenesis</a>. <em>Viruses</em> <strong>13</strong>, 1466 (2021).</div>
</div>
<div id="ref-emermanHIV1VprCell1996" class="csl-entry">
<div class="csl-left-margin">503. </div><div class="csl-right-inline">Emerman, M. <a href="https://doi.org/10.1016/S0960-9822(02)00676-0">HIV-1, Vpr and the cell cycle</a>. <em>Current Biology</em> <strong>6</strong>, 1096–1103 (1996).</div>
</div>
<div id="ref-millerHumanImmunodeficiencyVirus1988" class="csl-entry">
<div class="csl-left-margin">504. </div><div class="csl-right-inline">Miller, R. H. <a href="https://doi.org/10.1126/science.3347840">Human immunodeficiency virus may encode a novel protein on the genomic DNA plus strand</a>. <em>Science (New York, N.Y.)</em> <strong>239</strong>, 1420–1422 (1988).</div>
</div>
<div id="ref-briquetImmunolocalizationStudiesAntisense2002" class="csl-entry">
<div class="csl-left-margin">505. </div><div class="csl-right-inline">Briquet, S. &amp; Vaquero, C. <a href="https://doi.org/10.1006/viro.2001.1224">Immunolocalization Studies of an Antisense Protein in HIV-1-Infected Cells and Viral Particles</a>. <em>Virology</em> <strong>292</strong>, 177–184 (2002).</div>
</div>
<div id="ref-cassanConcomitantEmergenceAntisense2016" class="csl-entry">
<div class="csl-left-margin">506. </div><div class="csl-right-inline">Cassan, E., Arigon-Chifolleau, A.-M., Mesnard, J.-M., Gross, A. &amp; Gascuel, O. <a href="https://doi.org/10.1073/pnas.1605739113">Concomitant emergence of the antisense protein gene of HIV-1 and of the pandemic</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>113</strong>, 11537–11542 (2016).</div>
</div>
<div id="ref-savoretPilotStudyHumoral2020" class="csl-entry">
<div class="csl-left-margin">507. </div><div class="csl-right-inline">Savoret, J. <em>et al.</em> A pilot study of the humoral response against the AntiSense protein (ASP) in HIV-1-infected patients. <em>Frontiers in Microbiology</em> <strong>11</strong>, (2020).</div>
</div>
<div id="ref-zardecki2022" class="csl-entry">
<div class="csl-left-margin">508. </div><div class="csl-right-inline">Zardecki, C. <em>et al.</em> <a href="https://doi.org/10.1002/pro.4200">PDB-101: Educational resources supporting molecular explorations through biology and medicine</a>. <em>Protein Science</em> <strong>31</strong>, 129–140 (2022).</div>
</div>
<div id="ref-eisingerHIVViralLoad2019" class="csl-entry">
<div class="csl-left-margin">509. </div><div class="csl-right-inline">Eisinger, R. W., Dieffenbach, C. W. &amp; Fauci, A. S. <a href="https://doi.org/10.1001/jama.2018.21167">HIV viral load and transmissibility of HIV infection: Undetectable equals untransmittable</a>. <em>JAMA</em> <strong>321</strong>, 451–452 (2019).</div>
</div>
<div id="ref-palellaDecliningMorbidityMortality1998" class="csl-entry">
<div class="csl-left-margin">510. </div><div class="csl-right-inline">Palella, F. J. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM199803263381301">Declining morbidity and mortality among patients with advanced human immunodeficiency virus infection</a>. <em>New England Journal of Medicine</em> <strong>338</strong>, 853–860 (1998).</div>
</div>
<div id="ref-forsytheTwentyYearsAntiretroviral2019" class="csl-entry">
<div class="csl-left-margin">511. </div><div class="csl-right-inline">Forsythe, S. S. <em>et al.</em> <a href="https://doi.org/10.1377/hlthaff.2018.05391">Twenty years of antiretroviral therapy for people living with HIV: Global costs, health achievements, economic benefits</a>. <em>Health Affairs</em> <strong>38</strong>, 1163–1172 (2019).</div>
</div>
<div id="ref-fischlEfficacyAzidothymidineAZT1987" class="csl-entry">
<div class="csl-left-margin">512. </div><div class="csl-right-inline">Fischl, M. A. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM198707233170401">The efficacy of azidothymidine (AZT) in the treatment of patients with AIDS and AIDS-related complex</a>. <em>New England Journal of Medicine</em> <strong>317</strong>, 185–191 (1987).</div>
</div>
<div id="ref-richmanSusceptibilityNucleosideAnalogues1990" class="csl-entry">
<div class="csl-left-margin">513. </div><div class="csl-right-inline">Richman, D. D. <a href="https://doi.org/10.1016/0002-9343(90)90414-9">Susceptibility to nucleoside analogues of zidovudine-resistant isolates of human immunodeficiency virus</a>. <em>The American Journal of Medicine</em> <strong>88</strong>, S8–S10 (1990).</div>
</div>
<div id="ref-yeoDeterminationHIV1RT2020" class="csl-entry">
<div class="csl-left-margin">514. </div><div class="csl-right-inline">Yeo, J. Y., Goh, G.-R., Su, C. T.-T. &amp; Gan, S. K.-E. <a href="https://doi.org/10.3390/v12030297">The Determination of HIV-1 RT Mutation Rate, Its Possible Allosteric Effects, and Its Implications on Drug Resistance</a>. <em>Viruses</em> <strong>12</strong>, 297 (2020).</div>
</div>
<div id="ref-cuevasExtremelyHighMutation2015" class="csl-entry">
<div class="csl-left-margin">515. </div><div class="csl-right-inline">Cuevas, J. M., Geller, R., Garijo, R., López-Aldeguer, J. &amp; Sanjuán, R. <a href="https://doi.org/10.1371/journal.pbio.1002251">Extremely High Mutation Rate of HIV-1 In Vivo</a>. <em>PLOS Biology</em> <strong>13</strong>, e1002251 (2015).</div>
</div>
<div id="ref-gulickTreatmentIndinavirZidovudine1997" class="csl-entry">
<div class="csl-left-margin">516. </div><div class="csl-right-inline">Gulick, R. M. <em>et al.</em> <a href="https://doi.org/10.1056/NEJM199709113371102">Treatment with indinavir, zidovudine, and lamivudine in adults with human immunodeficiency virus infection and prior antiretroviral therapy</a>. <em>The New England Journal of Medicine</em> <strong>337</strong>, 734–739 (1997).</div>
</div>
<div id="ref-wensingFifteenYearsHIV2010" class="csl-entry">
<div class="csl-left-margin">517. </div><div class="csl-right-inline">Wensing, A. M. J., Maarseveen, N. M. van &amp; Nijhuis, M. <a href="https://doi.org/10.1016/j.antiviral.2009.10.003">Fifteen years of HIV Protease Inhibitors: raising the barrier to resistance</a>. <em>Antiviral Research</em> <strong>85</strong>, 59–74 (2010).</div>
</div>
<div id="ref-pedersenNonNucleosideReverseTranscriptase1999" class="csl-entry">
<div class="csl-left-margin">518. </div><div class="csl-right-inline">Pedersen, O. S. &amp; Pedersen, E. B. <a href="https://doi.org/10.1177/095632029901000601">Non-Nucleoside Reverse Transcriptase Inhibitors: The NNRTI Boom</a>. <em>Antiviral Chemistry and Chemotherapy</em> <strong>10</strong>, 285–314 (1999).</div>
</div>
<div id="ref-scarsiHIV1IntegraseInhibitors2020" class="csl-entry">
<div class="csl-left-margin">519. </div><div class="csl-right-inline">Scarsi, K. K., Havens, J. P., Podany, A. T., Avedissian, S. N. &amp; Fletcher, C. V. <a href="https://doi.org/10.1007/s40265-020-01379-9">HIV-1 Integrase Inhibitors: A Comparative Review of Efficacy and Safety</a>. <em>Drugs</em> <strong>80</strong>, 1649–1676 (2020).</div>
</div>
<div id="ref-fletcherEnfuvirtideNewDrug2003" class="csl-entry">
<div class="csl-left-margin">520. </div><div class="csl-right-inline">Fletcher, C. V. <a href="https://doi.org/10.1016/S0140-6736(03)13323-5">Enfuvirtide, a new drug for HIV infection</a>. <em>The Lancet</em> <strong>361</strong>, 1577–1578 (2003).</div>
</div>
<div id="ref-esteHIVEntryInhibitors2007" class="csl-entry">
<div class="csl-left-margin">521. </div><div class="csl-right-inline">Esté, J. A. &amp; Telenti, A. <a href="https://doi.org/10.1016/S0140-6736(07)61052-6">HIV entry inhibitors</a>. <em>The Lancet</em> <strong>370</strong>, 81–88 (2007).</div>
</div>
<div id="ref-kilbyNovelTherapiesBased2003" class="csl-entry">
<div class="csl-left-margin">522. </div><div class="csl-right-inline">Kilby, J. M. &amp; Eron, J. J. <a href="https://doi.org/10.1056/NEJMra022812">Novel therapies based on mechanisms of HIV-1 cell entry</a>. <em>New England Journal of Medicine</em> <strong>348</strong>, 2228–2238 (2003).</div>
</div>
<div id="ref-yeniUpdateHAARTHIV2006" class="csl-entry">
<div class="csl-left-margin">523. </div><div class="csl-right-inline">Yeni, P. <a href="https://doi.org/10.1016/j.jhep.2005.11.021">Update on HAART in HIV</a>. <em>Journal of Hepatology</em> <strong>44</strong>, S100–S103 (2006).</div>
</div>
<div id="ref-palmisanoBriefHistoryAntiretroviral2011" class="csl-entry">
<div class="csl-left-margin">524. </div><div class="csl-right-inline">Palmisano, L. &amp; Vella, S. <a href="https://doi.org/10.4415/ANN_11_01_10">A brief history of antiretroviral therapy of HIV infection: success and challenges</a>. <em>Annali dell’Istituto Superiore Di Sanita</em> <strong>47</strong>, 44–48 (2011).</div>
</div>
<div id="ref-mehtaPotentialFactorsAffecting1997" class="csl-entry">
<div class="csl-left-margin">525. </div><div class="csl-right-inline">Mehta, S., Moore, R. D. &amp; Graham, N. M. H. Potential factors affecting adherence with HIV therapy. <em>AIDS</em> <strong>11</strong>, 16651670 (1997).</div>
</div>
<div id="ref-millerComplianceTreatmentRegimens1997" class="csl-entry">
<div class="csl-left-margin">526. </div><div class="csl-right-inline">Miller, N. H. <a href="https://doi.org/10.1016/S0002-9343(97)00467-1">Compliance with treatment regimens in chronic asymptomatic diseases</a>. <em>The American Journal of Medicine</em> <strong>102</strong>, 43–49 (1997).</div>
</div>
<div id="ref-chesneyAdherenceHIVCombination2000" class="csl-entry">
<div class="csl-left-margin">527. </div><div class="csl-right-inline">Chesney, M. A., Morin, M. &amp; Sherr, L. <a href="https://doi.org/10.1016/S0277-9536(99)00468-2">Adherence to HIV combination therapy</a>. <em>Social Science &amp; Medicine</em> <strong>50</strong>, 1599–1605 (2000).</div>
</div>
<div id="ref-aldirSingletabletRegimensHIV2014" class="csl-entry">
<div class="csl-left-margin">528. </div><div class="csl-right-inline">Aldir, I., Horta, A. &amp; Serrado, M. <a href="https://doi.org/10.1185/03007995.2013.844685">Single-tablet regimens in HIV: Does it really make a difference?</a> <em>Current Medical Research and Opinion</em> <strong>30</strong>, 89–97 (2014).</div>
</div>
<div id="ref-grantPreexposureChemoprophylaxisHIV2010" class="csl-entry">
<div class="csl-left-margin">529. </div><div class="csl-right-inline">Grant, R. M. <em>et al.</em> <a href="https://doi.org/10.1056/NEJMoa1011205">Preexposure chemoprophylaxis for HIV prevention in men who have sex with men</a>. <em>New England Journal of Medicine</em> <strong>363</strong>, 2587–2599 (2010).</div>
</div>
<div id="ref-buchbinderPreexposureProphylaxisPromise2011" class="csl-entry">
<div class="csl-left-margin">531. </div><div class="csl-right-inline">Buchbinder, S. P. &amp; Liu, A. <a href="https://doi.org/10.1007/s10461-011-9894-1">Pre-exposure prophylaxis and the promise of combination prevention approaches</a>. <em>AIDS and behavior</em> <strong>15 Suppl 1</strong>, S72–79 (2011).</div>
</div>
<div id="ref-riddellHIVPreexposureProphylaxis2018" class="csl-entry">
<div class="csl-left-margin">532. </div><div class="csl-right-inline">Riddell, J. I., Amico, K. R. &amp; Mayer, K. H. <a href="https://doi.org/10.1001/jama.2018.1917">HIV preexposure prophylaxis: A review</a>. <em>JAMA</em> <strong>319</strong>, 1261–1268 (2018).</div>
</div>
<div id="ref-emaTruvada2018" class="csl-entry">
<div class="csl-left-margin">533. </div><div class="csl-right-inline">EMA. Truvada. (2018).</div>
</div>
<div id="ref-PrEPPrEPHIV2022" class="csl-entry">
<div class="csl-left-margin">534. </div><div class="csl-right-inline">About PrEP | PrEP | HIV basics | HIV/AIDS | CDC. (2022).</div>
</div>
<div id="ref-zolopaEvolutionHIVTreatment2010" class="csl-entry">
<div class="csl-left-margin">535. </div><div class="csl-right-inline">Zolopa, A. R. <a href="https://doi.org/10.1016/j.antiviral.2009.10.018">The evolution of HIV treatment guidelines: Current state-of-the-art of ART</a>. <em>Antiviral Research</em> <strong>85</strong>, 241–244 (2010).</div>
</div>
<div id="ref-worldhealthorganizationConsolidatedGuidelinesHIV2021" class="csl-entry">
<div class="csl-left-margin">536. </div><div class="csl-right-inline">World Health Organization. <em>Consolidated guidelines on HIV prevention, testing, treatment, service delivery and monitoring: recommendations for a public health approach</em>. (2021).</div>
</div>
<div id="ref-ammaranondMechanismHIVAntiretroviral2012" class="csl-entry">
<div class="csl-left-margin">537. </div><div class="csl-right-inline">Ammaranond, P. &amp; Sanguansittianan, S. <a href="https://doi.org/10.1111/j.1472-8206.2011.01009.x">Mechanism of HIV antiretroviral drugs progress toward drug resistance</a>. <em>Fundamental &amp; Clinical Pharmacology</em> <strong>26</strong>, 146–161 (2012).</div>
</div>
<div id="ref-clavelHIVDrugResistance2004" class="csl-entry">
<div class="csl-left-margin">538. </div><div class="csl-right-inline">Clavel, F. &amp; Hance, A. J. <a href="https://doi.org/10.1056/NEJMra025195">HIV drug resistance</a>. <em>New England Journal of Medicine</em> <strong>350</strong>, 1023–1035 (2004).</div>
</div>
<div id="ref-sarafianosStructureFunctionHIV12009" class="csl-entry">
<div class="csl-left-margin">539. </div><div class="csl-right-inline">Sarafianos, S. G. <em>et al.</em> <a href="https://doi.org/10.1016/j.jmb.2008.10.071">Structure and function of <span>HIV</span>-1 reverse transcriptase: Molecular mechanisms of polymerization and inhibition</a>. <em>Journal of molecular biology</em> <strong>385</strong>, 693–713 (2009).</div>
</div>
<div id="ref-goodsellIllustrateSoftwareBiomolecular2019" class="csl-entry">
<div class="csl-left-margin">540. </div><div class="csl-right-inline">Goodsell, D. S., Autin, L. &amp; Olson, A. J. <a href="https://doi.org/10.1016/j.str.2019.08.011">Illustrate: Software for Biomolecular Illustration</a>. <em>Structure (London, England: 1993)</em> <strong>27</strong>, 1716–1720.e1 (2019).</div>
</div>
<div id="ref-esnoufUniqueFeaturesStructure1997" class="csl-entry">
<div class="csl-left-margin">541. </div><div class="csl-right-inline">Esnouf, R. M. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.94.8.3984">Unique features in the structure of the complex between HIV-1 reverse transcriptase and the bis(heteroaryl)piperazine (BHAP) U-90152 explain resistance mutations for this nonnucleoside inhibitor</a>. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>94</strong>, 3984–3989 (1997).</div>
</div>
<div id="ref-hangActivityIsolatedHIV2004" class="csl-entry">
<div class="csl-left-margin">542. </div><div class="csl-right-inline">Hang, J. Q. <em>et al.</em> <a href="https://doi.org/10.1016/j.bbrc.2004.03.061">Activity of the isolated HIV RNase H domain and specific inhibition by N-hydroxyimides</a>. <em>Biochemical and Biophysical Research Communications</em> <strong>317</strong>, 321–329 (2004).</div>
</div>
<div id="ref-klumppRecentProgressDesign2006" class="csl-entry">
<div class="csl-left-margin">543. </div><div class="csl-right-inline">Klumpp, K. &amp; Mirzadegan, T. <a href="https://doi.org/10.2174/138161206776873653">Recent progress in the design of small molecule inhibitors of HIV RNase h</a>. <em>Current Pharmaceutical Design</em> <strong>12</strong>, 1909–1922 (2006).</div>
</div>
<div id="ref-menendez-ariasMechanismsResistanceNucleoside2008" class="csl-entry">
<div class="csl-left-margin">544. </div><div class="csl-right-inline">Menéndez-Arias, L. <a href="https://doi.org/10.1016/j.virusres.2007.12.015">Mechanisms of resistance to nucleoside analogue inhibitors of HIV-1 reverse transcriptase</a>. <em>Virus Research</em> <strong>134</strong>, 124–146 (2008).</div>
</div>
<div id="ref-sluis-cremerMolecularMechanismsHIV12000" class="csl-entry">
<div class="csl-left-margin">545. </div><div class="csl-right-inline">Sluis-Cremer, N., Arion, D. &amp; Parniak*, M. A. <a href="https://doi.org/10.1007/PL00000626">Molecular mechanisms of HIV-1 resistance to nucleoside reverse transcriptase inhibitors (NRTIs)</a>. <em>Cellular and Molecular Life Sciences CMLS</em> <strong>57</strong>, 1408–1422 (2000).</div>
</div>
<div id="ref-sarafianosLamivudine3TCResistance1999" class="csl-entry">
<div class="csl-left-margin">546. </div><div class="csl-right-inline">Sarafianos, S. G. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.96.18.10027">Lamivudine (3TC) resistance in HIV-1 reverse transcriptase involves steric hindrance with beta-branched amino acids</a>. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>96</strong>, 10027–10032 (1999).</div>
</div>
<div id="ref-meyerMechanismAZTResistance1999" class="csl-entry">
<div class="csl-left-margin">547. </div><div class="csl-right-inline">Meyer, P. R., Matsuura, S. E., Mian, A. M., So, A. G. &amp; Scott, W. A. <a href="https://doi.org/10.1016/s1097-2765(00)80185-9">A mechanism of AZT resistance: an increase in nucleotide-dependent primer unblocking by mutant HIV-1 reverse transcriptase</a>. <em>Molecular Cell</em> <strong>4</strong>, 35–43 (1999).</div>
</div>
<div id="ref-boyerSelectiveExcisionAZTMP2001" class="csl-entry">
<div class="csl-left-margin">548. </div><div class="csl-right-inline">Boyer, P. L., Sarafianos, S. G., Arnold, E. &amp; Hughes, S. H. <a href="https://doi.org/10.1128/JVI.75.10.4832-4842.2001">Selective excision of AZTMP by drug-resistant human immunodeficiency virus reverse transcriptase</a>. <em>Journal of Virology</em> <strong>75</strong>, 4832–4842 (2001).</div>
</div>
<div id="ref-deeksNonnucleosideReverseTranscriptase2001" class="csl-entry">
<div class="csl-left-margin">549. </div><div class="csl-right-inline">Deeks, S. G. Nonnucleoside reverse transcriptase inhibitor resistance. <em>JAIDS Journal of Acquired Immune Deficiency Syndromes</em> <strong>26</strong>, S25 (2001).</div>
</div>
<div id="ref-renStructuralBasisDrug2008" class="csl-entry">
<div class="csl-left-margin">550. </div><div class="csl-right-inline">Ren, J. &amp; Stammers, D. K. <a href="https://doi.org/10.1016/j.virusres.2007.12.018">Structural basis for drug resistance mechanisms for non-nucleoside inhibitors of HIV reverse transcriptase</a>. <em>Virus Research</em> <strong>134</strong>, 157–170 (2008).</div>
</div>
<div id="ref-lloydHighCostFidelity2014" class="csl-entry">
<div class="csl-left-margin">551. </div><div class="csl-right-inline">Lloyd, S. B., Kent, S. J. &amp; Winnall, W. R. <a href="https://doi.org/10.1089/aid.2013.0153">The high cost of fidelity</a>. <em>AIDS Research and Human Retroviruses</em> <strong>30</strong>, 8–16 (2014).</div>
</div>
<div id="ref-pearlStructuralModelRetroviral1987" class="csl-entry">
<div class="csl-left-margin">552. </div><div class="csl-right-inline">Pearl, L. H. &amp; Taylor, W. R. <a href="https://doi.org/10.1038/329351a0">A structural model for the retroviral proteases</a>. <em>Nature</em> <strong>329</strong>, 351–354 (1987).</div>
</div>
<div id="ref-gulnikHIVProteaseEnzyme2000" class="csl-entry">
<div class="csl-left-margin">553. </div><div class="csl-right-inline">Gulnik, S., Erickson, J. W. &amp; Xie, D. <a href="https://doi.org/10.1016/S0083-6729(00)58026-1">HIV protease: Enzyme function and drug resistance</a>. in vol. 58 213–256 (Academic Press, 2000).</div>
</div>
<div id="ref-silvaInhibitionCatalyticMechanism1996" class="csl-entry">
<div class="csl-left-margin">554. </div><div class="csl-right-inline">Silva, A. M., Cachau, R. E., Sham, H. L. &amp; Erickson, J. W. <a href="https://doi.org/10.1006/jmbi.1996.0026">Inhibition and catalytic mechanism of HIV-1 aspartic protease</a>. <em>Journal of Molecular Biology</em> <strong>255</strong>, 321–340 (1996).</div>
</div>
<div id="ref-hornakHIV1ProteaseFlaps2006" class="csl-entry">
<div class="csl-left-margin">555. </div><div class="csl-right-inline">Hornak, V., Okur, A., Rizzo, R. C. &amp; Simmerling, C. <a href="https://doi.org/10.1073/pnas.0508452103">HIV-1 protease flaps spontaneously open and reclose in molecular dynamics simulations</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>103</strong>, 915–920 (2006).</div>
</div>
<div id="ref-freedbergRapidStructuralFluctuations2002" class="csl-entry">
<div class="csl-left-margin">556. </div><div class="csl-right-inline">Freedberg, D. I. <em>et al.</em> Rapid structural fluctuations of the free HIV protease flaps in solution: Relationship to crystal structures and comparison with predictions of dynamics calculations. <em>Protein Science : A Publication of the Protein Society</em> <strong>11</strong>, 221–232 (2002).</div>
</div>
<div id="ref-yuStructuralInsightsHIV12017" class="csl-entry">
<div class="csl-left-margin">557. </div><div class="csl-right-inline">Yu, Y. <em>et al.</em> <a href="https://doi.org/10.1039/C7RA09691G">Structural insights into HIV-1 protease flap opening processes and key intermediates</a>. <em>RSC Advances</em> <strong>7</strong>, 45121–45128 (2017).</div>
</div>
<div id="ref-robertsRationalDesignPeptideBased1990" class="csl-entry">
<div class="csl-left-margin">558. </div><div class="csl-right-inline">Roberts, N. A. <em>et al.</em> <a href="https://doi.org/10.1126/science.2183354">Rational design of peptide-based HIV proteinase inhibitors</a>. <em>Science</em> <strong>248</strong>, 358–361 (1990).</div>
</div>
<div id="ref-lvHIVProteaseInhibitors2015" class="csl-entry">
<div class="csl-left-margin">559. </div><div class="csl-right-inline">Lv, Z., Chu, Y. &amp; Wang, Y. <a href="https://doi.org/10.2147/HIV.S79956">HIV protease inhibitors: A review of molecular selectivity and toxicity</a>. <em>HIV/AIDS (Auckland, N.Z.)</em> <strong>7</strong>, 95–104 (2015).</div>
</div>
<div id="ref-prabu-jeyabalanSubstrateShapeDetermines2002" class="csl-entry">
<div class="csl-left-margin">560. </div><div class="csl-right-inline">Prabu-Jeyabalan, M., Nalivaika, E. &amp; Schiffer, C. A. <a href="https://doi.org/10.1016/S0969-2126(02)00720-7">Substrate shape determines specificity of recognition for HIV-1 protease: Analysis of crystal structures of six substrate complexes</a>. <em>Structure</em> <strong>10</strong>, 369–381 (2002).</div>
</div>
<div id="ref-prabu-jeyabalanSubstrateEnvelopeDrug2006" class="csl-entry">
<div class="csl-left-margin">561. </div><div class="csl-right-inline">Prabu-Jeyabalan, M. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.50.4.1518-1521.2006">Substrate envelope and drug resistance: Crystal structure of RO1 in complex with wild-type human immunodeficiency virus type 1 protease</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>50</strong>, 1518–1521 (2006).</div>
</div>
<div id="ref-kurtyilmazImprovingViralProtease2016" class="csl-entry">
<div class="csl-left-margin">562. </div><div class="csl-right-inline">Kurt Yilmaz, N., Swanstrom, R. &amp; Schiffer, C. A. <a href="https://doi.org/10.1016/j.tim.2016.03.010">Improving Viral Protease Inhibitors to Counter Drug Resistance</a>. <em>Trends in Microbiology</em> <strong>24</strong>, 547–557 (2016).</div>
</div>
<div id="ref-chiuStructureFunctionHIV12004" class="csl-entry">
<div class="csl-left-margin">563. </div><div class="csl-right-inline">Chiu, T. K. &amp; Davies, D. R. <a href="https://doi.org/10.2174/1568026043388547">Structure and function of HIV-1 integrase</a>. <em>Current Topics in Medicinal Chemistry</em> <strong>4</strong>, 965–977 (2004).</div>
</div>
<div id="ref-espositoHIVIntegraseStructure1999" class="csl-entry">
<div class="csl-left-margin">564. </div><div class="csl-right-inline">Esposito, D. &amp; Craigie, R. HIV Integrase Structure and Function. in (eds. Rlaramorosch, K., Murphy, F. A. &amp; Shawn, A. J.) vol. 52 319–333 (Academic Press, 1999).</div>
</div>
<div id="ref-delelisIntegraseIntegrationBiochemical2008" class="csl-entry">
<div class="csl-left-margin">565. </div><div class="csl-right-inline">Delelis, O., Carayon, K., Saïb, A., Deprez, E. &amp; Mouscadet, J.-F. <a href="https://doi.org/10.1186/1742-4690-5-114">Integrase and integration: Biochemical activities of HIV-1 integrase</a>. <em>Retrovirology</em> <strong>5</strong>, 114 (2008).</div>
</div>
<div id="ref-maertensStructureFunctionRetroviral2022" class="csl-entry">
<div class="csl-left-margin">566. </div><div class="csl-right-inline">Maertens, G. N., Engelman, A. N. &amp; Cherepanov, P. <a href="https://doi.org/10.1038/s41579-021-00586-9">Structure and function of retroviral integrase</a>. <em>Nature Reviews Microbiology</em> <strong>20</strong>, 20–34 (2022).</div>
</div>
<div id="ref-pommierIntegraseInhibitorsTreat2005" class="csl-entry">
<div class="csl-left-margin">567. </div><div class="csl-right-inline">Pommier, Y., Johnson, A. A. &amp; Marchand, C. <a href="https://doi.org/10.1038/nrd1660">Integrase inhibitors to treat HIV/Aids</a>. <em>Nature Reviews Drug Discovery</em> <strong>4</strong>, 236–248 (2005).</div>
</div>
<div id="ref-blancoHIV1IntegraseInhibitor2011" class="csl-entry">
<div class="csl-left-margin">568. </div><div class="csl-right-inline">Blanco, J.-L., Varghese, V., Rhee, S.-Y., Gatell, J. M. &amp; Shafer, R. W. <a href="https://doi.org/10.1093/infdis/jir025">HIV-1 integrase inhibitor resistance and its clinical implications</a>. <em>The Journal of Infectious Diseases</em> <strong>203</strong>, 1204–1214 (2011).</div>
</div>
<div id="ref-gerettiEmergingPatternsImplications2012" class="csl-entry">
<div class="csl-left-margin">569. </div><div class="csl-right-inline">Geretti, A. M., Armenia, D. &amp; Ceccherini-Silberstein, F. <a href="https://doi.org/10.1097/QCO.0b013e32835a1de7">Emerging patterns and implications of HIV-1 integrase inhibitor resistance</a>. <em>Current Opinion in Infectious Diseases</em> <strong>25</strong>, 677686 (2012).</div>
</div>
<div id="ref-knoxMultidrugResistantHIV1Infection2017" class="csl-entry">
<div class="csl-left-margin">570. </div><div class="csl-right-inline">Knox, D. C., Anderson, P. L., Harrigan, P. R. &amp; Tan, D. H. S. <a href="https://doi.org/10.1056/NEJMc1611639">Multidrug-resistant HIV-1 infection despite preexposure prophylaxis</a>. <em>New England Journal of Medicine</em> <strong>376</strong>, 501–502 (2017).</div>
</div>
<div id="ref-hurtPreexposureProphylaxisAntiretroviral2011" class="csl-entry">
<div class="csl-left-margin">571. </div><div class="csl-right-inline">Hurt, C. B., Eron, J. J. &amp; Cohen, M. S. <a href="https://doi.org/10.1093/cid/cir684">Pre-exposure prophylaxis and antiretroviral resistance: HIV prevention at a cost?</a> <em>Clinical Infectious Diseases: An Official Publication of the Infectious Diseases Society of America</em> <strong>53</strong>, 1265–1270 (2011).</div>
</div>
<div id="ref-gibasDrugResistanceHIV2019" class="csl-entry">
<div class="csl-left-margin">572. </div><div class="csl-right-inline">Gibas, K. M., Berg, P. van den, Powell, V. E. &amp; Krakower, D. S. <a href="https://doi.org/10.1007/s40265-019-01108-x">Drug Resistance During HIV Pre-Exposure Prophylaxis</a>. <em>Drugs</em> <strong>79</strong>, 609–619 (2019).</div>
</div>
<div id="ref-mouradPhylotypebasedAnalysisHighlights2015" class="csl-entry">
<div class="csl-left-margin">573. </div><div class="csl-right-inline">Mourad, R. <em>et al.</em> <a href="https://doi.org/10.1097/QAD.0000000000000768">A phylotype-based analysis highlights the role of drug-naive <span>HIV</span>-positive individuals in the transmission of antiretroviral resistance in the <span>UK</span></a>. <em>Aids</em> <strong>29</strong>, 1917–1925 (2015).</div>
</div>
<div id="ref-hueDemonstrationSustainedDrugResistant2009" class="csl-entry">
<div class="csl-left-margin">574. </div><div class="csl-right-inline">Hué, S. <em>et al.</em> <a href="https://doi.org/10.1128/JVI.01556-08">Demonstration of <span>Sustained Drug</span>-<span>Resistant Human Immunodeficiency Virus Type</span> 1 <span>Lineages Circulating</span> among <span>Treatment</span>-<span>Naïve Individuals</span></a>. <em>Journal of Virology</em> <strong>83</strong>, 2645–2654 (2009).</div>
</div>
<div id="ref-drescherTreatmentNaiveIndividualsAre2014" class="csl-entry">
<div class="csl-left-margin">575. </div><div class="csl-right-inline">Drescher, S. M. <em>et al.</em> <a href="https://doi.org/10.1093/cid/cit694">Treatment-naive individuals are the major source of transmitted HIV-1 drug resistance in men who have sex with men in the swiss HIV cohort study</a>. <em>Clinical Infectious Diseases</em> <strong>58</strong>, 285–294 (2014).</div>
</div>
<div id="ref-boermaHighLevelsPretreatment2016" class="csl-entry">
<div class="csl-left-margin">576. </div><div class="csl-right-inline">Boerma, R. S. <em>et al.</em> <a href="https://doi.org/10.7448/IAS.19.1.21140">High levels of pre-treatment HIV drug resistance and treatment failure in Nigerian children</a>. <em>Journal of the International AIDS Society</em> <strong>19</strong>, 21140 (2016).</div>
</div>
<div id="ref-clutterHIV1DrugResistance2016" class="csl-entry">
<div class="csl-left-margin">577. </div><div class="csl-right-inline">Clutter, D. S., Jordan, M. R., Bertagnolio, S. &amp; Shafer, R. W. <a href="https://doi.org/10.1016/j.meegid.2016.08.031">HIV-1 drug resistance and resistance testing</a>. <em>Infection, Genetics and Evolution</em> <strong>46</strong>, 292–307 (2016).</div>
</div>
<div id="ref-kuhnertQuantifyingFitnessCost2018" class="csl-entry">
<div class="csl-left-margin">578. </div><div class="csl-right-inline">Kühnert, D. <em>et al.</em> <a href="https://doi.org/10.1371/journal.ppat.1006895">Quantifying the fitness cost of HIV-1 drug resistance mutations through phylodynamics</a>. <em>PLOS Pathogens</em> <strong>14</strong>, e1006895 (2018).</div>
</div>
<div id="ref-mespledeViralFitnessCost2013" class="csl-entry">
<div class="csl-left-margin">579. </div><div class="csl-right-inline">Mesplède, T. <em>et al.</em> <a href="https://doi.org/10.1186/1742-4690-10-22">Viral fitness cost prevents HIV-1 from evading dolutegravir drug pressure</a>. <em>Retrovirology</em> <strong>10</strong>, 22 (2013).</div>
</div>
<div id="ref-castroPersistenceHIV1Transmitted2013" class="csl-entry">
<div class="csl-left-margin">580. </div><div class="csl-right-inline">Castro, H. <em>et al.</em> <a href="https://doi.org/10.1093/infdis/jit345">Persistence of HIV-1 Transmitted Drug Resistance Mutations</a>. <em>The Journal of Infectious Diseases</em> <strong>208</strong>, 1459–1463 (2013).</div>
</div>
<div id="ref-blasselDrugResistanceMutations2021" class="csl-entry">
<div class="csl-left-margin">581. </div><div class="csl-right-inline">Blassel, L. <em>et al.</em> <a href="https://doi.org/10.1016/j.coviro.2021.09.009">Drug resistance mutations in HIV: new bioinformatics approaches and challenges</a>. <em>Current Opinion in Virology</em> <strong>51</strong>, 56–64 (2021).</div>
</div>
<div id="ref-ukchicsteeringcommitteeCreationLargeUKbased2004" class="csl-entry">
<div class="csl-left-margin">582. </div><div class="csl-right-inline">UK CHIC Steering Committee. <a href="https://doi.org/10.1111/j.1468-1293.2004.00197.x">The creation of a large UK-based multicentre cohort of HIV-infected individuals: The UK Collaborative HIV Cohort (UK CHIC) Study</a>. <em>HIV Medicine</em> <strong>5</strong>, 115–124 (2004).</div>
</div>
<div id="ref-abeler-dornerPANGEAHIVPhylogeneticsNetworks2019" class="csl-entry">
<div class="csl-left-margin">583. </div><div class="csl-right-inline">Abeler-Dörner, L. <em>et al.</em> <a href="https://doi.org/10.1097/COH.0000000000000542">PANGEA-HIV 2: Phylogenetics and networks for generalised epidemics in africa</a>. <em>Current Opinion in HIV and AIDS</em> <strong>14</strong>, 173180 (2019).</div>
</div>
<div id="ref-shaferRationaleUsesPublic2006" class="csl-entry">
<div class="csl-left-margin">584. </div><div class="csl-right-inline">Shafer, R. W. <a href="https://doi.org/10.1086/505356">Rationale and Uses of a Public HIV Drug<span>-</span>Resistance Database</a>. <em>The Journal of Infectious Diseases</em> <strong>194</strong>, S51–S58 (2006).</div>
</div>
<div id="ref-kuikenHIVSequenceDatabases2003" class="csl-entry">
<div class="csl-left-margin">585. </div><div class="csl-right-inline">Kuiken, C., Korber, B. &amp; Shafer, R. W. HIV sequence databases. <em>AIDS reviews</em> <strong>5</strong>, 52–61 (2003).</div>
</div>
<div id="ref-wensing2019UpdateDrug2019" class="csl-entry">
<div class="csl-left-margin">586. </div><div class="csl-right-inline">Wensing, A. M. <em>et al.</em> 2019 update of the drug resistance mutations in HIV-1. <em>Topics in Antiviral Medicine</em> <strong>27</strong>, 111–121 (2019).</div>
</div>
<div id="ref-clarkMutationsRetroviralGenes2007" class="csl-entry">
<div class="csl-left-margin">587. </div><div class="csl-right-inline">Clark, S. A., Calef, C. &amp; Mellors, J. W. Mutations in retroviral genes associated with drug resistance. <em>HIV sequence compendium</em> 58158 (2007).</div>
</div>
<div id="ref-liuWebResourcesHIV2006d" class="csl-entry">
<div class="csl-left-margin">588. </div><div class="csl-right-inline">Liu, T. F. &amp; Shafer, R. W. <a href="https://doi.org/10.1086/503914">Web <span>Resources</span> for <span>HIV Type</span> 1 <span>Genotypic</span>-<span>Resistance Test Interpretation</span></a>. <em>Clinical Infectious Diseases</em> <strong>42</strong>, 1608–1618 (2006).</div>
</div>
<div id="ref-johnsonUpdateDrugResistance2016" class="csl-entry">
<div class="csl-left-margin">589. </div><div class="csl-right-inline">Johnson, V. A. <em>et al.</em> Update of the drug resistance mutations in HIV-1: March 2013. <em>Topics in Antiviral Medicine</em> <strong>21</strong>, 6–7 (2016).</div>
</div>
<div id="ref-villabona-arenasIndepthAnalysisHIV12016" class="csl-entry">
<div class="csl-left-margin">590. </div><div class="csl-right-inline">Villabona-Arenas, C. J. <em>et al.</em> <a href="https://doi.org/10.1097/QAD.0000000000001233">In-depth analysis of <span>HIV</span>-1 drug resistance mutations in <span>HIV</span>-infected individuals failing first-line regimens in <span>West</span> and <span>Central Africa</span></a>. <em>AIDS</em> <strong>30</strong>, 2577 (2016).</div>
</div>
<div id="ref-shulmanGeneticCorrelatesEfavirenz2004" class="csl-entry">
<div class="csl-left-margin">591. </div><div class="csl-right-inline">Shulman, N. S., Bosch, R. J., Mellors, J. W., Albrecht, M. A. &amp; Katzenstein, D. A. Genetic correlates of efavirenz hypersusceptibility. <em>AIDS</em> <strong>18</strong>, 17811785 (2004).</div>
</div>
<div id="ref-millerGenotypicPhenotypicPredictors2004" class="csl-entry">
<div class="csl-left-margin">592. </div><div class="csl-right-inline">Miller, M. D. <em>et al.</em> <a href="https://doi.org/10.1086/381784">Genotypic and phenotypic predictors of the magnitude of response to tenofovir disoproxil fumarate treatment in antiretroviral-experienced patients</a>. <em>The Journal of Infectious Diseases</em> <strong>189</strong>, 837–846 (2004).</div>
</div>
<div id="ref-brownMethodsCorrectingMultiple1997" class="csl-entry">
<div class="csl-left-margin">593. </div><div class="csl-right-inline">Brown, B. W. &amp; Russell, K. <a href="https://doi.org/10.1002/(SICI)1097-0258(19971130)16:22&lt;2511::AID-SIM693&gt;3.0.CO;2-4">Methods correcting for multiple testing: operating characteristics</a>. <em>Statistics in Medicine</em> <strong>16</strong>, 2511–2528 (1997).</div>
</div>
<div id="ref-austinTestingMultipleStatistical2006" class="csl-entry">
<div class="csl-left-margin">594. </div><div class="csl-right-inline">Austin, P. C., Mamdani, M. M., Juurlink, D. N. &amp; Hux, J. E. <a href="https://doi.org/10.1016/j.jclinepi.2006.01.012">Testing multiple statistical hypotheses resulted in spurious associations: a study of astrological signs and health</a>. <em>Journal of Clinical Epidemiology</em> <strong>59</strong>, 964–969 (2006).</div>
</div>
<div id="ref-hochbergMultipleComparisonProcedures1987" class="csl-entry">
<div class="csl-left-margin">595. </div><div class="csl-right-inline">Hochberg, Y. &amp; Tamhane, A. C. <em>Multiple comparison procedures</em>. (1987).</div>
</div>
<div id="ref-benjaminiControllingFalseDiscovery1995" class="csl-entry">
<div class="csl-left-margin">596. </div><div class="csl-right-inline">Benjamini, Y. &amp; Hochberg, Y. Controlling the false discovery rate: A practical and powerful approach to multiple testing. <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> <strong>57</strong>, 289–300 (1995).</div>
</div>
<div id="ref-gonzalesExtendedSpectrumHIV12003" class="csl-entry">
<div class="csl-left-margin">597. </div><div class="csl-right-inline">Gonzales, M. J. <em>et al.</em> <a href="https://doi.org/10.1097/01.aids.0000050860.71999.23">Extended spectrum of HIV-1 reverse transcriptase mutations in patients receiving multiple nucleoside analog inhibitors</a>. <em>AIDS (London, England)</em> <strong>17</strong>, 791–799 (2003).</div>
</div>
<div id="ref-seoigheModelDirectionalSelection2007" class="csl-entry">
<div class="csl-left-margin">598. </div><div class="csl-right-inline">Seoighe, C. <em>et al.</em> <a href="https://doi.org/10.1093/molbev/msm021">A model of directional selection applied to the evolution of drug resistance in HIV-1</a>. <em>Molecular Biology and Evolution</em> <strong>24</strong>, 1025–1031 (2007).</div>
</div>
<div id="ref-shamStatisticalPowerSignificance2014" class="csl-entry">
<div class="csl-left-margin">599. </div><div class="csl-right-inline">Sham, P. C. &amp; Purcell, S. M. <a href="https://doi.org/10.1038/nrg3706">Statistical power and significance testing in large-scale genetic studies</a>. <em>Nature Reviews Genetics</em> <strong>15</strong>, 335–346 (2014).</div>
</div>
<div id="ref-alizonPhylogeneticApproachReveals2010" class="csl-entry">
<div class="csl-left-margin">600. </div><div class="csl-right-inline">Alizon, S. <em>et al.</em> <a href="https://doi.org/10.1371/journal.ppat.1001123">Phylogenetic Approach Reveals That Virus Genotype Largely Determines HIV Set-Point Viral Load</a>. <em>PLOS Pathogens</em> <strong>6</strong>, e1001123 (2010).</div>
</div>
<div id="ref-flynnDeepSequencingProtease2015" class="csl-entry">
<div class="csl-left-margin">601. </div><div class="csl-right-inline">Flynn, W. F. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pcbi.1004249">Deep Sequencing of Protease Inhibitor Resistant HIV Patient Isolates Reveals Patterns of Correlated Mutations in Gag and Protease</a>. <em>PLOS Computational Biology</em> <strong>11</strong>, e1004249 (2015).</div>
</div>
<div id="ref-petropoulosNovelPhenotypicDrug2000" class="csl-entry">
<div class="csl-left-margin">602. </div><div class="csl-right-inline">Petropoulos, C. J. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.44.4.920-928.2000">A novel phenotypic drug susceptibility assay for human immunodeficiency virus type 1</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>44</strong>, 920–928 (2000).</div>
</div>
<div id="ref-hertogsRapidMethodSimultaneous1998" class="csl-entry">
<div class="csl-left-margin">603. </div><div class="csl-right-inline">Hertogs, K. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.42.2.269">A rapid method for simultaneous detection of phenotypic resistance to inhibitors of protease and reverse transcriptase in recombinant human immunodeficiency virus type 1 isolates from patients treated with antiretroviral drugs</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>42</strong>, 269–276 (1998).</div>
</div>
<div id="ref-heilek-snyderRoleHIVPhenotypic2002" class="csl-entry">
<div class="csl-left-margin">604. </div><div class="csl-right-inline">Heilek-Snyder, G. &amp; Bean, P. Role of HIV phenotypic assays in the management of HIV infection. <em>American Clinical Laboratory</em> <strong>21</strong>, 40–43 (2002).</div>
</div>
<div id="ref-gartlandSusceptibilityGlobalHIV12021" class="csl-entry">
<div class="csl-left-margin">606. </div><div class="csl-right-inline">Gartland, M. <em>et al.</em> <a href="https://doi.org/10.1093/jac/dkaa474">Susceptibility of global HIV-1 clinical isolates to fostemsavir using the PhenoSense® entry assay</a>. <em>Journal of Antimicrobial Chemotherapy</em> <strong>76</strong>, 648–652 (2021).</div>
</div>
<div id="ref-masquelierGenotypicPhenotypicResistance2001" class="csl-entry">
<div class="csl-left-margin">607. </div><div class="csl-right-inline">Masquelier, B. <em>et al.</em> <a href="https://doi.org/10.1128/AAC.45.6.1836-1842.2001">Genotypic and phenotypic resistance patterns of human immunodeficiency virus type 1 variants with insertions or deletions in the reverse transcriptase (RT): Multicenter study of patients treated with RT inhibitors</a>. <em>Antimicrobial Agents and Chemotherapy</em> <strong>45</strong>, 1836–1842 (2001).</div>
</div>
<div id="ref-larderMultipleMutationsHIV11989" class="csl-entry">
<div class="csl-left-margin">608. </div><div class="csl-right-inline">Larder, B. A. &amp; Kemp, S. D. <a href="https://doi.org/10.1126/science.2479983">Multiple mutations in HIV-1 reverse transcriptase confer high-level resistance to zidovudine (AZT)</a>. <em>Science (New York, N.Y.)</em> <strong>246</strong>, 1155–1158 (1989).</div>
</div>
<div id="ref-devreeseResistanceHumanImmunodeficiency1992" class="csl-entry">
<div class="csl-left-margin">609. </div><div class="csl-right-inline">Vreese, K. de <em>et al.</em> <a href="https://doi.org/10.1016/0042-6822(92)90550-9">Resistance of human immunodeficiency virus type 1 reverse transcriptase to TIBO derivatives induced by site-directed mutagenesis</a>. <em>Virology</em> <strong>188</strong>, 900–904 (1992).</div>
</div>
<div id="ref-tambuyzerEffectMutationsPosition2011" class="csl-entry">
<div class="csl-left-margin">610. </div><div class="csl-right-inline">Tambuyzer, L., Nijs, S., Daems, B., Picchio, G. &amp; Vingerhoets, J. <a href="https://doi.org/10.1097/QAI.0b013e3182237f74">Effect of mutations at position E138 in HIV-1 reverse transcriptase on phenotypic susceptibility and virologic response to etravirine</a>. <em>JAIDS Journal of Acquired Immune Deficiency Syndromes</em> <strong>58</strong>, 1822 (2011).</div>
</div>
<div id="ref-katzensteinPhenotypicSusceptibilityVirological2003" class="csl-entry">
<div class="csl-left-margin">611. </div><div class="csl-right-inline">Katzenstein, D. A. <em>et al.</em> Phenotypic susceptibility and virological outcome in nucleoside-experienced patients receiving three or four antiretroviral drugs. <em>AIDS</em> <strong>17</strong>, 821830 (2003).</div>
</div>
<div id="ref-blasselUsingMachineLearning2021" class="csl-entry">
<div class="csl-left-margin">612. </div><div class="csl-right-inline">Blassel, L. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pcbi.1008873">Using machine learning and big data to explore the drug resistance landscape in HIV</a>. <em>PLOS Computational Biology</em> <strong>17</strong>, e1008873 (2021).</div>
</div>
<div id="ref-sheikamamuddyImprovingFoldResistance2017b" class="csl-entry">
<div class="csl-left-margin">613. </div><div class="csl-right-inline">Sheik Amamuddy, O., Bishop, N. T. &amp; Tastan Bishop, Ö. <a href="https://doi.org/10.1186/s12859-017-1782-x">Improving fold resistance prediction of <span>HIV</span>-1 against protease and reverse transcriptase inhibitors using artificial neural networks</a>. <em>BMC Bioinformatics</em> <strong>18</strong>, 369 (2017).</div>
</div>
<div id="ref-beerenwinkelGeno2phenoInterpretingGenotypic2001" class="csl-entry">
<div class="csl-left-margin">614. </div><div class="csl-right-inline">Beerenwinkel, N. <em>et al.</em> <a href="https://doi.org/10.1109/5254.972080">Geno2pheno: Interpreting genotypic <span>HIV</span> drug resistance tests</a>. <em>IEEE Intelligent Systems</em> <strong>16</strong>, 35–41 (2001).</div>
</div>
<div id="ref-riemenschneiderSHIVAWebApplication2016b" class="csl-entry">
<div class="csl-left-margin">615. </div><div class="csl-right-inline">Riemenschneider, M., Hummel, T. &amp; Heider, D. <a href="https://doi.org/10.1186/s12859-016-1179-2"><span>SHIVA</span> - a web application for drug resistance and tropism testing in <span>HIV</span></a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 314 (2016).</div>
</div>
<div id="ref-beerenwinkelDiversityComplexityHIV12002c" class="csl-entry">
<div class="csl-left-margin">616. </div><div class="csl-right-inline">Beerenwinkel, N. <em>et al.</em> Diversity and complexity of <span>HIV</span>-1 drug resistance: <span>A</span> bioinformatics approach to predicting phenotype from genotype. <em>Proceedings of the National Academy of Sciences</em> <strong>99</strong>, 8271–8276 (2002).</div>
</div>
<div id="ref-heiderMultilabelClassificationExploiting2013b" class="csl-entry">
<div class="csl-left-margin">617. </div><div class="csl-right-inline">Heider, D., Senge, R., Cheng, W. &amp; Hüllermeier, E. <a href="https://doi.org/10.1093/bioinformatics/btt331">Multilabel classification for exploiting cross-resistance information in <span>HIV</span>-1 drug resistance prediction</a>. <em>Bioinformatics</em> <strong>29</strong>, 1946–1952 (2013).</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="5">
<li id="fn5"><p>This sections build upon a review I participated in during my PhD<span class="citation"><sup><a href="#ref-blasselDrugResistanceMutations2021" role="doc-biblioref">581</a></sup></span><a href="viruses-hiv-and-drug-resistance.html#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="learning-from-sequences-and-alignments.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="HIV-paper.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/05-HIV-context.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
