<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments</title>
  <meta name="description" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments" />
  
  
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-09-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="general-introduction.html"/>
<link rel="next" href="aligning-sequence-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />
<link href="libs/tabwid-1.0.0/scrool.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">From sequences to knowledge,</br> improving and learning from sequence alignments.</a></li>

<li class="divider"></li>
<li><a href="index.html#section"></a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#résumé"><i class="fa fa-check"></i>Résumé</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="list-of-acronyms-and-abbreviations.html"><a href="list-of-acronyms-and-abbreviations.html"><i class="fa fa-check"></i>List of Acronyms and Abbreviations</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html"><i class="fa fa-check"></i>General Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#research-output"><i class="fa fa-check"></i>Research output</a>
<ul>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#journal-publications"><i class="fa fa-check"></i>Journal publications</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html#presentations-and-posters"><i class="fa fa-check"></i>Presentations and posters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>1</b> What is Sequence data ?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>1.1</b> Biological sequences, a primer</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>1.1.1</b> What is DNA ?</a></li>
<li class="chapter" data-level="1.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#from-information-to-action"><i class="fa fa-check"></i><b>1.1.2</b> From Information to action</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining sequence data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough"><i class="fa fa-check"></i><b>1.2.1</b> Sanger sequencing, a breakthrough</a></li>
<li class="chapter" data-level="1.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> Next-generation sequencing</a></li>
<li class="chapter" data-level="1.2.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#long-read-sequencing"><i class="fa fa-check"></i><b>1.2.3</b> Long read sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>1.3</b> Sequencing errors, how to account for them ?</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#error-correction-methods"><i class="fa fa-check"></i><b>1.3.1</b> Error correction methods</a></li>
<li class="chapter" data-level="1.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#more-accurate-sequencing-methods"><i class="fa fa-check"></i><b>1.3.2</b> More accurate sequencing methods</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#the-special-case-of-homopolymers"><i class="fa fa-check"></i><b>1.4</b> The special case of homopolymers</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-the-human-genome"><i class="fa fa-check"></i><b>1.4.1</b> Homopolymers and the human genome</a></li>
<li class="chapter" data-level="1.4.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-long-reads"><i class="fa fa-check"></i><b>1.4.2</b> Homopolymers and long reads</a></li>
<li class="chapter" data-level="1.4.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#accounting-for-homopolymers"><i class="fa fa-check"></i><b>1.4.3</b> Accounting for homopolymers</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>2</b> Aligning sequence data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>2.1</b> What is an alignment ?</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#why-align"><i class="fa fa-check"></i><b>2.1.1</b> Why align ?</a></li>
<li class="chapter" data-level="2.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-align-two-sequences"><i class="fa fa-check"></i><b>2.1.2</b> How to align two sequences ?</a></li>
<li class="chapter" data-level="2.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#scoring-and-substitution-models"><i class="fa fa-check"></i><b>2.1.3</b> Scoring and substitution models</a></li>
<li class="chapter" data-level="2.1.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#dealing-with-gaps"><i class="fa fa-check"></i><b>2.1.4</b> Dealing with gaps</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-speed-up-pairwise-alignment"><i class="fa fa-check"></i><b>2.2</b> How to speed up pairwise alignment ?</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#changing-the-method"><i class="fa fa-check"></i><b>2.2.1</b> Changing the method</a></li>
<li class="chapter" data-level="2.2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#seed-and-extend-with-data-structures"><i class="fa fa-check"></i><b>2.2.2</b> Seed and extend with data structures</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#the-specificities-of-read-mapping"><i class="fa fa-check"></i><b>2.3</b> The specificities of read-mapping</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-read-mapping"><i class="fa fa-check"></i><b>2.3.1</b> What is read-mapping ?</a></li>
<li class="chapter" data-level="2.3.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#challenges-of-read-mapping"><i class="fa fa-check"></i><b>2.3.2</b> Challenges of read-mapping</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.4</b> Multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#progressive-alignment"><i class="fa fa-check"></i><b>2.4.1</b> Progressive alignment</a></li>
<li class="chapter" data-level="2.4.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#other-methods"><i class="fa fa-check"></i><b>2.4.2</b> Other methods</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#conclusion-1"><i class="fa fa-check"></i><b>2.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="HPC-paper.html"><a href="HPC-paper.html"><i class="fa fa-check"></i><b>3</b> Contribution 1: Improving read alignment by exploring a sequence transformation space</a>
<ul>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#highlights"><i class="fa fa-check"></i>Highlights</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#graphical-abstract"><i class="fa fa-check"></i>Graphical Abstract</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="3.1" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="HPC-paper.html"><a href="HPC-paper.html#hpc-results"><i class="fa fa-check"></i><b>3.2</b> Results</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:msr-def"><i class="fa fa-check"></i><b>3.2.1</b> Streaming sequence reductions</a></li>
<li class="chapter" data-level="3.2.2" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:enum"><i class="fa fa-check"></i><b>3.2.2</b> Restricting the space of streaming sequence reductions</a></li>
<li class="chapter" data-level="3.2.3" data-path="HPC-paper.html"><a href="HPC-paper.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>3.2.3</b> Selection of mapping-friendly sequence reductions</a></li>
<li class="chapter" data-level="3.2.4" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>3.2.4</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes</a></li>
<li class="chapter" data-level="3.2.5" data-path="HPC-paper.html"><a href="HPC-paper.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>3.2.5</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome</a></li>
<li class="chapter" data-level="3.2.6" data-path="HPC-paper.html"><a href="HPC-paper.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>3.2.6</b> Raw mapping improves upon HPC on centromeric regions</a></li>
<li class="chapter" data-level="3.2.7" data-path="HPC-paper.html"><a href="HPC-paper.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>3.2.7</b> Positions of incorrectly mapped reads across the entire human genome</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="HPC-paper.html"><a href="HPC-paper.html#discussion"><i class="fa fa-check"></i><b>3.3</b> Discussion</a></li>
<li class="chapter" data-level="3.4" data-path="HPC-paper.html"><a href="HPC-paper.html#limitations-of-this-study"><i class="fa fa-check"></i><b>3.4</b> Limitations of this study</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#author-contributions"><i class="fa fa-check"></i>Author contributions</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#declaration-of-interests"><i class="fa fa-check"></i>Declaration of interests</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#sec:methods"><i class="fa fa-check"></i>STAR Methods</a>
<ul>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#lead-contact"><i class="fa fa-check"></i>Lead contact</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#materials-availability"><i class="fa fa-check"></i>Materials availability</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#data-and-code-availability"><i class="fa fa-check"></i>Data and code availability</a></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#method-details"><i class="fa fa-check"></i>Method details</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="HPC-paper.html"><a href="HPC-paper.html#supplementary-information"><i class="fa fa-check"></i>Supplementary information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html"><i class="fa fa-check"></i><b>4</b> Learning from sequences and alignments</a>
<ul>
<li class="chapter" data-level="4.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#why-learn-from-sequences-and-alignments"><i class="fa fa-check"></i><b>4.1</b> Why learn from sequences and alignments ?</a></li>
<li class="chapter" data-level="4.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#what-to-learn"><i class="fa fa-check"></i><b>4.2</b> What to learn ?</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#supervised-learning-from-biological-sequences"><i class="fa fa-check"></i><b>4.2.1</b> Supervised learning from biological sequences</a></li>
<li class="chapter" data-level="4.2.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#unsupervised-learning-from-biological-sequences"><i class="fa fa-check"></i><b>4.2.2</b> Unsupervised learning from biological sequences</a></li>
<li class="chapter" data-level="4.2.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#others-paradigms"><i class="fa fa-check"></i><b>4.2.3</b> Others paradigms</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#how-to-learn"><i class="fa fa-check"></i><b>4.3</b> How to learn ?</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#general-setting"><i class="fa fa-check"></i><b>4.3.1</b> General setting</a></li>
<li class="chapter" data-level="4.3.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>4.3.2</b> Tests and statistical learning</a></li>
<li class="chapter" data-level="4.3.3" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#more-complex-methods"><i class="fa fa-check"></i><b>4.3.3</b> More complex methods</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#pre-processing-the-alignment-for-machine-learning"><i class="fa fa-check"></i><b>4.4</b> Pre-processing the alignment for machine learning</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#general-purpose-encodings"><i class="fa fa-check"></i><b>4.4.1</b> General purpose encodings</a></li>
<li class="chapter" data-level="4.4.2" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#biological-sequence-specific-encodings"><i class="fa fa-check"></i><b>4.4.2</b> Biological sequence-specific encodings</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="learning-from-sequences-and-alignments.html"><a href="learning-from-sequences-and-alignments.html#conclusion-2"><i class="fa fa-check"></i><b>4.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html"><i class="fa fa-check"></i><b>5</b> Viruses, HIV and drug resistance</a>
<ul>
<li class="chapter" data-level="5.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#what-are-viruses"><i class="fa fa-check"></i><b>5.1</b> What are viruses ?</a></li>
<li class="chapter" data-level="5.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#getting-to-know-hiv"><i class="fa fa-check"></i><b>5.2</b> Getting to know HIV</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#quick-presentation-of-hiv"><i class="fa fa-check"></i><b>5.2.1</b> Quick Presentation of HIV</a></li>
<li class="chapter" data-level="5.2.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#the-replication-cycle-of-hiv"><i class="fa fa-check"></i><b>5.2.2</b> The replication cycle of HIV</a></li>
<li class="chapter" data-level="5.2.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#genetics-of-hiv"><i class="fa fa-check"></i><b>5.2.3</b> Genetics of HIV</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#drug-resistance-in-hiv"><i class="fa fa-check"></i><b>5.3</b> Drug resistance in HIV</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#a-quick-history-of-art"><i class="fa fa-check"></i><b>5.3.1</b> A quick history of ART</a></li>
<li class="chapter" data-level="5.3.2" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#drug-mechanisms"><i class="fa fa-check"></i><b>5.3.2</b> Main mechanisms of viral proteins, antiretroviral drugs and associated resistance.</a></li>
<li class="chapter" data-level="5.3.3" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#consequences-of-resistance-on-global-health"><i class="fa fa-check"></i><b>5.3.3</b> Consequences of resistance on global health</a></li>
<li class="chapter" data-level="5.3.4" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#finding-drms"><i class="fa fa-check"></i><b>5.3.4</b> Finding DRMs </a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="viruses-hiv-and-drug-resistance.html"><a href="viruses-hiv-and-drug-resistance.html#conclusion-3"><i class="fa fa-check"></i><b>5.4</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>6</b> Contribution 2: Inferring mutation roles from sequence alignments using machine learning</a>
<ul>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#abstract-paper"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#author-summary"><i class="fa fa-check"></i>Author summary</a></li>
<li class="chapter" data-level="6.1" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>6.2</b> Materials and methods</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>6.2.2</b> Classifier training</a></li>
<li class="chapter" data-level="6.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>6.2.3</b> Measuring classifier performance</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-results"><i class="fa fa-check"></i><b>6.3</b> Results</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>6.3.1</b> Classifier performance &amp; interpretation</a></li>
<li class="chapter" data-level="6.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>6.3.2</b> Additional classification results</a></li>
<li class="chapter" data-level="6.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>6.3.3</b> Identifying new mutations from classifiers</a></li>
<li class="chapter" data-level="6.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>6.3.4</b> Detailed analysis of potentially resistance-associated mutations</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>6.4</b> Discussion and perspectives</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-acknowledgments"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#supporting-information"><i class="fa fa-check"></i>Supporting Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>7</b> Learning alignments, an interesting perspective</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#deep-learning-and-sequences"><i class="fa fa-check"></i><b>7.1</b> Deep learning and sequences</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#intro-to-deep-learning"><i class="fa fa-check"></i><b>7.1.1</b> Intro to deep learning</a></li>
<li class="chapter" data-level="7.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learned-sequence-embeddings"><i class="fa fa-check"></i><b>7.1.2</b> Learned sequence embeddings</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-sequence-alignment"><i class="fa fa-check"></i><b>7.2</b> Learning sequence alignment</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-a-substitution-matrix"><i class="fa fa-check"></i><b>7.2.1</b> Predicting a substitution matrix</a></li>
<li class="chapter" data-level="7.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-an-alignment"><i class="fa fa-check"></i><b>7.2.2</b> predicting an alignment</a></li>
<li class="chapter" data-level="7.2.3" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#the-attention-limitation"><i class="fa fa-check"></i><b>7.2.3</b> The attention limitation</a></li>
<li class="chapter" data-level="7.2.4" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-read-mappings"><i class="fa fa-check"></i><b>7.2.4</b> Predicting read-mappings</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#conclusion-4"><i class="fa fa-check"></i><b>7.3</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html"><i class="fa fa-check"></i>Global conclusion</a>
<ul>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#improving-read-mapping-with-msrs"><i class="fa fa-check"></i>Improving read-mapping with MSRs</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#searching-for-resistance-mutations-in-hiv"><i class="fa fa-check"></i>Searching for resistance mutations in HIV</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#final-words"><i class="fa fa-check"></i>Final words</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-references.html"><a href="global-references.html"><i class="fa fa-check"></i>Global References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”</a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation</a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#msr-performance-comparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison</a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome"><i class="fa fa-check"></i><b>A.3</b> Analyzing read origin on whole human genome</a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.4</b> Performance of MSRs on the Drosophila genome</a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#key-resource-table"><i class="fa fa-check"></i><b>A.5</b> Key Resource Table</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “HIV and DRMs”</a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html#detailed-list-of-hiv-1-protein-structures-used-for-figure-generation."><i class="fa fa-check"></i><b>B.1</b> Detailed list of HIV-1 protein structures used for figure generation.</a></li>
<li class="chapter" data-level="B.2" data-path="HIV-intro-appendix.html"><a href="HIV-intro-appendix.html#list-of-all-antiretroviral-drugs"><i class="fa fa-check"></i><b>B.2</b> List of all antiretroviral drugs</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>C</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”</a>
<ul>
<li class="chapter" data-level="C.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>C.1</b> S1 Appendix (Technical appendix).</a>
<ul>
<li class="chapter" data-level="C.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>C.1.1</b> Data</a></li>
<li class="chapter" data-level="C.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>C.1.2</b> Classifiers</a></li>
<li class="chapter" data-level="C.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>C.1.3</b> Scoring</a></li>
</ul></li>
<li class="chapter" data-level="C.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>C.2</b> S1 Fig.</a></li>
<li class="chapter" data-level="C.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>C.3</b> S2 Fig.</a></li>
<li class="chapter" data-level="C.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>C.4</b> S3 Fig.</a></li>
<li class="chapter" data-level="C.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>C.5</b> S1 Table.</a></li>
<li class="chapter" data-level="C.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>C.6</b> S2 Appendix. (Fisher exact tests)</a></li>
<li class="chapter" data-level="C.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>C.7</b> S1 Data.</a></li>
<li class="chapter" data-level="C.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>C.8</b> S2 Data.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequences to knowledge, improving and learning from sequence alignments</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="what-is-sequence-data" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> What is Sequence data ?<a href="what-is-sequence-data.html#what-is-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="biological-sequences-a-primer" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Biological sequences, a primer<a href="what-is-sequence-data.html#biological-sequences-a-primer" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To fully understand the work that was done during this thesis, as well as the choices that were made, some basic knowledge of molecular biology and genetics is needed. If you are already familiar with biological sequences, feel free to skip ahead to section <a href="what-is-sequence-data.html#obtaining-sequence-data">1.2</a>.</p>
<div id="what-is-dna" class="section level3 hasAnchor" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> What is DNA ?<a href="what-is-sequence-data.html#what-is-dna" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>D</strong>esoxyribo<strong>N</strong>ucleic <strong>A</strong>cid (DNA) is one of the most important molecules there is, without it complex life as we know it is impossible. It contains all the genetic information of a given organism, that is to say all the information necessary for the organism to: 1) function as a living being and 2) make a perfect copy of itself. This is the case for the overwhelming majority of living organisms on planet earth, from elephants to potatoes, to micro-organisms like bacteria.</p>
<p>DNA is a polymer, composed of monomeric units called nucleotides. Each nucleotide is composed of ribose (a five carbon sugar) on which are attached a phosphate group as well as one of four nucleobases: Adenine (A), Cytosine (C), Guanine (G) of Thymine (T). These four types of nucleotide monomers link up with one-another, through phosphate-sugar bonds, creating a single strand of DNA. The ordered sequence of these four types of nucleotides in strand encodes all the genetic information necessary for the organism to function. Nucleotides in a strand form strong complementary bonds with nucleotides from another strand, A with T and C with G. These bonds allow two strands of DNA to form the double-helix structure of DNA<span class="citation"><sup><a href="#ref-watson1953" role="doc-biblioref">1</a></sup></span> shown in Figure <a href="what-is-sequence-data.html#fig:figDNA">1.1</a>. The specificity of nucleotide bonds ensure that the two strands of the double helix are complementary and that the information contained in one strand can be recovered from the other. This ensures a certain structural stability to the DNA molecule and a way to recover the important information that could be lost due to a damaged strand.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figDNA"></span>
<img src="figures/Sequence-Intro/DNA.png" alt="**Double-helix structure of DNA.**  
Each strand of DNA has a phosphate-sugar backbone on which are attached nucleobases. The two strands are linked by complementary bonds between the nucleobases of different strands (A bonding with T and C bonding with G), encoding the same information of both strands." width="90%" />
<p class="caption">
Figure 1.1: <strong>Double-helix structure of DNA.</strong><br />
Each strand of DNA has a phosphate-sugar backbone on which are attached nucleobases. The two strands are linked by complementary bonds between the nucleobases of different strands (A bonding with T and C bonding with G), encoding the same information of both strands.
</p>
</div>
<p>The amount of DNA necessary to encode the information varies greatly from organism to organism: 5400 base pairs (5.4kBp) for the <span class="math inline">\(\varphi X174\)</span> phage<span class="citation"><sup><a href="#ref-sangerNucleotideSequenceBacteriophage1977" role="doc-biblioref">2</a></sup></span>, 4.9MBp for <em>Escherichia coli</em><span class="citation"><sup><a href="#ref-archer2011" role="doc-biblioref">3</a></sup></span>, 3.1GBp for <em>Homo sapiens</em><span class="citation"><sup><a href="#ref-nurk2022" role="doc-biblioref">4</a></sup></span> all the way up to almost 150GBp for <em>Paris japonica,</em> a Japanese mountain flowering plant<span class="citation"><sup><a href="#ref-pellicer2010" role="doc-biblioref">5</a></sup></span>. While very small genome size tend to occur in smaller, simpler organisms, genome size does not correlate with organism complexity<span class="citation"><sup><a href="#ref-macgregor2001" role="doc-biblioref">6</a></sup></span>.</p>
</div>
<div id="from-information-to-action" class="section level3 hasAnchor" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> From Information to action<a href="what-is-sequence-data.html#from-information-to-action" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="proteins-their-structure-and-their-role" class="section level4 hasAnchor" number="1.1.2.1">
<h4><span class="header-section-number">1.1.2.1</span> Proteins, their structure and their role<a href="what-is-sequence-data.html#proteins-their-structure-and-their-role" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The double stranded DNA molecules present in the cells of a living organism contain information only; in order for the organism to live this information must be read and translated into actions. Most of the actions necessary for “life” are taken by large molecules called proteins, they have a very wide range of functions from catalyzing reactions in the cell to giving it its structure<span class="citation"><sup><a href="#ref-alberts2002" role="doc-biblioref">7</a></sup></span>.</p>
<p>Proteins are macromolecules that are made up of one or several chains of amino acids. These chains then link together and fold up in a specific three dimensional structure, giving the protein the shape it needs to fulfill its goal. This structure is determined by the sequence of amino acids, and a given protein can be identified by this amino acid sequence<span class="citation"><sup><a href="#ref-alberts2002" role="doc-biblioref">7</a></sup></span>.</p>
<p>This sequence is directly dependent on the information contained in the DNA. First the DNA is transcribed in a similar, but single stranded, molecule called RNA (Ribonucleic Acid) which encodes the same sequence. This RNA molecule is then translated into a protein by the following process<span class="citation"><sup><a href="#ref-crick1961" role="doc-biblioref">8</a></sup></span>:</p>
<ol style="list-style-type: decimal">
<li>Nucleotides in the RNA sequence are read in groups of three called codons.</li>
<li>These codons are read sequentially along the RNA molecule.</li>
<li>Each codon corresponds to an amino acid, according to the genetic code.</li>
<li>The sequence of codons in RNA <em>(and by extension DNA)</em> determines the sequence of amino acids.</li>
<li>The translation process is stopped when a specific type of codon <em>(a “Stop” codon)</em> is read.</li>
</ol>
<p>With four types of nucleotides and codons grouping three nucleotides there are <span class="math inline">\(4^3=64\)</span> possible codons. However, as stated above, proteins are only made up of 20 different amino acids, meaning that several different codons correspond to the same amino acid. This gives the translation process a certain robustness to errors that can occur when the DNA is copied to create a new cell, or when it is transformed into RNA prior to protein translation.</p>
<p>The portion of DNA that is read to create the protein is said to be “coding”, and is called a gene. There are several thousands of genes in the human genome<span class="citation"><sup><a href="#ref-internationalhumangenomesequencingconsortiumFinishingEuchromaticSequence2004" role="doc-biblioref">9</a></sup></span> resulting in proteins executing thousands of different functions in a cell. In human beings, coding DNA represents only 1% to 2% of the total genome<span class="citation"><sup><a href="#ref-elkonCharacterizationNoncodingRegulatory2017" role="doc-biblioref">10</a>,<a href="#ref-omennReflectionsHUPOHuman2021" role="doc-biblioref">11</a></sup></span>. The large majority of the DNA in a human being is not translated into proteins, a portion of it has a regulatory role, controlling transcription and translation, but the role remains unknown for the rest of the human genome<span class="citation"><sup><a href="#ref-shabalinaMammalianTranscriptomeFunction2004" role="doc-biblioref">12</a>,<a href="#ref-IntegratedEncyclopediaDNA2012" role="doc-biblioref">13</a></sup></span>.</p>
</div>
<div id="making-mistakes" class="section level4 hasAnchor" number="1.1.2.2">
<h4><span class="header-section-number">1.1.2.2</span> Making mistakes<a href="what-is-sequence-data.html#making-mistakes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Going from DNA sequence to protein is quite a complicated process involving several steps, it is therefore possible for a mistake to happen. There are several mechanisms to avoid mistakes and alteration of the genetic information: the complementary nature of the two strands of DNA, the redundant nature of the genetic code as well as error correction mechanisms in the molecules <em>(called “polymerases”)</em> that read and write DNA and RNA being some of them. Despite all that, some errors in the nucleic acid (DNA and RNA) or protein sequences still make it through, these are called mutations.</p>
<div id="where-can-mistakes-happen" class="section level5 hasAnchor" number="1.1.2.2.1">
<h5><span class="header-section-number">1.1.2.2.1</span> Where can mistakes happen ?<a href="what-is-sequence-data.html#where-can-mistakes-happen" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>There are several sources of error that can alter the genetic information<span class="citation"><sup><a href="#ref-chatterjee2017" role="doc-biblioref">14</a></sup></span>:</p>
<ul>
<li><p><strong>DNA replication:</strong> When a cell divides, or when an organism reproduces, the DNA molecule must be copied in order to preserve and transmit genetic information. This process has a very low rate of errors, with as low as one error for every billion to every hundred billion of replicated base pairs<span class="citation"><sup><a href="#ref-fijalkowska2012" role="doc-biblioref">15</a></sup></span>. This is due to the fact that the DNA polymerase (the protein responsible for copying DNA molecules), has a relatively low error rate to start with, but mostly to the error correcting mechanisms that are present in certain cells and bacteria<span class="citation"><sup><a href="#ref-pray2008dna" role="doc-biblioref">16</a></sup></span>.</p></li>
<li><p><strong>RNA transcription:</strong> Since errors in RNA transcripts are less important than in replicated DNA, RNA polymerases have a much higher error rate than their DNA counterparts. This error rate has been estimated to be between four errors for each million<span class="citation"><sup><a href="#ref-gout2013" role="doc-biblioref">17</a></sup></span> to two errors for each hundred thousand<span class="citation"><sup><a href="#ref-gout2017" role="doc-biblioref">18</a></sup></span> transcribed bases.</p></li>
<li><p><strong>Protein translation:</strong> The process of translating RNA to a protein is done by proteins called ribosomes. This is a very error prone process with a mistranslation rate estimated to be of the order of one error for every 10,000 codons translated<span class="citation"><sup><a href="#ref-shcherbakovRibosomalMistranslationLeads2019" role="doc-biblioref">19</a></sup></span></p></li>
<li><p><strong>Other mutagenic events:</strong> Many external events and factors have been shown to provoke mutations in exposed DNA such as Ionizing radiation<span class="citation"><sup><a href="#ref-desouky2015" role="doc-biblioref">20</a></sup></span>, UV rays<span class="citation"><sup><a href="#ref-kiefer2007" role="doc-biblioref">21</a></sup></span>, Toxins<span class="citation"><sup><a href="#ref-bennett2003" role="doc-biblioref">22</a></sup></span>, heat Stress<span class="citation"><sup><a href="#ref-kantidze2016" role="doc-biblioref">23</a></sup></span>, cold stress<span class="citation"><sup><a href="#ref-gregory1994" role="doc-biblioref">24</a></sup></span> or oxidative stress<span class="citation"><sup><a href="#ref-gafter-gvili2013" role="doc-biblioref">25</a></sup></span>.</p></li>
</ul>
</div>
<div id="what-kind-of-errors-are-possible" class="section level5 hasAnchor" number="1.1.2.2.2">
<h5><span class="header-section-number">1.1.2.2.2</span> What kind of errors are possible?<a href="what-is-sequence-data.html#what-kind-of-errors-are-possible" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>In biological sequences (nucleic acids and proteins), mutations can result from one of three error modes:</p>
<ul>
<li><strong>Substitutions</strong>, where the original base unit (nucleotide or amino acid) is mistakenly replaced by another one, for instance inserting an A instead of a G during RNA transcription.</li>
<li><strong>Insertions</strong>,where a new base unit not present in the original sequence is added to the newly synthesized biological sequence.</li>
<li><strong>deletions</strong>, where a base unit from the original sequence is skipped and not taken into account when synthesizing the new sequence.</li>
</ul>
<p>While these three types of errors occur both in nucleic acids and proteins there are some things to consider about the consequences of nucleic acid mutations on protein synthesis. Due to the redundant nature of the genetic code mentioned in Section <a href="what-is-sequence-data.html#proteins-their-structure-and-their-role">1.1.2.1</a>, some substitutions in the nucleic acid sequence will result in the same protein sequence and therefore not have altered protein activity. Some mutations however will result in a substitution at the amino acid level which could potentially lead to a physicochemically altered or even non-functional protein. Finally, insertion and deletion errors (collectively called indels) can have big consequences on resulting proteins. Inserting or deleting nucleotides in multiples of three will result in the insertion/deletion of amino acids in the protein, any other length of indel will result in what is called a frameshift mutation<span class="citation"><sup><a href="#ref-rothFrameshiftMutations1974" role="doc-biblioref">26</a></sup></span>. These mutations causes changes in all the codons, potentially resulting in a completely different amino-acid sequence, including premature stop codon apparition as shown in Figure <a href="what-is-sequence-data.html#fig:frameshift">1.2</a>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:frameshift"></span>
<img src="figures/Sequence-Intro/Frameshift.png" alt="**Effect of frameshift mutations.**  
The deletion of a single C (highlighted in red) in the original DNA sequence leads to a change in the codons read during translation. The original codons (shown in green, with corresponding amino acids, above the sequence) translate to a functional protein `MLIRG...`. The new codons caused by the deletion (shown in blue, with corresponding amino acids, below the sequence), induce a premature STOP codon leading to a non-functional protein `M`. The Serine and Valine codons are not translated due to the STOP codon. 
" width="50%" />
<p class="caption">
Figure 1.2: <strong>Effect of frameshift mutations.</strong><br />
The deletion of a single C (highlighted in red) in the original DNA sequence leads to a change in the codons read during translation. The original codons (shown in green, with corresponding amino acids, above the sequence) translate to a functional protein <code>MLIRG...</code>. The new codons caused by the deletion (shown in blue, with corresponding amino acids, below the sequence), induce a premature STOP codon leading to a non-functional protein <code>M</code>. The Serine and Valine codons are not translated due to the STOP codon.
</p>
</div>
</div>
<div id="what-effect-can-mutations-have" class="section level5 hasAnchor" number="1.1.2.2.3">
<h5><span class="header-section-number">1.1.2.2.3</span> What effect can mutations have ?<a href="what-is-sequence-data.html#what-effect-can-mutations-have" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>As we stated above, some mutations in DNA may have no repercussions, some others can lead to non-functional proteins. In some cases mutations can be associated with a trait in the mutated individual. For example a single mutation in a gene linked with coagulation can lead to pathological Leiden thrombophilia<span class="citation"><sup><a href="#ref-kujovichFactorLeidenThrombophilia2011" role="doc-biblioref">27</a></sup></span>, a single amino acid deletion in the CFTR protein leads to <em>(the very deadly)</em> cystic fibrosis<span class="citation"><sup><a href="#ref-cuttingCysticFibrosisGenetics2015" role="doc-biblioref">28</a></sup></span>, and many mutations have been linked to complex diseases like type 2 diabetes<span class="citation"><sup><a href="#ref-fuchsbergerGeneticArchitectureType2016" role="doc-biblioref">29</a>,<a href="#ref-morrisLargescaleAssociationAnalysis2012" role="doc-biblioref">30</a></sup></span>. All mutational effects are not necessarily bad for the organism though, and mutations are essential for bacteria<span class="citation"><sup><a href="#ref-woodfordEmergenceAntibioticResistance2007" role="doc-biblioref">31</a></sup></span> or viruses like HIV<span class="citation"><sup><a href="#ref-rheeHumanImmunodeficiencyVirus2003" role="doc-biblioref">32</a></sup></span> to develop resistance to treatment <em>(more on that in Chapters</em> <a href="viruses-hiv-and-drug-resistance.html#viruses-hiv-and-drug-resistance">5</a> <em>and</em> <a href="HIV-paper.html#HIV-paper">6</a><em>).</em></p>
<p>While some mutations, have had their mechanisms and consequences thoroughly studied, in many cases mutations are simply linked to a trait. Since it is easier to show correlation than causation, and that the former does not necessarily imply the latter, it is important to further study mutations of notice to understand their potential consequences.</p>
</div>
</div>
</div>
</div>
<div id="obtaining-sequence-data" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Obtaining sequence data<a href="what-is-sequence-data.html#obtaining-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In many fields, especially in computational biology, we need to know what genetic information the studied organism has. That is to say: what is the exact sequence of nucleotides that make up its DNA? The process of figuring out this sequence is, perhaps unsurprisingly, called sequencing. And a sequence that is produced by this process is called a <em>sequencing read</em> or, more commonly, just a <em>read</em>.</p>
<div id="sanger-sequencing-a-breakthrough" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Sanger sequencing, a breakthrough<a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The first sequencing method was developed in 1977<span class="citation"><sup><a href="#ref-sangerDNASequencingChainterminating1977" role="doc-biblioref">33</a></sup></span>. Sanger <em>et al.</em> devised a simple method to read the sequence of nucleotides that make up a DNA sequence known as <em>chain termination sequencing</em> or simply <em>Sanger sequencing</em> <em>(represented in Figure</em> <a href="what-is-sequence-data.html#fig:sanger">1.3</a><em>)</em>. Although this method is now mostly obsolete, it established some key concepts in sequencing, some of which are in action in the most modern sequencers.</p>
<p>To understand Sanger sequencing, one must first understand how to synthesize DNA. As we stated in Section <a href="what-is-sequence-data.html#what-is-dna">1.1.1</a>, DNA is built up from building blocks that we called nucleotides, more specifically deoxynucleotide triphosphates or dNTPs. These dNTPs are made up of a sugar (deoxyribose), a nucleobase (A, T, G or C) and 3 phosphate groups. By successively adding these dNTPs at the end of an existing DNA molecule, we extend it, linking one of the phospates of the dNTP to an oxygen atom on the last nucleotide of the DNA molecule. Let us now consider a dideoxynucleotide triphosphate (ddNTP), which is identical to a dNTP except we remove a specific oxygen atom. This ddNTP can be added to the growing molecule of DNA like regular dNTPs, but since it is missing that one oxygen atom no more dNTPs or ddNTPs can be added to the DNA molecule after this one. The elongation is terminated and we call these ddNTPs chain-terminators. This combination of DNA synthesis followed by termination are at the heart of Sanger sequencing.</p>
<p>It is important to note that dNTPs and ddNTPs refer to nucleotides with any nucleobase. We can refer to specific dNTPs by replacing the “N” with the base of choice. For example, dATP refers to the dNTP that has adenine as a base. Similarly we have dCTP, dGTP and dTTP (as well as ddATP, ddCTP, ddGTP and ddTTP).</p>
<ol style="list-style-type: decimal">
<li>The first step of Sanger sequencing (and most sequencing methods) is to amplify the DNA molecule we wish to sequence, <em>i.e.</em> make many copies of it (usually through a process called PCR). These clones of the sequence are then separated into their two complementary strands one of which will be used as a template for the sequencing steps.</li>
<li>The second step is to prepare 4 different sequencing environments <em>(think of it as 4 test tubes)</em>. In each environment we introduce an equal mix of the 4 dNTPs, that will be used to elongate new DNA molecules from the amplified templates, and a single type of ddNTP. So in the first test tube we will have only ddATP, ddCTP in the second, <em>et cetera.</em> In addition, these ddNTP are marked, at first with radioactive isotopes, and later on, as the technology matured, with dyes. This marking means that we can observe the location of these ddNTPs later on.</li>
<li>Then an equal portion of the template is introduced in each environment with DNA polymerases (that will add the nucleotides to elongate a sequence that is complementary to the template), and short specific DNA molecules called primers that are necessary for the polymerases to start synthesizing new DNA.</li>
<li>During synthesis the chain is elongated with dNTPs by the polymerase and the reaction stops once a ddNTP is incorporated. At the end of this process we have plenty of fragments of DNA in each test tube, and we know that these fragments end with a specific base in a given environment. For example, in the test tube where we added ddATP, we know that all the fragments end with an A, and that we have all the possible fragments that start at the beginning of the template and end with an A. If the template is AACTA, then the fragments we would get in the ddATP test tube would be A, AA, and AACTA.</li>
<li>Then, a sample from each environment is taken and deposited on a gel, each in its own lane. A process called electrophoresis is then used to separate the fragments according to their weight. By applying an electrical current to the gel, the fragments of DNA will migrate away from where they were deposited along their lane in the gel. Lighter, shorter DNA fragments will travel further than heavier ones. We then get clusters of fragments ordered by weight (and therefore by length) called bands. With the marked ddNTP we can reveal these bands in the gel.</li>
<li>We know that: 1) bands are ordered by length; 2) consecutive bands correspond to the addition of a single nucleotide; 3) in a specific lane fragments corresponding to a band end with a specific base. This knowledge is enough to deduce the sequence of the template. An example gel is shown in Figure <a href="what-is-sequence-data.html#fig:sanger">1.3</a>.</li>
</ol>
<p>This process allowed Sanger <em>et al.</em> to sequence the first genome, of a <span class="math inline">\(\varphi X174\)</span> bacteriophage, in 1977<span class="citation"><sup><a href="#ref-sangerNucleotideSequenceBacteriophage1977" role="doc-biblioref">2</a></sup></span>. Although revolutionary, this method was costly, time consuming and labor intensive. Adjustments to this method were made in order to make it faster and less expensive. An important step was to change the way ddNTPs were marked. By using fluorescent markers, each base having a distinct “color”, we can eliminate the need to have 4 different environments and lanes in the gel<span class="citation"><sup><a href="#ref-smithSynthesisOligonucleotidesContaining1985" role="doc-biblioref">34</a>,<a href="#ref-smithFluorescenceDetectionAutomated1986" role="doc-biblioref">35</a></sup></span>. This also paved the way for automating sequencing, each fluorescently marked band can be excited with a laser, and the resulting specific wavelength can be recorded by optical systems and the corresponding base automatically deduced<span class="citation"><sup><a href="#ref-ansorgeAutomatedDNASequencing1987" role="doc-biblioref">36</a></sup></span> (Also see Figure <a href="what-is-sequence-data.html#fig:sanger">1.3</a>). Other improvements were made such as using capillary electrophoresis instead of gel electrophoresis.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:sanger"></span>
<img src="figures/Sequence-Intro/Sanger.png" alt="**Overview of the sanger sequencing protocol.**  
**A)** The sequence to read and all the generated fragments, with highlighted ddNTP chain terminators, ordered by molecular weight (i.e. length). **B)** Classical Sanger sequencing. The fragments are separated by electrophoresis and the lighter fragments travel further from the wells at the botom of the gel. Each lane in the gel correpsonds to a specific ddNTP. The radioactivly marked ddNTPs appear as black band in the gel and we can reconstruct the sequence by reading the bands from top to bottom, the column in wich the band appears indicating which base is at each position. **C)** Automated Sanger sequencing. The fragments are also separated by electrophoresis, as in panel B. Chain terminators are marked with fluorescent markers. When excited by a laser, each ddNTP emits a specific wavelength. This is read by an optical sensor and the corresponding ddNTP is recorded. By exciting each band we can quickly deduce the sequence. 
" width="80%" />
<p class="caption">
Figure 1.3: <strong>Overview of the sanger sequencing protocol.</strong><br />
<strong>A)</strong> The sequence to read and all the generated fragments, with highlighted ddNTP chain terminators, ordered by molecular weight (i.e. length). <strong>B)</strong> Classical Sanger sequencing. The fragments are separated by electrophoresis and the lighter fragments travel further from the wells at the botom of the gel. Each lane in the gel correpsonds to a specific ddNTP. The radioactivly marked ddNTPs appear as black band in the gel and we can reconstruct the sequence by reading the bands from top to bottom, the column in wich the band appears indicating which base is at each position. <strong>C)</strong> Automated Sanger sequencing. The fragments are also separated by electrophoresis, as in panel B. Chain terminators are marked with fluorescent markers. When excited by a laser, each ddNTP emits a specific wavelength. This is read by an optical sensor and the corresponding ddNTP is recorded. By exciting each band we can quickly deduce the sequence.
</p>
</div>
<p>These gradual improvements to the Sanger sequencing protocol, allowed to sequence longer and more accurate reads, with the latest technologies resulting in reads reaching 1 ,000 base pairs with an accuracy of 99.999%<span class="citation"><sup><a href="#ref-shendureNextgenerationDNASequencing2008" role="doc-biblioref">37</a></sup></span>. These improvements also resulted in a lower cost for sequencing, which was greatly decreased from around $1000 per base-pair<span class="citation"><sup><a href="#ref-collinsHumanGenomeProject2003" role="doc-biblioref">38</a></sup></span> to only $0.5 per kilobase<span class="citation"><sup><a href="#ref-shendureNextgenerationDNASequencing2008" role="doc-biblioref">37</a></sup></span>. Finally these technological improvements also increased the throughput of sequencing machines from around 1 kilobase per day<span class="citation"><sup><a href="#ref-collinsHumanGenomeProject2003" role="doc-biblioref">38</a></sup></span> to 120 kilobases per hour<span class="citation"><sup><a href="#ref-liuComparisonNextGenerationSequencing2012" role="doc-biblioref">39</a></sup></span>.</p>
<p>Despite these improvements, for ambitious endeavors such as the human genome project, sequencing was a massive undertaking: the first human genome is estimated to have cost between 500 million and 1 billion US dollars to sequence<span class="citation"><sup><a href="#ref-CostSequencingHuman" role="doc-biblioref">40</a></sup></span>.</p>
</div>
<div id="next-generation-sequencing" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Next-generation sequencing<a href="what-is-sequence-data.html#next-generation-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Thanks to these large sequencing projects and the genomics field in general, the richness and usefulness of sequence data was made ever more apparent. This growing need of sequence data ushered in a new era of sequencing with the development of many new sequencing methods designed to have a higher throughput and a lower cost than Sanger sequencing. This second generation of sequencing technologies is often referred to as <em>Next-Generation Sequencing</em> (NGS) or <em>Massively parallel</em> sequencing. While there are different technologies, they share a few common key points<span class="citation"><sup><a href="#ref-metzkerSequencingTechnologiesNext2010" role="doc-biblioref">41</a></sup></span>:</p>
<ul>
<li><p>As with Sanger sequencing, we first need to amplify and clone the DNA template, however since these technologies result in shorter reads than Sanger sequencing, the DNA we want to sequence must first be randomly broken up into small template fragments before being amplified.</p></li>
<li><p>The amplified template fragments are attached to some sort of solid support, resulting in a physical support with billions of template fragments attached to it.</p></li>
<li><p>As in Sanger sequencing, DNA molecules, complementary to the template fragments, are elongated. This happens for billions of fragments at the same time (hence the “massively parallel” epithet).</p></li>
<li><p>The addition of specific nucleotides to a chain are detected in real time, and there is no permanent chain termination. There is no need for the long step of electrophoresis. These detections are simultaneous for all the molecules being elongated at once.</p></li>
</ul>
<p>The result of these steps is a very large number of short reads. With data analysis these short reads can be used to deduce longer sequences and eventually a fragmented approximation of the original whole genome sequence through a process called <em>assembly</em>.</p>
<p>The main NGS method is called “sequencing by synthesis”, developed by a company: Illumina. It is commonly referred to as <em>Illumina sequencing</em>. This method is based on <em>reversible chain terminators</em>, developed at the Institut Pasteur in the 90’s<span class="citation"><sup><a href="#ref-canardDNAPolymeraseFluorescent1994" role="doc-biblioref">42</a></sup></span>. These are marked dNTPs that can be used to elongate DNA molecules, but that have an additional molecular group that makes them terminators by default. However this terminating group can be removed once the NTP is included in a DNA molecule allowing the elongation process to continue. These dNTPs are fluorescently marked and when excited with a laser they emit light with a distinctive color. During Illumina sequencing, these reversible chain terminators are included to millions of fragments at the same time, stopping elongation. At this point all the fragments are excited with a laser and an optical system takes a picture of the emitted colors for all the fragments as once. In this image, a pixel loosely corresponds to a sequenced fragment, and its color to the most recently added dNTP. The terminating groups are then cleaved and the process can start over by incorporating a new batch of reversible terminators. By observing the successive images we can deduce the sequence of added nucleotides for each sequenced fragment and obtain all of our reads.</p>
<p>Another NGS method is called pyrosequencing, commercialized by 454 Life Sciences. Contrary to Illumina sequencing, this method does not use reversible chain terminators. Instead it uses a special enzyme called luciferase that emits light as specific dNTPs are added. This process is repeated for the 4 dNTPs (similarly to Sanger sequencing) and from the light emissions we can deduce the sequence of nucleotides<span class="citation"><sup><a href="#ref-nyrenSolidPhaseDNA1993" role="doc-biblioref">43</a></sup></span>.</p>
<p>These technologies yield reads around 150 nucleotides for Illumina and 400nt for pyrosequencing<span class="citation"><sup><a href="#ref-mardisDecadePerspectiveDNA2011" role="doc-biblioref">44</a></sup></span>, this is much shorter than the 1kB reads obtainable from the latest Sanger sequencing technologies. However the throughputs are much higher<span class="citation"><sup><a href="#ref-liuComparisonNextGenerationSequencing2012" role="doc-biblioref">39</a></sup></span>: 2.5 to 12.5 Gigabases per hour for Illumina and 30 Megabases per hour for pyrosequencing. Costs are also quite low: $0.07 and $10 per Megabase for Illumina and pyrosequencing respectively. The per-base sequencing accuracies are also quite high, up to 99.9% for both Illumina<span class="citation"><sup><a href="#ref-stolerSequencingErrorProfiles2021" role="doc-biblioref">45</a></sup></span> and pyrosequencing<span class="citation"><sup><a href="#ref-liuComparisonNextGenerationSequencing2012" role="doc-biblioref">39</a></sup></span>. A summary of the key characteristics for various sequencing technologies can be found in Table <a href="what-is-sequence-data.html#tab:sequencing">1.1</a>. The lower cost and higher throughput has made the Illumina sequencing technology the dominant one. The company estimates that 90% of the world’s sequencing data was generated with Illumina machines in 2015<span class="citation"><sup><a href="#ref-SequencingTechnologySequencing" role="doc-biblioref">46</a></sup></span>.</p>
</div>
<div id="long-read-sequencing" class="section level3 hasAnchor" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> Long read sequencing<a href="what-is-sequence-data.html#long-read-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Although NGS technologies revolutionized the sequencing world, recent efforts have been made to get longer reads. These third-generation methods generate reads of tens of kilobases and are commonly called <em>long-read sequencing</em> method. Long reads have a host of applications<span class="citation"><sup><a href="#ref-pollardLongReadsTheir2018" role="doc-biblioref">47</a></sup></span> for which short NGS reads might not be well suited: <em>De novo</em> assembly of large complex genomes, studying complex repetitive regions such as centromeres or telomeres or detection of structural variants. They have recently been used to assemble the first truly complete human genome, including telomeric and centromeric regions<span class="citation"><sup><a href="#ref-nurk2022" role="doc-biblioref">4</a></sup></span>.</p>
<p>The two available long read technologies are: Single Molecule Real Time sequencing (SMRT), commercialized by Pacific Biosciences (PacBio) and Nanopore sequencing, commercialized by Oxford Nanopore Technologies (ONT). While these technologies are quite different, they both result in much longer reads than even Sanger sequencing in real time, without the need for chain terminators or separate sequencing reactions, all with a high throughput and at a reasonably low cost.</p>
<p>SMRT sequencing was first developed in 2009<span class="citation"><sup><a href="#ref-eidRealTimeDNASequencing2009" role="doc-biblioref">48</a></sup></span>, before being commercialized and furthered by PacBio. The basic principle is as follows:</p>
<ol style="list-style-type: decimal">
<li>Fragment and amplify DNA to obtain a very large number of DNA templates.</li>
<li>Link both strands of each DNA template together with known sequences called <em>bell adapters</em>. Denature the DNA to create a single stranded, circular DNA molecule.</li>
<li>Primers and polymerases are attached to the circular molecule specifically on one of the bell adapters.</li>
<li>Add the circular DNA template, primer, polymerases complexes to a SMRT chip. This chip is essentially a large aluminium surface with hundreds of thousands of microscopic wells called <em>Zero-Mode Waveguides</em> (ZMWs) only 100nm in diameter<span class="citation"><sup><a href="#ref-leveneZeroModeWaveguidesSingleMolecule2003" role="doc-biblioref">49</a></sup></span>. The polymerases are chemically bonded to the bottom of each of these ZMWs so we effectively get a single DNA template and polymerase per well.</li>
<li>Fluorescently marked dNTPs are incorporated progressively in each of the wells. When a marked dNTP is incorporated in the newly synthesized DNA brand, light of a specific wavelength is emitted.</li>
<li>The size of these ZMWs make the detection of the fluorescence possible with an optical system. Incorporation of dNTPs in each ZMW can be detected simultaneously in a parallel fashion and the resulting sequences deduced.</li>
</ol>
<p>Nanopore sequencing, thought of in the eighties, further developed along the years<span class="citation"><sup><a href="#ref-clarkeContinuousBaseIdentification2009" role="doc-biblioref">50</a></sup></span> and first commercialized by ONT in 2014<span class="citation"><sup><a href="#ref-deamerThreeDecadesNanopore2016" role="doc-biblioref">51</a></sup></span>, is completely different from all the sequencing technologies previously mentioned. Where all the other ones are based on synthesizing a complementary DNA strand and detecting specific dNTP incorporation in some way or another, there is no synthesis in nanopore sequencing. The principle relies on feeding a single strand of a DNA template through a small hole in a membrane, a <em>nanopore</em>, at a controlled speed. As the nucleotides go through the nanopore, an electric current is formed between both sides of the membrane. This current can be measured and is specific to the succession of 5 to 6 nucleotides inside the nanopore channel at any given time. By looking at the evolution of the electric current as the DNA strand goes through the nanopore, we can deduce the sequence of nucleotides through a process called <em>base calling.</em> Base calling is usually done with machine learning methods, mainly artificial neural networks<span class="citation"><sup><a href="#ref-wickPerformanceNeuralNetwork2019" role="doc-biblioref">52</a></sup></span>. In the flow cells used in ONT sequencers, there are hundreds of thousands of nanopores, spread out over a synthetic membrane, allowing for massively parallel sequencing as well. Theoretically, since this method is not based on synthesis, the upper limit for read length is only limited by the length of the template, and in practice ONT sequencing produces the longest reads.</p>
<p>Both technologies yield long reads, the median and highest read lengths being 10 kilobases and 60 kilobases respectively for PacBio sequencing<span class="citation"><sup><a href="#ref-rhoadsPacBioSequencingIts2015" role="doc-biblioref">53</a></sup></span>. For nanopore the median read lengths of 10 to 12 kilobases<span class="citation"><sup><a href="#ref-ipMinIONAnalysisReference2015" role="doc-biblioref">54</a>,<a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">55</a></sup></span> are similar to PacBio, but in it can also yield ultra-long reads of 1 up to 2.3 megabases long<span class="citation"><sup><a href="#ref-jainNanoporeSequencingAssembly2018" role="doc-biblioref">56</a>–<a href="#ref-payneBulkVisGraphicalViewer2019" role="doc-biblioref">58</a></sup></span>. The length of the reads and parallel nature of these two technologies allow these sequencers to have truly massive throughputs. PacBio sequencers can sequence between 2 and 11 gigabases per hour and ONT from 12.5 gigabases per hour, up to a staggering 260 gigabases per hour using the latest ONT PromethION machines<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">55</a></sup></span>. The cost of sequencing with these machines, while higher than for Illumina sequencers, remains reasonably affordable at $0.32 and $0.13 per megabase for PacBio and ONT respectively<span class="citation"><sup><a href="#ref-murigneuxComparisonLongreadMethods2020" role="doc-biblioref">59</a></sup></span>. These characteristics are summarized in Table <a href="what-is-sequence-data.html#tab:sequencing">1.1</a> along with other sequencing technologies.</p>
<p>The length, throughput and sequencing cost of both these technologies paint a pretty picture, and indeed they have proved useful in many settings, however sequencing accuracy is a problem with these technologies. The per-base sequencing accuracy has been estimated to be between 85% and 92% for PacBio sequencers and 87% to 98% for ONT machines<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">55</a>,<a href="#ref-chaissonResolvingComplexityHuman2015" role="doc-biblioref">60</a>,<a href="#ref-jainOxfordNanoporeMinION2016" role="doc-biblioref">61</a></sup></span>. This accuracy is much lower than either Sanger sequencing or Illumina reads. Characterizing, correcting and accounting for these errors is widely studied and it will be discussed in more detail in Sections <a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them">1.3</a> and <a href="what-is-sequence-data.html#the-special-case-of-homopolymers">1.4</a>.</p>
<template id="b0fc1b42-9bd2-4570-9f28-f97b5b91f5f7"><style>
.tabwid table{
  border-spacing:0px !important;
  border-collapse:collapse;
  line-height:1;
  margin-left:auto;
  margin-right:auto;
  border-width: 0;
  display: table;
  margin-top: 1.275em;
  margin-bottom: 1.275em;
  border-color: transparent;
}
.tabwid_left table{
  margin-left:0;
}
.tabwid_right table{
  margin-right:0;
}
.tabwid td {
    padding: 0;
}
.tabwid a {
  text-decoration: none;
}
.tabwid thead {
    background-color: transparent;
}
.tabwid tfoot {
    background-color: transparent;
}
.tabwid table tr {
background-color: transparent;
}
</style><div class="tabwid"><style>.cl-ce154c9e{table-layout:auto;width:100%;}.cl-ce0f4ec0{font-family:'Helvetica';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-ce0f4ee8{font-family:'Helvetica';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;bottom:3.3pt;}.cl-ce0f6432{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-ce0f96aa{background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ce0f96b4{background-color:transparent;vertical-align: middle;border-bottom: 2pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-ce0f96be{background-color:transparent;vertical-align: middle;border-bottom: 2pt solid rgba(102, 102, 102, 1.00);border-top: 2pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table class='cl-ce154c9e'><thead><tr style="overflow-wrap:break-word;"><td class="cl-ce0f96be"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">technology</span></p></td><td class="cl-ce0f96be"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">read length (nt)</span></p></td><td class="cl-ce0f96be"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">throughput (nt/hour)</span></p></td><td class="cl-ce0f96be"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">cost ($/Mb)</span></p></td><td class="cl-ce0f96be"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">accuracy</span></p></td></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">Sanger</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">1000</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">120</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">10</span><span class="cl-ce0f4ee8">3</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">$500</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">99.999%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">Illumina</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">150</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">2.5-12.5</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">10</span><span class="cl-ce0f4ee8">9</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">$0.07</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">99.9%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">Pyrosequencing</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">400</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">30</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">10</span><span class="cl-ce0f4ee8">6</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">$10</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">99.9%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">PacBio</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">SMRT</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">10000</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">(up</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">to</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">60000)</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">2-11</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">10</span><span class="cl-ce0f4ee8">9</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">$0.32</span></p></td><td class="cl-ce0f96aa"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">85-92%</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-ce0f96b4"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">Nanopore</span></p></td><td class="cl-ce0f96b4"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">12000</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">(up</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">to</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">2.5</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">10</span><span class="cl-ce0f4ee8">6</span><span class="cl-ce0f4ec0">)</span></p></td><td class="cl-ce0f96b4"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">12.5-260</span><span class="cl-ce0f4ec0"> </span><span class="cl-ce0f4ec0">10</span><span class="cl-ce0f4ee8">9</span></p></td><td class="cl-ce0f96b4"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">$0.13</span></p></td><td class="cl-ce0f96b4"><p class="cl-ce0f6432"><span class="cl-ce0f4ec0">87-98%</span></p></td></tr></tbody>
<caption class="caption">
<p><span id="tab:sequencing">Table 1.1: </span><strong>Comparison of sequencing technology characteristics.</strong><br />
Characteristics for the latest sequencers were used for the Sanger sequencing entry. The length is given in nucleotides, throughputs in sequenced nuctleotides per hour and cost in US dollars per megabase.</p>
</caption>
</table></div></template>
<div class="flextable-shadow-host" id="2968cbd2-5410-413d-9ec0-0dd8e8911cd4"></div>
<script>
var dest = document.getElementById("2968cbd2-5410-413d-9ec0-0dd8e8911cd4");
var template = document.getElementById("b0fc1b42-9bd2-4570-9f28-f97b5b91f5f7");
var caption = template.content.querySelector("caption");
if(caption) {
  caption.style.cssText = "display:block;text-align:center;";
  var newcapt = document.createElement("p");
  newcapt.appendChild(caption)
  dest.parentNode.insertBefore(newcapt, dest.nextSibling);
}
var fantome = dest.attachShadow({mode: 'open'});
var templateContent = template.content;
fantome.appendChild(templateContent);
</script>

<p>While most of the mentioned technologies can also be adapted and used to sequence RNA instead of DNA<span class="citation"><sup><a href="#ref-hongRNASequencingNew2020" role="doc-biblioref">62</a>,<a href="#ref-ozsolakRNASequencingAdvances2011" role="doc-biblioref">63</a></sup></span>, directly sequencing proteins remains a challenge. The sequence of amino acids making up a protein is usually deduced from the codons in sequenced DNA or RNA after detection of potentially coding regions called open reading frames (ORFs). Development of methods to directly sequence protein molecules using mass spectrometry was started not very long after Sanger sequencing<span class="citation"><sup><a href="#ref-huntProteinSequencingTandem1986" role="doc-biblioref">64</a></sup></span> and improved<span class="citation"><sup><a href="#ref-smithProteinSequencingProtocols2002" role="doc-biblioref">65</a></sup></span>. New methods are still being developed<span class="citation"><sup><a href="#ref-restrepo-perezPavingWaySinglemolecule2018" role="doc-biblioref">66</a></sup></span> but protein sequencing remains a challenge.</p>
</div>
</div>
<div id="sequencing-errors-how-to-account-for-them" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Sequencing errors, how to account for them ?<a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sequencing technologies are not perfect, they make errors, as we can see from the accuracy rates reported in Section <a href="what-is-sequence-data.html#obtaining-sequence-data">1.2</a>. For technologies based on nucleic acid synthesis (i.e. everything except ONT), since they use polymerases it stands to reason that the same three types of errors, described in Section <a href="what-is-sequence-data.html#making-mistakes">1.1.2.2</a>, occur: substitutions, insertions and deletions. For long read technologies though, most of the errors do not come from the polymerase, but from signal processing used to deduce the sequence. Since both technologies execute single molecule sequencing, the signal to noise ratio is low<span class="citation"><sup><a href="#ref-weirather2017" role="doc-biblioref">67</a>,<a href="#ref-wangNanoporeSequencingTechnology2021" role="doc-biblioref">68</a></sup></span> making base calling more complicated.</p>
<p>This explains the discrepancy in error rates between short and long read sequencing technologies: the former getting as low as 10<sup>-4</sup> or 10<sup>-5</sup> after computational processing<span class="citation"><sup><a href="#ref-maAnalysisErrorProfiles2019" role="doc-biblioref">69</a></sup></span> where the latter are between 10% and 15%. This high error rate long reads is bothersome and many efforts have been made lower this error rate, computationally and technologically.</p>
<div id="error-correction-methods" class="section level3 hasAnchor" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Error correction methods<a href="what-is-sequence-data.html#error-correction-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The long read error-correction literature and toolset is rich and active<span class="citation"><sup><a href="#ref-limaComparativeAssessmentLongread2020" role="doc-biblioref">70</a>–<a href="#ref-zhangComprehensiveEvaluationLong2020" role="doc-biblioref">72</a></sup></span>. There are two main ways to correct errors: 1) hybrid methods where high-accuracy short reads are used, and 2) non-hybrid methods where only the long-reads are used.</p>
<p>In Non-hybrid methods<span class="citation"><sup><a href="#ref-limaComparativeAssessmentLongread2020" role="doc-biblioref">70</a>,<a href="#ref-amarasingheOpportunitiesChallengesLongread2020" role="doc-biblioref">73</a></sup></span>, by finding regions that overlap fairly well between reads and taking the consensus of the overlapped regions (i.e. the majority nucleotide at each position), some errors can be eliminated. In many analyses and sequencing data processing pipelines, the first step is to break up the reads into all possible overlapping subsequences of length <span class="math inline">\(k\)</span> called <span class="math inline">\(k\)</span>-mers (e.g the 3-mers of ATTGC are ATT, TTG and TGC). Rare <span class="math inline">\(k\)</span>-mers in the read dataset, i.e. <span class="math inline">\(k\)</span>-mers that appear only a handful of times in all the reads, are likely the result of an error and filtering them out can improve analysis. One or both of these procedures are implemented in several pieces of commonly used software such as assembler like <code>wtdbg2</code><span class="citation"><sup><a href="#ref-ruanFastAccurateLongread2020" role="doc-biblioref">74</a></sup></span>, and <code>canu</code><span class="citation"><sup><a href="#ref-korenCanuScalableAccurate2017" role="doc-biblioref">75</a></sup></span> or standalone long-read correctors like <code>daccord</code><span class="citation"><sup><a href="#ref-tischlerNonHybridLong2017" role="doc-biblioref">76</a></sup></span>. In some cases, errors are corrected not on the raw reads but after having assembled the long reads into long continuous sequences (contigs), this process is called polishing. The <code>ntEdit</code> polisher<span class="citation"><sup><a href="#ref-warrenNtEditScalableGenome2019" role="doc-biblioref">77</a></sup></span> also filters out rare <span class="math inline">\(k\)</span>-mers to correct errors. The <code>Arrow</code><span class="citation"><sup><a href="#ref-heplerImprovedCircularConsensus2016" role="doc-biblioref">78</a></sup></span> and <code>Nanopolish</code><span class="citation"><sup><a href="#ref-simpsonDetectingDNACytosine2017" role="doc-biblioref">79</a></sup></span> polishers correct the assembly using the raw PacBio and ONT long reads respectively.</p>
<p>Hybrid methods, as their name suggest, make use of short reads to correct errors in long reads. By finding similar regions between the short and long reads we can use the higher accuracy of short reads to correct the long ones. This is implemented in many pieces of software <code>proovread</code><span class="citation"><sup><a href="#ref-hacklProovreadLargescaleHighaccuracy2014" role="doc-biblioref">80</a></sup></span>, <code>Jabba</code><span class="citation"><sup><a href="#ref-miclotteJabbaHybridError2016" role="doc-biblioref">81</a></sup></span>, <code>PBcR</code><span class="citation"><sup><a href="#ref-korenHybridErrorCorrection2012" role="doc-biblioref">82</a></sup></span> or <code>LoRDEC</code><span class="citation"><sup><a href="#ref-salmelaLoRDECAccurateEfficient2014" role="doc-biblioref">83</a></sup></span>. Short reads can also be used to polish long read assemblies with tools like <code>Pilon</code><span class="citation"><sup><a href="#ref-walkerPilonIntegratedTool2014" role="doc-biblioref">84</a></sup></span> or <code>Racon</code><span class="citation"><sup><a href="#ref-vaserFastAccurateNovo2017" role="doc-biblioref">85</a></sup></span>. The first complete human genome was assembled and polished using many different sequencing technologies including PacBio, ONT and Illumina technologies<span class="citation"><sup><a href="#ref-nurk2022" role="doc-biblioref">4</a></sup></span>.</p>
</div>
<div id="more-accurate-sequencing-methods" class="section level3 hasAnchor" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> More accurate sequencing methods<a href="what-is-sequence-data.html#more-accurate-sequencing-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>While a lot of effort is being put into error correction, another angle of attack to lower the error rate of long reads is to improve the sequencing technology.</p>
<p>In 2019, PacBio introduced HiFi reads, based on a circular consensus (CCS) technique<span class="citation"><sup><a href="#ref-wenger2019" role="doc-biblioref">86</a></sup></span>. During SRMT sequencing the 2 strands are linked together by bell adapters to form a circular DNA template (c.f. Section <a href="what-is-sequence-data.html#long-read-sequencing">1.2.3</a>), the central idea of CCS is to sequence this molecule multiple times by going over the circle more than once. In the resulting long sequence the known bell adapter sequences can be removed, and a consensus sequence can be built from the multiple passes over the same DNA template. This results in long-read accuracies of 99.8% to 99.9%<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">55</a>,<a href="#ref-wenger2019" role="doc-biblioref">86</a></sup></span>. This works because PacBio sequencing errors are mostly randomly distributed along the sequenced template (more on that in Section <a href="what-is-sequence-data.html#homopolymers-and-long-reads">1.4.2</a>), therefore it is unlikely that the same error will appear in multiple passes over the same template portion.</p>
<p>For ONT sequencing, most improvement efforts have been focused on base-callers. These tools were originally based on Hidden Markov Models<span class="citation"><sup><a href="#ref-timpDNABaseCallingNanopore2012" role="doc-biblioref">87</a></sup></span> (HMMs), but gradually they have been shifting over to neural network based deep learning methods<span class="citation"><sup><a href="#ref-wickPerformanceNeuralNetwork2019" role="doc-biblioref">52</a>,<a href="#ref-amarasingheOpportunitiesChallengesLongread2020" role="doc-biblioref">73</a>,<a href="#ref-peresiniNanoporeBaseCalling2021" role="doc-biblioref">88</a>,<a href="#ref-bozaDeepNanoDeepRecurrent2017" role="doc-biblioref">89</a></sup></span> with faster inference times and better performance.</p>
<p>Similarly to PacBio HiFi reads, ONT developed 2D, and 1D<sup>2</sup> sequencing. In 2D sequencing both strands of the DNA molecule to sequence are linked with a hairpin adapter to form one long sequence. Each strand is sequenced once and a consensus is built from these 2 passes<span class="citation"><sup><a href="#ref-tylerEvaluationOxfordNanopore2018" role="doc-biblioref">90</a></sup></span>. 1D<sup>2</sup> sequencing operates in a similar fashion but without the need for a hairpin adapter<span class="citation"><sup><a href="#ref-linNanoporeTechnologyIts2021" role="doc-biblioref">91</a></sup></span>. 2D sequencing produces reads with 97% accuracy albeit much shorter than standard 1D sequencing<span class="citation"><sup><a href="#ref-tylerEvaluationOxfordNanopore2018" role="doc-biblioref">90</a></sup></span>. Recently, Oxford Nanopore Technologies announced the release of a new technology they call duplex. Using new chemistry, a new basecaller and sequencing of both strands (similarly to 2D and 1D<sup>2</sup>) they announce raw read accuracies of 99.3%<span class="citation"><sup><a href="#ref-OxfordNanoporeTech" role="doc-biblioref">92</a></sup></span>. Pre-printed research seems to confirm these numbers with one experiment yielding duplex reads with a 99.9% accuracy<span class="citation"><sup><a href="#ref-sandersonComparisonR9Kit102022" role="doc-biblioref">93</a></sup></span>.</p>
<p>A technologically agnostic method using unique molecular identifiers added during the template preparation phase, and consensus sequencing has been shown, in specific contexts, to improve the accuracies of both ONT and PacBio CCS long reads to 99.59% and 99.93% respectively<span class="citation"><sup><a href="#ref-karstHighaccuracyLongreadAmplicon2021" role="doc-biblioref">94</a></sup></span>.</p>
<p>Finally, new sequencing technologies are being developed, like built in error-correction short-read technologies yielding error-free reads of up to 200 nucleotides long<span class="citation"><sup><a href="#ref-chenHighlyAccurateFluorogenic2017" role="doc-biblioref">95</a></sup></span>. Illumina also recently announced its own high-throughput, high-accuracy long-read sequencing technology in 2022<span class="citation"><sup><a href="#ref-HighPerformanceLong" role="doc-biblioref">96</a></sup></span>, although details about the performance and technology are scarce.</p>
</div>
</div>
<div id="the-special-case-of-homopolymers" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> The special case of homopolymers<a href="what-is-sequence-data.html#the-special-case-of-homopolymers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Despite improvement in error correction methods and sequencing technologies, certain genetic patterns are particularly difficult to process, homopolymers are one such pattern.</p>
<div id="homopolymers-and-the-human-genome" class="section level3 hasAnchor" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Homopolymers and the human genome<a href="what-is-sequence-data.html#homopolymers-and-the-human-genome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>Homopolymers</em> consist of a stretch of repeated nucleotides (i.e. <span class="math inline">\(\geq 2\)</span>) occurring at some point in the genome. For example the sequence AAAA is a length 4 adenine homopolymer. In the complete human genome assembly (CHM13 v1.1 from the T2T consortium<span class="citation"><sup><a href="#ref-nurk2022" role="doc-biblioref">4</a></sup></span>), 50% of its three gigabases are in homopolymers of size 2 or more, and 10% are in homopolymers of size 4 or more. As can be seen in Figure <a href="what-is-sequence-data.html#fig:HPpercent">1.4</a>, short and medium length homopolymers make up a significant part of the genome. In a previous GRCh38 human genome assembly, more than 1.9 megabases are in homopolymers of length 8 or higher<span class="citation"><sup><a href="#ref-booeshaghiPseudoalignmentFacilitatesAssignment2022" role="doc-biblioref">97</a></sup></span>, representing about 1‰ of that assembly. The longest homopolymer run in the CHM13 v1.1 assembly is 86 <em>(90 in GRCh38<span class="citation"><sup><a href="#ref-booeshaghiPseudoalignmentFacilitatesAssignment2022" role="doc-biblioref">97</a></sup></span>)</em>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HPpercent"></span>
<img src="_main_files/figure-html/HPpercent-1.png" alt="**Homopolymer fraction of the whole human genome by homopolymer length.**  
The homopolymer counts were calculated from the T2T consortium full human genome assembly CHM13 v1.1. This figure was inspired by Figure 3b of reference [@booeshaghiPseudoalignmentFacilitatesAssignment2022]." width="70%" />
<p class="caption">
Figure 1.4: <strong>Homopolymer fraction of the whole human genome by homopolymer length.</strong><br />
The homopolymer counts were calculated from the T2T consortium full human genome assembly CHM13 v1.1. This figure was inspired by Figure 3b of reference<span class="citation"><sup><a href="#ref-booeshaghiPseudoalignmentFacilitatesAssignment2022" role="doc-biblioref">97</a></sup></span>.
</p>
</div>
<p>In the human genome, homopolymers tend to occur more often in adenine and thymine runs than guanine and cytosine. There are are approximately twice as many nucleotides within A or T homopolymers (481 Mb and 484 Mb) than G or C (278 Mb and 279 Mb). This discrepancy is even more pronounced when looking at homopolymers longer than four nucleotides (c.f. Figure <a href="what-is-sequence-data.html#fig:HPdistrib">1.5</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:HPdistrib"></span>
<img src="_main_files/figure-html/HPdistrib-1.png" alt="**Distribution of homopolymer lengths per base in the human genome, for homopolymers of length $\geq$ 4.**  
The homopolymer counts were calculated from the T2T consortium full human genome assembly CHM13 v1.1." width="70%" />
<p class="caption">
Figure 1.5: <strong>Distribution of homopolymer lengths per base in the human genome, for homopolymers of length <span class="math inline">\(\geq\)</span> 4.</strong><br />
The homopolymer counts were calculated from the T2T consortium full human genome assembly CHM13 v1.1.
</p>
</div>
</div>
<div id="homopolymers-and-long-reads" class="section level3 hasAnchor" number="1.4.2">
<h3><span class="header-section-number">1.4.2</span> Homopolymers and long reads<a href="what-is-sequence-data.html#homopolymers-and-long-reads" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Unfortunately homopolymers are a source of errors in sequencing, particularly for long-read technologies: while substitutions seem to be randomly distributed along the reads for PacBio and ONT, the main error mode seems to be indels in homopolymeric sections, <em>i.e.</em> reading the same nucleotide several times or skipping over one of the repeated nucleotides. Many studies show that homopolymeric indels is the main type of error for PacBIO SMRT and ONT long-read sequencing<span class="citation"><sup><a href="#ref-weirather2017" role="doc-biblioref">67</a>,<a href="#ref-delahaye2021" role="doc-biblioref">98</a>–<a href="#ref-dohmBenchmarkingLongreadCorrection2020" role="doc-biblioref">100</a></sup></span>. This is even the case for PacBio HiFi reads, while the circular consensus approach eliminates the randomly distributed substitutions but homopolymer indels remain<span class="citation"><sup><a href="#ref-wenger2019" role="doc-biblioref">86</a></sup></span>. It seems that ONT reads are more prone to this type of error than PacBIo<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">55</a></sup></span>. The rate of these errors is independent of the length of the homopolymer for ONT, but it rises with homopolymer length for short-read and PacBio technologies<span class="citation"><sup><a href="#ref-fooxPerformanceAssessmentDNA2021" role="doc-biblioref">101</a></sup></span>.</p>
</div>
<div id="accounting-for-homopolymers" class="section level3 hasAnchor" number="1.4.3">
<h3><span class="header-section-number">1.4.3</span> Accounting for homopolymers<a href="what-is-sequence-data.html#accounting-for-homopolymers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The fact that they make up a significant part of the human genome, and that they are a source of errors for long read technologies means that homopolymers warrant special attention and care. Methods have been devised and implemented, specifically to counter homopolymer-linked errors.</p>
<div id="specific-error-correction" class="section level4 hasAnchor" number="1.4.3.1">
<h4><span class="header-section-number">1.4.3.1</span> Specific error correction<a href="what-is-sequence-data.html#specific-error-correction" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Homopolymer errors are taken under special consideration during assembly polishing when using certain tools like <code>HomoPolish</code><span class="citation"><sup><a href="#ref-huangHomopolishMethodRemoval2021" role="doc-biblioref">102</a></sup></span>, <code>NanoPolish</code><span class="citation"><sup><a href="#ref-simpsonDetectingDNACytosine2017" role="doc-biblioref">79</a></sup></span> or <code>Pilon</code><span class="citation"><sup><a href="#ref-walkerPilonIntegratedTool2014" role="doc-biblioref">84</a></sup></span>. Methods to improve base calling of homopolymer stretches have been developed for nanopore sequencing<span class="citation"><sup><a href="#ref-rangSquiggleBasepairComputational2018" role="doc-biblioref">103</a>,<a href="#ref-sarkozyCallingHomopolymerStretches2018" role="doc-biblioref">104</a></sup></span>, and implemented in state of the art base-callers such as <code>guppy</code> or <code>scrappie</code><span class="citation"><sup><a href="#ref-wickPerformanceNeuralNetwork2019" role="doc-biblioref">52</a></sup></span>.</p>
<p>Steps before sequencing can also be taken in order to reduce the effect of these errors, like avoiding homopolymers in barcode sequences<span class="citation"><sup><a href="#ref-hawkinsIndelcorrectingDNABarcodes2018" role="doc-biblioref">105</a>,<a href="#ref-srivathsanMinIONBasedPipeline2018" role="doc-biblioref">106</a></sup></span>, or during the development of DNA based storage systems<span class="citation"><sup><a href="#ref-wangConstructionBioConstrainedCode2019" role="doc-biblioref">107</a></sup></span>.</p>
<p>Improving the sequencing technologies can also be a solution, by reducing the number of homopolymer errors straight from the source. The latest ONT chemistry R.10 reportedly improves accuracy in homopolymer rich regions<span class="citation"><sup><a href="#ref-amarasingheOpportunitiesChallengesLongread2020" role="doc-biblioref">73</a>,<a href="#ref-R10NewestNanopore" role="doc-biblioref">108</a></sup></span>, non-biological solid-state nanopores also reduces errors in homopolymers<span class="citation"><sup><a href="#ref-zhouDetectionDNAHomopolymer2019" role="doc-biblioref">109</a>,<a href="#ref-gotoIdentificationFourSinglestranded2018" role="doc-biblioref">110</a></sup></span>.</p>
</div>
<div id="hpc-trick" class="section level4 hasAnchor" number="1.4.3.2">
<h4><span class="header-section-number">1.4.3.2</span> Homopolymer compression, a nifty trick<a href="what-is-sequence-data.html#hpc-trick" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In many cases, reads cannot be re-sequenced with newer technologies, or base-called with better base callers. Only the read sequences, potentially containing homopolymer errors, are available for usage. In order to account for this sort of error, a simple pre-processing trick was developed: <em>homopolymer compression</em> (HPC).</p>
<p>The idea is very simple: for any sequence, replace a repeated run of any nucleotide (i.e. homopolymers) by a single occurrence of that nucleotide. This means that after going through HPC the sequence AAACTGGG will yield the sequence ACTG. This simple pre-processing step, applied to all the reads and sequences to analyze removes all indels in homopolymers, and can resolve some ambiguities (c.f. Figure <a href="what-is-sequence-data.html#fig:hpcSchema">1.6</a>). It can also remove legitimate information contained in homopolymers, however the trade-off with the reduced error rate has been deemed advantageous.</p>
<p>HPC has been implemented in many sequence bioinformatics software tools. The <code>HiCanu</code><span class="citation"><sup><a href="#ref-nurkHiCanuAccurateAssembly2020" role="doc-biblioref">111</a></sup></span>, <code>MDBG</code><span class="citation"><sup><a href="#ref-ekimMinimizerspaceBruijnGraphs2021" role="doc-biblioref">112</a></sup></span>, <code>wtdbg2</code><span class="citation"><sup><a href="#ref-ruanFastAccurateLongread2020" role="doc-biblioref">74</a></sup></span>, <code>shasta</code><span class="citation"><sup><a href="#ref-shafinNanoporeSequencingShasta2020" role="doc-biblioref">113</a></sup></span> assemblers all use HPC under the hood to provide better assemblies, and HPC was used to assemble the complete human genome sequence<span class="citation"><sup><a href="#ref-nurk2022" role="doc-biblioref">4</a></sup></span>. The first published usage of HPC, was actually in the <code>CABOG</code> assembler<span class="citation"><sup><a href="#ref-millerAggressiveAssemblyPyrosequencing2008" role="doc-biblioref">114</a></sup></span> developed for pyrosequencing reads. HPC has also been implemented for other tasks, like clustering<span class="citation"><sup><a href="#ref-sahlinNovoClusteringLongRead2020" role="doc-biblioref">115</a></sup></span>, long read error correction with <code>LSC</code><span class="citation"><sup><a href="#ref-auImprovingPacBioLong2012" role="doc-biblioref">116</a></sup></span> and <code>LSCPlus</code><span class="citation"><sup><a href="#ref-huLSCplusFastSolution2016" role="doc-biblioref">117</a></sup></span>, alignment with <code>minimap2</code><span class="citation"><sup><a href="#ref-liMinimap2PairwiseAlignment2018" role="doc-biblioref">118</a></sup></span> and <code>winnowmap2</code><span class="citation"><sup><a href="#ref-jainWeightedMinimizerSampling2020" role="doc-biblioref">119</a></sup></span>, or specific analysis pipelines for satellite tandem repeats<span class="citation"><sup><a href="#ref-vannesteForensicSTRAnalysis2012" role="doc-biblioref">120</a></sup></span>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hpcSchema"></span>
<img src="figures/Sequence-Intro/Hpc.png" alt="**Homopolymer compression can help resolve ambiguities due to sequencing errors.**  
A read with homopolymer related sequencing errors can be homologous to two different regions of the reference genome, with one discrepancy for each region. After applying HPC, this ambiguity is properly accounted for and the read is homologous to only one region. This figure, however, only shows one way homopolymers can be detrimental and others are possible^[Homopolymer indels can be harmful in opposite circumstances as well. Let us consider, for example, a read that should correspond to several repetitions of a conserved motif. Homopolymer indels can artificially resolve an ambiguity by making the read unique and prefer a specific repetition of the motif or entirely misplace the read.].
" width="70%" />
<p class="caption">
Figure 1.6: <strong>Homopolymer compression can help resolve ambiguities due to sequencing errors.</strong><br />
A read with homopolymer related sequencing errors can be homologous to two different regions of the reference genome, with one discrepancy for each region. After applying HPC, this ambiguity is properly accounted for and the read is homologous to only one region. This figure, however, only shows one way homopolymers can be detrimental and others are possible<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.
</p>
</div>
</div>
</div>
</div>
<div id="conclusion" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Conclusion<a href="what-is-sequence-data.html#conclusion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I hope, after reading this chapter, you will agree with me that sequencing is fundamental for furthering our knowledge of biological processes, organisms and Life in general. And as such, the sequencing field is still very active with new technologies being developed, to improve the current technologies in various aspects: Illumina promising high accuracy long reads with Infinity<span class="citation"><sup><a href="#ref-HighPerformanceLong" role="doc-biblioref">96</a></sup></span>, PacBio developing its own short read sequencing technology, moving away from sequencing by synthesis<span class="citation"><sup><a href="#ref-ShortreadSequencingBinding" role="doc-biblioref">121</a>,<a href="#ref-cetinPlasmonicSensorCould2018" role="doc-biblioref">122</a></sup></span>. Finally efforts are also being made to make sequencing more affordable and available in a greater number settings with Ultima genomics promising accurate short reads for as low as $1 per gigabase<span class="citation"><sup><a href="#ref-almogyCostefficientWholeGenomesequencing2022" role="doc-biblioref">123</a></sup></span>.</p>
<p>With all these technological improvements we are approaching an era where sequencing is easy and quick, opening the door for massive projects like Tara Oceans<span class="citation"><sup><a href="#ref-sunagawaTaraOceansGlobal2020" role="doc-biblioref">124</a></sup></span> or the BioGenome project<span class="citation"><sup><a href="#ref-lewinEarthBioGenomeProject2018" role="doc-biblioref">125</a></sup></span> to better understand biodiversity. Routine whole-genome sequencing could also usher in an era personalized medicine<span class="citation"><sup><a href="#ref-lightbodyReviewApplicationsHighthroughput2019" role="doc-biblioref">126</a></sup></span>.</p>
<p>Despite all these advancements, sequencing errors remain an obstacle to certain analyses. This is particularly true for the ever more used and useful long reads, and the important fraction of genomes made up of homopolymers. Detecting, removing or accounting for these errors in some way is a crucial step to improve any analysis based on sequencing data, and to make sure that no theory or conclusion are built upon erroneous sequence data.</p>
<p>Finally, it is important to note (at least for the remainder of this thesis) that, from a computational standpoint, a biological sequence is simply a succession of letters and a set of reads is simply a text file. Therefore many analyses and data processing methods are inspired or directly transposed from the field of text algorithmics.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-watson1953" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Watson, J. D. &amp; Crick, F. H. C. <a href="https://doi.org/10.1101/SQB.1953.018.01.020">The Structure of Dna</a>. <em>Cold Spring Harbor Symposia on Quantitative Biology</em> <strong>18</strong>, 123–131 (1953).</div>
</div>
<div id="ref-sangerNucleotideSequenceBacteriophage1977" class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Sanger, F. <em>et al.</em> <a href="https://doi.org/10.1038/265687a0">Nucleotide sequence of bacteriophage <span class="nocase">φX174 DNA</span></a>. <em>Nature</em> <strong>265</strong>, 687–695 (1977).</div>
</div>
<div id="ref-archer2011" class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Archer, C. T. <em>et al.</em> <a href="https://doi.org/10.1186/1471-2164-12-9">The genome sequence of e. Coli w (ATCC 9637): Comparative genome analysis and an improved genome-scale reconstruction of e. coli</a>. <em>BMC Genomics</em> <strong>12</strong>, 9 (2011).</div>
</div>
<div id="ref-nurk2022" class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Nurk, S. <em>et al.</em> <a href="https://doi.org/10.1126/science.abj6987">The complete sequence of a human genome</a>. <em>Science</em> <strong>376</strong>, 44–53 (2022).</div>
</div>
<div id="ref-pellicer2010" class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Pellicer, J., Fay, M. F. &amp; Leitch, I. J. <a href="https://doi.org/10.1111/j.1095-8339.2010.01072.x">The largest eukaryotic genome of them all?</a> <em>Botanical Journal of the Linnean Society</em> <strong>164</strong>, 10–15 (2010).</div>
</div>
<div id="ref-macgregor2001" class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Macgregor, H. C. C-Value Paradox. in (eds. Brenner, S. &amp; Miller, J. H.) 249–250 (Academic Press, 2001). doi:<a href="https://doi.org/10.1006/rwgn.2001.0301">10.1006/rwgn.2001.0301</a>.</div>
</div>
<div id="ref-alberts2002" class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Alberts, B. <em>et al.</em> <em>Molecular Biology of the Cell. 4th edition</em>. (Garland Science, 2002). doi:<a href="https://doi.org/10.1201/9781315735368">10.1201/9781315735368</a>.</div>
</div>
<div id="ref-crick1961" class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Crick, F. H. C., Barnett, L., Brenner, S. &amp; Watts-Tobin, R. J. <a href="https://doi.org/10.1038/1921227a0">General Nature of the Genetic Code for Proteins</a>. <em>Nature</em> <strong>192</strong>, 1227–1232 (1961).</div>
</div>
<div id="ref-internationalhumangenomesequencingconsortiumFinishingEuchromaticSequence2004" class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">International Human Genome Sequencing Consortium. <a href="https://doi.org/10.1038/nature03001">Finishing the euchromatic sequence of the human genome</a>. <em>Nature</em> <strong>431</strong>, 931–945 (2004).</div>
</div>
<div id="ref-elkonCharacterizationNoncodingRegulatory2017" class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Elkon, R. &amp; Agami, R. <a href="https://doi.org/10.1038/nbt.3863">Characterization of noncoding regulatory DNA in the human genome</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 732–746 (2017).</div>
</div>
<div id="ref-omennReflectionsHUPOHuman2021" class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Omenn, G. S. <a href="https://doi.org/10.1016/j.mcpro.2021.100062">Reflections on the HUPO human proteome project, the flagship project of the human proteome organization, at 10 years</a>. <em>Molecular &amp; Cellular Proteomics : MCP</em> <strong>20</strong>, 100062 (2021).</div>
</div>
<div id="ref-shabalinaMammalianTranscriptomeFunction2004" class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Shabalina, S. A. &amp; Spiridonov, N. A. <a href="https://doi.org/10.1186/gb-2004-5-4-105">The mammalian transcriptome and the function of non-coding DNA sequences</a>. <em>Genome Biology</em> <strong>5</strong>, 105 (2004).</div>
</div>
<div id="ref-IntegratedEncyclopediaDNA2012" class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">ENCODE Project Consortium. <a href="https://doi.org/10.1038/nature11247">An integrated encyclopedia of DNA elements in the human genome</a>. <em>Nature</em> <strong>489</strong>, 57–74 (2012).</div>
</div>
<div id="ref-chatterjee2017" class="csl-entry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Chatterjee, N. &amp; Walker, G. C. <a href="https://doi.org/10.1002/em.22087">Mechanisms of DNA damage, repair, and mutagenesis: DNA Damage and Repair</a>. <em>Environmental and Molecular Mutagenesis</em> <strong>58</strong>, 235–263 (2017).</div>
</div>
<div id="ref-fijalkowska2012" class="csl-entry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Fijalkowska, I. J., Schaaper, R. M. &amp; Jonczyk, P. <a href="https://doi.org/10.1111/j.1574-6976.2012.00338.x">DNA replication fidelity in escherichia coli: A multi-DNA polymerase affair</a>. <em>FEMS microbiology reviews</em> <strong>36</strong>, 1105–1121 (2012).</div>
</div>
<div id="ref-pray2008dna" class="csl-entry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Pray, L. <a href="https://www.nature.com/scitable/topicpage/dna-replication-and-causes-of-mutation-409/">DNA replication and causes of mutation</a>. <em>Nature education</em> <strong>1</strong>, 214 (2008).</div>
</div>
<div id="ref-gout2013" class="csl-entry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Gout, J.-F., Thomas, W. K., Smith, Z., Okamoto, K. &amp; Lynch, M. <a href="https://doi.org/10.1073/pnas.1309843110">Large-scale detection of in vivo transcription errors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>110</strong>, 18584–18589 (2013).</div>
</div>
<div id="ref-gout2017" class="csl-entry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Gout, J.-F. <em>et al.</em> <a href="https://doi.org/10.1126/sciadv.1701484">The landscape of transcription errors in eukaryotic cells</a>. <em>Science Advances</em> <strong>3</strong>, e1701484 (2017).</div>
</div>
<div id="ref-shcherbakovRibosomalMistranslationLeads2019" class="csl-entry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Shcherbakov, D. <em>et al.</em> <a href="https://doi.org/10.1038/s42003-019-0626-9">Ribosomal mistranslation leads to silencing of the unfolded protein response and increased mitochondrial biogenesis</a>. <em>Communications Biology</em> <strong>2</strong>, 1–16 (2019).</div>
</div>
<div id="ref-desouky2015" class="csl-entry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Desouky, O., Ding, N. &amp; Zhou, G. <a href="https://doi.org/10.1016/j.jrras.2015.03.003">Targeted and non-targeted effects of ionizing radiation</a>. <em>Journal of Radiation Research and Applied Sciences</em> <strong>8</strong>, 247–254 (2015).</div>
</div>
<div id="ref-kiefer2007" class="csl-entry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Kiefer, J. Effects of Ultraviolet Radiation on DNA. in (eds. Obe, G. &amp; Vijayalaxmi) 39–53 (Springer, 2007). doi:<a href="https://doi.org/10.1007/978-3-540-71414-9_3">10.1007/978-3-540-71414-9_3</a>.</div>
</div>
<div id="ref-bennett2003" class="csl-entry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Bennett, J. W. &amp; Klich, M. <a href="https://doi.org/10.1128/CMR.16.3.497-516.2003">Mycotoxins</a>. <em>Clinical Microbiology Reviews</em> <strong>16</strong>, 497–516 (2003).</div>
</div>
<div id="ref-kantidze2016" class="csl-entry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Kantidze, O. L., Velichko, A. K., Luzhin, A. V. &amp; Razin, S. V. Heat stress-induced DNA damage. <em>Acta Naturae</em> <strong>8</strong>, 75–78 (2016).</div>
</div>
<div id="ref-gregory1994" class="csl-entry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Gregory, C. D. &amp; Milner, A. E. <a href="https://doi.org/10.1002/ijc.2910570321">Regulation of cell survival in Burkitt lymphoma: implications from studies of apoptosis following cold-shock treatment</a>. <em>International Journal of Cancer</em> <strong>57</strong>, 419–426 (1994).</div>
</div>
<div id="ref-gafter-gvili2013" class="csl-entry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Gafter-Gvili, A. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0068341">Oxidative stress-induced DNA damage and repair in human peripheral blood mononuclear cells: Protective role of hemoglobin</a>. <em>PLoS ONE</em> <strong>8</strong>, e68341 (2013).</div>
</div>
<div id="ref-rothFrameshiftMutations1974" class="csl-entry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Roth, J. R. <a href="https://doi.org/10.1146/annurev.ge.08.120174.001535">Frameshift mutations</a>. <em>Annual Review of Genetics</em> <strong>8</strong>, 319–346 (1974).</div>
</div>
<div id="ref-kujovichFactorLeidenThrombophilia2011" class="csl-entry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Kujovich, J. L. <a href="https://doi.org/10.1097/GIM.0b013e3181faa0f2">Factor V Leiden thrombophilia</a>. <em>Genetics in Medicine</em> <strong>13</strong>, 1–16 (2011).</div>
</div>
<div id="ref-cuttingCysticFibrosisGenetics2015" class="csl-entry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Cutting, G. R. <a href="https://doi.org/10.1038/nrg3849">Cystic fibrosis genetics: from molecular understanding to clinical application</a>. <em>Nature Reviews Genetics</em> <strong>16</strong>, 45–56 (2015).</div>
</div>
<div id="ref-fuchsbergerGeneticArchitectureType2016" class="csl-entry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Fuchsberger, C. <em>et al.</em> <a href="https://doi.org/10.1038/nature18642">The genetic architecture of type 2 diabetes</a>. <em>Nature</em> <strong>536</strong>, 41–47 (2016).</div>
</div>
<div id="ref-morrisLargescaleAssociationAnalysis2012" class="csl-entry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Morris, A. P. <em>et al.</em> <a href="https://doi.org/10.1038/ng.2383">Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes</a>. <em>Nature genetics</em> <strong>44</strong>, 981–990 (2012).</div>
</div>
<div id="ref-woodfordEmergenceAntibioticResistance2007" class="csl-entry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Woodford, N. &amp; Ellington, M. J. <a href="https://doi.org/10.1111/j.1469-0691.2006.01492.x">The emergence of antibiotic resistance by mutation</a>. <em>Clinical Microbiology and Infection</em> <strong>13</strong>, 5–18 (2007).</div>
</div>
<div id="ref-rheeHumanImmunodeficiencyVirus2003" class="csl-entry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Rhee, S.-Y. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkg100">Human immunodeficiency virus reverse transcriptase and protease sequence database</a>. <em>Nucleic Acids Research</em> <strong>31</strong>, 298–303 (2003).</div>
</div>
<div id="ref-sangerDNASequencingChainterminating1977" class="csl-entry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Sanger, F., Nicklen, S. &amp; Coulson, A. R. <a href="https://doi.org/10.1073/pnas.74.12.5463"><span>DNA</span> sequencing with chain-terminating inhibitors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>74</strong>, 5463–5467 (1977).</div>
</div>
<div id="ref-smithSynthesisOligonucleotidesContaining1985" class="csl-entry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Smith, L. M., Fung, S., Hunkapiller, M. W., Hunkapiller, T. J. &amp; Hood, L. E. <a href="https://doi.org/10.1093/nar/13.7.2399">The synthesis of oligonucleotides containing an aliphatic amino group at the 5′ terminus: Synthesis of fluorescent <span>DNA</span> primers for use in <span>DNA</span> sequence analysis</a>. <em>Nucleic Acids Research</em> <strong>13</strong>, 2399–2412 (1985).</div>
</div>
<div id="ref-smithFluorescenceDetectionAutomated1986" class="csl-entry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline">Smith, L. M. <em>et al.</em> <a href="https://doi.org/10.1038/321674a0">Fluorescence detection in automated <span>DNA</span> sequence analysis</a>. <em>Nature</em> <strong>321</strong>, 674–679 (1986).</div>
</div>
<div id="ref-ansorgeAutomatedDNASequencing1987" class="csl-entry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Ansorge, W., Sproat, B., Stegemann, J., Schwager, C. &amp; Zenke, M. <a href="https://doi.org/10.1093/nar/15.11.4593">Automated DNA sequencing: Ultrasensitive detection of fluorescent bands during electrophoresis</a>. <em>Nucleic Acids Research</em> <strong>15</strong>, 4593–4602 (1987).</div>
</div>
<div id="ref-shendureNextgenerationDNASequencing2008" class="csl-entry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Shendure, J. &amp; Ji, H. <a href="https://doi.org/10.1038/nbt1486">Next-generation DNA sequencing</a>. <em>Nature Biotechnology</em> <strong>26</strong>, 1135–1145 (2008).</div>
</div>
<div id="ref-collinsHumanGenomeProject2003" class="csl-entry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Collins, F. S., Morgan, M. &amp; Patrinos, A. <a href="https://doi.org/10.1126/science.1084564">The human genome project: Lessons from large-scale biology</a>. <em>Science</em> <strong>300</strong>, 286–290 (2003).</div>
</div>
<div id="ref-liuComparisonNextGenerationSequencing2012" class="csl-entry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Liu, L. <em>et al.</em> <a href="https://doi.org/10.1155/2012/251364">Comparison of <span>Next-Generation Sequencing Systems</span></a>. <em>Journal of Biomedicine and Biotechnology</em> <strong>2012</strong>, e251364 (2012).</div>
</div>
<div id="ref-CostSequencingHuman" class="csl-entry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><a href="https://www.genome.gov/about-genomics/fact-sheets/Sequencing-Human-Genome-cost">The Cost of Sequencing a Human Genome</a>.</div>
</div>
<div id="ref-metzkerSequencingTechnologiesNext2010" class="csl-entry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Metzker, M. L. <a href="https://doi.org/10.1038/nrg2626">Sequencing technologies — the next generation</a>. <em>Nat Rev Genet</em> <strong>11</strong>, 31–46 (2010).</div>
</div>
<div id="ref-canardDNAPolymeraseFluorescent1994" class="csl-entry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Canard, B. &amp; Sarfati, R. S. <a href="https://doi.org/10.1016/0378-1119(94)90226-7">DNA polymerase fluorescent substrates with reversible 3<span>’</span>-tags</a>. <em>Gene</em> <strong>148</strong>, 1–6 (1994).</div>
</div>
<div id="ref-nyrenSolidPhaseDNA1993" class="csl-entry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Nyren, P., Pettersson, B. &amp; Uhlen, M. <a href="https://doi.org/10.1006/abio.1993.1024">Solid Phase DNA Minisequencing by an Enzymatic Luminometric Inorganic Pyrophosphate Detection Assay</a>. <em>Analytical Biochemistry</em> <strong>208</strong>, 171–175 (1993).</div>
</div>
<div id="ref-mardisDecadePerspectiveDNA2011" class="csl-entry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Mardis, E. R. <a href="https://doi.org/10.1038/nature09796">A decade<span>’</span>s perspective on DNA sequencing technology</a>. <em>Nature</em> <strong>470</strong>, 198–203 (2011).</div>
</div>
<div id="ref-stolerSequencingErrorProfiles2021" class="csl-entry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Stoler, N. &amp; Nekrutenko, A. <a href="https://doi.org/10.1093/nargab/lqab019">Sequencing error profiles of illumina sequencing instruments</a>. <em>NAR Genomics and Bioinformatics</em> <strong>3</strong>, lqab019 (2021).</div>
</div>
<div id="ref-SequencingTechnologySequencing" class="csl-entry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><a href="https://www.illumina.com/science/technology/next-generation-sequencing/sequencing-technology.html">Sequencing Technology | Sequencing by synthesis</a>.</div>
</div>
<div id="ref-pollardLongReadsTheir2018" class="csl-entry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Pollard, M. O., Gurdasani, D., Mentzer, A. J., Porter, T. &amp; Sandhu, M. S. <a href="https://doi.org/10.1093/hmg/ddy177">Long reads: Their purpose and place</a>. <em>Human Molecular Genetics</em> <strong>27</strong>, R234–R241 (2018).</div>
</div>
<div id="ref-eidRealTimeDNASequencing2009" class="csl-entry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline">Eid, J. <em>et al.</em> <a href="https://doi.org/10.1126/science.1162986">Real-time DNA sequencing from single polymerase molecules</a>. <em>Science</em> <strong>323</strong>, 133–138 (2009).</div>
</div>
<div id="ref-leveneZeroModeWaveguidesSingleMolecule2003" class="csl-entry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">Levene, M. J. <em>et al.</em> <a href="https://doi.org/10.1126/science.1079700">Zero-mode waveguides for single-molecule analysis at high concentrations</a>. <em>Science</em> <strong>299</strong>, 682–686 (2003).</div>
</div>
<div id="ref-clarkeContinuousBaseIdentification2009" class="csl-entry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">Clarke, J. <em>et al.</em> <a href="https://doi.org/10.1038/nnano.2009.12">Continuous base identification for single-molecule nanopore DNA sequencing</a>. <em>Nature Nanotechnology</em> <strong>4</strong>, 265–270 (2009).</div>
</div>
<div id="ref-deamerThreeDecadesNanopore2016" class="csl-entry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Deamer, D., Akeson, M. &amp; Branton, D. <a href="https://doi.org/10.1038/nbt.3423">Three decades of nanopore sequencing</a>. <em>Nature Biotechnology</em> <strong>34</strong>, 518–524 (2016).</div>
</div>
<div id="ref-wickPerformanceNeuralNetwork2019" class="csl-entry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Wick, R. R., Judd, L. M. &amp; Holt, K. E. <a href="https://doi.org/10.1186/s13059-019-1727-y">Performance of neural network basecalling tools for Oxford Nanopore sequencing</a>. <em>Genome Biology</em> <strong>20</strong>, 129 (2019).</div>
</div>
<div id="ref-rhoadsPacBioSequencingIts2015" class="csl-entry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Rhoads, A. &amp; Au, K. F. <a href="https://doi.org/10.1016/j.gpb.2015.08.002">PacBio Sequencing and Its Applications</a>. <em>Genomics, Proteomics &amp; Bioinformatics</em> <strong>13</strong>, 278–289 (2015).</div>
</div>
<div id="ref-ipMinIONAnalysisReference2015" class="csl-entry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline">Ip, C. L. C. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.7201.1">MinION analysis and reference consortium: Phase 1 data release and analysis</a>. <em>F1000Research</em> <strong>4</strong>, 1075 (2015).</div>
</div>
<div id="ref-logsdonLongreadHumanGenome2020" class="csl-entry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline">Logsdon, G. A., Vollger, M. R. &amp; Eichler, E. E. <a href="https://doi.org/10.1038/s41576-020-0236-x">Long-read human genome sequencing and its applications</a>. <em>Nat Rev Genet</em> <strong>21</strong>, 597–614 (2020).</div>
</div>
<div id="ref-jainNanoporeSequencingAssembly2018" class="csl-entry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline">Jain, M. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.4060">Nanopore sequencing and assembly of a human genome with ultra-long reads</a>. <em>Nature Biotechnology</em> <strong>36</strong>, 338–345 (2018).</div>
</div>
<div id="ref-payneBulkVisGraphicalViewer2019" class="csl-entry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline">Payne, A., Holmes, N., Rakyan, V. &amp; Loose, M. <a href="https://doi.org/10.1093/bioinformatics/bty841">BulkVis: A graphical viewer for oxford nanopore bulk FAST5 files</a>. <em>Bioinformatics</em> <strong>35</strong>, 2193–2198 (2019).</div>
</div>
<div id="ref-murigneuxComparisonLongreadMethods2020" class="csl-entry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline">Murigneux, V. <em>et al.</em> <a href="https://doi.org/10.1093/gigascience/giaa146">Comparison of long-read methods for sequencing and assembly of a plant genome</a>. <em>GigaScience</em> <strong>9</strong>, giaa146 (2020).</div>
</div>
<div id="ref-chaissonResolvingComplexityHuman2015" class="csl-entry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline">Chaisson, M. J. P. <em>et al.</em> <a href="https://doi.org/10.1038/nature13907">Resolving the complexity of the human genome using single-molecule sequencing</a>. <em>Nature</em> <strong>517</strong>, 608–611 (2015).</div>
</div>
<div id="ref-jainOxfordNanoporeMinION2016" class="csl-entry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline">Jain, M., Olsen, H. E., Paten, B. &amp; Akeson, M. <a href="https://doi.org/10.1186/s13059-016-1103-0">The Oxford Nanopore MinION: delivery of nanopore sequencing to the genomics community</a>. <em>Genome Biology</em> <strong>17</strong>, 239 (2016).</div>
</div>
<div id="ref-hongRNASequencingNew2020" class="csl-entry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline">Hong, M. <em>et al.</em> <a href="https://doi.org/10.1186/s13045-020-01005-x">RNA sequencing: New technologies and applications in cancer research</a>. <em>Journal of Hematology &amp; Oncology</em> <strong>13</strong>, 166 (2020).</div>
</div>
<div id="ref-ozsolakRNASequencingAdvances2011" class="csl-entry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline">Ozsolak, F. &amp; Milos, P. M. <a href="https://doi.org/10.1038/nrg2934">RNA sequencing: advances, challenges and opportunities</a>. <em>Nature Reviews Genetics</em> <strong>12</strong>, 87–98 (2011).</div>
</div>
<div id="ref-huntProteinSequencingTandem1986" class="csl-entry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline">Hunt, D. F., Yates, J. R., Shabanowitz, J., Winston, S. &amp; Hauer, C. R. <a href="https://doi.org/10.1073/pnas.83.17.6233">Protein sequencing by tandem mass spectrometry.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>83</strong>, 6233–6237 (1986).</div>
</div>
<div id="ref-smithProteinSequencingProtocols2002" class="csl-entry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline">Smith, B. J. <em>Protein Sequencing Protocols</em>. (Springer Science &amp; Business Media, 2002). doi:<a href="https://doi.org/10.1385/1592593429">10.1385/1592593429</a>.</div>
</div>
<div id="ref-restrepo-perezPavingWaySinglemolecule2018" class="csl-entry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline">Restrepo-Pérez, L., Joo, C. &amp; Dekker, C. <a href="https://doi.org/10.1038/s41565-018-0236-6">Paving the way to single-molecule protein sequencing</a>. <em>Nature Nanotechnology</em> <strong>13</strong>, 786–796 (2018).</div>
</div>
<div id="ref-weirather2017" class="csl-entry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Weirather, J. L. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.10571.2">Comprehensive comparison of pacific biosciences and oxford nanopore technologies and their applications to transcriptome analysis</a>. <em>F1000Research</em> <strong>6</strong>, 100 (2017).</div>
</div>
<div id="ref-wangNanoporeSequencingTechnology2021" class="csl-entry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Wang, Y., Zhao, Y., Bollas, A., Wang, Y. &amp; Au, K. F. <a href="https://doi.org/10.1038/s41587-021-01108-x">Nanopore sequencing technology, bioinformatics and applications</a>. <em>Nature Biotechnology</em> <strong>39</strong>, 1348–1365 (2021).</div>
</div>
<div id="ref-maAnalysisErrorProfiles2019" class="csl-entry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline">Ma, X. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-019-1659-6">Analysis of error profiles in deep next-generation sequencing data</a>. <em>Genome Biology</em> <strong>20</strong>, 50 (2019).</div>
</div>
<div id="ref-limaComparativeAssessmentLongread2020" class="csl-entry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline">Lima, L. <em>et al.</em> <a href="https://doi.org/10.1093/bib/bbz058">Comparative assessment of long-read error correction software applied to nanopore RNA-sequencing data</a>. <em>Briefings in Bioinformatics</em> <strong>21</strong>, 1164–1181 (2020).</div>
</div>
<div id="ref-zhangComprehensiveEvaluationLong2020" class="csl-entry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline">Zhang, H., Jain, C. &amp; Aluru, S. <a href="https://doi.org/10.1186/s12864-020-07227-0">A comprehensive evaluation of long read error correction methods</a>. <em>BMC Genomics</em> <strong>21</strong>, 889 (2020).</div>
</div>
<div id="ref-amarasingheOpportunitiesChallengesLongread2020" class="csl-entry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline">Amarasinghe, S. L. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-020-1935-5">Opportunities and challenges in long-read sequencing data analysis</a>. <em>Genome Biology</em> <strong>21</strong>, 30 (2020).</div>
</div>
<div id="ref-ruanFastAccurateLongread2020" class="csl-entry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline">Ruan, J. &amp; Li, H. <a href="https://doi.org/10.1038/s41592-019-0669-3">Fast and accurate long-read assembly with wtdbg2</a>. <em>Nature Methods</em> <strong>17</strong>, 155–158 (2020).</div>
</div>
<div id="ref-korenCanuScalableAccurate2017" class="csl-entry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline">Koren, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.215087.116">Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation</a>. <em>Genome Research</em> <strong>27</strong>, 722–736 (2017).</div>
</div>
<div id="ref-tischlerNonHybridLong2017" class="csl-entry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline">Tischler, G. &amp; Myers, E. W. <em>Non Hybrid Long Read Consensus Using Local De Bruijn Graph Assembly</em>. 106252 (2017) doi:<a href="https://doi.org/10.1101/106252">10.1101/106252</a>.</div>
</div>
<div id="ref-warrenNtEditScalableGenome2019" class="csl-entry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline">Warren, R. L. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btz400">ntEdit: Scalable genome sequence polishing</a>. <em>Bioinformatics</em> <strong>35</strong>, 4430–4432 (2019).</div>
</div>
<div id="ref-heplerImprovedCircularConsensus2016" class="csl-entry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline">Hepler, N. L. <em>et al.</em> <a href="https://www.pacb.com/wp-content/uploads/improved-circular-consensus-algorithm-with-application-to-detection-hiv-1-drams.pdf">An improved circular consensus algorithm with an application to detect HIV-1 drug-resistance associated mutations (DRAMs)</a>. in (2016).</div>
</div>
<div id="ref-simpsonDetectingDNACytosine2017" class="csl-entry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline">Simpson, J. T. <em>et al.</em> <a href="https://doi.org/10.1038/nmeth.4184">Detecting DNA cytosine methylation using nanopore sequencing</a>. <em>Nature Methods</em> <strong>14</strong>, 407–410 (2017).</div>
</div>
<div id="ref-hacklProovreadLargescaleHighaccuracy2014" class="csl-entry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline">Hackl, T., Hedrich, R., Schultz, J. &amp; Förster, F. <a href="https://doi.org/10.1093/bioinformatics/btu392">Proovread : Large-scale high-accuracy PacBio correction through iterative short read consensus</a>. <em>Bioinformatics</em> <strong>30</strong>, 3004–3011 (2014).</div>
</div>
<div id="ref-miclotteJabbaHybridError2016" class="csl-entry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline">Miclotte, G. <em>et al.</em> <a href="https://doi.org/10.1186/s13015-016-0075-7">Jabba: Hybrid error correction for long sequencing reads</a>. <em>Algorithms for Molecular Biology</em> <strong>11</strong>, 10 (2016).</div>
</div>
<div id="ref-korenHybridErrorCorrection2012" class="csl-entry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline">Koren, S. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.2280">Hybrid error correction and de novo assembly of single-molecule sequencing reads</a>. <em>Nature Biotechnology</em> <strong>30</strong>, 693–700 (2012).</div>
</div>
<div id="ref-salmelaLoRDECAccurateEfficient2014" class="csl-entry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline">Salmela, L. &amp; Rivals, E. <a href="https://doi.org/10.1093/bioinformatics/btu538">LoRDEC: Accurate and efficient long read error correction</a>. <em>Bioinformatics</em> <strong>30</strong>, 3506–3514 (2014).</div>
</div>
<div id="ref-walkerPilonIntegratedTool2014" class="csl-entry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline">Walker, B. J. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</a>. <em>PLOS ONE</em> <strong>9</strong>, e112963 (2014).</div>
</div>
<div id="ref-vaserFastAccurateNovo2017" class="csl-entry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline">Vaser, R., Sović, I., Nagarajan, N. &amp; Šikić, M. <a href="https://doi.org/10.1101/gr.214270.116">Fast and accurate de novo genome assembly from long uncorrected reads</a>. <em>Genome Research</em> <strong>27</strong>, 737–746 (2017).</div>
</div>
<div id="ref-wenger2019" class="csl-entry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline">Wenger, A. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-019-0217-9">Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome</a>. <em>Nature Biotechnology</em> <strong>37</strong>, 1155–1162 (2019).</div>
</div>
<div id="ref-timpDNABaseCallingNanopore2012" class="csl-entry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline">Timp, W., Comer, J. &amp; Aksimentiev, A. <a href="https://doi.org/10.1016/j.bpj.2012.04.009">DNA Base-Calling from a Nanopore Using a Viterbi Algorithm</a>. <em>Biophysical Journal</em> <strong>102</strong>, L37–L39 (2012).</div>
</div>
<div id="ref-peresiniNanoporeBaseCalling2021" class="csl-entry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline">Perešíni, P., Boža, V., Brejová, B. &amp; Vinař, T. <a href="https://doi.org/10.1093/bioinformatics/btab528">Nanopore base calling on the edge</a>. <em>Bioinformatics</em> <strong>37</strong>, 4661–4667 (2021).</div>
</div>
<div id="ref-bozaDeepNanoDeepRecurrent2017" class="csl-entry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline">Boža, V., Brejová, B. &amp; Vinař, T. <a href="https://doi.org/10.1371/journal.pone.0178751">DeepNano: Deep recurrent neural networks for base calling in MinION nanopore reads</a>. <em>PLOS ONE</em> <strong>12</strong>, e0178751 (2017).</div>
</div>
<div id="ref-tylerEvaluationOxfordNanopore2018" class="csl-entry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline">Tyler, A. D. <em>et al.</em> <a href="https://doi.org/10.1038/s41598-018-29334-5">Evaluation of Oxford Nanopore<span>’</span>s MinION Sequencing Device for Microbial Whole Genome Sequencing Applications</a>. <em>Scientific Reports</em> <strong>8</strong>, 10931 (2018).</div>
</div>
<div id="ref-linNanoporeTechnologyIts2021" class="csl-entry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline">Lin, B., Hui, J. &amp; Mao, H. <a href="https://doi.org/10.3390/bios11070214">Nanopore Technology and Its Applications in Gene Sequencing</a>. <em>Biosensors</em> <strong>11</strong>, 214 (2021).</div>
</div>
<div id="ref-OxfordNanoporeTech" class="csl-entry">
<div class="csl-left-margin">92. </div><div class="csl-right-inline"><a href="http://nanoporetech.com/about-us/news/oxford-nanopore-tech-update-new-duplex-method-q30-nanopore-single-molecule-reads-0">Oxford Nanopore Tech Update: new Duplex method for Q30 nanopore single molecule reads, PromethION 2, and more</a>.</div>
</div>
<div id="ref-sandersonComparisonR9Kit102022" class="csl-entry">
<div class="csl-left-margin">93. </div><div class="csl-right-inline">Sanderson, N. <em>et al.</em> Comparison of R9.4.1/Kit10 and R10/Kit12 Oxford Nanopore flowcells and chemistries in bacterial genome reconstruction. doi:<a href="https://doi.org/10.1101/2022.04.29.490057">10.1101/2022.04.29.490057</a>.</div>
</div>
<div id="ref-karstHighaccuracyLongreadAmplicon2021" class="csl-entry">
<div class="csl-left-margin">94. </div><div class="csl-right-inline">Karst, S. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-020-01041-y">High-accuracy long-read amplicon sequences using unique molecular identifiers with Nanopore or PacBio sequencing</a>. <em>Nature Methods</em> <strong>18</strong>, 165–169 (2021).</div>
</div>
<div id="ref-chenHighlyAccurateFluorogenic2017" class="csl-entry">
<div class="csl-left-margin">95. </div><div class="csl-right-inline">Chen, Z. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.3982">Highly accurate fluorogenic DNA sequencing with information theory<span></span>based error correction</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 1170–1178 (2017).</div>
</div>
<div id="ref-HighPerformanceLong" class="csl-entry">
<div class="csl-left-margin">96. </div><div class="csl-right-inline"><a href="https://www.illumina.com/content/illumina-marketing/amr/en_US/science/genomics-research/articles/infinity-high-performance-long-read-assay.html">High Performance Long Read Assay Enables Contiguous Data up to 10Kb on Existing Illumina Platforms</a>.</div>
</div>
<div id="ref-booeshaghiPseudoalignmentFacilitatesAssignment2022" class="csl-entry">
<div class="csl-left-margin">97. </div><div class="csl-right-inline">Booeshaghi, A. S. &amp; Pachter, L. Pseudoalignment facilitates assignment of error-prone Ultima Genomics reads. doi:<a href="https://doi.org/10.1101/2022.06.04.494845">10.1101/2022.06.04.494845</a>.</div>
</div>
<div id="ref-delahaye2021" class="csl-entry">
<div class="csl-left-margin">98. </div><div class="csl-right-inline">Delahaye, C. &amp; Nicolas, J. <a href="https://doi.org/10.1371/journal.pone.0257521">Sequencing DNA with nanopores: Troubles and biases</a>. <em>PLOS ONE</em> <strong>16</strong>, e0257521 (2021).</div>
</div>
<div id="ref-dohmBenchmarkingLongreadCorrection2020" class="csl-entry">
<div class="csl-left-margin">100. </div><div class="csl-right-inline">Dohm, J. C., Peters, P., Stralis-Pavese, N. &amp; Himmelbauer, H. <a href="https://doi.org/10.1093/nargab/lqaa037">Benchmarking of long-read correction methods</a>. <em>NAR Genomics and Bioinformatics</em> <strong>2</strong>, (2020).</div>
</div>
<div id="ref-fooxPerformanceAssessmentDNA2021" class="csl-entry">
<div class="csl-left-margin">101. </div><div class="csl-right-inline">Foox, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-021-01049-5">Performance assessment of DNA sequencing platforms in the ABRF Next-Generation Sequencing Study</a>. <em>Nature Biotechnology</em> <strong>39</strong>, 1129–1140 (2021).</div>
</div>
<div id="ref-huangHomopolishMethodRemoval2021" class="csl-entry">
<div class="csl-left-margin">102. </div><div class="csl-right-inline">Huang, Y.-T., Liu, P.-Y. &amp; Shih, P.-W. <a href="https://doi.org/10.1186/s13059-021-02282-6">Homopolish: A method for the removal of systematic errors in nanopore sequencing by homologous polishing</a>. <em>Genome Biology</em> <strong>22</strong>, 95 (2021).</div>
</div>
<div id="ref-rangSquiggleBasepairComputational2018" class="csl-entry">
<div class="csl-left-margin">103. </div><div class="csl-right-inline">Rang, F. J., Kloosterman, W. P. &amp; Ridder, J. de. <a href="https://doi.org/10.1186/s13059-018-1462-9">From squiggle to basepair: Computational approaches for improving nanopore sequencing read accuracy</a>. <em>Genome Biology</em> <strong>19</strong>, 90 (2018).</div>
</div>
<div id="ref-sarkozyCallingHomopolymerStretches2018" class="csl-entry">
<div class="csl-left-margin">104. </div><div class="csl-right-inline">Sarkozy, P., Jobbágy, Á. &amp; Antal, P. Calling Homopolymer Stretches from Raw Nanopore Reads by Analyzing k-mer Dwell Times. in (eds. Eskola, H., Väisänen, O., Viik, J. &amp; Hyttinen, J.) 241–244 (Springer, 2018). doi:<a href="https://doi.org/10.1007/978-981-10-5122-7_61">10.1007/978-981-10-5122-7_61</a>.</div>
</div>
<div id="ref-hawkinsIndelcorrectingDNABarcodes2018" class="csl-entry">
<div class="csl-left-margin">105. </div><div class="csl-right-inline">Hawkins, J. A., Jones, S. K., Finkelstein, I. J. &amp; Press, W. H. <a href="https://doi.org/10.1073/pnas.1802640115">Indel-correcting DNA barcodes for high-throughput sequencing</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>115</strong>, E6217–E6226 (2018).</div>
</div>
<div id="ref-srivathsanMinIONBasedPipeline2018" class="csl-entry">
<div class="csl-left-margin">106. </div><div class="csl-right-inline">Srivathsan, A. <em>et al.</em> <a href="https://doi.org/10.1111/1755-0998.12890">A MinION<span></span>-based pipeline for fast and cost-effective DNA barcoding</a>. <em>Molecular Ecology Resources</em> <strong>18</strong>, 1035–1049 (2018).</div>
</div>
<div id="ref-wangConstructionBioConstrainedCode2019" class="csl-entry">
<div class="csl-left-margin">107. </div><div class="csl-right-inline">Wang, Y., Noor-A-Rahim, Md., Gunawan, E., Guan, Y. L. &amp; Poh, C. L. <a href="https://doi.org/10.1109/LCOMM.2019.2912572">Construction of bio-constrained code for DNA data storage</a>. <em>IEEE Communications Letters</em> <strong>23</strong>, 963–966 (2019).</div>
</div>
<div id="ref-R10NewestNanopore" class="csl-entry">
<div class="csl-left-margin">108. </div><div class="csl-right-inline"><a href="http://nanoporetech.com/about-us/news/r103-newest-nanopore-high-accuracy-nanopore-sequencing-now-available-store">R10.3: the newest nanopore for high accuracy nanopore sequencing <span></span> now available in store</a>.</div>
</div>
<div id="ref-zhouDetectionDNAHomopolymer2019" class="csl-entry">
<div class="csl-left-margin">109. </div><div class="csl-right-inline">Zhou, L. <em>et al.</em> <a href="https://doi.org/10.1116/1.5116295">Detection of DNA homopolymer with graphene nanopore</a>. <em>Journal of Vacuum Science &amp; Technology B</em> <strong>37</strong>, 061809 (2019).</div>
</div>
<div id="ref-gotoIdentificationFourSinglestranded2018" class="csl-entry">
<div class="csl-left-margin">110. </div><div class="csl-right-inline">Goto, Y., Yanagi, I., Matsui, K., Yokoi, T. &amp; Takeda, K. <a href="https://doi.org/10.1039/C8NR04238A">Identification of four single-stranded DNA homopolymers with a solid-state nanopore in alkaline CsCl solution</a>. <em>Nanoscale</em> <strong>10</strong>, 20844–20850 (2018).</div>
</div>
<div id="ref-nurkHiCanuAccurateAssembly2020" class="csl-entry">
<div class="csl-left-margin">111. </div><div class="csl-right-inline">Nurk, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.263566.120"><span>HiCanu</span>: Accurate assembly of segmental duplications, satellites, and allelic variants from high-fidelity long reads</a>. <em>Genome Res.</em> <strong>30</strong>, 1291–1305 (2020).</div>
</div>
<div id="ref-ekimMinimizerspaceBruijnGraphs2021" class="csl-entry">
<div class="csl-left-margin">112. </div><div class="csl-right-inline">Ekim, B., Berger, B. &amp; Chikhi, R. <a href="https://doi.org/10.1016/j.cels.2021.08.009">Minimizer-space de <span>Bruijn</span> graphs: Whole-genome assembly of long reads in minutes on a personal computer</a>. <em>Cell Systems</em> <strong>12</strong>, 958–968.e6 (2021).</div>
</div>
<div id="ref-shafinNanoporeSequencingShasta2020" class="csl-entry">
<div class="csl-left-margin">113. </div><div class="csl-right-inline">Shafin, K. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-020-0503-6">Nanopore sequencing and the Shasta toolkit enable efficient de novo assembly of eleven human genomes</a>. <em>Nature Biotechnology</em> <strong>38</strong>, 1044–1053 (2020).</div>
</div>
<div id="ref-millerAggressiveAssemblyPyrosequencing2008" class="csl-entry">
<div class="csl-left-margin">114. </div><div class="csl-right-inline">Miller, J. R. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btn548">Aggressive assembly of pyrosequencing reads with mates</a>. <em>Bioinformatics</em> <strong>24</strong>, 2818–2824 (2008).</div>
</div>
<div id="ref-sahlinNovoClusteringLongRead2020" class="csl-entry">
<div class="csl-left-margin">115. </div><div class="csl-right-inline">Sahlin, K. &amp; Medvedev, P. <a href="https://doi.org/10.1089/cmb.2019.0299">De <span>Novo Clustering</span> of <span>Long-Read Transcriptome Data Using</span> a <span>Greedy</span>, <span>Quality Value-Based Algorithm</span></a>. <em>Journal of Computational Biology</em> <strong>27</strong>, 472–484 (2020).</div>
</div>
<div id="ref-auImprovingPacBioLong2012" class="csl-entry">
<div class="csl-left-margin">116. </div><div class="csl-right-inline">Au, K. F., Underwood, J. G., Lee, L. &amp; Wong, W. H. <a href="https://doi.org/10.1371/journal.pone.0046679">Improving <span>PacBio Long Read Accuracy</span> by <span>Short Read Alignment</span></a>. <em>PLOS ONE</em> <strong>7</strong>, e46679 (2012).</div>
</div>
<div id="ref-huLSCplusFastSolution2016" class="csl-entry">
<div class="csl-left-margin">117. </div><div class="csl-right-inline">Hu, R., Sun, G. &amp; Sun, X. <a href="https://doi.org/10.1186/s12859-016-1316-y">LSCplus: a fast solution for improving long read accuracy by short read alignment</a>. <em>BMC Bioinformatics</em> <strong>17</strong>, 451 (2016).</div>
</div>
<div id="ref-liMinimap2PairwiseAlignment2018" class="csl-entry">
<div class="csl-left-margin">118. </div><div class="csl-right-inline">Li, H. <a href="https://doi.org/10.1093/bioinformatics/bty191">Minimap2: Pairwise alignment for nucleotide sequences</a>. <em>Bioinformatics</em> <strong>34</strong>, 3094–3100 (2018).</div>
</div>
<div id="ref-jainWeightedMinimizerSampling2020" class="csl-entry">
<div class="csl-left-margin">119. </div><div class="csl-right-inline">Jain, C. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btaa435">Weighted minimizer sampling improves long read mapping</a>. <em>Bioinformatics</em> <strong>36</strong>, i111–i118 (2020).</div>
</div>
<div id="ref-vannesteForensicSTRAnalysis2012" class="csl-entry">
<div class="csl-left-margin">120. </div><div class="csl-right-inline">Van Neste, C., Van Nieuwerburgh, F., Van Hoofstat, D. &amp; Deforce, D. <a href="https://doi.org/10.1016/j.fsigen.2012.03.004">Forensic STR analysis using massive parallel sequencing</a>. <em>Forensic Science International: Genetics</em> <strong>6</strong>, 810–818 (2012).</div>
</div>
<div id="ref-ShortreadSequencingBinding" class="csl-entry">
<div class="csl-left-margin">121. </div><div class="csl-right-inline"><a href="https://www.pacb.com/technology/sequencing-by-binding/">Short-read sequencing by binding</a>.</div>
</div>
<div id="ref-cetinPlasmonicSensorCould2018" class="csl-entry">
<div class="csl-left-margin">122. </div><div class="csl-right-inline">Cetin, A. E. <em>et al.</em> <a href="https://doi.org/10.1021/acssensors.7b00957">Plasmonic sensor could enable label-free DNA sequencing</a>. <em>ACS Sensors</em> <strong>3</strong>, 561–568 (2018).</div>
</div>
<div id="ref-almogyCostefficientWholeGenomesequencing2022" class="csl-entry">
<div class="csl-left-margin">123. </div><div class="csl-right-inline">Almogy, G. <em>et al.</em> Cost-efficient whole genome-sequencing using novel mostly natural sequencing-by-synthesis chemistry and open fluidics platform. doi:<a href="https://doi.org/10.1101/2022.05.29.493900">10.1101/2022.05.29.493900</a>.</div>
</div>
<div id="ref-sunagawaTaraOceansGlobal2020" class="csl-entry">
<div class="csl-left-margin">124. </div><div class="csl-right-inline">Sunagawa, S. <em>et al.</em> <a href="https://doi.org/10.1038/s41579-020-0364-5">Tara Oceans: towards global ocean ecosystems biology</a>. <em>Nature Reviews Microbiology</em> <strong>18</strong>, 428–445 (2020).</div>
</div>
<div id="ref-lewinEarthBioGenomeProject2018" class="csl-entry">
<div class="csl-left-margin">125. </div><div class="csl-right-inline">Lewin, H. A. <em>et al.</em> <a href="https://doi.org/10.1073/pnas.1720115115">Earth BioGenome project: Sequencing life for the future of life</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>115</strong>, 4325–4333 (2018).</div>
</div>
<div id="ref-lightbodyReviewApplicationsHighthroughput2019" class="csl-entry">
<div class="csl-left-margin">126. </div><div class="csl-right-inline">Lightbody, G. <em>et al.</em> <a href="https://doi.org/10.1093/bib/bby051">Review of applications of high-throughput sequencing in personalized medicine: Barriers and facilitators of future progress in research and clinical application</a>. <em>Briefings in Bioinformatics</em> <strong>20</strong>, 1795–1811 (2019).</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="3">
<li id="fn3"><p>Homopolymer indels can be harmful in opposite circumstances as well. Let us consider, for example, a read that should correspond to several repetitions of a conserved motif. Homopolymer indels can artificially resolve an ambiguity by making the read unique and prefer a specific repetition of the motif or entirely misplace the read.<a href="what-is-sequence-data.html#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="general-introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="aligning-sequence-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/01-sequence-data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
