<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments.</title>
  <meta name="description" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments." />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments." />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments." />
  
  
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-06-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="general-introduction.html"/>
<link rel="next" href="aligning-sequence-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />
<link href="libs/tabwid-1.0.0/scrool.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">My PhD thesis.</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="résumé.html"><a href="résumé.html"><i class="fa fa-check"></i>Résumé</a></li>
<li class="chapter" data-level="" data-path="acknowledgments.html"><a href="acknowledgments.html"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="glossary.html"><a href="glossary.html"><i class="fa fa-check"></i>Glossary</a></li>
<li class="chapter" data-level="" data-path="general-introduction.html"><a href="general-introduction.html"><i class="fa fa-check"></i>General Introduction</a></li>
<li class="chapter" data-level="1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>1</b> What is Sequence data ?</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>1.1</b> Biological sequences, a primer</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>1.1.1</b> What is DNA ?</a></li>
<li class="chapter" data-level="1.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#from-information-to-action"><i class="fa fa-check"></i><b>1.1.2</b> From Information to action</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining sequence data</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough"><i class="fa fa-check"></i><b>1.2.1</b> Sanger sequencing, a breakthrough</a></li>
<li class="chapter" data-level="1.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> Next-generation sequencing</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>1.3</b> Sequencing errors, how to account for them ?</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#error-correction-methods"><i class="fa fa-check"></i><b>1.3.1</b> Error correction methods</a></li>
<li class="chapter" data-level="1.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#more-accurate-sequencing-methods"><i class="fa fa-check"></i><b>1.3.2</b> More accurate sequencing methods</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#the-special-case-of-homopolymers"><i class="fa fa-check"></i><b>1.4</b> The special case of homopolymers</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-the-genome"><i class="fa fa-check"></i><b>1.4.1</b> Homopolymers and the genome</a></li>
<li class="chapter" data-level="1.4.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#homopolymers-and-long-reads"><i class="fa fa-check"></i><b>1.4.2</b> Homopolymers and long reads</a></li>
<li class="chapter" data-level="1.4.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#accounting-for-homopolymers"><i class="fa fa-check"></i><b>1.4.3</b> Accounting for homopolymers</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#conclusion"><i class="fa fa-check"></i><b>1.5</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>2</b> Aligning sequence data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>2.1</b> What is an alignment ?</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-align"><i class="fa fa-check"></i><b>2.1.1</b> How to align ?</a></li>
<li class="chapter" data-level="2.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#substitution-models"><i class="fa fa-check"></i><b>2.1.2</b> Substitution models</a></li>
<li class="chapter" data-level="2.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#why-align"><i class="fa fa-check"></i><b>2.1.3</b> Why align ?</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-do-we-speed-up-local-alignment"><i class="fa fa-check"></i><b>2.2</b> How do we speed up local alignment ?</a></li>
<li class="chapter" data-level="2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#msa-short"><i class="fa fa-check"></i><b>2.3</b> MSA (short)</a></li>
<li class="chapter" data-level="2.4" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#mapping"><i class="fa fa-check"></i><b>2.4</b> Mapping</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><i class="fa fa-check"></i><b>3</b> Contribution 1: Improving local alignment by exploring a sequence transformation space</a>
<ul>
<li class="chapter" data-level="" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#hpc-abstract"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="3.1" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#hpc-introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#hpc-methods"><i class="fa fa-check"></i><b>3.2</b> Methods</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#sec:msr-def"><i class="fa fa-check"></i><b>3.2.1</b> Streaming sequence reductions</a></li>
<li class="chapter" data-level="3.2.2" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#sec:enum"><i class="fa fa-check"></i><b>3.2.2</b> Restricting the space of Streaming sequence reductions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#datasets-and-pipelines"><i class="fa fa-check"></i><b>3.3</b> Datasets and Pipelines</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#datasets"><i class="fa fa-check"></i><b>3.3.1</b> Datasets</a></li>
<li class="chapter" data-level="3.3.2" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#simulation-pipeline"><i class="fa fa-check"></i><b>3.3.2</b> Simulation pipeline</a></li>
<li class="chapter" data-level="3.3.3" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#sec:evalpipeline"><i class="fa fa-check"></i><b>3.3.3</b> Evaluation pipeline</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#hpc-results"><i class="fa fa-check"></i><b>3.4</b> Results</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>3.4.1</b> Selection of mapping-friendly sequence reductions</a></li>
<li class="chapter" data-level="3.4.2" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>3.4.2</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes</a></li>
<li class="chapter" data-level="3.4.3" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>3.4.3</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome</a></li>
<li class="chapter" data-level="3.4.4" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>3.4.4</b> Raw mapping improves upon HPC on centromeric regions</a></li>
<li class="chapter" data-level="3.4.5" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>3.4.5</b> Positions of incorrectly mapped reads across the entire human genome</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#hpc-discussion"><i class="fa fa-check"></i><b>3.5</b> Discussion</a></li>
<li class="chapter" data-level="3.6" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#limitations-of-this-study"><i class="fa fa-check"></i><b>3.6</b> Limitations of this study</a></li>
<li class="chapter" data-level="3.7" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#code-availability"><i class="fa fa-check"></i><b>3.7</b> Code availability</a></li>
<li class="chapter" data-level="" data-path="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html"><a href="contribution-1-improving-local-alignment-by-exploring-a-sequence-transformation-space.html#supplementary-information"><i class="fa fa-check"></i>Supplementary information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html"><i class="fa fa-check"></i><b>4</b> Learning from alignments</a>
<ul>
<li class="chapter" data-level="4.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#alignments-are-a-rich-source-of-information"><i class="fa fa-check"></i><b>4.1</b> Alignments are a rich source of information</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#pairwise-alns"><i class="fa fa-check"></i><b>4.1.1</b> Pairwise alns</a></li>
<li class="chapter" data-level="4.1.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#msa"><i class="fa fa-check"></i><b>4.1.2</b> MSA</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#preprocessing-the-alignment-for-machine-learning"><i class="fa fa-check"></i><b>4.2</b> Preprocessing the alignment for machine learning</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#embedding-the-alignment"><i class="fa fa-check"></i><b>4.2.1</b> Embedding the alignment</a></li>
<li class="chapter" data-level="4.2.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#choosing-a-learning-target"><i class="fa fa-check"></i><b>4.2.2</b> Choosing a learning target</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#how-to-learn-from-alns"><i class="fa fa-check"></i><b>4.3</b> How to learn from ALNs</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>4.3.1</b> Tests and statistical learning</a></li>
<li class="chapter" data-level="4.3.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#taking-interactions-into-account"><i class="fa fa-check"></i><b>4.3.2</b> Taking interactions into account</a></li>
<li class="chapter" data-level="4.3.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#deep-learning"><i class="fa fa-check"></i><b>4.3.3</b> Deep Learning</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html"><i class="fa fa-check"></i><b>5</b> HIV and DRMs</a>
<ul>
<li class="chapter" data-level="5.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-are-viruses"><i class="fa fa-check"></i><b>5.1</b> What are viruses ?</a></li>
<li class="chapter" data-level="5.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-is-hiv"><i class="fa fa-check"></i><b>5.2</b> What is HIV ?</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#quick-presentation-of-hiv"><i class="fa fa-check"></i><b>5.2.1</b> Quick Presentation of HIV</a></li>
<li class="chapter" data-level="5.2.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#replication-cycle-of-hiv"><i class="fa fa-check"></i><b>5.2.2</b> Replication cycle of HIV</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#drug-resistance-in-hiv"><i class="fa fa-check"></i><b>5.3</b> Drug resistance in HIV</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#how-does-art-work"><i class="fa fa-check"></i><b>5.3.1</b> How does ART work</a></li>
<li class="chapter" data-level="5.3.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#different-types-of-resistance"><i class="fa fa-check"></i><b>5.3.2</b> different types of resistance</a></li>
<li class="chapter" data-level="5.3.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#consequences-on-global-health"><i class="fa fa-check"></i><b>5.3.3</b> Consequences on global health</a></li>
<li class="chapter" data-level="5.3.4" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#finding-drms"><i class="fa fa-check"></i><b>5.3.4</b> Finding DRMS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>6</b> Contribution 2: Inferring mutation roles from sequence alignments using machine learning</a>
<ul>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#abstract-paper"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#author-summary"><i class="fa fa-check"></i>Author summary</a></li>
<li class="chapter" data-level="6.1" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction</a></li>
<li class="chapter" data-level="6.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>6.2</b> Materials and methods</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>6.2.2</b> Classifier training</a></li>
<li class="chapter" data-level="6.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>6.2.3</b> Measuring classifier performance</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-results"><i class="fa fa-check"></i><b>6.3</b> Results</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>6.3.1</b> Classifier performance &amp; interpretation</a></li>
<li class="chapter" data-level="6.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>6.3.2</b> Additional classification results</a></li>
<li class="chapter" data-level="6.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>6.3.3</b> Identifying new mutations from classifiers</a></li>
<li class="chapter" data-level="6.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>6.3.4</b> Detailed analysis of potentially resistance-associated mutations</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>6.4</b> Discussion and perspectives</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-acknowledgments"><i class="fa fa-check"></i>Acknowledgments</a></li>
<li class="chapter" data-level="" data-path="HIV-paper.html"><a href="HIV-paper.html#supporting-information"><i class="fa fa-check"></i>Supporting Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>7</b> Learning alignments, an interesting perspective</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-pairwise-alignment"><i class="fa fa-check"></i><b>7.1</b> Learning pairwise alignment</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#dedal"><i class="fa fa-check"></i><b>7.1.1</b> DEDAL</a></li>
<li class="chapter" data-level="7.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-an-alignment"><i class="fa fa-check"></i><b>7.1.2</b> predicting an alignment</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#what-else-could-we-learn"><i class="fa fa-check"></i><b>7.2</b> What else could we learn ?</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-to-predict-seeds-or-starting-positions"><i class="fa fa-check"></i><b>7.2.1</b> Learn to predict seeds or starting positions</a></li>
<li class="chapter" data-level="7.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-pre-processing-functions"><i class="fa fa-check"></i><b>7.2.2</b> Learn pre-processing functions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html"><i class="fa fa-check"></i>Global conclusion</a>
<ul>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hpc-part"><i class="fa fa-check"></i>HPC part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#hiv-part"><i class="fa fa-check"></i>HIV part</a></li>
<li class="chapter" data-level="" data-path="global-conclusion.html"><a href="global-conclusion.html#final-words"><i class="fa fa-check"></i>Final words</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="global-references.html"><a href="global-references.html"><i class="fa fa-check"></i>Global References</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”</a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation</a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#msr-performance-comparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison</a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#origin-of-incorrectly-mapped-reads-of-high-mapping-quality-on-whole-human-genome."><i class="fa fa-check"></i><b>A.3</b> Origin of incorrectly mapped reads of high mapping quality on whole human genome.</a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome"><i class="fa fa-check"></i><b>A.4</b> Analyzing read origin on whole human genome</a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.5</b> Performance of MSRs on the Drosophila genome</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”</a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>B.1</b> S1 Appendix (Technical appendix).</a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>B.1.1</b> Data</a></li>
<li class="chapter" data-level="B.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>B.1.2</b> Classifiers</a></li>
<li class="chapter" data-level="B.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>B.1.3</b> Scoring</a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>B.2</b> S1 Fig.</a></li>
<li class="chapter" data-level="B.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>B.3</b> S2 Fig.</a></li>
<li class="chapter" data-level="B.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>B.4</b> S3 Fig.</a></li>
<li class="chapter" data-level="B.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>B.5</b> S1 Table.</a></li>
<li class="chapter" data-level="B.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>B.6</b> S2 Appendix. (Fisher exact tests)</a></li>
<li class="chapter" data-level="B.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>B.7</b> S1 Data.</a></li>
<li class="chapter" data-level="B.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>B.8</b> S2 Data.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequences to knowledge, improving and learning from sequence alignments.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="what-is-sequence-data" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> What is Sequence data ?<a href="what-is-sequence-data.html#what-is-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="biological-sequences-a-primer" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Biological sequences, a primer<a href="what-is-sequence-data.html#biological-sequences-a-primer" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To fully understand the work that was done during this thesis, as well as the choices that were made some basic knowledge of biology and more particularly genetics are needed. If you are already familiar with biological sequences, feel free to skip ahead to section <a href="what-is-sequence-data.html#obtaining-sequence-data">1.2</a>.</p>
<div id="what-is-dna" class="section level3 hasAnchor" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> What is DNA ?<a href="what-is-sequence-data.html#what-is-dna" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>D</strong>esoxyribo<strong>N</strong>ucleic <strong>A</strong>cid (DNA) is one of the most important molecules there is, without it complex life as we know it is impossible. It contains all the genetic information of a given organism, that is to say all the information necessary for the organism to: 1) function as a living being and 2) make a perfect copy of itself. This is the case for the overwhelming majority of living organisms on planet earth, from elephants to potatoes, to micro-organisms like bacteria.</p>
<p>DNA is a polymer, composed of monomeric units called nucleotides. Each nucleotide is composed of Ribose (a five carbon sugar) on which are attached a phosphate group as well as one of four nucleobases: Adenine (A), Cytosine (C), Guanine (G) of Thymine (T). These 4 types of nucleotide monomers link up with one-another, through phosphate-sugar bonds, creating a single strand of DNA. The ordered sequence of these four types of nucleotides in strand encodes all the genetic information necessary for the organism to function. Nucleotides in a strand form strong complementary bonds with nucleotides from another strand, A with T and C with G. These bonds allows two strands of DNA to form the double-helix structure of DNA<span class="citation"><sup><a href="#ref-watson1953" role="doc-biblioref">1</a></sup></span> shown in Figure <a href="what-is-sequence-data.html#fig:figDNA">1.1</a>. The specificity of nucleotide bonds ensure that the two strands of the double helix are complementary and that the information contained in one strand can be recovered from the other. This ensures a certain structural stability to the DNA molecule and a way to recover the important information that could be lost due to a damaged strand.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figDNA"></span>
<img src="figures/Basic-Intro/DNA.png" alt="**Double-helix structure of DNA**.
Each strand of DNA has a phosphate-sugar backbone on which are attached nucleobases. The two strands are linked by complementary bonds between the nucleobases of different strands (A bonding with T and C bonding with G)." width="90%" />
<p class="caption">
Figure 1.1: <strong>Double-helix structure of DNA</strong>.
Each strand of DNA has a phosphate-sugar backbone on which are attached nucleobases. The two strands are linked by complementary bonds between the nucleobases of different strands (A bonding with T and C bonding with G).
</p>
</div>
<p>The amount of DNA necessary to encode the information varies greatly from organism to organism: 5400 base pairs (5.4kBp) for the <span class="math inline">\(\varphi X174\)</span> phage<span class="citation"><sup><a href="#ref-sangerNucleotideSequenceBacteriophage1977" role="doc-biblioref">2</a></sup></span>, 4.9MBp for <em>Escherichia coli</em><span class="citation"><sup><a href="#ref-archer2011" role="doc-biblioref">3</a></sup></span>, 3.1GBp for <em>Homo sapiens</em><span class="citation"><sup><a href="#ref-nurk2022" role="doc-biblioref">4</a></sup></span> all the way up to almost 150GBp for <em>Paris japonica,</em> a Japanese mountain flowering plant<span class="citation"><sup><a href="#ref-pellicer2010" role="doc-biblioref">5</a></sup></span>. While very small genome size tend to occur in smaller, simpler organisms genome size does not correlate with organism complexity<span class="citation"><sup><a href="#ref-macgregor2001" role="doc-biblioref">6</a></sup></span>.</p>
</div>
<div id="from-information-to-action" class="section level3 hasAnchor" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> From Information to action<a href="what-is-sequence-data.html#from-information-to-action" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="proteins-their-structure-and-their-role" class="section level4 hasAnchor" number="1.1.2.1">
<h4><span class="header-section-number">1.1.2.1</span> Proteins, their structure and their role<a href="what-is-sequence-data.html#proteins-their-structure-and-their-role" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The double stranded DNA molecules present in the cells of a living organism contains only information; in order for the organism to live, this information must be read and transformed into actions. Most of the actions necessary for “life” are taken by large molecules called proteins, they have a very wide range of functions from catalyzing reactions in the cell to giving it it’s structure<span class="citation"><sup><a href="#ref-alberts2002" role="doc-biblioref">7</a></sup></span>.</p>
<p>Proteins are macromolecules, that are made up of one or several chains of amino acids. These chains then link together and fold up in a specific 3 dimensional structure, giving the protein the shape it needs to fulfill it’s goal. This structure is determined by the sequence of amino acids, and a given protein can be identified by this amino acid sequence<span class="citation"><sup><a href="#ref-alberts2002" role="doc-biblioref">7</a></sup></span>.</p>
<p>This sequence is directly dependent on the information contained in the DNA. First the DNA is transcripted in a similar, but single stranded, molecule called RNA which encodes the same sequence. This RNA molecule is then translated into a protein by the following process<span class="citation"><sup><a href="#ref-crick1961" role="doc-biblioref">8</a></sup></span>:</p>
<ol style="list-style-type: decimal">
<li>Nucleotides in the RNA sequence are read in groups of 3 called a codon.</li>
<li>These codons are read sequentially along the RNA molecule</li>
<li>Each codon corresponds to an amino acid, according to the genetic code.</li>
<li>The sequence of codons in RNA <em>(and by extension DNA)</em> determines the sequence of amino acids.</li>
<li>The translation process is stopped when a specific type of codon is read.</li>
</ol>
<p>With 4 types of nucleotides and codons grouping 3 nucleotides there are <span class="math inline">\(4^3=64\)</span> possible codons. However, as stated above, proteins are only made up of 20 different amino acids, meaning that several different codons correspond to the same amino acid. This gives the translation process a certain robustness to errors that can occur when the DNA is copied to create a new cell, or when it is transformed into RNA prior to protein translation.</p>
<p>The portion of DNA that is read to create the protein is said to be “coding”, and is called a gene. There are several thousands of genes in the human genome<span class="citation"><sup><a href="#ref-internationalhumangenomesequencingconsortiumFinishingEuchromaticSequence2004" role="doc-biblioref">9</a></sup></span> resulting in proteins executing thousands of different functions in a cell. In human beings, coding DNA represents only 1% to 2% of the total genome<span class="citation"><sup><a href="#ref-elkonCharacterizationNoncodingRegulatory2017" role="doc-biblioref">10</a>,<a href="#ref-omennReflectionsHUPOHuman2021" role="doc-biblioref">11</a></sup></span>. The large majority of the DNA in a human being is not translated into proteins, a portion of it has a regulatory role, controlling transcription and translation, but the role remains unknown for a portion of the human genome<span class="citation"><sup><a href="#ref-shabalinaMammalianTranscriptomeFunction2004" role="doc-biblioref">12</a>,<a href="#ref-IntegratedEncyclopediaDNA2012" role="doc-biblioref">13</a></sup></span>.</p>
</div>
<div id="making-mistakes" class="section level4 hasAnchor" number="1.1.2.2">
<h4><span class="header-section-number">1.1.2.2</span> Making mistakes<a href="what-is-sequence-data.html#making-mistakes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Going from DNA sequence to protein is quite a complicated process involving several steps, it is therefore possible for a mistake to happen. There are several mechanisms to avoid mistakes and alteration of the genetic information: the complementary nature of the 2 strands of DNA, the redundant nature of the genetic code as well as error correction mechanisms in the molecules that read and write DNA and RNA <em>(polymerases)</em>. However, despite all that, some errors still make it through.</p>
<div id="where-can-mistakes-happen" class="section level5 hasAnchor" number="1.1.2.2.1">
<h5><span class="header-section-number">1.1.2.2.1</span> Where can mistakes happen ?<a href="what-is-sequence-data.html#where-can-mistakes-happen" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>There are several sources of error that can alter the genetic information<span class="citation"><sup><a href="#ref-chatterjee2017" role="doc-biblioref">14</a></sup></span>:</p>
<ul>
<li><p><strong>DNA replication:</strong> When a cell divides, or when an organism reproduces, the DNA molecule must be copied in order to transmit genetic information. This process has a very low rate of errors, with as low as 1 error for every billion to every hundred billions of base pairs replicated<span class="citation"><sup><a href="#ref-fijalkowska2012" role="doc-biblioref">15</a></sup></span>. This is due to the fact that the DNA polymerase (the protein that is responsible for copying DNA molecules), has a relatively error rate to start with, but mostly due to the error correcting mechanisms that are present in certain cells and bacteria<span class="citation"><sup><a href="#ref-pray2008dna" role="doc-biblioref">16</a></sup></span>.</p></li>
<li><p><strong>RNA transcription:</strong> error rate between 4 errors for each million<span class="citation"><sup><a href="#ref-gout2013" role="doc-biblioref">17</a></sup></span> to 2 errors for each hundred thousand<span class="citation"><sup><a href="#ref-gout2017" role="doc-biblioref">18</a></sup></span> base pairs transcribed.</p></li>
<li><p><strong>Other mutagenic events:</strong> Ionizing radiation<span class="citation"><sup><a href="#ref-desouky2015" role="doc-biblioref">19</a></sup></span>, UV rays<span class="citation"><sup><a href="#ref-kiefer2007" role="doc-biblioref">20</a></sup></span>, Toxins<span class="citation"><sup><a href="#ref-bennett2003" role="doc-biblioref">21</a></sup></span>, heat Stress<span class="citation"><sup><a href="#ref-kantidze2016" role="doc-biblioref">22</a></sup></span>, cold stress<span class="citation"><sup><a href="#ref-gregory1994" role="doc-biblioref">23</a></sup></span>, oxidative stress<span class="citation"><sup><a href="#ref-gafter-gvili2013" role="doc-biblioref">24</a></sup></span>.</p></li>
</ul>
</div>
<div id="what-kind-of-errors-are-possible" class="section level5 hasAnchor" number="1.1.2.2.2">
<h5><span class="header-section-number">1.1.2.2.2</span> What kind of errors are possible?<a href="what-is-sequence-data.html#what-kind-of-errors-are-possible" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li>substitution</li>
<li>insertion</li>
<li>deletion</li>
</ul>
</div>
<div id="what-effect-can-mutations-have" class="section level5 hasAnchor" number="1.1.2.2.3">
<h5><span class="header-section-number">1.1.2.2.3</span> What effect can mutations have ?<a href="what-is-sequence-data.html#what-effect-can-mutations-have" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>SNPs, many individual mutations have been linked to certain traits (however, a trait is not usually caused only by a single mutation, plus hard to prove causality)</p>
<ul>
<li><p>Diseases:</p>
<ul>
<li><p>Pathologic coagulation linked to a single mutation in DNA coding for a coagulation protein<span class="citation"><sup><a href="#ref-kujovichFactorLeidenThrombophilia2011" role="doc-biblioref">25</a></sup></span>.</p></li>
<li><p>Cystic Fibrosis, a deletion of a single amino acid in the protein leads to fatal disease<span class="citation"><sup><a href="#ref-cuttingCysticFibrosisGenetics2015" role="doc-biblioref">26</a></sup></span></p></li>
<li><p>Plenty of mutation in specific genes have been linked with type 2 diabetes<span class="citation"><sup><a href="#ref-fuchsbergerGeneticArchitectureType2016" role="doc-biblioref">27</a>,<a href="#ref-morrisLargescaleAssociationAnalysis2012" role="doc-biblioref">28</a></sup></span></p></li>
</ul></li>
<li><p>Added traits, like drug resistance:</p>
<ul>
<li><p>Dozens of mutations proven to confer HIV resistance to commonly used drugs<span class="citation"><sup><a href="#ref-rheeHumanImmunodeficiencyVirus2003" role="doc-biblioref">29</a></sup></span></p></li>
<li><p>A large amount of mutations have been shown to confer antibacterial resistance in a range of bacteria<span class="citation"><sup><a href="#ref-woodfordEmergenceAntibioticResistance2007" role="doc-biblioref">30</a></sup></span></p></li>
</ul></li>
</ul>
</div>
</div>
</div>
</div>
<div id="obtaining-sequence-data" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Obtaining sequence data<a href="what-is-sequence-data.html#obtaining-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In order to study living organisms we need to be able to obtain their genetic information, i.e figure out a way to get the sequence of nucleobases that make up their DNA.</p>
<div id="sanger-sequencing-a-breakthrough" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Sanger sequencing, a breakthrough<a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The first true sequencing method was developed in 1977<span class="citation"><sup><a href="#ref-sangerDNASequencingChainterminating1977" role="doc-biblioref">31</a></sup></span>. Sanger <em>et al.</em> devised a simple method to read the sequence of nucleotides that make up a DNA sequence.</p>
<ol style="list-style-type: decimal">
<li>Clone sequence / amplify</li>
<li>Prepare 4 different sequencing environments with a majority of dNTP (ie regular nucleotides) and in each a single type of ddNTP (a terminator). ddNTP are marked</li>
<li>In each test tube add DNA polymerase, primers and denatures DNA fragments you want to sequence</li>
<li>Sequence is replicated until incorporation of ddNTP stopping reaction</li>
<li>Separate replicated fragments by gel electrophoresis (i.e shorter fragments go further), 1 ddNTP type in each lane</li>
<li>With marked you can see which nucleotide is present at a given position</li>
</ol>
<p>This allowed Sanger <em>et al.</em> to sequence the first whole <span class="math inline">\(\varphi X174\)</span> bacteriophage genome<span class="citation"><sup><a href="#ref-sangerNucleotideSequenceBacteriophage1977" role="doc-biblioref">2</a></sup></span>. This method, although revolutionnary was costly and time consuming.</p>
<p>The marking of primers and ddNTP with fluorescence allowed to do the polymerization in a single test tube and use a single lane for electrophoresis<span class="citation"><sup><a href="#ref-smithSynthesisOligonucleotidesContaining1985" role="doc-biblioref">32</a>,<a href="#ref-smithFluorescenceDetectionAutomated1986" role="doc-biblioref">33</a></sup></span>. The fluorescence also allowed for automated reading with optical systems. Plus switching to capillary electrophoresis allowed for automation and speeding up. Shotgun sequencing also speed up stuff at the reconstruct the whole genome.</p>
<p><strong><em>I need to speak of performance / throughput of these methods here.</em></strong></p>
<p>With the latest Sanger sequencing technologies, sequenced sequences can reach 1kBp with an accuracy of 99.999%<span class="citation"><sup><a href="#ref-shendureNextgenerationDNASequencing2008" role="doc-biblioref">34</a></sup></span>. Although very time consuming and expensive at it’s dawn, the sequencing of the first human genome costing between 500 million and 1 billion dollars<span class="citation"><sup><a href="#ref-CostSequencingHuman" role="doc-biblioref">35</a></sup></span>. Over time and with technological advancments, the cost of Sanger sequencing was greatly reduced from $1000 per Bp<span class="citation"><sup><a href="#ref-collinsHumanGenomeProject2003" role="doc-biblioref">36</a></sup></span> to $0.5 per kBp<span class="citation"><sup><a href="#ref-shendureNextgenerationDNASequencing2008" role="doc-biblioref">34</a></sup></span>, and the throughput increased from 1 kBp per day<span class="citation"><sup><a href="#ref-collinsHumanGenomeProject2003" role="doc-biblioref">36</a></sup></span> to up to 120 kBp per hour<span class="citation"><sup><a href="#ref-liuComparisonNextGenerationSequencing2012" role="doc-biblioref">37</a></sup></span>.</p>
</div>
<div id="next-generation-sequencing" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Next-generation sequencing<a href="what-is-sequence-data.html#next-generation-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Developed to lower cost and more throughput</p>
<ul>
<li><p>Massively parallel <em>(no need to detail technology, just numbers i.e. throughput and cost) (i.e. 2nd gen)</em><span class="citation"><sup><a href="#ref-metzkerSequencingTechnologiesNext2010" role="doc-biblioref">38</a></sup></span></p>
<ul>
<li><p>Key points:</p>
<ul>
<li><p>Break up DNA to sequence in smaller fragments</p></li>
<li><p>Amplify fragments (i.e. clone them)</p></li>
<li><p>Attach fragments to solid support so we can thousands to billions of sequencing reactions at once</p></li>
<li><p>detect the polymerization as it happens, speeding up the process.</p></li>
</ul></li>
<li><p>Illumina:</p>
<ul>
<li><p>150nt<span class="citation"><sup><a href="#ref-mardisDecadePerspectiveDNA2011" role="doc-biblioref">39</a></sup></span></p></li>
<li><p>98% accuracy, $0.07 per MBp, 2.5 to 12.5 Gbp per hour<span class="citation"><sup><a href="#ref-liuComparisonNextGenerationSequencing2012" role="doc-biblioref">37</a></sup></span></p></li>
<li><p>Detect polymerization by adding marked terminating NTP, imaging and cleaving terminating group to allow further polymerization.</p></li>
</ul></li>
<li><p>pyrosequencing:</p>
<ul>
<li><p>400nt<span class="citation"><sup><a href="#ref-mardisDecadePerspectiveDNA2011" role="doc-biblioref">39</a></sup></span></p></li>
<li><p>99.9% accuracy, $10 per MBp, 30Mbp per hour<span class="citation"><sup><a href="#ref-liuComparisonNextGenerationSequencing2012" role="doc-biblioref">37</a></sup></span></p></li>
<li><p>Detect polymerization with a reaction that emits light during the incorporation of a dNTP to the synthesized DNA, measure the light emission to determine the base that was added.</p></li>
</ul></li>
</ul></li>
<li><p>Long reads sequencing (i.e. 3rd gen):</p>
<ul>
<li><p>Introduce need for longer reads:</p>
<ul>
<li><p>Better for analyzing complex and repetitive parts of the genome, and were used for assembly <strong>[CITE]</strong>, structural variant detection <strong>[CITE]</strong>, etc…</p></li>
<li><p>Real time sequencing, i.e.</p></li>
<li><p>High throughput and</p></li>
</ul></li>
<li><p>PacBIO SMRT<span class="citation"><sup><a href="#ref-eidRealTimeDNASequencing2009" role="doc-biblioref">40</a></sup></span>:</p>
<ul>
<li>Tech summary:
<ul>
<li><p>Link 2 strands of DNA to sequence with a hairpin sequence to create circular DNA</p></li>
<li><p>Capture inserted with polymerase fixed to bottom of a well (up to 8million wells in 1 sequencing cell)</p></li>
<li><p>Florescently marked dNTP are added and excited with laser and light emission recorded by camera</p></li>
<li><p>Stop when you reach the hairpin</p></li>
</ul></li>
<li>Most reads around 10kb in length with reads up to 60kb<span class="citation"><sup><a href="#ref-rhoadsPacBioSequencingIts2015" role="doc-biblioref">41</a></sup></span></li>
<li>85% to 92% accuracy<span class="citation"><sup><a href="#ref-chaissonResolvingComplexityHuman2015" role="doc-biblioref">42</a>,<a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">43</a></sup></span></li>
<li>$0.32 per Mbp<span class="citation"><sup><a href="#ref-murigneuxComparisonLongreadMethods2020" role="doc-biblioref">44</a></sup></span></li>
<li>2Gbp to 11Gbp per hour<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">43</a></sup></span></li>
</ul></li>
<li><p>NanoPore<span class="citation"><sup><a href="#ref-clarkeContinuousBaseIdentification2009" role="doc-biblioref">45</a></sup></span>:</p>
<ul>
<li><p>Tech summary:</p>
<ul>
<li><p>100k nanopores spread out on synthetic membrane</p></li>
<li><p>Protein splits the 2 strands and pushes 1 of the brands through a nanopore at a controlled rate</p></li>
<li><p>As DNA goes through the pore it disrupts the electrical current between 2 sides of the membrane</p></li>
<li><p>Disruptions are characteristic of the 5/6 bases in the nanopore channel -&gt; deduce sequence from current disruptions (i.e. base calling)</p></li>
</ul></li>
<li><p>Median read length around 10kb - 12kb<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">43</a>,<a href="#ref-ipMinIONAnalysisReference2015" role="doc-biblioref">46</a></sup></span>.</p></li>
<li><p>you can get ultra long reads, of almost 1Mb<span class="citation"><sup><a href="#ref-jainNanoporeSequencingAssembly2018" role="doc-biblioref">47</a>,<a href="#ref-TharSheBlows" role="doc-biblioref">48</a></sup></span> and even up to 2.3 Mb<span class="citation"><sup><a href="#ref-payneBulkVisGraphicalViewer2019" role="doc-biblioref">49</a></sup></span>.</p></li>
<li><p>87% to 98% accuracy<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">43</a>,<a href="#ref-jainOxfordNanoporeMinION2016" role="doc-biblioref">50</a></sup></span></p></li>
<li><p>$0.13 per Mbp<span class="citation"><sup><a href="#ref-murigneuxComparisonLongreadMethods2020" role="doc-biblioref">44</a></sup></span></p></li>
<li><p>12.5Gbp to 260Gbp per hour !!<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">43</a></sup></span></p></li>
</ul></li>
</ul></li>
</ul>
<p>As a conclusion, quickly mention the work done on protein sequencing:</p>
<ul>
<li><p>has been going on for a long time<span class="citation"><sup><a href="#ref-huntProteinSequencingTandem1986" role="doc-biblioref">51</a>,<a href="#ref-smithProteinSequencingProtocols2002" role="doc-biblioref">52</a></sup></span> with mass spectrometry</p></li>
<li><p>But still ongoing with new methods<span class="citation"><sup><a href="#ref-restrepo-perezPavingWaySinglemolecule2018" role="doc-biblioref">53</a></sup></span></p></li>
</ul>
<p>but we usually deduce the protein sequence from the DNA sequence that is translated from codons after detecting ORFs.</p>
</div>
</div>
<div id="sequencing-errors-how-to-account-for-them" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Sequencing errors, how to account for them ?<a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Just like mutations we can get substitutions and indels.</p>
<p>general context of sequencing errors, i.e. short reads have high accuracy (After computational correction you can get 10<sup>-4</sup> to 10<sup>-5</sup> error rate<span class="citation"><sup><a href="#ref-maAnalysisErrorProfiles2019" role="doc-biblioref">54</a></sup></span> ) , long reads high error rate as stated before, 10% to 15%.</p>
<div id="error-correction-methods" class="section level3 hasAnchor" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Error correction methods<a href="what-is-sequence-data.html#error-correction-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>In Non-Hybrid error correcting methods only the long reads are used, by overlapping them and taking the consensus of the overlap it is possible to eliminate some errors, as well as filtering very rare subsequences likely due to a sequencing error<span class="citation"><sup><a href="#ref-limaComparativeAssessmentLongread2020" role="doc-biblioref">55</a>,<a href="#ref-amarasingheOpportunitiesChallengesLongread2020" role="doc-biblioref">56</a></sup></span>. These methods are applied in several pieces of software like wtdbg2<span class="citation"><sup><a href="#ref-ruanFastAccurateLongread2020" role="doc-biblioref">57</a></sup></span>, canu<span class="citation"><sup><a href="#ref-korenCanuScalableAccurate2017" role="doc-biblioref">58</a></sup></span>, or daccord<span class="citation"><sup><a href="#ref-tischlerNonHybridLong2017" role="doc-biblioref">59</a></sup></span>.</p></li>
<li><p>In Hybrid methods, shorter more accurate reads are used to correct long-read errors, it has been used in software like proovread<span class="citation"><sup><a href="#ref-hacklProovreadLargescaleHighaccuracy2014" role="doc-biblioref">60</a></sup></span>, Jabba<span class="citation"><sup><a href="#ref-miclotteJabbaHybridError2016" role="doc-biblioref">61</a></sup></span> or LoRDEC<span class="citation"><sup><a href="#ref-salmelaLoRDECAccurateEfficient2014" role="doc-biblioref">62</a></sup></span></p></li>
<li><p>In some cases it can be useful to correct errors after having assembled the raw reads in a process called polishing. One can use non-hybrid methods by filtering out rare subsequences like in ntEdits , or correcting with raw PacBio or ONT data as is done in Arrow<span class="citation"><sup><a href="#ref-heplerImprovedCircularConsensus2016" role="doc-biblioref">63</a></sup></span> or Nanopolish<span class="citation"><sup><a href="#ref-simpsonDetectingDNACytosine2017" role="doc-biblioref">64</a></sup></span> respectively. Conversly, hybrid methods using short reads also exist to correct assembled reads like Pilon<span class="citation"><sup><a href="#ref-walkerPilonIntegratedTool2014" role="doc-biblioref">65</a></sup></span> or Racon<span class="citation"><sup><a href="#ref-vaserFastAccurateNovo2017" role="doc-biblioref">66</a></sup></span>.</p></li>
<li><p>Error correction bibliography is rich and many people are working on this<span class="citation"><sup><a href="#ref-limaComparativeAssessmentLongread2020" role="doc-biblioref">55</a>,<a href="#ref-fuComparativeEvaluationHybrid2019" role="doc-biblioref">67</a>,<a href="#ref-zhangComprehensiveEvaluationLong2020" role="doc-biblioref">68</a></sup></span></p></li>
</ul>
</div>
<div id="more-accurate-sequencing-methods" class="section level3 hasAnchor" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> More accurate sequencing methods<a href="what-is-sequence-data.html#more-accurate-sequencing-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>CCS / HiFi stuff -&gt; 99.8% to 99.9% accuracy on long reads<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">43</a>,<a href="#ref-wenger2019" role="doc-biblioref">69</a></sup></span> because errors are randomly distributed. Most errors are still indels in homopolymers<span class="citation"><sup><a href="#ref-wenger2019" role="doc-biblioref">69</a></sup></span></p></li>
<li><p>Using unique molecular identifiers and consensus reads, you can reduce error rate to 99.59% and 99.93% accuracies for ONT and PacBio CCS<span class="citation"><sup><a href="#ref-karstHighaccuracyLongreadAmplicon2021" role="doc-biblioref">70</a></sup></span>.</p></li>
<li><p>Improve base calling models for ONT by switching from HMMs to deep learning,<span class="citation"><sup><a href="#ref-amarasingheOpportunitiesChallengesLongread2020" role="doc-biblioref">56</a>,<a href="#ref-peresiniNanoporeBaseCalling2021" role="doc-biblioref">71</a>–<a href="#ref-wickPerformanceNeuralNetwork2019" role="doc-biblioref">73</a></sup></span></p></li>
<li><p>Built in error correction<span class="citation"><sup><a href="#ref-chenHighlyAccurateFluorogenic2017" role="doc-biblioref">74</a></sup></span>, results in shorter reads with 99.82% accuracy (200-250nt reads though). Error free up to 200bp.</p></li>
<li><p>To come Illumina Infinity<span class="citation"><sup><a href="#ref-HighPerformanceLong" role="doc-biblioref">75</a></sup></span>, high throughput long reads.</p></li>
</ul>
</div>
</div>
<div id="the-special-case-of-homopolymers" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> The special case of homopolymers<a href="what-is-sequence-data.html#the-special-case-of-homopolymers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="homopolymers-and-the-genome" class="section level3 hasAnchor" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Homopolymers and the genome<a href="what-is-sequence-data.html#homopolymers-and-the-genome" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>What are homopolymers -&gt; repeated stretches of identical nucleotides</li>
</ul>
<!-- -->
<ul>
<li><p>On the CHM13 whole human genome assembly v1.1, 29.7% of the 3Gb are in homopolymers (i.e. are identical to the preceding nucleotide in the sequence) Longest HP is 86.</p></li>
<li><p>More than 1.4 million homopolymers 4-mers and up in the exome (i.e. DNA that is translated into proteins)<span class="citation"><sup><a href="#ref-ivadyAnalyticalParametersValidation2018" role="doc-biblioref">76</a></sup></span></p></li>
<li><p>according to<span class="citation"><sup><a href="#ref-booeshaghiPseudoalignmentFacilitatesAssignment2022" role="doc-biblioref">77</a></sup></span>, in the GRCh38 human genome assembly, more than 1.9 Mb are in homopolymers of length 8 or higher, representing about a thousandth of the genome, still a lot of bases to make mistakes on! Longest HP is 90.</p></li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:big-hp-distrib"></span>
<img src="_main_files/figure-html/big-hp-distrib-1.png" alt="Distribution of homopolymer lengths per base in the human genome" width="90%" />
<p class="caption">
Figure 1.2: Distribution of homopolymer lengths per base in the human genome
</p>
</div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:hp-freq"></span>
<img src="_main_files/figure-html/hp-freq-1.png" alt="Homopolymer length as fraction of the genome" width="90%" />
<p class="caption">
Figure 1.3: Homopolymer length as fraction of the genome
</p>
</div>
</div>
<div id="homopolymers-and-long-reads" class="section level3 hasAnchor" number="1.4.2">
<h3><span class="header-section-number">1.4.2</span> Homopolymers and long reads<a href="what-is-sequence-data.html#homopolymers-and-long-reads" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Mainly indels, in certain regions of the genomes, particularly homopolymers. For ONT<span class="citation"><sup><a href="#ref-delahaye2021" role="doc-biblioref">78</a>,<a href="#ref-goodwin2015" role="doc-biblioref">79</a></sup></span>, for PacBIO and ONT<span class="citation"><sup><a href="#ref-dohmBenchmarkingLongreadCorrection2020" role="doc-biblioref">80</a>,<a href="#ref-weirather2017" role="doc-biblioref">81</a></sup></span>. ONT has more homopolymer errors than PacBIO<span class="citation"><sup><a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">43</a></sup></span>. For short reads and PacBio the error rate usually goes up as the homopolymer length grows, however ONT’s error rate in HPs does not depend on the length (even though it’s the highest it’s flat)<span class="citation"><sup><a href="#ref-fooxPerformanceAssessmentDNA2021" role="doc-biblioref">82</a></sup></span>.</p>
</div>
<div id="accounting-for-homopolymers" class="section level3 hasAnchor" number="1.4.3">
<h3><span class="header-section-number">1.4.3</span> Accounting for homopolymers<a href="what-is-sequence-data.html#accounting-for-homopolymers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="specific-error-correction" class="section level4 hasAnchor" number="1.4.3.1">
<h4><span class="header-section-number">1.4.3.1</span> Specific error correction<a href="what-is-sequence-data.html#specific-error-correction" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>In some cases homopolymers are taken into special consideration during polishing<span class="citation"><sup><a href="#ref-simpsonDetectingDNACytosine2017" role="doc-biblioref">64</a>,<a href="#ref-huangHomopolishMethodRemoval2021" role="doc-biblioref">83</a></sup></span>, and computational approaches are developed to deal with homopolymers<span class="citation"><sup><a href="#ref-rangSquiggleBasepairComputational2018" role="doc-biblioref">84</a>,<a href="#ref-sarkozyCallingHomopolymerStretches2018" role="doc-biblioref">85</a></sup></span></li>
<li>Specific technologies:
<ul>
<li><p>R.10 Nanopore:<span class="citation"><sup><a href="#ref-amarasingheOpportunitiesChallengesLongread2020" role="doc-biblioref">56</a>,<a href="#ref-R10NewestNanopore" role="doc-biblioref">86</a></sup></span></p></li>
<li><p>Solid state nanopores:<span class="citation"><sup><a href="#ref-zhouDetectionDNAHomopolymer2019" role="doc-biblioref">87</a>,<a href="#ref-gotoIdentificationFourSinglestranded2018" role="doc-biblioref">88</a></sup></span></p></li>
<li><p>BaseCallers which deal with homopolymers, like can be seen in Fig 3 of<span class="citation"><sup><a href="#ref-wickPerformanceNeuralNetwork2019" role="doc-biblioref">73</a></sup></span> (scrappie raw 1.4.1 or guppy)</p></li>
</ul></li>
<li>Avoid using HPs in barcode sequences<span class="citation"><sup><a href="#ref-hawkinsIndelcorrectingDNABarcodes2018" role="doc-biblioref">89</a>,<a href="#ref-srivathsanMinIONBasedPipeline2018" role="doc-biblioref">90</a></sup></span>.</li>
<li>A little different but HPs have to be taken into account when developing DNA based data storage<span class="citation"><sup><a href="#ref-wangConstructionBioConstrainedCode2019" role="doc-biblioref">91</a></sup></span></li>
</ul>
</div>
<div id="hpc-trick" class="section level4 hasAnchor" number="1.4.3.2">
<h4><span class="header-section-number">1.4.3.2</span> HPC Trick<a href="what-is-sequence-data.html#hpc-trick" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>homopolymers are still tricky regions even with long high accuracy reads -&gt; so HPC</p>
<ul>
<li><p>HPC takes repeated runs of a single nucleotide and compresses them to a single occurrence<br />
<em>Short schematic example of HPC with example where it removes an error ? (don’t talk about mapping yet)</em></p></li>
<li><p>Used in many software tools for long reads<span class="citation"><sup><a href="#ref-ruanFastAccurateLongread2020" role="doc-biblioref">57</a>,<a href="#ref-auImprovingPacBioLong2012" role="doc-biblioref">92</a>–<a href="#ref-ekimMinimizerspaceBruijnGraphs2021" role="doc-biblioref">97</a></sup></span> and even pyrosequencing reads<span class="citation"><sup><a href="#ref-millerAggressiveAssemblyPyrosequencing2008" role="doc-biblioref">98</a></sup></span></p></li>
</ul>
</div>
</div>
</div>
<div id="conclusion" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Conclusion<a href="what-is-sequence-data.html#conclusion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p><strong>Sequencing is an active field:</strong> new methods, both in short reads (ultima) and long reads (illumina infinity), as well as improving existing technologies</p></li>
<li><p><strong>Long reads are good:</strong> Super useful for plenty of very important tasks but…</p></li>
<li><p><strong>Errors are bad:</strong> particularly in HPs since they are an important part of the genome and a major source of error in long reads</p></li>
<li><p><strong>Computational view:</strong> sequences Essentially a text file. Just a long succession of letters, so we can apply String/text algorithmics to it.</p></li>
</ul>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-watson1953" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Watson, J. D. &amp; Crick, F. H. C. <a href="https://doi.org/10.1101/SQB.1953.018.01.020">The Structure of Dna</a>. <em>Cold Spring Harbor Symposia on Quantitative Biology</em> <strong>18</strong>, 123–131 (1953).</div>
</div>
<div id="ref-sangerNucleotideSequenceBacteriophage1977" class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Sanger, F. <em>et al.</em> <a href="https://doi.org/10.1038/265687a0">Nucleotide sequence of bacteriophage <span class="nocase">φX174 DNA</span></a>. <em>Nature</em> <strong>265</strong>, 687–695 (1977).</div>
</div>
<div id="ref-archer2011" class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Archer, C. T. <em>et al.</em> <a href="https://doi.org/10.1186/1471-2164-12-9">The genome sequence of e. Coli w (ATCC 9637): Comparative genome analysis and an improved genome-scale reconstruction of e. coli</a>. <em>BMC Genomics</em> <strong>12</strong>, 9 (2011).</div>
</div>
<div id="ref-nurk2022" class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Nurk, S. <em>et al.</em> <a href="https://doi.org/10.1126/science.abj6987">The complete sequence of a human genome</a>. <em>Science</em> <strong>376</strong>, 44–53 (2022).</div>
</div>
<div id="ref-pellicer2010" class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Pellicer, J., Fay, M. F. &amp; Leitch, I. J. <a href="https://doi.org/10.1111/j.1095-8339.2010.01072.x">The largest eukaryotic genome of them all?</a> <em>Botanical Journal of the Linnean Society</em> <strong>164</strong>, 10–15 (2010).</div>
</div>
<div id="ref-macgregor2001" class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Macgregor, H. C. C-Value Paradox. in (eds. Brenner, S. &amp; Miller, J. H.) 249–250 (Academic Press, 2001). doi:<a href="https://doi.org/10.1006/rwgn.2001.0301">10.1006/rwgn.2001.0301</a>.</div>
</div>
<div id="ref-alberts2002" class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Alberts, B. <em>et al.</em> <em><a href="https://www.ncbi.nlm.nih.gov/books/NBK26916/">Molecular Biology of the Cell. 4th edition</a></em>. (Garland Science, 2002).</div>
</div>
<div id="ref-crick1961" class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Crick, F. H. C., Barnett, L., Brenner, S. &amp; Watts-Tobin, R. J. <a href="https://doi.org/10.1038/1921227a0">General Nature of the Genetic Code for Proteins</a>. <em>Nature</em> <strong>192</strong>, 1227–1232 (1961).</div>
</div>
<div id="ref-internationalhumangenomesequencingconsortiumFinishingEuchromaticSequence2004" class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">International Human Genome Sequencing Consortium. <a href="https://doi.org/10.1038/nature03001">Finishing the euchromatic sequence of the human genome</a>. <em>Nature</em> <strong>431</strong>, 931–945 (2004).</div>
</div>
<div id="ref-elkonCharacterizationNoncodingRegulatory2017" class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Elkon, R. &amp; Agami, R. <a href="https://doi.org/10.1038/nbt.3863">Characterization of noncoding regulatory DNA in the human genome</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 732–746 (2017).</div>
</div>
<div id="ref-omennReflectionsHUPOHuman2021" class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Omenn, G. S. <a href="https://doi.org/10.1016/j.mcpro.2021.100062">Reflections on the HUPO human proteome project, the flagship project of the human proteome organization, at 10 years</a>. <em>Molecular &amp; Cellular Proteomics : MCP</em> <strong>20</strong>, 100062 (2021).</div>
</div>
<div id="ref-shabalinaMammalianTranscriptomeFunction2004" class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Shabalina, S. A. &amp; Spiridonov, N. A. <a href="https://doi.org/10.1186/gb-2004-5-4-105">The mammalian transcriptome and the function of non-coding DNA sequences</a>. <em>Genome Biology</em> <strong>5</strong>, 105 (2004).</div>
</div>
<div id="ref-IntegratedEncyclopediaDNA2012" class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">ENCODE Project Consortium. <a href="https://doi.org/10.1038/nature11247">An integrated encyclopedia of DNA elements in the human genome</a>. <em>Nature</em> <strong>489</strong>, 57–74 (2012).</div>
</div>
<div id="ref-chatterjee2017" class="csl-entry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Chatterjee, N. &amp; Walker, G. C. <a href="https://doi.org/10.1002/em.22087">Mechanisms of DNA damage, repair, and mutagenesis: DNA Damage and Repair</a>. <em>Environmental and Molecular Mutagenesis</em> <strong>58</strong>, 235–263 (2017).</div>
</div>
<div id="ref-fijalkowska2012" class="csl-entry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Fijalkowska, I. J., Schaaper, R. M. &amp; Jonczyk, P. <a href="https://doi.org/10.1111/j.1574-6976.2012.00338.x">DNA replication fidelity in escherichia coli: A multi-DNA polymerase affair</a>. <em>FEMS microbiology reviews</em> <strong>36</strong>, 1105–1121 (2012).</div>
</div>
<div id="ref-pray2008dna" class="csl-entry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Pray, L. DNA replication and causes of mutation. <em>Nature education</em> <strong>1</strong>, 214 (2008).</div>
</div>
<div id="ref-gout2013" class="csl-entry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Gout, J.-F., Thomas, W. K., Smith, Z., Okamoto, K. &amp; Lynch, M. <a href="https://doi.org/10.1073/pnas.1309843110">Large-scale detection of in vivo transcription errors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>110</strong>, 18584–18589 (2013).</div>
</div>
<div id="ref-gout2017" class="csl-entry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Gout, J.-F. <em>et al.</em> <a href="https://doi.org/10.1126/sciadv.1701484">The landscape of transcription errors in eukaryotic cells</a>. <em>Science Advances</em> <strong>3</strong>, e1701484 (2017).</div>
</div>
<div id="ref-desouky2015" class="csl-entry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Desouky, O., Ding, N. &amp; Zhou, G. <a href="https://doi.org/10.1016/j.jrras.2015.03.003">Targeted and non-targeted effects of ionizing radiation</a>. <em>Journal of Radiation Research and Applied Sciences</em> <strong>8</strong>, 247–254 (2015).</div>
</div>
<div id="ref-kiefer2007" class="csl-entry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Kiefer, J. Effects of Ultraviolet Radiation on DNA. in (eds. Obe, G. &amp; Vijayalaxmi) 39–53 (Springer, 2007). doi:<a href="https://doi.org/10.1007/978-3-540-71414-9_3">10.1007/978-3-540-71414-9_3</a>.</div>
</div>
<div id="ref-bennett2003" class="csl-entry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Bennett, J. W. &amp; Klich, M. <a href="https://doi.org/10.1128/CMR.16.3.497-516.2003">Mycotoxins</a>. <em>Clinical Microbiology Reviews</em> <strong>16</strong>, 497–516 (2003).</div>
</div>
<div id="ref-kantidze2016" class="csl-entry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Kantidze, O. L., Velichko, A. K., Luzhin, A. V. &amp; Razin, S. V. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4947990/">Heat stress-induced DNA damage</a>. <em>Acta Naturae</em> <strong>8</strong>, 75–78 (2016).</div>
</div>
<div id="ref-gregory1994" class="csl-entry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Gregory, C. D. &amp; Milner, A. E. <a href="https://doi.org/10.1002/ijc.2910570321">Regulation of cell survival in Burkitt lymphoma: implications from studies of apoptosis following cold-shock treatment</a>. <em>International Journal of Cancer</em> <strong>57</strong>, 419–426 (1994).</div>
</div>
<div id="ref-gafter-gvili2013" class="csl-entry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Gafter-Gvili, A. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0068341">Oxidative stress-induced DNA damage and repair in human peripheral blood mononuclear cells: Protective role of hemoglobin</a>. <em>PLoS ONE</em> <strong>8</strong>, e68341 (2013).</div>
</div>
<div id="ref-kujovichFactorLeidenThrombophilia2011" class="csl-entry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Kujovich, J. L. <a href="https://doi.org/10.1097/GIM.0b013e3181faa0f2">Factor V Leiden thrombophilia</a>. <em>Genetics in Medicine</em> <strong>13</strong>, 1–16 (2011).</div>
</div>
<div id="ref-cuttingCysticFibrosisGenetics2015" class="csl-entry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Cutting, G. R. <a href="https://doi.org/10.1038/nrg3849">Cystic fibrosis genetics: from molecular understanding to clinical application</a>. <em>Nature Reviews Genetics</em> <strong>16</strong>, 45–56 (2015).</div>
</div>
<div id="ref-fuchsbergerGeneticArchitectureType2016" class="csl-entry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Fuchsberger, C. <em>et al.</em> <a href="https://doi.org/10.1038/nature18642">The genetic architecture of type 2 diabetes</a>. <em>Nature</em> <strong>536</strong>, 41–47 (2016).</div>
</div>
<div id="ref-morrisLargescaleAssociationAnalysis2012" class="csl-entry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Morris, A. P. <em>et al.</em> <a href="https://doi.org/10.1038/ng.2383">Large-scale association analysis provides insights into the genetic architecture and pathophysiology of type 2 diabetes</a>. <em>Nature genetics</em> <strong>44</strong>, 981–990 (2012).</div>
</div>
<div id="ref-rheeHumanImmunodeficiencyVirus2003" class="csl-entry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Rhee, S.-Y. <em>et al.</em> <a href="https://doi.org/10.1093/nar/gkg100">Human immunodeficiency virus reverse transcriptase and protease sequence database</a>. <em>Nucleic Acids Research</em> <strong>31</strong>, 298–303 (2003).</div>
</div>
<div id="ref-woodfordEmergenceAntibioticResistance2007" class="csl-entry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Woodford, N. &amp; Ellington, M. J. <a href="https://doi.org/10.1111/j.1469-0691.2006.01492.x">The emergence of antibiotic resistance by mutation</a>. <em>Clinical Microbiology and Infection</em> <strong>13</strong>, 5–18 (2007).</div>
</div>
<div id="ref-sangerDNASequencingChainterminating1977" class="csl-entry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Sanger, F., Nicklen, S. &amp; Coulson, A. R. <a href="https://doi.org/10.1073/pnas.74.12.5463"><span>DNA</span> sequencing with chain-terminating inhibitors</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>74</strong>, 5463–5467 (1977).</div>
</div>
<div id="ref-smithSynthesisOligonucleotidesContaining1985" class="csl-entry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Smith, L. M., Fung, S., Hunkapiller, M. W., Hunkapiller, T. J. &amp; Hood, L. E. <a href="https://doi.org/10.1093/nar/13.7.2399">The synthesis of oligonucleotides containing an aliphatic amino group at the 5′ terminus: Synthesis of fluorescent <span>DNA</span> primers for use in <span>DNA</span> sequence analysis</a>. <em>Nucleic Acids Research</em> <strong>13</strong>, 2399–2412 (1985).</div>
</div>
<div id="ref-smithFluorescenceDetectionAutomated1986" class="csl-entry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Smith, L. M. <em>et al.</em> <a href="https://doi.org/10.1038/321674a0">Fluorescence detection in automated <span>DNA</span> sequence analysis</a>. <em>Nature</em> <strong>321</strong>, 674–679 (1986).</div>
</div>
<div id="ref-shendureNextgenerationDNASequencing2008" class="csl-entry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Shendure, J. &amp; Ji, H. <a href="https://doi.org/10.1038/nbt1486">Next-generation DNA sequencing</a>. <em>Nature Biotechnology</em> <strong>26</strong>, 1135–1145 (2008).</div>
</div>
<div id="ref-CostSequencingHuman" class="csl-entry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><a href="https://www.genome.gov/about-genomics/fact-sheets/Sequencing-Human-Genome-cost">The Cost of Sequencing a Human Genome</a>.</div>
</div>
<div id="ref-collinsHumanGenomeProject2003" class="csl-entry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Collins, F. S., Morgan, M. &amp; Patrinos, A. <a href="https://doi.org/10.1126/science.1084564">The human genome project: Lessons from large-scale biology</a>. <em>Science</em> <strong>300</strong>, 286–290 (2003).</div>
</div>
<div id="ref-liuComparisonNextGenerationSequencing2012" class="csl-entry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Liu, L. <em>et al.</em> <a href="https://doi.org/10.1155/2012/251364">Comparison of <span>Next-Generation Sequencing Systems</span></a>. <em>Journal of Biomedicine and Biotechnology</em> <strong>2012</strong>, e251364 (2012).</div>
</div>
<div id="ref-metzkerSequencingTechnologiesNext2010" class="csl-entry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Metzker, M. L. <a href="https://doi.org/10.1038/nrg2626">Sequencing technologies — the next generation</a>. <em>Nat Rev Genet</em> <strong>11</strong>, 31–46 (2010).</div>
</div>
<div id="ref-mardisDecadePerspectiveDNA2011" class="csl-entry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Mardis, E. R. <a href="https://doi.org/10.1038/nature09796">A decade<span>’</span>s perspective on DNA sequencing technology</a>. <em>Nature</em> <strong>470</strong>, 198–203 (2011).</div>
</div>
<div id="ref-eidRealTimeDNASequencing2009" class="csl-entry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline">Eid, J. <em>et al.</em> <a href="https://doi.org/10.1126/science.1162986">Real-time DNA sequencing from single polymerase molecules</a>. <em>Science</em> <strong>323</strong>, 133–138 (2009).</div>
</div>
<div id="ref-rhoadsPacBioSequencingIts2015" class="csl-entry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Rhoads, A. &amp; Au, K. F. <a href="https://doi.org/10.1016/j.gpb.2015.08.002">PacBio Sequencing and Its Applications</a>. <em>Genomics, Proteomics &amp; Bioinformatics</em> <strong>13</strong>, 278–289 (2015).</div>
</div>
<div id="ref-chaissonResolvingComplexityHuman2015" class="csl-entry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Chaisson, M. J. P. <em>et al.</em> <a href="https://doi.org/10.1038/nature13907">Resolving the complexity of the human genome using single-molecule sequencing</a>. <em>Nature</em> <strong>517</strong>, 608–611 (2015).</div>
</div>
<div id="ref-logsdonLongreadHumanGenome2020" class="csl-entry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Logsdon, G. A., Vollger, M. R. &amp; Eichler, E. E. <a href="https://doi.org/10.1038/s41576-020-0236-x">Long-read human genome sequencing and its applications</a>. <em>Nat Rev Genet</em> <strong>21</strong>, 597–614 (2020).</div>
</div>
<div id="ref-murigneuxComparisonLongreadMethods2020" class="csl-entry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Murigneux, V. <em>et al.</em> <a href="https://doi.org/10.1093/gigascience/giaa146">Comparison of long-read methods for sequencing and assembly of a plant genome</a>. <em>GigaScience</em> <strong>9</strong>, giaa146 (2020).</div>
</div>
<div id="ref-clarkeContinuousBaseIdentification2009" class="csl-entry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Clarke, J. <em>et al.</em> <a href="https://doi.org/10.1038/nnano.2009.12">Continuous base identification for single-molecule nanopore DNA sequencing</a>. <em>Nature Nanotechnology</em> <strong>4</strong>, 265–270 (2009).</div>
</div>
<div id="ref-ipMinIONAnalysisReference2015" class="csl-entry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline">Ip, C. L. C. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.7201.1">MinION analysis and reference consortium: Phase 1 data release and analysis</a>. <em>F1000Research</em> <strong>4</strong>, 1075 (2015).</div>
</div>
<div id="ref-jainNanoporeSequencingAssembly2018" class="csl-entry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Jain, M. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.4060">Nanopore sequencing and assembly of a human genome with ultra-long reads</a>. <em>Nature Biotechnology</em> <strong>36</strong>, 338–345 (2018).</div>
</div>
<div id="ref-TharSheBlows" class="csl-entry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><a href="http://lab.loman.net/2017/03/09/ultrareads-for-nanopore/">Thar she blows! Ultra long read method for nanopore sequencing · loman labs</a>.</div>
</div>
<div id="ref-payneBulkVisGraphicalViewer2019" class="csl-entry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">Payne, A., Holmes, N., Rakyan, V. &amp; Loose, M. <a href="https://doi.org/10.1093/bioinformatics/bty841">BulkVis: A graphical viewer for oxford nanopore bulk FAST5 files</a>. <em>Bioinformatics</em> <strong>35</strong>, 2193–2198 (2019).</div>
</div>
<div id="ref-jainOxfordNanoporeMinION2016" class="csl-entry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">Jain, M., Olsen, H. E., Paten, B. &amp; Akeson, M. <a href="https://doi.org/10.1186/s13059-016-1103-0">The Oxford Nanopore MinION: delivery of nanopore sequencing to the genomics community</a>. <em>Genome Biology</em> <strong>17</strong>, 239 (2016).</div>
</div>
<div id="ref-huntProteinSequencingTandem1986" class="csl-entry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Hunt, D. F., Yates, J. R., Shabanowitz, J., Winston, S. &amp; Hauer, C. R. <a href="https://doi.org/10.1073/pnas.83.17.6233">Protein sequencing by tandem mass spectrometry.</a> <em>Proceedings of the National Academy of Sciences</em> <strong>83</strong>, 6233–6237 (1986).</div>
</div>
<div id="ref-smithProteinSequencingProtocols2002" class="csl-entry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Smith, B. J. <em>Protein Sequencing Protocols</em>. (Springer Science &amp; Business Media, 2002).</div>
</div>
<div id="ref-restrepo-perezPavingWaySinglemolecule2018" class="csl-entry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Restrepo-Pérez, L., Joo, C. &amp; Dekker, C. <a href="https://doi.org/10.1038/s41565-018-0236-6">Paving the way to single-molecule protein sequencing</a>. <em>Nature Nanotechnology</em> <strong>13</strong>, 786–796 (2018).</div>
</div>
<div id="ref-maAnalysisErrorProfiles2019" class="csl-entry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline">Ma, X. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-019-1659-6">Analysis of error profiles in deep next-generation sequencing data</a>. <em>Genome Biology</em> <strong>20</strong>, 50 (2019).</div>
</div>
<div id="ref-limaComparativeAssessmentLongread2020" class="csl-entry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline">Lima, L. <em>et al.</em> <a href="https://doi.org/10.1093/bib/bbz058">Comparative assessment of long-read error correction software applied to nanopore RNA-sequencing data</a>. <em>Briefings in Bioinformatics</em> <strong>21</strong>, 1164–1181 (2020).</div>
</div>
<div id="ref-amarasingheOpportunitiesChallengesLongread2020" class="csl-entry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline">Amarasinghe, S. L. <em>et al.</em> <a href="https://doi.org/10.1186/s13059-020-1935-5">Opportunities and challenges in long-read sequencing data analysis</a>. <em>Genome Biology</em> <strong>21</strong>, 30 (2020).</div>
</div>
<div id="ref-ruanFastAccurateLongread2020" class="csl-entry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline">Ruan, J. &amp; Li, H. <a href="https://doi.org/10.1038/s41592-019-0669-3">Fast and accurate long-read assembly with wtdbg2</a>. <em>Nature Methods</em> <strong>17</strong>, 155–158 (2020).</div>
</div>
<div id="ref-korenCanuScalableAccurate2017" class="csl-entry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline">Koren, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.215087.116">Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation</a>. <em>Genome Research</em> <strong>27</strong>, 722–736 (2017).</div>
</div>
<div id="ref-tischlerNonHybridLong2017" class="csl-entry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline">Tischler, G. &amp; Myers, E. W. <em>Non Hybrid Long Read Consensus Using Local De Bruijn Graph Assembly</em>. 106252 <a href="https://www.biorxiv.org/content/10.1101/106252v1">https://www.biorxiv.org/content/10.1101/106252v1</a> (2017) doi:<a href="https://doi.org/10.1101/106252">10.1101/106252</a>.</div>
</div>
<div id="ref-hacklProovreadLargescaleHighaccuracy2014" class="csl-entry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline">Hackl, T., Hedrich, R., Schultz, J. &amp; Förster, F. <a href="https://doi.org/10.1093/bioinformatics/btu392">Proovread : Large-scale high-accuracy PacBio correction through iterative short read consensus</a>. <em>Bioinformatics</em> <strong>30</strong>, 3004–3011 (2014).</div>
</div>
<div id="ref-miclotteJabbaHybridError2016" class="csl-entry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline">Miclotte, G. <em>et al.</em> <a href="https://doi.org/10.1186/s13015-016-0075-7">Jabba: Hybrid error correction for long sequencing reads</a>. <em>Algorithms for Molecular Biology</em> <strong>11</strong>, 10 (2016).</div>
</div>
<div id="ref-salmelaLoRDECAccurateEfficient2014" class="csl-entry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline">Salmela, L. &amp; Rivals, E. <a href="https://doi.org/10.1093/bioinformatics/btu538">LoRDEC: Accurate and efficient long read error correction</a>. <em>Bioinformatics</em> <strong>30</strong>, 3506–3514 (2014).</div>
</div>
<div id="ref-heplerImprovedCircularConsensus2016" class="csl-entry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline">Hepler, N. L. <em>et al.</em> An improved circular consensus algorithm with an application to detect HIV-1 drug-resistance associated mutations (DRAMs). in (2016).</div>
</div>
<div id="ref-simpsonDetectingDNACytosine2017" class="csl-entry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline">Simpson, J. T. <em>et al.</em> <a href="https://doi.org/10.1038/nmeth.4184">Detecting DNA cytosine methylation using nanopore sequencing</a>. <em>Nature Methods</em> <strong>14</strong>, 407–410 (2017).</div>
</div>
<div id="ref-walkerPilonIntegratedTool2014" class="csl-entry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline">Walker, B. J. <em>et al.</em> <a href="https://doi.org/10.1371/journal.pone.0112963">Pilon: An Integrated Tool for Comprehensive Microbial Variant Detection and Genome Assembly Improvement</a>. <em>PLOS ONE</em> <strong>9</strong>, e112963 (2014).</div>
</div>
<div id="ref-vaserFastAccurateNovo2017" class="csl-entry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline">Vaser, R., Sović, I., Nagarajan, N. &amp; Šikić, M. <a href="https://doi.org/10.1101/gr.214270.116">Fast and accurate de novo genome assembly from long uncorrected reads</a>. <em>Genome Research</em> <strong>27</strong>, 737–746 (2017).</div>
</div>
<div id="ref-fuComparativeEvaluationHybrid2019" class="csl-entry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Fu, S., Wang, A. &amp; Au, K. F. <a href="https://doi.org/10.1186/s13059-018-1605-z">A comparative evaluation of hybrid error correction methods for error-prone long reads</a>. <em>Genome Biology</em> <strong>20</strong>, 26 (2019).</div>
</div>
<div id="ref-zhangComprehensiveEvaluationLong2020" class="csl-entry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Zhang, H., Jain, C. &amp; Aluru, S. <a href="https://doi.org/10.1186/s12864-020-07227-0">A comprehensive evaluation of long read error correction methods</a>. <em>BMC Genomics</em> <strong>21</strong>, 889 (2020).</div>
</div>
<div id="ref-wenger2019" class="csl-entry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline">Wenger, A. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-019-0217-9">Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome</a>. <em>Nature Biotechnology</em> <strong>37</strong>, 1155–1162 (2019).</div>
</div>
<div id="ref-karstHighaccuracyLongreadAmplicon2021" class="csl-entry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline">Karst, S. M. <em>et al.</em> <a href="https://doi.org/10.1038/s41592-020-01041-y">High-accuracy long-read amplicon sequences using unique molecular identifiers with Nanopore or PacBio sequencing</a>. <em>Nature Methods</em> <strong>18</strong>, 165–169 (2021).</div>
</div>
<div id="ref-peresiniNanoporeBaseCalling2021" class="csl-entry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline">Perešíni, P., Boža, V., Brejová, B. &amp; Vinař, T. <a href="https://doi.org/10.1093/bioinformatics/btab528">Nanopore base calling on the edge</a>. <em>Bioinformatics</em> <strong>37</strong>, 4661–4667 (2021).</div>
</div>
<div id="ref-wickPerformanceNeuralNetwork2019" class="csl-entry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline">Wick, R. R., Judd, L. M. &amp; Holt, K. E. <a href="https://doi.org/10.1186/s13059-019-1727-y">Performance of neural network basecalling tools for Oxford Nanopore sequencing</a>. <em>Genome Biology</em> <strong>20</strong>, 129 (2019).</div>
</div>
<div id="ref-chenHighlyAccurateFluorogenic2017" class="csl-entry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline">Chen, Z. <em>et al.</em> <a href="https://doi.org/10.1038/nbt.3982">Highly accurate fluorogenic DNA sequencing with information theory<span></span>based error correction</a>. <em>Nature Biotechnology</em> <strong>35</strong>, 1170–1178 (2017).</div>
</div>
<div id="ref-HighPerformanceLong" class="csl-entry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline"><a href="https://www.illumina.com/content/illumina-marketing/amr/en_US/science/genomics-research/articles/infinity-high-performance-long-read-assay.html">High Performance Long Read Assay Enables Contiguous Data up to 10Kb on Existing Illumina Platforms</a>.</div>
</div>
<div id="ref-ivadyAnalyticalParametersValidation2018" class="csl-entry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline">Ivády, G. <em>et al.</em> <a href="https://doi.org/10.1186/s12864-018-4544-x">Analytical parameters and validation of homopolymer detection in a pyrosequencing-based next generation sequencing system</a>. <em>BMC Genomics</em> <strong>19</strong>, 158 (2018).</div>
</div>
<div id="ref-booeshaghiPseudoalignmentFacilitatesAssignment2022" class="csl-entry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline">Booeshaghi, A. S. &amp; Pachter, L. Pseudoalignment facilitates assignment of error-prone Ultima Genomics reads. doi:<a href="https://doi.org/10.1101/2022.06.04.494845">10.1101/2022.06.04.494845</a>.</div>
</div>
<div id="ref-delahaye2021" class="csl-entry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline">Delahaye, C. &amp; Nicolas, J. <a href="https://doi.org/10.1371/journal.pone.0257521">Sequencing DNA with nanopores: Troubles and biases</a>. <em>PLOS ONE</em> <strong>16</strong>, e0257521 (2021).</div>
</div>
<div id="ref-goodwin2015" class="csl-entry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline">Goodwin, S. <em>et al.</em> <a href="https://doi.org/10.1101/gr.191395.115">Oxford Nanopore sequencing, hybrid error correction, and de novo assembly of a eukaryotic genome</a>. <em>Genome Research</em> <strong>25</strong>, 1750–1756 (2015).</div>
</div>
<div id="ref-dohmBenchmarkingLongreadCorrection2020" class="csl-entry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline">Dohm, J. C., Peters, P., Stralis-Pavese, N. &amp; Himmelbauer, H. <a href="https://doi.org/10.1093/nargab/lqaa037">Benchmarking of long-read correction methods</a>. <em>NAR Genomics and Bioinformatics</em> <strong>2</strong>, (2020).</div>
</div>
<div id="ref-weirather2017" class="csl-entry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline">Weirather, J. L. <em>et al.</em> <a href="https://doi.org/10.12688/f1000research.10571.2">Comprehensive comparison of pacific biosciences and oxford nanopore technologies and their applications to transcriptome analysis</a>. <em>F1000Research</em> <strong>6</strong>, 100 (2017).</div>
</div>
<div id="ref-fooxPerformanceAssessmentDNA2021" class="csl-entry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline">Foox, J. <em>et al.</em> <a href="https://doi.org/10.1038/s41587-021-01049-5">Performance assessment of DNA sequencing platforms in the ABRF Next-Generation Sequencing Study</a>. <em>Nature Biotechnology</em> <strong>39</strong>, 1129–1140 (2021).</div>
</div>
<div id="ref-huangHomopolishMethodRemoval2021" class="csl-entry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline">Huang, Y.-T., Liu, P.-Y. &amp; Shih, P.-W. <a href="https://doi.org/10.1186/s13059-021-02282-6">Homopolish: A method for the removal of systematic errors in nanopore sequencing by homologous polishing</a>. <em>Genome Biology</em> <strong>22</strong>, 95 (2021).</div>
</div>
<div id="ref-rangSquiggleBasepairComputational2018" class="csl-entry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline">Rang, F. J., Kloosterman, W. P. &amp; Ridder, J. de. <a href="https://doi.org/10.1186/s13059-018-1462-9">From squiggle to basepair: Computational approaches for improving nanopore sequencing read accuracy</a>. <em>Genome Biology</em> <strong>19</strong>, 90 (2018).</div>
</div>
<div id="ref-sarkozyCallingHomopolymerStretches2018" class="csl-entry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline">Sarkozy, P., Jobbágy, Á. &amp; Antal, P. Calling Homopolymer Stretches from Raw Nanopore Reads by Analyzing k-mer Dwell Times. in (eds. Eskola, H., Väisänen, O., Viik, J. &amp; Hyttinen, J.) 241–244 (Springer, 2018). doi:<a href="https://doi.org/10.1007/978-981-10-5122-7_61">10.1007/978-981-10-5122-7_61</a>.</div>
</div>
<div id="ref-R10NewestNanopore" class="csl-entry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline"><a href="http://nanoporetech.com/about-us/news/r103-newest-nanopore-high-accuracy-nanopore-sequencing-now-available-store">R10.3: the newest nanopore for high accuracy nanopore sequencing <span></span> now available in store</a>.</div>
</div>
<div id="ref-zhouDetectionDNAHomopolymer2019" class="csl-entry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline">Zhou, L. <em>et al.</em> <a href="https://doi.org/10.1116/1.5116295">Detection of DNA homopolymer with graphene nanopore</a>. <em>Journal of Vacuum Science &amp; Technology B</em> <strong>37</strong>, 061809 (2019).</div>
</div>
<div id="ref-gotoIdentificationFourSinglestranded2018" class="csl-entry">
<div class="csl-left-margin">88. </div><div class="csl-right-inline">Goto, Y., Yanagi, I., Matsui, K., Yokoi, T. &amp; Takeda, K. <a href="https://doi.org/10.1039/C8NR04238A">Identification of four single-stranded DNA homopolymers with a solid-state nanopore in alkaline CsCl solution</a>. <em>Nanoscale</em> <strong>10</strong>, 20844–20850 (2018).</div>
</div>
<div id="ref-hawkinsIndelcorrectingDNABarcodes2018" class="csl-entry">
<div class="csl-left-margin">89. </div><div class="csl-right-inline">Hawkins, J. A., Jones, S. K., Finkelstein, I. J. &amp; Press, W. H. <a href="https://doi.org/10.1073/pnas.1802640115">Indel-correcting DNA barcodes for high-throughput sequencing</a>. <em>Proceedings of the National Academy of Sciences</em> <strong>115</strong>, E6217–E6226 (2018).</div>
</div>
<div id="ref-srivathsanMinIONBasedPipeline2018" class="csl-entry">
<div class="csl-left-margin">90. </div><div class="csl-right-inline">Srivathsan, A. <em>et al.</em> <a href="https://doi.org/10.1111/1755-0998.12890">A MinION<span></span>-based pipeline for fast and cost-effective DNA barcoding</a>. <em>Molecular Ecology Resources</em> <strong>18</strong>, 1035–1049 (2018).</div>
</div>
<div id="ref-wangConstructionBioConstrainedCode2019" class="csl-entry">
<div class="csl-left-margin">91. </div><div class="csl-right-inline">Wang, Y., Noor-A-Rahim, Md., Gunawan, E., Guan, Y. L. &amp; Poh, C. L. <a href="https://doi.org/10.1109/LCOMM.2019.2912572">Construction of bio-constrained code for DNA data storage</a>. <em>IEEE Communications Letters</em> <strong>23</strong>, 963–966 (2019).</div>
</div>
<div id="ref-auImprovingPacBioLong2012" class="csl-entry">
<div class="csl-left-margin">92. </div><div class="csl-right-inline">Au, K. F., Underwood, J. G., Lee, L. &amp; Wong, W. H. <a href="https://doi.org/10.1371/journal.pone.0046679">Improving <span>PacBio Long Read Accuracy</span> by <span>Short Read Alignment</span></a>. <em>PLOS ONE</em> <strong>7</strong>, e46679 (2012).</div>
</div>
<div id="ref-ekimMinimizerspaceBruijnGraphs2021" class="csl-entry">
<div class="csl-left-margin">97. </div><div class="csl-right-inline">Ekim, B., Berger, B. &amp; Chikhi, R. <a href="https://doi.org/10.1016/j.cels.2021.08.009">Minimizer-space de <span>Bruijn</span> graphs: Whole-genome assembly of long reads in minutes on a personal computer</a>. <em>Cell Systems</em> <strong>12</strong>, 958–968.e6 (2021).</div>
</div>
<div id="ref-millerAggressiveAssemblyPyrosequencing2008" class="csl-entry">
<div class="csl-left-margin">98. </div><div class="csl-right-inline">Miller, J. R. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btn548">Aggressive assembly of pyrosequencing reads with mates</a>. <em>Bioinformatics</em> <strong>24</strong>, 2818–2824 (2008).</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="general-introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="aligning-sequence-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/01-sequence-data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
