<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 What is Sequence data ? | From sequence to knowledge, improving and learning from sequence alignments.</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.24.11 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 What is Sequence data ? | From sequence to knowledge, improving and learning from sequence alignments." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 What is Sequence data ? | From sequence to knowledge, improving and learning from sequence alignments." />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-05-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="general-introduction.html"/>
<link rel="next" href="aligning-sequence-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">My PhD thesis.</a></li>

<li class="divider"></li>
<li><a href="index.html#abstract">Abstract<span></span></a></li>
<li><a href="resume.html#resume">Resume<span></span></a></li>
<li><a href="aknowledgments.html#aknowledgments">Aknowledgments<span></span></a></li>
<li><a href="glosssary.html#glosssary">Glosssary<span></span></a></li>
<li class="chapter" data-level="1" data-path="general-introduction.html"><a href="general-introduction.html"><i class="fa fa-check"></i><b>1</b> General Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="general-introduction.html"><a href="general-introduction.html#goal-of-this-thesis"><i class="fa fa-check"></i><b>1.1</b> Goal of this Thesis<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="general-introduction.html"><a href="general-introduction.html#thesis-organization"><i class="fa fa-check"></i><b>1.2</b> Thesis organization<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>2</b> What is Sequence data ?<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>2.1</b> Biological sequences, a primer<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>2.1.1</b> What is DNA<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#rna-and-transcription"><i class="fa fa-check"></i><b>2.1.2</b> RNA and transcription<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-are-proteins"><i class="fa fa-check"></i><b>2.1.3</b> What are proteins ?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>2.2</b> Obtaining sequence data<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequenceing-a-breakthrough"><i class="fa fa-check"></i><b>2.2.1</b> Sanger sequenceing, a breakthrough<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Next-generation sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#long-read-sequencing-the-next-step-in-genomics"><i class="fa fa-check"></i><b>2.2.3</b> Long read sequencing, the next step in genomics<span></span></a></li>
<li class="chapter" data-level="2.2.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#protein-sequencing"><i class="fa fa-check"></i><b>2.2.4</b> Protein sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>2.3</b> Sequencing errors, how to account for them ?<span></span></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#ngs-errors"><i class="fa fa-check"></i><b>2.3.1</b> NGS errors<span></span></a></li>
<li class="chapter" data-level="2.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#long-read-errors"><i class="fa fa-check"></i><b>2.3.2</b> Long read errors<span></span></a></li>
<li class="chapter" data-level="2.3.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#hpc"><i class="fa fa-check"></i><b>2.3.3</b> HPC<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>3</b> Aligning sequence data<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>3.1</b> What is an alignment ?<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#local-vs-global"><i class="fa fa-check"></i><b>3.1.1</b> local vs global<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#dynamic-programming-and-nw-sw"><i class="fa fa-check"></i><b>3.1.2</b> dynamic programming and NW, SW<span></span></a></li>
<li class="chapter" data-level="3.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#substitution-models"><i class="fa fa-check"></i><b>3.1.3</b> Substitution models<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#pairwise-sequence-alignment-and-mapping"><i class="fa fa-check"></i><b>3.2</b> Pairwise sequence alignment and mapping<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#naive-pairwise"><i class="fa fa-check"></i><b>3.2.1</b> naive pairwise<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#indexed-blast"><i class="fa fa-check"></i><b>3.2.2</b> Indexed … BLAST,…<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#seeded-minimap2"><i class="fa fa-check"></i><b>3.2.3</b> Seeded, minimap2, …<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#msa"><i class="fa fa-check"></i><b>3.3</b> MSA<span></span></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#msa-is-a-difficult-problem"><i class="fa fa-check"></i><b>3.3.1</b> MSA is a difficult problem<span></span></a></li>
<li class="chapter" data-level="3.3.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#methods"><i class="fa fa-check"></i><b>3.3.2</b> Methods<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><i class="fa fa-check"></i><b>4</b> Mapping-friendly sequence reductions: going beyond homopolymer compression<span></span></a>
<ul>
<li><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-abstract">Abstract<span></span></a></li>
<li class="chapter" data-level="4.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#methods-1"><i class="fa fa-check"></i><b>4.2</b> Methods<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:msr-def"><i class="fa fa-check"></i><b>4.2.1</b> Mapping-friendly sequence reductions<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:enum"><i class="fa fa-check"></i><b>4.2.2</b> Restricting the space of mapping-friendly sequence reductions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#datasets-and-pipelines"><i class="fa fa-check"></i><b>4.3</b> Datasets and Pipelines<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#datasets"><i class="fa fa-check"></i><b>4.3.1</b> Datasets<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#simulation-pipeline"><i class="fa fa-check"></i><b>4.3.2</b> Simulation pipeline<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:evalpipeline"><i class="fa fa-check"></i><b>4.3.3</b> Evaluation pipeline<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#results"><i class="fa fa-check"></i><b>4.4</b> Results<span></span></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>4.4.1</b> Selection of mapping-friendly sequence reductions<span></span></a></li>
<li class="chapter" data-level="4.4.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>4.4.2</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes<span></span></a></li>
<li class="chapter" data-level="4.4.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>4.4.3</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome<span></span></a></li>
<li class="chapter" data-level="4.4.4" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>4.4.4</b> Raw mapping improves upon HPC on centromeric regions<span></span></a></li>
<li class="chapter" data-level="4.4.5" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>4.4.5</b> Positions of incorrectly mapped reads across the entire human genome<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#discussion"><i class="fa fa-check"></i><b>4.5</b> Discussion<span></span></a></li>
<li class="chapter" data-level="4.6" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#limitations-of-this-study"><i class="fa fa-check"></i><b>4.6</b> Limitations of this study<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#code-availability"><i class="fa fa-check"></i><b>4.7</b> Code availability<span></span></a></li>
<li><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#supplementary-information">Supplementary information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html"><i class="fa fa-check"></i><b>5</b> Learning from alignments<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#alignments-are-a-rich-source-of-information"><i class="fa fa-check"></i><b>5.1</b> Alignments are a rich source of information<span></span></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#pairwise-alns"><i class="fa fa-check"></i><b>5.1.1</b> Pairwise alns<span></span></a></li>
<li class="chapter" data-level="5.1.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#msa-1"><i class="fa fa-check"></i><b>5.1.2</b> MSA<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#preprocessing-the-alignment"><i class="fa fa-check"></i><b>5.2</b> Preprocessing the alignment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#embedding-the-alignment"><i class="fa fa-check"></i><b>5.2.1</b> Embedding the alignment<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#choosing-a-learning-target"><i class="fa fa-check"></i><b>5.2.2</b> Choosing a learning target<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#how-to-learn-from-alns"><i class="fa fa-check"></i><b>5.3</b> How to learn from ALNs<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>5.3.1</b> Tests and statistical learning<span></span></a></li>
<li class="chapter" data-level="5.3.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#taking-interactions-into-account"><i class="fa fa-check"></i><b>5.3.2</b> Taking interactions into account<span></span></a></li>
<li class="chapter" data-level="5.3.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#deep-learning"><i class="fa fa-check"></i><b>5.3.3</b> Deep Learning<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html"><i class="fa fa-check"></i><b>6</b> HIV and DRMs<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-are-viruses"><i class="fa fa-check"></i><b>6.1</b> What are viruses<span></span></a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#small-presentationdefinition-of-viruses"><i class="fa fa-check"></i><b>6.1.1</b> small presentation/definition of viruses<span></span></a></li>
<li class="chapter" data-level="6.1.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#dnarna-viruses"><i class="fa fa-check"></i><b>6.1.2</b> DNA/RNA viruses<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#hiv"><i class="fa fa-check"></i><b>6.2</b> HIV<span></span></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#presentation-of-hiv"><i class="fa fa-check"></i><b>6.2.1</b> Presentation of HIV<span></span></a></li>
<li class="chapter" data-level="6.2.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#replication-cycle-of-hiv"><i class="fa fa-check"></i><b>6.2.2</b> Replication cycle of HIV<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#drms"><i class="fa fa-check"></i><b>6.3</b> DRMs<span></span></a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#how-does-art-work"><i class="fa fa-check"></i><b>6.3.1</b> How does ART work<span></span></a></li>
<li class="chapter" data-level="6.3.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#different-types-of-resistance"><i class="fa fa-check"></i><b>6.3.2</b> different types of resistance<span></span></a></li>
<li class="chapter" data-level="6.3.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#consequences-on-global-health"><i class="fa fa-check"></i><b>6.3.3</b> Consequences on global health<span></span></a></li>
<li class="chapter" data-level="6.3.4" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#finding-drms"><i class="fa fa-check"></i><b>6.3.4</b> Finding DRMS<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>7</b> Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV<span></span></a>
<ul>
<li><a href="HIV-paper.html#abstract-paper">Abstract<span></span></a></li>
<li><a href="HIV-paper.html#author-summary">Author summary<span></span></a></li>
<li class="chapter" data-level="7.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#introduction"><i class="fa fa-check"></i><b>7.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>7.2</b> Materials and methods<span></span></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>7.2.1</b> Data<span></span></a></li>
<li class="chapter" data-level="7.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>7.2.2</b> Classifier training<span></span></a></li>
<li class="chapter" data-level="7.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>7.2.3</b> Measuring classifier performance<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#results"><i class="fa fa-check"></i><b>7.3</b> Results<span></span></a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>7.3.1</b> Classifier performance &amp; interpretation<span></span></a></li>
<li class="chapter" data-level="7.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>7.3.2</b> Additional classification results<span></span></a></li>
<li class="chapter" data-level="7.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>7.3.3</b> Identifying new mutations from classifiers<span></span></a></li>
<li class="chapter" data-level="7.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>7.3.4</b> Detailed analysis of potentially resistance-associated mutations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>7.4</b> Discussion and perspectives<span></span></a></li>
<li><a href="HIV-paper.html#acknowledgments">Acknowledgments<span></span></a></li>
<li><a href="HIV-paper.html#supporting-information">Supporting Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>8</b> Learning alignments, an interesting perspective<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-pairwise-alignment"><i class="fa fa-check"></i><b>8.1</b> Learning pairwise alignment<span></span></a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#transformers-and-deep-embedding"><i class="fa fa-check"></i><b>8.1.1</b> Transformers and deep embedding<span></span></a></li>
<li class="chapter" data-level="8.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#dedal"><i class="fa fa-check"></i><b>8.1.2</b> DEDAL<span></span></a></li>
<li class="chapter" data-level="8.1.3" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predictind-an-alignment"><i class="fa fa-check"></i><b>8.1.3</b> predictind an alignment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#what-else-could-we-learn"><i class="fa fa-check"></i><b>8.2</b> What else could we learn ?<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-to-predict-seeds-or-starting-positions"><i class="fa fa-check"></i><b>8.2.1</b> Learn to predict seeds or starting positions<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-pre-processing-functions"><i class="fa fa-check"></i><b>8.2.2</b> Learn pre-processing functions<span></span></a></li>
</ul></li>
</ul></li>
<li><a href="global-references.html#global-references">Global References<span></span></a></li>
<li class="appendix"><span><b>Appendix<span></span></b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”<span></span></a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation<span></span></a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#tab:mapperComparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison<span></span></a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#fig:error-hists"><i class="fa fa-check"></i><b>A.3</b> Origin of incorrectly mapped reads of high mapping quality on whole human genome.<span></span></a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:all-reads-hist"><i class="fa fa-check"></i><b>A.4</b> Analyzing read origin on whole human genome<span></span></a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.5</b> Performance of MSRs on the Drosophila genome<span></span></a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”<span></span></a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>B.1</b> S1 Appendix (Technical appendix).<span></span></a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>B.1.1</b> Data<span></span></a></li>
<li class="chapter" data-level="B.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>B.1.2</b> Classifiers<span></span></a></li>
<li class="chapter" data-level="B.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>B.1.3</b> Scoring<span></span></a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>B.2</b> S1 Fig.<span></span></a></li>
<li class="chapter" data-level="B.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>B.3</b> S2 Fig.<span></span></a></li>
<li class="chapter" data-level="B.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>B.4</b> S3 Fig.<span></span></a></li>
<li class="chapter" data-level="B.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>B.5</b> S1 Table.<span></span></a></li>
<li class="chapter" data-level="B.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>B.6</b> S2 Appendix. (Fisher exact tests)<span></span></a></li>
<li class="chapter" data-level="B.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>B.7</b> S1 Data.<span></span></a></li>
<li class="chapter" data-level="B.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>B.8</b> S2 Data.<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequence to knowledge, improving and learning from sequence alignments.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="what-is-sequence-data" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> What is Sequence data ?<a href="what-is-sequence-data.html#what-is-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="biological-sequences-a-primer" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Biological sequences, a primer<a href="what-is-sequence-data.html#biological-sequences-a-primer" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To fully understand the work that was done during this thesis, as well as the choices that were made some basic knowledge of biology and more particularly genetics are needed. If you are already familiar with biological sequences, feel free to skip ahead.</p>
<div id="what-is-dna" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> What is DNA<a href="what-is-sequence-data.html#what-is-dna" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>D</strong>esoxyribo<strong>N</strong>ucleic <strong>A</strong>cid (DNA) is one of the most important molecules there is, without it complex life as we know it is impossible. It contains all the genetic information of a given organism, that is to say all the information necessary for the organism to: 1) function as a living being and 2) make a perfect copy of itself. This is the case for the overwhelming majority of living organisms on planet earth, from elephants to potatoes, to micro-organisms like bacteria.</p>
<p>DNA is a polymer, composed of monomeric units called nucleotides. Each nucleotide is composed of Ribose (a five carbon sugar) on which are attached a phosphate group as well as one of four nucleobases: Adenine (A), Cytosine (C), Guanine (G) of Thymine (T). These 4 types of nucleotide monomers link up with one-another, creating the phosphate-sugar backbone of single strand of DNA. The ordered sequence of these four types of nucleotides in a single strand encodes all the genetic information necessary to life. Nucleotides can form strong complementary bonds with each other, A with T and C with G. These bonds allows two strands of DNA to link up with each other, forming the double-helix structure of DNA as in Figure <a href="what-is-sequence-data.html#fig:figDNA">2.1</a>. The specificity of nucleotide bonds ensure that the two strands of the double helix are complementary and that the information contained in one strand can be recovered from the other.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figDNA"></span>
<img src="figures/Basic-Intro/DNA.png" alt="**Double-helix** structure of DNA" width="90%" />
<p class="caption">
Figure 2.1: <strong>Double-helix</strong> structure of DNA
</p>
</div>
</div>
<div id="rna-and-transcription" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> RNA and transcription<a href="what-is-sequence-data.html#rna-and-transcription" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="what-are-proteins" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> What are proteins ?<a href="what-is-sequence-data.html#what-are-proteins" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
</div>
<div id="obtaining-sequence-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Obtaining sequence data<a href="what-is-sequence-data.html#obtaining-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="sanger-sequenceing-a-breakthrough" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Sanger sequenceing, a breakthrough<a href="what-is-sequence-data.html#sanger-sequenceing-a-breakthrough" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="next-generation-sequencing" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Next-generation sequencing<a href="what-is-sequence-data.html#next-generation-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="long-read-sequencing-the-next-step-in-genomics" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Long read sequencing, the next step in genomics<a href="what-is-sequence-data.html#long-read-sequencing-the-next-step-in-genomics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="protein-sequencing" class="section level3 hasAnchor" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Protein sequencing<a href="what-is-sequence-data.html#protein-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
</div>
<div id="sequencing-errors-how-to-account-for-them" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Sequencing errors, how to account for them ?<a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="ngs-errors" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> NGS errors<a href="what-is-sequence-data.html#ngs-errors" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="long-read-errors" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Long read errors<a href="what-is-sequence-data.html#long-read-errors" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="hpc" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> HPC<a href="what-is-sequence-data.html#hpc" class="anchor-section" aria-label="Anchor link to header"></a></h3>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="general-introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="aligning-sequence-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/01-sequence-data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
