<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments.</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.24.11 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments." />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="lucblassel/phd-manuscript" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 What is Sequence data ? | From sequences to knowledge, improving and learning from sequence alignments." />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Luc Blassel" />


<meta name="date" content="2022-06-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="general-introduction.html"/>
<link rel="next" href="aligning-sequence-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/tabwid-1.0.0/tabwid.css" rel="stylesheet" />



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">My PhD thesis.</a></li>

<li class="divider"></li>
<li><a href="index.html#abstract">Abstract<span></span></a></li>
<li><a href="résumé.html#résumé">Résumé<span></span></a></li>
<li><a href="acknowledgments.html#acknowledgments">Acknowledgments<span></span></a></li>
<li><a href="glossary.html#glossary">Glossary<span></span></a></li>
<li><a href="general-introduction.html#general-introduction">General Introduction<span></span></a></li>
<li class="chapter" data-level="1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html"><i class="fa fa-check"></i><b>1</b> What is Sequence data ?<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-primer"><i class="fa fa-check"></i><b>1.1</b> Biological sequences, a primer<span></span></a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#what-is-dna"><i class="fa fa-check"></i><b>1.1.1</b> What is DNA ?<span></span></a></li>
<li class="chapter" data-level="1.1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#from-information-to-action"><i class="fa fa-check"></i><b>1.1.2</b> From Information to action<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#obtaining-sequence-data"><i class="fa fa-check"></i><b>1.2</b> Obtaining sequence data<span></span></a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough-wip"><i class="fa fa-check"></i><b>1.2.1</b> Sanger sequencing, a breakthrough (WIP)<span></span></a></li>
<li class="chapter" data-level="1.2.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#next-generation-sequencing"><i class="fa fa-check"></i><b>1.2.2</b> Next-generation sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them"><i class="fa fa-check"></i><b>1.3</b> Sequencing errors, how to account for them ?<span></span></a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#long-read-errors"><i class="fa fa-check"></i><b>1.3.1</b> Long read errors<span></span></a></li>
<li class="chapter" data-level="1.3.2" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#hpc"><i class="fa fa-check"></i><b>1.3.2</b> HPC<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="what-is-sequence-data.html"><a href="what-is-sequence-data.html#biological-sequences-a-computational-view-here"><i class="fa fa-check"></i><b>1.4</b> Biological sequences, a computational view (here ?)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html"><i class="fa fa-check"></i><b>2</b> Aligning sequence data<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#what-is-an-alignment"><i class="fa fa-check"></i><b>2.1</b> What is an alignment ?<span></span></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-to-align"><i class="fa fa-check"></i><b>2.1.1</b> How to align ?<span></span></a></li>
<li class="chapter" data-level="2.1.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#substitution-models"><i class="fa fa-check"></i><b>2.1.2</b> Substitution models<span></span></a></li>
<li class="chapter" data-level="2.1.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#why-align"><i class="fa fa-check"></i><b>2.1.3</b> Why align ?<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#how-do-we-speed-up-local-alignment"><i class="fa fa-check"></i><b>2.2</b> How do we speed up local alignment ?<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="aligning-sequence-data.html"><a href="aligning-sequence-data.html#msa-short"><i class="fa fa-check"></i><b>2.3</b> MSA (short)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><i class="fa fa-check"></i><b>3</b> Mapping-friendly sequence reductions: going beyond homopolymer compression<span></span></a>
<ul>
<li><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-abstract">Abstract<span></span></a></li>
<li class="chapter" data-level="3.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-methods"><i class="fa fa-check"></i><b>3.2</b> Methods<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:msr-def"><i class="fa fa-check"></i><b>3.2.1</b> Streaming sequence reductions<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:enum"><i class="fa fa-check"></i><b>3.2.2</b> Restricting the space of Streaming sequence reductions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#datasets-and-pipelines"><i class="fa fa-check"></i><b>3.3</b> Datasets and Pipelines<span></span></a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#datasets"><i class="fa fa-check"></i><b>3.3.1</b> Datasets<span></span></a></li>
<li class="chapter" data-level="3.3.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#simulation-pipeline"><i class="fa fa-check"></i><b>3.3.2</b> Simulation pipeline<span></span></a></li>
<li class="chapter" data-level="3.3.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#sec:evalpipeline"><i class="fa fa-check"></i><b>3.3.3</b> Evaluation pipeline<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-results"><i class="fa fa-check"></i><b>3.4</b> Results<span></span></a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#selection-of-mapping-friendly-sequence-reductions"><i class="fa fa-check"></i><b>3.4.1</b> Selection of mapping-friendly sequence reductions<span></span></a></li>
<li class="chapter" data-level="3.4.2" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#mapping-friendly-sequence-reductions-lead-to-lower-mapping-errors-on-whole-genomes"><i class="fa fa-check"></i><b>3.4.2</b> Mapping-friendly sequence reductions lead to lower mapping errors on whole genomes<span></span></a></li>
<li class="chapter" data-level="3.4.3" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#mapping-friendly-sequence-reductions-increase-mapping-quality-on-repeated-regions-of-the-human-genome"><i class="fa fa-check"></i><b>3.4.3</b> Mapping-friendly sequence reductions increase mapping quality on repeated regions of the human genome<span></span></a></li>
<li class="chapter" data-level="3.4.4" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#raw-mapping-improves-upon-hpc-on-centromeric-regions"><i class="fa fa-check"></i><b>3.4.4</b> Raw mapping improves upon HPC on centromeric regions<span></span></a></li>
<li class="chapter" data-level="3.4.5" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#positions-of-incorrectly-mapped-reads-across-the-entire-human-genome"><i class="fa fa-check"></i><b>3.4.5</b> Positions of incorrectly mapped reads across the entire human genome<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#hpc-discussion"><i class="fa fa-check"></i><b>3.5</b> Discussion<span></span></a></li>
<li class="chapter" data-level="3.6" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#limitations-of-this-study"><i class="fa fa-check"></i><b>3.6</b> Limitations of this study<span></span></a></li>
<li class="chapter" data-level="3.7" data-path="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html"><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#code-availability"><i class="fa fa-check"></i><b>3.7</b> Code availability<span></span></a></li>
<li><a href="mapping-friendly-sequence-reductions-going-beyond-homopolymer-compression.html#supplementary-information">Supplementary information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html"><i class="fa fa-check"></i><b>4</b> Learning from alignments<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#alignments-are-a-rich-source-of-information"><i class="fa fa-check"></i><b>4.1</b> Alignments are a rich source of information<span></span></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#pairwise-alns"><i class="fa fa-check"></i><b>4.1.1</b> Pairwise alns<span></span></a></li>
<li class="chapter" data-level="4.1.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#msa"><i class="fa fa-check"></i><b>4.1.2</b> MSA<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#preprocessing-the-alignment-for-machine-learning"><i class="fa fa-check"></i><b>4.2</b> Preprocessing the alignment for machine learning<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#embedding-the-alignment"><i class="fa fa-check"></i><b>4.2.1</b> Embedding the alignment<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#choosing-a-learning-target"><i class="fa fa-check"></i><b>4.2.2</b> Choosing a learning target<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#how-to-learn-from-alns"><i class="fa fa-check"></i><b>4.3</b> How to learn from ALNs<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#tests-and-statistical-learning"><i class="fa fa-check"></i><b>4.3.1</b> Tests and statistical learning<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#taking-interactions-into-account"><i class="fa fa-check"></i><b>4.3.2</b> Taking interactions into account<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="learning-from-alignments.html"><a href="learning-from-alignments.html#deep-learning"><i class="fa fa-check"></i><b>4.3.3</b> Deep Learning<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html"><i class="fa fa-check"></i><b>5</b> HIV and DRMs<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-are-viruses"><i class="fa fa-check"></i><b>5.1</b> What are viruses ?<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#what-is-hiv"><i class="fa fa-check"></i><b>5.2</b> What is HIV ?<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#quick-presentation-of-hiv"><i class="fa fa-check"></i><b>5.2.1</b> Quick Presentation of HIV<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#replication-cycle-of-hiv"><i class="fa fa-check"></i><b>5.2.2</b> Replication cycle of HIV<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#drug-resistance-in-hiv"><i class="fa fa-check"></i><b>5.3</b> Drug resistance in HIV<span></span></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#how-does-art-work"><i class="fa fa-check"></i><b>5.3.1</b> How does ART work<span></span></a></li>
<li class="chapter" data-level="5.3.2" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#different-types-of-resistance"><i class="fa fa-check"></i><b>5.3.2</b> different types of resistance<span></span></a></li>
<li class="chapter" data-level="5.3.3" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#consequences-on-global-health"><i class="fa fa-check"></i><b>5.3.3</b> Consequences on global health<span></span></a></li>
<li class="chapter" data-level="5.3.4" data-path="hiv-and-drms.html"><a href="hiv-and-drms.html#finding-drms"><i class="fa fa-check"></i><b>5.3.4</b> Finding DRMS<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="HIV-paper.html"><a href="HIV-paper.html"><i class="fa fa-check"></i><b>6</b> Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV<span></span></a>
<ul>
<li><a href="HIV-paper.html#abstract-paper">Abstract<span></span></a></li>
<li><a href="HIV-paper.html#author-summary">Author summary<span></span></a></li>
<li class="chapter" data-level="6.1" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-introduction"><i class="fa fa-check"></i><b>6.1</b> Introduction<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="HIV-paper.html"><a href="HIV-paper.html#materials-and-methods"><i class="fa fa-check"></i><b>6.2</b> Materials and methods<span></span></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="HIV-paper.html"><a href="HIV-paper.html#data"><i class="fa fa-check"></i><b>6.2.1</b> Data<span></span></a></li>
<li class="chapter" data-level="6.2.2" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-training"><i class="fa fa-check"></i><b>6.2.2</b> Classifier training<span></span></a></li>
<li class="chapter" data-level="6.2.3" data-path="HIV-paper.html"><a href="HIV-paper.html#measuring-classifier-performance"><i class="fa fa-check"></i><b>6.2.3</b> Measuring classifier performance<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="HIV-paper.html"><a href="HIV-paper.html#hiv-results"><i class="fa fa-check"></i><b>6.3</b> Results<span></span></a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="HIV-paper.html"><a href="HIV-paper.html#classifier-performance-interpretation"><i class="fa fa-check"></i><b>6.3.1</b> Classifier performance &amp; interpretation<span></span></a></li>
<li class="chapter" data-level="6.3.2" data-path="HIV-paper.html"><a href="HIV-paper.html#additional-classification-results"><i class="fa fa-check"></i><b>6.3.2</b> Additional classification results<span></span></a></li>
<li class="chapter" data-level="6.3.3" data-path="HIV-paper.html"><a href="HIV-paper.html#identifying-new-mutations-from-classifiers"><i class="fa fa-check"></i><b>6.3.3</b> Identifying new mutations from classifiers<span></span></a></li>
<li class="chapter" data-level="6.3.4" data-path="HIV-paper.html"><a href="HIV-paper.html#detailed-analysis-of-potentially-resistance-associated-mutations"><i class="fa fa-check"></i><b>6.3.4</b> Detailed analysis of potentially resistance-associated mutations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="HIV-paper.html"><a href="HIV-paper.html#discussion-and-perspectives"><i class="fa fa-check"></i><b>6.4</b> Discussion and perspectives<span></span></a></li>
<li><a href="HIV-paper.html#hiv-acknowledgments">Acknowledgments<span></span></a></li>
<li><a href="HIV-paper.html#supporting-information">Supporting Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html"><i class="fa fa-check"></i><b>7</b> Learning alignments, an interesting perspective<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learning-pairwise-alignment"><i class="fa fa-check"></i><b>7.1</b> Learning pairwise alignment<span></span></a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#dedal"><i class="fa fa-check"></i><b>7.1.1</b> DEDAL<span></span></a></li>
<li class="chapter" data-level="7.1.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#predicting-an-alignment"><i class="fa fa-check"></i><b>7.1.2</b> predicting an alignment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#what-else-could-we-learn"><i class="fa fa-check"></i><b>7.2</b> What else could we learn ?<span></span></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-to-predict-seeds-or-starting-positions"><i class="fa fa-check"></i><b>7.2.1</b> Learn to predict seeds or starting positions<span></span></a></li>
<li class="chapter" data-level="7.2.2" data-path="learning-alignments-an-interesting-perspective.html"><a href="learning-alignments-an-interesting-perspective.html#learn-pre-processing-functions"><i class="fa fa-check"></i><b>7.2.2</b> Learn pre-processing functions<span></span></a></li>
</ul></li>
</ul></li>
<li><a href="global-references.html#global-references">Global References<span></span></a></li>
<li class="appendix"><span><b>Appendix<span></span></b></span></li>
<li class="chapter" data-level="A" data-path="HPC-appendix.html"><a href="HPC-appendix.html"><i class="fa fa-check"></i><b>A</b> Supporting Information for “Mapping-friendly sequence reductions: going beyond homopolymer compression”<span></span></a>
<ul>
<li class="chapter" data-level="A.1" data-path="HPC-appendix.html"><a href="HPC-appendix.html#appendix:tandemtools"><i class="fa fa-check"></i><b>A.1</b> “TandemTools” dataset generation<span></span></a></li>
<li class="chapter" data-level="A.2" data-path="HPC-appendix.html"><a href="HPC-appendix.html#msr-performance-comparison"><i class="fa fa-check"></i><b>A.2</b> MSR performance comparison<span></span></a></li>
<li class="chapter" data-level="A.3" data-path="HPC-appendix.html"><a href="HPC-appendix.html#origin-of-incorrectly-mapped-reads-of-high-mapping-quality-on-whole-human-genome."><i class="fa fa-check"></i><b>A.3</b> Origin of incorrectly mapped reads of high mapping quality on whole human genome.<span></span></a></li>
<li class="chapter" data-level="A.4" data-path="HPC-appendix.html"><a href="HPC-appendix.html#analyzing-read-origin-on-whole-human-genome"><i class="fa fa-check"></i><b>A.4</b> Analyzing read origin on whole human genome<span></span></a></li>
<li class="chapter" data-level="A.5" data-path="HPC-appendix.html"><a href="HPC-appendix.html#performance-of-msrs-on-the-drosophila-genome"><i class="fa fa-check"></i><b>A.5</b> Performance of MSRs on the Drosophila genome<span></span></a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="HIV-appendix.html"><a href="HIV-appendix.html"><i class="fa fa-check"></i><b>B</b> Supporting Information for “Using Machine Learning and Big Data to Explore the Drug Resistance Landscape in HIV”<span></span></a>
<ul>
<li class="chapter" data-level="B.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Appendix"><i class="fa fa-check"></i><b>B.1</b> S1 Appendix (Technical appendix).<span></span></a>
<ul>
<li class="chapter" data-level="B.1.1" data-path="HIV-appendix.html"><a href="HIV-appendix.html#data-appendix"><i class="fa fa-check"></i><b>B.1.1</b> Data<span></span></a></li>
<li class="chapter" data-level="B.1.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#classifiers"><i class="fa fa-check"></i><b>B.1.2</b> Classifiers<span></span></a></li>
<li class="chapter" data-level="B.1.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#scoring"><i class="fa fa-check"></i><b>B.1.3</b> Scoring<span></span></a></li>
</ul></li>
<li class="chapter" data-level="B.2" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-fig."><i class="fa fa-check"></i><b>B.2</b> S1 Fig.<span></span></a></li>
<li class="chapter" data-level="B.3" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-fig."><i class="fa fa-check"></i><b>B.3</b> S2 Fig.<span></span></a></li>
<li class="chapter" data-level="B.4" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s3-fig."><i class="fa fa-check"></i><b>B.4</b> S3 Fig.<span></span></a></li>
<li class="chapter" data-level="B.5" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S1-Table"><i class="fa fa-check"></i><b>B.5</b> S1 Table.<span></span></a></li>
<li class="chapter" data-level="B.6" data-path="HIV-appendix.html"><a href="HIV-appendix.html#S2-Appendix"><i class="fa fa-check"></i><b>B.6</b> S2 Appendix. (Fisher exact tests)<span></span></a></li>
<li class="chapter" data-level="B.7" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s1-data."><i class="fa fa-check"></i><b>B.7</b> S1 Data.<span></span></a></li>
<li class="chapter" data-level="B.8" data-path="HIV-appendix.html"><a href="HIV-appendix.html#s2-data."><i class="fa fa-check"></i><b>B.8</b> S2 Data.<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://lucblassel.com" target="blank">Back to main website</a></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">From sequences to knowledge, improving and learning from sequence alignments.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="what-is-sequence-data" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> What is Sequence data ?<a href="what-is-sequence-data.html#what-is-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="biological-sequences-a-primer" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Biological sequences, a primer<a href="what-is-sequence-data.html#biological-sequences-a-primer" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To fully understand the work that was done during this thesis, as well as the choices that were made some basic knowledge of biology and more particularly genetics are needed. If you are already familiar with biological sequences, feel free to skip ahead to section <a href="what-is-sequence-data.html#obtaining-sequence-data">1.2</a>.</p>
<div id="what-is-dna" class="section level3 hasAnchor" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> What is DNA ?<a href="what-is-sequence-data.html#what-is-dna" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>D</strong>esoxyribo<strong>N</strong>ucleic <strong>A</strong>cid (DNA) is one of the most important molecules there is, without it complex life as we know it is impossible. It contains all the genetic information of a given organism, that is to say all the information necessary for the organism to: 1) function as a living being and 2) make a perfect copy of itself. This is the case for the overwhelming majority of living organisms on planet earth, from elephants to potatoes, to micro-organisms like bacteria.!g molecular biology tetbo</p>
<p>DNA is a polymer, composed of monomeric units called nucleotides. Each nucleotide is composed of Ribose (a five carbon sugar) on which are attached a phosphate group as well as one of four nucleobases: Adenine (A), Cytosine (C), Guanine (G) of Thymine (T). These 4 types of nucleotide monomers link up with one-another, through phosphate-sugar bonds, creating a single strand of DNA. The ordered sequence of these four types of nucleotides in strand encodes all the genetic information necessary for the organism to function. Nucleotides in a strand form strong complementary bonds with nucleotides from another strand, A with T and C with G. These bonds allows two strands of DNA to form the double-helix structure of DNA <span class="citation">[<a href="#ref-watson1953" role="doc-biblioref">1</a>]</span> shown in Figure <a href="what-is-sequence-data.html#fig:figDNA">1.1</a>. The specificity of nucleotide bonds ensure that the two strands of the double helix are complementary and that the information contained in one strand can be recovered from the other. This ensures a certain structural stability to the DNA molecule and a way to recover the important information that could be lost due to a damaged strand.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figDNA"></span>
<img src="figures/Basic-Intro/DNA.png" alt="**Double-helix** structure of DNA. Each strand of DNA has a phosphate-sugar backbone on which are attached nucleobases. The two strands are linked by complementary bonds between the nucleobases of different strands (A bonding with T and C bonding with G)." width="90%" />
<p class="caption">
Figure 1.1: <strong>Double-helix</strong> structure of DNA. Each strand of DNA has a phosphate-sugar backbone on which are attached nucleobases. The two strands are linked by complementary bonds between the nucleobases of different strands (A bonding with T and C bonding with G).
</p>
</div>
<p>The amount of DNA necessary to encode the information varies greatly from organism to organism: 5400 base pairs (5.4kBp) for the <span class="math inline">\(\varphi X174\)</span> phage <span class="citation">[<a href="#ref-sangerNucleotideSequenceBacteriophage1977" role="doc-biblioref">2</a>]</span>, 4.9MBp for <em>Escherichia coli</em> <span class="citation">[<a href="#ref-archer2011" role="doc-biblioref">3</a>]</span>, 3.1GBp for <em>Homo sapiens</em> <span class="citation">[<a href="#ref-nurk2022" role="doc-biblioref">4</a>]</span> all the way up to almost 150GBp for <em>Paris japonica,</em> a Japanese mountain flowering plant <span class="citation">[<a href="#ref-pellicer2010" role="doc-biblioref">5</a>]</span>. While very small genome size tend to occur in smaller, simpler organisms genome size does not correlate with organism complexity <span class="citation">[<a href="#ref-macgregor2001" role="doc-biblioref">6</a>]</span></p>
</div>
<div id="from-information-to-action" class="section level3 hasAnchor" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> From Information to action<a href="what-is-sequence-data.html#from-information-to-action" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="proteins-their-structure-and-their-role" class="section level4 hasAnchor" number="1.1.2.1">
<h4><span class="header-section-number">1.1.2.1</span> Proteins, their structure and their role<a href="what-is-sequence-data.html#proteins-their-structure-and-their-role" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The double stranded DNA molecules present in the cells of a living organism contains only information; in order for the organism to live, this information must be read and transformed into actions. Most of the actions necessary for “life” are taken by large molecules called proteins, they have a very wide range of functions from catalyzing reactions in the cell to giving it it’s structure <span class="citation">[<a href="#ref-alberts2002" role="doc-biblioref">7</a>]</span>.</p>
<p>Proteins are macromolecules, that are made up of one or several chains of amino acids. These chains then link together and fold up in a specific 3 dimensional structure, giving the protein the shape it needs to fulfill it’s goal. This structure is determined by the sequence of amino acids, and a given protein can be identified by this amino acid sequence <span class="citation">[<a href="#ref-alberts2002" role="doc-biblioref">7</a>]</span>.</p>
<p>This sequence is directly dependent on the information contained in the DNA. First the DNA is transcripted in a similar, but single stranded, molecule called RNA which encodes the same sequence. This RNA molecule is then translated into a protein by the following process <span class="citation">[<a href="#ref-crick1961" role="doc-biblioref">8</a>]</span>:</p>
<ol style="list-style-type: decimal">
<li>Nucleotides in the RNA sequence are read in groups of 3 called a codon.</li>
<li>These codons are read sequentially along the RNA molecule</li>
<li>Each codon corresponds to an amino acid, according to the genetic code.</li>
<li>The sequence of codons in RNA <em>(and by extension DNA)</em> determines the sequence of amino acids.</li>
<li>The translation process is stopped when a specific type of codon is read.</li>
</ol>
<p>With 4 types of nucleotides and codons grouping 3 nucleotides there are <span class="math inline">\(4^3=64\)</span> possible codons. However, as stated above, proteins are only made up of 20 different amino acids, meaning that several different codons correspond to the same amino acid. This gives the translation process a certain robustness to errors that can occur when the DNA is copied to create a new cell, or when it is transformed into RNA prior to protein translation.</p>
</div>
<div id="making-mistakes" class="section level4 hasAnchor" number="1.1.2.2">
<h4><span class="header-section-number">1.1.2.2</span> Making mistakes<a href="what-is-sequence-data.html#making-mistakes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Going from DNA sequence to protein is quite a complicated process involving several steps, it is therefore possible for a mistake to happen. There are several mechanisms to avoid mistakes and alteration of the genetic information: the complementary nature of the 2 strands of DNA, the redundant nature of the genetic code as well as error correction mechanisms in the molecules that read and write DNA and RNA <em>(polymerases)</em>. However, despite all that, some errors still make it through.</p>
<div id="where-can-mistakes-happen" class="section level5 hasAnchor" number="1.1.2.2.1">
<h5><span class="header-section-number">1.1.2.2.1</span> Where can mistakes happen ?<a href="what-is-sequence-data.html#where-can-mistakes-happen" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>There are several sources of error that can alter the genetic information <span class="citation">[<a href="#ref-chatterjee2017" role="doc-biblioref">9</a>]</span>:</p>
<ul>
<li><p><strong>DNA replication:</strong> When a cell divides, or when an organism reproduces, the DNA molecule must be copied in order to transmit genetic information. This process has a very low rate of errors, with as low as 1 error for every billion to every hundred billions of base pairs replicated <span class="citation">[<a href="#ref-fijalkowska2012" role="doc-biblioref">10</a>]</span>. This is due to the fact that the DNA polymerase (the protein that is responsible for copying DNA molecules), has a relatively error rate to start with, but mostly due to the error correcting mechanisms that are present in certain cells and bacteria <span class="citation">[<a href="#ref-pray2008dna" role="doc-biblioref">11</a>]</span>.</p></li>
<li><p><strong>RNA transcription:</strong> error rate between 4 errors for each million <span class="citation">[<a href="#ref-gout2013" role="doc-biblioref">12</a>]</span> to 2 errors for each hundred thousand <span class="citation">[<a href="#ref-gout2017" role="doc-biblioref">13</a>]</span> base pairs transcribed.</p></li>
<li><p><strong>Other mutagenic events:</strong> Ionizing radiation <span class="citation">[<a href="#ref-desouky2015" role="doc-biblioref">14</a>]</span>, UV rays <span class="citation">[<a href="#ref-kiefer2007" role="doc-biblioref">15</a>]</span>, Toxins <span class="citation">[<a href="#ref-bennett2003" role="doc-biblioref">16</a>]</span>, heat Stress <span class="citation">[<a href="#ref-kantidze2016" role="doc-biblioref">17</a>]</span>, cold stress <span class="citation">[<a href="#ref-gregory1994" role="doc-biblioref">18</a>]</span>, oxidative stress <span class="citation">[<a href="#ref-gafter-gvili2013" role="doc-biblioref">19</a>]</span>.</p></li>
</ul>
</div>
<div id="what-kind-of-errors-are-possible" class="section level5 hasAnchor" number="1.1.2.2.2">
<h5><span class="header-section-number">1.1.2.2.2</span> What kind of errors are possible?<a href="what-is-sequence-data.html#what-kind-of-errors-are-possible" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li>substitution</li>
<li>insertion</li>
<li>deletion</li>
</ul>
</div>
<div id="what-effect-can-mutations-have" class="section level5 hasAnchor" number="1.1.2.2.3">
<h5><span class="header-section-number">1.1.2.2.3</span> What effect can mutations have ?<a href="what-is-sequence-data.html#what-effect-can-mutations-have" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p>illnesses (SNPs, groups of mutations,…)</p></li>
<li><p>acquired traits (drug resistance, …)</p></li>
</ul>
</div>
</div>
</div>
</div>
<div id="obtaining-sequence-data" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Obtaining sequence data<a href="what-is-sequence-data.html#obtaining-sequence-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In order to study living organisms we need to be able to obtain their genetic information, i.e figure out a way to get the sequence of nucleobases that make up their DNA.</p>
<div id="sanger-sequencing-a-breakthrough-wip" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Sanger sequencing, a breakthrough (WIP)<a href="what-is-sequence-data.html#sanger-sequencing-a-breakthrough-wip" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The first true sequencing method was developed in 1977 <span class="citation">[<a href="#ref-sangerDNASequencingChainterminating1977" role="doc-biblioref">20</a>]</span>. Sanger <em>et al.</em> devised a simple method to read the sequence of nucleotides that make up a DNA sequence.</p>
<ol style="list-style-type: decimal">
<li>Clone sequence / amplify</li>
<li>Prepare 4 different sequencing environments with a majority of dNTP (ie regular nucleotides) and in each a single type of ddNTP (a terminator). ddNTP are marked</li>
<li>In each test tube add DNA polymerase, primers and denatures DNA fragments you want to sequence</li>
<li>Sequence is replicated until incorporation of ddNTP stopping reaction</li>
<li>Separate replicated fragments by electrophoresis (i.e shorter fragments go further), 1 ddNTP type in each lane</li>
<li>With marked you can see which nucleotide is present at a given position</li>
</ol>
<p>This allowed Sanger <em>et al.</em> to sequence the first whole <span class="math inline">\(\varphi X174\)</span> bacteriophage genome <span class="citation">[<a href="#ref-sangerNucleotideSequenceBacteriophage1977" role="doc-biblioref">2</a>]</span>. This method, although revolutionnary was costly and time consuming.</p>
<p>The marking of primers and ddNTP with fluorescence allowed to do the polymerization in a single test tube and use a single lane for electrophoresis <span class="citation">[<a href="#ref-smithSynthesisOligonucleotidesContaining1985" role="doc-biblioref">21</a>,<a href="#ref-smithFluorescenceDetectionAutomated1986" role="doc-biblioref">22</a>]</span>. The fluorescence also allowed for automated reading <strong><em>(base-calling ?)</em></strong> with optical systems.</p>
<p><strong><em>I need to speak of performance / throughput of these methods here.</em></strong></p>
</div>
<div id="next-generation-sequencing" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Next-generation sequencing<a href="what-is-sequence-data.html#next-generation-sequencing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Developed to lower cost and more throughput</p>
<ul>
<li><p>Massively parallel, Illumina</p></li>
<li><p>Long reads sequencing</p></li>
</ul>
<p><strong><em>Quick summary on PacBio and ONT</em></strong></p>
<p>As a conclusion, quickly mention the work done on protein sequencing, but we usually get the protein sequence from the DNA sequence that is translated from codons.</p>
</div>
</div>
<div id="sequencing-errors-how-to-account-for-them" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Sequencing errors, how to account for them ?<a href="what-is-sequence-data.html#sequencing-errors-how-to-account-for-them" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>general context of sequencing errors</p>
<div id="long-read-errors" class="section level3 hasAnchor" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Long read errors<a href="what-is-sequence-data.html#long-read-errors" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Mainly indels, in certain regions of the genomes, particularily homopolymers. For ONT <span class="citation">[<a href="#ref-delahaye2021" role="doc-biblioref">23</a>,<a href="#ref-goodwin2015" role="doc-biblioref">24</a>]</span>, for PacBIO and ONT <span class="citation">[<a href="#ref-dohmBenchmarkingLongreadCorrection2020" role="doc-biblioref">25</a>,<a href="#ref-weirather2017" role="doc-biblioref">26</a>]</span>, for PacBIO circular consensus <span class="citation">[<a href="#ref-wenger2019" role="doc-biblioref">27</a>]</span>. ONT has more homopolymer errors than PacBIO <span class="citation">[<a href="#ref-logsdonLongreadHumanGenome2020" role="doc-biblioref">28</a>]</span>.</p>
</div>
<div id="hpc" class="section level3 hasAnchor" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> HPC<a href="what-is-sequence-data.html#hpc" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>HPC takes repeated runs of a single nucleotide and compresses them to a single occurence</p></li>
<li><p>Empirically is has been shown to improve mapping and other applications</p></li>
</ul>
</div>
</div>
<div id="biological-sequences-a-computational-view-here" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Biological sequences, a computational view (here ?)<a href="what-is-sequence-data.html#biological-sequences-a-computational-view-here" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Essentially a text file. Just a long succession of letters, so we can apply text algorithmics to it.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body">
<div id="ref-watson1953" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Watson JD, Crick FHC. The Structure of Dna. Cold Spring Harbor Symposia on Quantitative Biology. 1953;18: 123–131. doi:<a href="https://doi.org/10.1101/SQB.1953.018.01.020">10.1101/SQB.1953.018.01.020</a></div>
</div>
<div id="ref-sangerNucleotideSequenceBacteriophage1977" class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Sanger F, Air GM, Barrell BG, Brown NL, Coulson AR, Fiddes JC, et al. Nucleotide sequence of bacteriophage <span class="nocase">φX174 DNA</span>. Nature. 1977;265: 687–695. doi:<a href="https://doi.org/10.1038/265687a0">10.1038/265687a0</a></div>
</div>
<div id="ref-archer2011" class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Archer CT, Kim JF, Jeong H, Park JH, Vickers CE, Lee SY, et al. The genome sequence of e. Coli w (ATCC 9637): Comparative genome analysis and an improved genome-scale reconstruction of e. coli. BMC Genomics. 2011;12: 9. doi:<a href="https://doi.org/10.1186/1471-2164-12-9">10.1186/1471-2164-12-9</a></div>
</div>
<div id="ref-nurk2022" class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Nurk S, Koren S, Rhie A, Rautiainen M, Bzikadze AV, Mikheenko A, et al. The complete sequence of a human genome. Science. 2022;376: 44–53. doi:<a href="https://doi.org/10.1126/science.abj6987">10.1126/science.abj6987</a></div>
</div>
<div id="ref-pellicer2010" class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Pellicer J, Fay MF, Leitch IJ. The largest eukaryotic genome of them all? Botanical Journal of the Linnean Society. 2010;164: 10–15. doi:<a href="https://doi.org/10.1111/j.1095-8339.2010.01072.x">10.1111/j.1095-8339.2010.01072.x</a></div>
</div>
<div id="ref-macgregor2001" class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Macgregor HC. C-Value Paradox. In: Brenner S, Miller JH, editors. New York: Academic Press; 2001. pp. 249–250. doi:<a href="https://doi.org/10.1006/rwgn.2001.0301">10.1006/rwgn.2001.0301</a></div>
</div>
<div id="ref-alberts2002" class="csl-entry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Alberts B, Johnson A, Lewis J, Raff M, Roberts K, Walter P. Molecular Biology of the Cell. 4th edition. Garland Science; 2002. Available: <a href="https://www.ncbi.nlm.nih.gov/books/NBK26916/">https://www.ncbi.nlm.nih.gov/books/NBK26916/</a></div>
</div>
<div id="ref-crick1961" class="csl-entry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Crick FHC, Barnett L, Brenner S, Watts-Tobin RJ. General Nature of the Genetic Code for Proteins. Nature. 1961;192: 1227–1232. doi:<a href="https://doi.org/10.1038/1921227a0">10.1038/1921227a0</a></div>
</div>
<div id="ref-chatterjee2017" class="csl-entry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Chatterjee N, Walker GC. Mechanisms of DNA damage, repair, and mutagenesis: DNA Damage and Repair. Environmental and Molecular Mutagenesis. 2017;58: 235–263. doi:<a href="https://doi.org/10.1002/em.22087">10.1002/em.22087</a></div>
</div>
<div id="ref-fijalkowska2012" class="csl-entry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Fijalkowska IJ, Schaaper RM, Jonczyk P. DNA replication fidelity in escherichia coli: A multi-DNA polymerase affair. FEMS microbiology reviews. 2012;36: 1105–1121. doi:<a href="https://doi.org/10.1111/j.1574-6976.2012.00338.x">10.1111/j.1574-6976.2012.00338.x</a></div>
</div>
<div id="ref-pray2008dna" class="csl-entry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">Pray L. DNA replication and causes of mutation. Nature education. 2008;1: 214. </div>
</div>
<div id="ref-gout2013" class="csl-entry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Gout J-F, Thomas WK, Smith Z, Okamoto K, Lynch M. Large-scale detection of in vivo transcription errors. Proceedings of the National Academy of Sciences. 2013;110: 18584–18589. doi:<a href="https://doi.org/10.1073/pnas.1309843110">10.1073/pnas.1309843110</a></div>
</div>
<div id="ref-gout2017" class="csl-entry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">Gout J-F, Li W, Fritsch C, Li A, Haroon S, Singh L, et al. The landscape of transcription errors in eukaryotic cells. Science Advances. 2017;3: e1701484. doi:<a href="https://doi.org/10.1126/sciadv.1701484">10.1126/sciadv.1701484</a></div>
</div>
<div id="ref-desouky2015" class="csl-entry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Desouky O, Ding N, Zhou G. Targeted and non-targeted effects of ionizing radiation. Journal of Radiation Research and Applied Sciences. 2015;8: 247–254. doi:<a href="https://doi.org/10.1016/j.jrras.2015.03.003">10.1016/j.jrras.2015.03.003</a></div>
</div>
<div id="ref-kiefer2007" class="csl-entry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Kiefer J. Effects of Ultraviolet Radiation on DNA. In: Obe G, Vijayalaxmi, editors. Berlin, Heidelberg: Springer; 2007. pp. 39–53. doi:<a href="https://doi.org/10.1007/978-3-540-71414-9_3">10.1007/978-3-540-71414-9_3</a></div>
</div>
<div id="ref-bennett2003" class="csl-entry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Bennett JW, Klich M. Mycotoxins. Clinical Microbiology Reviews. 2003;16: 497–516. doi:<a href="https://doi.org/10.1128/CMR.16.3.497-516.2003">10.1128/CMR.16.3.497-516.2003</a></div>
</div>
<div id="ref-kantidze2016" class="csl-entry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Kantidze OL, Velichko AK, Luzhin AV, Razin SV. Heat stress-induced DNA damage. Acta Naturae. 2016;8: 75–78. Available: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4947990/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4947990/</a></div>
</div>
<div id="ref-gregory1994" class="csl-entry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Gregory CD, Milner AE. Regulation of cell survival in Burkitt lymphoma: implications from studies of apoptosis following cold-shock treatment. International Journal of Cancer. 1994;57: 419–426. doi:<a href="https://doi.org/10.1002/ijc.2910570321">10.1002/ijc.2910570321</a></div>
</div>
<div id="ref-gafter-gvili2013" class="csl-entry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Gafter-Gvili A, Zingerman B, Rozen-Zvi B, Ori Y, Green H, Lubin I, et al. Oxidative stress-induced DNA damage and repair in human peripheral blood mononuclear cells: Protective role of hemoglobin. PLoS ONE. 2013;8: e68341. doi:<a href="https://doi.org/10.1371/journal.pone.0068341">10.1371/journal.pone.0068341</a></div>
</div>
<div id="ref-sangerDNASequencingChainterminating1977" class="csl-entry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Sanger F, Nicklen S, Coulson AR. <span>DNA</span> sequencing with chain-terminating inhibitors. Proceedings of the National Academy of Sciences. 1977;74: 5463–5467. doi:<a href="https://doi.org/10.1073/pnas.74.12.5463">10.1073/pnas.74.12.5463</a></div>
</div>
<div id="ref-smithSynthesisOligonucleotidesContaining1985" class="csl-entry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Smith LM, Fung S, Hunkapiller MW, Hunkapiller TJ, Hood LE. The synthesis of oligonucleotides containing an aliphatic amino group at the 5′ terminus: Synthesis of fluorescent <span>DNA</span> primers for use in <span>DNA</span> sequence analysis. Nucleic Acids Research. 1985;13: 2399–2412. doi:<a href="https://doi.org/10.1093/nar/13.7.2399">10.1093/nar/13.7.2399</a></div>
</div>
<div id="ref-smithFluorescenceDetectionAutomated1986" class="csl-entry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Smith LM, Sanders JZ, Kaiser RJ, Hughes P, Dodd C, Connell CR, et al. Fluorescence detection in automated <span>DNA</span> sequence analysis. Nature. 1986;321: 674–679. doi:<a href="https://doi.org/10.1038/321674a0">10.1038/321674a0</a></div>
</div>
<div id="ref-delahaye2021" class="csl-entry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Delahaye C, Nicolas J. Sequencing DNA with nanopores: Troubles and biases. PLOS ONE. 2021;16: e0257521. doi:<a href="https://doi.org/10.1371/journal.pone.0257521">10.1371/journal.pone.0257521</a></div>
</div>
<div id="ref-goodwin2015" class="csl-entry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Goodwin S, Gurtowski J, Ethe-Sayers S, Deshpande P, Schatz MC, McCombie WR. Oxford Nanopore sequencing, hybrid error correction, and de novo assembly of a eukaryotic genome. Genome Research. 2015;25: 1750–1756. doi:<a href="https://doi.org/10.1101/gr.191395.115">10.1101/gr.191395.115</a></div>
</div>
<div id="ref-dohmBenchmarkingLongreadCorrection2020" class="csl-entry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Dohm JC, Peters P, Stralis-Pavese N, Himmelbauer H. Benchmarking of long-read correction methods. NAR Genomics and Bioinformatics. 2020;2. doi:<a href="https://doi.org/10.1093/nargab/lqaa037">10.1093/nargab/lqaa037</a></div>
</div>
<div id="ref-weirather2017" class="csl-entry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Weirather JL, Cesare M de, Wang Y, Piazza P, Sebastiano V, Wang X-J, et al. Comprehensive comparison of pacific biosciences and oxford nanopore technologies and their applications to transcriptome analysis. F1000Research. 2017;6: 100. doi:<a href="https://doi.org/10.12688/f1000research.10571.2">10.12688/f1000research.10571.2</a></div>
</div>
<div id="ref-wenger2019" class="csl-entry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Wenger AM, Peluso P, Rowell WJ, Chang P-C, Hall RJ, Concepcion GT, et al. Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome. Nature Biotechnology. 2019;37: 1155–1162. doi:<a href="https://doi.org/10.1038/s41587-019-0217-9">10.1038/s41587-019-0217-9</a></div>
</div>
<div id="ref-logsdonLongreadHumanGenome2020" class="csl-entry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Logsdon GA, Vollger MR, Eichler EE. Long-read human genome sequencing and its applications. Nat Rev Genet. 2020;21: 597–614. doi:<a href="https://doi.org/10.1038/s41576-020-0236-x">10.1038/s41576-020-0236-x</a></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="general-introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="aligning-sequence-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/lucblassel/phd-manuscript/edit/main/01-sequence-data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
